---
title: "16S Thermal Tolerance"
html_document:
code_folding: hide
toc:yes
toc_depth:3
toc_float:yes
output: 
chunk_output_type: console
editor_options: 
  chunk_output_type: console
---
Last updated: 8 apr 2021


16s data analysis
#Data and libraries
```{r} 
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(tidyverse)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(microbiome)
```


#Load in data and get into phyloseq
```{r}   

#load in 3000 rare instead
#read in sample data
MetaData16<-read.csv("Data/Meta16s_3k.csv") #run in physio script first
MetaData16$Parent.ID<-as.factor(as.character(MetaData16$Parent.ID))
MetaData16$Tank.num<-as.factor(as.character(MetaData16$Tank.num))

sam0 <- MetaData16
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))

#load in OUT:raw_abundanceTable_100.shared

OTU3k<-read.table("Data/TT17_3000_summer_16S-pipeline_outputs/abundance_table_100_shared.csv", sep=',', header=T)

#from its  
#otu
otu3k1 <- as.matrix(OTU3k[, -1])
rownames(otu3k1) <- OTU3k$sample_name
otu <- otu_table(otu3k1, taxa_are_rows = FALSE)


#tax table annotations_100.taxonomy.csv
TAX<- read.csv("Data/TT17_3000_summer_16S-pipeline_outputs/annotations_100.taxonomy.csv", colClasses = "character")
tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- TAX$OTU
tax <- tax_table(tax1)


# Read the data into phyloseq
Bac.seq = phyloseq(otu, tax,sam) #THIS WORKS
Bac.seq
Bac.seq.df <- sample_data(Bac.seq)

save(Bac.seq, file = "Data/Bac.seq_phyloseq.RData")








########################################################
# ## THIS IS FOR INCLUDING ALL SAMPLES YOU NEED FOR phylotype analysis.
# #read in sample data
# MetaData16<-read.csv("Data/Meta_16s_new.csv") #run in physio script first
# MetaData16$Parent.ID<-as.factor(as.character(MetaData16$Parent.ID))
# MetaData16$Tank.num<-as.factor(as.character(MetaData16$Tank.num))
# 
# sam0 <- MetaData16
# sam1 <- as.matrix(sam0[, -1])
# rownames(sam1) <- sam0$sample_name
# sam <- sample_data(data.frame(sam1))
# 
# 
# #load in OUT:raw_abundanceTable_100.shared
# OTU2<-read.table("Data/TT17_3000_summer_16S-pipeline_outputs/raw_abundanceTable_100_shared.csv", sep=',', header=T)
# 
# #from its3
# #otu
# otu5 <- as.matrix(OTU2[, -1])
# rownames(otu5) <- OTU2$sample_name
# otu <- otu_table(otu5, taxa_are_rows = FALSE)
# 
# 
# #tax
# TAX<- read.csv("Data/TT17_3000_summer_16S-pipeline_outputs/raw_consensusClassification_100_taxonomy.csv", colClasses = "character")
# tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
# rownames(tax1) <- TAX$OTU
# tax <- tax_table(tax1)
# 
# # Read the data into phyloseq
# Bac.seqAll = phyloseq(otu, tax,sam) #THIS WORKS
# Bac.seqAll
# 
# save(Bac.seqAll, file = "Data/Bac.seqAll_phyloseq.RData")

```
Look at the data, Bac.seq for 3k
```{r}  
## check out data
ntaxa(Bac.seq)  #num taxa
nsamples(Bac.seq)   #num samples
sample_names(Bac.seq)[1:300] #samp names
 #sampsy<-sample_names(physeq)[1:300]
#write.csv(sampsy,"sampsy.csv")

# OK Now you need to do the steps in the pipeline. ONLY for >3K dfs
#1 filter out taxa: get rid of unknown, mitochondria and chloroplast
# Bac.seq<-subset_taxa(Bac.seq, Family !="Mitochondria")
# Bac.seq<-subset_taxa(Bac.seq, Order !="Chloroplast")
# Bac.seq<-subset_taxa(Bac.seq, Domain !="unknown")
# Bac.seq<-subset_taxa(Bac.seq, Domain !="Archaea")

# now get rid of things not represnted 2x or more
#Prune singletons (these are reads that are only found once)
# Bac.seq.prune <- prune_taxa(taxa_sums(Bac.seq) > 2, Bac.seq)
# 
#   save(Bac.seq, file = "Data/Bac.seq.prune.RData")
  
```

#16S
Load in your data
```{r} 
# Load data (see data_exploration.Rmd)
#load("Data/Bac.seq_phyloseq.RData") #use RAW data that has been pruned. 
# the df is Bac.seq
rank_names(Bac.seq) #ranks are DIV and clade


#remove field samples
Bac.s = subset_samples(Bac.seq, Type== "sample" )

# #Bac.seq  not 3k use to check who you lose
# # create df of sample data to view 
# sample.data <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view
# sample.data$LibrarySize <- sample_sums(Bac.seq)
# #sample.data <- sample.data[order(sample.data$LibrarySize),]
# #sample.data$Index <- seq(nrow(sample.data))  # check 3k rare
# ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = Species)) +
#   geom_point()
# write.csv(sample.data, "sampscheck.csv")

## check out data
ntaxa(Bac.s)  #num taxa
nsamples(Bac.s)   #num samples
sample_names(Bac.s)[1:5] #samp names
rank_names(Bac.s) #ranks are DIV and clade
sample_variables(Bac.s) # metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(Bac.s), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.s)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  # check 3k rare
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = Species)) +
  geom_point()



#remove duplicates 
Dups<-subset_samples(Bac.s, Dups=="Yes")

TopNOTUsB = names(sort(taxa_sums(Dups), TRUE)[1:100])
bac50 = prune_taxa(TopNOTUsB, Dups)
bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot
#ggsave("mMcapBarPlot_DIVrelARare.pdf", McapBarPlot, dpi=300, width=8)
Bac.s <- subset_samples(Bac.s, sample_name.1 != "S1753b" )
Bac.s <- subset_samples(Bac.s, sample_name.1 != "S208" )
Bac.s <- subset_samples(Bac.s, sample_name.1 != "S3476" )
Bac.s <- subset_samples(Bac.s, sample_name.1 != "SN35b" )
Bac.s <- subset_samples(Bac.s, sample_name.1 != "SN111b" )

Bac.s <- subset_samples(Bac.s, sample_name.1 != "Field" )


# Subsample to 2K <- only for phylotypes
# Bac.s2 <- prune_samples(sample_sums(Bac.s)>=1764, Bac.s) #keep things over 2k
#Rarefied to 2000 reads
# Bac.r <- rarefy_even_depth(Bac.s2, sample.size = 1764, rngseed = 711) 


#### Create a duplicate of T0s for the other A and H treatments for comparision: df coralDIV2_rare
#not
controlT0_Amb = subset_samples(Bac.s, Time.Point == "T0"&Treatment == "Ambient")  #copy

#tax table
tax  = tax_table(controlT0_Amb)

  # sample data
  sd = sample_data(controlT0_Amb)
  rownames(sd) <- paste0(rownames(sd), "_dup")
  # otu_table
  OTU = otu_table(controlT0_Amb)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  # sd$Treatment <- as.character(sd$Treatment)
  # sd$ID <- as.character(sd$ID)
  sd$Treatment <- "High"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
  sd$Treatment <- as.factor(sd$Treatment)
  # sd$ID <- as.factor(sd$ID)

  Amb2<-phyloseq(OTU, sd, tax)
#sample.dataAmb2 <- as(sample_data(Amb2), "data.frame") #create sample data frame to view

##High
  controlT0_High = subset_samples(Bac.s, Time.Point == "T0"&Treatment == "High")  #copy

#tax table
tax  = tax_table(controlT0_High)

  # sample data
  sd = sample_data(controlT0_High)
  rownames(sd) <- paste0(rownames(sd), "_dup")
# otu_table
OTU = otu_table(controlT0_High)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  sd$Treatment <- as.character(sd$Treatment)
  # sd$ID <- as.character(sd$ID)
  sd$Treatment <- "Ambient"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
  sd$Treatment <- as.factor(sd$Treatment)
  sd$ID <- as.factor(sd$ID)

  High2<-phyloseq(OTU, sd, tax)
  #sample.dataHigh2 <- as(sample_data(High2), "data.frame") #create sample data frame to view

#Merge control dups H and A  
Controldups = merge_phyloseq(Amb2,High2)
Controldups
    #sample.dataControldups2 <- as(sample_data(Controldups), "data.frame") #create sample data frame to view

#merg them to the original df:

Bac.s2 = merge_phyloseq(Bac.s,Controldups)
Bac.s2
Bac.s2.sd <- as(sample_data(Bac.s2), "data.frame") #create sample data frame to view
Bac.s2.sd
#write.csv(Bac.s2.sd,"16sMetadata.csv")
```


Set up, Bac.s2
```{r} 
#make relative abundance df Bac.s2
Bac.s.rel  = transform_sample_counts(Bac.s2, function(x) x / sum(x) )  #save as RELA DIVs
data.Bac.s.rel <- as(sample_data(Bac.s.rel), "data.frame") #look at samp data

#heatmap
  #plot_heatmap(Bac.RelA, method = "NMDS", distance = "bray")# way too much data



# # export into table 
# taxCN  = tax_table(Bac.s2)
# taxCN = as(tax_table(Bac.s2), "matrix")
# taxCN2 = as.data.frame(taxCN)
# write.csv(taxCN2, "taxCN.csv")
# 
# taxCN2$OTU <- rownames(taxCN2) #make row names a col
# taxCN3<-taxCN2[,c("OTU","Family")] # keep only columns you need
# rownames(taxCN3) <- c() #remove row namds
# 
# 
# OTUCN = otu_table(Bac.s2)
# # Extract abundance matrix from the phyloseq object
# OTU.cn = as(otu_table(OTUCN), "matrix")
# if(taxa_are_rows(Bac.s2)){OTU.cn <- t(OTU.cn)}
# OTU.cn2 = as.data.frame(OTU.cn)
# write.csv(OTU.cn2, "OTU.cn2.csv")
# 
# #OK can we fucking merge
# CNtab<-merge(OTU.cn2,taxCN3)


``` 
For plotting only, subset the data:
- Families with >0.5% in one or more samples, AND nonzero in 3 or more samples. somehow relax the latter criteria if they are greater than 1% in any one sample. https://bioconductor.riken.jp/packages/3.0/bioc/vignettes/phyloseq/inst/doc/phyloseq-basics.html
```{r}
library(genefilter)
#df use counts not Rel A
#tax glom should let you glom at Family
# phyFam<-tax_glom(Bac.s, taxrank="Family")  #this isn't doing it by family. wtf

otu_table(phyFam)[1:5, 1:5]
tax_table(phyFam)[1:5, 1:4]
taxFAM  = tax_table(phyFam)
taxFAM<-taxFAM[!(taxFAM$Size)]
# write.csv(taxFAM,"taxFAM.csv")


# goal: only keep families with >0.5% in one ore more samples and nonzero in 3 or more samples, unless >1% in any sample

#this works for fam level and relA. now we have 543 families
phyFamTest2 <- Bac.s %>% 
  tax_glom(taxrank = "Family", NArm=FALSE) %>% #glom to family and remove anything not classified at family level
  transform_sample_counts(function(x) {x/sum(x)} ) #then turn to REl A
    sum(is.na(otu_table(phyFamTest2))) #check if there are any NA
        otu_table(phyFamTest2)[1:5, 1:5]
        tax_table(phyFamTest2)[1:6, 1:6]


#      #filter taxa not >.5% rela bund per sample, removes 25 taxa <- not exactly sure what this is doing tbh
# phyFamTest2.f = filter_taxa(phyFamTest2, function(x) mean(x) < .005,TRUE) 
    
 
test_function<-function(x){x>0} #keep only taxa with abundance at least 2 in at least 10 samples
#keep only taxa with abundance at least 1 in at least 3 samples
taxa.to.keep<-genefilter_sample(phyFamTest2, test_function, A=3) #trying with skipping the above step
head(taxa.to.keep)

phyFamTest2.fb<-prune_taxa(taxa.to.keep,phyFamTest2 )  #ok now that you did the above, PRUNE them out
    #now we are donw to 334 taxa. This is without the filter step. now there are no empty rows (samps)
    
taxFAM  = tax_table(phyFamTest2.fb) #look at it
tax_table(phyFamTest2.fb)[1:7, 1:6]


# can we change mean to max? wtf did that do?
phyFamTest2TEST = filter_taxa(phyFamTest2, function(x) max(x) > 0.005, TRUE)
tax_table(phyFamTest2TEST)[1:7, 1:6]
otu_table(phyFamTest2TEST)[1:5, 1:5]
#ok now you are down to 257 taxa. this is the 2nd heat
    OTUfam = as(otu_table(phyFamTest2TEST), "matrix")
    # transpose if necessary
    if(taxa_are_rows(phyFamTest2TEST)){OTUfam <- t(OTUfam)}
    # Coerce to data.frame
    OTUfam2.df = as.data.frame(OTUfam)
    write.csv(OTUfam2.df,"phyFamTest2TEST.csv")



# Extract abundance matrix from the phyloseq object< to look at OTU table, why isn't this working below!?
#OK some of your samples now have zero taxa. 
OTUfam = as(otu_table(phyFamTest2.fb), "matrix")
# transpose if necessary
if(taxa_are_rows(phyFamTest2.fb)){OTUfam <- t(OTUfam)}
# Coerce to data.frame
OTUfam.df = as.data.frame(OTUfam)
write.csv(OTUfam.df,"OTUfam.df.csv")



 #ok so in theory you got rid of stuff sort of how CN wanted. 
#helpful https://www.gdc-docs.ethz.ch/MDA/handouts/MDA20_PhyloseqFormation_Mahendra_Mariadassou.pdf


#try heatmap again
theme_set(theme_bw())

#ok....
  plot_heatmap(phyFamTest2TEST, "NMDS", "bray", "Species","Family", sample.order = "Species") #use to show split

#try by speces: 
#Mcap
  Mcap_phyFam<-subset_samples(phyFamTest2TEST, Species=="Montipora_capitata")
  plot_heatmap(Mcap_phyFam, "NMDS", "bray", "Parent.ID","Family") #use to show split

  
  
  Mcap_phyFam.T1<-subset_samples(Mcap_phyFam, Time.Point=="T1")
  plot_heatmap(Mcap_phyFam.T1, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment")
    plot_heatmap(Mcap_phyFam.T1, "NMDS", "bray", "Clade","Family")

  Mcap_phyFam.TF<-subset_samples(Mcap_phyFam, Time.Point=="TF")
  plot_heatmap(Mcap_phyFam.TF, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment") 
  plot_heatmap(Mcap_phyFam.TF, "NMDS", "bray", "Clade","Family") 
  
#Pcomp
  Pcomp_phyFam<-subset_samples(phyFamTest2TEST, Species=="Porites_compressa")
  plot_heatmap(Pcomp_phyFam, "NMDS", "bray", "Parent.ID","Family") #use to show split
  
  Pcomp_phyFam.T1<-subset_samples(Pcomp_phyFam, Time.Point=="T1")
  plot_heatmap(Pcomp_phyFam.T1, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment")
    plot_heatmap(Pcomp_phyFam.T1, "NMDS", "bray", "Clade","Family")

  Pcomp_phyFam.TF<-subset_samples(Pcomp_phyFam, Time.Point=="TF")
  plot_heatmap(Pcomp_phyFam.TF, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment") 
  plot_heatmap(Pcomp_phyFam.TF, "NMDS", "bray", "Clade","Family") 

#Pvar
  Pvar_phyFam<-subset_samples(phyFamTest2TEST, Species=="Pavona_varians")
  plot_heatmap(Pvar_phyFam, "NMDS", "bray", "Parent.ID","Family") #use to show split
  
  Pvar_phyFam.T1<-subset_samples(Pvar_phyFam, Time.Point=="T1")
  plot_heatmap(Pvar_phyFam.T1, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment")
    plot_heatmap(Pvar_phyFam.T1, "NMDS", "bray", "Clade","Family")

  Pvar_phyFam.TF<-subset_samples(Pvar_phyFam, Time.Point=="TF")
  plot_heatmap(Pvar_phyFam.TF, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment") 
  plot_heatmap(Pvar_phyFam.TF, "NMDS", "bray", "Clade","Family") 
  
  
#Pvar
  Pacu_phyFam<-subset_samples(phyFamTest2TEST, Species=="Pocillopora_acuta")
  plot_heatmap(Pacu_phyFam, "NMDS", "bray", "Parent.ID","Family") #use to show split
  
  Pacu_phyFam.T1<-subset_samples(Pacu_phyFam, Time.Point=="T1")
  plot_heatmap(Pacu_phyFam.T1, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment")
    plot_heatmap(Pacu_phyFam.T1, "NMDS", "bray", "Clade","Family")

  Pacu_phyFam.TF<-subset_samples(Pacu_phyFam, Time.Point=="TF")
  plot_heatmap(Pacu_phyFam.TF, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment") 
  plot_heatmap(Pacu_phyFam.TF, "NMDS", "bray", "Clade","Family")
  
  
#lets do some PERMANOVAs at Family level
  
  
```
Crap you were working on for above
```{r}
#everything below here is shit that doesn't work

# phyFamTest2.f2 = filter_taxa(phyFamTest2.f, function(x) sum(x > 3) > (0.013*length(x)), TRUE)

# 
# #This function will remove taxa (OTUs) with low prevalence, where prevalence is the fraction of total samples in which an OTU is observed.
# phyloseq_filter_prevalence(GlobalPatterns, prev.trh = 0.05, abund.trh = NULL)  # 15389 taxa
# 
# 
# 
# 
# library(metagMisc)
# #https://rdrr.io/github/vmikk/metagMisc/man/phyloseq_filter_prevalence.html
# #then non zero in 3 or more samples unless >1% in a sample
# phyFamTest2.f3<-phyloseq_filter_prevalence(phyFamTest2.f, prev.trh = 0.05, abund.trh = .01, threshold_condition = "OR")  # 15639 taxa
# 
# 
# tax_tab <- phyloseq::tax_table(phyFam.REL)
# tax_tab[1:7,1:7] # for my table show me [1 to 5 otu ids, 1 to 5 first five ranks]
# 
# 
# 
# 
# 
# #remove taxa not seen more than 3 times in at least 3 samples (do do this by %, 3/230 is 1.3%)
# phyFam.rel.f = filter_taxa(phyFam, function(x) sum(x > 3) > (0.013*length(x)), TRUE)
# sum(is.na(otu_table(phyFam))) #check if there are any NA
# 
# 
# #aparently you have to use count not rel a
# #first set the genefilter, then send to prune_taxa
# filter <- phyloseq::genefilter_sample(phyFam.REL, filterfun_sample(function(x) x >= 3), 
#                                        A = 0.013*nsamples(phyFam.REL))
# ps_filtered <- prune_taxa(filter, phyFam.REL)
# 
# 
# f <- filter_taxa(phyFam.REL, function(x) var(x) > 1e-05, TRUE) #this worked. this is for variance wtf
# 
# #global test
# otu_table(GlobalPatterns)[1:5, 1:5]
# tax_table(GlobalPatterns)[1:5, 1:4]
# 
# 
# require("genefilter")
#  flist    <- filterfun(kOverA(3, 2e-05))
#  ent.logi <- filter_taxa(phyFamTest2, flist)
#  ent.trim <- filter_taxa(phyFamTest2, flist, TRUE)
#  identical(ent.trim, prune_taxa(ent.logi, phyFamTest2)) 
#  identical(sum(ent.logi), ntaxa(ent.trim))
#  filter_taxa(phyFamTest2, flist, TRUE)   
# 
# 
# 
# 
# 
# 
# data("enterotype")
# library("genefilter")
# flist<- filterfun(kOverA(5, 2e-05))
# ent.logi <- filter_taxa(enterotype, flist)
# ent.trim <- filter_taxa(enterotype, flist, TRUE)
# identical(ent.trim, prune_taxa(ent.logi, enterotype)) 

```

16 bar plots in ggplot (to make faster than next section)  
#MCAP
PHYLUM
```{r} 
#try glom use raw counts Bac.s2
Mcap_Bac.s2<-subset_samples(Bac.s2, Species=="Montipora_capitata")

Mcap_Bac.s2.phy <- Mcap_Bac.s2 %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Mcap_Bac.s2.melt <- psmelt(Mcap_Bac.s2.phy)

# sum.ra.mcap <- ddply(Mcap_Bac.s2.melt, c("Phylum", "Time.Point", "Treatment","Parent.ID"), summarise,
#                 N = length(Abundance),
#                 mean = mean(Abundance),
#                 sd = sd(Abundance), 
#                 se = sd/sqrt(N)
# )
# sum.ra.mcap

Mcap_Bac.s2.melt$Code<-paste(Mcap_Bac.s2.melt$Treatment, Mcap_Bac.s2.melt$Time.Point)
Mcap_Bac.s2.melt$Parent.ID <- factor(Mcap_Bac.s2.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Mcap_Bac.s2.melt$Code <- factor(Mcap_Bac.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

mcap.p <- ggplot(Mcap_Bac.s2.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)


################################## 
#by C or CD
# C ONLY

Mcap_Bac.s2.C<-subset_samples(Mcap_Bac.s2.melt, Clade=="C")

mcap.p <- ggplot(Mcap_Bac.s2.C, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap C")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Mcap_Bac.s2.CD<-subset_samples(Mcap_Bac.s2.melt, Clade=="CD")

mcap.p <- ggplot(Mcap_Bac.s2.CD.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

```

Family
```{r}  
library(tidy)

Mcap_Bac.s2.melt <- psmelt(Mcap_phyFam)

#set 
Mcap_Bac.s2.melt$Code<-paste(Mcap_Bac.s2.melt$Treatment, Mcap_Bac.s2.melt$Time.Point)
Mcap_Bac.s2.melt$Parent.ID <- factor(Mcap_Bac.s2.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Mcap_Bac.s2.melt$Code <- factor(Mcap_Bac.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
#FAMILY COLORS
nb.cols <- 252
mycolors <- colorRampPalette(brewer.pal(11, "BrBG"))(nb.cols)

mcap.p <- ggplot(Mcap_Bac.s2.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Mcap_Bac.s2.melt.C<-subset(Mcap_Bac.s2.melt, Clade=="C")

mcap.p <- ggplot(Mcap_Bac.s2.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Mcap_Bac.s2.melt.CD<-subset(Mcap_Bac.s2.melt, Clade=="CD")


mcap.p <- ggplot(Mcap_Bac.s2.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Mcap CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)



```
#Pcomp
PHYLUM
```{r} 
#try glom use raw counts Bac.s2
Pcomp_Bac.s2<-subset_samples(Bac.s2, Species=="Porites_compressa")

Pcomp_Bac.s2.phy <- Pcomp_Bac.s2 %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pcomp_Bac.s2.melt <- psmelt(Pcomp_Bac.s2.phy)

Pcomp_Bac.s2.melt$Code<-paste(Pcomp_Bac.s2.melt$Treatment, Pcomp_Bac.s2.melt$Time.Point)
Pcomp_Bac.s2.melt$Parent.ID <- factor(Pcomp_Bac.s2.melt$Parent.ID, levels = c("313",  "314","315","316", "317","318","319",  "320", "322","323",  "324","321"))

Pcomp_Bac.s2.melt$Code <- factor(Pcomp_Bac.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pcomp.p <- ggplot(Pcomp_Bac.s2.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pcomp")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)



### by C15cj or C15n
# C15cj ONLY

Pcomp_Bac.s2.C15cj<-subset(Pcomp_Bac.s2.melt, Clade=="C15cj")

Pcomp.p <- ggplot(Pcomp_Bac.s2.C15cj, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C15cj")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

# C15n ONLY
Pcomp_Bac.s2.C15n<-subset(Pcomp_Bac.s2.melt, Clade=="C15n")

Pcomp.p <- ggplot(Pcomp_Bac.s2.C15n, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C15n")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

```
Family
```{r}  

Pcomp_Bac.s2.melt <- psmelt(Pcomp_phyFam)

#set 
Pcomp_Bac.s2.melt$Code<-paste(Pcomp_Bac.s2.melt$Treatment, Pcomp_Bac.s2.melt$Time.Point)
Pcomp_Bac.s2.melt$Parent.ID <- factor(Pcomp_Bac.s2.melt$Parent.ID, levels = c("313",  "314","315","316", "317","318","319",  "320", "322","323",  "324","321"))
Pcomp_Bac.s2.melt$Code <- factor(Pcomp_Bac.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
#FAMILY COLORS
nb.cols <- 252
mycolors <- colorRampPalette(brewer.pal(11, "BrBG"))(nb.cols)

Pcomp.p <- ggplot(Pcomp_Bac.s2.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Pcomp_Bac.s2.melt.C<-subset(Pcomp_Bac.s2.melt, Clade=="C")

Pcomp.p <- ggplot(Pcomp_Bac.s2.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Pcomp_Bac.s2.melt.CD<-subset(Pcomp_Bac.s2.melt, Clade=="CD")


Pcomp.p <- ggplot(Pcomp_Bac.s2.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Pcomp CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)



```
#Pvar
PHYLUM
```{r}
#try glom use raw counts Bac.s2
Pvar_Bac.s2<-subset_samples(Bac.s2, Species=="Pavona_varians")

Pvar_Bac.s2.phy <- Pvar_Bac.s2 %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pvar_Bac.s2.melt <- psmelt(Pvar_Bac.s2.phy)

Pvar_Bac.s2.melt$Code<-paste(Pvar_Bac.s2.melt$Treatment, Pvar_Bac.s2.melt$Time.Point)
Pvar_Bac.s2.melt$Parent.ID <- factor(Pvar_Bac.s2.melt$Parent.ID, levels = c("361",  "358","353","356", "325","354",  "359", "357","362", "351", "355","360"))

Pvar_Bac.s2.melt$Code <- factor(Pvar_Bac.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pvar.p <- ggplot(Pvar_Bac.s2.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pvar")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)



### by C1 or C127
# C15cj ONLY

Pvar_Bac.s2.C15cj<-subset(Pvar_Bac.s2.melt, Clade=="C1")

Pvar.p <- ggplot(Pvar_Bac.s2.C15cj, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pvar C1")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)

# C15n ONLY
Pvar_Bac.s2.C27<-subset(Pvar_Bac.s2.melt, Clade=="C27")

Pvar.p <- ggplot(Pvar_Bac.s2.C27, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pvar C27")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)

```
Family
```{r}  

Pvar_Bac.s2.melt <- psmelt(Pvar_phyFam)

#set 
Pvar_Bac.s2.melt$Code<-paste(Pvar_Bac.s2.melt$Treatment, Pvar_Bac.s2.melt$Time.Point)
Pvar_Bac.s2.melt$Parent.ID <- factor(Pvar_Bac.s2.melt$Parent.ID, levels = c("361",  "358","353","356", "325","354",  "359", "357","362", "351", "355","360"))

Pvar_Bac.s2.melt$Code <- factor(Pvar_Bac.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
#FAMILY COLORS
nb.cols <- 252
mycolors <- colorRampPalette(brewer.pal(11, "BrBG"))(nb.cols)

Pvar.p <- ggplot(Pvar_Bac.s2.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Pvar_Bac.s2.melt.C<-subset(Pvar_Bac.s2.melt, Clade=="C")

Pvar.p <- ggplot(Pvar_Bac.s2.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pvar C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Pvar_Bac.s2.melt.CD<-subset(Pvar_Bac.s2.melt, Clade=="CD")


Pvar.p <- ggplot(Pvar_Bac.s2.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Pvar CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)



```


#Pacu
PHYLUM
```{r}
#try glom use raw counts Bac.s2
Pacu_Bac.s2<-subset_samples(Bac.s2, Species=="Pocillopora_acuta")

Pacu_Bac.s2.phy <- Pacu_Bac.s2 %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pacu_Bac.s2.melt <- psmelt(Pacu_Bac.s2.phy)

Pacu_Bac.s2.melt$Code<-paste(Pacu_Bac.s2.melt$Treatment, Pacu_Bac.s2.melt$Time.Point)
Pacu_Bac.s2.melt$Parent.ID <- factor(Pacu_Bac.s2.melt$Parent.ID, levels = c("304",  "308","312","303", "306","302",  "310", "301","311", "307", "309","305"))

Pacu_Bac.s2.melt$Code <- factor(Pacu_Bac.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pacu.p <- ggplot(Pacu_Bac.s2.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pacu")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)



### by CCC or C42
# CCC ONLY

Pacu_Bac.s2.CCC<-subset(Pacu_Bac.s2.melt, Clade=="CCC")

Pacu.p <- ggplot(Pacu_Bac.s2.CCC, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pacu CCC")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)

# C42 ONLY
Pacu_Bac.s2.C42<-subset(Pacu_Bac.s2.melt, Clade=="C42")

Pacu.p <- ggplot(Pacu_Bac.s2.C42, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pacu C42")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)

```
Family
```{r}  

Pacu_Bac.s2.melt <- psmelt(Pacu_phyFam)

#set 
Pacu_Bac.s2.melt$Code<-paste(Pacu_Bac.s2.melt$Treatment, Pacu_Bac.s2.melt$Time.Point)
PPacu_Bac.s2.melt$Parent.ID <- factor(Pacu_Bac.s2.melt$Parent.ID, levels = c("304",  "308","312","303", "306","302",  "310", "301","311", "307", "309","305"))
Pacu_Bac.s2.melt$Code <- factor(Pacu_Bac.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
#FAMILY COLORS
nb.cols <- 252
mycolors <- colorRampPalette(brewer.pal(11, "BrBG"))(nb.cols)

Pacu.p <- ggplot(Pacu_Bac.s2.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Pacu_Bac.s2.melt.C<-subset(Pacu_Bac.s2.melt, Clade=="C")

Pacu.p <- ggplot(Pacu_Bac.s2.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pacu C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Pacu_Bac.s2.melt.CD<-subset(Pacu_Bac.s2.melt, Clade=="CD")


Pacu.p <- ggplot(Pacu_Bac.s2.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Pacu CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)



```




#16s - bar plots df Bac.s2RelA YOU Don't need now
```{r}    
#ordination time - CUT WHEN YOU HAVE OTHER 4 SPECIES ORDINATION SORTED 
Bac.s.rel.ord <- ordinate(Bac.s.rel, "NMDS", "bray", set.seed(30))
Bac.RelA.ord_plot = plot_ordination(Bac.s.rel, Bac.s.rel.ord, type="samples", shape="Treatment",color="Species", title="species")
print(Bac.RelA.ord_plot)
Bac.RelA.ord_plot + facet_wrap(~Species, 3)

### let's do this again and run the NMDS by species now that we know they are different. 
#subset by species
Mcap_16s_RA <- subset_samples(Bac.s.rel, Species=="Montipora_capitata")
Pcomp_16s_RA <- subset_samples(Bac.s.rel, Species=="Porites_compressa")
Pvar_16s_RA <- subset_samples(Bac.s.rel, Species=="Pavona_varians")
Pacu_16s_RA <- subset_samples(Bac.s.rel, Species=="Pocillopora_acuta")

#top 50 DIVs across by species 

#Mcap
 TopNOTUsMcap = names(sort(taxa_sums(Mcap_16s_RA), TRUE)[1:300])
 Mcapent50 = prune_taxa(TopNOTUsMcap, Mcap_16s_RA)
McapBarPlot<-plot_bar(Mcap_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot
#ggsave("mMcapBarPlot_16srelARare.pdf", McapBarPlot, dpi=300, width=8)


#### try with gg
Mcap_16s_RA.gg<-phyloseq_to_df(Mcap_16s_RA, addtax = T, addtot = F, addmaxrank = F,
  sorting = "abundance")
Mcap_16s_RA.gg.Phylum<-subset(Mcap_16s_RA.gg, select=-c(2:3,5:8))
#convert data frame from a "wide" format to a "long" format
pcm = melt(Mcap_16s_RA.gg, id = c("OTU","Size","Phylum"))

  mx = ggplot(Mcap_16s_RA.gg, aes(x = Parent.ID, fill = Phylum, y = value)) + 
    geom_bar(stat = "identity", colour = "black") + 
    theme(axis.text.x = element_text(angle = 90, size = 14, colour = "black", vjust = 0.5, hjust = 1, face= "bold"), 
    axis.title.y = element_text(size = 16, face = "bold"), legend.title = element_text(size = 16, face = "bold"), 
    legend.text = element_text(size = 12, face = "bold", colour = "black"), 
    axis.text.y = element_text(colour = "black", size = 12, face = "bold")) + 
    scale_y_continuous(expand = c(0,0)) + 
    labs(x = "", y = "Relative Abundance (%)", fill = "OTU") + 
    scale_fill_manual(values = colours)
    
  mx




#Mcap C and CD
#C
Mcap_16s_RA.C<-subset_samples(Mcap_16s_RA, Clade=="C")
TopNOTUsMcap.C = names(sort(taxa_sums(Mcap_16s_RA.C), TRUE)[1:300])
Mcap.50.C = prune_taxa(TopNOTUsMcap.C, Mcap_16s_RA.C)
McapBarPlot.C<-plot_bar(Mcap_16s_RA.C, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot.C
#ggsave("McapBarPlot_16s_C.pdf", McapBarPlot.C, dpi=300, width=8)

#CD
Mcap_16s_RA.CD<-subset_samples(Mcap_16s_RA, Clade=="CD")
TopNOTUsMcap.CD = names(sort(taxa_sums(Mcap_16s_RA.CD), TRUE)[1:300])
 Mcap.50.CD = prune_taxa(TopNOTUsMcap.CD, Mcap_16s_RA.CD)
McapBarPlot.CD<-plot_bar(Mcap_16s_RA.CD, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot.CD
#ggsave("McapBarPlot_16s_CD.pdf", McapBarPlot.CD, dpi=300, width=8)


#Pcomp
TopNOTUsPcomp = names(sort(taxa_sums(Pcomp_16s_RA), TRUE)[1:300])
# Pcompent50 = prune_taxa(TopNOTUsPcomp,Pcomp_16s_RA )
PcompBarPlot<-plot_bar(Pcomp_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot
#ggsave("PcompBarPlot_16srelARare.pdf", PcompBarPlot, dpi=300, width=8)

#C15cj and C15n
  #C15cj
  Pcomp_16s_RA.C15cj<-subset_samples(Pcomp_16s_RA, Clade=="C15cj")
  TopNOTUsPcomp.C15cj = names(sort(taxa_sums(Pcomp_16s_RA.C15cj), TRUE)[1:300])
  Pcomp.50.C15cj = prune_taxa(TopNOTUsPcomp.C15cj, Pcomp_16s_RA.C15cj)
  PcompBarPlot.C15cj<-plot_bar(Pcomp_16s_RA.C15cj, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot.C15cj
  #ggsave("PcompBarPlot_16s_C.pdf", PcompBarPlot.C15cj, dpi=300, width=8)

  #C15n
  Pcomp_16s_RA.C15n<-subset_samples(Pcomp_16s_RA, Clade=="C15n")
  TopNOTUsPcomp.C15n = names(sort(taxa_sums(Pcomp_16s_RA.C15n), TRUE)[1:300])
   Pcomp.50.C15n = prune_taxa(TopNOTUsPcomp.C15n, Pcomp_16s_RA.C15n)
  PcompBarPlot.C15n<-plot_bar(Pcomp_16s_RA.C15n, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot.C15n
#ggsave("PcompBarPlot_16s_CD.pdf", PcompBarPlot.C15n, dpi=300, width=8)


#Pvar
TopNOTUsVar = names(sort(taxa_sums(Pvar_16s_RA), TRUE)[1:300])
Pvarent50 = prune_taxa(TopNOTUsVar, Pvar_16s_RA)
PvarBarPlot<-plot_bar(Pvar_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot
#ggsave("PvarBarPlot_16srelARare.pdf", PvarBarPlot, dpi=300, width=8)


#C1 and C27
  #C15cj
  Pvar_16s_RA.C1<-subset_samples(Pvar_16s_RA, Clade=="C1")
  TopNOTUsPvar.C1 = names(sort(taxa_sums(Pvar_16s_RA.C1), TRUE)[1:300])
  Pvar.50.C1 = prune_taxa(TopNOTUsPvar.C1, Pvar_16s_RA.C1)
  PvarBarPlot.C1<-plot_bar(Pvar_16s_RA.C1, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot.C1
  #ggsave("PvarBarPlot_16s_C.pdf", PvarBarPlot.C15cj, dpi=300, width=8)

  #C27
  Pvar_16s_RA.C27<-subset_samples(Pvar_16s_RA, Clade=="C27")
  TopNOTUsPvar.C27 = names(sort(taxa_sums(Pvar_16s_RA.C27), TRUE)[1:300])
   Pvar.50.C27 = prune_taxa(TopNOTUsPvar.C27, Pvar_16s_RA.C27)
  PvarBarPlot.C27<-plot_bar(Pvar_16s_RA.C27, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot.C27
  #ggsave("PvarBarPlot_16s_CD.pdf", PvarBarPlot.C27, dpi=300, width=8)



#Pacu
# TopNOTUsPacu = names(sort(taxa_sums(Pacu_16s_RA), TRUE)[1:300])
Pacuent50 = prune_taxa(TopNOTUsPacu, Pacu_16s_RA)
PacuBarPlot<-plot_bar(Pacu_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot
#ggsave("PacuBarPlot_16srelARare.pdf", PacuBarPlot, dpi=300, width=8)

#CCC and C42
  #CCC
  Pacu_16s_RA.CCC<-subset_samples(Pacu_16s_RA, Clade=="CCC")
  TopNOTUsPacu.CCC = names(sort(taxa_sums(Pacu_16s_RA.CCC), TRUE)[1:300])
  Pacu.50.CCC = prune_taxa(TopNOTUsPacu.CCC, Pacu_16s_RA.CCC)
  PacuBarPlot.CCC<-plot_bar(Pacu_16s_RA.CCC, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot.CCC
  #ggsave("PacuBarPlot_16s_C.pdf", PacuBarPlotCCC5cj, dpi=300, width=8)

  #C42
  Pacu_16s_RA.C42<-subset_samples(Pacu_16s_RA, Clade=="C42")
  TopNOTUsPacu.C42 = names(sort(taxa_sums(Pacu_16s_RA.C42), TRUE)[1:300])
   Pacu.50.C42 = prune_taxa(TopNOTUsPacu.C42, Pacu_16s_RA.C42)
  PacuBarPlot.C42<-plot_bar(Pacu_16s_RA.C42, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot.C42
  #ggsave("PacuBarPlot_16s_CD.pdf", PacuBarPlot.C42, dpi=300, width=8)
```


#Ordination DIV_RelA <- you might not need this section, code below for each species  
try vegan nmds - its the same, delete when CN says cool, Bac.s.rel
```{r}  
  ##this isn't working, not converging. 
# convert the sample_data() within a phyloseq object to a vegan compatible data object
df16s2veg <- function(Bac.s.rel) { 
  sd <- sample_data(Bac.s.rel)
  return(as(sd,"data.frame"))
}

# convert the otu_table() within a phyloseq object to a vegan compatible data object
df16sotu2veg <- function(Bac.s.rel) {
  OTU <- otu_table(Bac.s.rel)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

# Calculate distance matrix
df16s.spp_distmat <- vegdist(OTU, method = "bray")
# Creating easy to view matrix and writing .csv
df16s.spp_distmat <- 
  as.matrix(df16s.spp_distmat, labels = T)
#write.csv(island.spp_distmat, "island_spp_distmat.csv")

# Running NMDS in vegan (metaMDS)
df16s.spp_NMS <-
  metaMDS(df16s.spp_distmat,
          distance = "bray",
          k =2,
          maxit = 999, 
          trymax = 500,
          wascores = TRUE, set.seed(10))

# Shepards test/goodness of fit
goodness(df16s.spp_NMS) # Produces a results of test statistics for goodness of fit for each point

stressplot(df16s.spp_NMS) # Produces a Shepards diagram

# Plotting points in ordination space
plot(df16s.spp_NMS, "sites")   # Produces distance 
orditorp(df16s.spp_NMS, "sites")   # Gives points labels




```
Heatmap and first crack at ordination, delete when solid
```{r}           
# CUT?? OrdiSpecies<-plot_ordination(DIV_RelA, ordinate(DIV_RelA, "NMDS"), color = "Species", label="ID") + geom_point(size = 5) 
# #ggsave("OrdiSpecies.pdf", OrdiSpecies, dpi=300, width=8)

  
#heat map
plot_heatmap(Bac.s.rel, "NMDS", "bray", "Species", "Phylum")

#by species 
  #mcap
Mcapent50.T1<- subset_samples(Mcapent50, Time.Point=="T1")
Mcapent50.T1.C<- subset_samples(Mcapent50.T1, Clade=="C")
Mcapent50.T1.CD<- subset_samples(Mcapent50.T1, Clade=="CD")

Mcapent50.TF<- subset_samples(Mcapent50, Time.Point=="TF")


  plot_heatmap(Mcapent50.T1, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split
  plot_heatmap(Mcapent50.T1.C, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split
  plot_heatmap(Mcapent50.T1.CD, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split

 #Pcomp
  Pcomp_DIV_RA_T1 <- subset_samples(Pcomp_DIV_RA, Time.Point=="T1")
  plot_heatmap(Pcomp_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
  #Pvar
  Pvar_DIV_RA_T1 <- subset_samples(Pvar_DIV_RA, Time.Point=="T1")
  plot_heatmap(Pvar_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
  plot_heatmap(Pvar_DIV_RA, "NMDS", "bray", "Parent.ID", "DIV")

  
  
  
  
#ordination time - CUT WHEN YOU HAVE OTHER 4 SPECIES ORDINATION SORTED 
DIVrareRel.ord <- ordinate(DIV_RelA, "NMDS", "bray", set.seed(30))
DIVrareRel1 = plot_ordination(DIV_RelA, DIVrareRel.ord, type="samples", shape="Treatment",color="Time.Point", title="species")
print(DIVrareRel1)
DIVrareRel1 + facet_wrap(~Species, 3)

### let's do this again and run the NMDS by species now that we know they are different. 
# mcap
Mcap_DIV_RA.ord <- ordinate(Mcap_DIV_RA, "NMDS", "bray", set.seed(30))
DIVrareRel1_Mcap = plot_ordination(Mcap_DIV_RA, Mcap_DIV_RA.ord, type="samples", shape="Treatment",color="Time.Point", title="Mcap NMDS")
print(DIVrareRel1_Mcap) 
  #add ellipses
  DIVrareRel1_Mcap + 
  stat_ellipse(type = "norm", linetype = 2, alpha=.2) +
  stat_ellipse(type = "t") +
  theme_bw()

# pcomp
Pcomp_DIV_RA.ord <- ordinate(Pcomp_DIV_RA, "NMDS", "bray", set.seed(30))
DIVrareRel1_Pcomp = plot_ordination(Pcomp_DIV_RA, Pcomp_DIV_RA.ord, type="samples", shape="Treatment",color="Time.Point", title="Pcomp NMDS")
print(DIVrareRel1_Pcomp)
#add ellipses
  DIVrareRel1_Pcomp + 
  stat_ellipse(type = "norm",  alpha=.2) +
  theme_bw()
```

#Beta Diversity: df rarefied and relative abudnaces  
All Species Bac.s.rel .  relative abundances 
```{r}         
#Bac.s.rel 

Bac.s.rel_stats<- Bac.s.rel #make a copy
otu_table(Bac.s.rel_stats)[1:5, 1:5] #check rela worked

#get rid of NA OTUs now!
Bac.s.rel_stats = prune_taxa(taxa_sums(Bac.s.rel_stats) > 0, Bac.s.rel_stats) #this removes any OTU with 0s


#bray curtis presence-absense and abundance
## REMOVE T0 for models
Bac_RelA_stats.T1F<-subset_samples(Bac.s.rel_stats, Time.Point=="T1"|Time.Point=="TF")

#bray curtis presence-absense and abundance

#all speices
set.seed(30)

#make bray curtis data matrix
bc.all <- phyloseq::distance(Bac_RelA_stats.T1F, method = "bray")
  samp.df <- as(sample_data(Bac_RelA_stats.T1F), "data.frame") #sample df
#adonis test
adonis(bc.all ~ Species*Treatment*Time.Point, data = samp.df)
    adonis(bc.all ~ Species*Treatment*Time.Point*Clade, data = samp.df)

adonis(bc.all ~ Species*Treatment*Time.Point, data = samp.df, contr.unordered = "contr.treatement")

    
    
    
# Homogeneity of dispersion test
betaAll <- betadisper(bc.all, samp.df$Species)
permutest(betaAll, pairwise = T) 
(bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
plot(bc.all.HSD)


#Ordination NMDS for all species, T1Tf
set.seed(50)
ord.Bac_RelA_stats <- ordinate(Bac_RelA_stats.T1F, "NMDS", "bray", trymax = 500) 
stressplot(ord.Bac_RelA_stats)
scores.ord.Bac_RelA_stats<-as.data.frame(cbind(vegan::scores(ord.Bac_RelA_stats, display="sites"))) # these are "points" ask hannah 

scores.ord.Bac_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.Bac_RelA_stats$Species <- samp.df$Species


ggplot(scores.ord.Bac_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")


```
All species by family df phy,FamTest2TEST relA
```{r}         
#phyFamTest2TEST 

phyFamTest2TEST_stats<- phyFamTest2TEST #make a copy
otu_table(phyFamTest2TEST_stats)[1:5, 1:5] #check rela worked

#get rid of NA OTUs now!
phyFamTest2TEST_stats = prune_taxa(taxa_sums(phyFamTest2TEST_stats) > 0, phyFamTest2TEST_stats) #this removes any OTU with 0s


#bray curtis presence-absense and abundance
## REMOVE T0 for models
Bac_RelA_stats.T1F<-subset_samples(phyFamTest2TEST_stats, Time.Point=="T1"|Time.Point=="TF")

#bray curtis presence-absense and abundance

#all speices
set.seed(30)

#make bray curtis data matrix
bc.all <- phyloseq::distance(Bac_RelA_stats.T1F, method = "bray")
  samp.df <- as(sample_data(Bac_RelA_stats.T1F), "data.frame") #sample df
#adonis test
adonis(bc.all ~ Species*Treatment*Time.Point, data = samp.df)
    adonis(bc.all ~ Species*Treatment*Time.Point*Clade, data = samp.df)
    
  
# Homogeneity of dispersion test
betaAll <- betadisper(bc.all, samp.df$Species)
permutest(betaAll, pairwise = T) 
(bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
plot(bc.all.HSD)


#Ordination NMDS for all species, T1Tf
set.seed(50)
ord.Bac_RelA_stats <- ordinate(Bac_RelA_stats.T1F, "NMDS", "bray", trymax = 500) 
stressplot(ord.Bac_RelA_stats)
scores.ord.Bac_RelA_stats<-as.data.frame(cbind(vegan::scores(ord.Bac_RelA_stats, display="sites"))) # these are "points" ask hannah 

scores.ord.Bac_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.Bac_RelA_stats$Species <- samp.df$Species
scores.ord.Bac_RelA_stats$Code <- paste(samp.df$Species,samp.df$Treatment, samp.df$Time.Point)


ggplot(scores.ord.Bac_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Species), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")
 

```
# Beta Split data by species 
Beta div
Montipora - T1 and TF together, separet, and then by phylotype. PERMANOVA
```{r}     
# df Bac_RelA_stats.T1F
#Mcap
mcap.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Montipora_capitata")
mcap.bac.T1F = prune_taxa(taxa_sums(mcap.bac.T1F) > 0, mcap.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
mcap.bac.T1 <- subset_samples(mcap.bac.T1F, Time.Point == "T1")
  mcap.bac.T1 = prune_taxa(taxa_sums(mcap.bac.T1) > 0, mcap.bac.T1) #this removes any OTU with 0s

mcap.bac.TF <- subset_samples(mcap.bac.T1F, Time.Point == "TF")
  mcap.bac.TF = prune_taxa(taxa_sums(mcap.bac.TF) > 0, mcap.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.mcap.T1F <- phyloseq::distance(mcap.bac.T1F, method = "bray")
     mcap.samp.df.T1F <- as(sample_data(mcap.bac.T1F), "data.frame") #sample df
    adonis(bc.mcap.T1F ~ Treatment*Time.Point, data = mcap.samp.df.T1F) #adonis test no clade
    adonis(bc.mcap.T1F ~ Clade*Treatment*Time.Point, data = mcap.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(mcap.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
  bc.mcapT1 <- phyloseq::distance(mcap.bac.T1, method = "bray") 
      mcap.samp.df.T1 <- as(sample_data(mcap.bac.T1), "data.frame") 
    adonis(bc.mcapT1 ~ Clade*Treatment, data = mcap.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Clade)  #clade
    permutest(beta.mcap.T1, pairwise = T) 
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Treatment) #treatment
    permutest(beta.mcap.T1, pairwise = T) 

#TF
  bc.mcap.TF <- phyloseq::distance(mcap.bac.TF, method = "bray")
     mcap.samp.df.TF <- as(sample_data(mcap.bac.TF), "data.frame") 
    adonis(bc.mcap.TF ~ Clade*Treatment, data = mcap.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Clade) 
    permutest(beta.mcap.TF, pairwise = T) 
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Treatment) 
    permutest(beta.mcap.TF, pairwise = T) 

######### by phylotype: C vs CD
# C ONLY
  mcap.bac.T1F.C <- subset_samples(mcap.bac.T1F, Clade == "C")
  mcap.bac.T1.C <- subset_samples(mcap.bac.T1, Clade == "C")
  mcap.bac.TF.C <- subset_samples(mcap.bac.TF, Clade == "C")
  
#T1TF.C
  bc.mcap.T1F.C <- phyloseq::distance(mcap.bac.T1F.C, method = "bray")
    mcap.samp.df.T1F.C <- as(sample_data(mcap.bac.T1F.C), "data.frame") 
  adonis(bc.mcap.T1F.C ~ Treatment*Time.Point, data = mcap.samp.df.T1F.C)

#T1.C
  bc.mcapT1.C <- phyloseq::distance(mcap.bac.T1.C, method = "bray")
    mcap.samp.df.T1.C <- as(sample_data(mcap.bac.T1.C), "data.frame") #sample df
  adonis(bc.mcapT1.C ~ Treatment, data = mcap.samp.df.T1.C)
  beta.mcap.T1.C <- betadisper(bc.mcapT1.C, mcap.samp.df.T1.C$Treatment)
    permutest(beta.mcap.T1.C, pairwise = T) 

#TF.C
  bc.mcap.TF.C <- phyloseq::distance(mcap.bac.TF.C, method = "bray")
    mcap.samp.df.TF.C <- as(sample_data(mcap.bac.TF.C), "data.frame") #sample df
  adonis(bc.mcap.TF.C ~ Treatment, data = mcap.samp.df.TF.C)
  beta.mcap.TF.C <- betadisper(bc.mcap.TF.C, mcap.samp.df.TF.C$Treatment)
    permutest(beta.mcap.TF.C, pairwise = T) 

# CD ONLY
mcap.bac.T1F.CD <- subset_samples(mcap.bac.T1F, Clade == "CD")
mcap.bac.T1.CD <- subset_samples(mcap.bac.T1, Clade == "CD")
mcap.bac.TF.CD <- subset_samples(mcap.bac.TF, Clade == "CD")
  
#T1TF.CD
  bc.mcap.T1F.CD <- phyloseq::distance(mcap.bac.T1F.CD, method = "bray")
  mcap.samp.df.T1F.CD <- as(sample_data(mcap.bac.T1F.CD), "data.frame") 
  adonis(bc.mcap.T1F.CD ~ Treatment*Time.Point, data = mcap.samp.df.T1F.CD)

#T1.CD
  bc.mcapT1.CD <- phyloseq::distance(mcap.bac.T1.CD, method = "bray")
    mcap.samp.df.T1.CD <- as(sample_data(mcap.bac.T1.CD), "data.frame") 
  adonis(bc.mcapT1.CD ~ Treatment, data = mcap.samp.df.T1.CD)
    beta.mcap.T1.CD <- betadisper(bc.mcapT1.CD, mcap.samp.df.T1.CD$Treatment)
  permutest(beta.mcap.T1.CD, pairwise = T) 

#TF.CD
  bc.mcap.TF.CD <- phyloseq::distance(mcap.bac.TF.CD, method = "bray")
    mcap.samp.df.TF.CD <- as(sample_data(mcap.bac.TF.CD), "data.frame") 
  adonis(bc.mcap.TF.CD ~ Treatment, data = mcap.samp.df.TF.CD)
  beta.mcap.TF.CD <- betadisper(bc.mcap.TF.CD, mcap.samp.df.TF.CD$Treatment)
    permutest(beta.mcap.TF.CD, pairwise = T) 
``` 
Mcap NMDS
```{r} 
## create palette for parent.ID and treatments
Mcap.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", 
                        "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#DDCC77","C High"="#999933", "CD Ambient"="#44AA99", "CD High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 

#T1 and TF
ord.mcap2 <- ordinate(mcap.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap2)

scores.mcap.samp.df<-as.data.frame(cbind(vegan::scores(ord.mcap2, display="sites"))) # these are "points" ask hannah 
scores.mcap.samp.df$Time.Point <- mcap.samp.df.T1F$Time.Point
scores.mcap.samp.df$Parent.ID <- mcap.samp.df.T1F$Parent.ID
scores.mcap.samp.df$Treatment <- mcap.samp.df.T1F$Treatment
scores.mcap.samp.df$Clade <- mcap.samp.df.T1F$Clade
scores.mcap.samp.df$Code <- paste(mcap.samp.df.T1F$Parent.ID,mcap.samp.df.T1F$Time.Point)
scores.mcap.samp.df$Code2 <- paste(mcap.samp.df.T1F$Treatment,mcap.samp.df.T1F$Time.Point)
scores.mcap.samp.df$Code3 <- paste(mcap.samp.df.T1F$Treatment,mcap.samp.df.T1F$Time.Point,mcap.samp.df.T1F$Parent.ID)

ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Mcap.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()

ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#T1
ord.mcap.T1 <- ordinate(mcap.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap.T1)

ord.mcap.T1.df<-as.data.frame(cbind(vegan::scores(ord.mcap.T1, display="sites"))) # these are "points" ask hannah 
ord.mcap.T1.df$Time.Point <- mcap.samp.df.T1$Time.Point
ord.mcap.T1.df$Parent.ID <- mcap.samp.df.T1$Parent.ID
ord.mcap.T1.df$Treatment <- mcap.samp.df.T1$Treatment
ord.mcap.T1.df$Clade <- mcap.samp.df.T1$Clade
ord.mcap.T1.df$Code <- paste(mcap.samp.df.T1$Parent.ID,mcap.samp.df.T1$Time.Point)
ord.mcap.T1.df$Code2 <- paste(mcap.samp.df.T1$Clade,mcap.samp.df.T1$Treatment)
ord.mcap.T1.df$Code2 <- as.factor(ord.mcap.T1.df$Code2)


ggplot(ord.mcap.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata T1") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(mcap.bac.TF, ordinate(mcap.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.mcapTF <- ordinate(mcap.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcapTF)
ord.mcap.TF.df<-as.data.frame(cbind(vegan::scores(ord.mcapTF, display="sites"))) # these are "points" ask hannah 
ord.mcap.TF.df$Treatment <- mcap.samp.df.TF$Treatment
ord.mcap.TF.df$Parent.ID <- mcap.samp.df.TF$Parent.ID
ord.mcap.TF.df$Clade <- mcap.samp.df.TF$Clade
ord.mcap.TF.df$CladeTreat <- as.factor(paste(mcap.samp.df.TF$Clade,mcap.samp.df.TF$Treatment))
ord.mcap.TF.df$Code <- as.factor(paste(mcap.samp.df.TF$Parent.ID,mcap.samp.df.TF$Treatment))
ord.mcap.TF.df$Code2 <- paste(mcap.samp.df.TF$Clade,mcap.samp.df.TF$Treatment)
ord.mcap.TF.df$Code2 <- as.factor(ord.mcap.TF.df$Code2)

ggplot(ord.mcap.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")



#### BY CLADE
#ords


#C T1 and TF 
set.seed(30)
ord.mcap_C <- ordinate(mcap.bac.T1F.C , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C)
scores.mcap_C<-as.data.frame(cbind(vegan::scores(ord.mcap_C, display="sites"))) # these are "points" ask hannah 
scores.mcap_C$Treatment <- mcap.samp.df.T1F.C$Treatment
scores.mcap_C$Parent.ID <- mcap.samp.df.T1F.C$Parent.ID
scores.mcap_C$Time.Point <- mcap.samp.df.T1F.C$Time.Point
scores.mcap_C$Code2<-as.factor(paste(scores.mcap_C$Treatment, scores.mcap_C$Time.Point))

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape =Code2 , color = Parent.ID, fill=Code2), size = 5,stroke = 1) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Mcap.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata C only T1F") +
  theme_classic()


#C T1
set.seed(30)
ord.mcap_C.T1 <- ordinate(mcap.bac.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.T1)
scores.mcap_C.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_C.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.T1$Treatment <- mcap.samp.df.T1.C$Treatment
scores.mcap_C.T1$Parent.ID <- mcap.samp.df.T1.C$Parent.ID
scores.mcap_C.T1$Time.Point <- mcap.samp.df.T1.C$Time.Point

ggplot(scores.mcap_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("Mcap_T1_its_C.pdf")

#C TF
set.seed(30)
ord.mcap_C.TF <- ordinate(mcap.bac.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.TF)
scores.mcap_C.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_C.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.TF$Treatment <- mcap.samp.df.TF.C$Treatment
scores.mcap_C.TF$Parent.ID <- mcap.samp.df.TF.C$Parent.ID
scores.mcap_C.TF$Time.Point <- mcap.samp.df.TF.C$Time.Point

ggplot(scores.mcap_C.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("Mcap_TF_its_TF_C.pdf")


#CD mcap.bacT1F.CD
set.seed(30)
ord.mcap_CD <- ordinate(mcap.bac.T1F.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD)
scores.mcap_CD<-as.data.frame(cbind(vegan::scores(ord.mcap_CD, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD$Treatment <- mcap.samp.df.T1F.CD$Treatment
scores.mcap_CD$Parent.ID <- mcap.samp.df.T1F.CD$Parent.ID
scores.mcap_CD$Time.Point <- mcap.samp.df.T1F.CD$Time.Point

ggplot(scores.mcap_CD, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Mcap.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.mcap_CD.T1 <- ordinate(mcap.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.T1)
scores.mcap_CD.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.T1$Treatment <- mcap.samp.df.T1.CD$Treatment
scores.mcap_CD.T1$Parent.ID <- mcap.samp.df.T1.CD$Parent.ID
scores.mcap_CD.T1$Time.Point <- mcap.samp.df.T1.CD$Time.Point

ggplot(scores.mcap_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_T1.pdf")

#CD TF
set.seed(30)
ord.mcap_CD.TF <- ordinate(mcap.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.TF)
scores.mcap_CD.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.TF$Treatment <- mcap.samp.df.TF.CD$Treatment
scores.mcap_CD.TF$Parent.ID <- mcap.samp.df.TF.CD$Parent.ID
scores.mcap_CD.TF$Time.Point <- mcap.samp.df.TF.CD$Time.Point

ggplot(scores.mcap_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Mcap_ITS2_CD_TF.pdf")



```

Montipora - T1 and TF together, separet, and then by phylotype. PERMANOVA  
Family good to go
```{r}     
# df   Bac_RelA_stats.T1F
#Mcap
mcap.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Montipora_capitata")
mcap.bac.T1F = prune_taxa(taxa_sums(mcap.bac.T1F) > 0, mcap.bac.T1F) #this removes any OTU with 0s




#make dfs you need 
mcap.bac.T1 <- subset_samples(mcap.bac.T1F, Time.Point == "T1")
  mcap.bac.T1 = prune_taxa(taxa_sums(mcap.bac.T1) > 0, mcap.bac.T1) #this removes any OTU with 0s

mcap.bac.TF <- subset_samples(mcap.bac.T1F, Time.Point == "TF")
  mcap.bac.TF = prune_taxa(taxa_sums(mcap.bac.TF) > 0, mcap.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.mcap.T1F <- phyloseq::distance(mcap.bac.T1F, method = "bray")
     mcap.samp.df.T1F <- as(sample_data(mcap.bac.T1F), "data.frame") #sample df
    adonis(bc.mcap.T1F ~ Treatment*Time.Point, data = mcap.samp.df.T1F) #adonis test no clade
    adonis(bc.mcap.T1F ~ Clade*Treatment*Time.Point, data = mcap.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(mcap.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
  bc.mcapT1 <- phyloseq::distance(mcap.bac.T1, method = "bray") 
      mcap.samp.df.T1 <- as(sample_data(mcap.bac.T1), "data.frame") 
    adonis(bc.mcapT1 ~ Clade*Treatment, data = mcap.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Clade)  #clade
    permutest(beta.mcap.T1, pairwise = T) 
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Treatment) #treatment
    permutest(beta.mcap.T1, pairwise = T) 

#TF
  bc.mcap.TF <- phyloseq::distance(mcap.bac.TF, method = "bray")
     mcap.samp.df.TF <- as(sample_data(mcap.bac.TF), "data.frame") 
    adonis(bc.mcap.TF ~ Clade*Treatment, data = mcap.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Clade) 
    permutest(beta.mcap.TF, pairwise = T) 
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Treatment) 
    permutest(beta.mcap.TF, pairwise = T) 

######### by phylotype: C vs CD
# C ONLY
  mcap.bac.T1F.C <- subset_samples(mcap.bac.T1F, Clade == "C")
  mcap.bac.T1.C <- subset_samples(mcap.bac.T1, Clade == "C")
  mcap.bac.TF.C <- subset_samples(mcap.bac.TF, Clade == "C")
  
#T1TF.C
  bc.mcap.T1F.C <- phyloseq::distance(mcap.bac.T1F.C, method = "bray")
    mcap.samp.df.T1F.C <- as(sample_data(mcap.bac.T1F.C), "data.frame") 
  adonis(bc.mcap.T1F.C ~ Treatment*Time.Point, data = mcap.samp.df.T1F.C)

#T1.C
  bc.mcapT1.C <- phyloseq::distance(mcap.bac.T1.C, method = "bray")
    mcap.samp.df.T1.C <- as(sample_data(mcap.bac.T1.C), "data.frame") #sample df
  adonis(bc.mcapT1.C ~ Treatment, data = mcap.samp.df.T1.C)
  beta.mcap.T1.C <- betadisper(bc.mcapT1.C, mcap.samp.df.T1.C$Treatment)
    permutest(beta.mcap.T1.C, pairwise = T) 

#TF.C
  bc.mcap.TF.C <- phyloseq::distance(mcap.bac.TF.C, method = "bray")
    mcap.samp.df.TF.C <- as(sample_data(mcap.bac.TF.C), "data.frame") #sample df
  adonis(bc.mcap.TF.C ~ Treatment, data = mcap.samp.df.TF.C)
  beta.mcap.TF.C <- betadisper(bc.mcap.TF.C, mcap.samp.df.TF.C$Treatment)
    permutest(beta.mcap.TF.C, pairwise = T) 

# CD ONLY
mcap.bac.T1F.CD <- subset_samples(mcap.bac.T1F, Clade == "CD")
mcap.bac.T1.CD <- subset_samples(mcap.bac.T1, Clade == "CD")
mcap.bac.TF.CD <- subset_samples(mcap.bac.TF, Clade == "CD")
  
#T1TF.CD
  bc.mcap.T1F.CD <- phyloseq::distance(mcap.bac.T1F.CD, method = "bray")
  mcap.samp.df.T1F.CD <- as(sample_data(mcap.bac.T1F.CD), "data.frame") 
  adonis(bc.mcap.T1F.CD ~ Treatment*Time.Point, data = mcap.samp.df.T1F.CD)

#T1.CD
  bc.mcapT1.CD <- phyloseq::distance(mcap.bac.T1.CD, method = "bray")
    mcap.samp.df.T1.CD <- as(sample_data(mcap.bac.T1.CD), "data.frame") 
  adonis(bc.mcapT1.CD ~ Treatment, data = mcap.samp.df.T1.CD)
    beta.mcap.T1.CD <- betadisper(bc.mcapT1.CD, mcap.samp.df.T1.CD$Treatment)
  permutest(beta.mcap.T1.CD, pairwise = T) 

#TF.CD
  bc.mcap.TF.CD <- phyloseq::distance(mcap.bac.TF.CD, method = "bray")
    mcap.samp.df.TF.CD <- as(sample_data(mcap.bac.TF.CD), "data.frame") 
  adonis(bc.mcap.TF.CD ~ Treatment, data = mcap.samp.df.TF.CD)
  beta.mcap.TF.CD <- betadisper(bc.mcap.TF.CD, mcap.samp.df.TF.CD$Treatment)
    permutest(beta.mcap.TF.CD, pairwise = T) 
``` 
Mcap NMDS Family good 
```{r} 
## create palette for parent.ID and treatments
Mcap.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", 
                        "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#DDCC77","C High"="#999933", "CD Ambient"="#44AA99", "CD High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 

#T1 and TF
ord.mcap2 <- ordinate(mcap.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap2)

scores.mcap.samp.df<-as.data.frame(cbind(vegan::scores(ord.mcap2, display="sites"))) # these are "points" ask hannah 
scores.mcap.samp.df$Time.Point <- mcap.samp.df.T1F$Time.Point
scores.mcap.samp.df$Parent.ID <- mcap.samp.df.T1F$Parent.ID
scores.mcap.samp.df$Treatment <- mcap.samp.df.T1F$Treatment
scores.mcap.samp.df$Clade <- mcap.samp.df.T1F$Clade
scores.mcap.samp.df$Code <- paste(mcap.samp.df.T1F$Parent.ID,mcap.samp.df.T1F$Time.Point)
scores.mcap.samp.df$Code2 <- paste(mcap.samp.df.T1F$Treatment,mcap.samp.df.T1F$Time.Point)
scores.mcap.samp.df$Code3 <- paste(mcap.samp.df.T1F$Treatment,mcap.samp.df.T1F$Time.Point,mcap.samp.df.T1F$Parent.ID)

ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Mcap.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()

ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#T1
ord.mcap.T1 <- ordinate(mcap.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap.T1)

ord.mcap.T1.df<-as.data.frame(cbind(vegan::scores(ord.mcap.T1, display="sites"))) # these are "points" ask hannah 
ord.mcap.T1.df$Time.Point <- mcap.samp.df.T1$Time.Point
ord.mcap.T1.df$Parent.ID <- mcap.samp.df.T1$Parent.ID
ord.mcap.T1.df$Treatment <- mcap.samp.df.T1$Treatment
ord.mcap.T1.df$Clade <- mcap.samp.df.T1$Clade
ord.mcap.T1.df$Code <- paste(mcap.samp.df.T1$Parent.ID,mcap.samp.df.T1$Time.Point)
ord.mcap.T1.df$Code2 <- paste(mcap.samp.df.T1$Clade,mcap.samp.df.T1$Treatment)
ord.mcap.T1.df$Code2 <- as.factor(ord.mcap.T1.df$Code2)


ggplot(ord.mcap.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata T1") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(mcap.bac.TF, ordinate(mcap.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.mcapTF <- ordinate(mcap.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcapTF)
ord.mcap.TF.df<-as.data.frame(cbind(vegan::scores(ord.mcapTF, display="sites"))) # these are "points" ask hannah 
ord.mcap.TF.df$Treatment <- mcap.samp.df.TF$Treatment
ord.mcap.TF.df$Parent.ID <- mcap.samp.df.TF$Parent.ID
ord.mcap.TF.df$Clade <- mcap.samp.df.TF$Clade
ord.mcap.TF.df$CladeTreat <- as.factor(paste(mcap.samp.df.TF$Clade,mcap.samp.df.TF$Treatment))
ord.mcap.TF.df$Code <- as.factor(paste(mcap.samp.df.TF$Parent.ID,mcap.samp.df.TF$Treatment))
ord.mcap.TF.df$Code2 <- paste(mcap.samp.df.TF$Clade,mcap.samp.df.TF$Treatment)
ord.mcap.TF.df$Code2 <- as.factor(ord.mcap.TF.df$Code2)

ggplot(ord.mcap.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")



#### BY CLADE
#ords


#C T1 and TF 
set.seed(30)
ord.mcap_C <- ordinate(mcap.bac.T1F.C , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C)
scores.mcap_C<-as.data.frame(cbind(vegan::scores(ord.mcap_C, display="sites"))) # these are "points" ask hannah 
scores.mcap_C$Treatment <- mcap.samp.df.T1F.C$Treatment
scores.mcap_C$Parent.ID <- mcap.samp.df.T1F.C$Parent.ID
scores.mcap_C$Time.Point <- mcap.samp.df.T1F.C$Time.Point
scores.mcap_C$Code2<-as.factor(paste(scores.mcap_C$Treatment, scores.mcap_C$Time.Point))

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape =Code2 , color = Parent.ID, fill=Code2), size = 5,stroke = 1) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Mcap.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata C only T1F") +
  theme_classic()


#C T1
set.seed(30)
ord.mcap_C.T1 <- ordinate(mcap.bac.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.T1)
scores.mcap_C.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_C.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.T1$Treatment <- mcap.samp.df.T1.C$Treatment
scores.mcap_C.T1$Parent.ID <- mcap.samp.df.T1.C$Parent.ID
scores.mcap_C.T1$Time.Point <- mcap.samp.df.T1.C$Time.Point

ggplot(scores.mcap_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("Mcap_T1_its_C.pdf")

#C TF
set.seed(30)
ord.mcap_C.TF <- ordinate(mcap.bac.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.TF)
scores.mcap_C.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_C.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.TF$Treatment <- mcap.samp.df.TF.C$Treatment
scores.mcap_C.TF$Parent.ID <- mcap.samp.df.TF.C$Parent.ID
scores.mcap_C.TF$Time.Point <- mcap.samp.df.TF.C$Time.Point

ggplot(scores.mcap_C.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("Mcap_TF_its_TF_C.pdf")


#CD mcap.bacT1F.CD
set.seed(30)
ord.mcap_CD <- ordinate(mcap.bac.T1F.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD)
scores.mcap_CD<-as.data.frame(cbind(vegan::scores(ord.mcap_CD, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD$Treatment <- mcap.samp.df.T1F.CD$Treatment
scores.mcap_CD$Parent.ID <- mcap.samp.df.T1F.CD$Parent.ID
scores.mcap_CD$Time.Point <- mcap.samp.df.T1F.CD$Time.Point

ggplot(scores.mcap_CD, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Mcap.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.mcap_CD.T1 <- ordinate(mcap.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.T1)
scores.mcap_CD.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.T1$Treatment <- mcap.samp.df.T1.CD$Treatment
scores.mcap_CD.T1$Parent.ID <- mcap.samp.df.T1.CD$Parent.ID
scores.mcap_CD.T1$Time.Point <- mcap.samp.df.T1.CD$Time.Point

ggplot(scores.mcap_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_T1.pdf")

#CD TF
set.seed(30)
ord.mcap_CD.TF <- ordinate(mcap.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.TF)
scores.mcap_CD.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.TF$Treatment <- mcap.samp.df.TF.CD$Treatment
scores.mcap_CD.TF$Parent.ID <- mcap.samp.df.TF.CD$Parent.ID
scores.mcap_CD.TF$Time.Point <- mcap.samp.df.TF.CD$Time.Point

ggplot(scores.mcap_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Mcap_ITS2_CD_TF.pdf")



```


Porites - T1 and TF together, separet, and then by phylotype. PERMANOVA
CHANGE PHYLOTYPES
```{r}     
# df Bac_RelA_stats.T1F
#Pcomp
Pcomp.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Porites_compressa")
Pcomp.bac.T1F = prune_taxa(taxa_sums(Pcomp.bac.T1F) > 0, Pcomp.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
Pcomp.bac.T1 <- subset_samples(Pcomp.bac.T1F, Time.Point == "T1")
  Pcomp.bac.T1 = prune_taxa(taxa_sums(Pcomp.bac.T1) > 0, Pcomp.bac.T1) #this removes any OTU with 0s

Pcomp.bac.TF <- subset_samples(Pcomp.bac.T1F, Time.Point == "TF")
  Pcomp.bac.TF = prune_taxa(taxa_sums(Pcomp.bac.TF) > 0, Pcomp.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F <- phyloseq::distance(Pcomp.bac.T1F, method = "bray")
     Pcomp.samp.df.T1F <- as(sample_data(Pcomp.bac.T1F), "data.frame") #sample df
    adonis(bc.Pcomp.T1F ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F) #adonis test no clade
    adonis(bc.Pcomp.T1F ~ Clade*Treatment*Time.Point, data = Pcomp.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(Pcomp.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
  bc.PcompT1 <- phyloseq::distance(Pcomp.bac.T1, method = "bray") 
      Pcomp.samp.df.T1 <- as(sample_data(Pcomp.bac.T1), "data.frame") 
    adonis(bc.PcompT1 ~ Clade*Treatment, data = Pcomp.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Clade)  #clade
    permutest(beta.Pcomp.T1, pairwise = T) 
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Treatment) #treatment
    permutest(beta.Pcomp.T1, pairwise = T) 

#TF
  bc.Pcomp.TF <- phyloseq::distance(Pcomp.bac.TF, method = "bray")
     Pcomp.samp.df.TF <- as(sample_data(Pcomp.bac.TF), "data.frame") 
    adonis(bc.Pcomp.TF ~ Clade*Treatment, data = Pcomp.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Clade) 
    permutest(beta.Pcomp.TF, pairwise = T) 
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Treatment) 
    permutest(beta.Pcomp.TF, pairwise = T) 

######### by phylotype: C15cj vs C15n
# C ONLY
  Pcomp.bac.T1F.C15cj <- subset_samples(Pcomp.bac.T1F, Clade == "C15cj")
  Pcomp.bac.T1.C15cj <- subset_samples(Pcomp.bac.T1, Clade == "C15cj")
  Pcomp.bac.TF.C15cj <- subset_samples(Pcomp.bac.TF, Clade == "C15cj")
  
#T1TF.C15cj
  bc.Pcomp.T1F.C15cj <- phyloseq::distance(Pcomp.bac.T1F.C15cj, method = "bray")
    Pcomp.samp.df.T1F.C15cj <- as(sample_data(Pcomp.bac.T1F.C15cj), "data.frame") 
  adonis(bc.Pcomp.T1F.C15cj ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C15cj)

#T1.C15cj
  bc.PcompT1.C15cj <- phyloseq::distance(Pcomp.bac.T1.C15cj, method = "bray")
    Pcomp.samp.df.T1.C15cj <- as(sample_data(Pcomp.bac.T1.C15cj), "data.frame") #sample df
  adonis(bc.PcompT1.C15cj ~ Treatment, data = Pcomp.samp.df.T1.C15cj)
  beta.Pcomp.T1.C15cj <- betadisper(bc.PcompT1.C15cj, Pcomp.samp.df.T1.C15cj$Treatment)
    permutest(beta.Pcomp.T1.C15cj, pairwise = T) 

#TF.C15cj
  bc.Pcomp.TF.C15cj <- phyloseq::distance(Pcomp.bac.TF.C15cj, method = "bray")
    Pcomp.samp.df.TF.C15cj <- as(sample_data(Pcomp.bac.TF.C15cj), "data.frame") #sample df
  adonis(bc.Pcomp.TF.C15cj ~ Treatment, data = Pcomp.samp.df.TF.C15cj)
  beta.Pcomp.TF.C15cj <- betadisper(bc.Pcomp.TF.C15cj, Pcomp.samp.df.TF.C15cj$Treatment)
    permutest(beta.Pcomp.TF.C15cj, pairwise = T) 

# CD ONLY
Pcomp.bac.T1F.C15n <- subset_samples(Pcomp.bac.T1F, Clade == "C15n")
Pcomp.bac.T1.C15n <- subset_samples(Pcomp.bac.T1, Clade == "C15n")
Pcomp.bac.TF.C15n <- subset_samples(Pcomp.bac.TF, Clade == "C15n")
  
#T1TF.C15n
  bc.Pcomp.T1F.C15n <- phyloseq::distance(Pcomp.bac.T1F.C15n, method = "bray")
  Pcomp.samp.df.T1F.C15n <- as(sample_data(Pcomp.bac.T1F.C15n), "data.frame") 
  adonis(bc.Pcomp.T1F.C15n ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C15n)

#T1.C15n
  bc.PcompT1.C15n <- phyloseq::distance(Pcomp.bac.T1.C15n, method = "bray")
    Pcomp.samp.df.T1.C15n <- as(sample_data(Pcomp.bac.T1.C15n), "data.frame") 
  adonis(bc.PcompT1.C15n ~ Treatment, data = Pcomp.samp.df.T1.C15n)
    beta.Pcomp.T1.C15n <- betadisper(bc.PcompT1.C15n, Pcomp.samp.df.T1.C15n$Treatment)
  permutest(beta.Pcomp.T1.C15n, pairwise = T) 

#TF.C15n
  bc.Pcomp.TF.C15n <- phyloseq::distance(Pcomp.bac.TF.C15n, method = "bray")
    Pcomp.samp.df.TF.C15n <- as(sample_data(Pcomp.bac.TF.C15n), "data.frame") 
  adonis(bc.Pcomp.TF.C15n ~ Treatment, data = Pcomp.samp.df.TF.C15n)
  beta.Pcomp.TF.C15n <- betadisper(bc.Pcomp.TF.C15n, Pcomp.samp.df.TF.C15n$Treatment)
    permutest(beta.Pcomp.TF.C15n, pairwise = T) 
``` 
Pcomp NMDS
```{r} 
## create palette for parent.ID and treatments
Pcomp.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", Ambient = "Blue", High = "Red","C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred", "Ambient T1"="darkred", "Ambient TF"="darkblue", "High T1"="pink","High TF"="darkred") 

#T1 and TF
ord.Pcomp2 <- ordinate(Pcomp.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pcomp2)

scores.Pcomp.samp.df<-as.data.frame(cbind(vegan::scores(ord.Pcomp2, display="sites"))) # these are "points" ask hannah 
scores.Pcomp.samp.df$Time.Point <- Pcomp.samp.df.T1F$Time.Point
scores.Pcomp.samp.df$Parent.ID <- Pcomp.samp.df.T1F$Parent.ID
scores.Pcomp.samp.df$Treatment <- Pcomp.samp.df.T1F$Treatment
scores.Pcomp.samp.df$Clade <- Pcomp.samp.df.T1F$Clade
scores.Pcomp.samp.df$Code <- paste(Pcomp.samp.df.T1F$Parent.ID,Pcomp.samp.df.T1F$Time.Point)
scores.Pcomp.samp.df$Code2 <- paste(Pcomp.samp.df.T1F$Treatment,Pcomp.samp.df.T1F$Time.Point)
scores.Pcomp.samp.df$Code3 <- paste(Pcomp.samp.df.T1F$Treatment,Pcomp.samp.df.T1F$Time.Point,Pcomp.samp.df.T1F$Parent.ID)

ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Pcomp.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. com T1TF") +
  theme_classic()

ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. comp T1TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")





ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2), linetype = 2) +
  ggtitle("Pcmop T1TF") +
  theme_classic()
#ggsave("PcompT1its2.pdf")



#T1
ord.Pcomp.T1 <- ordinate(Pcomp.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pcomp.T1)

ord.Pcomp.T1.df<-as.data.frame(cbind(vegan::scores(ord.Pcomp.T1, display="sites"))) # these are "points" ask hannah 
ord.Pcomp.T1.df$Time.Point <- Pcomp.samp.df.T1$Time.Point
ord.Pcomp.T1.df$Parent.ID <- Pcomp.samp.df.T1$Parent.ID
ord.Pcomp.T1.df$Treatment <- Pcomp.samp.df.T1$Treatment
ord.Pcomp.T1.df$Clade <- Pcomp.samp.df.T1$Clade
ord.Pcomp.T1.df$Code <- paste(Pcomp.samp.df.T1$Parent.ID,Pcomp.samp.df.T1$Time.Point)

#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-Section-01-Main-Lab.html#procrustes-and-ggplot2 
ggplot(ord.Pcomp.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
    geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("Pcomp T1") +
  theme_classic()
#ggsave("PcompT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pcomp.bac.TF, ordinate(Pcomp.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.PcompTF <- ordinate(Pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompTF)
ord.Pcomp.TF.df<-as.data.frame(cbind(vegan::scores(ord.PcompTF, display="sites"))) # these are "points" ask hannah 
ord.Pcomp.TF.df$Treatment <- Pcomp.samp.df.TF$Treatment
ord.Pcomp.TF.df$Parent.ID <- Pcomp.samp.df.TF$Parent.ID
ord.Pcomp.TF.df$Clade <- Pcomp.samp.df.TF$Clade
ord.Pcomp.TF.df$CladeTreat <- as.factor(paste(Pcomp.samp.df.TF$Clade,Pcomp.samp.df.TF$Treatment))
ord.Pcomp.TF.df$Code <- as.factor(paste(Pcomp.samp.df.TF$Parent.ID,Pcomp.samp.df.TF$Treatment))

ggplot(ord.Pcomp.TF.df, aes(x = NMDS1, y = NMDS2)) + 
 geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
    scale_color_manual(values=Pcomp.Parent.colors)+
    #geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp TF") +
  theme_classic()
#ggsave("PcompT1its2.pdf")



#### BY CLADE
#ords


#C15cj T1 and TF 
set.seed(30)
ord.Pcomp.C15cj <- ordinate(Pcomp.bac.T1F.C15cj , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp.C15cj)
scores.Pcomp.C15cj<-as.data.frame(cbind(vegan::scores(ord.Pcomp.C15cj, display="sites"))) # these are "points" ask hannah
scores.Pcomp.C15cj$Treatment <- Pcomp.samp.df.T1F.C15cj$Treatment
scores.Pcomp.C15cj$Parent.ID <- Pcomp.samp.df.T1F.C15cj$Parent.ID
scores.Pcomp.C15cj$Time.Point <- Pcomp.samp.df.T1F.C15cj$Time.Point

ggplot(scores.Pcomp.C15cj, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P.comp C15cj") +
  theme_classic()
#ggsave("PcompTFits_C.pdf")


#C15cj T1
set.seed(30)
ord.Pcomp_C.T1 <- ordinate(Pcomp.bac.T1.C15cj, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_C.T1)
scores.Pcomp_C.T1<-as.data.frame(cbind(vegan::scores(ord.Pcomp_C.T1, display="sites"))) # these are "points" ask hannah 
scores.Pcomp_C.T1$Treatment <- Pcomp.samp.df.T1.C15cj$Treatment
scores.Pcomp_C.T1$Parent.ID <- Pcomp.samp.df.T1.C15cj$Parent.ID
scores.Pcomp_C.T1$Time.Point <- Pcomp.samp.df.T1.C15cj$Time.Point

ggplot(scores.Pcomp_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P comp T1 Cladocopium") +
  theme_classic()
#ggsave("Pcomp_T1_its_C.pdf")

#C TF
set.seed(30)
ord.Pcomp_C15n.TF <- ordinate(Pcomp.bac.TF.C15cj, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_C15n.TF)
scores.Pcomp_C15n.TF<-as.data.frame(cbind(vegan::scores(ord.Pcomp_C15n.TF, display="sites"))) # 
scores.Pcomp_C15n.TF$Treatment <- Pcomp.samp.df.TF.C15cj$Treatment
scores.Pcomp_C15n.TF$Parent.ID <- Pcomp.samp.df.TF.C15cj$Parent.ID
scores.Pcomp_C15n.TF$Time.Point <- Pcomp.samp.df.TF.C15cj$Time.Point

ggplot(scores.Pcomp_C15n.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P.comp C15 TF") +
  theme_classic()
#ggsave("Pcomp_TF_its_TF_C.pdf")


#CD Pcomp.bacT1F.C15n
set.seed(30)
ord.Pcomp.C15n <- ordinate(Pcomp.bac.T1F.C15n, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp.C15n)
scores.Pcomp.C15n<-as.data.frame(cbind(vegan::scores(ord.Pcomp.C15n, display="sites"))) # these are "points" ask hannah 
scores.Pcomp.C15n$Treatment <- Pcomp.samp.df.T1F.CD$Treatment
scores.Pcomp.C15n$Parent.ID <- Pcomp.samp.df.T1F.CD$Parent.ID
scores.Pcomp.C15n$Time.Point <- Pcomp.samp.df.T1F.CD$Time.Point

ggplot(scores.Pcomp.C15n, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("Pcomp_ITS2.C15n_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.Pcomp.C15n.T1 <- ordinate(Pcomp.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp.C15n.T1)
scores.Pcomp.C15n.T1<-as.data.frame(cbind(vegan::scores(ord.Pcomp.C15n.T1, display="sites"))) # these are "points" ask hannah 
scores.Pcomp.C15n.T1$Treatment <- Pcomp.samp.df.T1.CD$Treatment
scores.Pcomp.C15n.T1$Parent.ID <- Pcomp.samp.df.T1.CD$Parent.ID
scores.Pcomp.C15n.T1$Time.Point <- Pcomp.samp.df.T1.CD$Time.Point

ggplot(scores.Pcomp.C15n.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("Pcomp_ITS2.C15n_T1.pdf")

#CD TF
set.seed(30)
ord.Pcomp.C15n.TF <- ordinate(Pcomp.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp.C15n.TF)
scores.Pcomp.C15n.TF<-as.data.frame(cbind(vegan::scores(ord.Pcomp.C15n.TF, display="sites"))) # these are "points" ask hannah 
scores.Pcomp.C15n.TF$Treatment <- Pcomp.samp.df.TF.CD$Treatment
scores.Pcomp.C15n.TF$Parent.ID <- Pcomp.samp.df.TF.CD$Parent.ID
scores.Pcomp.C15n.TF$Time.Point <- Pcomp.samp.df.TF.CD$Time.Point

ggplot(scores.Pcomp.C15n.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Pcomp_ITS2.C15n_TF.pdf")



```

Pavona - T1 and TF together, separet, and then by phylotype. PERMANOVA
CHANGE PHYLOTYPES

```{r}     
# df Bac_RelA_stats.T1F
#Pvar
Pvar.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Pavona_varians")
Pvar.bac.T1F = prune_taxa(taxa_sums(Pvar.bac.T1F) > 0, Pvar.bac.T1F) #this removes any OTU with 0s
 
#make dfs you need 
Pvar.bac.T1 <- subset_samples(Pvar.bac.T1F, Time.Point == "T1")
  Pvar.bac.T1 = prune_taxa(taxa_sums(Pvar.bac.T1) > 0, Pvar.bac.T1) #this removes any OTU with 0s

Pvar.bac.TF <- subset_samples(Pvar.bac.T1F, Time.Point == "TF")
  Pvar.bac.TF = prune_taxa(taxa_sums(Pvar.bac.TF) > 0, Pvar.bac.TF) #this removes any OTU with 0s
 
#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F <- phyloseq::distance(Pvar.bac.T1F, method = "bray")
     Pvar.samp.df.T1F <- as(sample_data(Pvar.bac.T1F), "data.frame") #sample df
    adonis(bc.Pvar.T1F ~ Treatment*Time.Point, data = Pvar.samp.df.T1F) #adonis test no clade
    adonis(bc.Pvar.T1F ~ Clade*Treatment*Time.Point, data = Pvar.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(Pvar.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
  bc.PvarT1 <- phyloseq::distance(Pvar.bac.T1, method = "bray") 
      Pvar.samp.df.T1 <- as(sample_data(Pvar.bac.T1), "data.frame") 
    adonis(bc.PvarT1 ~ Clade*Treatment, data = Pvar.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Clade)  #clade
    permutest(beta.Pvar.T1, pairwise = T) 
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Treatment) #treatment
    permutest(beta.Pvar.T1, pairwise = T) 

#TF
  bc.Pvar.TF <- phyloseq::distance(Pvar.bac.TF, method = "bray")
     Pvar.samp.df.TF <- as(sample_data(Pvar.bac.TF), "data.frame") 
    adonis(bc.Pvar.TF ~ Clade*Treatment, data = Pvar.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Clade) 
    permutest(beta.Pvar.TF, pairwise = T) 
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Treatment) 
    permutest(beta.Pvar.TF, pairwise = T) 

######### by phylotype: C vs CD
# C ONLY
  Pvar.bac.T1F.C27 <- subset_samples(Pvar.bac.T1F, Clade == "C27")
  Pvar.bac.T1.C27 <- subset_samples(Pvar.bac.T1, Clade == "C27")
  Pvar.bac.TF.C27 <- subset_samples(Pvar.bac.TF, Clade == "C27")
  
#T1TF.C27
  bc.Pvar.T1F.C27 <- phyloseq::distance(Pvar.bac.T1F.C27, method = "bray")
    Pvar.samp.df.T1F.C27 <- as(sample_data(Pvar.bac.T1F.C27), "data.frame") 
  adonis(bc.Pvar.T1F.C27 ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C27)

#T1.C27
  bc.PvarT1.C27 <- phyloseq::distance(Pvar.bac.T1.C27, method = "bray")
    Pvar.samp.df.T1.C27 <- as(sample_data(Pvar.bac.T1.C27), "data.frame") #sample df
  adonis(bc.PvarT1.C27 ~ Treatment, data = Pvar.samp.df.T1.C27)
  beta.Pvar.T1.C27 <- betadisper(bc.PvarT1.C27, Pvar.samp.df.T1.C27$Treatment)
    permutest(beta.Pvar.T1.C27, pairwise = T) 

#TF.C27
  bc.Pvar.TF.C27 <- phyloseq::distance(Pvar.bac.TF.C27, method = "bray")
    Pvar.samp.df.TF.C27 <- as(sample_data(Pvar.bac.TF.C27), "data.frame") #sample df
  adonis(bc.Pvar.TF.C27 ~ Treatment, data = Pvar.samp.df.TF.C27)
  beta.Pvar.TF.C27 <- betadisper(bc.Pvar.TF.C27, Pvar.samp.df.TF.C27$Treatment)
    permutest(beta.Pvar.TF.C27, pairwise = T) 

# C1 ONLY
Pvar.bac.T1F.C1 <- subset_samples(Pvar.bac.T1F, Clade == "C1")
Pvar.bac.T1.C1 <- subset_samples(Pvar.bac.T1, Clade == "C1")
Pvar.bac.TF.C1 <- subset_samples(Pvar.bac.TF, Clade == "C1")
  
#T1TF.C1
  bc.Pvar.T1F.C1 <- phyloseq::distance(Pvar.bac.T1F.C1, method = "bray")
  Pvar.samp.df.T1F.C1 <- as(sample_data(Pvar.bac.T1F.C1), "data.frame") 
  adonis(bc.Pvar.T1F.C1 ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C1)

#T1.C1
  bc.PvarT1.C1 <- phyloseq::distance(Pvar.bac.T1.C1, method = "bray")
    Pvar.samp.df.T1.C1 <- as(sample_data(Pvar.bac.T1.C1), "data.frame") 
  adonis(bc.PvarT1.C1 ~ Treatment, data = Pvar.samp.df.T1.C1)
    beta.Pvar.T1.C1 <- betadisper(bc.PvarT1.C1, Pvar.samp.df.T1.C1$Treatment)
  permutest(beta.Pvar.T1.C1, pairwise = T) 

#TF.C1
  bc.Pvar.TF.C1 <- phyloseq::distance(Pvar.bac.TF.C1, method = "bray")
    Pvar.samp.df.TF.C1 <- as(sample_data(Pvar.bac.TF.C1), "data.frame") 
  adonis(bc.Pvar.TF.C1 ~ Treatment, data = Pvar.samp.df.TF.C1)
  beta.Pvar.TF.C1 <- betadisper(bc.Pvar.TF.C1, Pvar.samp.df.TF.C1$Treatment)
    permutest(beta.Pvar.TF.C1, pairwise = T) 
``` 
Pvar NMDS
```{r} 
## create palette for parent.ID and treatments
Pvar.Parent.colors <- c("325" = "green4",  "353" = "firebrick3", "356" ="deeppink4", "358" = "orange", "361" = "green1","354" = "gold",  "351"= "hotpink", "355" ="darkslateblue", "357" = "cyan", "359" = "darkorange","360" ="darkorchid1", "362" = "lightblue", 
                       "Ambient" = "dodgerblue", "High" = "darkred",
                        "C1 Ambient"="#DDCC77","C1 High"="#999933", "C27 Ambient"="#44AA99", "C27 High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 


 
#T1 and TF
ord.Pvar2 <- ordinate(Pvar.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pvar2)

scores.Pvar.samp.df<-as.data.frame(cbind(vegan::scores(ord.Pvar2, display="sites"))) # these are "points" ask hannah 
scores.Pvar.samp.df$Time.Point <- Pvar.samp.df.T1F$Time.Point
scores.Pvar.samp.df$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar.samp.df$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar.samp.df$Clade <- Pvar.samp.df.T1F$Clade
scores.Pvar.samp.df$Code <- paste(Pvar.samp.df.T1F$Parent.ID,Pvar.samp.df.T1F$Time.Point)
scores.Pvar.samp.df$Code2 <- paste(Pvar.samp.df.T1F$Treatment,Pvar.samp.df.T1F$Time.Point)
scores.Pvar.samp.df$Code3 <- paste(Pvar.samp.df.T1F$Treatment,Pvar.samp.df.T1F$Time.Point,Pvar.samp.df.T1F$Parent.ID)

ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2), linetype = 2) +
  ggtitle("P. varians T1TF") +
  theme_classic()
#ggsave("PvarT1its2.pdf")


ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Pvar.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. var T1TF") +
  theme_classic()

ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Pvar.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. var T1TF") +
  theme_classic()

#T1
ord.Pvar.T1 <- ordinate(Pvar.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pvar.T1)

ord.Pvar.T1.df<-as.data.frame(cbind(vegan::scores(ord.Pvar.T1, display="sites"))) # these are "points" ask hannah 
ord.Pvar.T1.df$Time.Point <- Pvar.samp.df.T1$Time.Point
ord.Pvar.T1.df$Parent.ID <- Pvar.samp.df.T1$Parent.ID
ord.Pvar.T1.df$Treatment <- Pvar.samp.df.T1$Treatment
ord.Pvar.T1.df$Clade <- Pvar.samp.df.T1$Clade
ord.Pvar.T1.df$Code <- paste(Pvar.samp.df.T1$Parent.ID,Pvar.samp.df.T1$Time.Point)
ord.Pvar.T1.df$Code2 <- paste(Pvar.samp.df.T1$Clade,Pvar.samp.df.T1$Treatment)
ord.Pvar.T1.df$Code2 <- as.factor(ord.Pvar.T1.df$Code2)


ggplot(ord.Pvar.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. var T1") +
  theme_classic()
#ggsave("pvarT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(pvar.bac.TF, ordinate(pvar.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.pvarTF <- ordinate(Pvar.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvarTF)
ord.pvar.TF.df<-as.data.frame(cbind(vegan::scores(ord.pvarTF, display="sites"))) # these are "points" ask hannah 
ord.pvar.TF.df$Treatment <- Pvar.samp.df.TF$Treatment
ord.pvar.TF.df$Parent.ID <- Pvar.samp.df.TF$Parent.ID
ord.pvar.TF.df$Clade <- Pvar.samp.df.TF$Clade
ord.pvar.TF.df$CladeTreat <- as.factor(paste(Pvar.samp.df.TF$Clade,Pvar.samp.df.TF$Treatment))
ord.pvar.TF.df$Code <- as.factor(paste(Pvar.samp.df.TF$Parent.ID,Pvar.samp.df.TF$Treatment))
ord.pvar.TF.df$Code2 <- paste(Pvar.samp.df.TF$Clade,Pvar.samp.df.TF$Treatment)
ord.pvar.TF.df$Code2 <- as.factor(ord.pvar.TF.df$Code2)

ggplot(ord.pvar.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. var TF") +
  theme_classic()
#ggsave("pvarT1its2.pdf")



#### BY CLADE
#ords


#C T1 and TF 
set.seed(30)
ord.pvar_C <- ordinate(pvar.bac.T1F.C , "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_C)
scores.pvar_C<-as.data.frame(cbind(vegan::scores(ord.pvar_C, display="sites"))) # these are "points" ask hannah 
scores.pvar_C$Treatment <- pvar.samp.df.T1F.C$Treatment
scores.pvar_C$Parent.ID <- pvar.samp.df.T1F.C$Parent.ID
scores.pvar_C$Time.Point <- pvar.samp.df.T1F.C$Time.Point
scores.pvar_C$Code2<-as.factor(paste(scores.pvar_C$Treatment, scores.pvar_C$Time.Point))

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape =Code2 , color = Parent.ID, fill=Code2), size = 5,stroke = 1) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Mcap.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata C only T1F") +
  theme_classic()






#C T1
set.seed(30)
ord.mcap_C.T1 <- ordinate(mcap.bac.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.T1)
scores.mcap_C.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_C.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.T1$Treatment <- mcap.samp.df.T1.C$Treatment
scores.mcap_C.T1$Parent.ID <- mcap.samp.df.T1.C$Parent.ID
scores.mcap_C.T1$Time.Point <- mcap.samp.df.T1.C$Time.Point

ggplot(scores.pvar_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("Pvar T1") +
  theme_classic()
#ggsave("pvar_T1_its_C.pdf")

#C TF
set.seed(30)
ord.pvar_C.TF <- ordinate(pvar.bac.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_C.TF)
scores.pvar_C.TF<-as.data.frame(cbind(vegan::scores(ord.pvar_C.TF, display="sites"))) # these are "points" ask hannah 
scores.pvar_C.TF$Treatment <- pvar.samp.df.TF.C$Treatment
scores.pvar_C.TF$Parent.ID <- pvar.samp.df.TF.C$Parent.ID
scores.pvar_C.TF$Time.Point <- pvar.samp.df.TF.C$Time.Point

ggplot(scores.pvar_C.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("pvar_TF_its_TF_C.pdf")


#CD pvar.bacT1F.CD
set.seed(30)
ord.pvar_CD <- ordinate(pvar.bac.T1F.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_CD)
scores.pvar_CD<-as.data.frame(cbind(vegan::scores(ord.pvar_CD, display="sites"))) # these are "points" ask hannah 
scores.pvar_CD$Treatment <- pvar.samp.df.T1F.CD$Treatment
scores.pvar_CD$Parent.ID <- pvar.samp.df.T1F.CD$Parent.ID
scores.pvar_CD$Time.Point <- pvar.samp.df.T1F.CD$Time.Point

ggplot(scores.pvar_CD, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("pvar_ITS2_CD_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.pvar_CD.T1 <- ordinate(pvar.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_CD.T1)
scores.pvar_CD.T1<-as.data.frame(cbind(vegan::scores(ord.pvar_CD.T1, display="sites"))) # these are "points" ask hannah 
scores.pvar_CD.T1$Treatment <- pvar.samp.df.T1.CD$Treatment
scores.pvar_CD.T1$Parent.ID <- pvar.samp.df.T1.CD$Parent.ID
scores.pvar_CD.T1$Time.Point <- pvar.samp.df.T1.CD$Time.Point

ggplot(scores.pvar_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("pvar_ITS2_CD_T1.pdf")

#CD TF
set.seed(30)
ord.pvar_CD.TF <- ordinate(pvar.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_CD.TF)
scores.pvar_CD.TF<-as.data.frame(cbind(vegan::scores(ord.pvar_CD.TF, display="sites"))) # these are "points" ask hannah 
scores.pvar_CD.TF$Treatment <- pvar.samp.df.TF.CD$Treatment
scores.pvar_CD.TF$Parent.ID <- pvar.samp.df.TF.CD$Parent.ID
scores.pvar_CD.TF$Time.Point <- pvar.samp.df.TF.CD$Time.Point

ggplot(scores.pvar_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("pvar_ITS2_CD_TF.pdf")


```

Pocillopora - T1 and TF together, separet, and then by phylotype. PERMANOVA
CHANGE PHYLOTYPES

```{r}    
# df Bac_RelA_stats.T1F 
#Pacu
Pacu.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Pocillopora_acuta")
Pacu.bac.T1F = prune_taxa(taxa_sums(Pacu.bac.T1F) > 0, Pacu.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
Pacu.bac.T1 <- subset_samples(Pacu.bac.T1F, Time.Point == "T1")
  Pacu.bac.T1 = prune_taxa(taxa_sums(Pacu.bac.T1) > 0, Pacu.bac.T1) #this removes any OTU with 0s

Pacu.bac.TF <- subset_samples(Pacu.bac.T1F, Time.Point == "TF")
  Pacu.bac.TF = prune_taxa(taxa_sums(Pacu.bac.TF) > 0, Pacu.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F <- phyloseq::distance(Pacu.bac.T1F, method = "bray")
     Pacu.samp.df.T1F <- as(sample_data(Pacu.bac.T1F), "data.frame") #sample df
    adonis(bc.Pacu.T1F ~ Treatment*Time.Point, data = Pacu.samp.df.T1F) #adonis test no clade
    adonis(bc.Pacu.T1F ~ Clade*Treatment*Time.Point, data = Pacu.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(Pacu.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
  bc.PacuT1 <- phyloseq::distance(Pacu.bac.T1, method = "bray") 
      Pacu.samp.df.T1 <- as(sample_data(Pacu.bac.T1), "data.frame") 
    adonis(bc.PacuT1 ~ Clade*Treatment, data = Pacu.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Clade)  #clade
    permutest(beta.Pacu.T1, pairwise = T) 
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Treatment) #treatment
    permutest(beta.Pacu.T1, pairwise = T) 

#TF
  bc.Pacu.TF <- phyloseq::distance(Pacu.bac.TF, method = "bray")
     Pacu.samp.df.TF <- as(sample_data(Pacu.bac.TF), "data.frame") 
    adonis(bc.Pacu.TF ~ Clade*Treatment, data = Pacu.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Clade) 
    permutest(beta.Pacu.TF, pairwise = T) 
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Treatment) 
    permutest(beta.Pacu.TF, pairwise = T) 

######### by phylotype: CCC vs C42
# C ONLY
  Pacu.bac.T1F.CCC <- subset_samples(Pacu.bac.T1F, Clade == "CCC")
  Pacu.bac.T1.CCC <- subset_samples(Pacu.bac.T1, Clade == "CCC")
  Pacu.bac.TF.CCC <- subset_samples(Pacu.bac.TF, Clade == "CCC")
  
#T1TF.CCC
  bc.Pacu.T1F.CCC <- phyloseq::distance(Pacu.bac.T1F.CCC, method = "bray")
    Pacu.samp.df.T1F.CCC <- as(sample_data(Pacu.bac.T1F.CCC), "data.frame") 
  adonis(bc.Pacu.T1F.CCC ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.CCC)

#T1.CCC
  bc.PacuT1.CCC <- phyloseq::distance(Pacu.bac.T1.CCC, method = "bray")
    Pacu.samp.df.T1.CCC <- as(sample_data(Pacu.bac.T1.CCC), "data.frame") #sample df
  adonis(bc.PacuT1.CCC ~ Treatment, data = Pacu.samp.df.T1.CCC)
  beta.Pacu.T1.CCC <- betadisper(bc.PacuT1.CCC, Pacu.samp.df.T1.CCC$Treatment)
    permutest(beta.Pacu.T1.CCC, pairwise = T) 

#TF.CCC
  bc.Pacu.TF.CCC <- phyloseq::distance(Pacu.bac.TF.CCC, method = "bray")
    Pacu.samp.df.TF.CCC <- as(sample_data(Pacu.bac.TF.CCC), "data.frame") #sample df
  adonis(bc.Pacu.TF.CCC ~ Treatment, data = Pacu.samp.df.TF.CCC)
  beta.Pacu.TF.CCC <- betadisper(bc.Pacu.TF.CCC, Pacu.samp.df.TF.CCC$Treatment)
    permutest(beta.Pacu.TF.CCC, pairwise = T) 

# C42 ONLY
Pacu.bac.T1F.C42 <- subset_samples(Pacu.bac.T1F, Clade == "C42")
Pacu.bac.T1.C42 <- subset_samples(Pacu.bac.T1, Clade == "C42")
Pacu.bac.TF.C42 <- subset_samples(Pacu.bac.TF, Clade == "C42")
  
#T1TF.C42
  bc.Pacu.T1F.C42 <- phyloseq::distance(Pacu.bac.T1F.C42, method = "bray")
  Pacu.samp.df.T1F.C42 <- as(sample_data(Pacu.bac.T1F.C42), "data.frame") 
  adonis(bc.Pacu.T1F.C42 ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.C42)

#T1.C42
  bc.PacuT1.C42 <- phyloseq::distance(Pacu.bac.T1.C42, method = "bray")
    Pacu.samp.df.T1.C42 <- as(sample_data(Pacu.bac.T1.C42), "data.frame") 
  adonis(bc.PacuT1.C42 ~ Treatment, data = Pacu.samp.df.T1.C42)
    beta.Pacu.T1.C42 <- betadisper(bc.PacuT1.C42, Pacu.samp.df.T1.C42$Treatment)
  permutest(beta.Pacu.T1.C42, pairwise = T) 

#TF.C42
  bc.Pacu.TF.C42 <- phyloseq::distance(Pacu.bac.TF.C42, method = "bray")
    Pacu.samp.df.TF.C42 <- as(sample_data(Pacu.bac.TF.C42), "data.frame") 
  adonis(bc.Pacu.TF.C42 ~ Treatment, data = Pacu.samp.df.TF.C42)
  beta.Pacu.TF.C42 <- betadisper(bc.Pacu.TF.C42, Pacu.samp.df.TF.C42$Treatment)
    permutest(beta.Pacu.TF.C42, pairwise = T) 
``` 
Pacu NMDS
```{r}
## create palette for parent.ID and treatments
Pacu.Parent.colors <- c("305" = "green4",  "307" = "firebrick3", "309" ="deeppink4", "311" = "orange", "301" = "green1","302" = "gold",  "308"= "hotpink", "310" ="darkslateblue", "312" = "cyan", "303" = "darkorange","304" ="darkorchid1", "306" = "lightblue", 
                        "Ambient" = "dodgerblue", "High" = "darkred",
                        "C42 Ambient"="#DDCC77","C42 High"="#999933", "CCC Ambient"="#44AA99", "CCC High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 



#T1 and TF
ord.Pacu2 <- ordinate(Pacu.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pacu2)

scores.Pacu.samp.df<-as.data.frame(cbind(vegan::scores(ord.Pacu2, display="sites"))) # these are "points" ask hannah 
scores.Pacu.samp.df$Time.Point <- Pacu.samp.df.T1F$Time.Point
scores.Pacu.samp.df$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu.samp.df$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu.samp.df$Clade <- Pacu.samp.df.T1F$Clade
scores.Pacu.samp.df$Code <- paste(Pacu.samp.df.T1F$Parent.ID,Pacu.samp.df.T1F$Time.Point)
scores.Pacu.samp.df$Code2 <- paste(Pacu.samp.df.T1F$Treatment,Pacu.samp.df.T1F$Time.Point)
scores.Pacu.samp.df$Code3 <- paste(Pacu.samp.df.T1F$Treatment,Pacu.samp.df.T1F$Time.Point,Pacu.samp.df.T1F$Parent.ID)

ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Pacu.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. acuta T1TF") +
  theme_classic()

ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. acuta T1TF") +
  theme_classic()
#ggsave("PacuT1its2.pdf")

#T1
ord.Pacu.T1 <- ordinate(Pacu.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pacu.T1)

ord.Pacu.T1.df<-as.data.frame(cbind(vegan::scores(ord.Pacu.T1, display="sites"))) # these are "points" ask hannah 
ord.Pacu.T1.df$Time.Point <- Pacu.samp.df.T1$Time.Point
ord.Pacu.T1.df$Parent.ID <- Pacu.samp.df.T1$Parent.ID
ord.Pacu.T1.df$Treatment <- Pacu.samp.df.T1$Treatment
ord.Pacu.T1.df$Clade <- Pacu.samp.df.T1$Clade
ord.Pacu.T1.df$Code <- paste(Pacu.samp.df.T1$Parent.ID,Pacu.samp.df.T1$Time.Point)
ord.Pacu.T1.df$Code2 <- paste(Pacu.samp.df.T1$Clade,Pacu.samp.df.T1$Treatment)
ord.Pacu.T1.df$Code2 <- as.factor(ord.Pacu.T1.df$Code2)


ggplot(ord.Pacu.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta T1") +
  theme_classic()
#ggsave("PacuT1its2.pdf")



#TF only
set.seed(30)
ord.PacuTF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuTF)
ord.Pacu.TF.df<-as.data.frame(cbind(vegan::scores(ord.PacuTF, display="sites"))) # these are "points" ask hannah 
ord.Pacu.TF.df$Treatment <- Pacu.samp.df.TF$Treatment
ord.Pacu.TF.df$Parent.ID <- Pacu.samp.df.TF$Parent.ID
ord.Pacu.TF.df$Clade <- Pacu.samp.df.TF$Clade
ord.Pacu.TF.df$CladeTreat <- as.factor(paste(Pacu.samp.df.TF$Clade,Pacu.samp.df.TF$Treatment))
ord.Pacu.TF.df$Code <- as.factor(paste(Pacu.samp.df.TF$Parent.ID,Pacu.samp.df.TF$Treatment))
ord.Pacu.TF.df$Code2 <- paste(Pacu.samp.df.TF$Clade,Pacu.samp.df.TF$Treatment)
ord.Pacu.TF.df$Code2 <- as.factor(ord.Pacu.TF.df$Code2)

ggplot(ord.Pacu.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta TF") +
  theme_classic()
#ggsave("PacuT1its2.pdf")



#### BY CLADE
#ords


#CCC T1 and TF 
set.seed(30)
ord.Pacu_CCC <- ordinate(Pacu.bac.T1F.CCC , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC)
scores.Pacu_CCC<-as.data.frame(cbind(vegan::scores(ord.Pacu_CCC, display="sites"))) # these are "points" ask hannah 
scores.Pacu_CCC$Treatment <- Pacu.samp.df.T1F.CCC$Treatment
scores.Pacu_CCC$Parent.ID <- Pacu.samp.df.T1F.CCC$Parent.ID
scores.Pacu_CCC$Time.Point <- Pacu.samp.df.T1F.CCC$Time.Point
scores.Pacu_CCC$Code2<-as.factor(paste(scores.Pacu_CCC$Treatment, scores.Pacu_CCC$Time.Point))


ggplot(scores.Pacu_CCC, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. acuta C only T1TF") +
  theme_classic()

ggplot(scores.Pacu_CCC, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta C only T1F") +
  theme_classic()


#C T1
set.seed(30)
ord.Pacu_CCC.T1 <- ordinate(Pacu.bac.T1.CCC, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.T1)
scores.Pacu_CCC.T1<-as.data.frame(cbind(vegan::scores(ord.Pacu_CCC.T1, display="sites"))) # these are "points" ask hannah 
scores.Pacu_CCC.T1$Treatment <- Pacu.samp.df.T1.CCC$Treatment
scores.Pacu_CCC.T1$Parent.ID <- Pacu.samp.df.T1.CCC$Parent.ID
scores.Pacu_CCC.T1$Time.Point <- Pacu.samp.df.T1.CCC$Time.Point

ggplot(scores.Pacu_CCC.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. acuta T1 CCC") +
  theme_classic()
#ggsave("Pacu_T1_its_CCC.pdf")

#C TF
set.seed(30)
ord.Pacu_CCC.TF <- ordinate(Pacu.bac.TF.CCC, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.TF)
scores.Pacu_CCC.TF<-as.data.frame(cbind(vegan::scores(ord.Pacu_CCC.TF, display="sites"))) # these are "points" ask hannah 
scores.Pacu_CCC.TF$Treatment <- Pacu.samp.df.TF.CCC$Treatment
scores.Pacu_CCC.TF$Parent.ID <- Pacu.samp.df.TF.CCC$Parent.ID
scores.Pacu_CCC.TF$Time.Point <- Pacu.samp.df.TF.CCC$Time.Point

ggplot(scores.Pacu_C42, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta 42  T1F") +
  theme_classic()
#ggsave("Pacu_TF_its_TF_CCC.pdf")


#C42 Pacu.bacT1F C42
set.seed(30)
ord.Pacu_C42 <- ordinate(Pacu.bac.T1F.C42, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C42)
scores.Pacu_C42<-as.data.frame(cbind(vegan::scores(ord.Pacu_C42, display="sites"))) # these are "points" ask hannah 
scores.Pacu_C42$Treatment <- Pacu.samp.df.T1F.C42$Treatment
scores.Pacu_C42$Parent.ID <- Pacu.samp.df.T1F.C42$Parent.ID
scores.Pacu_C42$Time.Point <- Pacu.samp.df.T1F.C42$Time.Point
scores.Pacu_C42$Time.Point <- Pacu.samp.df.T1F.C42$Time.Point
scores.Pacu_C42$Code2<-as.factor(paste(scores.Pacu_C42$Treatment, scores.Pacu_C42$Time.Point))

ggplot(scores.Pacu_C42, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta CC42 T1F") +
  theme_classic()

#CD T1
set.seed(30)
ord.Pacu_C42.T1 <- ordinate(Pacu.bac.T1.C42 , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C42.T1)
scores.Pacu_C42.T1<-as.data.frame(cbind(vegan::scores(ord.Pacu_C42.T1, display="sites"))) # these are "points" ask hannah 
scores.Pacu_C42.T1$Treatment <- Pacu.samp.df.T1.C42$Treatment
scores.Pacu_C42.T1$Parent.ID <- Pacu.samp.df.T1.C42$Parent.ID
scores.Pacu_C42.T1$Time.Point <- Pacu.samp.df.T1.C42$Time.Point

ggplot(scores.Pacu_C42.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. acuta T1 C42") +
  theme_classic()
#ggsave("Pacu_ITS2_C42_T1.pdf")

#CD TF
set.seed(30)
ord.Pacu_C42.TF <- ordinate(Pacu.bac.TF.C42, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C42.TF)
scores.Pacu_C42.TF<-as.data.frame(cbind(vegan::scores(ord.Pacu_C42.TF, display="sites"))) # these are "points" ask hannah 
scores.Pacu_C42.TF$Treatment <- Pacu.samp.df.TF.C42$Treatment
scores.Pacu_C42.TF$Parent.ID <- Pacu.samp.df.TF.C42$Parent.ID
scores.Pacu_C42.TF$Time.Point <- Pacu.samp.df.TF.C42$Time.Point

ggplot(scores.Pacu_C42.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. acuta CD TF") +
  theme_classic()
ggsave("Pacu_ITS2_C42_TF.pdf")




```









##### DELETE BELOW once redon above
Porites - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pcomp
Pcomp.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "porites")
Pcomp.bac.T1 <- subset_samples(Pcomp.bac.T1F, Time.Point == "T1")
Pcomp.bac.TF <- subset_samples(Pcomp.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F <- phyloseq::distance(Pcomp.bac.T1F, method = "bray")
  Pcomp.samp.df.T1F <- as(sample_data(Pcomp.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pcomp.T1F ~ Clade*Treatment*Time.Point, data = Pcomp.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PcompT1 <- phyloseq::distance(Pcomp.bac.T1, method = "bray")
  Pcomp.samp.df.T1 <- as(sample_data(Pcomp.bac.T1), "data.frame") #sample df
  adonis(bc.PcompT1 ~ Clade*Treatment, data = Pcomp.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Clade)
  permutest(beta.Pcomp.T1, pairwise = T) 
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Treatment)
  permutest(beta.Pcomp.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pcomp.TF <- phyloseq::distance(Pcomp.bac.TF, method = "bray")
  Pcomp.samp.df.TF <- as(sample_data(Pcomp.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF ~ Clade*Treatment, data = Pcomp.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Clade)
  permutest(beta.Pcomp.TF, pairwise = T) 
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Treatment)
  permutest(beta.Pcomp.TF, pairwise = T) 


## by clade: C vs CD
  
# C15cj only, new objs
Pcomp.bacT1F.C15cj <- subset_samples(Pcomp.bac.T1F, Clade == "C15cj")
Pcomp.bac.T1.C15cj <- subset_samples(Pcomp.bac.T1, Clade == "C15cj")
Pcomp.bac.TF.C15cj <- subset_samples(Pcomp.bac.TF, Clade == "C15cj")
  
#T1TF.C
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F.C15cj <- phyloseq::distance(Pcomp.bacT1F.C15cj, method = "bray")
  Pcomp.samp.df.T1F.C15cj <- as(sample_data(Pcomp.bacT1F.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.T1F.C15cj ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C15cj)

#T1.C15cj
#make bray curtis data matrix 
  bc.PcompT1.C15cj <- phyloseq::distance(Pcomp.bac.T1.C15cj, method = "bray")
  Pcomp.samp.df.T1.C15cj <- as(sample_data(Pcomp.bac.T1.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.PcompT1.C15cj ~ Treatment, data = Pcomp.samp.df.T1.C15cj)
# Homogeneity of dispersion test  
  beta.Pcomp.T1.C15cj <- betadisper(bc.PcompT1.C15cj, Pcomp.samp.df.T1.C15cj$Treatment)
  permutest(beta.Pcomp.T1.C15cj, pairwise = T) 

#TF.C15cj
#make bray curtis data matrix 
  bc.Pcomp.TF.C15cj <- phyloseq::distance(Pcomp.bac.TF.C15cj, method = "bray")
  Pcomp.samp.df.TF.C15cj <- as(sample_data(Pcomp.bac.TF.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF.C15cj ~ Treatment, data = Pcomp.samp.df.TF.C15cj)
# Homogeneity of dispersion test  
  beta.Pcomp.TF.C15cj <- betadisper(bc.Pcomp.TF.C15cj, Pcomp.samp.df.TF.C15cj$Treatment)
  permutest(beta.Pcomp.TF.C15cj, pairwise = T) 


# C15n only, new objs
Pcomp.bacT1F.C15n <- subset_samples(Pcomp.bac.T1F, Clade == "C15n")
Pcomp.bac.T1.C15n <- subset_samples(Pcomp.bac.T1, Clade == "C15n")
Pcomp.bac.TF.C15n <- subset_samples(Pcomp.bac.TF, Clade == "C15n")
  
#T1TF.C15n
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F.C15n <- phyloseq::distance(Pcomp.bacT1F.C15n, method = "bray")
  Pcomp.samp.df.T1F.C15n <- as(sample_data(Pcomp.bacT1F.C15n), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.T1F.C15n ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C15n)

#T1.C15n
#make bray curtis data matrix 
  bc.PcompT1.C15n <- phyloseq::distance(Pcomp.bac.T1.C15n, method = "bray")
  Pcomp.samp.df.T1.C15n <- as(sample_data(Pcomp.bac.T1.C15n), "data.frame") #sample df
  adonis(bc.PcompT1.C15n ~ Treatment, data = Pcomp.samp.df.T1.C15n)#adonis test
# Homogeneity of dispersion test  
  beta.Pcomp.T1.C15n <- betadisper(bc.PcompT1.C15n, Pcomp.samp.df.T1.C15n$Treatment)
  permutest(beta.Pcomp.T1.C15n, pairwise = T) 

#TF.C15n
#make bray curtis data matrix 
  bc.Pcomp.TF.C15n <- phyloseq::distance(Pcomp.bac.TF.C15n, method = "bray")
  Pcomp.samp.df.TF.C15n <- as(sample_data(Pcomp.bac.TF.C15n), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF.C15n ~ Treatment, data = Pcomp.samp.df.TF.C15n)
# Homogeneity of dispersion test  
  beta.Pcomp.TF.C15n <- betadisper(bc.Pcomp.TF.C15n, Pcomp.samp.df.TF.C15n$Treatment)
  permutest(beta.Pcomp.TF.C15n, pairwise = T) 
``` 
Pcomp NMDS
```{r}
library(RColorBrewer)
library(colorRamps)
library(devtools)
 
## create palette for parent.ID and treatments
Pcomp.Parent.colors <- c("313" = "green4",  "314" = "firebrick3", "315" ="deeppink4", "316" = "orange", "317" = "green1","318" = "gold",  "319"= "hotpink", "320" ="darkslateblue", "321" = "cyan", "322" = "darkorange","323" ="darkorchid1", "324" = "lightblue", "Ambient" = "Blue", "High" = "Red","C15cj Ambient"="blue","C15cj High"="red", "C15n Ambient"="darkblue", "C15n High"="darkred") 

#T1 and TF
ord.Pcomp <- ordinate(Pcomp.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pcomp)

scores.Pcomp.samp.df <- as.data.frame(scores(ord.Pcomp) ) 
scores.Pcomp.samp.df$Time.Point <- Pcomp.samp.df.T1F$Time.Point
scores.Pcomp.samp.df$Parent.ID <- Pcomp.samp.df.T1F$Parent.ID
scores.Pcomp.samp.df$Treatment <- Pcomp.samp.df.T1F$Treatment
scores.Pcomp.samp.df$Clade <- Pcomp.samp.df.T1F$Clade
scores.Pcomp.samp.df$Code <- paste(Pcomp.samp.df.T1F$Parent.ID, Pcomp.samp.df.T1F$Time.Point)


ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P. comp Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_T1TF_its2.pdf")

ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PcompT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pcomp.bac.TF, ordinate(Pcomp.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PcompT1 <- ordinate(Pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompT1)
scores.PcompT1 <- as.data.frame(scores(ord.PcompT1))
scores.PcompT1$Treatment <- Pcomp.samp.df.T1$Treatment
scores.PcompT1$Parent.ID <- Pcomp.samp.df.T1$Parent.ID
scores.PcompT1$Clade <- Pcomp.samp.df.T1$Clade
scores.PcompT1$CladeTreat <- as.factor(paste(Pcomp.samp.df.T1$Clade,Pcomp.samp.df.T1$Treatment))


ggplot(scores.PcompT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. comp T1 Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PcompTF <- ordinate(Pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompTF)
scores.PcompTF <- as.data.frame(scores(ord.PcompTF))
scores.PcompTF$Treatment <- Pcomp.samp.df.TF$Treatment
scores.PcompTF$Parent.ID <- Pcomp.samp.df.TF$Parent.ID
scores.PcompTF$Clade <- Pcomp.samp.df.TF$Clade
scores.PcompTF$CladeTreat <- as.factor(paste(Pcomp.samp.df.TF$Clade,Pcomp.samp.df.TF$Treatment))

ggplot(scores.PcompTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. comp TF Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_TF_its_nmds.pdf")



 Pcomp.bac.T1F
Pcomp.bac.T1
Pcomp.bac.TF. 

#### BY CLADE
#ords
#C T1 and TF
set.seed(30)
ord.pcomp_C15cj  <- ordinate(Pcomp.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj)
scores.pcomp_C15cj <- as.data.frame(scores(ord.pcomp_C15cj))
scores.pcomp_C15cj$Treatment <- Pcomp.samp.df.T1F$Treatment
scores.pcomp_C15cj$Parent.ID <- Pcomp.samp.df.T1F$Parent.ID
scores.pcomp_C15cj$Time.Point <- Pcomp.samp.df.T1F$Time.Point

ggplot(scores.pcomp_C15cj, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P comp C15cj") +
  theme_classic()
#ggsave("pcompTFits_C15cj.pdf")


#C T1
set.seed(30)
ord.pcomp_C15cj.T1 <- ordinate(Pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj.T1)
scores.pcomp_C15cj.T1 <- as.data.frame(scores(ord.pcomp_C15cj.T1))
scores.pcomp_C15cj.T1$Treatment <- pcomp.samp.df.T1$Treatment
scores.pcomp_C15cj.T1$Parent.ID <- pcomp.samp.df.T1$Parent.ID
scores.pcomp_C15cj.T1$Time.Point <- pcomp.samp.df.T1$Time.Point

ggplot(scores.pcomp_C15cj.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp T1 C15cj") +
  theme_classic()
#ggsave("pcomp_T1_its_C15cj.pdf")

#C TF
set.seed(30)
ord.pcomp_C15cj.TF <- ordinate(pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj.TF)
scores.pcomp_C15cj.TF <- as.data.frame(scores(ord.pcomp_C15cj.TF))
scores.pcomp_C15cj.TF$Treatment <- pcomp.samp.df.TF$Treatment
scores.pcomp_C15cj.TF$Parent.ID <- pcomp.samp.df.TF$Parent.ID
scores.pcomp_C15cj.TF$Time.Point <- pcomp.samp.df.TF$Time.Point

ggplot(scores.pcomp_C15cj.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("pcomp_TF_its_TF_C15cj.pdf")


#C15n
set.seed(30)
ord.pcomp_C15n <- ordinate(pcomp_DIV_C15n, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n)
scores.pcomp_C15n <- as.data.frame(scores(ord.pcomp_C15n))
scores.pcomp_C15n$Treatment <- pcomp.samp.df.T1F.CD$Treatment
scores.pcomp_C15n$Parent.ID <- pcomp.samp.df.T1F.CD$Parent.ID
scores.pcomp_C15n$Time.Point <- pcomp.samp.df.T1F.CD$Time.Point

ggplot(scores.pcomp_C15n, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_its_T1TF.pdf")

#C15n T1
set.seed(30)
ord.pcomp_C15n.T1 <- ordinate(pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n.T1)
scores.pcomp_C15n.T1 <- as.data.frame(scores(ord.pcomp_C15n.T1))
scores.pcomp_C15n.T1$Treatment <- pcomp.samp.df.T1$Treatment
scores.pcomp_C15n.T1$Parent.ID <- pcomp.samp.df.T1$Parent.ID
scores.pcomp_C15n.T1$Time.Point <- pcomp.samp.df.T1$Time.Point

ggplot(scores.pcomp_C15n.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P comp T1 C15n") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_T1.pdf")

#CD TF
set.seed(30)
ord.pcomp_C15n.TF <- ordinate(pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n.TF)
scores.pcomp_C15n.TF <- as.data.frame(scores(ord.pcomp_C15n.TF))
scores.pcomp_C15n.TF$Treatment <- pcomp.samp.df.TF$Treatment
scores.pcomp_C15n.TF$Parent.ID <- pcomp.samp.df.TF$Parent.ID
scores.pcomp_C15n.TF$Time.Point <- pcomp.samp.df.TF$Time.Point

ggplot(scores.pcomp_C15n.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp C15n TF") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_TF.pdf")


```
Pavona - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pvar
Pvar.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pavona")
Pvar.bac.T1 <- subset_samples(Pvar.bac.T1F, Time.Point == "T1")
Pvar.bac.TF <- subset_samples(Pvar.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F <- phyloseq::distance(Pvar.bac.T1F, method = "bray")
  Pvar.samp.df.T1F <- as(sample_data(Pvar.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pvar.T1F ~ Clade*Treatment*Time.Point, data = Pvar.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PvarT1 <- phyloseq::distance(Pvar.bac.T1, method = "bray")
  Pvar.samp.df.T1 <- as(sample_data(Pvar.bac.T1), "data.frame") #sample df
  adonis(bc.PvarT1 ~ Clade*Treatment, data = Pvar.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Clade)
  permutest(beta.Pvar.T1, pairwise = T) 
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Treatment)
  permutest(beta.Pvar.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pvar.TF <- phyloseq::distance(Pvar.bac.TF, method = "bray")
  Pvar.samp.df.TF <- as(sample_data(Pvar.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF ~ Clade*Treatment, data = Pvar.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Clade)
  permutest(beta.Pvar.TF, pairwise = T) 
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Treatment)
  permutest(beta.Pvar.TF, pairwise = T) 


## by clade: C1 vs C27
  
# C1 only, new objs
Pvar.bacT1F.C1 <- subset_samples(Pvar.bac.T1F, Clade == "C1")
Pvar.bac.T1.C1 <- subset_samples(Pvar.bac.T1, Clade == "C1")
Pvar.bac.TF.C1 <- subset_samples(Pvar.bac.TF, Clade == "C1")
  
#T1TF.C1
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F.C1 <- phyloseq::distance(Pvar.bacT1F.C1, method = "bray")
  Pvar.samp.df.T1F.C1 <- as(sample_data(Pvar.bacT1F.C1), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.T1F.C1 ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C1)

#T1.C1
#make bray curtis data matrix 
  bc.PvarT1.C1 <- phyloseq::distance(Pvar.bac.T1.C1, method = "bray")
  Pvar.samp.df.T1.C1 <- as(sample_data(Pvar.bac.T1.C1), "data.frame") #sample df
#adonis test
  adonis(bc.PvarT1.C1 ~ Treatment, data = Pvar.samp.df.T1.C1)
# Homogeneity of dispersion test  
  beta.Pvar.T1.C1 <- betadisper(bc.PvarT1.C1, Pvar.samp.df.T1.C1$Treatment)
  permutest(beta.Pvar.T1.C1, pairwise = T) 

#TF.C1
#make bray curtis data matrix 
  bc.Pvar.TF.C1 <- phyloseq::distance(Pvar.bac.TF.C1, method = "bray")
  Pvar.samp.df.TF.C1 <- as(sample_data(Pvar.bac.TF.C1), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF.C1 ~ Treatment, data = Pvar.samp.df.TF.C1)
# Homogeneity of dispersion test  
  beta.Pvar.TF.C1 <- betadisper(bc.Pvar.TF.C1, Pvar.samp.df.TF.C1$Treatment)
  permutest(beta.Pvar.TF.C1, pairwise = T) 


# C27 only, new objs
Pvar.bacT1F.C27 <- subset_samples(Pvar.bac.T1F, Clade == "C27")
Pvar.bac.T1.C27 <- subset_samples(Pvar.bac.T1, Clade == "C27")
Pvar.bac.TF.C27 <- subset_samples(Pvar.bac.TF, Clade == "C27")
  
#T1TF.C27
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F.C27 <- phyloseq::distance(Pvar.bacT1F.C27, method = "bray")
  Pvar.samp.df.T1F.C27 <- as(sample_data(Pvar.bacT1F.C27), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.T1F.C27 ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C27)

#T1.C27
#make bray curtis data matrix 
  bc.PvarT1.C27 <- phyloseq::distance(Pvar.bac.T1.C27, method = "bray")
  Pvar.samp.df.T1.C27 <- as(sample_data(Pvar.bac.T1.C27), "data.frame") #sample df
  adonis(bc.PvarT1.C27 ~ Treatment, data = Pvar.samp.df.T1.C27)#adonis test
# Homogeneity of dispersion test  
  beta.Pvar.T1.C27 <- betadisper(bc.PvarT1.C27, Pvar.samp.df.T1.C27$Treatment)
  permutest(beta.Pvar.T1.C27, pairwise = T) 

#TF.C27
#make bray curtis data matrix 
  bc.Pvar.TF.C27 <- phyloseq::distance(Pvar.bac.TF.C27, method = "bray")
  Pvar.samp.df.TF.C27 <- as(sample_data(Pvar.bac.TF.C27), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF.C27 ~ Treatment, data = Pvar.samp.df.TF.C27)
# Homogeneity of dispersion test  
  beta.Pvar.TF.C27 <- betadisper(bc.Pvar.TF.C27, Pvar.samp.df.TF.C27$Treatment)
  permutest(beta.Pvar.TF.C27, pairwise = T) 
``` 
Pvar NMDS
```{r}
#dfs
# C1 only, new objs
# Pvar.bacT1F.C1 <- subset_samples(Pvar.bac.T1F, Clade == "C1")
# Pvar.bac.T1.C1 <- subset_samples(Pvar.bac.T1, Clade == "C1")
# Pvar.bac.TF.C1 <- subset_samples(Pvar.bac.TF, Clade == "C1")


library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pvar.Parent.colors <- c("325" = "green4",  "351" = "firebrick3", "353" ="deeppink4", "354" = "orange", "355" = "green1","356" = "gold",  "357"= "hotpink", "358" ="darkslateblue", "359" = "honeydew4", "360" = "khaki4","361" ="ivory4", "362" = "lightblue", Ambient = "Blue", High = "Red","C1 Ambient"="blue","C1 High"="red", "C27 Ambient"="darkblue", "C27 High"="darkred") 

#T1 and TF
ord.Pvar <- ordinate(Pvar.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pvar)

scores.Pvar.samp.df <- as.data.frame(scores(ord.Pvar) ) 
scores.Pvar.samp.df$Time.Point <- Pvar.samp.df.T1F$Time.Point
scores.Pvar.samp.df$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar.samp.df$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar.samp.df$Clade <- Pvar.samp.df.T1F$Clade
scores.Pvar.samp.df$Code <- paste(Pvar.samp.df.T1F$Parent.ID, Pvar.samp.df.T1F$Time.Point)


ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P.var Bray Curtis") +
  theme_classic()
#ggsave("Pvar_T1TF_its2.pdf")

ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var Bray Curtis") +
  theme_classic()
#ggsave("PvarT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pvar.bac.TF, ordinate(Pvar.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PvarT1 <- ordinate(Pvar.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PvarT1)
scores.PvarT1 <- as.data.frame(scores(ord.PvarT1))
scores.PvarT1$Treatment <- Pvar.samp.df.T1$Treatment
scores.PvarT1$Parent.ID <- Pvar.samp.df.T1$Parent.ID
scores.PvarT1$Clade <- Pvar.samp.df.T1$Clade
scores.PvarT1$CladeTreat <- as.factor(paste(Pvar.samp.df.T1$Clade,Pvar.samp.df.T1$Treatment))


ggplot(scores.PvarT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var T1 Bray Curtis") +
  theme_classic()
#ggsave("Pvar_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PvarTF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PvarTF)
scores.PvarTF <- as.data.frame(scores(ord.PvarTF))
scores.PvarTF$Treatment <- Pvar.samp.df.TF$Treatment
scores.PvarTF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.PvarTF$Clade <- Pvar.samp.df.TF$Clade
scores.PvarTF$CladeTreat <- as.factor(paste(Pvar.samp.df.TF$Clade,Pvar.samp.df.TF$Treatment))

ggplot(scores.PvarTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. var TF Bray Curtis") +
  theme_classic()
#gsave("Pvar_TF_its_nmds.pdf")



# C1 only, new objs
#Pvar.bacT1F.C1 , Pvar.samp.df.T1F.C1
#Pvar.bac.T1.C1 , Pvar.samp.df.T1.C1
#Pvar.bac.TF.C1 ,Pvar.samp.df.TF.C1

#### BY CLADE

#CCC T1 and TF
set.seed(30)
ord.Pvar_CCC  <- ordinate(Pvar.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC)
scores.Pvar_CCC <- as.data.frame(scores(ord.Pvar_CCC))
scores.Pvar_CCC$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar_CCC$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar_CCC$Time.Point <- Pvar.samp.df.T1F$Time.Point

ggplot(scores.Pvar_CCC, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var CCC") +
  theme_classic()
#ggsave("PvarTFits_CCC.pdf")


#C T1
set.seed(30)
ord.Pvar_CCC.T1 <- ordinate(Pvar.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC.T1)
scores.Pvar_CCC.T1 <- as.data.frame(scores(ord.Pvar_CCC.T1))
scores.Pvar_CCC.T1$Treatment <- Pvar.samp.df.T1$Treatment
scores.Pvar_CCC.T1$Parent.ID <- Pvar.samp.df.T1$Parent.ID
scores.Pvar_CCC.T1$Time.Point <- Pvar.samp.df.T1$Time.Point

ggplot(scores.Pvar_CCC.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var T1 CCC") +
  theme_classic()
#ggsave("Pvar_T1_its_CCC.pdf")

#C TF
set.seed(30)
ord.Pvar_CCC.TF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC.TF)
scores.Pvar_CCC.TF <- as.data.frame(scores(ord.Pvar_CCC.TF))
scores.Pvar_CCC.TF$Treatment <- Pvar.samp.df.TF$Treatment
scores.Pvar_CCC.TF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.Pvar_CCC.TF$Time.Point <- Pvar.samp.df.TF$Time.Point

ggplot(scores.Pvar_CCC.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var CCC TF") +
  theme_classic()
#ggsave("Pvar_TF_its_TF_CCC.pdf")





#C27
set.seed(30)
ord.Pvar_C27 <- ordinate(Pvar.bacT1F.C27, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27)
scores.Pvar_C27 <- as.data.frame(scores(ord.Pvar_C27))
scores.Pvar_C27$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar_C27$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar_C27$Time.Point <- Pvar.samp.df.T1F$Time.Point

ggplot(scores.Pvar_C27, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C&D") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_its_T1TF.pdf")

#C27 T1
set.seed(30)
ord.Pvar_C27.T1 <- ordinate(Pvar.bac.T1.C27 , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27.T1)
scores.Pvar_C27.T1 <- as.data.frame(scores(ord.Pvar_C27.T1))
scores.Pvar_C27.T1$Treatment <- Pvar.samp.df.T1.C27$Treatment
scores.Pvar_C27.T1$Parent.ID <- Pvar.samp.df.T1.C27$Parent.ID
scores.Pvar_C27.T1$Time.Point <- Pvar.samp.df.T1.C27$Time.Point

ggplot(scores.Pvar_C27.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var T1 C27") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_T1.pdf")

#CD TF
set.seed(30)
ord.Pvar_C27.TF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27.TF)
scores.Pvar_C27.TF <- as.data.frame(scores(ord.Pvar_C27.TF))
scores.Pvar_C27.TF$Treatment <- Pvar.samp.df.TF$Treatment
scores.Pvar_C27.TF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.Pvar_C27.TF$Time.Point <- Pvar.samp.df.TF$Time.Point

ggplot(scores.Pvar_C27.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C27 TF") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_TF.pdf")



```

Pocillopora - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pacu
Pacu.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pocillopora")
Pacu.bac.T1 <- subset_samples(Pacu.bac.T1F, Time.Point == "T1")
Pacu.bac.TF <- subset_samples(Pacu.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F <- phyloseq::distance(Pacu.bac.T1F, method = "bray")
  Pacu.samp.df.T1F <- as(sample_data(Pacu.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pacu.T1F ~ Treatment*Time.Point, data = Pacu.samp.df.T1F)      #adonis test
#write.csv(Pacu.samp.df.T1F, "Pacu_its2_samp_df.csv")
#T1
#make bray curtis data matrix 
  bc.PacuT1 <- phyloseq::distance(Pacu.bac.T1, method = "bray")
  Pacu.samp.df.T1 <- as(sample_data(Pacu.bac.T1), "data.frame") #sample df
  adonis(bc.PacuT1 ~ Clade*Treatment, data = Pacu.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Clade)
  permutest(beta.Pacu.T1, pairwise = T) 
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Treatment)
  permutest(beta.Pacu.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pacu.TF <- phyloseq::distance(Pacu.bac.TF, method = "bray")
  Pacu.samp.df.TF <- as(sample_data(Pacu.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF ~ Clade*Treatment, data = Pacu.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Clade)
  permutest(beta.Pacu.TF, pairwise = T) 
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Treatment)
  permutest(beta.Pacu.TF, pairwise = T) 


## by clade: CCC vs C42
  
# C only, new objs
Pacu.bacT1F.CCC <- subset_samples(Pacu.bac.T1F, Clade == "CCC")
Pacu.bac.T1.CCC <- subset_samples(Pacu.bac.T1, Clade == "CCC")
Pacu.bac.TF.CCC <- subset_samples(Pacu.bac.TF, Clade == "CCC")
  
#T1TF.CCC
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F.CCC <- phyloseq::distance(Pacu.bacT1F.CCC, method = "bray")
  Pacu.samp.df.T1F.CCC <- as(sample_data(Pacu.bacT1F.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.T1F.CCC ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.CCC)

#T1.CCC
#make bray curtis data matrix 
  bc.PacuT1.CCC <- phyloseq::distance(Pacu.bac.T1.CCC, method = "bray")
  Pacu.samp.df.T1.CCC <- as(sample_data(Pacu.bac.T1.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.PacuT1.CCC ~ Treatment, data = Pacu.samp.df.T1.CCC)
# Homogeneity of dispersion test  
  beta.Pacu.T1.CCC <- betadisper(bc.PacuT1.CCC, Pacu.samp.df.T1.CCC$Treatment)
  permutest(beta.Pacu.T1.CCC, pairwise = T) 

#TF.CCC
#make bray curtis data matrix 
  bc.Pacu.TF.CCC <- phyloseq::distance(Pacu.bac.TF.CCC, method = "bray")
  Pacu.samp.df.TF.CCC <- as(sample_data(Pacu.bac.TF.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF.CCC ~ Treatment, data = Pacu.samp.df.TF.CCC)
# Homogeneity of dispersion test  
  beta.Pacu.TF.CCC <- betadisper(bc.Pacu.TF.CCC, Pacu.samp.df.TF.CCC$Treatment)
  permutest(beta.Pacu.TF.CCC, pairwise = T) 


# C42 only, new objs
Pacu.bacT1F.C42 <- subset_samples(Pacu.bac.T1F, Clade == "C42")
Pacu.bac.T1.C42 <- subset_samples(Pacu.bac.T1, Clade == "C42")
Pacu.bac.TF.C42 <- subset_samples(Pacu.bac.TF, Clade == "C42")
  
#T1TF.C42
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F.C42 <- phyloseq::distance(Pacu.bacT1F.C42, method = "bray")
  Pacu.samp.df.T1F.C42 <- as(sample_data(Pacu.bacT1F.C42), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.T1F.C42 ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.C42)

#T1.C42
#make bray curtis data matrix 
  bc.PacuT1.C42 <- phyloseq::distance(Pacu.bac.T1.C42, method = "bray")
  Pacu.samp.df.T1.C42 <- as(sample_data(Pacu.bac.T1.C42), "data.frame") #sample df
  adonis(bc.PacuT1.C42 ~ Treatment, data = Pacu.samp.df.T1.C42)#adonis test
# Homogeneity of dispersion test  
  beta.Pacu.T1.C42 <- betadisper(bc.PacuT1.C42, Pacu.samp.df.T1.C42$Treatment)
  permutest(beta.Pacu.T1.C42, pairwise = T) 

#TF.C42
#make bray curtis data matrix 
  bc.Pacu.TF.C42 <- phyloseq::distance(Pacu.bac.TF.C42, method = "bray")
  Pacu.samp.df.TF.C42 <- as(sample_data(Pacu.bac.TF.C42), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF.C42 ~ Treatment, data = Pacu.samp.df.TF.C42)
# Homogeneity of dispersion test  
  beta.Pacu.TF.C42 <- betadisper(bc.Pacu.TF.C42, Pacu.samp.df.TF.C42$Treatment)
  permutest(beta.Pacu.TF.C42, pairwise = T) 
``` 
Pacu NMDS
```{r}
# C only, new objs
Pacu.bacT1F.CCC <- subset_samples(Pacu.bac.T1F, Clade == "CCC")
Pacu.bac.T1.CCC <- subset_samples(Pacu.bac.T1, Clade == "CCC")
Pacu.bac.TF.CCC <- subset_samples(Pacu.bac.TF, Clade == "CCC")
# # pull df sections for craig Pacu.bac.T1F
# OTU_Pacu<-out_table(Pacu.bac.T1F)
# tax  = tax_table(coralDIV2_rare2)
# otu  = otu_table(coralDIV2_rare2)
# 
# OTU1 = as(otu_table(Pacu.bac.T1F), "matrix")
# # transpose if necessary
# if(taxa_are_rows(Pacu.bac.T1F)){OTU1 <- t(OTU1)}
# # Coerce to data.frame
# OTUdf = as.data.frame(OTU1)
# write.csv(OTUdf, "Pacuta_its2_T1TF.csv")
# 


library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pacu.Parent.colors <- c("305" = "green4",  "306" = "firebrick3", "307" ="deeppink4", "308" = "orange", "309" = "green1","310" = "gold",  "311"= "hotpink", "312" ="darkslateblue", "302" = "honeydew4", "303" = "khaki4","304" ="ivory4", "305" = "lightblue", Ambient = "Blue", High = "Red","CCC Ambient"="blue","CCC High"="red", "C42 Ambient"="darkblue", "C42 High"="darkred") 





#T1 and TF
ord.Pacu <- ordinate(Pacu.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pacu)

scores.Pacu.samp.df <- as.data.frame(scores(ord.Pacu) ) 
scores.Pacu.samp.df$Time.Point <- Pacu.samp.df.T1F$Time.Point
scores.Pacu.samp.df$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu.samp.df$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu.samp.df$Clade <- Pacu.samp.df.T1F$Clade
scores.Pacu.samp.df$Code <- paste(Pacu.samp.df.T1F$Parent.ID, Pacu.samp.df.T1F$Time.Point)


ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P.acu Bray Curtis") +
  theme_classic()
#ggsave("Pacu_T1TF_its2.pdf")

ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var Bray Curtis") +
  theme_classic()
#ggsave("PacuT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pacu.bac.TF, ordinate(Pacu.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PacuT1 <- ordinate(Pacu.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuT1)
scores.PacuT1 <- as.data.frame(scores(ord.PacuT1))
scores.PacuT1$Treatment <- Pacu.samp.df.T1$Treatment
scores.PacuT1$Parent.ID <- Pacu.samp.df.T1$Parent.ID
scores.PacuT1$Clade <- Pacu.samp.df.T1$Clade
scores.PacuT1$CladeTreat <- as.factor(paste(Pacu.samp.df.T1$Clade,Pacu.samp.df.T1$Treatment))


ggplot(scores.PacuT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var T1 Bray Curtis") +
  theme_classic()
#ggsave("Pacu_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PacuTF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuTF)
scores.PacuTF <- as.data.frame(scores(ord.PacuTF))
scores.PacuTF$Treatment <- Pacu.samp.df.TF$Treatment
scores.PacuTF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.PacuTF$Clade <- Pacu.samp.df.TF$Clade
scores.PacuTF$CladeTreat <- as.factor(paste(Pacu.samp.df.TF$Clade,Pacu.samp.df.TF$Treatment))

ggplot(scores.PacuTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. var TF Bray Curtis") +
  theme_classic()
#gsave("Pacu_TF_its_nmds.pdf")










#### BY CLADE

#CCC T1 and TF
set.seed(30)
ord.Pacu_CCC  <- ordinate(Pacu.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC)
scores.Pacu_CCC <- as.data.frame(scores(ord.Pacu_CCC))
scores.Pacu_CCC$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu_CCC$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu_CCC$Time.Point <- Pacu.samp.df.T1F$Time.Point

ggplot(scores.Pacu_CCC, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var CCC") +
  theme_classic()
#ggsave("PacuTFits_CCC.pdf")


#C T1
set.seed(30)
ord.Pacu_CCC.T1 <- ordinate(Pacu.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.T1)
scores.Pacu_CCC.T1 <- as.data.frame(scores(ord.Pacu_CCC.T1))
scores.Pacu_CCC.T1$Treatment <- Pacu.samp.df.T1$Treatment
scores.Pacu_CCC.T1$Parent.ID <- Pacu.samp.df.T1$Parent.ID
scores.Pacu_CCC.T1$Time.Point <- Pacu.samp.df.T1$Time.Point

ggplot(scores.Pacu_CCC.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var T1 CCC") +
  theme_classic()
#ggsave("Pacu_T1_its_CCC.pdf")

#C TF
set.seed(30)
ord.Pacu_CCC.TF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.TF)
scores.Pacu_CCC.TF <- as.data.frame(scores(ord.Pacu_CCC.TF))
scores.Pacu_CCC.TF$Treatment <- Pacu.samp.df.TF$Treatment
scores.Pacu_CCC.TF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.Pacu_CCC.TF$Time.Point <- Pacu.samp.df.TF$Time.Point

ggplot(scores.Pacu_CCC.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var CCC TF") +
  theme_classic()
#ggsave("Pacu_TF_its_TF_CCC.pdf")



Pacu.bacT1F.C27  
Pacu.bac.T1.C27 
Pacu.bac.TF.C27 

#C27
set.seed(30)
ord.Pacu_C27 <- ordinate(Pacu.bacT1F.C27, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27)
scores.Pacu_C27 <- as.data.frame(scores(ord.Pacu_C27))
scores.Pacu_C27$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu_C27$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu_C27$Time.Point <- Pacu.samp.df.T1F$Time.Point

ggplot(scores.Pacu_C27, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C&D") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_its_T1TF.pdf")

#C27 T1
set.seed(30)
ord.Pacu_C27.T1 <- ordinate(Pacu.bac.T1.C27 , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27.T1)
scores.Pacu_C27.T1 <- as.data.frame(scores(ord.Pacu_C27.T1))
scores.Pacu_C27.T1$Treatment <- Pacu.samp.df.T1.C27$Treatment
scores.Pacu_C27.T1$Parent.ID <- Pacu.samp.df.T1.C27$Parent.ID
scores.Pacu_C27.T1$Time.Point <- Pacu.samp.df.T1.C27$Time.Point

ggplot(scores.Pacu_C27.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var T1 C27") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_T1.pdf")

#CD TF
set.seed(30)
ord.Pacu_C27.TF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27.TF)
scores.Pacu_C27.TF <- as.data.frame(scores(ord.Pacu_C27.TF))
scores.Pacu_C27.TF$Treatment <- Pacu.samp.df.TF$Treatment
scores.Pacu_C27.TF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.Pacu_C27.TF$Time.Point <- Pacu.samp.df.TF$Time.Point

ggplot(scores.Pacu_C27.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C27 TF") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_TF.pdf")





```












































#Alpha diversit: observed richness, Shannon (+evenness) - Hannah  
Use rarefied data, not relative abunances   
df:Bac.r2 and Bac.r2.sd sample data
```{r}  
#trying to install microbiome on old R version: https://microbiome.github.io/tutorials/Installation.html https://bioconductor.org/install/
Sp.colors <- c("Montipora_capitata"= "#D43F3AFF","Porites_compressa"="#EEA236FF","Pavona_varians"="#5CB85CFF","Pocillopora_acuta"="#46B8DAFF", "High"="darkRed","Ambient"="navyBlue")

Bac.r2.alpha<-Bac.r2 #make a copy of df

Bac.r2.alpha<-subset_samples(Bac.r2.alpha, Time.Point=="T1" |Time.Point=="TF")  # ONLY KEEP T1 and TF!
Bac.r2.alpha2 <- prune_taxa(taxa_sums(Bac.r2.alpha) > 0, Bac.r2.alpha) #prune taxa 0s
Bac.r2.alpha2.sd <- as(sample_data(Bac.r2.alpha2), "data.frame") #create sample data frame to view


#####try microbiome package for evenness and alpha diversity. use raw counts
library(microbiome)  
library(knitr)

erich.tab <-microbiome::alpha(Bac.r2.alpha2, index = "all") # pull out: observed, diversity_shannon, evenness_pielou

erich.tab2<-erich.tab[-c(2:4,6:8,10:22)] #keep the cols you need

erich.tab2
 erich.tab2$Treatment <- Bac.r2.alpha2.sd$Treatment       #add back in meta
  erich.tab2$Time.Point <- Bac.r2.alpha2.sd$Time.Point
  erich.tab2$Parent.ID <- Bac.r2.alpha2.sd$Parent.ID
  erich.tab2$ID <- Bac.r2.alpha2.sd$Frag.ID
  erich.tab2$Species <- Bac.r2.alpha2.sd$Species
  erich.tab2$Time.Point = factor(erich.tab2$Time.Point, levels = c("T0", "T1", "TF"))


#all species - to show significance and to subset

#ANOVA
#shannon
  aov.shannon.species = aov(diversity_shannon ~ Species, data=erich.tab2)
  summary(aov.shannon.species) # sig diff between species
  TukeyHSD(aov.shannon.species)
#observed  
  aov.observed.species = aov(observed ~ Species, data=erich.tab2)
  summary(aov.observed.species) # sig diff between species
  TukeyHSD(aov.observed.species)
#evenness
  aov.evenness.species = aov(evenness_pielou ~ Species, data=erich.tab2)
  summary(aov.evenness.species) # sig diff between species
  TukeyHSD(aov.evenness.species)  
#Plot
  boxplot(diversity_shannon ~ Species, data=erich.tab2, ylab="Shannon's diversity") 
  boxplot(observed ~ Species, data=erich.tab2, ylab="Observed") 
  boxplot(evenness_pielou ~ Species, data=erich.tab2, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.species <- levels(erich.tab2$Species) # get the variables

# make a pairwise list that we want to compare.
  al.species.pairs <- combn(seq_along(al.species), 2, simplify = FALSE, FUN = function(i)al.species[i])
  print(al.species.pairs)

#observed
p1 <- ggviolin(erich.tab2, x = "Species", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.species.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
    
#Shannon
p1 <- ggviolin(erich.tab2, x = "Species", y = "diversity_shannon",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.species.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2, x = "Species", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.species.pairs) #non-parametric test (Wilcoxon test)
  print(p1)


#by species:
erich.tab2.mcap<-subset(erich.tab2, Species=="Montipora_capitata")
erich.tab2.pcomp<-subset(erich.tab2, Species=="Porites_compressa")
erich.tab2.pvar<-subset(erich.tab2, Species=="Pavona_varians")
erich.tab2.pacu<-subset(erich.tab2, Species=="Pocillopora_acuta")

```
OLDER alpha, delete when ready
```{r}
#### older alpha

#########OK you are removing T0 here and then you need to make a new samp data frame and then do below. 




erich <- estimate_richness(Bac.r2.alpha2, measures = c("Observed", "Shannon")) #select which estimates and estimate
  erich$Treatment <- Bac.r2.sd$Treatment       #add back in meta
  erich$Time.Point <- Bac.r2.sd$Time.Point
  erich$Parent.ID <- Bac.r2.sd$Parent.ID
  erich$ID <- Bac.r2.sd$ID
  erich$Species <- Bac.r2.sd$Species
  erich$Time.Point = factor(erich$Time.Point, levels = c("T0", "T1", "TF"))

  
  
 
#stats ALL THE DATA
 library(ggpubr)
  #look at data
  
#observed
# hist(erich$Observed) 
# ggqqplot(erich$Observed)
# erich$Observed_log<-log10(erich$Observed)
# hist(erich$Observed_log) 

## subset by coral species for OBSERVED
erich.mcap<-subset(erich, Species=="Montipora_capitata")
  hist(erich.mcap$Observed) 
  # hist(erich.mcap$Observed_log) 

erich.pcomp<-subset(erich, Species=="Porites_compressa")
  hist(erich.pcomp$Observed) #meh
  #   hist(erich.pcomp$Observed_log) #Better 
  # ggqqplot(erich.pcomp$Observed_log) 
 
erich.pvar<-subset(erich, Species=="Pavona_varians")
 hist(erich.pvar$Observed) #
 # hist(erich.pvar$Observed_log) #better
 #  ggqqplot(erich.pvar$Observed_log) #

erich.pacu<-subset(erich, Species=="Pocillopora_acuta")
 hist(erich.pacu$Observed) 
  hist(erich.pacu$Observed_log) 
  ggqqplot(erich.pacu$Observed_log) 

  
## subset for Shannon
#shannon
hist(erich$Shannon)
ggqqplot(erich$Shannon)
# erich$Shannon_log<-log10(erich$Shannon)
# hist(erich$Shannon_log)
# ggqqplot(erich$Shannon_log)
# erich$Shannon_sq<-sqrt(erich$Shannon)
# hist(erich$Shannon_sq)
# ggqqplot(erich$Shannon_sq)


erich.mcap<-subset(erich, Species=="Montipora_capitata")
  hist(erich.mcap$Shannon) # 
  ggqqplot(erich.mcap$Shannon) #
  #   hist(erich.mcap$Shannon_log) # 
  # ggqqplot(erich.mcap$Shannon) #
 
erich.pcomp<-subset(erich, Species=="Porites_compressa")
  hist(erich.pcomp$Shannon) # normal 
  ggqqplot(erich.pcomp$Shannon) 
  # hist(erich.pcomp$Shannon_log) # normal 
  # ggqqplot(erich.pcomp$Shannon) 

erich.pvar<-subset(erich, Species=="Pavona_varians")
 hist(erich.pvar$Shannon) #not normal 
  ggqqplot(erich.pvar$Shannon) #
  
erich.pacu<-subset(erich, Species=="Pocillopora_acuta")
 hist(erich.pacu$Shannon) #not awfuk 
  ggqqplot(erich.pacu$Shannon) #not bad
  

### ALL SPECIES
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.species = aov(Shannon ~ Species, data=erich)

#Call for the summary of that ANOVA, which will include P-values
  summary(aov.shannon.species) # sig diff between species
  TukeyHSD(aov.shannon.species)

#Plot
  boxplot(Shannon ~ Species, data=erich, ylab="Shannon's diversity") 

  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.species = aov(Observed ~ Species*Treatment*Time.Point, data=erich)
  #aov.observed.species = aov(Observed ~ Species, data=erich)

#Call for the summary of that ANOVA, which will include P-values
  summary(aov.observed.species) # sig diff between species
  TukeyHSD(aov.observed.species)

#Plot
  boxplot(Observed ~ Species, data=erich, ylab="observed's diversity") 
  
#### LHK paper code
  
alphaplot <- plot_richness(Bac.r2.alpha, x="Species", measures=c("Observed","Shannon", "Simpson"), color="Time.Point") + 
  labs(x="Genet") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(strip.text.x = element_text(size=10, face="bold"),
        strip.background = element_rect(fill="#FFFFFF"))


alphaplot


#stats
erich <- estimate_richness(Bac.r2.alpha, measures=c("Observed","Shannon", "Simpson"))
#make dataframe
SampleID <- row.names(erich) 
# makes a new dataframe with the specified columnes
alpha <- data.frame(SampleID, erich) 
s <- data.frame(sample_data(Bac.r2.alpha)) 
# Change first column title to SampleID to match distances dataframe
colnames(s)[1] <- "SampleID" 
# merges metadata df and distances df
alphaGC <- merge(alpha, s, by = "SampleID") 

```

### MCAP alpha, df: erich.tab2.mcap
```{r} 
#ANOVA
erich.tab2.mcap$treat.time<-paste(erich.tab2.mcap$Treatment,erich.tab2.mcap$Time.Point)
erich.tab2.mcap$treat.time<-as.factor(erich.tab2.mcap$treat.time)

#shannon
  aov.shannon.treat.time = aov(diversity_shannon ~ treat.time, data=erich.tab2.mcap)
  summary(aov.shannon.treat.time) # sig diff between treat.time
  TukeyHSD(aov.shannon.treat.time)
#observed  
  aov.observed.treat.time = aov(observed ~ treat.time, data=erich.tab2.mcap)
  summary(aov.observed.treat.time) # sig diff between treat.time
  TukeyHSD(aov.observed.treat.time)
#evenness
  aov.evenness.treat.time = aov(evenness_pielou ~ treat.time, data=erich.tab2.mcap)
  summary(aov.evenness.treat.time) # sig diff between treat.time
  TukeyHSD(aov.evenness.treat.time)  
#Plot
  boxplot(diversity_shannon ~ treat.time, data=erich.tab2.mcap, ylab="Shannon's diversity") 
  boxplot(observed ~ treat.time, data=erich.tab2.mcap, ylab="Observed") 
  boxplot(evenness_pielou ~ treat.time, data=erich.tab2.mcap, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.treat.time <- levels(erich.tab2.mcap$treat.time) # get the variables

# make a pairwise list that we want to compare.
  al.treat.time.pairs <- combn(seq_along(al.treat.time), 2, simplify = FALSE, FUN = function(i)al.treat.time[i])
  print(al.treat.time.pairs)

#observed
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "observed",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
    
#Shannon
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "diversity_shannon",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "evenness_pielou",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)






#OLD

#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesMcap = aov(Shannon ~ Treatment*Time.Point, data=erich.mcap)
  summary(aov.shannon.speciesMcap) # sig diff between species
  TukeyHSD(aov.shannon.speciesMcap)

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.mcap, ylab="Shannon's diversity") 

  
  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesMcap = aov(Observed ~ Treatment*Time.Point, data=erich.mcap)
  summary(aov.observed.speciesMcap) # sig diff between species
  TukeyHSD(aov.observed.speciesMcap)
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.mcap, ylab="observed's diversity") 
```
#### Do by C and CD
```{r}
#mcap  erich.mcap


  # by parent ID set clade dom
  erich.mcap$Clade[erich.mcap$Parent.ID==363]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==364]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==365]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==366]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==367]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==368]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==369]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==370]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==371]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==372]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==373]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==374]<-"CD"
  erich.mcap$Clade<-as.factor(erich.mcap$Clade) #change to factor

## let look at C vs CD
  #Run the ANOVA and save it as an object
  aov.shannon.speciesMcap_CCD = aov(Shannon ~ Clade*Treatment*Time.Point, data=erich.mcap)
  summary(aov.shannon.speciesMcap_CCD) # sig diff between species
  TukeyHSD(aov.shannon.speciesMcap_CCD)

#Plot
  boxplot(Shannon ~ Clade*Treatment*Time.Point, data=erich.mcap, ylab="Shannon's diversity") 

#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesMcap_CCD = aov(Observed ~ Clade*Treatment*Time.Point, data=erich.mcap)
  summary(aov.observed.speciesMcap_CCD) # sig diff between species
  TukeyHSD(aov.observed.speciesMcap_CCD)
#Plot
  boxplot(Observed ~ Clade*Treatment*Time.Point, data=erich.mcap, ylab="observed's diversity") 
  
``` 
# Pcomp alpha
```{r} 
#ANOVA
erich.tab2.pcomp$treat.time<-paste(erich.tab2.pcomp$Treatment,erich.tab2.pcomp$Time.Point)
erich.tab2.pcomp$treat.time<-as.factor(erich.tab2.pcomp$treat.time)

#shannon
  aov.shannon.treat.time = aov(diversity_shannon ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.shannon.treat.time) # sig diff between treat.time
  TukeyHSD(aov.shannon.treat.time)
#observed  
  aov.observed.treat.time = aov(observed ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.observed.treat.time) # sig diff between treat.time
  TukeyHSD(aov.observed.treat.time)
#evenness
  aov.evenness.treat.time = aov(evenness_pielou ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.evenness.treat.time) # sig diff between treat.time
  TukeyHSD(aov.evenness.treat.time)  
#Plot
  boxplot(diversity_shannon ~ treat.time, data=erich.tab2.pcomp, ylab="Shannon's diversity") 
  boxplot(observed ~ treat.time, data=erich.tab2.pcomp, ylab="Observed") 
  boxplot(evenness_pielou ~ treat.time, data=erich.tab2.pcomp, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.treat.time <- levels(erich.tab2.pcomp$treat.time) # get the variables

# make a pairwise list that we want to compare.
  al.treat.time.pairs <- combn(seq_along(al.treat.time), 2, simplify = FALSE, FUN = function(i)al.treat.time[i])
  print(al.treat.time.pairs)

#observed
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "observed",
   add = "boxplot", fill = "treat.time", palette = c(Pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
    
#Shannon
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "diversity_shannon",
   add = "boxplot", fill = "treat.time", palette = c(pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "evenness_pielou",
   add = "boxplot", fill = "treat.time", palette = c(pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)















#old
#Shannon 
#Run the ANOVA and save it as an object
  aov.shannon.speciesPcomp = aov(Shannon ~ Treatment*Time.Point, data=erich.pcomp)
  summary(aov.shannon.speciesPcomp) # sig diff between species
  TukeyHSD(aov.shannon.speciesPcomp)
  
 
 

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pcomp, ylab="Shannon's diversity") 

#Pcomp  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPcomp = aov(Observed ~ Treatment*Time.Point, data=erich.pcomp)
  summary(aov.observed.speciesPcomp) # sig diff between species
  TukeyHSD(aov.observed.speciesPcomp)
  


  
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.pcomp, ylab="observed's diversity") 


  
``` 
# Pvar
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesPvar = aov(Shannon ~ Treatment*Time.Point, data=erich.pvar)
  summary(aov.shannon.speciesPvar) # sig diff between species
  TukeyHSD(aov.shannon.speciesPvar)

  

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pvar, ylab="Shannon's diversity") 
  
 
  
  
  
  
  
  

#Pvar  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPvar = aov(Observed ~ Treatment*Time.Point, data=erich.pvar)
  summary(aov.observed.speciesPvar) # sig diff between species
  TukeyHSD(aov.observed.speciesPvar)


  
#Plot
  boxplot(log(Observed) ~ Treatment*Time.Point, data=erich.pvar, ylab="observed's diversity") 
  
  
``` 

# Pacu
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesPacu = aov(Shannon ~ Treatment*Time.Point, data=erich.pacu)
  summary(aov.shannon.speciesPacu) # sig diff between species
  TukeyHSD(aov.shannon.speciesPacu)


#Plot
  aov.shannon.speciesPacu$Time.Point = factor(aov.shannon.speciesPacu$Time.Point, levels = c("T0", "T1", "TF"))

  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pacu, ylab="Shannon's diversity") 
  
 
  
  
  
  
  
  

#Pvar  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPacu = aov(Observed ~ Treatment*Time.Point, data=erich.pacu)
  summary(aov.observed.speciesPacu) # sig diff between species
  TukeyHSD(aov.observed.speciesPacu)


  
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.pacu, ylab="observed's diversity") 

# Just T1
  erich.pacuT1<-subset(erich.pacu, Time.Point=="T1")
  hist(log10(erich.pacuT1$Observed))
  aov.observed.speciesPacuT1 = aov(log10(Observed) ~ Treatment, data=erich.pacuT1)
  summary(aov.observed.speciesPacuT1) # sig diff between species
  TukeyHSD(aov.observed.speciesPacuT1)
  
  hist((erich.pacuT1$Shannon))
  aov.shannon.speciesPacuT1 = aov(log10(Shannon) ~ Treatment, data=erich.pacuT1)
  summary(aov.shannon.speciesPacuT1) # sig diff between species
  TukeyHSD(aov.shannon.speciesPacuT1)
  
  
``` 
 





 
```{r}
#for non normal Kruskal wallis rank sum  
  kruskal.test(x ~ AgeGroup, data=meta)

  
#testing
  
  coral <- get_variable(coralDIV2_rare.alpha2, "host_genus") %in% c("montipora",  "pocillopora")
sample_data(coralDIV2_rare.alpha2)$human <- factor(coral)

 #estimate_richness(coralDIV2_rare, "coral","host_genus", measures = c("Observed", "Shannon")) #select which estimates and estimate CANNOT GET TO WORK with X
  
## from tutorial can't get it to wokr in my code.   
  GP <- prune_taxa(taxa_sums(GlobalPatterns) > 0, GlobalPatterns)

human <- get_variable(GP, "SampleType") %in% c("Feces", "Mock", "Skin", "Tongue")
sample_data(GP)$human <- factor(human)

alpha_meas = c("Observed", "Shannon")
(p <- plot_richness(GP, "human", "SampleType", measures=alpha_meas))






##
## make bar plot of rank abundance of all DIVs
par(mar = c(10, 4, 4, 2) + 0.1) # make more room on bottom margin
N <- 50
barplot(sort(taxa_sums(coralDIV2_rare.alpha2), TRUE)[1:N]/nsamples(coralDIV2_rare.alpha2), las=2)

sample_variables(coralDIV2_rare.alpha2) #check sample variables

#plot_bar(coralDIV2_rare.alpha2, "Species", fill="DIV", facet_grid=~DIV)


#make network - not helpful for ITS2 here

ig <- make_network(coralDIV2_rare.alpha2, max.dist=0.3)
plot_network(ig, coralDIV2_rare.alpha2, color="Parent.ID", shape="Species", line_weight=0.4, label=NULL)






```




