---
title: "16S Thermal Tolerance"
html_document:
code_folding: hide
toc:yes
toc_depth:3
toc_float:yes
output: 
chunk_output_type: console
editor_options: 
  chunk_output_type: console
---
This is an RMarkdown for the All Coral Experiment 16S data. 

#16S
Load in your data
```{r} 
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr) 
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(tidyverse)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(microbiome)
library(vegan)
library(geosphere)
library(ade4)
library(microbiome)  
library(knitr)

# Load data (data_exploration.Rmd)
load("Data/Bac.seq_phyloseq.RData") 

## view data
ntaxa(Bac.seq)  #num taxa
nsamples(Bac.seq)   #num samples
sample_names(Bac.seq)[1:5] #samp names
rank_names(Bac.seq) #ranks are DIV and clade
sample_variables(Bac.seq) # metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.seq)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  # check 3k rare
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = Species)) +
  geom_point()

#remove duplicates and tests
Dups<-subset_samples(Bac.seq, Dups=="Yes")
TopNOTUsB = names(sort(taxa_sums(Dups), TRUE)[1:100])
bac50 = prune_taxa(TopNOTUsB, Dups)
bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S1753b" ) #remove duplicates
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S208" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3476" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "SN35b" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3149" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3149" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S1240" )
Bac.seq <- subset_samples(Bac.seq, Type != "extra" )
Bac.seq <- subset_samples(Bac.seq, Type != " " )
Bac.seq.df <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view

Bac.s = subset_samples(Bac.seq, Type== "sample" )
Bac.s.df <- as(sample_data(Bac.s), "data.frame") #create sample data frame to view

Bac.field = subset_samples(Bac.seq, Type== "field" )
Bac.field.df <- as(sample_data(Bac.field), "data.frame") #create sample data frame to view
```

#Make relative abundance df
```{r}        
Bac.s.rel  = transform_sample_counts(Bac.s, function(x) x / sum(x) )
Bac.s.rel = prune_taxa(taxa_sums(Bac.s.rel) > 0, Bac.s.rel) #this removes any OTU with 0s
data.Bac.s.rel <- as(sample_data(Bac.s.rel), "data.frame") 

Bac.seq.rel<- transform_sample_counts(Bac.seq, function(x) x / sum(x) ) 
Bac.seq.rel = prune_taxa(taxa_sums(Bac.seq.rel) > 0, Bac.seq.rel) #this removes any OTU with 0s
data.Bac.seq.rel <- as(sample_data(Bac.seq.rel), "data.frame") 

#number of ASVs by species
Bac.s.rel_Mcap<-subset_samples(Bac.s.rel, Species=="Montipora_capitata")
    Bac.s.rel_Mcap = prune_taxa(taxa_sums(Bac.s.rel_Mcap) > 0, Bac.s.rel_Mcap) #this removes any OTU with 0s

  Bac.s.rel_Pcomp<-subset_samples(Bac.s.rel, Species=="Porites_compressa")
    Bac.s.rel_Pcomp = prune_taxa(taxa_sums(Bac.s.rel_Pcomp) > 0, Bac.s.rel_Pcomp) #this removes any OTU with 0s

Bac.s.rel_Pvar<-subset_samples(Bac.s.rel, Species=="Pavona_varians")
    Bac.s.rel_Pvar = prune_taxa(taxa_sums(Bac.s.rel_Pvar) > 0, Bac.s.rel_Pvar) #this removes any OTU with 0s
  
    #by lineage for Pvar
    Bac.s.rel_Pvar.C1<-subset_samples(Bac.s.rel_Pvar, Clade=="C1")
    Bac.s.rel_Pvar.C1 = prune_taxa(taxa_sums(Bac.s.rel_Pvar.C1) > 0, Bac.s.rel_Pvar.C1)
    Bac.s.rel_Pvar.C27<-subset_samples(Bac.s.rel_Pvar, Clade=="C27")
    Bac.s.rel_Pvar.C27 = prune_taxa(taxa_sums(Bac.s.rel_Pvar.C27) > 0, Bac.s.rel_Pvar.C27)
    
Bac.s.rel_Pacu<-subset_samples(Bac.s.rel, Species=="Pocillopora_acuta")
    Bac.s.rel_Pacu = prune_taxa(taxa_sums(Bac.s.rel_Pacu) > 0, Bac.s.rel_Pacu) #this removes any OTU with 0s

#anything shared between species: ####

# Step 1: Convert to presence/absence
bac.presabs <- transform_sample_counts(Bac.s.rel, function(x) ifelse(x > 0, 1, 0))
unique(sample_data(Bac.s.rel)$Species)

# Step 2: List of species (must exactly match what's in your sample_data)
species_list <- c("Montipora_capitata", "Porites_compressa", "Pavona_varians", "Pocillopora_acuta")

# Step 3: Function to get taxa present in a species
get_present_taxa <- function(sp) {
  sp_samples <- rownames(as.data.frame(sample_data(bac.presabs)))[
    as.data.frame(sample_data(bac.presabs))$Species == sp
  ]
  sp_otu <- prune_samples(sp_samples, bac.presabs)
  otu_tab <- otu_table(sp_otu)
  if (!taxa_are_rows(sp_otu)) {
    otu_tab <- t(otu_tab)
  }
  taxa_names(sp_otu)[rowSums(otu_tab) > 0]
}

# Step 4: Create list of taxa per species
taxa_by_species <- setNames(lapply(species_list, get_present_taxa), species_list)

# Step 5: Get shared taxa for each species pair
pairwise_shared <- combn(species_list, 2, simplify = FALSE, FUN = function(pair) {
  taxa1 <- taxa_by_species[[pair[1]]]
  taxa2 <- taxa_by_species[[pair[2]]]
  shared <- intersect(taxa1, taxa2)
  list(
    species1 = pair[1],
    species2 = pair[2],
    n_shared = length(shared),
    shared_taxa = shared
  )
})

# Step 6: Create summary table
shared_summary <- map_df(pairwise_shared, ~{
  tibble(
    Species1 = .x$species1,
    Species2 = .x$species2,
    Shared_OTUs = .x$n_shared
  )
})

print(shared_summary)

# Optional: View OTUs or taxonomy for a pair
# First pair's shared OTUs:
pairwise_shared[[1]]$shared_taxa

# With taxonomy:
sharedtax<-as.data.frame(tax_table(Bac.s.rel)[pairwise_shared[[1]]$shared_taxa, ])
``` 
#Permanova UNIFRAC by ASV T0 ONLY
Tables S3-S6
```{r}         
 #T0 only #### 
Bac.s.rel.u.T0<-subset_samples(Bac.s.rel, Time.Point=="T0")
  Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s
#make unifrac matrix
  Bac.s.u.T0 <- phyloseq::distance(Bac.s.rel.u.T0, method = "wunifrac")
  s.u.samp.df <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df
  s.u.samp.df$Code<-paste(s.u.samp.df$Species,s.u.samp.df$Clade)

# PERMANOVA: Table S3
  set.seed(30)
    adonis2(Bac.s.u.T0 ~ Species, data = s.u.samp.df)
  set.seed(30)    
    adonis2(Bac.s.u.T0 ~ Species*Clade, data = s.u.samp.df)

# Homogeneity of dispersion test: Table S4 and S5
  set.seed(30)
    permutest(betadisper(Bac.s.u.T0, s.u.samp.df$Species, type = "centroid")) 
      ball<-  betadisper(Bac.s.u.T0, s.u.samp.df$Species, type = "centroid")
      TukeyHSD(ball,  ordered = FALSE,conf.level = 0.95)  #PCOMP has sig different dispersion
    
#Within Species: Table S6  ####
#Mcap
  #t0 only Bac.s.rel.u.T0
  mcap.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Montipora_capitata")
  mcap.u.bac.T0 = prune_taxa(taxa_sums(mcap.u.bac.T0) > 0, mcap.u.bac.T0) #this removes any OTU with 0s
  
  #make bray curtis data matrix
  u.mcap.u.T0 <- phyloseq::distance(mcap.u.bac.T0, method = "wunifrac")
     mcap.u.samp.df.T0 <- as(sample_data(mcap.u.bac.T0), "data.frame") #sample df
    
  set.seed(30)
    adonis2(u.mcap.u.T0 ~ Clade, data = mcap.u.samp.df.T0) 
  set.seed(30)
    permutest(betadisper(u.mcap.u.T0, mcap.u.samp.df.T0$Clade, type = "centroid")) 

#Pcomp  
  Pcomp.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Porites_compressa")
  Pcomp.u.bac.T0 = prune_taxa(taxa_sums(Pcomp.u.bac.T0) > 0, Pcomp.u.bac.T0) 

  #make bray curtis data matrix -T0
  u.Pcomp.u.T0 <- phyloseq::distance(Pcomp.u.bac.T0, method = "wunifrac")
     Pcomp.u.samp.df.T0 <- as(sample_data(Pcomp.u.bac.T0), "data.frame") 
     
  set.seed(30)
    adonis2(u.Pcomp.u.T0 ~ Clade, data = Pcomp.u.samp.df.T0)
  set.seed(30)
    permutest(betadisper(u.Pcomp.u.T0, Pcomp.u.samp.df.T0$Clade, type = "centroid")) 

#Pvar  
  #t0 only Bac.s.rel.u.T0
  Pvar.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Pavona_varians")
  Pvar.u.bac.T0 = prune_taxa(taxa_sums(Pvar.u.bac.T0) > 0, Pvar.u.bac.T0) 

  #make bray curtis data matrix -T0
  u.Pvar.u.T0 <- phyloseq::distance(Pvar.u.bac.T0, method = "wunifrac")
     Pvar.u.samp.df.T0 <- as(sample_data(Pvar.u.bac.T0), "data.frame") #sample df
  
  set.seed(30)
    adonis2(u.Pvar.u.T0 ~ Clade, data = Pvar.u.samp.df.T0) #adonis test no clade
  set.seed(30)
    permutest(betadisper(u.Pvar.u.T0, Pvar.u.samp.df.T0$Clade, type = "centroid"))     
    betadisper(u.Pvar.u.T0, Pvar.u.samp.df.T0$Time.Point, type = "centroid")
    
#Pacu  
  #t0 only Bac.s.rel.u.T0
  Pacu.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Pocillopora_acuta")
  Pacu.u.bac.T0 = prune_taxa(taxa_sums(Pacu.u.bac.T0) > 0, Pacu.u.bac.T0) 

  #make bray curtis data matrix -T0
  u.Pacu.u.T0 <- phyloseq::distance(Pacu.u.bac.T0, method = "wunifrac")
     Pacu.u.samp.df.T0 <- as(sample_data(Pacu.u.bac.T0), "data.frame") 
     
  set.seed(30)
    adonis2(u.Pacu.u.T0 ~ Clade, data = Pacu.u.samp.df.T0) #adonis test no clade
  set.seed(30)
    permutest(betadisper(u.Pacu.u.T0, Pacu.u.samp.df.T0$Clade, type = "centroid")) 
    betadisper(u.Pacu.u.T0, Pacu.u.samp.df.T0$Time.Point, type = "centroid")

``` 
    
#Permanova UNIFRAC by ASV
T0-T1, T1-TF by TREATMENT (high only, ambient only)
Models run within treatment group (high, ambient) with all T0 genotypes represented in both treatments (not split at T0)
Table S11
```{r} 
#Make T0 for Ambient and High with all genotypes represented ####
Bac.s.rel.T0Dupsamps<-Bac.s.rel #copy
 
#Ambient T0 ####
Bac.s.rel.T0Dupsamps_amb = subset_samples(Bac.s.rel, Time.Point == "T0" & Treatment == "Ambient")  #copy

#tax table
  tax  = tax_table(Bac.s.rel.T0Dupsamps_amb)
# sample data
  sd = sample_data(Bac.s.rel.T0Dupsamps_amb)
    rownames(sd) <- paste0(rownames(sd), "_dup")
# otu_table
  OTU = otu_table(Bac.s.rel.T0Dupsamps_amb)
    sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
    sd$Treatment <- as.character(sd$Treatment)
    sd$Treatment <- "High"    #rename HIGH
    sd$ID <- paste0(sd$ID, "_dup")
      bac.Amb.T0dup<-phyloseq(OTU, sd, tax)

##High ####
  Bac.s.rel.T0Dupsamps_amb = subset_samples(Bac.s.rel, Time.Point == "T0"&Treatment == "High")  #copy

#tax table
tax  = tax_table(Bac.s.rel.T0Dupsamps_amb)
# sample data
  sd = sample_data(Bac.s.rel.T0Dupsamps_amb)
  rownames(sd) <- paste0(rownames(sd), "_dup")
# otu_table
  OTU = otu_table(Bac.s.rel.T0Dupsamps_amb)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  sd$Treatment <- as.character(sd$Treatment)
  sd$Treatment <- "Ambient"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
    bac.High.T0dup<-phyloseq(OTU, sd, tax)

#Merge ####
Control16dups = merge_phyloseq(bac.Amb.T0dup,bac.High.T0dup)
Control16dups

  #merge  to the original df:
  Bac.s.rel.T0duplicated = merge_phyloseq(Bac.s.rel.T0Dupsamps,Control16dups)
   Bac.s.rel.T0duplicated
  Bac.s.rel.T0duplicated.sd <- as(sample_data(Bac.s.rel.T0duplicated), "data.frame") #create sample data frame to view
    Bac.s.rel.T0duplicated.sd

# Mcap  #### 

#Ambient only  #### 
  
#T0-T1 
  mcap.u.bac<- subset_samples(Bac.s.rel.T0duplicated, Species == "Montipora_capitata")
    mcap.u.bac.amb <- subset_samples(mcap.u.bac, Treatment == "Ambient")
        mcap.u.bac.T0T1.amb <- subset_samples(mcap.u.bac.amb, Time.Point == "T0"|Time.Point=="T1")
        mcap.u.bac.T0T1.amb = prune_taxa(taxa_sums(mcap.u.bac.T0T1.amb) > 0, mcap.u.bac.T0T1.amb) #this removes any OTU with 0s
        sample_sums(mcap.u.bac.T0T1.amb)
      
  #Make bray curtis data matrix
  u.mcap.u.bac.T0T1.amb <- phyloseq::distance(mcap.u.bac.T0T1.amb, method = "wunifrac")
     mcap.u.samp.df.T0T1.amb <- as(sample_data(mcap.u.bac.T0T1.amb), "data.frame") #sample df

  set.seed(30)
    adonis2(u.mcap.u.bac.T0T1.amb ~ Time.Point, data = mcap.u.samp.df.T0T1.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.mcap.u.bac.T0T1.amb, mcap.u.samp.df.T0T1.amb$Time.Point, type = "centroid")) 

#T1-TF 
  mcap.u.bac.T1TF.amb <- subset_samples(mcap.u.bac.amb, Time.Point == "T1"|Time.Point=="TF")
     mcap.u.bac.T1TF.amb = prune_taxa(taxa_sums(mcap.u.bac.T1TF.amb) > 0, mcap.u.bac.T1TF.amb) #this removes any OTU with 0s
        sample_sums(mcap.u.bac.T1TF.amb)
      
  #make bray curtis data matrix
  u.mcap.u.bac.T1TF.amb <- phyloseq::distance(mcap.u.bac.T1TF.amb, method = "wunifrac")
     mcap.u.samp.df.T1TF.amb <- as(sample_data(mcap.u.bac.T1TF.amb), "data.frame") #sample df
     mcap.u.samp.df.T1TF.amb$Code<-paste(mcap.u.samp.df.T1TF.amb$Treatment, mcap.u.samp.df.T1TF.amb$Time.Point)
   
  set.seed(30)
    adonis2(u.mcap.u.bac.T1TF.amb ~ Time.Point, data = mcap.u.samp.df.T1TF.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.mcap.u.bac.T1TF.amb, mcap.u.samp.df.T1TF.amb$Time.Point, type = "centroid")) 

#High only   
  #T0-T1 
    mcap.u.bac.high <- subset_samples(mcap.u.bac, Treatment == "High")
        mcap.u.bac.T0T1.high <- subset_samples(mcap.u.bac.high, Time.Point == "T0"|Time.Point=="T1")
       mcap.u.bac.T0T1.high = prune_taxa(taxa_sums(mcap.u.bac.T0T1.high) > 0, mcap.u.bac.T0T1.high) #this removes any OTU with 0s
        sample_sums(mcap.u.bac.T0T1.high)
      
  #make bray curtis data matrix
  u.mcap.u.bac.T0T1.high <- phyloseq::distance(mcap.u.bac.T0T1.high, method = "wunifrac")
     mcap.u.samp.df.T0T1.high <- as(sample_data(mcap.u.bac.T0T1.high), "data.frame") #sample df

  set.seed(30)
    adonis2(u.mcap.u.bac.T0T1.high ~ Time.Point, data = mcap.u.samp.df.T0T1.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.mcap.u.bac.T0T1.high, mcap.u.samp.df.T0T1.high$Time.Point, type = "centroid")) 

#T1-TF 
        mcap.u.bac.T1TF.high <- subset_samples(mcap.u.bac.high, Time.Point == "T1"|Time.Point=="TF")
       mcap.u.bac.T1TF.high = prune_taxa(taxa_sums(mcap.u.bac.T1TF.high) > 0, mcap.u.bac.T1TF.high) #this removes any OTU with 0s
        sample_sums(mcap.u.bac.T1TF.high)
      
  #make bray curtis data matrix
  u.mcap.u.bac.T1TF.high <- phyloseq::distance(mcap.u.bac.T1TF.high, method = "wunifrac")
     mcap.u.samp.df.T1TF.high <- as(sample_data(mcap.u.bac.T1TF.high), "data.frame") #sample df
     mcap.u.samp.df.T1TF.high$Code<-paste(mcap.u.samp.df.T1TF.high$Treatment, mcap.u.samp.df.T1TF.high$Time.Point)
   
  set.seed(30)
    adonis2(u.mcap.u.bac.T1TF.high ~ Time.Point, data = mcap.u.samp.df.T1TF.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.mcap.u.bac.T1TF.high, mcap.u.samp.df.T1TF.high$Time.Point, type = "centroid")) 
    
#Pcomp  #### 

#Ambient only  
#T0-T1 
  pcomp.u.bac<- subset_samples(Bac.s.rel.T0duplicated, Species == "Porites_compressa")
    pcomp.u.bac.amb <- subset_samples(pcomp.u.bac, Treatment == "Ambient")
        pcomp.u.bac.T0T1.amb <- subset_samples(pcomp.u.bac.amb, Time.Point == "T0"|Time.Point=="T1")
        pcomp.u.bac.T0T1.amb = prune_taxa(taxa_sums(pcomp.u.bac.T0T1.amb) > 0, pcomp.u.bac.T0T1.amb) #this removes any OTU with 0s
        sample_sums(pcomp.u.bac.T0T1.amb)
      
  #make bray curtis data matrix
  u.pcomp.u.bac.T0T1.amb <- phyloseq::distance(pcomp.u.bac.T0T1.amb, method = "wunifrac")
     pcomp.u.samp.df.T0T1.amb <- as(sample_data(pcomp.u.bac.T0T1.amb), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pcomp.u.bac.T0T1.amb ~ Time.Point, data = pcomp.u.samp.df.T0T1.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pcomp.u.bac.T0T1.amb, pcomp.u.samp.df.T0T1.amb$Time.Point, type = "centroid")) 

#T1-TF 
      pcomp.u.bac.T1TF.amb <- subset_samples(pcomp.u.bac.amb, Time.Point == "T1"|Time.Point=="TF")
       pcomp.u.bac.T1TF.amb = prune_taxa(taxa_sums(pcomp.u.bac.T1TF.amb) > 0, pcomp.u.bac.T1TF.amb) #this removes any OTU with 0s
        sample_sums(pcomp.u.bac.T1TF.amb)
      
  #make bray curtis data matrix
  u.pcomp.u.bac.T1TF.amb <- phyloseq::distance(pcomp.u.bac.T1TF.amb, method = "wunifrac")
     pcomp.u.samp.df.T1TF.amb <- as(sample_data(pcomp.u.bac.T1TF.amb), "data.frame") #sample df
     pcomp.u.samp.df.T1TF.amb$Code<-paste(pcomp.u.samp.df.T1TF.amb$Treatment, pcomp.u.samp.df.T1TF.amb$Time.Point)
   
  set.seed(30)
    adonis2(u.pcomp.u.bac.T1TF.amb ~ Time.Point, data = pcomp.u.samp.df.T1TF.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pcomp.u.bac.T1TF.amb, pcomp.u.samp.df.T1TF.amb$Time.Point, type = "centroid")) 

#High only   
  #T0-T1 
    pcomp.u.bac.high <- subset_samples(pcomp.u.bac, Treatment == "High")
        pcomp.u.bac.T0T1.high <- subset_samples(pcomp.u.bac.high, Time.Point == "T0"|Time.Point=="T1")
       pcomp.u.bac.T0T1.high = prune_taxa(taxa_sums(pcomp.u.bac.T0T1.high) > 0, pcomp.u.bac.T0T1.high) 
        sample_sums(pcomp.u.bac.T0T1.high)
      
  #make bray curtis data matrix
  u.pcomp.u.bac.T0T1.high <- phyloseq::distance(pcomp.u.bac.T0T1.high, method = "wunifrac")
     pcomp.u.samp.df.T0T1.high <- as(sample_data(pcomp.u.bac.T0T1.high), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pcomp.u.bac.T0T1.high ~ Time.Point, data = pcomp.u.samp.df.T0T1.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pcomp.u.bac.T0T1.high, pcomp.u.samp.df.T0T1.high$Time.Point, type = "centroid")) 

#T1-TF 
        pcomp.u.bac.T1TF.high <- subset_samples(pcomp.u.bac.high, Time.Point == "T1"|Time.Point=="TF")
       pcomp.u.bac.T1TF.high = prune_taxa(taxa_sums(pcomp.u.bac.T1TF.high) > 0, pcomp.u.bac.T1TF.high) 
        sample_sums(pcomp.u.bac.T1TF.high)
      
  #make bray curtis data matrix
  u.pcomp.u.bac.T1TF.high <- phyloseq::distance(pcomp.u.bac.T1TF.high, method = "wunifrac")
     pcomp.u.samp.df.T1TF.high <- as(sample_data(pcomp.u.bac.T1TF.high), "data.frame") #sample df
     pcomp.u.samp.df.T1TF.high$Code<-paste(pcomp.u.samp.df.T1TF.high$Treatment, pcomp.u.samp.df.T1TF.high$Time.Point)
   
  set.seed(30)
    adonis2(u.pcomp.u.bac.T1TF.high ~ Time.Point, data = pcomp.u.samp.df.T1TF.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pcomp.u.bac.T1TF.high, pcomp.u.samp.df.T1TF.high$Time.Point, type = "centroid")) 
    
#pvar  #### 

#Ambient only  
#T0-T1 
  pvar.u.bac<- subset_samples(Bac.s.rel.T0duplicated, Species == "Pavona_varians")
    pvar.u.bac.amb <- subset_samples(pvar.u.bac, Treatment == "Ambient")
        pvar.u.bac.T0T1.amb <- subset_samples(pvar.u.bac.amb, Time.Point == "T0"|Time.Point=="T1")
       pvar.u.bac.T0T1.amb = prune_taxa(taxa_sums(pvar.u.bac.T0T1.amb) > 0, pvar.u.bac.T0T1.amb) #this removes any OTU with 0s
        sample_sums(pvar.u.bac.T0T1.amb)
      
  #make bray curtis data matrix
  u.pvar.u.bac.T0T1.amb <- phyloseq::distance(pvar.u.bac.T0T1.amb, method = "wunifrac")
     pvar.u.samp.df.T0T1.amb <- as(sample_data(pvar.u.bac.T0T1.amb), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pvar.u.bac.T0T1.amb ~ Time.Point, data = pvar.u.samp.df.T0T1.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pvar.u.bac.T0T1.amb, pvar.u.samp.df.T0T1.amb$Time.Point, type = "centroid")) 

#T1-TF 
        pvar.u.bac.T1TF.amb <- subset_samples(pvar.u.bac.amb, Time.Point == "T1"|Time.Point=="TF")
       pvar.u.bac.T1TF.amb = prune_taxa(taxa_sums(pvar.u.bac.T1TF.amb) > 0, pvar.u.bac.T1TF.amb) #this removes any OTU with 0s
        sample_sums(pvar.u.bac.T1TF.amb)
      
  #make bray curtis data matrix
  u.pvar.u.bac.T1TF.amb <- phyloseq::distance(pvar.u.bac.T1TF.amb, method = "wunifrac")
     pvar.u.samp.df.T1TF.amb <- as(sample_data(pvar.u.bac.T1TF.amb), "data.frame") #sample df
     pvar.u.samp.df.T1TF.amb$Code<-paste(pvar.u.samp.df.T1TF.amb$Treatment, pvar.u.samp.df.T1TF.amb$Time.Point)
   
  set.seed(30)
    adonis2(u.pvar.u.bac.T1TF.amb ~ Time.Point, data = pvar.u.samp.df.T1TF.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pvar.u.bac.T1TF.amb, pvar.u.samp.df.T1TF.amb$Time.Point, type = "centroid")) #0.028
    dispersion <- betadisper(u.pvar.u.bac.T1TF.amb, 
                         pvar.u.samp.df.T1TF.amb$Time.Point, 
                         type = "centroid")
    plot(dispersion) #more dispersion at TF

#High only   
  #T0-T1 
    pvar.u.bac.high <- subset_samples(pvar.u.bac, Treatment == "High")
        pvar.u.bac.T0T1.high <- subset_samples(pvar.u.bac.high, Time.Point == "T0"|Time.Point=="T1")
       pvar.u.bac.T0T1.high = prune_taxa(taxa_sums(pvar.u.bac.T0T1.high) > 0, pvar.u.bac.T0T1.high) #this removes any OTU with 0s
        sample_sums(pvar.u.bac.T0T1.high)
      
  #make bray curtis data matrix
  u.pvar.u.bac.T0T1.high <- phyloseq::distance(pvar.u.bac.T0T1.high, method = "wunifrac")
     pvar.u.samp.df.T0T1.high <- as(sample_data(pvar.u.bac.T0T1.high), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pvar.u.bac.T0T1.high ~ Time.Point, data = pvar.u.samp.df.T0T1.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pvar.u.bac.T0T1.high, pvar.u.samp.df.T0T1.high$Time.Point, type = "centroid")) 

#T1-TF 
        pvar.u.bac.T1TF.high <- subset_samples(pvar.u.bac.high, Time.Point == "T1"|Time.Point=="TF")
       pvar.u.bac.T1TF.high = prune_taxa(taxa_sums(pvar.u.bac.T1TF.high) > 0, pvar.u.bac.T1TF.high) #this removes any OTU with 0s
        sample_sums(pvar.u.bac.T1TF.high)
      
  #make bray curtis data matrix
  u.pvar.u.bac.T1TF.high <- phyloseq::distance(pvar.u.bac.T1TF.high, method = "wunifrac")
     pvar.u.samp.df.T1TF.high <- as(sample_data(pvar.u.bac.T1TF.high), "data.frame") #sample df
     pvar.u.samp.df.T1TF.high$Code<-paste(pvar.u.samp.df.T1TF.high$Treatment, pvar.u.samp.df.T1TF.high$Time.Point)
   
  set.seed(30)
    adonis2(u.pvar.u.bac.T1TF.high ~ Time.Point, data = pvar.u.samp.df.T1TF.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pvar.u.bac.T1TF.high, pvar.u.samp.df.T1TF.high$Time.Point, type = "centroid")) 
    
#pacu  #### 

#Ambient only  
#T0-T1 
  pacu.u.bac<- subset_samples(Bac.s.rel.T0duplicated, Species == "Pocillopora_acuta")
    pacu.u.bac.amb <- subset_samples(pacu.u.bac, Treatment == "Ambient")
        pacu.u.bac.T0T1.amb <- subset_samples(pacu.u.bac.amb, Time.Point == "T0"|Time.Point=="T1")
        pacu.u.bac.T0T1.amb = prune_taxa(taxa_sums(pacu.u.bac.T0T1.amb) > 0, pacu.u.bac.T0T1.amb) #this removes any OTU with 0s
        sample_sums(pacu.u.bac.T0T1.amb)
      
  #make bray curtis data matrix
  u.pacu.u.bac.T0T1.amb <- phyloseq::distance(pacu.u.bac.T0T1.amb, method = "wunifrac")
     pacu.u.samp.df.T0T1.amb <- as(sample_data(pacu.u.bac.T0T1.amb), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pacu.u.bac.T0T1.amb ~ Time.Point, data = pacu.u.samp.df.T0T1.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pacu.u.bac.T0T1.amb, pacu.u.samp.df.T0T1.amb$Time.Point, type = "centroid")) 

#T1-TF 
        pacu.u.bac.T1TF.amb <- subset_samples(pacu.u.bac.amb, Time.Point == "T1"|Time.Point=="TF")
       pacu.u.bac.T1TF.amb = prune_taxa(taxa_sums(pacu.u.bac.T1TF.amb) > 0, pacu.u.bac.T1TF.amb) #this removes any OTU with 0s
        sample_sums(pacu.u.bac.T1TF.amb)
      
  #make bray curtis data matrix
  u.pacu.u.bac.T1TF.amb <- phyloseq::distance(pacu.u.bac.T1TF.amb, method = "wunifrac")
     pacu.u.samp.df.T1TF.amb <- as(sample_data(pacu.u.bac.T1TF.amb), "data.frame") #sample df
     pacu.u.samp.df.T1TF.amb$Code<-paste(pacu.u.samp.df.T1TF.amb$Treatment, pacu.u.samp.df.T1TF.amb$Time.Point)
   
  set.seed(30)
    adonis2(u.pacu.u.bac.T1TF.amb ~ Time.Point, data = pacu.u.samp.df.T1TF.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pacu.u.bac.T1TF.amb, pacu.u.samp.df.T1TF.amb$Time.Point, type = "centroid")) 
    
#High only   
  #T0-T1 
    pacu.u.bac.high <- subset_samples(pacu.u.bac, Treatment == "High")
        pacu.u.bac.T0T1.high <- subset_samples(pacu.u.bac.high, Time.Point == "T0"|Time.Point=="T1")
       pacu.u.bac.T0T1.high = prune_taxa(taxa_sums(pacu.u.bac.T0T1.high) > 0, pacu.u.bac.T0T1.high) #this removes any OTU with 0s
        sample_sums(pacu.u.bac.T0T1.high)
      
  #make bray curtis data matrix
  u.pacu.u.bac.T0T1.high <- phyloseq::distance(pacu.u.bac.T0T1.high, method = "wunifrac")
     pacu.u.samp.df.T0T1.high <- as(sample_data(pacu.u.bac.T0T1.high), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pacu.u.bac.T0T1.high ~ Time.Point, data = pacu.u.samp.df.T0T1.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pacu.u.bac.T0T1.high, pacu.u.samp.df.T0T1.high$Time.Point, type = "centroid")) 

#T1-TF 
        pacu.u.bac.T1TF.high <- subset_samples(pacu.u.bac.high, Time.Point == "T1"|Time.Point=="TF")
       pacu.u.bac.T1TF.high = prune_taxa(taxa_sums(pacu.u.bac.T1TF.high) > 0, pacu.u.bac.T1TF.high) #this removes any OTU with 0s
        sample_sums(pacu.u.bac.T1TF.high)
      
  #make bray curtis data matrix
  u.pacu.u.bac.T1TF.high <- phyloseq::distance(pacu.u.bac.T1TF.high, method = "wunifrac")
     pacu.u.samp.df.T1TF.high <- as(sample_data(pacu.u.bac.T1TF.high), "data.frame") #sample df
     pacu.u.samp.df.T1TF.high$Code<-paste(pacu.u.samp.df.T1TF.high$Treatment, pacu.u.samp.df.T1TF.high$Time.Point)
   
  set.seed(30)
    adonis2(u.pacu.u.bac.T1TF.high ~ Time.Point, data = pacu.u.samp.df.T1TF.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pacu.u.bac.T1TF.high, pacu.u.samp.df.T1TF.high$Time.Point, type = "centroid")) 
``` 

#Unifrac NMDS 
Figure 2B, S13-S15
```{r}      
 Sp.colors <- c("Montipora_capitata"= "#E69F00","Porites_compressa"="#56B4E9","Pavona_varians"="#CC79A7","Pocillopora_acuta"="#009E73", "High"="darkRed","Ambient"="navyBlue")

clade_colors <- c("C31" = "goldenrod1", "C31D4" = "darkorange3", "C15cj" = "powderblue", "C15n" = "#0072B2", "C1"="pink1", "C27"="firebrick", "C1d"="palegreen2", "C42_2"="darkgreen", "High"="red", "Ambient"="blue")

#T0 only  ####
Bac.s.rel.u.T0<-subset_samples(Bac.s.rel, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s

unifrac.all.T0 <- ordinate(Bac.s.rel.u.T0, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T0)
scores.ord.u_RelA_stats.T0<-as.data.frame(cbind(vegan::scores(unifrac.all.T0, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.T0 <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df

scores.ord.u_RelA_stats.T0$Treatment <- unifrac.all.sd.T0$Treatment
scores.ord.u_RelA_stats.T0$Species <- unifrac.all.sd.T0$Species
scores.ord.u_RelA_stats.T0$Clade <- unifrac.all.sd.T0$Clade
scores.ord.u_RelA_stats.T0$Code <- paste(scores.ord.u_RelA_stats.T0$Clade, scores.ord.u_RelA_stats.T0$Species)

#plot Figure 2B
ggplot(scores.ord.u_RelA_stats.T0, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  scale_color_manual(values = Sp.colors) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Species, fill = Species), 
               geom = "polygon", alpha = 0.1, linetype = 1) +  # Solid line and filled
  scale_fill_manual(values = Sp.colors) +  # Fill ellipses with the same color as points
  ggtitle("NMDS 16S TO") +
  theme_classic()
ggsave("16S_T0.pdf")

#T0 Only Figure S13
ggplot(scores.ord.u_RelA_stats.T0, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  #scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
 # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  ggtitle("NMDS 16s unifrac T0") +
  theme_classic()+
  facet_wrap(~Species)
ggsave("16S_T0Clade.pdf")

#T1 only####
Bac.s.rel.u.T1<-subset_samples(Bac.s.rel, Time.Point=="T1")
    Bac.s.rel.u.T1 = prune_taxa(taxa_sums(Bac.s.rel.u.T1) > 0, Bac.s.rel.u.T1) #this removes any OTU with 0s

unifrac.all.T1 <- ordinate(Bac.s.rel.u.T1, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T1)
scores.ord.u_RelA_stats.T1<-as.data.frame(cbind(vegan::scores(unifrac.all.T1, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.T1 <- as(sample_data(Bac.s.rel.u.T1), "data.frame") #sample df

scores.ord.u_RelA_stats.T1$Treatment <- unifrac.all.sd.T1$Treatment
scores.ord.u_RelA_stats.T1$Species <- unifrac.all.sd.T1$Species
scores.ord.u_RelA_stats.T1$Clade <- unifrac.all.sd.T1$Clade
scores.ord.u_RelA_stats.T1$Code <- paste(scores.ord.u_RelA_stats.T1$Clade, scores.ord.u_RelA_stats.T1$Treatment)
scores.ord.u_RelA_stats.T1$Code3 <- paste(scores.ord.u_RelA_stats.T1$Species, scores.ord.u_RelA_stats.T1$Treatment)

p<-ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species, shape=Treatment), size = 4) +
  scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
stat_ellipse(aes(x = NMDS1, y = NMDS2, color=Species, group=Code3,lty = Species)) +
scale_linetype_manual(values=c(1,2,1,2,1,2,1,2))+
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic();p
p+facet_wrap(~Species)

#T1 by lineage and treatment ####
ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape = Treatment), size = 4) +
  stat_ellipse(aes(fill = Code), 
               geom = "polygon", 
               alpha = 0.3, 
               colour = NA) +  # remove ellipse outlines (optional)
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic() +
  facet_wrap(~Species)

ggsave("16S T1 by lineage.pdf")


# ADD PCOMP ONLY T1 AMB/HIGH ####
scores.ord.u_RelA_stats.T1_Pcomp<-subset(scores.ord.u_RelA_stats.T1, Species=="Porites_compressa")
# Plot
# Subset Porites compressa
scores.ord.u_RelA_stats.T1_Pcomp <- subset(scores.ord.u_RelA_stats.T1, Species == "Porites_compressa")

# Plot with matched fill and color using Sp.colors: Figure S13
p <- ggplot(scores.ord.u_RelA_stats.T1_Pcomp, aes(x = NMDS1, y = NMDS2)) + 
  # Points
  geom_point(aes(color = Treatment, shape = Treatment), size = 8) +

  # 95% filled ellipses
  stat_ellipse(aes(fill = Treatment, linetype = Treatment),
               geom = "polygon",
               level = 0.95,
               alpha = 0.25,
               color = NA) +

  # Apply the same custom colors to both color and fill
  scale_color_manual(values = Sp.colors) +
  scale_fill_manual(values = Sp.colors) +

  ggtitle("NMDS 16S – T1 Porites compressa (95% Ellipses by Treatment)") +
  theme_classic()
p
ggsave("16S T1 Pcomp by treatment")



#TF only ####
Bac.s.rel.u.TF<-subset_samples(Bac.s.rel, Time.Point=="TF")
    Bac.s.rel.u.TF = prune_taxa(taxa_sums(Bac.s.rel.u.TF) > 0, Bac.s.rel.u.TF) #this removes any OTU with 0s

unifrac.all.TF <- ordinate(Bac.s.rel.u.TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.TF)
scores.ord.u_RelA_stats.TF<-as.data.frame(cbind(vegan::scores(unifrac.all.TF, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.TF <- as(sample_data(Bac.s.rel.u.TF), "data.frame") #sample df

scores.ord.u_RelA_stats.TF$Treatment <- unifrac.all.sd.TF$Treatment
scores.ord.u_RelA_stats.TF$Species <- unifrac.all.sd.TF$Species
scores.ord.u_RelA_stats.TF$Clade <- unifrac.all.sd.TF$Clade
scores.ord.u_RelA_stats.TF$Code <- paste(scores.ord.u_RelA_stats.TF$Clade, scores.ord.u_RelA_stats.TF$Species)
scores.ord.u_RelA_stats.TF$Code3 <- paste(scores.ord.u_RelA_stats.TF$Species, scores.ord.u_RelA_stats.TF$Treatment)



ggplot(scores.ord.u_RelA_stats.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species, shape=Treatment), size = 4) +
  scale_color_manual(values=Sp.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, color = Species, group = Code3, linetype = Treatment, fill = Treatment),
             geom = "polygon", alpha = 0.2)+
    scale_fill_manual(values = c("Ambient" = "blue", "High" = "red")) +  # flipped here
scale_linetype_manual(values=c(1,2,1,2,1,2,1,2))+
  ggtitle("NMDS 16s unifrac TF") +
  theme_classic()+
  facet_wrap(~Species)
ggsave("16s TF by treatment")

#Table S13  
ggplot(scores.ord.u_RelA_stats.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  scale_color_manual(values=clade_colors)+
 stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  ggtitle("NMDS 16s unifrac TF") +
  theme_classic()+
  facet_wrap(~Species)

#Ordination NMDS for all species, all time points #### 
unifrac.all <- ordinate(Bac.s.rel, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all)
scores.ord.u_RelA_stats<-as.data.frame(cbind(vegan::scores(unifrac.all, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd <- as(sample_data(Bac.s.rel), "data.frame") #sample df

scores.ord.u_RelA_stats$Treatment <- unifrac.all.sd$Treatment
scores.ord.u_RelA_stats$Species <- unifrac.all.sd$Species
scores.ord.u_RelA_stats$Clade <- unifrac.all.sd$Clade
scores.ord.u_RelA_stats$Time.Point <- unifrac.all.sd$Time.Point
scores.ord.u_RelA_stats$Code <- paste(scores.ord.u_RelA_stats$Clade, scores.ord.u_RelA_stats$Species)

ggplot(scores.ord.u_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS 16s bray") +
  theme_classic()+
  facet_wrap(~ Species + Time.Point, nrow = 4)
#ggsave("bc_all.pdf")

#T1 and TF only
#### mcap #####   
Bac.s.rel_McapT1TF <- subset_samples(Bac.s.rel_Mcap, Time.Point == "T1" | Time.Point == "TF")

ord.mcap2 <- ordinate(Bac.s.rel_McapT1TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.mcap2)

scores.mcap.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.mcap2, display="sites"))) # these are "points" ask hannah 
     Bac.s.rel_Mcap.df <- as(sample_data(Bac.s.rel_McapT1TF), "data.frame") #sample df
scores.mcap.u.samp.df$Time.Point <- Bac.s.rel_Mcap.df$Time.Point
scores.mcap.u.samp.df$Parent.ID <- Bac.s.rel_Mcap.df$Parent.ID
scores.mcap.u.samp.df$Treatment <- Bac.s.rel_Mcap.df$Treatment
scores.mcap.u.samp.df$Clade <- Bac.s.rel_Mcap.df$Clade
scores.mcap.u.samp.df$Code <- paste(Bac.s.rel_Mcap.df$Parent.ID,Bac.s.rel_Mcap.df$Time.Point)
scores.mcap.u.samp.df$Code2 <- paste(Bac.s.rel_Mcap.df$Treatment,Bac.s.rel_Mcap.df$Time.Point)
scores.mcap.u.samp.df$Code3 <- paste(Bac.s.rel_Mcap.df$Treatment,Bac.s.rel_Mcap.df$Time.Point,mcap.u.samp.df.T1F$Parent.ID)

mcap_16S<-ggplot(scores.mcap.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  stat_ellipse(aes(fill = Treatment, group = interaction(Treatment, Time.Point),
                   linetype = factor(Time.Point)),
               geom = "polygon", alpha = 0.2, color = NA) +
  scale_fill_manual(values = mcap.u.Parent.colors) +
  scale_linetype_manual(values = c(2, 3, 2, 3)) +
  guides(linetype = "none") +  
  ggtitle("M. capitata") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);

mcap_16S_clade<-ggplot(scores.mcap.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Clade, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  ggtitle("M. capitata") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);mcap_16S_clade

#### Pcomp #####  
Bac.s.rel_PcompT1TF <- subset_samples(Bac.s.rel_Pcomp, Time.Point == "T1" | Time.Point == "TF")

ord.pcomp2 <- ordinate(Bac.s.rel_PcompT1TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pcomp2)

scores.pcomp.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.pcomp2, display="sites"))) 
  Bac.s.rel_Pcomp.df <- as(sample_data(Bac.s.rel_PcompT1TF), "data.frame") #sample df
scores.pcomp.u.samp.df$Time.Point <- Bac.s.rel_Pcomp.df$Time.Point
scores.pcomp.u.samp.df$Parent.ID <- Bac.s.rel_Pcomp.df$Parent.ID
scores.pcomp.u.samp.df$Treatment <- Bac.s.rel_Pcomp.df$Treatment
scores.pcomp.u.samp.df$Clade <- Bac.s.rel_Pcomp.df$Clade
scores.pcomp.u.samp.df$Code <- paste(Bac.s.rel_Pcomp.df$Parent.ID,Bac.s.rel_Pcomp.df$Time.Point)
scores.pcomp.u.samp.df$Code2 <- paste(Bac.s.rel_Pcomp.df$Treatment,Bac.s.rel_Pcomp.df$Time.Point)
scores.pcomp.u.samp.df$Code3 <- paste(Bac.s.rel_Pcomp.df$Treatment,Bac.s.rel_Pcomp.df$Time.Point,pcomp.u.samp.df.T1F$Parent.ID)

pcomp_16S<-ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  scale_color_manual(values = pcomp.u.Parent.colors) +
  stat_ellipse(aes(fill = Treatment, group = interaction(Treatment, Time.Point),
                   linetype = factor(Time.Point)),
               geom = "polygon", alpha = 0.2, color = NA) +
  scale_fill_manual(values = pcomp.u.Parent.colors) +
  scale_linetype_manual(values = c(2, 3, 2, 3)) +
  guides(linetype = "none") +  # 👈 removes line type legend
  ggtitle("P. compressa") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pcomp_16S

pcomp_16S_clade<-ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Clade, color = Treatment), size = 5) + 
  scale_color_manual(values = pcomp.u.Parent.colors) +
  #stat_ellipse(aes(group = interaction(Treatment, Clade)),
              # geom = "polygon", alpha = 0.2, color = NA) +
  # scale_fill_manual(values = pcomp.u.Parent.colors) +
  guides(linetype = "none") +  # 👈 removes line type legend
  ggtitle("P. compressa") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pcomp_16S_clade

#### Pvar #####   
Bac.s.rel_PvarT1TF <- subset_samples(Bac.s.rel_Pvar, Time.Point == "T1" | Time.Point == "TF")

ord.pvar2 <- ordinate(Bac.s.rel_PvarT1TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pvar2)

scores.pvar.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.pvar2, display="sites"))) 
     Bac.s.rel_Pvar.df <- as(sample_data(Bac.s.rel_PvarT1TF), "data.frame") 
scores.pvar.u.samp.df$Time.Point <- Bac.s.rel_Pvar.df$Time.Point
scores.pvar.u.samp.df$Parent.ID <- Bac.s.rel_Pvar.df$Parent.ID
scores.pvar.u.samp.df$Treatment <- Bac.s.rel_Pvar.df$Treatment
scores.pvar.u.samp.df$Clade <- Bac.s.rel_Pvar.df$Clade
scores.pvar.u.samp.df$Code <- paste(Bac.s.rel_Pvar.df$Parent.ID,Bac.s.rel_Pvar.df$Time.Point)
scores.pvar.u.samp.df$Code2 <- paste(Bac.s.rel_Pvar.df$Treatment,Bac.s.rel_Pvar.df$Time.Point)
scores.pvar.u.samp.df$Code3 <- paste(Bac.s.rel_Pvar.df$Treatment,Bac.s.rel_Pvar.df$Time.Point,pvar.u.samp.df.T1F$Parent.ID)

pvar_16S<-ggplot(scores.pvar.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  stat_ellipse(aes(fill = Treatment, group = interaction(Treatment, Time.Point),
                   linetype = factor(Time.Point)),
               geom = "polygon", alpha = 0.2, color = NA) +
  scale_fill_manual(values = mcap.u.Parent.colors) +
  scale_linetype_manual(values = c(2, 3, 2, 3)) +
  guides(linetype = "none") +  # 👈 removes line type legend
  ggtitle("P. varians") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pvar_16S

pvar_16S_clade<-ggplot(scores.pvar.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Clade, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  #stat_ellipse(aes(group = interaction(Treatment, Clade)),
              # geom = "polygon", alpha = 0.2, color = NA) +
  # scale_fill_manual(values = pvar.u.Parent.colors) +
  guides(linetype = "none") +  # 👈 removes line type legend
  ggtitle("P. varians") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pvar_16S_clade

#### Pacu #####   
Bac.s.rel_PacuT1TF <- subset_samples(Bac.s.rel_Pacu, Time.Point == "T1" | Time.Point == "TF")

ord.pacu2 <- ordinate(Bac.s.rel_PacuT1TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pacu2)

scores.pacu.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.pacu2, display="sites")))
     Bac.s.rel_Pacu.df <- as(sample_data(Bac.s.rel_PacuT1TF), "data.frame") #sample df
scores.pacu.u.samp.df$Time.Point <- Bac.s.rel_Pacu.df$Time.Point
scores.pacu.u.samp.df$Parent.ID <- Bac.s.rel_Pacu.df$Parent.ID
scores.pacu.u.samp.df$Treatment <- Bac.s.rel_Pacu.df$Treatment
scores.pacu.u.samp.df$Clade <- Bac.s.rel_Pacu.df$Clade
scores.pacu.u.samp.df$Code <- paste(Bac.s.rel_Pacu.df$Parent.ID,Bac.s.rel_Pacu.df$Time.Point)
scores.pacu.u.samp.df$Code2 <- paste(Bac.s.rel_Pacu.df$Treatment,Bac.s.rel_Pacu.df$Time.Point)
scores.pacu.u.samp.df$Code3 <- paste(Bac.s.rel_Pacu.df$Treatment,Bac.s.rel_Pacu.df$Time.Point,pacu.u.samp.df.T1F$Parent.ID)

pacu_16S<-ggplot(scores.pacu.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  stat_ellipse(aes(fill = Treatment, group = interaction(Treatment, Time.Point),
                   linetype = factor(Time.Point)),
               geom = "polygon", alpha = 0.2, color = NA) +
  scale_fill_manual(values = mcap.u.Parent.colors) +
  scale_linetype_manual(values = c(2, 3, 2, 3)) +
  guides(linetype = "none") +  
  ggtitle("P. acuta") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pacu_16S

pacu_16S_clade<-ggplot(scores.pacu.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Clade, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  #stat_ellipse(aes(group = interaction(Treatment, Clade)),
              # geom = "polygon", alpha = 0.2, color = NA) +
  # scale_fill_manual(values = pacu.u.Parent.colors) +
  guides(linetype = "none") +  
  ggtitle("P. acuta") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pacu_16S_clade

#plot all together: Figure S14 #### 
combined_16S <- grid.arrange(mcap_16S, pcomp_16S, pvar_16S, pacu_16S, ncol = 1)
ggsave("16S_over_time.png", combined_16S, width = 8, height = 16, dpi = 300)


#plot all together: Figure S15 #### 
combined_16S <- grid.arrange(mcap_16S_clade, pcomp_16S_clade, pvar_16S_clade, pacu_16S_clade, ncol = 1)
ggsave("16S_over_time_clade.png", combined_16S, width = 8, height = 16, dpi = 300)
```

# Unifrac with field samples to compare: Figure S16
```{r} 
 # Check for microbiome tank drift compare Ambient to Field samples #### 
Bac.subset <- subset_samples(Bac.seq, Type == "field" | Treatment == "Ambient")
Bac.subset<-subset_samples(Bac.subset, Time.Point=="T1"|Time.Point=="TF")

unifrac.all.field <- ordinate(Bac.subset, method="NMDS", "unifrac",weighted=TRUE, set.seed(40)) 
stressplot(unifrac.all.field)
scores.ord.u_ALLfield_stats<-as.data.frame(cbind(vegan::scores(unifrac.all.field, display="sites"))) 
     unifrac.samp.field.sd <- as(sample_data(Bac.subset), "data.frame") #sample df
scores.ord.u_ALLfield_stats$Treatment <- unifrac.samp.field.sd$Treatment
scores.ord.u_ALLfield_stats$Species <- unifrac.samp.field.sd$Species
scores.ord.u_ALLfield_stats$Type <- unifrac.samp.field.sd$Type
scores.ord.u_ALLfield_stats$Time.Point <- unifrac.samp.field.sd$Time.Point
scores.ord.u_ALLfield_stats$Code <- paste(scores.ord.u_ALLfield_stats$Species, scores.ord.u_ALLfield_stats$Type)
scores.ord.u_ALLfield_stats$Time.Point[scores.ord.u_ALLfield_stats$Time.Point == "F1"] <- "T1"          # Replace b by XXX
scores.ord.u_ALLfield_stats$Time.Point[scores.ord.u_ALLfield_stats$Time.Point == "TC"] <- "T0"          # Replace b by XXX

pf<-ggplot(scores.ord.u_ALLfield_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Type, shape=Type), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code), linetype = 2) +
  ggtitle("NMDS 16s ") +
  theme_classic();pf
#ggsave("bc_all.pdf")
field<-pf+facet_wrap(~Species~Time.Point, nrow=4);field
ggsave("16S_field_compare.png", field, width = 8, height = 13, dpi = 300)

#PERMANOVA by species: Table S18, no interactive effect of microbial communities between sample times over time

#Mcap
Bac.subset.mcap<-subset_samples(Bac.subset, Species=="Montipora_capitata")
#make unifrac matrix
s.u.samp.df.mcap <- as(sample_data(Bac.subset.mcap), "data.frame") #sample df
s.u.samp.df.mcap$Code<-paste(s.u.samp.df.mcap$Species,s.u.samp.df.mcap$Clade)
s.u.samp.df.mcap$Time.Point[s.u.samp.df.mcap$Time.Point == "F1"] <- "T1"          # Replace b by XXX

FieldSamps.mcap <- phyloseq::distance(Bac.subset.mcap, method = "wunifrac")
     FieldSamps.df.mcap <- as(sample_data(Bac.subset.mcap), "data.frame")
     FieldSamps.df.mcap$Code<-paste(FieldSamps.df.mcap$Treatment,FieldSamps.df.mcap$Clade)
 
      set.seed(30)
    adonis2(FieldSamps.mcap ~ Type*Time.Point, data = FieldSamps.df.mcap, by="margin")

#Pcomp
Bac.subset.pcomp<-subset_samples(Bac.subset, Species=="Porites_compressa")
#make unifrac matrix
s.u.samp.df.pcomp <- as(sample_data(Bac.subset.pcomp), "data.frame") #sample df

FieldSamps.pcomp <- phyloseq::distance(Bac.subset.pcomp, method = "wunifrac")
     FieldSamps.df.pcomp <- as(sample_data(Bac.subset.pcomp), "data.frame")

      set.seed(30)
    adonis2(FieldSamps.pcomp ~ Type*Time.Point, data = FieldSamps.df.pcomp, by="margin")
    
#Pvar
Bac.subset.pvar<-subset_samples(Bac.subset, Species=="Pavona_varians")
#make unifrac matrix
s.u.samp.df.pvar <- as(sample_data(Bac.subset.pvar), "data.frame") #sample df

FieldSamps.pvar <- phyloseq::distance(Bac.subset.pvar, method = "wunifrac")
     FieldSamps.df.pvar <- as(sample_data(Bac.subset.pvar), "data.frame")

      set.seed(30)
    adonis2(FieldSamps.pvar ~ Type*Time.Point, data = FieldSamps.df.pvar, by="margin")
    
#Pacu
Bac.subset.pacu<-subset_samples(Bac.subset, Species=="Pocillopora_acuta")
#make unifrac matrix
s.u.samp.df.pacu <- as(sample_data(Bac.subset.pacu), "data.frame") #sample df

FieldSamps.pacu <- phyloseq::distance(Bac.subset.pacu, method = "wunifrac")
     FieldSamps.df.pacu <- as(sample_data(Bac.subset.pacu), "data.frame")

      set.seed(30)
    adonis2(FieldSamps.pacu ~ Type*Time.Point, data = FieldSamps.df.pacu, by="margin")
``` 

#Alpha diversit: observed richness, Shannon (+evenness) - Hannah  
Use rarefied data, not relative abunances   
df:Bac.s and
```{r}           
Sp.colors <- c("Montipora_capitata"= "#D43F3AFF","Porites_compressa"="#EEA236FF","Pavona_varians"="#5CB85CFF","Pocillopora_acuta"="#46B8DAFF", "High"="darkRed","Ambient"="navyBlue")

Bac.al.alpha<-Bac.s #make a copy of df
Bac.al.alpha2 <- prune_taxa(taxa_sums(Bac.al.alpha) > 0, Bac.al.alpha) #prune taxa 0s
Bac.al.alpha2.sd <- as(sample_data(Bac.al.alpha2), "data.frame") #create sample data frame to view


##### microbiome package for evenness and alpha diversity. use raw counts ####
erich.tab <-microbiome::alpha(Bac.al.alpha2, index = "all") # pull out: observed, diversity_shannon, evenness_pielou

erich.tab2<-erich.tab[-c(2:4,6:8,10:22)] #keep the cols you need

 erich.tab2$Treatment <- Bac.al.alpha2.sd$Treatment       #add back in meta
  erich.tab2$Time.Point <- Bac.al.alpha2.sd$Time.Point
  erich.tab2$Parent.ID <- Bac.al.alpha2.sd$Parent.ID
  erich.tab2$ID <- Bac.al.alpha2.sd$Frag.ID
  erich.tab2$Species <- Bac.al.alpha2.sd$Species
    erich.tab2$Clade <- Bac.al.alpha2.sd$Clade
    erich.tab2$Tank.num <- Bac.al.alpha2.sd$Tank.num
erich.tab2$Time.Point = factor(erich.tab2$Time.Point, levels = c("T0", "T1", "TF"))


#all species - to show significance and to subset
  erich.tab2T0 <- subset(erich.tab2, Time.Point=="T0")#make T0 to use below
  erich.tab2 <- subset(erich.tab2, Time.Point=="T1"|Time.Point=="TF")

#ANOVA
#shannon
  aov.shannon.species = aov(diversity_shannon ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.shannon.species) # sig diff between species
  TukeyHSD(aov.shannon.species)
#observed  
  aov.observed.species = aov(observed ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.observed.species) # sig diff between species
  TukeyHSD(aov.observed.species)
#evenness
  aov.evenness.species = aov(evenness_pielou ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.evenness.species) # sig diff between species
  TukeyHSD(aov.evenness.species)  
#Plot
  boxplot(diversity_shannon ~ Species, data=erich.tab2, ylab="Shannon's diversity") 
  boxplot(observed ~ Species, data=erich.tab2, ylab="Observed") 
  boxplot(evenness_pielou ~ Species, data=erich.tab2, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.species <- levels(erich.tab2$Species) # get the variables

# make a pairwise list that we want to compare.
  al.species.pairs <- combn(seq_along(al.species), 2, simplify = FALSE, FUN = function(i)al.species[i])
  print(al.species.pairs)

 erich.tab2$Code<-paste(erich.tab2$Species,erich.tab2$Clade)
  
#observed
p1 <- ggviolin(erich.tab2, x = "Species", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

    p1 <- ggviolin(erich.tab2, x = "Code", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

  #Shannon
p1 <- ggviolin(erich.tab2, x = "Species", y = "diversity_shannon",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.species.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2, x = "Species", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.species.pairs) #non-parametric test (Wilcoxon test)
  print(p1)

p1 <- ggviolin(erich.tab2, x = "Code", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

#by species:
erich.tab2.mcap<-subset(erich.tab2, Species=="Montipora_capitata")
erich.tab2.pcomp<-subset(erich.tab2, Species=="Porites_compressa")
erich.tab2.pvar<-subset(erich.tab2, Species=="Pavona_varians")
erich.tab2.pacu<-subset(erich.tab2, Species=="Pocillopora_acuta")


#look at data
#observed
hist(erich.tab2$observed)
ggqqplot(erich.tab2$observed)
hist(log(erich.tab2$observed))
ggqqplot(log10(erich.tab2$observed))
erich.tab2$observed_log<-log10(erich.tab2$observed)

hist(erich.tab2.mcap$observed_log)
ggqqplot(erich.tab2.mcap$observed_log)
 hist(erich.tab2.pcomp$observed_log)
ggqqplot(erich.tab2.pcomp$observed_log)
   hist(erich.tab2.pvar$observed_log)
ggqqplot(erich.tab2.pvar$observed_log)
        hist(erich.tab2.pacu$observed_log)
ggqqplot(erich.tab2.pacu$observed_log)



#shannon
hist(erich.tab2$diversity_shannon)
ggqqplot(erich.tab2$diversity_shannon)
hist(log10(erich.tab2$diversity_shannon))
ggqqplot(log10(erich.tab2$diversity_shannon))
hist(sqrt(erich.tab2$diversity_shannon))
ggqqplot(sqrt(erich.tab2$diversity_shannon))
hist((erich.tab2$diversity_shannon)^(1/4))
ggqqplot(sqrt(erich.tab2$diversity_shannon))


#evenness
hist(erich.tab2$evenness_pielou) #best
ggqqplot(erich.tab2$evenness_pielou)
hist(log10(erich.tab2$evenness_pielou))#worse
ggqqplot(erich.tab2$evenness_pielou)
erich.tab2$even.t<-sqrt(max(erich.tab2$evenness_pielou+1)-erich.tab2$evenness_pielou)
hist(sqrt(max(erich.tab2$evenness_pielou+1)-erich.tab2$evenness_pielou))
ggqqplot(sqrt(max(erich.tab2$evenness_pielou+1)-erich.tab2$evenness_pielou))

hist(erich.tab2.mcap$evenness_pielou)
ggqqplot(erich.tab2.mcap$evenness_pielou)
 hist(erich.tab2.pcomp$evenness_pielou)
ggqqplot(erich.tab2.pcomp$evenness_pielou)
   hist(erich.tab2.pvar$evenness_pielou)
ggqqplot(erich.tab2.pvar$evenness_pielou)
        hist(erich.tab2.pacu$evenness_pielou)
ggqqplot(erich.tab2.pacu$evenness_pielou)

hist(erich.tab2.mcap$even.t)
ggqqplot(erich.tab2.mcap$even.t)
 hist(erich.tab2.pcomp$even.t)
ggqqplot(erich.tab2.pcomp$even.t)
   hist(erich.tab2.pvar$even.t)
ggqqplot(erich.tab2.pvar$even.t)
        hist(erich.tab2.pacu$even.t)
ggqqplot(erich.tab2.pacu$even.t)






#by species:
erich.tab2.mcap<-subset(erich.tab2, Species=="Montipora_capitata")
erich.tab2.pcomp<-subset(erich.tab2, Species=="Porites_compressa")
erich.tab2.pvar<-subset(erich.tab2, Species=="Pavona_varians")
erich.tab2.pacu<-subset(erich.tab2, Species=="Pocillopora_acuta")

hist(erich.tab2.mcap$observed_log)
hist(erich.tab2.mcap$observed) #better
hist(erich.tab2.pcomp$observed_log)#better
hist(erich.tab2.pcomp$observed) 
hist(erich.tab2.pvar$observed_log)#better
hist(erich.tab2.pvar$observed) 
hist(erich.tab2.pacu$observed_log)#better
hist(erich.tab2.pacu$observed) 





```
#Observed Richness 
```{r}      
#observed    
 #log it
#T0
hist(erich.tab2T0$observed)
ggqqplot(erich.tab2T0$observed)
erich.tab2T0$observed_log<-log10(erich.tab2T0$observed)
 hist(erich.tab2T0$observed_log)

observed_alpha_modT0<-lmer(observed_log ~ Species+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  observed_alpha_modT0<-lmer(observed_log ~ Species*Clade+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  
#Richness
p1 <- ggviolin(erich.tab2T0, x = "Species", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
#Evenness
p1 <- ggviolin(erich.tab2T0, x = "Species", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

 
 
 
 
 erich.tab2.T1F<-subset(erich.tab2, Time.Point=="T1"|Time.Point=="TF")
 
 #T1TF
observed_alpha_mod<-lmer(observed_log ~ Species*Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.T1F) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_mod) 
  qqPlot(residuals(observed_alpha_mod)) 
  emm<-emmeans(observed_alpha_mod,  ~ Species*Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  observed_alpha_mod<-lmer(observed_log ~ Species*Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2.T1F) #adding tank is singularity
  anova(observed_alpha_mod) 
  qqPlot(residuals(observed_alpha_mod)) 
  emm<-emmeans(observed_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)



################Mcap only
  hist(erich.tab2.mcap$observed_log)
ggqqplot(erich.tab2.mcap$observed_log)
  
 #t1TF
erich.tab2.mcap.T1F<-subset(erich.tab2.mcap, Time.Point=="T1"|Time.Point=="TF")
  Mcap_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment+(1|Parent.ID), data=erich.tab2.mcap.T1F) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.T1F) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.mcap.T1<-subset(erich.tab2.mcap.T1F, Time.Point=="T1")

  Mcap_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.mcap.TF<-subset(erich.tab2.mcap.T1F, Time.Point=="TF")

   Mcap_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)

 #t0
  erich.tab2.mcap.T0<-subset(erich.tab2T0, Species=="Montipora_capitata")

  Mcap_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.mcap.T0) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)

  
  
####### PCOMP only ####
   erich.tab2.pcomp.T1F<-subset(erich.tab2.pcomp, Time.Point=="T1"|Time.Point=="TF")

  #t1TF
  pcomp_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp.T1F) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp.T1F) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pcomp.T1<-subset(erich.tab2.pcomp, Time.Point=="T1")

  pcomp_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Clade*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pcomp.TF<-subset(erich.tab2.pcomp, Time.Point=="TF")

   pcomp_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

 #t0
  erich.tab2.pcomp.T0<-subset(erich.tab2T0, Species=="Porites_compressa")

  pcomp_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.pcomp.T0) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
####### pvar only####
   
  #t1TF
  pvar_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pvar.T1<-subset(erich.tab2.pvar, Time.Point=="T1")

  pvar_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pvar.TF<-subset(erich.tab2.pvar, Time.Point=="TF")

   pvar_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #t0
  erich.tab2.pvar.T0<-subset(erich.tab2T0, Species=="Pavona_varians")

  pvar_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.pvar.T0) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  ####### pacu only####

  #t1TF
  pacu_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pacu.T1<-subset(erich.tab2.pacu, Time.Point=="T1")

  pacu_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pacu.TF<-subset(erich.tab2.pacu, Time.Point=="TF")

   pacu_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #t0
  erich.tab2.pacu.T0<-subset(erich.tab2T0, Species=="Pocilopora_acuta")

  pacu_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.pacu.T0) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  
```
alpha Observed Plots, no clade
```{r}  
#Means   
observed_mean <- ddply(erich.tab2, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                 N    = length(observed[!is.na(observed)]), #calculate the length of the data frame, excluding NA’s
                 mean = mean(observed, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(observed, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(observed, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
observed_mean

#Mcap 
Mcap_observed_mean<-subset(observed_mean, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_observed_mean<-subset(observed_mean, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_observed_mean<-subset(observed_mean, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_observed_mean<-subset(observed_mean, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_observed_plot<-ggplot(data=Mcap_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_observed_plot


#plot Pcomp
Pcomp_observed_plot<-ggplot(data=Pcomp_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_observed_plot

#Pvar plots
#plot Pvar
Pvar_observed_plot<-ggplot(data=Pvar_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
    xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_observed_plot


#Pacu plots
#plot Pacu
Pacu_observed_plot<-ggplot(data=Pacu_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_observed_plot

###### All plots together
Mcap_observed_plot<-Mcap_observed_plot+ theme(legend.position = "none") #remove legend
Pcomp_observed_plot<-Pcomp_observed_plot+ theme(legend.position = "none") #remove legend
Pvar_observed_plot<-Pvar_observed_plot+ theme(legend.position = "none") #remove legend
Pacu_observed_plot<-Pacu_observed_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_observed_plot, Pcomp_observed_plot, Pvar_observed_plot,Pacu_observed_plot, nrow = 1)
 
```  

alpha Observed Plots, clade  
```{r}   
# color pal from eta mcap.g.Parent.colors
alpha.colors <- c( "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#1F78B4","C High"="#6A3D9A", "CD Ambient"="#E31A1C", "CD High"="#FF7F00") 

#Means
observed_mean.C <- ddply(erich.tab2, c("Treatment", "Species", "Time.Point","Clade"), summarise, #summarize cell counts
                 N    = length(observed[!is.na(observed)]), #calculate the length of the data frame, excluding NA’s
                 mean = mean(observed, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(observed, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(observed, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
observed_mean.C
observed_mean.C$Code<-paste(observed_mean.C$Clade, observed_mean.C$Treatment)

#Mcap 
Mcap_observed_mean.C<-subset(observed_mean.C, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_observed_mean.C<-subset(observed_mean.C, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_observed_mean.C<-subset(observed_mean.C, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_observed_mean.C<-subset(observed_mean.C, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_observed_plot<-ggplot(data=Mcap_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
    xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  # ylim(0, 190) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_observed_plot

#boxplot

Rich16.plot<-ggplot(erich.tab2.mcap.T1, aes(x=Treatment, y=mean, color=Treatment)) + 
  scale_color_manual(values=c("blue", "red"))+
     # coord_cartesian(ylim = c(0, 10))+
   theme_classic()+
 theme(legend.position="none") +# Remove legend
  geom_boxplot(notch=F);Rich16.plot

library(tidyverse)
library(hrbrthemes)
library(viridis)

erich.tab2.mcap.T1$code2<-paste(erich.tab2.mcap.T1$Code, erich.tab2.mcap.T1$Treatment)


erich.tab2.mcap.T1 %>%
  ggplot( aes(x=code2, y=observed, fill=Treatment)) +
    geom_boxplot() +
    scale_color_manual(values=c("red", "blue"))+
    # scale_fill_viridis(discrete = TRUE, alpha=0.6, option="A") +
    # theme_ipsum() +
    theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
    ggtitle("Mcap Rich T1 ") +
    xlab("")

#Pcomp plots
#plot Pcomp
Pcomp_observed_plot<-ggplot(data=Pcomp_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
   xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_observed_plot

#Pvar plots
#plot Pvar
Pvar_observed_plot<-ggplot(data=Pvar_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
   xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_observed_plot


#plot Pacu
Pacu_observed_plot<-ggplot(data=Pacu_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_observed_plot

###### All plots together
Mcap_observed_plot<-Mcap_observed_plot+ theme(legend.position = "none") #remove legend
Pcomp_observed_plot<-Pcomp_observed_plot+ theme(legend.position = "none") #remove legend
Pvar_observed_plot<-Pvar_observed_plot+ theme(legend.position = "none") #remove legend
Pacu_observed_plot<-Pacu_observed_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_observed_plot, Pcomp_observed_plot, Pvar_observed_plot,Pacu_observed_plot, nrow = 1)
 
```
 
#evenness_pielou Richness 
```{r}   
#evenness_pielou  
#not transformeds

evenness_pielou_alpha_mod<-lmer(evenness_pielou ~ Species*Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2) #adding tank is singularity
  anova(evenness_pielou_alpha_mod) 
  qqPlot(residuals(evenness_pielou_alpha_mod)) 
  emm<-emmeans(evenness_pielou_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  evenness_pielou_alpha_mod<-lmer(evenness_pielou ~ Species*Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2) #adding tank is singularity
  anova(evenness_pielou_alpha_mod) 
  qqPlot(residuals(evenness_pielou_alpha_mod)) 
  emm<-emmeans(evenness_pielou_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

#T0
hist(erich.tab2T0$evenness_pielou)
ggqqplot(erich.tab2T0$evenness_pielou)


observed_alpha_modT0<-lmer(evenness_pielou ~ Species+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #this is nesting so not sure you can do this this way. 
  observed_alpha_modT0<-lmer(evenness_pielou ~ Species*Clade+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  

################Mcap only
 #t1TF
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID), data=erich.tab2.mcap) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.mcap.T1<-subset(erich.tab2.mcap, Time.Point=="T1")

  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Clade*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.mcap.TF<-subset(erich.tab2.mcap, Time.Point=="TF")

   Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)

#T0
  #t1
  erich.tab2.mcap.T0<-subset(erich.tab2.mcap, Time.Point=="T0")
hist(erich.tab2.mcap.T0$evenness_pielou)
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.mcap.T0) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
####### PCOMP only
  #t1TF
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pcomp.T1<-subset(erich.tab2.pcomp, Time.Point=="T1")

  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pcomp.TF<-subset(erich.tab2.pcomp, Time.Point=="TF")

   pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

#t0
  #t1
  erich.tab2.pcomp.T0<-subset(erich.tab2.pcomp, Time.Point=="T0")
hist(erich.tab2.pcomp.T0$evenness_pielou)
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.pcomp.T0) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
####### pvar only
  #t1TF
  pvar_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pvar.T1<-subset(erich.tab2.pvar, Time.Point=="T1")

  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pvar.TF<-subset(erich.tab2.pvar, Time.Point=="TF")

   pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
 
  
    #t0
  erich.tab2.pvar.T0<-subset(erich.tab2.pvar, Time.Point=="T0")
hist(erich.tab2.pvar.T0$evenness_pielou)
   pvar_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.pvar.T0) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 
  
  ####### pacu only
  #t1TF
  pacu_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pacu.T1<-subset(erich.tab2.pacu, Time.Point=="T1")

  pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pacu.TF<-subset(erich.tab2.pacu, Time.Point=="TF")

   pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #t0
  erich.tab2.pacu.T0<-subset(erich.tab2.pacu, Time.Point=="T0")
hist(erich.tab2.pacu.T0$evenness_pielou)#medium
   pacu_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.pacu.T0) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
```
alpha evenness_pielou Plots, no clade
```{r} 
#Means
evenness_pielou_mean <- ddply(erich.tab2, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NA’s
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
evenness_pielou_mean

#Mcap 
Mcap_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_evenness_pielou_plot<-ggplot(data=Mcap_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_evenness_pielou_plot


#plot Pcomp
Pcomp_evenness_pielou_plot<-ggplot(data=Pcomp_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_evenness_pielou_plot

#Pvar plots
#plot Pvar
Pvar_evenness_pielou_plot<-ggplot(data=Pvar_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_evenness_pielou_plot


#Pacu plots
#plot Pacu
Pacu_evenness_pielou_plot<-ggplot(data=Pacu_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_evenness_pielou_plot

###### All plots together
Mcap_evenness_pielou_plot<-Mcap_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pcomp_evenness_pielou_plot<-Pcomp_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pvar_evenness_pielou_plot<-Pvar_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pacu_evenness_pielou_plot<-Pacu_evenness_pielou_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_evenness_pielou_plot, Pcomp_evenness_pielou_plot, Pvar_evenness_pielou_plot,Pacu_evenness_pielou_plot, nrow = 1)
 
```  

alpha evenness_pielou Plots, clade  
```{r}
# color pal from eta mcap.g.Parent.colors
alpha.colors <- c( "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#1F78B4","C High"="#6A3D9A", "CD Ambient"="#E31A1C", "CD High"="#FF7F00") 

#Means
evenness_pielou_mean.C <- ddply(erich.tab2, c("Treatment", "Species", "Time.Point","Clade"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NA’s
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
evenness_pielou_mean.C
evenness_pielou_mean.C$Code<-paste(evenness_pielou_mean.C$Clade, evenness_pielou_mean.C$Treatment)

#Mcap 
Mcap_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_evenness_pielou_plot<-ggplot(data=Mcap_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_evenness_pielou_plot

#Pcomp plots
#plot Pcomp
Pcomp_evenness_pielou_plot<-ggplot(data=Pcomp_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
 xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_evenness_pielou_plot

#Pvar plots
#plot Pvar
Pvar_evenness_pielou_plot<-ggplot(data=Pvar_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
 xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_evenness_pielou_plot


#plot Pacu
Pacu_evenness_pielou_plot<-ggplot(data=Pacu_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
 xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_evenness_pielou_plot

###### All plots together
Mcap_evenness_pielou_plot<-Mcap_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pcomp_evenness_pielou_plot<-Pcomp_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pvar_evenness_pielou_plot<-Pvar_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pacu_evenness_pielou_plot<-Pacu_evenness_pielou_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_evenness_pielou_plot, Pcomp_evenness_pielou_plot, Pvar_evenness_pielou_plot,Pacu_evenness_pielou_plot, nrow = 1)
 
```
 




















#Observed Richness OLD ANOVA CODE DELETE WHEN UPDATED ABOVE
```{r}
#observed  


################Mcap only


  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Time.Point*Treatment, data=erich.tab2.mcap)
  summary(aov.diversity_shannon.mcap) # sig diff between species
  TukeyHSD(aov.diversity_shannon.mcap)
  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Time.Point*Treatment*Clade, data=erich.tab2.mcap)
  summary(aov.diversity_shannon.mcap) # sig diff between species
  
  #lmer
shannon_Mcap_model<-lmer(diversity_shannon~Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), erich.tab2.mcap)
  anova(shannon_Mcap_model) 
  qqPlot(residuals(shannon_Mcap_model)) 
  emm<-emmeans(shannon_Mcap_model,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  
  
  
 #t1
  erich.tab2.mcap.T1<-subset(erich.tab2.mcap, Time.Point=="T1")

  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Treatment, data=erich.tab2.mcap.T1)
  summary(aov.diversity_shannon.mcap) # sig diff between species
  TukeyHSD(aov.diversity_shannon.mcap)
  
  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.mcap.T1)
  summary(aov.diversity_shannon.mcap) # sig diff between species 
  
  #tf
  erich.tab2.mcap.TF<-subset(erich.tab2.mcap, Time.Point=="TF")

  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Treatment, data=erich.tab2.mcap.TF)
  summary(aov.diversity_shannon.mcap) # sig diff between species
  TukeyHSD(aov.diversity_shannon.mcap)
  
  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.mcap.TF)
  summary(aov.diversity_shannon.mcap) # sig diff between species 

  
####### PCOMP only
 #t1TF
  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Time.Point*Treatment, data=erich.tab2.pcomp)
  summary(aov.diversity_shannon.pcomp) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pcomp)
  
  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Time.Point*Treatment*Clade, data=erich.tab2.pcomp)
  summary(aov.diversity_shannon.pcomp) # sig diff between species
  
 #t1
  erich.tab2.pcomp.T1<-subset(erich.tab2.pcomp, Time.Point=="T1")

  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Treatment, data=erich.tab2.pcomp.T1)
  summary(aov.diversity_shannon.pcomp) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pcomp)
  
  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.pcomp.T1)
  summary(aov.diversity_shannon.pcomp) # sig diff between species 
  
  #tf
  erich.tab2.pcomp.TF<-subset(erich.tab2.pcomp, Time.Point=="TF")

  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Treatment, data=erich.tab2.pcomp.TF)
  summary(aov.diversity_shannon.pcomp) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pcomp)
  
  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.pcomp.TF)
  summary(aov.diversity_shannon.pcomp) # sig diff between species 
  
################pacu only
 #t1TF
  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Time.Point*Treatment, data=erich.tab2.pacu)
  summary(aov.diversity_shannon.pacu) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pacu)
  
  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Time.Point*Treatment*Clade, data=erich.tab2.pacu)
  summary(aov.diversity_shannon.pacu) # sig diff between species
  
 #t1
  erich.tab2.pacu.T1<-subset(erich.tab2.pacu, Time.Point=="T1")

  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Treatment, data=erich.tab2.pacu.T1)
  summary(aov.diversity_shannon.pacu) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pacu)
  
  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.pacu.T1)
  summary(aov.diversity_shannon.pacu) # sig diff between species 
  
  #tf
  erich.tab2.pacu.TF<-subset(erich.tab2.pacu, Time.Point=="TF")

  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Treatment, data=erich.tab2.pacu.TF)
  summary(aov.diversity_shannon.pacu) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pacu)
  
  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.pacu.TF)
  summary(aov.diversity_shannon.pacu) # sig diff between species 

```
OLDER alpha, delete when ready
```{r}
#### older alpha

#########OK you are removing T0 here and then you need to make a new samp data frame and then do below. 

 


erich <- estimate_richness(Bac.r2.alpha2, measures = c("Observed", "Shannon")) #select which estimates and estimate
  erich$Treatment <- Bac.r2.sd$Treatment       #add back in meta
  erich$Time.Point <- Bac.r2.sd$Time.Point
  erich$Parent.ID <- Bac.r2.sd$Parent.ID
  erich$ID <- Bac.r2.sd$ID
  erich$Species <- Bac.r2.sd$Species
  erich$Time.Point = factor(erich$Time.Point, levels = c("T0", "T1", "TF"))

  
  
 
#stats ALL THE DATA
 library(ggpubr)
  #look at data
  
#observed
# hist(erich$Observed) 
# ggqqplot(erich$Observed)
# erich$Observed_log<-log10(erich$Observed)
# hist(erich$Observed_log) 

## subset by coral species for OBSERVED
erich.mcap<-subset(erich, Species=="Montipora_capitata")
  hist(erich.mcap$Observed) 
  # hist(erich.mcap$Observed_log) 

erich.pcomp<-subset(erich, Species=="Porites_compressa")
  hist(erich.pcomp$Observed) #meh
  #   hist(erich.pcomp$Observed_log) #Better 
  # ggqqplot(erich.pcomp$Observed_log) 
 
erich.pvar<-subset(erich, Species=="Pavona_varians")
 hist(erich.pvar$Observed) #
 # hist(erich.pvar$Observed_log) #better
 #  ggqqplot(erich.pvar$Observed_log) #

erich.pacu<-subset(erich, Species=="Pocillopora_acuta")
 hist(erich.pacu$Observed) 
  hist(erich.pacu$Observed_log) 
  ggqqplot(erich.pacu$Observed_log) 

  
## subset for Shannon
#shannon
hist(erich$Shannon)
ggqqplot(erich$Shannon)
# erich$Shannon_log<-log10(erich$Shannon)
# hist(erich$Shannon_log)
# ggqqplot(erich$Shannon_log)
# erich$Shannon_sq<-sqrt(erich$Shannon)
# hist(erich$Shannon_sq)
# ggqqplot(erich$Shannon_sq)


erich.mcap<-subset(erich, Species=="Montipora_capitata")
  hist(erich.mcap$Shannon) # 
  ggqqplot(erich.mcap$Shannon) #
  #   hist(erich.mcap$Shannon_log) # 
  # ggqqplot(erich.mcap$Shannon) #
 
erich.pcomp<-subset(erich, Species=="Porites_compressa")
  hist(erich.pcomp$Shannon) # normal 
  ggqqplot(erich.pcomp$Shannon) 
  # hist(erich.pcomp$Shannon_log) # normal 
  # ggqqplot(erich.pcomp$Shannon) 

erich.pvar<-subset(erich, Species=="Pavona_varians")
 hist(erich.pvar$Shannon) #not normal 
  ggqqplot(erich.pvar$Shannon) #
  
erich.pacu<-subset(erich, Species=="Pocillopora_acuta")
 hist(erich.pacu$Shannon) #not awfuk 
  ggqqplot(erich.pacu$Shannon) #not bad
  

### ALL SPECIES
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.species = aov(Shannon ~ Species, data=erich)

#Call for the summary of that ANOVA, which will include P-values
  summary(aov.shannon.species) # sig diff between species
  TukeyHSD(aov.shannon.species)

#Plot
  boxplot(Shannon ~ Species, data=erich, ylab="Shannon's diversity") 

  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.species = aov(Observed ~ Species*Treatment*Time.Point, data=erich)
  #aov.observed.species = aov(Observed ~ Species, data=erich)

#Call for the summary of that ANOVA, which will include P-values
  summary(aov.observed.species) # sig diff between species
  TukeyHSD(aov.observed.species)

#Plot
  boxplot(Observed ~ Species, data=erich, ylab="observed's diversity") 
  
#### LHK paper code
  
alphaplot <- plot_richness(Bac.r2.alpha, x="Species", measures=c("Observed","Shannon", "Simpson"), color="Time.Point") + 
  labs(x="Genet") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(strip.text.x = element_text(size=10, face="bold"),
        strip.background = element_rect(fill="#FFFFFF"))


alphaplot


#stats
erich <- estimate_richness(Bac.r2.alpha, measures=c("Observed","Shannon", "Simpson"))
#make dataframe
SampleID <- row.names(erich) 
# makes a new dataframe with the specified columnes
alpha <- data.frame(SampleID, erich) 
s <- data.frame(sample_data(Bac.r2.alpha)) 
# Change first column title to SampleID to match distances dataframe
colnames(s)[1] <- "SampleID" 
# merges metadata df and distances df
alphaGC <- merge(alpha, s, by = "SampleID") 

```

### MCAP alpha, df: erich.tab2.mcap
```{r} 
#ANOVA
erich.tab2.mcap$treat.time<-paste(erich.tab2.mcap$Treatment,erich.tab2.mcap$Time.Point)
erich.tab2.mcap$treat.time<-as.factor(erich.tab2.mcap$treat.time)

#shannon
  aov.shannon.treat.time = aov(diversity_shannon ~ treat.time, data=erich.tab2.mcap)
  summary(aov.shannon.treat.time) # sig diff between treat.time
  TukeyHSD(aov.shannon.treat.time)
#observed  
  aov.observed.treat.time = aov(observed ~ treat.time, data=erich.tab2.mcap)
  summary(aov.observed.treat.time) # sig diff between treat.time
  TukeyHSD(aov.observed.treat.time)
#evenness
  aov.evenness.treat.time = aov(evenness_pielou ~ treat.time, data=erich.tab2.mcap)
  summary(aov.evenness.treat.time) # sig diff between treat.time
  TukeyHSD(aov.evenness.treat.time)  
#Plot
  boxplot(diversity_shannon ~ treat.time, data=erich.tab2.mcap, ylab="Shannon's diversity") 
  boxplot(observed ~ treat.time, data=erich.tab2.mcap, ylab="Observed") 
  boxplot(evenness_pielou ~ treat.time, data=erich.tab2.mcap, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.treat.time <- levels(erich.tab2.mcap$treat.time) # get the variables

# make a pairwise list that we want to compare.
  al.treat.time.pairs <- combn(seq_along(al.treat.time), 2, simplify = FALSE, FUN = function(i)al.treat.time[i])
  print(al.treat.time.pairs)

#observed
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "observed",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
    
#Shannon
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "diversity_shannon",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "evenness_pielou",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)






#OLD

#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesMcap = aov(Shannon ~ Treatment*Time.Point, data=erich.mcap)
  summary(aov.shannon.speciesMcap) # sig diff between species
  TukeyHSD(aov.shannon.speciesMcap)

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.mcap, ylab="Shannon's diversity") 

  
  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesMcap = aov(Observed ~ Treatment*Time.Point, data=erich.mcap)
  summary(aov.observed.speciesMcap) # sig diff between species
  TukeyHSD(aov.observed.speciesMcap)
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.mcap, ylab="observed's diversity") 
```
#### Do by C and CD
```{r}
#mcap  erich.mcap


  # by parent ID set clade dom
  erich.mcap$Clade[erich.mcap$Parent.ID==363]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==364]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==365]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==366]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==367]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==368]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==369]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==370]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==371]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==372]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==373]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==374]<-"CD"
  erich.mcap$Clade<-as.factor(erich.mcap$Clade) #change to factor

## let look at C vs CD
  #Run the ANOVA and save it as an object
  aov.shannon.speciesMcap_CCD = aov(Shannon ~ Clade*Treatment*Time.Point, data=erich.mcap)
  summary(aov.shannon.speciesMcap_CCD) # sig diff between species
  TukeyHSD(aov.shannon.speciesMcap_CCD)

#Plot
  boxplot(Shannon ~ Clade*Treatment*Time.Point, data=erich.mcap, ylab="Shannon's diversity") 

#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesMcap_CCD = aov(Observed ~ Clade*Treatment*Time.Point, data=erich.mcap)
  summary(aov.observed.speciesMcap_CCD) # sig diff between species
  TukeyHSD(aov.observed.speciesMcap_CCD)
#Plot
  boxplot(Observed ~ Clade*Treatment*Time.Point, data=erich.mcap, ylab="observed's diversity") 
  
``` 
# Pcomp alpha
```{r} 
#ANOVA
erich.tab2.pcomp$treat.time<-paste(erich.tab2.pcomp$Treatment,erich.tab2.pcomp$Time.Point)
erich.tab2.pcomp$treat.time<-as.factor(erich.tab2.pcomp$treat.time)

#shannon
  aov.shannon.treat.time = aov(diversity_shannon ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.shannon.treat.time) # sig diff between treat.time
  TukeyHSD(aov.shannon.treat.time)
#observed  
  aov.observed.treat.time = aov(observed ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.observed.treat.time) # sig diff between treat.time
  TukeyHSD(aov.observed.treat.time)
#evenness
  aov.evenness.treat.time = aov(evenness_pielou ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.evenness.treat.time) # sig diff between treat.time
  TukeyHSD(aov.evenness.treat.time)  
#Plot
  boxplot(diversity_shannon ~ treat.time, data=erich.tab2.pcomp, ylab="Shannon's diversity") 
  boxplot(observed ~ treat.time, data=erich.tab2.pcomp, ylab="Observed") 
  boxplot(evenness_pielou ~ treat.time, data=erich.tab2.pcomp, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.treat.time <- levels(erich.tab2.pcomp$treat.time) # get the variables

# make a pairwise list that we want to compare.
  al.treat.time.pairs <- combn(seq_along(al.treat.time), 2, simplify = FALSE, FUN = function(i)al.treat.time[i])
  print(al.treat.time.pairs)

#observed
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "observed",
   add = "boxplot", fill = "treat.time", palette = c(Pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
    
#Shannon
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "diversity_shannon",
   add = "boxplot", fill = "treat.time", palette = c(pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "evenness_pielou",
   add = "boxplot", fill = "treat.time", palette = c(pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)















#old
#Shannon 
#Run the ANOVA and save it as an object
  aov.shannon.speciesPcomp = aov(Shannon ~ Treatment*Time.Point, data=erich.pcomp)
  summary(aov.shannon.speciesPcomp) # sig diff between species
  TukeyHSD(aov.shannon.speciesPcomp)
  
 
 

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pcomp, ylab="Shannon's diversity") 

#Pcomp  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPcomp = aov(Observed ~ Treatment*Time.Point, data=erich.pcomp)
  summary(aov.observed.speciesPcomp) # sig diff between species
  TukeyHSD(aov.observed.speciesPcomp)
  


  
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.pcomp, ylab="observed's diversity") 


  
``` 
# Pvar
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesPvar = aov(Shannon ~ Treatment*Time.Point, data=erich.pvar)
  summary(aov.shannon.speciesPvar) # sig diff between species
  TukeyHSD(aov.shannon.speciesPvar)

  

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pvar, ylab="Shannon's diversity") 
  
 
  
  
  
  
  
  

#Pvar  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPvar = aov(Observed ~ Treatment*Time.Point, data=erich.pvar)
  summary(aov.observed.speciesPvar) # sig diff between species
  TukeyHSD(aov.observed.speciesPvar)


  
#Plot
  boxplot(log(Observed) ~ Treatment*Time.Point, data=erich.pvar, ylab="observed's diversity") 
  
  
``` 

# Pacu
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesPacu = aov(Shannon ~ Treatment*Time.Point, data=erich.pacu)
  summary(aov.shannon.speciesPacu) # sig diff between species
  TukeyHSD(aov.shannon.speciesPacu)


#Plot
  aov.shannon.speciesPacu$Time.Point = factor(aov.shannon.speciesPacu$Time.Point, levels = c("T0", "T1", "TF"))

  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pacu, ylab="Shannon's diversity") 
  
 
  
  
  
  
  
  

#Pvar  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPacu = aov(Observed ~ Treatment*Time.Point, data=erich.pacu)
  summary(aov.observed.speciesPacu) # sig diff between species
  TukeyHSD(aov.observed.speciesPacu)


  
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.pacu, ylab="observed's diversity") 

# Just T1
  erich.pacuT1<-subset(erich.pacu, Time.Point=="T1")
  hist(log10(erich.pacuT1$Observed))
  aov.observed.speciesPacuT1 = aov(log10(Observed) ~ Treatment, data=erich.pacuT1)
  summary(aov.observed.speciesPacuT1) # sig diff between species
  TukeyHSD(aov.observed.speciesPacuT1)
  
  hist((erich.pacuT1$Shannon))
  aov.shannon.speciesPacuT1 = aov(log10(Shannon) ~ Treatment, data=erich.pacuT1)
  summary(aov.shannon.speciesPacuT1) # sig diff between species
  TukeyHSD(aov.shannon.speciesPacuT1)
  
  
``` 
 





 
```{r}
#for non normal Kruskal wallis rank sum  
  kruskal.test(x ~ AgeGroup, data=meta)

  
#testing
  
  coral <- get_variable(coralDIV2_rare.alpha2, "host_genus") %in% c("montipora",  "pocillopora")
sample_data(coralDIV2_rare.alpha2)$human <- factor(coral)

 #estimate_richness(coralDIV2_rare, "coral","host_genus", measures = c("Observed", "Shannon")) #select which estimates and estimate CANNOT GET TO WORK with X
  
## from tutorial can't get it to wokr in my code.   
  GP <- prune_taxa(taxa_sums(GlobalPatterns) > 0, GlobalPatterns)

human <- get_variable(GP, "SampleType") %in% c("Feces", "Mock", "Skin", "Tongue")
sample_data(GP)$human <- factor(human)

alpha_meas = c("Observed", "Shannon")
(p <- plot_richness(GP, "human", "SampleType", measures=alpha_meas))






##
## make bar plot of rank abundance of all DIVs
par(mar = c(10, 4, 4, 2) + 0.1) # make more room on bottom margin
N <- 50
barplot(sort(taxa_sums(coralDIV2_rare.alpha2), TRUE)[1:N]/nsamples(coralDIV2_rare.alpha2), las=2)

sample_variables(coralDIV2_rare.alpha2) #check sample variables

#plot_bar(coralDIV2_rare.alpha2, "Species", fill="DIV", facet_grid=~DIV)


#make network - not helpful for ITS2 here

ig <- make_network(coralDIV2_rare.alpha2, max.dist=0.3)
plot_network(ig, coralDIV2_rare.alpha2, color="Parent.ID", shape="Species", line_weight=0.4, label=NULL)






```






# for defense NMDS
```{r}
AA_16S<-Bac.s.rel.u.T1F #copy df

fullNMDS<-AA_16S
fullNMDS = prune_taxa(taxa_sums(fullNMDS) > 0, fullNMDS) #this removes any OTU with 0s


# extract OTU table with metadata. 

# Extract abundance matrix from the phyloseq object< 
#OK some of your samples now have zero taxa. 
OTU.16S.v = as(otu_table(fullNMDS), "matrix")
# transpose if necessary
if(taxa_are_rows(fullNMDS)){OTU.16S.v <- t(OTU.16S.v)}
# Coerce to data.frame
OTU.16S.v.df = as.data.frame(OTU.16S.v)
    sample_data<-rownames(OTU.16S.v.df)  #make row names a column
  rownames(OTU.16S.v.df) <- NULL
  OTU.16S.v.df2 <- cbind(sample_data,OTU.16S.v.df)



#smash in meta in vegan, multivariate stats
Asample.data16S <- as(sample_data(fullNMDS), "data.frame") #create sample data frame to view

sample_data<-rownames(Asample.data16S)  #get meta organized, row names a column
  rownames(Asample.data16S) <- NULL
  Asample.data16S2 <- cbind(sample_data,Asample.data16S)
  

B16S.data<-merge(Asample.data16S2,OTU.16S.v.df2) #merge on the metadata to OTU table. Bam

B16S.data<-subset(B16S.data, Time.Point=="T1"|Time.Point=="TF")
#CW code
# cleaned data for full dataframe NMDS (Site, Period, dom)

B16S.data$SpTrTi<-interaction(B16S.data$Species, B16S.data$Treatment, B16S.data$Time.Point)

 fac.B16S.data<-B16S.data[,c("Treatment","Species", "Time.Point", "SpTrTi")] #  sans responses

resp.B16S.data<-B16S.data[, 13:6461] # just responses

# resp.B16S.data <- scale(resp.B16S.data, center = TRUE, scale = TRUE) 
# 
# ##  Run MANOVA 
# # all.Manova<-adonis2(resp.B16S.data~Species*Treatment*Time.Point, data=B16S.data, permutations=1000,  method="euclidian")
# # all.Manova
# 

allNMDS<-metaMDS(resp.B16S.data, distance='bray', k=3, trymax=300)
# 
# stressplot(allNMDS)
# 
# ##### next  ####
# 
allNMDS.df<-data.frame(x=allNMDS$point[,1], y=allNMDS$point[,2],
                            Species=as.factor(fac.B16S.data[,2]),
                            Treatment=as.factor(fac.B16S.data[,1]),
                            Time.Point=as.factor(fac.B16S.data[,3]),
                            SpTrTi=as.factor(fac.B16S.data[,4]))

colnames(allNMDS.df)[1:2]<-c("MDS1", "MDS2")


# scores.ord.B16S_RelA_stats  df from above nmds you can't get it to work here. 

#centroid means scores.ord.B16S_RelA_stats
NMDS.mean=aggregate(allNMDS.df[,1:2],list(group=allNMDS.df$SpTrTi), mean) # mean by species*treatment*time

############### All groups NMDS
# B16S.data$Code<-droplevels(B16S.data$Code)
groups<-B16S.data$Code
groups2<-c("Montipora_capitata Ambient T1", 
           "Montipora_capitata Ambient TF", 
           "Montipora_capitata High T1",
           "Montipora_capitata High TF", 
           "Porites_compressa Ambient T1", 
           "Porites_compressa Ambient TF", 
           "Porites_compressa High T1",
           "Porites_compressa High TF",
           "Pavona_varians Ambient T1", 
           "Pavona_varians Ambient TF", 
           "Pavona_varians High T1",
           "Pavona_varians High TF",
           "Pocillopora_acuta Ambient T1", 
           "Pocillopora_acuta Ambient TF", 
           "Pocillopora_acutaa High T1",
           "Pocillopora_acuta High TF")
colors=c(
  "#F8766D", #Mcap 
  "#C77CFF", #Porites
  "#7CAE00", # Pvar
  "#00BFC4" # Pacu
   ) 
plot<-ordiplot(allNMDS, type="n", cex.main=1, display="sites", cex.lab=0.8, cex.axis=0.8) #ylim=c(-0.3, 0.3))
axis(side = 1, labels = FALSE)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
ordihull(plot, groups=B16S.data$SpTrTi, draw="polygon", alpha=20, col=c("#F8766D", "#7CAE00","#00BFC4","#C77CFF"), border=F)

# points for start and end
with(plot, points(NMDS.mean[5,2], NMDS.mean[5,3], bg="white", col=colors[1], pch=21, cex=2, lwd=1)) #mcap T1 H
with(plot, points(NMDS.mean[1,2], NMDS.mean[1,3], col=colors[1], pch=16, cex=2, lwd=1)) # Mcap A T1
with(plot, points(NMDS.mean[13,2], NMDS.mean[13,3], bg="white",col=colors[1], pch=24, cex=2, lwd=1)) #mcap TF H
with(plot, points(NMDS.mean[9,2], NMDS.mean[9,3], col=colors[1], pch=17, cex=2, lwd=1)) # Mcap A TF

with(plot, points(NMDS.mean[8,2], NMDS.mean[8,3], bg="white", col=colors[2], pch=21, cex=2, lwd=1)) # Pcomp H T1
with(plot, points(NMDS.mean[4,2], NMDS.mean[4,3], col=colors[2], pch=16, cex=2, lwd=1)) # Pcomp  A T1
with(plot, points(NMDS.mean[16,2], NMDS.mean[16,3], bg="white", col=colors[2], pch=24, cex=2, lwd=1)) # Pcomp H TF
with(plot, points(NMDS.mean[12,2], NMDS.mean[12,3], col=colors[2], pch=17, cex=2, lwd=1)) # PcompA TF

with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], bg="white", col=colors[3], pch=21, cex=2, lwd=1)) # Pvar H T1
with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[3], pch=16, cex=2, lwd=1)) # Pvar A T1
with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], bg="white", col=colors[3], pch=24, cex=2, lwd=1)) # Pvar H TF
with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=17, cex=2, lwd=1)) # Pvar A TF

with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], bg="white", col=colors[4], pch=21, cex=2, lwd=1)) # Pacu H T1
with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[4], pch=16, cex=2, lwd=1)) # Pac A Ta
with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], bg="white", col=colors[4], pch=24, cex=2, lwd=1)) # Pacu H TF
with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[4], pch=17, cex=2, lwd=1)) # Pac A TF



# # points for others (if want)
# with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D
# with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D

# # lines for start and end
# with(plot, lines(NMDS.mean[c(1,9),2], NMDS.mean[c(1,9),3], lwd=1.5, col =colors[1], lty=1))# Mcap A
# with(plot, lines(NMDS.mean[c(5,13),2], NMDS.mean[c(5,13),3], lwd=1.5, col =colors[1], lty=2))# Mcap H
# with(plot, lines(NMDS.mean[c(8,16),2], NMDS.mean[c(8,16),3], lwd=1.5, col =colors[2], lty=2))# Pcomp H
# with(plot, lines(NMDS.mean[c(4,12),2], NMDS.mean[c(4,12),3], lwd=1.5, col =colors[2], lty=1))# Pcomp A
# with(plot, lines(NMDS.mean[c(6, 14),2], NMDS.mean[c(6,14),3], lwd=1.5, col =colors[3], lty=2))# Pvar H
# with(plot, lines(NMDS.mean[c(2,10),2], NMDS.mean[c(2,10),3], lwd=1.5, col =colors[3], lty=1))# Pvar A
# with(plot, lines(NMDS.mean[c(7,15),2], NMDS.mean[c(7,15),3], lwd=1.5, col =colors[4], lty=2))# Pacu H
# with(plot, lines(NMDS.mean[c(3,11),2], NMDS.mean[c(3,11),3], lwd=1.5, col =colors[4], lty=1))# Pacu A

# add labels
# text(0, -4, "Montipora capitata", cex=0.8, col=colors[1])
# text(-5, -3, "Porites compressa", cex=0.8, col=colors[2])
# text(4, -3, "Pavona varians", cex=0.8, col=colors[3])
# text(4, 2, "Pocillopora acuta", cex=0.8, col=colors[4])

text(-3, -1.8, "2D Stress = 0.15", cex=0.8, col="black")
legend("topright", legend=groups2, inset=c(0.03, -0.01), x.intersp=0.6, y.intersp=0.9, cex=0.7, pch=16, col=colors, lty=c(1,3,1,3), seg.len=2, pt.cex=1, bty="n")

dev.copy(pdf, "ITSNMDS.pdf", height=5.5, width=6, useDingbats=FALSE)
dev.off() 

```


#try again with weighted unifrac _ You are stucks....
```{r}
Bac.s.rel.u.T1TF2<-subset_samples(Bac.s.rel.u, Time.Point=="T1"|Time.Point=="TF")
    Bac.s.rel.u.T1TF2 = prune_taxa(taxa_sums(Bac.s.rel.u.T1TF2) > 0, Bac.s.rel.u.T1TF2) #this removes any OTU with 0s

unifrac.all.T1TF <- ordinate(Bac.s.rel.u.T1TF2, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T1TF)
scores.ord.unifrac.all.T1TF<-as.data.frame(cbind(vegan::scores(unifrac.all.T1TF, display="sites"))) #  
     sd.unifrac.T1TF2 <- as(sample_data(Bac.s.rel.u.T1TF2), "data.frame") #sample df

scores.ord.unifrac.all.T1TF$Treatment <- sd.unifrac.T1TF2$Treatment
scores.ord.unifrac.all.T1TF$Species <- sd.unifrac.T1TF2$Species
scores.ord.unifrac.all.T1TF$Clade <- sd.unifrac.T1TF2$Clade
scores.ord.unifrac.all.T1TF$Time.Point <- sd.unifrac.T1TF2$Time.Point
scores.ord.unifrac.all.T1TF$Code <- paste(sd.unifrac.T1TF2$Species, sd.unifrac.T1TF2$Treatment,sd.unifrac.T1TF2$Time.Point)



OTU.16.v = as(otu_table(Bac.s.rel.u.T1TF2), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.rel.u.T1TF2)){OTU.16.v <- t(OTU.16.v)}
# Coerce to data.frame
OTU.16.v.df = as.data.frame(OTU.16.v)
    sample_data16<-rownames(OTU.16.v.df)  #make row names a column
  rownames(OTU.16.v.df) <- NULL
  OTU.16.v.df2 <- cbind(sample_data16,OTU.16.v.df)



#smash in meta in vegan, multivariate stats
Asample.data16 <- as(sample_data(Bac.s.rel.u.T1TF2), "data.frame") #create sample data frame to view

sample_data<-rownames(Asample.data16)  #get meta organized, row names a column
  rownames(Asample.data16) <- NULL
  Asample.data162 <- cbind(sample_data,Asample.data16)
  

B16S.data<-merge(Asample.data162,OTU.16.v.df2) #merge on the metadata to OTU table. Bam

B16S.data<-subset(B16S.data, Time.Point=="T1"|Time.Point=="TF")
#CW code
# cleaned data for full dataframe NMDS (Site, Period, dom)

B16S.data$SpTrTi<-interaction(B16S.data$Species, B16S.data$Treatment, B16S.data$Time.Point)




# ##### next  ####
# 
# allNMDS.df<-data.frame(x=allNMDS$point[,1], y=allNMDS$point[,2],
#                             Species=as.factor(fac.B16S.data[,2]),
#                             Treatment=as.factor(fac.B16S.data[,1]),
#                             Time.Point=as.factor(fac.B16S.data[,3]),
#                             SpTrTi=as.factor(fac.B16S.data[,4]))

# colnames(allNMDS.df)[1:2]<-c("MDS1", "MDS2")


# scores.ord.B16S_RelA_stats  df from above nmds you can't get it to work here. 
#centroid means scores.ord.B16S_RelA_stats
NMDS.mean=aggregate(scores.ord.unifrac.all.T1TF[,1:2],list(group=B16S.data$Code), mean) # mean by species*treatment*time

############### All groups NMDS
# B16S.data$Code<-droplevels(B16S.data$Code)
groups<-B16S.data$Code
groups2<-c("Montipora_capitata Ambient T1", 
           "Montipora_capitata Ambient TF", 
           "Montipora_capitata High T1",
           "Montipora_capitata High TF", 
           "Porites_compressa Ambient T1", 
           "Porites_compressa Ambient TF", 
           "Porites_compressa High T1",
           "Porites_compressa High TF",
           "Pavona_varians Ambient T1", 
           "Pavona_varians Ambient TF", 
           "Pavona_varians High T1",
           "Pavona_varians High TF",
           "Pocillopora_acuta Ambient T1", 
           "Pocillopora_acuta Ambient TF", 
           "Pocillopora_acutaa High T1",
           "Pocillopora_acuta High TF")
colors=c(
  "#F8766D", #Mcap 
  "#C77CFF", #Porites
  "#7CAE00", # Pvar
  "#00BFC4" # Pacu
   ) 
plot<-ordiplot(unifrac.all.T1TF, type="n", cex.main=1, display="sites", cex.lab=0.8, cex.axis=0.8) #ylim=c(-0.3, 0.3))
axis(side = 1, labels = FALSE)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
ordihull(plot, groups=unifrac.all.T1TF$Code, draw="polygon", alpha=20, col=c("#F8766D", "#7CAE00","#00BFC4","#C77CFF"), border=F)

# points for start and end
with(plot, points(NMDS.mean[5,2], NMDS.mean[5,3], bg="white", col=colors[1], pch=21, cex=2, lwd=1)) #mcap T1 H
with(plot, points(NMDS.mean[1,2], NMDS.mean[1,3], col=colors[1], pch=16, cex=2, lwd=1)) # Mcap A T1
with(plot, points(NMDS.mean[13,2], NMDS.mean[13,3], bg="white",col=colors[1], pch=24, cex=2, lwd=1)) #mcap TF H
with(plot, points(NMDS.mean[9,2], NMDS.mean[9,3], col=colors[1], pch=17, cex=2, lwd=1)) # Mcap A TF

with(plot, points(NMDS.mean[8,2], NMDS.mean[8,3], bg="white", col=colors[2], pch=21, cex=2, lwd=1)) # Pcomp H T1
with(plot, points(NMDS.mean[4,2], NMDS.mean[4,3], col=colors[2], pch=16, cex=2, lwd=1)) # Pcomp  A T1
with(plot, points(NMDS.mean[16,2], NMDS.mean[16,3], bg="white", col=colors[2], pch=24, cex=2, lwd=1)) # Pcomp H TF
with(plot, points(NMDS.mean[12,2], NMDS.mean[12,3], col=colors[2], pch=17, cex=2, lwd=1)) # PcompA TF

with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], bg="white", col=colors[3], pch=21, cex=2, lwd=1)) # Pvar H T1
with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[3], pch=16, cex=2, lwd=1)) # Pvar A T1
with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], bg="white", col=colors[3], pch=24, cex=2, lwd=1)) # Pvar H TF
with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=17, cex=2, lwd=1)) # Pvar A TF

with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], bg="white", col=colors[4], pch=21, cex=2, lwd=1)) # Pacu H T1
with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[4], pch=16, cex=2, lwd=1)) # Pac A Ta
with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], bg="white", col=colors[4], pch=24, cex=2, lwd=1)) # Pacu H TF
with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[4], pch=17, cex=2, lwd=1)) # Pac A TF



# # points for others (if want)
# with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D
# with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D

# lines for start and end
with(plot, lines(NMDS.mean[c(1,9),2], NMDS.mean[c(1,9),3], lwd=1.5, col =colors[1], lty=1))# Mcap A
with(plot, lines(NMDS.mean[c(5,13),2], NMDS.mean[c(5,13),3], lwd=1.5, col =colors[1], lty=2))# Mcap H
with(plot, lines(NMDS.mean[c(8,16),2], NMDS.mean[c(8,16),3], lwd=1.5, col =colors[2], lty=2))# Pcomp H
with(plot, lines(NMDS.mean[c(4,12),2], NMDS.mean[c(4,12),3], lwd=1.5, col =colors[2], lty=1))# Pcomp A
with(plot, lines(NMDS.mean[c(6, 14),2], NMDS.mean[c(6,14),3], lwd=1.5, col =colors[3], lty=2))# Pvar H
with(plot, lines(NMDS.mean[c(2,10),2], NMDS.mean[c(2,10),3], lwd=1.5, col =colors[3], lty=1))# Pvar A
with(plot, lines(NMDS.mean[c(7,15),2], NMDS.mean[c(7,15),3], lwd=1.5, col =colors[4], lty=2))# Pacu H
with(plot, lines(NMDS.mean[c(3,11),2], NMDS.mean[c(3,11),3], lwd=1.5, col =colors[4], lty=1))# Pacu A

# add labels
# text(0, -4, "Montipora capitata", cex=0.8, col=colors[1])
# text(-5, -3, "Porites compressa", cex=0.8, col=colors[2])
# text(4, -3, "Pavona varians", cex=0.8, col=colors[3])
# text(4, 2, "Pocillopora acuta", cex=0.8, col=colors[4])

text(-30, -22, "2D Stress = 0.15", cex=0.8, col="black")
legend("topright", legend=groups2, inset=c(0.03, -0.01), x.intersp=0.6, y.intersp=0.9, cex=0.7, pch=16, col=colors, lty=c(1,3,1,3), seg.len=2, pt.cex=1, bty="n")

dev.copy(pdf, "ITSNMDS.pdf", height=5.5, width=6, useDingbats=FALSE)
dev.off() 

```

```{r}
Bac.s.rel.u.T0<-subset_samples(Bac.s.rel.u, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s

unifrac.all.T0 <- ordinate(Bac.s.rel.u.T0, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T0)
scores.ord.unifrac.all.T0<-as.data.frame(cbind(vegan::scores(unifrac.all.T0, display="sites"))) #  
     sd.unifrac.T0 <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df

scores.ord.unifrac.all.T0$Treatment <- sd.unifrac.T0$Treatment
scores.ord.unifrac.all.T0$Species <- sd.unifrac.T0$Species
scores.ord.unifrac.all.T0$Clade <- sd.unifrac.T0$Clade
scores.ord.unifrac.all.T0$Time.Point <- sd.unifrac.T0$Time.Point
scores.ord.unifrac.all.T0$Code <- paste(sd.unifrac.T0$Species, sd.unifrac.T0$Treatment,sd.unifrac.T0$Time.Point)



OTU.16.v = as(otu_table(Bac.s.rel.u.T0), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.rel.u.T0)){OTU.16.v <- t(OTU.16.v)}
# Coerce to data.frame
OTU.16.v.df = as.data.frame(OTU.16.v)
    sample_data16<-rownames(OTU.16.v.df)  #make row names a column
  rownames(OTU.16.v.df) <- NULL
  OTU.16.v.df2 <- cbind(sample_data16,OTU.16.v.df)



#smash in meta in vegan, multivariate stats
Asample.data16 <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #create sample data frame to view

sample_data<-rownames(Asample.data16)  #get meta organized, row names a column
  rownames(Asample.data16) <- NULL
  Asample.data162 <- cbind(sample_data,Asample.data16)
  

B16S.data<-merge(Asample.data162,OTU.16.v.df2) #merge on the metadata to OTU table. Bam

B16S.data<-subset(B16S.data, Time.Point=="T1"|Time.Point=="TF")
#CW code
# cleaned data for full dataframe NMDS (Site, Period, dom)

B16S.data$SpTrTi<-interaction(B16S.data$Species, B16S.data$Treatment, B16S.data$Time.Point)




# ##### next  ####
# 
# allNMDS.df<-data.frame(x=allNMDS$point[,1], y=allNMDS$point[,2],
#                             Species=as.factor(fac.B16S.data[,2]),
#                             Treatment=as.factor(fac.B16S.data[,1]),
#                             Time.Point=as.factor(fac.B16S.data[,3]),
#                             SpTrTi=as.factor(fac.B16S.data[,4]))

# colnames(allNMDS.df)[1:2]<-c("MDS1", "MDS2")


# scores.ord.B16S_RelA_stats  df from above nmds you can't get it to work here. 
#centroid means scores.ord.B16S_RelA_stats
NMDS.mean=aggregate(scores.ord.unifrac.all.T0[,1:2],list(group=B16S.data$Code), mean) # mean by species*treatment*time

############### All groups NMDS
# B16S.data$Code<-droplevels(B16S.data$Code)
groups<-B16S.data$Code
groups2<-c("Montipora_capitata Ambient T1", 
           "Montipora_capitata Ambient TF", 
           "Montipora_capitata High T1",
           "Montipora_capitata High TF", 
           "Porites_compressa Ambient T1", 
           "Porites_compressa Ambient TF", 
           "Porites_compressa High T1",
           "Porites_compressa High TF",
           "Pavona_varians Ambient T1", 
           "Pavona_varians Ambient TF", 
           "Pavona_varians High T1",
           "Pavona_varians High TF",
           "Pocillopora_acuta Ambient T1", 
           "Pocillopora_acuta Ambient TF", 
           "Pocillopora_acutaa High T1",
           "Pocillopora_acuta High TF")
colors=c(
  "#F8766D", #Mcap 
  "#C77CFF", #Porites
  "#7CAE00", # Pvar
  "#00BFC4" # Pacu
   ) 
plot<-ordiplot(unifrac.all.T0, type="n", cex.main=1, display="sites", cex.lab=0.8, cex.axis=0.8) #ylim=c(-0.3, 0.3))
axis(side = 1, labels = FALSE)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
ordihull(plot, groups=unifrac.all.T0$Code, draw="polygon", alpha=20, col=c("#F8766D", "#7CAE00","#00BFC4","#C77CFF"), border=F)

# points for start and end
with(plot, points(NMDS.mean[5,2], NMDS.mean[5,3], bg="white", col=colors[1], pch=21, cex=2, lwd=1)) #mcap T1 H
with(plot, points(NMDS.mean[1,2], NMDS.mean[1,3], col=colors[1], pch=16, cex=2, lwd=1)) # Mcap A T1
with(plot, points(NMDS.mean[13,2], NMDS.mean[13,3], bg="white",col=colors[1], pch=24, cex=2, lwd=1)) #mcap TF H
with(plot, points(NMDS.mean[9,2], NMDS.mean[9,3], col=colors[1], pch=17, cex=2, lwd=1)) # Mcap A TF

with(plot, points(NMDS.mean[8,2], NMDS.mean[8,3], bg="white", col=colors[2], pch=21, cex=2, lwd=1)) # Pcomp H T1
with(plot, points(NMDS.mean[4,2], NMDS.mean[4,3], col=colors[2], pch=16, cex=2, lwd=1)) # Pcomp  A T1
with(plot, points(NMDS.mean[16,2], NMDS.mean[16,3], bg="white", col=colors[2], pch=24, cex=2, lwd=1)) # Pcomp H TF
with(plot, points(NMDS.mean[12,2], NMDS.mean[12,3], col=colors[2], pch=17, cex=2, lwd=1)) # PcompA TF

with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], bg="white", col=colors[3], pch=21, cex=2, lwd=1)) # Pvar H T1
with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[3], pch=16, cex=2, lwd=1)) # Pvar A T1
with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], bg="white", col=colors[3], pch=24, cex=2, lwd=1)) # Pvar H TF
with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=17, cex=2, lwd=1)) # Pvar A TF

with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], bg="white", col=colors[4], pch=21, cex=2, lwd=1)) # Pacu H T1
with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[4], pch=16, cex=2, lwd=1)) # Pac A Ta
with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], bg="white", col=colors[4], pch=24, cex=2, lwd=1)) # Pacu H TF
with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[4], pch=17, cex=2, lwd=1)) # Pac A TF



# # points for others (if want)
# with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D
# with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D

# lines for start and end
with(plot, lines(NMDS.mean[c(1,9),2], NMDS.mean[c(1,9),3], lwd=1.5, col =colors[1], lty=1))# Mcap A
with(plot, lines(NMDS.mean[c(5,13),2], NMDS.mean[c(5,13),3], lwd=1.5, col =colors[1], lty=2))# Mcap H
with(plot, lines(NMDS.mean[c(8,16),2], NMDS.mean[c(8,16),3], lwd=1.5, col =colors[2], lty=2))# Pcomp H
with(plot, lines(NMDS.mean[c(4,12),2], NMDS.mean[c(4,12),3], lwd=1.5, col =colors[2], lty=1))# Pcomp A
with(plot, lines(NMDS.mean[c(6, 14),2], NMDS.mean[c(6,14),3], lwd=1.5, col =colors[3], lty=2))# Pvar H
with(plot, lines(NMDS.mean[c(2,10),2], NMDS.mean[c(2,10),3], lwd=1.5, col =colors[3], lty=1))# Pvar A
with(plot, lines(NMDS.mean[c(7,15),2], NMDS.mean[c(7,15),3], lwd=1.5, col =colors[4], lty=2))# Pacu H
with(plot, lines(NMDS.mean[c(3,11),2], NMDS.mean[c(3,11),3], lwd=1.5, col =colors[4], lty=1))# Pacu A

# add labels
# text(0, -4, "Montipora capitata", cex=0.8, col=colors[1])
# text(-5, -3, "Porites compressa", cex=0.8, col=colors[2])
# text(4, -3, "Pavona varians", cex=0.8, col=colors[3])
# text(4, 2, "Pocillopora acuta", cex=0.8, col=colors[4])

text(-30, -22, "2D Stress = 0.15", cex=0.8, col="black")
legend("topright", legend=groups2, inset=c(0.03, -0.01), x.intersp=0.6, y.intersp=0.9, cex=0.7, pch=16, col=colors, lty=c(1,3,1,3), seg.len=2, pt.cex=1, bty="n")

dev.copy(pdf, "ITSNMDS.pdf", height=5.5, width=6, useDingbats=FALSE)
dev.off() 

```


T0 bray bc it works
```{r}
AA_16ST0<-Bac.s.rel.u.T0 #copy df

fullNMDS<-AA_16ST0
fullNMDS = prune_taxa(taxa_sums(fullNMDS) > 0, fullNMDS) #this removes any OTU with 0s


# extract OTU table with metadata. 

# Extract abundance matrix from the phyloseq object< 
#OK some of your samples now have zero taxa. 
OTU.16S.v = as(otu_table(fullNMDS), "matrix")
# transpose if necessary
if(taxa_are_rows(fullNMDS)){OTU.16S.v <- t(OTU.16S.v)}
# Coerce to data.frame
OTU.16S.v.df = as.data.frame(OTU.16S.v)
    sample_data<-rownames(OTU.16S.v.df)  #make row names a column
  rownames(OTU.16S.v.df) <- NULL
  OTU.16S.v.df2 <- cbind(sample_data,OTU.16S.v.df)



#smash in meta in vegan, multivariate stats
Asample.data16S <- as(sample_data(fullNMDS), "data.frame") #create sample data frame to view

sample_data<-rownames(Asample.data16S)  #get meta organized, row names a column
  rownames(Asample.data16S) <- NULL
  Asample.data16S2 <- cbind(sample_data,Asample.data16S)
  

B16S.data<-merge(Asample.data16S2,OTU.16S.v.df2) #merge on the metadata to OTU table. Bam

B16S.data<-subset(B16S.data, Time.Point=="T0")
#CW code
# cleaned data for full dataframe NMDS (Site, Period, dom)


 fac.B16S.data<-B16S.data[,c("Species","Clade")] #  sans responses

resp.B16S.data<-B16S.data[, 13:2449] # just responses

# resp.B16S.data <- scale(resp.B16S.data, center = TRUE, scale = TRUE) 
# 
# ##  Run MANOVA 
# # all.Manova<-adonis2(resp.B16S.data~Species*Treatment*Time.Point, data=B16S.data, permutations=1000,  method="euclidian")
# # all.Manova
# 
 
allNMDS<-metaMDS(resp.B16S.data, distance='bray', k=3, trymax=300)
# 
# stressplot(allNMDS)
# 
# ##### next  #### 
# 
allNMDS.df<-data.frame(x=allNMDS$point[,1], y=allNMDS$point[,2],
                            Species=as.factor(fac.B16S.data[,2]))

colnames(allNMDS.df)[1:2]<-c("MDS1", "MDS2")


# scores.ord.B16S_RelA_stats  df from above nmds you can't get it to work here. 

#centroid means scores.ord.B16S_RelA_stats
NMDS.mean=aggregate(allNMDS.df[,1:2],list(group=allNMDS.df$Species), mean) # mean by species*treatment*time

############### All groups NMDS
# B16S.data$Code<-droplevels(B16S.data$Code)
groups<-B16S.data$Species
groups2<-c("Montipora_capitata",
           "Porites_compressa",
           "Pavona_varians", 
           "Pocillopora_acuta")
colors=c(
  "#F8766D", #Mcap 
  "#C77CFF", #Porites
  "#7CAE00", # Pvar
  "#00BFC4" # Pacu
   ) 
plot<-ordiplot(allNMDS, type="n", cex.main=1, display="sites", cex.lab=0.8, cex.axis=0.8) #ylim=c(-0.3, 0.3))
axis(side = 1, labels = FALSE)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
ordihull(plot, groups=B16S.data$Species, draw="polygon", alpha=40, col=c("#F8766D", "#7CAE00","#00BFC4","#C77CFF"), border=F)

# points for start and end
with(plot, points(NMDS.mean[1,2], NMDS.mean[1,3],  col=colors[1], pch=18, cex=6, lwd=1)) #mcap C
with(plot, points(NMDS.mean[8,2], NMDS.mean[8,3], col=colors[1], pch=23, cex=5, lwd=2)) # McapCD
with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[3], pch=18, cex=6, lwd=1)) #pvar c1
with(plot, points(NMDS.mean[5,2], NMDS.mean[5,3], col=colors[3], pch=23, cex=5, lwd=2)) # pvar c27

with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3],  col=colors[2], pch=18, cex=6, lwd=1)) # pcomp cj
with(plot, points(NMDS.mean[4,2], NMDS.mean[4,3], col=colors[2], pch=23, cex=5, lwd=2)) # Pcomp  n
with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3],  col=colors[4], pch=18, cex=6, lwd=1)) # pacu c42
with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[4], pch=23, cex=5, lwd=2)) # Pacu ccc




# # points for others (if want)
# with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D
# with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D

# # lines for start and end
# with(plot, lines(NMDS.mean[c(1,9),2], NMDS.mean[c(1,9),3], lwd=1.5, col =colors[1], lty=1))# Mcap A
# with(plot, lines(NMDS.mean[c(5,13),2], NMDS.mean[c(5,13),3], lwd=1.5, col =colors[1], lty=2))# Mcap H
# with(plot, lines(NMDS.mean[c(8,16),2], NMDS.mean[c(8,16),3], lwd=1.5, col =colors[2], lty=2))# Pcomp H
# with(plot, lines(NMDS.mean[c(4,12),2], NMDS.mean[c(4,12),3], lwd=1.5, col =colors[2], lty=1))# Pcomp A
# with(plot, lines(NMDS.mean[c(6, 14),2], NMDS.mean[c(6,14),3], lwd=1.5, col =colors[3], lty=2))# Pvar H
# with(plot, lines(NMDS.mean[c(2,10),2], NMDS.mean[c(2,10),3], lwd=1.5, col =colors[3], lty=1))# Pvar A
# with(plot, lines(NMDS.mean[c(7,15),2], NMDS.mean[c(7,15),3], lwd=1.5, col =colors[4], lty=2))# Pacu H
# with(plot, lines(NMDS.mean[c(3,11),2], NMDS.mean[c(3,11),3], lwd=1.5, col =colors[4], lty=1))# Pacu A

# add labels
# text(0, -4, "Montipora capitata", cex=0.8, col=colors[1])
# text(-5, -3, "Porites compressa", cex=0.8, col=colors[2])
# text(4, -3, "Pavona varians", cex=0.8, col=colors[3])
# text(4, 2, "Pocillopora acuta", cex=0.8, col=colors[4])

text(-3, -1.8, "2D Stress = 0.15", cex=0.8, col="black")
legend("topright", legend=groups2, inset=c(0.03, -0.01), x.intersp=0.6, y.intersp=0.9, cex=0.7, pch=16, col=colors, lty=c(1,3,1,3), seg.len=2, pt.cex=1, bty="n")

dev.copy(pdf, "ITSNMDS.pdf", height=5.5, width=6, useDingbats=FALSE)
dev.off() 

```
 
#Mantel Test
import the its2 matrix from the ITS script

T0 only
```{r}
# figure out which samples are represented in the its2 and 16s dfs

# T0 only  #####

#set up 16s
    Bac.s.rel.u.T0<-subset_samples(Bac.s.rel.u, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s
  # look at sample names
    bac.T0.sd <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df
    #save just sample names
      bac.T0.sd.samps<-bac.T0.sd[['sample_name.1']]

#set up ITS2 T0, import  (already pruned)
    ITS2_T0_mantel <- readRDS("ITS2_T0_phyloseq.rds")
  # look at sample names
    ITS.T0.sd <- as(sample_data(ITS2_T0_mantel), "data.frame") #sample df
  #save just sample names
      ITS.T0.sd.samps<-ITS.T0.sd[['ID']]

# compare ITS to 16s at T0
  #identify missing samples btw datasets     
      setdiff(bac.T0.sd.samps,ITS.T0.sd.samps) #samples in 16s but not ITS: S208b, SN70b
      setdiff(ITS.T0.sd.samps,bac.T0.sd.samps) #samples in ITS but not 16s: "S208"  "SN60"  "SN70"  "S49"   "SN11"  "SN145" "SN40"  "SN111" "SN6"   "SN56"  "SN109"
        
      # in 16S change sample S208b name to S208

  # 16s changes: rename sample S208b = S208

    # pull OTU table
     OTU.16sT0 = as(otu_table(Bac.s.rel.u.T0), "matrix") #pull otu
       OTU.16sT0<-as.data.frame(OTU.16sT0) #make into df
        row.names(OTU.16sT0)[5] <- "S208"   #change row name by row number
                row.names(OTU.16sT0)[31] <- "SN70"   #change row name by row number

    #pull TAX table
     aTax.16sT0 = as(tax_table(Bac.s.rel.u.T0), "matrix") #pull tax
     
     #pull phy tree table
     tree.seq = phy_tree(Bac.s.rel.u.T0) #pull out tree

    #pull SAMPLE DATA
     aSD.16sT0 = as(sample_data(Bac.s.rel.u.T0), "matrix") #pull samp.data
     aSD.16sT0<-as.data.frame(aSD.16sT0) #change to df
        row.names(aSD.16sT0)[5] <- "S208"  #replace name S208b
        row.names(aSD.16sT0)[31] <- "SN70"  #replace name S208b
            

    #remake the phyloseq obj
      samdata = sample_data(aSD.16sT0)
        seqtab = otu_table(OTU.16sT0, taxa_are_rows = FALSE)
        taxtab = tax_table(aTax.16sT0)
        phy_tree = read_tree(phyTree.16sT0)
          Bac_T0_mantel = phyloseq(otu_table(seqtab), tax_table(taxtab), sample_data(samdata),phy_tree(tree.seq))
          
          
   #ITS2 - remove samps not in the 16s df - "SN60"  "S49"   "SN11"  "SN145" "SN40"  "SN111" "SN6"   "SN56"  "SN109" 
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN60" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "S49" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN11" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN145" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN40" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN111" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN6" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN56" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN109" )

        # both have 39 samples - Bac_T0_mantel ITS2_T0_mantel
      
        # reorder ps object so samples are in the same order
        library(microbiome)
        library(microViz)
        
        Bac_T0_mantel <- Bac_T0_mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac_T0_mantel) %>% head(8)
        
        ITS2_T0_mantel <- ITS2_T0_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS2_T0_mantel) %>% head(8)
        
  # MAKE data matrix
       library(vegan)
  
  #16s T0        
        unifrac.Bac.T0.mantel <- phyloseq::distance(Bac_T0_mantel, method = "wunifrac")
        
  #ITS2 T0
     bray.ITS2.T0.mantel <- phyloseq::distance(ITS2_T0_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T0.mantel, bray.ITS2.T0.mantel,  permutations=1000)
    
        
#### T0 Mcap only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for mcap
    Bac_T0_mantel.Mcap <- subset_samples(Bac_T0_mantel, Species == "Montipora_capitata")
    Bac_T0_mantel.Mcap = prune_taxa(taxa_sums(Bac_T0_mantel.Mcap) > 0, Bac_T0_mantel.Mcap) 
    
    ITS2_T0_mantel.Mcap  <- subset_samples(ITS2_T0_mantel, Species == "Montipora_capitata")
    ITS2_T0_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T0_mantel.Mcap) > 0, ITS2_T0_mantel.Mcap)
        
    #16s T0 mcap dist        
      unifrac.Bac_T0_mantel.Mcap <- phyloseq::distance(Bac_T0_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T0 mcap dist
     bray.ITS2.T0.mantel.Mcap <- phyloseq::distance(ITS2_T0_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Mcap, bray.ITS2.T0.mantel.Mcap,  permutations=1000)
    
#### T0 Pcomp only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pcomp
    Bac_T0_mantel.Pcomp <- subset_samples(Bac_T0_mantel, Species == "Porites_compressa")
    Bac_T0_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T0_mantel.Pcomp) > 0, Bac_T0_mantel.Pcomp) 
    
    ITS2_T0_mantel.Pcomp  <- subset_samples(ITS2_T0_mantel, Species == "Porites_compressa")
    ITS2_T0_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T0_mantel.Pcomp) > 0, ITS2_T0_mantel.Pcomp)
        
    #16s T0 Pcomp dist        
      unifrac.Bac_T0_mantel.Pcomp <- phyloseq::distance(Bac_T0_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T0 Pcomp dist
     bray.ITS2.T0.mantel.Pcomp <- phyloseq::distance(ITS2_T0_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pcomp, bray.ITS2.T0.mantel.Pcomp,  permutations=1000)        
        
        
  #### T0 Pvar only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pvar
    Bac_T0_mantel.Pvar <- subset_samples(Bac_T0_mantel, Species == "Pavona_varians")
    Bac_T0_mantel.Pvar = prune_taxa(taxa_sums(Bac_T0_mantel.Pvar) > 0, Bac_T0_mantel.Pvar) 
    
    ITS2_T0_mantel.Pvar  <- subset_samples(ITS2_T0_mantel, Species == "Pavona_varians")
    ITS2_T0_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T0_mantel.Pvar) > 0, ITS2_T0_mantel.Pvar)
        
    #16s T0 Pvar dist        
      unifrac.Bac_T0_mantel.Pvar <- phyloseq::distance(Bac_T0_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T0 Pvar dist
     bray.ITS2.T0.mantel.Pvar <- phyloseq::distance(ITS2_T0_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pvar, bray.ITS2.T0.mantel.Pvar,  permutations=1000)            
        
        
     #### T0 Pacu only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pacu
    Bac_T0_mantel.Pacu <- subset_samples(Bac_T0_mantel, Species == "Pavona_varians")
    Bac_T0_mantel.Pacu = prune_taxa(taxa_sums(Bac_T0_mantel.Pacu) > 0, Bac_T0_mantel.Pacu) 
    
    ITS2_T0_mantel.Pacu  <- subset_samples(ITS2_T0_mantel, Species == "Pavona_varians")
    ITS2_T0_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T0_mantel.Pacu) > 0, ITS2_T0_mantel.Pacu)
        
    #16s T0 Pacu dist        
      unifrac.Bac_T0_mantel.Pacu <- phyloseq::distance(Bac_T0_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T0 Pacu dist
     bray.ITS2.T0.mantel.Pacu <- phyloseq::distance(ITS2_T0_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pacu, bray.ITS2.T0.mantel.Pacu,  permutations=1000)            
        
        
    


```
T1 High only
```{r}
# figure out which samples are represented in the its2 and 16s dfs

# T1 only  #####

#high only 

#set up 16s
    Bac.s.rel.u.T1<-subset_samples(Bac.s.rel.u, Time.Point=="T1")
    Bac.s.rel.u.T1.High<-subset_samples(Bac.s.rel.u.T1, Treatment=="High")

    Bac.s.rel.u.T1.High = prune_taxa(taxa_sums(Bac.s.rel.u.T1.High) > 0, Bac.s.rel.u.T1.High) #this removes any OTU with 0s
  # look at sample names
    bac.T1.sd <- as(sample_data(Bac.s.rel.u.T1.High), "data.frame") #sample df
    #save just sample names
      bac.T1.High.sd.samps<-bac.T1.sd[['sample_name.1']]

#set up ITS2 T1, import  T1TF
    ITS2_T1TF_mantel <- readRDS("ITS2_T1TF_phyloseq.rds") 
     ITS2_T1_mantel<-subset_samples(ITS2_T1TF_mantel, Time.Point=="T1")
    ITS2_T1_mantel.High<-subset_samples(ITS2_T1_mantel, Treatment=="High")
      ITS2_T1_mantel.High = prune_taxa(taxa_sums(ITS2_T1_mantel.High) > 0, ITS2_T1_mantel.High) #this removes any OTU with 0s

    
  # look at sample names
    ITS.T1.High.sd <- as(sample_data(ITS2_T1_mantel.High), "data.frame") #sample df
  #save just sample names
      ITS.T1.High.sd.samps<-ITS.T1.High.sd[['ID']]

# compare ITS to 16s at T1
  #identify missing samples btw datasets     
      setdiff(bac.T1.High.sd.samps,ITS.T1.High.sd.samps) #samples in 16s but not ITS: "S1857" "S3469" "SN134"
      setdiff(ITS.T1.High.sd.samps,bac.T1.High.sd.samps) #samples in ITS but not 16s: "S1857b" "S3468"  "SN134b" "S670"   "S3025"  "S1388a"
        
      # in 16S remove S3469
      #in ITS change S1857b to S1857 and SN134b to SN134. Remove S3468, S670, S3025, S1388a

  # 16s changes: rename sample S3469
      
    Bac.T1.High.mantel <- subset_samples(Bac.s.rel.u.T1.High, sample_name.1 != "S3469" )
        #40 samples
          
   #ITS2 - remove samps not in the 16s df - " S3468, S670, S3025, S1388a
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S3468" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S670" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S3025" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S1388a" )

        
    #change names ITS2 S1857b to S1857 and SN134b to SN134

    # pull OTU table
     OTU.ITS2T1.High = as(otu_table(ITS2_T1_mantel.High), "matrix") #pull otu
       OTU.ITS2T1.High<-as.data.frame(OTU.ITS2T1.High) #make into df
        row.names(OTU.ITS2T1.High)[7] <- "S1857"   #change row name by row number
                row.names(OTU.ITS2T1.High)[17] <- "SN134"   #change row name by row number

    #pull TAX table
     aTax.ITS2T1.High = as(tax_table(ITS2_T1_mantel.High), "matrix") #pull tax

    #pull SAMPLE DATA
     aSD.ITS2T1.High = as(sample_data(ITS2_T1_mantel.High), "matrix") #pull samp.data
     aSD.ITS2T1.High<-as.data.frame(aSD.ITS2T1.High) #change to df
        row.names(aSD.ITS2T1.High)[7] <- "S1857"  #replace name S208b
        row.names(aSD.ITS2T1.High)[17] <- "SN134"  #replace name S208b
            

    #remake the phyloseq obj
      samdata.High = sample_data(aSD.ITS2T1.High)
        seqtab.High = otu_table(OTU.ITS2T1.High, taxa_are_rows = FALSE)
        taxtab.High = tax_table(aTax.ITS2T1.High)
          ITS_T1.High_mantel = phyloseq(otu_table(seqtab.High), tax_table(taxtab.High), sample_data(samdata.High))
             
        # both have 40 samples - 
      
        # reorder ps object so samples are in the same order
        
        Bac.T1.High.mantel <- Bac.T1.High.mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac.T1.High.mantel) %>% head(8)
        
        ITS_T1.High_mantel <- ITS_T1.High_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS_T1.High_mantel) %>% head(8)
        
  # MAKE data matrix
  
  #16s T1 High        
        unifrac.Bac.T1.High.mantel <- phyloseq::distance(Bac.T1.High.mantel, method = "wunifrac")
        
  #ITS2 T1 High
     bray.ITS2.T1.High.mantel <- phyloseq::distance(ITS_T1.High_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T1.High.mantel, bray.ITS2.T1.High.mantel,  permutations=1000)
    
        
#### T1.High Mcap only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for mcap
    Bac_T1.High_mantel.Mcap <- subset_samples(Bac.T1.High.mantel, Species == "Montipora_capitata")
    Bac_T1.High_mantel.Mcap = prune_taxa(taxa_sums(Bac_T1.High_mantel.Mcap) > 0, Bac_T1.High_mantel.Mcap) 
    
    ITS2_T1.High_mantel.Mcap  <- subset_samples(ITS_T1.High_mantel, Species == "Montipora_capitata")
    ITS2_T1.High_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Mcap) > 0, ITS2_T1.High_mantel.Mcap)
        
    #16s T1.High mcap dist        
      unifrac.Bac_T1.High_mantel.Mcap <- phyloseq::distance(Bac_T1.High_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T1.High mcap dist
     bray.ITS2.T1.High.mantel.Mcap <- phyloseq::distance(ITS2_T1.High_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Mcap, bray.ITS2.T1.High.mantel.Mcap,  permutations=1000)
    
#### T1.High Pcomp only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pcomp
    Bac_T1.High_mantel.Pcomp <- subset_samples(Bac.T1.High.mantel, Species == "Porites_compressa")
    Bac_T1.High_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pcomp) > 0, Bac_T1.High_mantel.Pcomp) 
    
    ITS2_T1.High_mantel.Pcomp  <- subset_samples(ITS_T1.High_mantel, Species == "Porites_compressa")
    ITS2_T1.High_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pcomp) > 0, ITS2_T1.High_mantel.Pcomp)
        
    #16s T1.High Pcomp dist        
      unifrac.Bac_T1.High_mantel.Pcomp <- phyloseq::distance(Bac_T1.High_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T1.High Pcomp dist
     bray.ITS2.T1.High.mantel.Pcomp <- phyloseq::distance(ITS2_T1.High_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pcomp, bray.ITS2.T1.High.mantel.Pcomp,  permutations=1000)        
        
    
#### T1.High Pvar only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pvar
    Bac_T1.High_mantel.Pvar <- subset_samples(Bac.T1.High.mantel, Species == "Pavona_varians")
    Bac_T1.High_mantel.Pvar = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pvar) > 0, Bac_T1.High_mantel.Pvar) 
    
    ITS2_T1.High_mantel.Pvar  <- subset_samples(ITS_T1.High_mantel, Species == "Pavona_varians")
    ITS2_T1.High_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pvar) > 0, ITS2_T1.High_mantel.Pvar)
        
    #16s T1.High Pvar dist        
      unifrac.Bac_T1.High_mantel.Pvar <- phyloseq::distance(Bac_T1.High_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T1.High Pvar dist
     bray.ITS2.T1.High.mantel.Pvar <- phyloseq::distance(ITS2_T1.High_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pvar, bray.ITS2.T1.High.mantel.Pvar,  permutations=1000)       
      
      #### T1.High Pacu only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pacu
    Bac_T1.High_mantel.Pacu <- subset_samples(Bac.T1.High.mantel, Species == "Pocillopora_acuta")
    Bac_T1.High_mantel.Pacu = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pacu) > 0, Bac_T1.High_mantel.Pacu) 
    
    ITS2_T1.High_mantel.Pacu  <- subset_samples(ITS_T1.High_mantel, Species == "Pocillopora_acuta")
    ITS2_T1.High_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pacu) > 0, ITS2_T1.High_mantel.Pacu)
        
    #16s T1.High Pacu dist        
      unifrac.Bac_T1.High_mantel.Pacu <- phyloseq::distance(Bac_T1.High_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T1.High Pacu dist
     bray.ITS2.T1.High.mantel.Pacu <- phyloseq::distance(ITS2_T1.High_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pacu, bray.ITS2.T1.High.mantel.Pacu,  permutations=1000)  

```
T1 Ambient only
```{r}
# figure out which samples are represented in the its2 and 16s dfs

# T1 only  #####

#Ambient only 

#set up 16s
    Bac.s.rel.u.T1<-subset_samples(Bac.s.rel.u, Time.Point=="T1")
    Bac.s.rel.u.T1.Ambient<-subset_samples(Bac.s.rel.u.T1, Treatment=="Ambient")

    Bac.s.rel.u.T1.Ambient = prune_taxa(taxa_sums(Bac.s.rel.u.T1.Ambient) > 0, Bac.s.rel.u.T1.Ambient) #this removes any OTU with 0s
  # look at sample names
    bac.T1.sd <- as(sample_data(Bac.s.rel.u.T1.Ambient), "data.frame") #sample df
    #save just sample names
      bac.T1.Ambient.sd.samps<-bac.T1.sd[['sample_name.1']]

#set up ITS2 T1, import  T1TF
    ITS2_T1TF_mantel <- readRDS("ITS2_T1TF_phyloseq.rds") 
     ITS2_T1_mantel<-subset_samples(ITS2_T1TF_mantel, Time.Point=="T1")
    ITS2_T1_mantel.Ambient<-subset_samples(ITS2_T1_mantel, Treatment=="Ambient")
      ITS2_T1_mantel.Ambient = prune_taxa(taxa_sums(ITS2_T1_mantel.Ambient) > 0, ITS2_T1_mantel.Ambient) #this removes any OTU with 0s

    
  # look at sample names
    ITS.T1.Ambient.sd <- as(sample_data(ITS2_T1_mantel.Ambient), "data.frame") #sample df
  #save just sample names
      ITS.T1.Ambient.sd.samps<-ITS.T1.Ambient.sd[['ID']]

# compare ITS to 16s at T1
  #identify missing samples btw datasets     
      setdiff(bac.T1.Ambient.sd.samps,ITS.T1.Ambient.sd.samps) #samples in 16s but not ITS: "S1443b" "SN108b" "SN197b" "SN222"
      setdiff(ITS.T1.Ambient.sd.samps,bac.T1.Ambient.sd.samps) #samples in ITS but not 16s: "S1573" "S2424" "S3035" "S1443" "S1895" "SN209" "SN63"  "S1418" "SN197" "SN126" "S785"  "SN108"
        
      # in 16S remove SN222,remove the bs on the rest "S1443b" "SN108b" "SN197b"
      #in ITS  Remove "S1573" "S2424" "S3035" "S1895" "SN209" "SN63"  "S1418" " "SN126" "S785"  

      
    #######stoped here#########  
      
      
      
      
      
  # 16s changes: rename sample S3469
      
    Bac.T1.Ambient.mantel <- subset_samples(Bac.s.rel.u.T1.Ambient, sample_name.1 != "S3469" )
        #40 samples
          
   #ITS2 - remove samps not in the 16s df - " S3468, S670, S3025, S1388a
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S3468" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S670" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S3025" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S1388a" )

        
    #change names ITS2 S1857b to S1857 and SN134b to SN134

    # pull OTU table
     OTU.ITS2T1.Ambient = as(otu_table(ITS2_T1_mantel.Ambient), "matrix") #pull otu
       OTU.ITS2T1.Ambient<-as.data.frame(OTU.ITS2T1.Ambient) #make into df
        row.names(OTU.ITS2T1.Ambient)[7] <- "S1857"   #change row name by row number
                row.names(OTU.ITS2T1.Ambient)[17] <- "SN134"   #change row name by row number

    #pull TAX table
     aTax.ITS2T1.Ambient = as(tax_table(ITS2_T1_mantel.Ambient), "matrix") #pull tax

    #pull SAMPLE DATA
     aSD.ITS2T1.Ambient = as(sample_data(ITS2_T1_mantel.Ambient), "matrix") #pull samp.data
     aSD.ITS2T1.Ambient<-as.data.frame(aSD.ITS2T1.Ambient) #change to df
        row.names(aSD.ITS2T1.Ambient)[7] <- "S1857"  #replace name S208b
        row.names(aSD.ITS2T1.Ambient)[17] <- "SN134"  #replace name S208b
            

    #remake the phyloseq obj
      samdata.Ambient = sample_data(aSD.ITS2T1.Ambient)
        seqtab.Ambient = otu_table(OTU.ITS2T1.Ambient, taxa_are_rows = FALSE)
        taxtab.Ambient = tax_table(aTax.ITS2T1.Ambient)
          ITS_T1.Ambient_mantel = phyloseq(otu_table(seqtab.Ambient), tax_table(taxtab.Ambient), sample_data(samdata.Ambient))
             
        # both have 40 samples - 
      
        # reorder ps object so samples are in the same order
        
        Bac.T1.Ambient.mantel <- Bac.T1.Ambient.mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac.T1.Ambient.mantel) %>% head(8)
        
        ITS_T1.Ambient_mantel <- ITS_T1.Ambient_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS_T1.Ambient_mantel) %>% head(8)
        
  # MAKE data matrix
  
  #16s T1 Ambient        
        unifrac.Bac.T1.Ambient.mantel <- phyloseq::distance(Bac.T1.Ambient.mantel, method = "wunifrac")
        
  #ITS2 T1 Ambient
     bray.ITS2.T1.Ambient.mantel <- phyloseq::distance(ITS_T1.Ambient_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T1.Ambient.mantel, bray.ITS2.T1.Ambient.mantel,  permutations=1000)
    
        
#### T1.Ambient Mcap only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for mcap
    Bac_T1.Ambient_mantel.Mcap <- subset_samples(Bac.T1.Ambient.mantel, Species == "Montipora_capitata")
    Bac_T1.Ambient_mantel.Mcap = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Mcap) > 0, Bac_T1.Ambient_mantel.Mcap) 
    
    ITS2_T1.Ambient_mantel.Mcap  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Montipora_capitata")
    ITS2_T1.Ambient_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Mcap) > 0, ITS2_T1.Ambient_mantel.Mcap)
        
    #16s T1.Ambient mcap dist        
      unifrac.Bac_T1.Ambient_mantel.Mcap <- phyloseq::distance(Bac_T1.Ambient_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T1.Ambient mcap dist
     bray.ITS2.T1.Ambient.mantel.Mcap <- phyloseq::distance(ITS2_T1.Ambient_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Mcap, bray.ITS2.T1.Ambient.mantel.Mcap,  permutations=1000)
    
#### T1.Ambient Pcomp only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pcomp
    Bac_T1.Ambient_mantel.Pcomp <- subset_samples(Bac.T1.Ambient.mantel, Species == "Porites_compressa")
    Bac_T1.Ambient_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pcomp) > 0, Bac_T1.Ambient_mantel.Pcomp) 
    
    ITS2_T1.Ambient_mantel.Pcomp  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Porites_compressa")
    ITS2_T1.Ambient_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pcomp) > 0, ITS2_T1.Ambient_mantel.Pcomp)
        
    #16s T1.Ambient Pcomp dist        
      unifrac.Bac_T1.Ambient_mantel.Pcomp <- phyloseq::distance(Bac_T1.Ambient_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T1.Ambient Pcomp dist
     bray.ITS2.T1.Ambient.mantel.Pcomp <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pcomp, bray.ITS2.T1.Ambient.mantel.Pcomp,  permutations=1000)        
        
    
#### T1.Ambient Pvar only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pvar
    Bac_T1.Ambient_mantel.Pvar <- subset_samples(Bac.T1.Ambient.mantel, Species == "Pavona_varians")
    Bac_T1.Ambient_mantel.Pvar = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pvar) > 0, Bac_T1.Ambient_mantel.Pvar) 
    
    ITS2_T1.Ambient_mantel.Pvar  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Pavona_varians")
    ITS2_T1.Ambient_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pvar) > 0, ITS2_T1.Ambient_mantel.Pvar)
        
    #16s T1.Ambient Pvar dist        
      unifrac.Bac_T1.Ambient_mantel.Pvar <- phyloseq::distance(Bac_T1.Ambient_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T1.Ambient Pvar dist
     bray.ITS2.T1.Ambient.mantel.Pvar <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pvar, bray.ITS2.T1.Ambient.mantel.Pvar,  permutations=1000)       
      
      #### T1.Ambient Pacu only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pacu
    Bac_T1.Ambient_mantel.Pacu <- subset_samples(Bac.T1.Ambient.mantel, Species == "Pocillopora_acuta")
    Bac_T1.Ambient_mantel.Pacu = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pacu) > 0, Bac_T1.Ambient_mantel.Pacu) 
    
    ITS2_T1.Ambient_mantel.Pacu  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Pocillopora_acuta")
    ITS2_T1.Ambient_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pacu) > 0, ITS2_T1.Ambient_mantel.Pacu)
        
    #16s T1.Ambient Pacu dist        
      unifrac.Bac_T1.Ambient_mantel.Pacu <- phyloseq::distance(Bac_T1.Ambient_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T1.Ambient Pacu dist
     bray.ITS2.T1.Ambient.mantel.Pacu <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pacu, bray.ITS2.T1.Ambient.mantel.Pacu,  permutations=1000)  

```
