---
title: "16S Thermal Tolerance"
html_document:
code_folding: hide
toc:yes
toc_depth:3
toc_float:yes
output: 
chunk_output_type: console
editor_options: 
  chunk_output_type: console
---
This is an RMarkdown for the All Coral Experiment 16S data. 

#16S
Load in your data
```{r} 
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr) 
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(tidyverse)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(microbiome)
library(vegan)
library(geosphere)
library(ade4)
library(microbiome)  
library(knitr)

# Load data (data_exploration.Rmd)
load("Data/Bac.seq_phyloseq.RData") 

## view data
ntaxa(Bac.seq)  #num taxa
nsamples(Bac.seq)   #num samples
sample_names(Bac.seq)[1:5] #samp names
rank_names(Bac.seq) #ranks are DIV and clade
sample_variables(Bac.seq) # metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.seq)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  # check 3k rare
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = Species)) +
  geom_point()

#remove duplicates and tests
Dups<-subset_samples(Bac.seq, Dups=="Yes")
TopNOTUsB = names(sort(taxa_sums(Dups), TRUE)[1:100])
bac50 = prune_taxa(TopNOTUsB, Dups)
bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S1753b" ) #remove duplicates
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S208" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3476" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "SN35b" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3149" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3149" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S1240" )
Bac.seq <- subset_samples(Bac.seq, Type != "extra" )
Bac.seq <- subset_samples(Bac.seq, Type != " " )
Bac.seq.df <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view

Bac.s = subset_samples(Bac.seq, Type== "sample" )
Bac.s.df <- as(sample_data(Bac.s), "data.frame") #create sample data frame to view

Bac.field = subset_samples(Bac.seq, Type== "field" )
Bac.field.df <- as(sample_data(Bac.field), "data.frame") #create sample data frame to view
```

#Make relative abundance df
```{r}        
Bac.s.rel  = transform_sample_counts(Bac.s, function(x) x / sum(x) )
Bac.s.rel = prune_taxa(taxa_sums(Bac.s.rel) > 0, Bac.s.rel) #this removes any OTU with 0s
data.Bac.s.rel <- as(sample_data(Bac.s.rel), "data.frame") 

Bac.seq.rel<- transform_sample_counts(Bac.seq, function(x) x / sum(x) ) 
Bac.seq.rel = prune_taxa(taxa_sums(Bac.seq.rel) > 0, Bac.seq.rel) #this removes any OTU with 0s
data.Bac.seq.rel <- as(sample_data(Bac.seq.rel), "data.frame") 

#number of ASVs by species
Bac.s.rel_Mcap<-subset_samples(Bac.s.rel, Species=="Montipora_capitata")
    Bac.s.rel_Mcap = prune_taxa(taxa_sums(Bac.s.rel_Mcap) > 0, Bac.s.rel_Mcap) #this removes any OTU with 0s

  Bac.s.rel_Pcomp<-subset_samples(Bac.s.rel, Species=="Porites_compressa")
    Bac.s.rel_Pcomp = prune_taxa(taxa_sums(Bac.s.rel_Pcomp) > 0, Bac.s.rel_Pcomp) #this removes any OTU with 0s

Bac.s.rel_Pvar<-subset_samples(Bac.s.rel, Species=="Pavona_varians")
    Bac.s.rel_Pvar = prune_taxa(taxa_sums(Bac.s.rel_Pvar) > 0, Bac.s.rel_Pvar) #this removes any OTU with 0s
  
    #by lineage for Pvar
    Bac.s.rel_Pvar.C1<-subset_samples(Bac.s.rel_Pvar, Clade=="C1")
    Bac.s.rel_Pvar.C1 = prune_taxa(taxa_sums(Bac.s.rel_Pvar.C1) > 0, Bac.s.rel_Pvar.C1)
    Bac.s.rel_Pvar.C27<-subset_samples(Bac.s.rel_Pvar, Clade=="C27")
    Bac.s.rel_Pvar.C27 = prune_taxa(taxa_sums(Bac.s.rel_Pvar.C27) > 0, Bac.s.rel_Pvar.C27)
    
Bac.s.rel_Pacu<-subset_samples(Bac.s.rel, Species=="Pocillopora_acuta")
    Bac.s.rel_Pacu = prune_taxa(taxa_sums(Bac.s.rel_Pacu) > 0, Bac.s.rel_Pacu) #this removes any OTU with 0s

#anything shared between species: ####

# Step 1: Convert to presence/absence
bac.presabs <- transform_sample_counts(Bac.s.rel, function(x) ifelse(x > 0, 1, 0))
unique(sample_data(Bac.s.rel)$Species)

# Step 2: List of species (must exactly match what's in your sample_data)
species_list <- c("Montipora_capitata", "Porites_compressa", "Pavona_varians", "Pocillopora_acuta")

# Step 3: Function to get taxa present in a species
get_present_taxa <- function(sp) {
  sp_samples <- rownames(as.data.frame(sample_data(bac.presabs)))[
    as.data.frame(sample_data(bac.presabs))$Species == sp
  ]
  sp_otu <- prune_samples(sp_samples, bac.presabs)
  otu_tab <- otu_table(sp_otu)
  if (!taxa_are_rows(sp_otu)) {
    otu_tab <- t(otu_tab)
  }
  taxa_names(sp_otu)[rowSums(otu_tab) > 0]
}

# Step 4: Create list of taxa per species
taxa_by_species <- setNames(lapply(species_list, get_present_taxa), species_list)

# Step 5: Get shared taxa for each species pair
pairwise_shared <- combn(species_list, 2, simplify = FALSE, FUN = function(pair) {
  taxa1 <- taxa_by_species[[pair[1]]]
  taxa2 <- taxa_by_species[[pair[2]]]
  shared <- intersect(taxa1, taxa2)
  list(
    species1 = pair[1],
    species2 = pair[2],
    n_shared = length(shared),
    shared_taxa = shared
  )
})

# Step 6: Create summary table
shared_summary <- map_df(pairwise_shared, ~{
  tibble(
    Species1 = .x$species1,
    Species2 = .x$species2,
    Shared_OTUs = .x$n_shared
  )
})

print(shared_summary)

# Optional: View OTUs or taxonomy for a pair
# First pair's shared OTUs:
pairwise_shared[[1]]$shared_taxa

# With taxonomy:
sharedtax<-as.data.frame(tax_table(Bac.s.rel)[pairwise_shared[[1]]$shared_taxa, ])
``` 
#Permanova UNIFRAC by ASV T0 ONLY
Tables S3-S6
```{r}         
 #T0 only #### 
Bac.s.rel.u.T0<-subset_samples(Bac.s.rel, Time.Point=="T0")
  Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s
#make unifrac matrix
  Bac.s.u.T0 <- phyloseq::distance(Bac.s.rel.u.T0, method = "wunifrac")
  s.u.samp.df <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df
  s.u.samp.df$Code<-paste(s.u.samp.df$Species,s.u.samp.df$Clade)

# PERMANOVA: Table S3
  set.seed(30)
    adonis2(Bac.s.u.T0 ~ Species, data = s.u.samp.df)
  set.seed(30)    
    adonis2(Bac.s.u.T0 ~ Species*Clade, data = s.u.samp.df)

# Homogeneity of dispersion test: Table S4 and S5
  set.seed(30)
    permutest(betadisper(Bac.s.u.T0, s.u.samp.df$Species, type = "centroid")) 
      ball<-  betadisper(Bac.s.u.T0, s.u.samp.df$Species, type = "centroid")
      TukeyHSD(ball,  ordered = FALSE,conf.level = 0.95)  #PCOMP has sig different dispersion
    
#Within Species: Table S6  ####
#Mcap
  #t0 only Bac.s.rel.u.T0
  mcap.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Montipora_capitata")
  mcap.u.bac.T0 = prune_taxa(taxa_sums(mcap.u.bac.T0) > 0, mcap.u.bac.T0) #this removes any OTU with 0s
  
  #make bray curtis data matrix
  u.mcap.u.T0 <- phyloseq::distance(mcap.u.bac.T0, method = "wunifrac")
     mcap.u.samp.df.T0 <- as(sample_data(mcap.u.bac.T0), "data.frame") #sample df
    
  set.seed(30)
    adonis2(u.mcap.u.T0 ~ Clade, data = mcap.u.samp.df.T0) 
  set.seed(30)
    permutest(betadisper(u.mcap.u.T0, mcap.u.samp.df.T0$Clade, type = "centroid")) 

#Pcomp  
  Pcomp.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Porites_compressa")
  Pcomp.u.bac.T0 = prune_taxa(taxa_sums(Pcomp.u.bac.T0) > 0, Pcomp.u.bac.T0) 

  #make bray curtis data matrix -T0
  u.Pcomp.u.T0 <- phyloseq::distance(Pcomp.u.bac.T0, method = "wunifrac")
     Pcomp.u.samp.df.T0 <- as(sample_data(Pcomp.u.bac.T0), "data.frame") 
     
  set.seed(30)
    adonis2(u.Pcomp.u.T0 ~ Clade, data = Pcomp.u.samp.df.T0)
  set.seed(30)
    permutest(betadisper(u.Pcomp.u.T0, Pcomp.u.samp.df.T0$Clade, type = "centroid")) 

#Pvar  
  #t0 only Bac.s.rel.u.T0
  Pvar.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Pavona_varians")
  Pvar.u.bac.T0 = prune_taxa(taxa_sums(Pvar.u.bac.T0) > 0, Pvar.u.bac.T0) 

  #make bray curtis data matrix -T0
  u.Pvar.u.T0 <- phyloseq::distance(Pvar.u.bac.T0, method = "wunifrac")
     Pvar.u.samp.df.T0 <- as(sample_data(Pvar.u.bac.T0), "data.frame") #sample df
  
  set.seed(30)
    adonis2(u.Pvar.u.T0 ~ Clade, data = Pvar.u.samp.df.T0) #adonis test no clade
  set.seed(30)
    permutest(betadisper(u.Pvar.u.T0, Pvar.u.samp.df.T0$Clade, type = "centroid"))     
    betadisper(u.Pvar.u.T0, Pvar.u.samp.df.T0$Time.Point, type = "centroid")
    
#Pacu  
  #t0 only Bac.s.rel.u.T0
  Pacu.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Pocillopora_acuta")
  Pacu.u.bac.T0 = prune_taxa(taxa_sums(Pacu.u.bac.T0) > 0, Pacu.u.bac.T0) 

  #make bray curtis data matrix -T0
  u.Pacu.u.T0 <- phyloseq::distance(Pacu.u.bac.T0, method = "wunifrac")
     Pacu.u.samp.df.T0 <- as(sample_data(Pacu.u.bac.T0), "data.frame") 
     
  set.seed(30)
    adonis2(u.Pacu.u.T0 ~ Clade, data = Pacu.u.samp.df.T0) #adonis test no clade
  set.seed(30)
    permutest(betadisper(u.Pacu.u.T0, Pacu.u.samp.df.T0$Clade, type = "centroid")) 
    betadisper(u.Pacu.u.T0, Pacu.u.samp.df.T0$Time.Point, type = "centroid")

``` 
    
#Permanova UNIFRAC by ASV
T0-T1, T1-TF by TREATMENT (high only, ambient only)
Models run within treatment group (high, ambient) with all T0 genotypes represented in both treatments (not split at T0)
Table S11
```{r} 
#Make T0 for Ambient and High with all genotypes represented ####
Bac.s.rel.T0Dupsamps<-Bac.s.rel #copy
 
#Ambient T0 ####
Bac.s.rel.T0Dupsamps_amb = subset_samples(Bac.s.rel, Time.Point == "T0" & Treatment == "Ambient")  #copy

#tax table
  tax  = tax_table(Bac.s.rel.T0Dupsamps_amb)
# sample data
  sd = sample_data(Bac.s.rel.T0Dupsamps_amb)
    rownames(sd) <- paste0(rownames(sd), "_dup")
# otu_table
  OTU = otu_table(Bac.s.rel.T0Dupsamps_amb)
    sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
    sd$Treatment <- as.character(sd$Treatment)
    sd$Treatment <- "High"    #rename HIGH
    sd$ID <- paste0(sd$ID, "_dup")
      bac.Amb.T0dup<-phyloseq(OTU, sd, tax)

##High ####
  Bac.s.rel.T0Dupsamps_amb = subset_samples(Bac.s.rel, Time.Point == "T0"&Treatment == "High")  #copy

#tax table
tax  = tax_table(Bac.s.rel.T0Dupsamps_amb)
# sample data
  sd = sample_data(Bac.s.rel.T0Dupsamps_amb)
  rownames(sd) <- paste0(rownames(sd), "_dup")
# otu_table
  OTU = otu_table(Bac.s.rel.T0Dupsamps_amb)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  sd$Treatment <- as.character(sd$Treatment)
  sd$Treatment <- "Ambient"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
    bac.High.T0dup<-phyloseq(OTU, sd, tax)

#Merge ####
Control16dups = merge_phyloseq(bac.Amb.T0dup,bac.High.T0dup)
Control16dups

  #merge  to the original df:
  Bac.s.rel.T0duplicated = merge_phyloseq(Bac.s.rel.T0Dupsamps,Control16dups)
   Bac.s.rel.T0duplicated
  Bac.s.rel.T0duplicated.sd <- as(sample_data(Bac.s.rel.T0duplicated), "data.frame") #create sample data frame to view
    Bac.s.rel.T0duplicated.sd

# Mcap  #### 

#Ambient only  #### 
  
#T0-T1 
  mcap.u.bac<- subset_samples(Bac.s.rel.T0duplicated, Species == "Montipora_capitata")
    mcap.u.bac.amb <- subset_samples(mcap.u.bac, Treatment == "Ambient")
        mcap.u.bac.T0T1.amb <- subset_samples(mcap.u.bac.amb, Time.Point == "T0"|Time.Point=="T1")
        mcap.u.bac.T0T1.amb = prune_taxa(taxa_sums(mcap.u.bac.T0T1.amb) > 0, mcap.u.bac.T0T1.amb) #this removes any OTU with 0s
        sample_sums(mcap.u.bac.T0T1.amb)
      
  #Make bray curtis data matrix
  u.mcap.u.bac.T0T1.amb <- phyloseq::distance(mcap.u.bac.T0T1.amb, method = "wunifrac")
     mcap.u.samp.df.T0T1.amb <- as(sample_data(mcap.u.bac.T0T1.amb), "data.frame") #sample df

  set.seed(30)
    adonis2(u.mcap.u.bac.T0T1.amb ~ Time.Point, data = mcap.u.samp.df.T0T1.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.mcap.u.bac.T0T1.amb, mcap.u.samp.df.T0T1.amb$Time.Point, type = "centroid")) 

#T1-TF 
  mcap.u.bac.T1TF.amb <- subset_samples(mcap.u.bac.amb, Time.Point == "T1"|Time.Point=="TF")
     mcap.u.bac.T1TF.amb = prune_taxa(taxa_sums(mcap.u.bac.T1TF.amb) > 0, mcap.u.bac.T1TF.amb) #this removes any OTU with 0s
        sample_sums(mcap.u.bac.T1TF.amb)
      
  #make bray curtis data matrix
  u.mcap.u.bac.T1TF.amb <- phyloseq::distance(mcap.u.bac.T1TF.amb, method = "wunifrac")
     mcap.u.samp.df.T1TF.amb <- as(sample_data(mcap.u.bac.T1TF.amb), "data.frame") #sample df
     mcap.u.samp.df.T1TF.amb$Code<-paste(mcap.u.samp.df.T1TF.amb$Treatment, mcap.u.samp.df.T1TF.amb$Time.Point)
   
  set.seed(30)
    adonis2(u.mcap.u.bac.T1TF.amb ~ Time.Point, data = mcap.u.samp.df.T1TF.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.mcap.u.bac.T1TF.amb, mcap.u.samp.df.T1TF.amb$Time.Point, type = "centroid")) 

#High only   
  #T0-T1 
    mcap.u.bac.high <- subset_samples(mcap.u.bac, Treatment == "High")
        mcap.u.bac.T0T1.high <- subset_samples(mcap.u.bac.high, Time.Point == "T0"|Time.Point=="T1")
       mcap.u.bac.T0T1.high = prune_taxa(taxa_sums(mcap.u.bac.T0T1.high) > 0, mcap.u.bac.T0T1.high) #this removes any OTU with 0s
        sample_sums(mcap.u.bac.T0T1.high)
      
  #make bray curtis data matrix
  u.mcap.u.bac.T0T1.high <- phyloseq::distance(mcap.u.bac.T0T1.high, method = "wunifrac")
     mcap.u.samp.df.T0T1.high <- as(sample_data(mcap.u.bac.T0T1.high), "data.frame") #sample df

  set.seed(30)
    adonis2(u.mcap.u.bac.T0T1.high ~ Time.Point, data = mcap.u.samp.df.T0T1.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.mcap.u.bac.T0T1.high, mcap.u.samp.df.T0T1.high$Time.Point, type = "centroid")) 

#T1-TF 
        mcap.u.bac.T1TF.high <- subset_samples(mcap.u.bac.high, Time.Point == "T1"|Time.Point=="TF")
       mcap.u.bac.T1TF.high = prune_taxa(taxa_sums(mcap.u.bac.T1TF.high) > 0, mcap.u.bac.T1TF.high) #this removes any OTU with 0s
        sample_sums(mcap.u.bac.T1TF.high)
      
  #make bray curtis data matrix
  u.mcap.u.bac.T1TF.high <- phyloseq::distance(mcap.u.bac.T1TF.high, method = "wunifrac")
     mcap.u.samp.df.T1TF.high <- as(sample_data(mcap.u.bac.T1TF.high), "data.frame") #sample df
     mcap.u.samp.df.T1TF.high$Code<-paste(mcap.u.samp.df.T1TF.high$Treatment, mcap.u.samp.df.T1TF.high$Time.Point)
   
  set.seed(30)
    adonis2(u.mcap.u.bac.T1TF.high ~ Time.Point, data = mcap.u.samp.df.T1TF.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.mcap.u.bac.T1TF.high, mcap.u.samp.df.T1TF.high$Time.Point, type = "centroid")) 
    
#Pcomp  #### 

#Ambient only  
#T0-T1 
  pcomp.u.bac<- subset_samples(Bac.s.rel.T0duplicated, Species == "Porites_compressa")
    pcomp.u.bac.amb <- subset_samples(pcomp.u.bac, Treatment == "Ambient")
        pcomp.u.bac.T0T1.amb <- subset_samples(pcomp.u.bac.amb, Time.Point == "T0"|Time.Point=="T1")
        pcomp.u.bac.T0T1.amb = prune_taxa(taxa_sums(pcomp.u.bac.T0T1.amb) > 0, pcomp.u.bac.T0T1.amb) #this removes any OTU with 0s
        sample_sums(pcomp.u.bac.T0T1.amb)
      
  #make bray curtis data matrix
  u.pcomp.u.bac.T0T1.amb <- phyloseq::distance(pcomp.u.bac.T0T1.amb, method = "wunifrac")
     pcomp.u.samp.df.T0T1.amb <- as(sample_data(pcomp.u.bac.T0T1.amb), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pcomp.u.bac.T0T1.amb ~ Time.Point, data = pcomp.u.samp.df.T0T1.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pcomp.u.bac.T0T1.amb, pcomp.u.samp.df.T0T1.amb$Time.Point, type = "centroid")) 

#T1-TF 
      pcomp.u.bac.T1TF.amb <- subset_samples(pcomp.u.bac.amb, Time.Point == "T1"|Time.Point=="TF")
       pcomp.u.bac.T1TF.amb = prune_taxa(taxa_sums(pcomp.u.bac.T1TF.amb) > 0, pcomp.u.bac.T1TF.amb) #this removes any OTU with 0s
        sample_sums(pcomp.u.bac.T1TF.amb)
      
  #make bray curtis data matrix
  u.pcomp.u.bac.T1TF.amb <- phyloseq::distance(pcomp.u.bac.T1TF.amb, method = "wunifrac")
     pcomp.u.samp.df.T1TF.amb <- as(sample_data(pcomp.u.bac.T1TF.amb), "data.frame") #sample df
     pcomp.u.samp.df.T1TF.amb$Code<-paste(pcomp.u.samp.df.T1TF.amb$Treatment, pcomp.u.samp.df.T1TF.amb$Time.Point)
   
  set.seed(30)
    adonis2(u.pcomp.u.bac.T1TF.amb ~ Time.Point, data = pcomp.u.samp.df.T1TF.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pcomp.u.bac.T1TF.amb, pcomp.u.samp.df.T1TF.amb$Time.Point, type = "centroid")) 

#High only   
  #T0-T1 
    pcomp.u.bac.high <- subset_samples(pcomp.u.bac, Treatment == "High")
        pcomp.u.bac.T0T1.high <- subset_samples(pcomp.u.bac.high, Time.Point == "T0"|Time.Point=="T1")
       pcomp.u.bac.T0T1.high = prune_taxa(taxa_sums(pcomp.u.bac.T0T1.high) > 0, pcomp.u.bac.T0T1.high) 
        sample_sums(pcomp.u.bac.T0T1.high)
      
  #make bray curtis data matrix
  u.pcomp.u.bac.T0T1.high <- phyloseq::distance(pcomp.u.bac.T0T1.high, method = "wunifrac")
     pcomp.u.samp.df.T0T1.high <- as(sample_data(pcomp.u.bac.T0T1.high), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pcomp.u.bac.T0T1.high ~ Time.Point, data = pcomp.u.samp.df.T0T1.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pcomp.u.bac.T0T1.high, pcomp.u.samp.df.T0T1.high$Time.Point, type = "centroid")) 

#T1-TF 
        pcomp.u.bac.T1TF.high <- subset_samples(pcomp.u.bac.high, Time.Point == "T1"|Time.Point=="TF")
       pcomp.u.bac.T1TF.high = prune_taxa(taxa_sums(pcomp.u.bac.T1TF.high) > 0, pcomp.u.bac.T1TF.high) 
        sample_sums(pcomp.u.bac.T1TF.high)
      
  #make bray curtis data matrix
  u.pcomp.u.bac.T1TF.high <- phyloseq::distance(pcomp.u.bac.T1TF.high, method = "wunifrac")
     pcomp.u.samp.df.T1TF.high <- as(sample_data(pcomp.u.bac.T1TF.high), "data.frame") #sample df
     pcomp.u.samp.df.T1TF.high$Code<-paste(pcomp.u.samp.df.T1TF.high$Treatment, pcomp.u.samp.df.T1TF.high$Time.Point)
   
  set.seed(30)
    adonis2(u.pcomp.u.bac.T1TF.high ~ Time.Point, data = pcomp.u.samp.df.T1TF.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pcomp.u.bac.T1TF.high, pcomp.u.samp.df.T1TF.high$Time.Point, type = "centroid")) 
    
#pvar  #### 

#Ambient only  
#T0-T1 
  pvar.u.bac<- subset_samples(Bac.s.rel.T0duplicated, Species == "Pavona_varians")
    pvar.u.bac.amb <- subset_samples(pvar.u.bac, Treatment == "Ambient")
        pvar.u.bac.T0T1.amb <- subset_samples(pvar.u.bac.amb, Time.Point == "T0"|Time.Point=="T1")
       pvar.u.bac.T0T1.amb = prune_taxa(taxa_sums(pvar.u.bac.T0T1.amb) > 0, pvar.u.bac.T0T1.amb) #this removes any OTU with 0s
        sample_sums(pvar.u.bac.T0T1.amb)
      
  #make bray curtis data matrix
  u.pvar.u.bac.T0T1.amb <- phyloseq::distance(pvar.u.bac.T0T1.amb, method = "wunifrac")
     pvar.u.samp.df.T0T1.amb <- as(sample_data(pvar.u.bac.T0T1.amb), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pvar.u.bac.T0T1.amb ~ Time.Point, data = pvar.u.samp.df.T0T1.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pvar.u.bac.T0T1.amb, pvar.u.samp.df.T0T1.amb$Time.Point, type = "centroid")) 

#T1-TF 
        pvar.u.bac.T1TF.amb <- subset_samples(pvar.u.bac.amb, Time.Point == "T1"|Time.Point=="TF")
       pvar.u.bac.T1TF.amb = prune_taxa(taxa_sums(pvar.u.bac.T1TF.amb) > 0, pvar.u.bac.T1TF.amb) #this removes any OTU with 0s
        sample_sums(pvar.u.bac.T1TF.amb)
      
  #make bray curtis data matrix
  u.pvar.u.bac.T1TF.amb <- phyloseq::distance(pvar.u.bac.T1TF.amb, method = "wunifrac")
     pvar.u.samp.df.T1TF.amb <- as(sample_data(pvar.u.bac.T1TF.amb), "data.frame") #sample df
     pvar.u.samp.df.T1TF.amb$Code<-paste(pvar.u.samp.df.T1TF.amb$Treatment, pvar.u.samp.df.T1TF.amb$Time.Point)
   
  set.seed(30)
    adonis2(u.pvar.u.bac.T1TF.amb ~ Time.Point, data = pvar.u.samp.df.T1TF.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pvar.u.bac.T1TF.amb, pvar.u.samp.df.T1TF.amb$Time.Point, type = "centroid")) #0.028
    dispersion <- betadisper(u.pvar.u.bac.T1TF.amb, 
                         pvar.u.samp.df.T1TF.amb$Time.Point, 
                         type = "centroid")
    plot(dispersion) #more dispersion at TF

#High only   
  #T0-T1 
    pvar.u.bac.high <- subset_samples(pvar.u.bac, Treatment == "High")
        pvar.u.bac.T0T1.high <- subset_samples(pvar.u.bac.high, Time.Point == "T0"|Time.Point=="T1")
       pvar.u.bac.T0T1.high = prune_taxa(taxa_sums(pvar.u.bac.T0T1.high) > 0, pvar.u.bac.T0T1.high) #this removes any OTU with 0s
        sample_sums(pvar.u.bac.T0T1.high)
      
  #make bray curtis data matrix
  u.pvar.u.bac.T0T1.high <- phyloseq::distance(pvar.u.bac.T0T1.high, method = "wunifrac")
     pvar.u.samp.df.T0T1.high <- as(sample_data(pvar.u.bac.T0T1.high), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pvar.u.bac.T0T1.high ~ Time.Point, data = pvar.u.samp.df.T0T1.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pvar.u.bac.T0T1.high, pvar.u.samp.df.T0T1.high$Time.Point, type = "centroid")) 

#T1-TF 
        pvar.u.bac.T1TF.high <- subset_samples(pvar.u.bac.high, Time.Point == "T1"|Time.Point=="TF")
       pvar.u.bac.T1TF.high = prune_taxa(taxa_sums(pvar.u.bac.T1TF.high) > 0, pvar.u.bac.T1TF.high) #this removes any OTU with 0s
        sample_sums(pvar.u.bac.T1TF.high)
      
  #make bray curtis data matrix
  u.pvar.u.bac.T1TF.high <- phyloseq::distance(pvar.u.bac.T1TF.high, method = "wunifrac")
     pvar.u.samp.df.T1TF.high <- as(sample_data(pvar.u.bac.T1TF.high), "data.frame") #sample df
     pvar.u.samp.df.T1TF.high$Code<-paste(pvar.u.samp.df.T1TF.high$Treatment, pvar.u.samp.df.T1TF.high$Time.Point)
   
  set.seed(30)
    adonis2(u.pvar.u.bac.T1TF.high ~ Time.Point, data = pvar.u.samp.df.T1TF.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pvar.u.bac.T1TF.high, pvar.u.samp.df.T1TF.high$Time.Point, type = "centroid")) 
    
#pacu  #### 

#Ambient only  
#T0-T1 
  pacu.u.bac<- subset_samples(Bac.s.rel.T0duplicated, Species == "Pocillopora_acuta")
    pacu.u.bac.amb <- subset_samples(pacu.u.bac, Treatment == "Ambient")
        pacu.u.bac.T0T1.amb <- subset_samples(pacu.u.bac.amb, Time.Point == "T0"|Time.Point=="T1")
        pacu.u.bac.T0T1.amb = prune_taxa(taxa_sums(pacu.u.bac.T0T1.amb) > 0, pacu.u.bac.T0T1.amb) #this removes any OTU with 0s
        sample_sums(pacu.u.bac.T0T1.amb)
      
  #make bray curtis data matrix
  u.pacu.u.bac.T0T1.amb <- phyloseq::distance(pacu.u.bac.T0T1.amb, method = "wunifrac")
     pacu.u.samp.df.T0T1.amb <- as(sample_data(pacu.u.bac.T0T1.amb), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pacu.u.bac.T0T1.amb ~ Time.Point, data = pacu.u.samp.df.T0T1.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pacu.u.bac.T0T1.amb, pacu.u.samp.df.T0T1.amb$Time.Point, type = "centroid")) 

#T1-TF 
        pacu.u.bac.T1TF.amb <- subset_samples(pacu.u.bac.amb, Time.Point == "T1"|Time.Point=="TF")
       pacu.u.bac.T1TF.amb = prune_taxa(taxa_sums(pacu.u.bac.T1TF.amb) > 0, pacu.u.bac.T1TF.amb) #this removes any OTU with 0s
        sample_sums(pacu.u.bac.T1TF.amb)
      
  #make bray curtis data matrix
  u.pacu.u.bac.T1TF.amb <- phyloseq::distance(pacu.u.bac.T1TF.amb, method = "wunifrac")
     pacu.u.samp.df.T1TF.amb <- as(sample_data(pacu.u.bac.T1TF.amb), "data.frame") #sample df
     pacu.u.samp.df.T1TF.amb$Code<-paste(pacu.u.samp.df.T1TF.amb$Treatment, pacu.u.samp.df.T1TF.amb$Time.Point)
   
  set.seed(30)
    adonis2(u.pacu.u.bac.T1TF.amb ~ Time.Point, data = pacu.u.samp.df.T1TF.amb) #adonis test
  set.seed(30)
    permutest(betadisper(u.pacu.u.bac.T1TF.amb, pacu.u.samp.df.T1TF.amb$Time.Point, type = "centroid")) 
    
#High only   
  #T0-T1 
    pacu.u.bac.high <- subset_samples(pacu.u.bac, Treatment == "High")
        pacu.u.bac.T0T1.high <- subset_samples(pacu.u.bac.high, Time.Point == "T0"|Time.Point=="T1")
       pacu.u.bac.T0T1.high = prune_taxa(taxa_sums(pacu.u.bac.T0T1.high) > 0, pacu.u.bac.T0T1.high) #this removes any OTU with 0s
        sample_sums(pacu.u.bac.T0T1.high)
      
  #make bray curtis data matrix
  u.pacu.u.bac.T0T1.high <- phyloseq::distance(pacu.u.bac.T0T1.high, method = "wunifrac")
     pacu.u.samp.df.T0T1.high <- as(sample_data(pacu.u.bac.T0T1.high), "data.frame") #sample df

  set.seed(30)
    adonis2(u.pacu.u.bac.T0T1.high ~ Time.Point, data = pacu.u.samp.df.T0T1.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pacu.u.bac.T0T1.high, pacu.u.samp.df.T0T1.high$Time.Point, type = "centroid")) 

#T1-TF 
        pacu.u.bac.T1TF.high <- subset_samples(pacu.u.bac.high, Time.Point == "T1"|Time.Point=="TF")
       pacu.u.bac.T1TF.high = prune_taxa(taxa_sums(pacu.u.bac.T1TF.high) > 0, pacu.u.bac.T1TF.high) #this removes any OTU with 0s
        sample_sums(pacu.u.bac.T1TF.high)
      
  #make bray curtis data matrix
  u.pacu.u.bac.T1TF.high <- phyloseq::distance(pacu.u.bac.T1TF.high, method = "wunifrac")
     pacu.u.samp.df.T1TF.high <- as(sample_data(pacu.u.bac.T1TF.high), "data.frame") #sample df
     pacu.u.samp.df.T1TF.high$Code<-paste(pacu.u.samp.df.T1TF.high$Treatment, pacu.u.samp.df.T1TF.high$Time.Point)
   
  set.seed(30)
    adonis2(u.pacu.u.bac.T1TF.high ~ Time.Point, data = pacu.u.samp.df.T1TF.high) #adonis test
  set.seed(30)
    permutest(betadisper(u.pacu.u.bac.T1TF.high, pacu.u.samp.df.T1TF.high$Time.Point, type = "centroid")) 
``` 

#Unifrac NMDS 
Figure 2B, S13-S15
```{r}      
 Sp.colors <- c("Montipora_capitata"= "#E69F00","Porites_compressa"="#56B4E9","Pavona_varians"="#CC79A7","Pocillopora_acuta"="#009E73", "High"="darkRed","Ambient"="navyBlue")

clade_colors <- c("C31" = "goldenrod1", "C31D4" = "darkorange3", "C15cj" = "powderblue", "C15n" = "#0072B2", "C1"="pink1", "C27"="firebrick", "C1d"="palegreen2", "C42_2"="darkgreen", "High"="red", "Ambient"="blue")

#T0 only  ####
Bac.s.rel.u.T0<-subset_samples(Bac.s.rel, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s

unifrac.all.T0 <- ordinate(Bac.s.rel.u.T0, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T0)
scores.ord.u_RelA_stats.T0<-as.data.frame(cbind(vegan::scores(unifrac.all.T0, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.T0 <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df

scores.ord.u_RelA_stats.T0$Treatment <- unifrac.all.sd.T0$Treatment
scores.ord.u_RelA_stats.T0$Species <- unifrac.all.sd.T0$Species
scores.ord.u_RelA_stats.T0$Clade <- unifrac.all.sd.T0$Clade
scores.ord.u_RelA_stats.T0$Code <- paste(scores.ord.u_RelA_stats.T0$Clade, scores.ord.u_RelA_stats.T0$Species)

#plot Figure 2B
ggplot(scores.ord.u_RelA_stats.T0, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  scale_color_manual(values = Sp.colors) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Species, fill = Species), 
               geom = "polygon", alpha = 0.1, linetype = 1) +  # Solid line and filled
  scale_fill_manual(values = Sp.colors) +  # Fill ellipses with the same color as points
  ggtitle("NMDS 16S TO") +
  theme_classic()
ggsave("16S_T0.pdf")

#T0 Only Figure S13
ggplot(scores.ord.u_RelA_stats.T0, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  #scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
 # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  ggtitle("NMDS 16s unifrac T0") +
  theme_classic()+
  facet_wrap(~Species)
ggsave("16S_T0Clade.pdf")

#T1 only####
Bac.s.rel.u.T1<-subset_samples(Bac.s.rel, Time.Point=="T1")
    Bac.s.rel.u.T1 = prune_taxa(taxa_sums(Bac.s.rel.u.T1) > 0, Bac.s.rel.u.T1) #this removes any OTU with 0s

unifrac.all.T1 <- ordinate(Bac.s.rel.u.T1, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T1)
scores.ord.u_RelA_stats.T1<-as.data.frame(cbind(vegan::scores(unifrac.all.T1, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.T1 <- as(sample_data(Bac.s.rel.u.T1), "data.frame") #sample df

scores.ord.u_RelA_stats.T1$Treatment <- unifrac.all.sd.T1$Treatment
scores.ord.u_RelA_stats.T1$Species <- unifrac.all.sd.T1$Species
scores.ord.u_RelA_stats.T1$Clade <- unifrac.all.sd.T1$Clade
scores.ord.u_RelA_stats.T1$Code <- paste(scores.ord.u_RelA_stats.T1$Clade, scores.ord.u_RelA_stats.T1$Treatment)
scores.ord.u_RelA_stats.T1$Code3 <- paste(scores.ord.u_RelA_stats.T1$Species, scores.ord.u_RelA_stats.T1$Treatment)

p<-ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species, shape=Treatment), size = 4) +
  scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
stat_ellipse(aes(x = NMDS1, y = NMDS2, color=Species, group=Code3,lty = Species)) +
scale_linetype_manual(values=c(1,2,1,2,1,2,1,2))+
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic();p
p+facet_wrap(~Species)

#T1 by lineage and treatment ####
ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape = Treatment), size = 4) +
  stat_ellipse(aes(fill = Code), 
               geom = "polygon", 
               alpha = 0.3, 
               colour = NA) +  # remove ellipse outlines (optional)
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic() +
  facet_wrap(~Species)

ggsave("16S T1 by lineage.pdf")


# ADD PCOMP ONLY T1 AMB/HIGH ####
scores.ord.u_RelA_stats.T1_Pcomp<-subset(scores.ord.u_RelA_stats.T1, Species=="Porites_compressa")
# Plot
# Subset Porites compressa
scores.ord.u_RelA_stats.T1_Pcomp <- subset(scores.ord.u_RelA_stats.T1, Species == "Porites_compressa")

# Plot with matched fill and color using Sp.colors: Figure S13
p <- ggplot(scores.ord.u_RelA_stats.T1_Pcomp, aes(x = NMDS1, y = NMDS2)) + 
  # Points
  geom_point(aes(color = Treatment, shape = Treatment), size = 8) +

  # 95% filled ellipses
  stat_ellipse(aes(fill = Treatment, linetype = Treatment),
               geom = "polygon",
               level = 0.95,
               alpha = 0.25,
               color = NA) +

  # Apply the same custom colors to both color and fill
  scale_color_manual(values = Sp.colors) +
  scale_fill_manual(values = Sp.colors) +

  ggtitle("NMDS 16S – T1 Porites compressa (95% Ellipses by Treatment)") +
  theme_classic()
p
ggsave("16S T1 Pcomp by treatment")



#TF only ####
Bac.s.rel.u.TF<-subset_samples(Bac.s.rel, Time.Point=="TF")
    Bac.s.rel.u.TF = prune_taxa(taxa_sums(Bac.s.rel.u.TF) > 0, Bac.s.rel.u.TF) #this removes any OTU with 0s

unifrac.all.TF <- ordinate(Bac.s.rel.u.TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.TF)
scores.ord.u_RelA_stats.TF<-as.data.frame(cbind(vegan::scores(unifrac.all.TF, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.TF <- as(sample_data(Bac.s.rel.u.TF), "data.frame") #sample df

scores.ord.u_RelA_stats.TF$Treatment <- unifrac.all.sd.TF$Treatment
scores.ord.u_RelA_stats.TF$Species <- unifrac.all.sd.TF$Species
scores.ord.u_RelA_stats.TF$Clade <- unifrac.all.sd.TF$Clade
scores.ord.u_RelA_stats.TF$Code <- paste(scores.ord.u_RelA_stats.TF$Clade, scores.ord.u_RelA_stats.TF$Species)
scores.ord.u_RelA_stats.TF$Code3 <- paste(scores.ord.u_RelA_stats.TF$Species, scores.ord.u_RelA_stats.TF$Treatment)



ggplot(scores.ord.u_RelA_stats.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species, shape=Treatment), size = 4) +
  scale_color_manual(values=Sp.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, color = Species, group = Code3, linetype = Treatment, fill = Treatment),
             geom = "polygon", alpha = 0.2)+
    scale_fill_manual(values = c("Ambient" = "blue", "High" = "red")) +  # flipped here
scale_linetype_manual(values=c(1,2,1,2,1,2,1,2))+
  ggtitle("NMDS 16s unifrac TF") +
  theme_classic()+
  facet_wrap(~Species)
ggsave("16s TF by treatment")

#Table S13  
ggplot(scores.ord.u_RelA_stats.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  scale_color_manual(values=clade_colors)+
 stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  ggtitle("NMDS 16s unifrac TF") +
  theme_classic()+
  facet_wrap(~Species)

#Ordination NMDS for all species, all time points #### 
unifrac.all <- ordinate(Bac.s.rel, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all)
scores.ord.u_RelA_stats<-as.data.frame(cbind(vegan::scores(unifrac.all, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd <- as(sample_data(Bac.s.rel), "data.frame") #sample df

scores.ord.u_RelA_stats$Treatment <- unifrac.all.sd$Treatment
scores.ord.u_RelA_stats$Species <- unifrac.all.sd$Species
scores.ord.u_RelA_stats$Clade <- unifrac.all.sd$Clade
scores.ord.u_RelA_stats$Time.Point <- unifrac.all.sd$Time.Point
scores.ord.u_RelA_stats$Code <- paste(scores.ord.u_RelA_stats$Clade, scores.ord.u_RelA_stats$Species)

ggplot(scores.ord.u_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS 16s bray") +
  theme_classic()+
  facet_wrap(~ Species + Time.Point, nrow = 4)
#ggsave("bc_all.pdf")

#T1 and TF only
#### mcap #####   
Bac.s.rel_McapT1TF <- subset_samples(Bac.s.rel_Mcap, Time.Point == "T1" | Time.Point == "TF")

ord.mcap2 <- ordinate(Bac.s.rel_McapT1TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.mcap2)

scores.mcap.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.mcap2, display="sites"))) # these are "points" ask hannah 
     Bac.s.rel_Mcap.df <- as(sample_data(Bac.s.rel_McapT1TF), "data.frame") #sample df
scores.mcap.u.samp.df$Time.Point <- Bac.s.rel_Mcap.df$Time.Point
scores.mcap.u.samp.df$Parent.ID <- Bac.s.rel_Mcap.df$Parent.ID
scores.mcap.u.samp.df$Treatment <- Bac.s.rel_Mcap.df$Treatment
scores.mcap.u.samp.df$Clade <- Bac.s.rel_Mcap.df$Clade
scores.mcap.u.samp.df$Code <- paste(Bac.s.rel_Mcap.df$Parent.ID,Bac.s.rel_Mcap.df$Time.Point)
scores.mcap.u.samp.df$Code2 <- paste(Bac.s.rel_Mcap.df$Treatment,Bac.s.rel_Mcap.df$Time.Point)
scores.mcap.u.samp.df$Code3 <- paste(Bac.s.rel_Mcap.df$Treatment,Bac.s.rel_Mcap.df$Time.Point,mcap.u.samp.df.T1F$Parent.ID)

mcap_16S<-ggplot(scores.mcap.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  stat_ellipse(aes(fill = Treatment, group = interaction(Treatment, Time.Point),
                   linetype = factor(Time.Point)),
               geom = "polygon", alpha = 0.2, color = NA) +
  scale_fill_manual(values = mcap.u.Parent.colors) +
  scale_linetype_manual(values = c(2, 3, 2, 3)) +
  guides(linetype = "none") +  
  ggtitle("M. capitata") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);

mcap_16S_clade<-ggplot(scores.mcap.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Clade, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  ggtitle("M. capitata") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);mcap_16S_clade

#### Pcomp #####  
Bac.s.rel_PcompT1TF <- subset_samples(Bac.s.rel_Pcomp, Time.Point == "T1" | Time.Point == "TF")

ord.pcomp2 <- ordinate(Bac.s.rel_PcompT1TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pcomp2)

scores.pcomp.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.pcomp2, display="sites"))) 
  Bac.s.rel_Pcomp.df <- as(sample_data(Bac.s.rel_PcompT1TF), "data.frame") #sample df
scores.pcomp.u.samp.df$Time.Point <- Bac.s.rel_Pcomp.df$Time.Point
scores.pcomp.u.samp.df$Parent.ID <- Bac.s.rel_Pcomp.df$Parent.ID
scores.pcomp.u.samp.df$Treatment <- Bac.s.rel_Pcomp.df$Treatment
scores.pcomp.u.samp.df$Clade <- Bac.s.rel_Pcomp.df$Clade
scores.pcomp.u.samp.df$Code <- paste(Bac.s.rel_Pcomp.df$Parent.ID,Bac.s.rel_Pcomp.df$Time.Point)
scores.pcomp.u.samp.df$Code2 <- paste(Bac.s.rel_Pcomp.df$Treatment,Bac.s.rel_Pcomp.df$Time.Point)
scores.pcomp.u.samp.df$Code3 <- paste(Bac.s.rel_Pcomp.df$Treatment,Bac.s.rel_Pcomp.df$Time.Point,pcomp.u.samp.df.T1F$Parent.ID)

pcomp_16S<-ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  scale_color_manual(values = pcomp.u.Parent.colors) +
  stat_ellipse(aes(fill = Treatment, group = interaction(Treatment, Time.Point),
                   linetype = factor(Time.Point)),
               geom = "polygon", alpha = 0.2, color = NA) +
  scale_fill_manual(values = pcomp.u.Parent.colors) +
  scale_linetype_manual(values = c(2, 3, 2, 3)) +
  guides(linetype = "none") +  # 👈 removes line type legend
  ggtitle("P. compressa") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pcomp_16S

pcomp_16S_clade<-ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Clade, color = Treatment), size = 5) + 
  scale_color_manual(values = pcomp.u.Parent.colors) +
  #stat_ellipse(aes(group = interaction(Treatment, Clade)),
              # geom = "polygon", alpha = 0.2, color = NA) +
  # scale_fill_manual(values = pcomp.u.Parent.colors) +
  guides(linetype = "none") +  # 👈 removes line type legend
  ggtitle("P. compressa") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pcomp_16S_clade

#### Pvar #####   
Bac.s.rel_PvarT1TF <- subset_samples(Bac.s.rel_Pvar, Time.Point == "T1" | Time.Point == "TF")

ord.pvar2 <- ordinate(Bac.s.rel_PvarT1TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pvar2)

scores.pvar.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.pvar2, display="sites"))) 
     Bac.s.rel_Pvar.df <- as(sample_data(Bac.s.rel_PvarT1TF), "data.frame") 
scores.pvar.u.samp.df$Time.Point <- Bac.s.rel_Pvar.df$Time.Point
scores.pvar.u.samp.df$Parent.ID <- Bac.s.rel_Pvar.df$Parent.ID
scores.pvar.u.samp.df$Treatment <- Bac.s.rel_Pvar.df$Treatment
scores.pvar.u.samp.df$Clade <- Bac.s.rel_Pvar.df$Clade
scores.pvar.u.samp.df$Code <- paste(Bac.s.rel_Pvar.df$Parent.ID,Bac.s.rel_Pvar.df$Time.Point)
scores.pvar.u.samp.df$Code2 <- paste(Bac.s.rel_Pvar.df$Treatment,Bac.s.rel_Pvar.df$Time.Point)
scores.pvar.u.samp.df$Code3 <- paste(Bac.s.rel_Pvar.df$Treatment,Bac.s.rel_Pvar.df$Time.Point,pvar.u.samp.df.T1F$Parent.ID)

pvar_16S<-ggplot(scores.pvar.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  stat_ellipse(aes(fill = Treatment, group = interaction(Treatment, Time.Point),
                   linetype = factor(Time.Point)),
               geom = "polygon", alpha = 0.2, color = NA) +
  scale_fill_manual(values = mcap.u.Parent.colors) +
  scale_linetype_manual(values = c(2, 3, 2, 3)) +
  guides(linetype = "none") +  # 👈 removes line type legend
  ggtitle("P. varians") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pvar_16S

pvar_16S_clade<-ggplot(scores.pvar.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Clade, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  #stat_ellipse(aes(group = interaction(Treatment, Clade)),
              # geom = "polygon", alpha = 0.2, color = NA) +
  # scale_fill_manual(values = pvar.u.Parent.colors) +
  guides(linetype = "none") +  # 👈 removes line type legend
  ggtitle("P. varians") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pvar_16S_clade

#### Pacu #####   
Bac.s.rel_PacuT1TF <- subset_samples(Bac.s.rel_Pacu, Time.Point == "T1" | Time.Point == "TF")

ord.pacu2 <- ordinate(Bac.s.rel_PacuT1TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pacu2)

scores.pacu.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.pacu2, display="sites")))
     Bac.s.rel_Pacu.df <- as(sample_data(Bac.s.rel_PacuT1TF), "data.frame") #sample df
scores.pacu.u.samp.df$Time.Point <- Bac.s.rel_Pacu.df$Time.Point
scores.pacu.u.samp.df$Parent.ID <- Bac.s.rel_Pacu.df$Parent.ID
scores.pacu.u.samp.df$Treatment <- Bac.s.rel_Pacu.df$Treatment
scores.pacu.u.samp.df$Clade <- Bac.s.rel_Pacu.df$Clade
scores.pacu.u.samp.df$Code <- paste(Bac.s.rel_Pacu.df$Parent.ID,Bac.s.rel_Pacu.df$Time.Point)
scores.pacu.u.samp.df$Code2 <- paste(Bac.s.rel_Pacu.df$Treatment,Bac.s.rel_Pacu.df$Time.Point)
scores.pacu.u.samp.df$Code3 <- paste(Bac.s.rel_Pacu.df$Treatment,Bac.s.rel_Pacu.df$Time.Point,pacu.u.samp.df.T1F$Parent.ID)

pacu_16S<-ggplot(scores.pacu.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  stat_ellipse(aes(fill = Treatment, group = interaction(Treatment, Time.Point),
                   linetype = factor(Time.Point)),
               geom = "polygon", alpha = 0.2, color = NA) +
  scale_fill_manual(values = mcap.u.Parent.colors) +
  scale_linetype_manual(values = c(2, 3, 2, 3)) +
  guides(linetype = "none") +  
  ggtitle("P. acuta") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pacu_16S

pacu_16S_clade<-ggplot(scores.pacu.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Clade, color = Treatment), size = 5) + 
  scale_color_manual(values = mcap.u.Parent.colors) +
  #stat_ellipse(aes(group = interaction(Treatment, Clade)),
              # geom = "polygon", alpha = 0.2, color = NA) +
  # scale_fill_manual(values = pacu.u.Parent.colors) +
  guides(linetype = "none") +  
  ggtitle("P. acuta") +
  theme_classic() +
    theme(plot.title = element_text(face = "italic")) +  
  facet_wrap(~Time.Point);pacu_16S_clade

#plot all together: Figure S14 #### 
combined_16S <- grid.arrange(mcap_16S, pcomp_16S, pvar_16S, pacu_16S, ncol = 1)
ggsave("16S_over_time.png", combined_16S, width = 8, height = 16, dpi = 300)


#plot all together: Figure S15 #### 
combined_16S <- grid.arrange(mcap_16S_clade, pcomp_16S_clade, pvar_16S_clade, pacu_16S_clade, ncol = 1)
ggsave("16S_over_time_clade.png", combined_16S, width = 8, height = 16, dpi = 300)
```

# Unifrac with field samples to compare: Figure S16
```{r} 
 # Check for microbiome tank drift compare Ambient to Field samples #### 
Bac.subset <- subset_samples(Bac.seq, Type == "field" | Treatment == "Ambient")
Bac.subset<-subset_samples(Bac.subset, Time.Point=="T1"|Time.Point=="TF")

unifrac.all.field <- ordinate(Bac.subset, method="NMDS", "unifrac",weighted=TRUE, set.seed(40)) 
stressplot(unifrac.all.field)
scores.ord.u_ALLfield_stats<-as.data.frame(cbind(vegan::scores(unifrac.all.field, display="sites"))) 
     unifrac.samp.field.sd <- as(sample_data(Bac.subset), "data.frame") #sample df
scores.ord.u_ALLfield_stats$Treatment <- unifrac.samp.field.sd$Treatment
scores.ord.u_ALLfield_stats$Species <- unifrac.samp.field.sd$Species
scores.ord.u_ALLfield_stats$Type <- unifrac.samp.field.sd$Type
scores.ord.u_ALLfield_stats$Time.Point <- unifrac.samp.field.sd$Time.Point
scores.ord.u_ALLfield_stats$Code <- paste(scores.ord.u_ALLfield_stats$Species, scores.ord.u_ALLfield_stats$Type)
scores.ord.u_ALLfield_stats$Time.Point[scores.ord.u_ALLfield_stats$Time.Point == "F1"] <- "T1"          # Replace b by XXX
scores.ord.u_ALLfield_stats$Time.Point[scores.ord.u_ALLfield_stats$Time.Point == "TC"] <- "T0"          # Replace b by XXX

pf<-ggplot(scores.ord.u_ALLfield_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Type, shape=Type), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code), linetype = 2) +
  ggtitle("NMDS 16s ") +
  theme_classic();pf
#ggsave("bc_all.pdf")
field<-pf+facet_wrap(~Species~Time.Point, nrow=4);field
ggsave("16S_field_compare.png", field, width = 8, height = 13, dpi = 300)

#PERMANOVA by species: Table S18, no interactive effect of microbial communities between sample times over time

#Mcap
Bac.subset.mcap<-subset_samples(Bac.subset, Species=="Montipora_capitata")
#make unifrac matrix
s.u.samp.df.mcap <- as(sample_data(Bac.subset.mcap), "data.frame") #sample df
s.u.samp.df.mcap$Code<-paste(s.u.samp.df.mcap$Species,s.u.samp.df.mcap$Clade)
s.u.samp.df.mcap$Time.Point[s.u.samp.df.mcap$Time.Point == "F1"] <- "T1"          # Replace b by XXX

FieldSamps.mcap <- phyloseq::distance(Bac.subset.mcap, method = "wunifrac")
     FieldSamps.df.mcap <- as(sample_data(Bac.subset.mcap), "data.frame")
     FieldSamps.df.mcap$Code<-paste(FieldSamps.df.mcap$Treatment,FieldSamps.df.mcap$Clade)
 
      set.seed(30)
    adonis2(FieldSamps.mcap ~ Type*Time.Point, data = FieldSamps.df.mcap, by="margin")

#Pcomp
Bac.subset.pcomp<-subset_samples(Bac.subset, Species=="Porites_compressa")
#make unifrac matrix
s.u.samp.df.pcomp <- as(sample_data(Bac.subset.pcomp), "data.frame") #sample df

FieldSamps.pcomp <- phyloseq::distance(Bac.subset.pcomp, method = "wunifrac")
     FieldSamps.df.pcomp <- as(sample_data(Bac.subset.pcomp), "data.frame")

      set.seed(30)
    adonis2(FieldSamps.pcomp ~ Type*Time.Point, data = FieldSamps.df.pcomp, by="margin")
    
#Pvar
Bac.subset.pvar<-subset_samples(Bac.subset, Species=="Pavona_varians")
#make unifrac matrix
s.u.samp.df.pvar <- as(sample_data(Bac.subset.pvar), "data.frame") #sample df

FieldSamps.pvar <- phyloseq::distance(Bac.subset.pvar, method = "wunifrac")
     FieldSamps.df.pvar <- as(sample_data(Bac.subset.pvar), "data.frame")

      set.seed(30)
    adonis2(FieldSamps.pvar ~ Type*Time.Point, data = FieldSamps.df.pvar, by="margin")
    
#Pacu
Bac.subset.pacu<-subset_samples(Bac.subset, Species=="Pocillopora_acuta")
#make unifrac matrix
s.u.samp.df.pacu <- as(sample_data(Bac.subset.pacu), "data.frame") #sample df

FieldSamps.pacu <- phyloseq::distance(Bac.subset.pacu, method = "wunifrac")
     FieldSamps.df.pacu <- as(sample_data(Bac.subset.pacu), "data.frame")

      set.seed(30)
    adonis2(FieldSamps.pacu ~ Type*Time.Point, data = FieldSamps.df.pacu, by="margin")
``` 

#Alpha diversit: observed richness, Shannon (+evenness) - Hannah  
Use rarefied data, not relative abundances   
```{r}           
Sp.colors <- c("Montipora_capitata"="#E69F00","Porites_compressa"="#56B4E9","Pavona_varians"="#009E73","Pocillopora_acuta"="#CC79A7", "High"="darkRed","Ambient"="navyBlue")

Bac.al.alpha<-Bac.s #make a copy of df
Bac.al.alpha2 <- prune_taxa(taxa_sums(Bac.al.alpha) > 0, Bac.al.alpha) #prune taxa 0s
Bac.al.alpha2.sd <- as(sample_data(Bac.al.alpha2), "data.frame") #create sample data frame to view

##### microbiome package for evenness and alpha diversity. use raw counts ####
erich.tab <-microbiome::alpha(Bac.al.alpha2, index = "all") # pull out: observed, diversity_shannon, evenness_pielou

erich.tab2<-erich.tab[-c(2:4,6:8,10:22)] #keep the cols you need

 erich.tab2$Treatment <- Bac.al.alpha2.sd$Treatment       #add back in meta
  erich.tab2$Time.Point <- Bac.al.alpha2.sd$Time.Point
  erich.tab2$Parent.ID <- Bac.al.alpha2.sd$Parent.ID
  erich.tab2$ID <- Bac.al.alpha2.sd$Frag.ID
  erich.tab2$Species <- Bac.al.alpha2.sd$Species
    erich.tab2$Clade <- Bac.al.alpha2.sd$Clade
    erich.tab2$Tank.num <- Bac.al.alpha2.sd$Tank.num
erich.tab2$Time.Point = factor(erich.tab2$Time.Point, levels = c("T0", "T1", "TF"))

#all species - to show significance and to subset
  erich.tab2T0 <- subset(erich.tab2, Time.Point=="T0") #T0 to use below
  erich.tab2T0$Species <- factor(erich.tab2T0$Species,
  levels = c("Montipora_capitata", "Porites_compressa", "Pavona_varians", "Pocillopora_acuta"))

  erich.tab2 <- subset(erich.tab2, Time.Point=="T1"|Time.Point=="TF")
 erich.tab2$Species <- factor(erich.tab2$Species,
  levels = c("Montipora_capitata", "Porites_compressa", "Pavona_varians", "Pocillopora_acuta"))

 #ANOVA
#shannon
  aov.shannon.species = aov(diversity_shannon ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.shannon.species) # sig diff between species
#observed  
  aov.observed.species = aov(observed ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.observed.species) # sig diff between species
#evenness
  aov.evenness.species = aov(evenness_pielou ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.evenness.species) # sig diff between species
  TukeyHSD(aov.evenness.species)  

#by species:
erich.tab2.mcap<-subset(erich.tab2, Species=="Montipora_capitata")
erich.tab2.pcomp<-subset(erich.tab2, Species=="Porites_compressa")
erich.tab2.pvar<-subset(erich.tab2, Species=="Pavona_varians")
erich.tab2.pacu<-subset(erich.tab2, Species=="Pocillopora_acuta")
```
#Observed Richness ANOVA
Tables: S7, S10, S19
```{r}      

#T0 ####
hist(erich.tab2T0$observed)
ggqqplot(erich.tab2T0$observed)
erich.tab2T0$observed_log<-log10(erich.tab2T0$observed)
 hist(erich.tab2T0$observed_log)

observed_alpha_modT0<-lmer(observed_log ~ Species+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)

#Richness
p1 <- ggviolin(erich.tab2T0, x = "Species", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
#Evenness
p1 <- ggviolin(erich.tab2T0, x = "Species", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

#T1TF: Table S7 S12C ####
  erich.tab2.T1F<-subset(erich.tab2, Time.Point=="T1"|Time.Point=="TF")
  erich.tab2.T1F$observed_log<-log10(erich.tab2.T1F$observed)
# By species
  erich.tab2.mcapT1F<-subset(erich.tab2.T1F, Species=="Montipora_capitata")
  erich.tab2.pcompT1F<-subset(erich.tab2.T1F, Species=="Porites_compressa")
  erich.tab2.pvarT1F<-subset(erich.tab2.T1F, Species=="Pavona_varians")
  erich.tab2.pacuT1F<-subset(erich.tab2.T1F, Species=="Pocillopora_acuta")

#Mcap ####
  hist(erich.tab2.mcapT1F$observed_log)
  ggqqplot(erich.tab2.mcapT1F$observed_log)
  
 #T0 
  erich.tab2.mcap.T0<-subset(erich.tab2T0, Species=="Montipora_capitata")

  Mcap_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.mcap.T0) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod))
  
 #T1
  erich.tab2.mcap.T1<-subset(erich.tab2.mcapT1F, Time.Point=="T1")

  Mcap_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.T1) 
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  
  Mcap_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.T1) 
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  
  #TF
  erich.tab2.mcap.TF<-subset(erich.tab2.mcapT1F, Time.Point=="TF")

  Mcap_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.TF) 
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  
  Mcap_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.TF) 
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Clade*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

#Pcomp ####
  hist(erich.tab2.pcompT1F$observed_log)
  ggqqplot(erich.tab2.pcompT1F$observed_log)
  
 #T0 
  erich.tab2.pcomp.T0<-subset(erich.tab2T0, Species=="Porites_compressa")

  Pcomp_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.pcomp.T0) #adding tank is singularity
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod))
  
 #T1
  erich.tab2.pcomp.T1<-subset(erich.tab2.pcompT1F, Time.Point=="T1")

  Pcomp_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pcomp.T1) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod)) 
  
  Pcomp_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.T1) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod)) 
  
  #TF
  erich.tab2.pcomp.TF<-subset(erich.tab2.pcompT1F, Time.Point=="TF")

  Pcomp_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pcomp.TF) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod)) 
  
  Pcomp_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.TF) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod)) 
  emm<-emmeans(Pcomp_alpha_mod,  ~ Clade*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)


#Pvar ####
  hist(erich.tab2.pvarT1F$observed_log)
  ggqqplot(erich.tab2.pvarT1F$observed_log)
  
 #T0 
  erich.tab2.pvar.T0<-subset(erich.tab2T0, Species=="Pavona_varians")

  Pvar_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.pvar.T0) #adding tank is singularity
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod))
  
 #T1
  erich.tab2.pvar.T1<-subset(erich.tab2.pvarT1F, Time.Point=="T1")

  Pvar_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pvar.T1) 
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod)) 
  
  Pvar_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.T1) 
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod)) 
  
  #TF
  erich.tab2.pvar.TF<-subset(erich.tab2.pvarT1F, Time.Point=="TF")

  Pvar_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pvar.TF) 
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod)) 
  
  Pvar_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.TF) 
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod)) 


#Pacu ####
  hist(erich.tab2.pacuT1F$observed_log)
  ggqqplot(erich.tab2.pacuT1F$observed_log)
  
 #T1
  erich.tab2.pacu.T1<-subset(erich.tab2.pacuT1F, Time.Point=="T1")

  Pacu_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pacu.T1) 
  anova(Pacu_alpha_mod) 
  qqPlot(residuals(Pacu_alpha_mod)) 
  
  #TF
  erich.tab2.pacu.TF<-subset(erich.tab2.pacuT1F, Time.Point=="TF")

  Pacu_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pacu.TF) 
  anova(Pacu_alpha_mod) 
  qqPlot(residuals(Pacu_alpha_mod)) 
```

Alpha Diversity - Observed Richness Plots: Figures S17
```{r}   
# color pal from eta mcap.g.Parent.colors
alpha.colors <- c( "Ambient" = "deepskyblue1", "High" = "orangered",
                        "C Ambient"="#1F78B4","C High"="#6A3D9A", "CD Ambient"="#E31A1C", "CD High"="#FF7F00") 


# df erich.tab2

RichnessT1TF<- ggplot(erich.tab2, aes(x=interaction(Treatment,Time.Point,Species), y=observed, fill=Treatment)) + 
      scale_fill_manual(values=c("deepskyblue1", "orangered"))+
      ggtitle("Observed Richness")+
      xlab("") + #Label the X Axis
      theme(axis.text.y = element_text(size=14))+
      theme(axis.title.y = element_text(size=14))+
      theme(plot.title = element_text(size=16, face="bold.italic"))+
      theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
      panel.background = element_blank(), axis.line = element_line(colour = "black")) +
      geom_boxplot(show.legend = F)+
          scale_x_discrete(guide = "axis_nested");RichnessT1TF
ggsave("16S_RichnessT1TF.pdf")

#by algal symbiont lineage
erich.tab2_noPacu<-subset(erich.tab2, Species!="Pocillopora_acuta")

    RichnessT1TF_clade<- ggplot(erich.tab2_noPacu, aes(x=interaction(Treatment,Clade,Time.Point, Species), y=observed, fill=Treatment)) + 
      scale_fill_manual(values=c("deepskyblue1", "orangered"))+
      ggtitle("Observed Richness")+
            xlab("") + #Label the X Axis
      theme(axis.text.x = element_text(size=14))+
      theme(axis.text.y = element_text(size=14))+
      theme(axis.title.y = element_text(size=14))+
      theme(plot.title = element_text(size=16, face="bold.italic"))+
      theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
      panel.background = element_blank(), axis.line = element_line(colour = "black")) +
      geom_boxplot(show.legend = F)+
          scale_x_discrete(guide = "axis_nested");RichnessT1TF_clade
    ggsave("16S_RichnessT1TF_clade.pdf", width=20, height=7)

```
 
#Evenness_pielou 
Tables: S7, S10, S19
```{r}   
#evenness_pielou  

#T0 ####
hist(erich.tab2T0$evenness_pielou)
ggqqplot(erich.tab2T0$evenness_pielou)

observed_alpha_modT0<-lmer(evenness_pielou ~ Species+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)

#Mcap ####
  hist(erich.tab2.mcapT1F$evenness_pielou)
  ggqqplot(erich.tab2.mcapT1F$evenness_pielou)
  
 #T0 
  erich.tab2.mcap.T0<-subset(erich.tab2T0, Species=="Montipora_capitata")

  Mcap_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.mcap.T0) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod))
  
 #T1
  erich.tab2.mcap.T1<-subset(erich.tab2.mcapT1F, Time.Point=="T1")

  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.T1) 
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.T1) 
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  
  #TF
  erich.tab2.mcap.TF<-subset(erich.tab2.mcapT1F, Time.Point=="TF")

  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.TF) 
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.TF) 
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 

#Pcomp ####
  hist(erich.tab2.pcompT1F$evenness_pielou)
  ggqqplot(erich.tab2.pcompT1F$evenness_pielou)
  
 #T0 
  erich.tab2.pcomp.T0<-subset(erich.tab2T0, Species=="Porites_compressa")

  Pcomp_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.pcomp.T0) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod))
  
 #T1
  erich.tab2.pcomp.T1<-subset(erich.tab2.pcompT1F, Time.Point=="T1")

  Pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pcomp.T1) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod)) 
  
  Pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.T1) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod)) 
  
  #TF
  erich.tab2.pcomp.TF<-subset(erich.tab2.pcompT1F, Time.Point=="TF")

  Pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pcomp.TF) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod)) 
  
  Pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.TF) 
  anova(Pcomp_alpha_mod) 
  qqPlot(residuals(Pcomp_alpha_mod)) 

#Pvar ####
  hist(erich.tab2.pvarT1F$evenness_pielou)
  ggqqplot(erich.tab2.pvarT1F$evenness_pielou)
  
 #T0 
  erich.tab2.pvar.T0<-subset(erich.tab2T0, Species=="Pavona_varians")

  Pvar_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.pvar.T0) #adding tank is singularity
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod))
  
 #T1
  erich.tab2.pvar.T1<-subset(erich.tab2.pvarT1F, Time.Point=="T1")

  Pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pvar.T1) 
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod)) 
  
  Pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.T1) 
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod)) 
  
  #TF
  erich.tab2.pvar.TF<-subset(erich.tab2.pvarT1F, Time.Point=="TF")

  Pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pvar.TF) 
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod)) 
  
  Pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.TF) 
  anova(Pvar_alpha_mod) 
  qqPlot(residuals(Pvar_alpha_mod)) 


#Pacu ####
  hist(erich.tab2.pacuT1F$evenness_pielou)
  ggqqplot(erich.tab2.pacuT1F$evenness_pielou)
  
 #T1
  erich.tab2.pacu.T1<-subset(erich.tab2.pacuT1F, Time.Point=="T1")

  Pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pacu.T1) 
  anova(Pacu_alpha_mod) 
  qqPlot(residuals(Pacu_alpha_mod)) 
  
  #TF
  erich.tab2.pacu.TF<-subset(erich.tab2.pacuT1F, Time.Point=="TF")

  Pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pacu.TF) 
  anova(Pacu_alpha_mod) 
  qqPlot(residuals(Pacu_alpha_mod)) 

```
Evenness Plots: Figure S18
```{r}   
EvennessT1TF<- ggplot(erich.tab2, aes(x=interaction(Treatment,Time.Point,Species), y=evenness_pielou, fill=Treatment)) + 
      scale_fill_manual(values=c("deepskyblue1", "orangered"))+
      ggtitle(" Evenness")+
      xlab("") + #Label the X Axis
      theme(axis.text.y = element_text(size=14))+
      theme(axis.title.y = element_text(size=14))+
      theme(plot.title = element_text(size=16, face="bold.italic"))+
      theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
      panel.background = element_blank(), axis.line = element_line(colour = "black")) +
      geom_boxplot(show.legend = F)+
          scale_x_discrete(guide = "axis_nested");EvennessT1TF
ggsave("16S_EvennessT1TF.pdf")

#by algal symbiont lineage
#no pacu
erich.tab2_noPacu<-subset(erich.tab2, Species!="Pocillopora_acuta")
    EvennessT1TF_clade<- ggplot(erich.tab2_noPacu, aes(x=interaction(Treatment,Clade,Time.Point, Species), y=evenness_pielou, fill=Treatment)) + 
      scale_fill_manual(values=c("deepskyblue1", "orangered"))+
      ggtitle(" Evenness")+
            xlab("") + #Label the X Axis
      theme(axis.text.x = element_text(size=14))+
      theme(axis.text.y = element_text(size=14))+
      theme(axis.title.y = element_text(size=14))+
      theme(plot.title = element_text(size=16, face="bold.italic"))+
      theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
      panel.background = element_blank(), axis.line = element_line(colour = "black")) +
      geom_boxplot(show.legend = F)+
          scale_x_discrete(guide = "axis_nested");EvennessT1TF_clade
    ggsave("16S_EvennessT1TF_clade.pdf", width=20, height=7)

```


#Mantel Test
T0 only
```{r}
# Identify which samples are represented in the its2 and 16s dfs

# T0 only  #####
#set up 16s
    Bac.s.rel.u.T0<-subset_samples(Bac.s.rel, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s
  # look at sample names
    bac.T0.sd <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df
    #save just sample names
      bac.T0.sd.samps<-bac.T0.sd[['sample_name.1']]

#set up ITS2 T0, import  (already pruned)
    ITS2_T0_mantel <- readRDS("Data/ITS2_T0_phyloseq.rds")
  # look at sample names
    ITS.T0.sd <- as(sample_data(ITS2_T0_mantel), "data.frame") #sample df
  #save just sample names
      ITS.T0.sd.samps<-ITS.T0.sd[['ID']]

# compare ITS to 16s at T0
  #identify missing samples btw datasets     
      setdiff(bac.T0.sd.samps,ITS.T0.sd.samps) #samples in 16s but not ITS: S208b, SN70b
      setdiff(ITS.T0.sd.samps,bac.T0.sd.samps) #samples in ITS but not 16s: "S208"  "SN60"  "SN70"  "S49"   "SN11"  "SN145" "SN40"  "SN111" "SN6"   "SN56"  "SN109"
        
      # in 16S change sample S208b name to S208

  # 16s changes: rename sample S208b = S208

    # pull OTU table
     OTU.16sT0 = as(otu_table(Bac.s.rel.u.T0), "matrix") #pull otu
       OTU.16sT0<-as.data.frame(OTU.16sT0) #make into df
        row.names(OTU.16sT0)[5] <- "S208"   #change row name by row number
                row.names(OTU.16sT0)[31] <- "SN70"   #change row name by row number

    #pull TAX table
     aTax.16sT0 = as(tax_table(Bac.s.rel.u.T0), "matrix") #pull tax
     
     #pull phy tree table
     tree.seq = phy_tree(Bac.s.rel.u.T0) #pull out tree

    #pull SAMPLE DATA
     aSD.16sT0 = as(sample_data(Bac.s.rel.u.T0), "matrix") #pull samp.data
     aSD.16sT0<-as.data.frame(aSD.16sT0) #change to df
        row.names(aSD.16sT0)[5] <- "S208"  #replace name S208b
        row.names(aSD.16sT0)[31] <- "SN70"  #replace name S208b
            

    #remake the phyloseq obj
      samdata = sample_data(aSD.16sT0)
        seqtab = otu_table(OTU.16sT0, taxa_are_rows = FALSE)
        taxtab = tax_table(aTax.16sT0)
        phy_tree = read_tree(phyTree.16sT0)
          Bac_T0_mantel = phyloseq(otu_table(seqtab), tax_table(taxtab), sample_data(samdata),phy_tree(tree.seq))
          
          
   #ITS2 - remove samps not in the 16s df - "SN60"  "S49"   "SN11"  "SN145" "SN40"  "SN111" "SN6"   "SN56"  "SN109" 
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN60" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "S49" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN11" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN145" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN40" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN111" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN6" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN56" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN109" )

        # both have 39 samples - Bac_T0_mantel ITS2_T0_mantel
      
        # reorder ps object so samples are in the same order
        library(microbiome)
        library(microViz)
        
        Bac_T0_mantel <- Bac_T0_mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac_T0_mantel) %>% head(8)
        
        ITS2_T0_mantel <- ITS2_T0_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS2_T0_mantel) %>% head(8)
        
  # MAKE data matrix
       library(vegan)
  
  #16s T0        
        unifrac.Bac.T0.mantel <- phyloseq::distance(Bac_T0_mantel, method = "wunifrac")
        
  #ITS2 T0
     bray.ITS2.T0.mantel <- phyloseq::distance(ITS2_T0_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T0.mantel, bray.ITS2.T0.mantel,  permutations=1000)
    
        
#### T0 Mcap only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for mcap
    Bac_T0_mantel.Mcap <- subset_samples(Bac_T0_mantel, Species == "Montipora_capitata")
    Bac_T0_mantel.Mcap = prune_taxa(taxa_sums(Bac_T0_mantel.Mcap) > 0, Bac_T0_mantel.Mcap) 
    
    ITS2_T0_mantel.Mcap  <- subset_samples(ITS2_T0_mantel, Species == "Montipora_capitata")
    ITS2_T0_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T0_mantel.Mcap) > 0, ITS2_T0_mantel.Mcap)
        
    #16s T0 mcap dist        
      unifrac.Bac_T0_mantel.Mcap <- phyloseq::distance(Bac_T0_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T0 mcap dist
     bray.ITS2.T0.mantel.Mcap <- phyloseq::distance(ITS2_T0_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Mcap, bray.ITS2.T0.mantel.Mcap,  permutations=1000)
    
#### T0 Pcomp only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pcomp
    Bac_T0_mantel.Pcomp <- subset_samples(Bac_T0_mantel, Species == "Porites_compressa")
    Bac_T0_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T0_mantel.Pcomp) > 0, Bac_T0_mantel.Pcomp) 
    
    ITS2_T0_mantel.Pcomp  <- subset_samples(ITS2_T0_mantel, Species == "Porites_compressa")
    ITS2_T0_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T0_mantel.Pcomp) > 0, ITS2_T0_mantel.Pcomp)
        
    #16s T0 Pcomp dist        
      unifrac.Bac_T0_mantel.Pcomp <- phyloseq::distance(Bac_T0_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T0 Pcomp dist
     bray.ITS2.T0.mantel.Pcomp <- phyloseq::distance(ITS2_T0_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pcomp, bray.ITS2.T0.mantel.Pcomp,  permutations=1000)        
        
        
  #### T0 Pvar only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pvar
    Bac_T0_mantel.Pvar <- subset_samples(Bac_T0_mantel, Species == "Pavona_varians")
    Bac_T0_mantel.Pvar = prune_taxa(taxa_sums(Bac_T0_mantel.Pvar) > 0, Bac_T0_mantel.Pvar) 
    
    ITS2_T0_mantel.Pvar  <- subset_samples(ITS2_T0_mantel, Species == "Pavona_varians")
    ITS2_T0_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T0_mantel.Pvar) > 0, ITS2_T0_mantel.Pvar)
        
    #16s T0 Pvar dist        
      unifrac.Bac_T0_mantel.Pvar <- phyloseq::distance(Bac_T0_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T0 Pvar dist
     bray.ITS2.T0.mantel.Pvar <- phyloseq::distance(ITS2_T0_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pvar, bray.ITS2.T0.mantel.Pvar,  permutations=1000)            
        
        
     #### T0 Pacu only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pacu
    Bac_T0_mantel.Pacu <- subset_samples(Bac_T0_mantel, Species == "Pavona_varians")
    Bac_T0_mantel.Pacu = prune_taxa(taxa_sums(Bac_T0_mantel.Pacu) > 0, Bac_T0_mantel.Pacu) 
    
    ITS2_T0_mantel.Pacu  <- subset_samples(ITS2_T0_mantel, Species == "Pavona_varians")
    ITS2_T0_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T0_mantel.Pacu) > 0, ITS2_T0_mantel.Pacu)
        
    #16s T0 Pacu dist        
      unifrac.Bac_T0_mantel.Pacu <- phyloseq::distance(Bac_T0_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T0 Pacu dist
     bray.ITS2.T0.mantel.Pacu <- phyloseq::distance(ITS2_T0_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pacu, bray.ITS2.T0.mantel.Pacu,  permutations=1000)            
        
        
    


```
T1 High only
```{r}
# figure out which samples are represented in the its2 and 16s dfs

# T1 only  #####

#high only 

#set up 16s
    Bac.s.rel.u.T1<-subset_samples(Bac.s.rel, Time.Point=="T1")
    Bac.s.rel.u.T1.High<-subset_samples(Bac.s.rel.u.T1, Treatment=="High")

    Bac.s.rel.u.T1.High = prune_taxa(taxa_sums(Bac.s.rel.u.T1.High) > 0, Bac.s.rel.u.T1.High) #this removes any OTU with 0s
  # look at sample names
    bac.T1.sd <- as(sample_data(Bac.s.rel.u.T1.High), "data.frame") #sample df
    #save just sample names
      bac.T1.High.sd.samps<-bac.T1.sd[['sample_name.1']]

#set up ITS2 T1, import  T1TF
    ITS2_T1TF_mantel <- readRDS("Data/ITS2_T1TF_phyloseq.rds") 

     ITS2_T1_mantel<-subset_samples(ITS2_T1TF_mantel, Time.Point=="T1")
    ITS2_T1_mantel.High<-subset_samples(ITS2_T1_mantel, Treatment=="High")
      ITS2_T1_mantel.High = prune_taxa(taxa_sums(ITS2_T1_mantel.High) > 0, ITS2_T1_mantel.High) #this removes any OTU with 0s

  # look at sample names
    ITS.T1.High.sd <- as(sample_data(ITS2_T1_mantel.High), "data.frame") #sample df
  #save just sample names
      ITS.T1.High.sd.samps<-ITS.T1.High.sd[['ID']]

# compare ITS to 16s at T1
  #identify missing samples btw datasets     
      setdiff(bac.T1.High.sd.samps,ITS.T1.High.sd.samps) #samples in 16s but not ITS: "S1857" "S3469" "SN134"
      setdiff(ITS.T1.High.sd.samps,bac.T1.High.sd.samps) #samples in ITS but not 16s: "S1857b" "S3468"  "SN134b" "S670"   "S3025"  "S1388a"
        
      # in 16S remove S3469
      #in ITS change S1857b to S1857 and SN134b to SN134. Remove S3468, S670, S3025, S1388a

  # 16s changes: rename sample S3469
      
    Bac.T1.High.mantel <- subset_samples(Bac.s.rel.u.T1.High, sample_name.1 != "S3469" )
        #40 samples
          
   #ITS2 - remove samps not in the 16s df - " S3468, S670, S3025, S1388a
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S3468" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S670" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S3025" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S1388a" )

        
    #change names ITS2 S1857b to S1857 and SN134b to SN134

    # pull OTU table
     OTU.ITS2T1.High = as(otu_table(ITS2_T1_mantel.High), "matrix") #pull otu
       OTU.ITS2T1.High<-as.data.frame(OTU.ITS2T1.High) #make into df
        row.names(OTU.ITS2T1.High)[7] <- "S1857"   #change row name by row number
                row.names(OTU.ITS2T1.High)[17] <- "SN134"   #change row name by row number

    #pull TAX table
     aTax.ITS2T1.High = as(tax_table(ITS2_T1_mantel.High), "matrix") #pull tax

    #pull SAMPLE DATA
     aSD.ITS2T1.High = as(sample_data(ITS2_T1_mantel.High), "matrix") #pull samp.data
     aSD.ITS2T1.High<-as.data.frame(aSD.ITS2T1.High) #change to df
        row.names(aSD.ITS2T1.High)[7] <- "S1857"  #replace name S208b
        row.names(aSD.ITS2T1.High)[17] <- "SN134"  #replace name S208b
            

    #remake the phyloseq obj
      samdata.High = sample_data(aSD.ITS2T1.High)
        seqtab.High = otu_table(OTU.ITS2T1.High, taxa_are_rows = FALSE)
        taxtab.High = tax_table(aTax.ITS2T1.High)
          ITS_T1.High_mantel = phyloseq(otu_table(seqtab.High), tax_table(taxtab.High), sample_data(samdata.High))
             
        # both have 40 samples - 
      
        # reorder ps object so samples are in the same order
        
        Bac.T1.High.mantel <- Bac.T1.High.mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac.T1.High.mantel) %>% head(8)
        
        ITS_T1.High_mantel <- ITS_T1.High_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS_T1.High_mantel) %>% head(8)
        
  # MAKE data matrix
  
  #16s T1 High        
        unifrac.Bac.T1.High.mantel <- phyloseq::distance(Bac.T1.High.mantel, method = "wunifrac")
        
  #ITS2 T1 High
     bray.ITS2.T1.High.mantel <- phyloseq::distance(ITS_T1.High_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T1.High.mantel, bray.ITS2.T1.High.mantel,  permutations=1000)
    
        
#### T1.High Mcap only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for mcap
    Bac_T1.High_mantel.Mcap <- subset_samples(Bac.T1.High.mantel, Species == "Montipora_capitata")
    Bac_T1.High_mantel.Mcap = prune_taxa(taxa_sums(Bac_T1.High_mantel.Mcap) > 0, Bac_T1.High_mantel.Mcap) 
    
    ITS2_T1.High_mantel.Mcap  <- subset_samples(ITS_T1.High_mantel, Species == "Montipora_capitata")
    ITS2_T1.High_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Mcap) > 0, ITS2_T1.High_mantel.Mcap)
        
    #16s T1.High mcap dist        
      unifrac.Bac_T1.High_mantel.Mcap <- phyloseq::distance(Bac_T1.High_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T1.High mcap dist
     bray.ITS2.T1.High.mantel.Mcap <- phyloseq::distance(ITS2_T1.High_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Mcap, bray.ITS2.T1.High.mantel.Mcap,  permutations=1000)
    
#### T1.High Pcomp only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pcomp
    Bac_T1.High_mantel.Pcomp <- subset_samples(Bac.T1.High.mantel, Species == "Porites_compressa")
    Bac_T1.High_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pcomp) > 0, Bac_T1.High_mantel.Pcomp) 
    
    ITS2_T1.High_mantel.Pcomp  <- subset_samples(ITS_T1.High_mantel, Species == "Porites_compressa")
    ITS2_T1.High_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pcomp) > 0, ITS2_T1.High_mantel.Pcomp)
        
    #16s T1.High Pcomp dist        
      unifrac.Bac_T1.High_mantel.Pcomp <- phyloseq::distance(Bac_T1.High_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T1.High Pcomp dist
     bray.ITS2.T1.High.mantel.Pcomp <- phyloseq::distance(ITS2_T1.High_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pcomp, bray.ITS2.T1.High.mantel.Pcomp,  permutations=1000)        
        
    
#### T1.High Pvar only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pvar
    Bac_T1.High_mantel.Pvar <- subset_samples(Bac.T1.High.mantel, Species == "Pavona_varians")
    Bac_T1.High_mantel.Pvar = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pvar) > 0, Bac_T1.High_mantel.Pvar) 
    
    ITS2_T1.High_mantel.Pvar  <- subset_samples(ITS_T1.High_mantel, Species == "Pavona_varians")
    ITS2_T1.High_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pvar) > 0, ITS2_T1.High_mantel.Pvar)
        
    #16s T1.High Pvar dist        
      unifrac.Bac_T1.High_mantel.Pvar <- phyloseq::distance(Bac_T1.High_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T1.High Pvar dist
     bray.ITS2.T1.High.mantel.Pvar <- phyloseq::distance(ITS2_T1.High_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pvar, bray.ITS2.T1.High.mantel.Pvar,  permutations=1000)       
      
      #### T1.High Pacu only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pacu
    Bac_T1.High_mantel.Pacu <- subset_samples(Bac.T1.High.mantel, Species == "Pocillopora_acuta")
    Bac_T1.High_mantel.Pacu = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pacu) > 0, Bac_T1.High_mantel.Pacu) 
    
    ITS2_T1.High_mantel.Pacu  <- subset_samples(ITS_T1.High_mantel, Species == "Pocillopora_acuta")
    ITS2_T1.High_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pacu) > 0, ITS2_T1.High_mantel.Pacu)
        
    #16s T1.High Pacu dist        
      unifrac.Bac_T1.High_mantel.Pacu <- phyloseq::distance(Bac_T1.High_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T1.High Pacu dist
     bray.ITS2.T1.High.mantel.Pacu <- phyloseq::distance(ITS2_T1.High_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pacu, bray.ITS2.T1.High.mantel.Pacu,  permutations=1000)  

```
T1 Ambient only
```{r}
# figure out which samples are represented in the its2 and 16s dfs

# T1 only  #####

#Ambient only 

#set up 16s
    Bac.s.rel.u.T1<-subset_samples(Bac.s.rel.u, Time.Point=="T1")
    Bac.s.rel.u.T1.Ambient<-subset_samples(Bac.s.rel.u.T1, Treatment=="Ambient")

    Bac.s.rel.u.T1.Ambient = prune_taxa(taxa_sums(Bac.s.rel.u.T1.Ambient) > 0, Bac.s.rel.u.T1.Ambient) #this removes any OTU with 0s
  # look at sample names
    bac.T1.sd <- as(sample_data(Bac.s.rel.u.T1.Ambient), "data.frame") #sample df
    #save just sample names
      bac.T1.Ambient.sd.samps<-bac.T1.sd[['sample_name.1']]

#set up ITS2 T1, import  T1TF
     ITS2_T1_mantel<-subset_samples(ITS2_T1TF_mantel, Time.Point=="T1")
    ITS2_T1_mantel.Ambient<-subset_samples(ITS2_T1_mantel, Treatment=="Ambient")
      ITS2_T1_mantel.Ambient = prune_taxa(taxa_sums(ITS2_T1_mantel.Ambient) > 0, ITS2_T1_mantel.Ambient) #this removes any OTU with 0s

    
  # look at sample names
    ITS.T1.Ambient.sd <- as(sample_data(ITS2_T1_mantel.Ambient), "data.frame") #sample df
  #save just sample names
      ITS.T1.Ambient.sd.samps<-ITS.T1.Ambient.sd[['ID']]

# compare ITS to 16s at T1
  #identify missing samples btw datasets     
      setdiff(bac.T1.Ambient.sd.samps,ITS.T1.Ambient.sd.samps) #samples in 16s but not ITS: "S1443b" "SN108b" "SN197b" "SN222"
      setdiff(ITS.T1.Ambient.sd.samps,bac.T1.Ambient.sd.samps) #samples in ITS but not 16s: "S1573" "S2424" "S3035" "S1443" "S1895" "SN209" "SN63"  "S1418" "SN197" "SN126" "S785"  "SN108"
        
      # in 16S remove SN222,remove the bs on the rest "S1443b" "SN108b" "SN197b"
      #in ITS  Remove "S1573" "S2424" "S3035" "S1895" "SN209" "SN63"  "S1418" " "SN126" "S785"  

      
    #######stoped here#########  
      
      
      
      
      
  # 16s changes: rename sample S3469
      
    Bac.T1.Ambient.mantel <- subset_samples(Bac.s.rel.u.T1.Ambient, sample_name.1 != "S3469" )
        #40 samples
          
   #ITS2 - remove samps not in the 16s df - " S3468, S670, S3025, S1388a
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S3468" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S670" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S3025" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S1388a" )

        
    #change names ITS2 S1857b to S1857 and SN134b to SN134

    # pull OTU table
     OTU.ITS2T1.Ambient = as(otu_table(ITS2_T1_mantel.Ambient), "matrix") #pull otu
       OTU.ITS2T1.Ambient<-as.data.frame(OTU.ITS2T1.Ambient) #make into df
        row.names(OTU.ITS2T1.Ambient)[7] <- "S1857"   #change row name by row number
                row.names(OTU.ITS2T1.Ambient)[17] <- "SN134"   #change row name by row number

    #pull TAX table
     aTax.ITS2T1.Ambient = as(tax_table(ITS2_T1_mantel.Ambient), "matrix") #pull tax

    #pull SAMPLE DATA
     aSD.ITS2T1.Ambient = as(sample_data(ITS2_T1_mantel.Ambient), "matrix") #pull samp.data
     aSD.ITS2T1.Ambient<-as.data.frame(aSD.ITS2T1.Ambient) #change to df
        row.names(aSD.ITS2T1.Ambient)[7] <- "S1857"  #replace name S208b
        row.names(aSD.ITS2T1.Ambient)[17] <- "SN134"  #replace name S208b
            

    #remake the phyloseq obj
      samdata.Ambient = sample_data(aSD.ITS2T1.Ambient)
        seqtab.Ambient = otu_table(OTU.ITS2T1.Ambient, taxa_are_rows = FALSE)
        taxtab.Ambient = tax_table(aTax.ITS2T1.Ambient)
          ITS_T1.Ambient_mantel = phyloseq(otu_table(seqtab.Ambient), tax_table(taxtab.Ambient), sample_data(samdata.Ambient))
             
        # both have 40 samples - 
      
        # reorder ps object so samples are in the same order
        
        Bac.T1.Ambient.mantel <- Bac.T1.Ambient.mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac.T1.Ambient.mantel) %>% head(8)
        
        ITS_T1.Ambient_mantel <- ITS_T1.Ambient_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS_T1.Ambient_mantel) %>% head(8)
        
  # MAKE data matrix
  
  #16s T1 Ambient        
        unifrac.Bac.T1.Ambient.mantel <- phyloseq::distance(Bac.T1.Ambient.mantel, method = "wunifrac")
        
  #ITS2 T1 Ambient
     bray.ITS2.T1.Ambient.mantel <- phyloseq::distance(ITS_T1.Ambient_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T1.Ambient.mantel, bray.ITS2.T1.Ambient.mantel,  permutations=1000)
    
        
#### T1.Ambient Mcap only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for mcap
    Bac_T1.Ambient_mantel.Mcap <- subset_samples(Bac.T1.Ambient.mantel, Species == "Montipora_capitata")
    Bac_T1.Ambient_mantel.Mcap = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Mcap) > 0, Bac_T1.Ambient_mantel.Mcap) 
    
    ITS2_T1.Ambient_mantel.Mcap  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Montipora_capitata")
    ITS2_T1.Ambient_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Mcap) > 0, ITS2_T1.Ambient_mantel.Mcap)
        
    #16s T1.Ambient mcap dist        
      unifrac.Bac_T1.Ambient_mantel.Mcap <- phyloseq::distance(Bac_T1.Ambient_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T1.Ambient mcap dist
     bray.ITS2.T1.Ambient.mantel.Mcap <- phyloseq::distance(ITS2_T1.Ambient_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Mcap, bray.ITS2.T1.Ambient.mantel.Mcap,  permutations=1000)
    
#### T1.Ambient Pcomp only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pcomp
    Bac_T1.Ambient_mantel.Pcomp <- subset_samples(Bac.T1.Ambient.mantel, Species == "Porites_compressa")
    Bac_T1.Ambient_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pcomp) > 0, Bac_T1.Ambient_mantel.Pcomp) 
    
    ITS2_T1.Ambient_mantel.Pcomp  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Porites_compressa")
    ITS2_T1.Ambient_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pcomp) > 0, ITS2_T1.Ambient_mantel.Pcomp)
        
    #16s T1.Ambient Pcomp dist        
      unifrac.Bac_T1.Ambient_mantel.Pcomp <- phyloseq::distance(Bac_T1.Ambient_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T1.Ambient Pcomp dist
     bray.ITS2.T1.Ambient.mantel.Pcomp <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pcomp, bray.ITS2.T1.Ambient.mantel.Pcomp,  permutations=1000)        
        
    
#### T1.Ambient Pvar only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pvar
    Bac_T1.Ambient_mantel.Pvar <- subset_samples(Bac.T1.Ambient.mantel, Species == "Pavona_varians")
    Bac_T1.Ambient_mantel.Pvar = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pvar) > 0, Bac_T1.Ambient_mantel.Pvar) 
    
    ITS2_T1.Ambient_mantel.Pvar  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Pavona_varians")
    ITS2_T1.Ambient_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pvar) > 0, ITS2_T1.Ambient_mantel.Pvar)
        
    #16s T1.Ambient Pvar dist        
      unifrac.Bac_T1.Ambient_mantel.Pvar <- phyloseq::distance(Bac_T1.Ambient_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T1.Ambient Pvar dist
     bray.ITS2.T1.Ambient.mantel.Pvar <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pvar, bray.ITS2.T1.Ambient.mantel.Pvar,  permutations=1000)       
      
      #### T1.Ambient Pacu only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pacu
    Bac_T1.Ambient_mantel.Pacu <- subset_samples(Bac.T1.Ambient.mantel, Species == "Pocillopora_acuta")
    Bac_T1.Ambient_mantel.Pacu = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pacu) > 0, Bac_T1.Ambient_mantel.Pacu) 
    
    ITS2_T1.Ambient_mantel.Pacu  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Pocillopora_acuta")
    ITS2_T1.Ambient_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pacu) > 0, ITS2_T1.Ambient_mantel.Pacu)
        
    #16s T1.Ambient Pacu dist        
      unifrac.Bac_T1.Ambient_mantel.Pacu <- phyloseq::distance(Bac_T1.Ambient_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T1.Ambient Pacu dist
     bray.ITS2.T1.Ambient.mantel.Pacu <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pacu, bray.ITS2.T1.Ambient.mantel.Pacu,  permutations=1000)  

```
