---
title: "16S Thermal Tolerance"
html_document:
code_folding: hide
toc:yes
toc_depth:3
toc_float:yes
output: 
chunk_output_type: console
editor_options: 
  chunk_output_type: console
---

16s data analysis
# load in 16S to phyloseq
```{r}

library(phyloseq)
library(readr)
```

Load in data
```{r}
#read in sample data
MetaData16<-read.csv("Data/Meta_16s_new.csv") #run in physio script first
MetaData16$Parent.ID<-as.factor(as.character(MetaData16$Parent.ID))
MetaData16$Tank.num<-as.factor(as.character(MetaData16$Tank.num))

sam0 <- MetaData16
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))


#load in OUT:raw_abundanceTable_100.shared
OTU2<-read.table("Data/TT17_3000_summer_16S-pipeline_outputs/raw_abundanceTable_100_shared.csv", sep=',', header=T)

#from its3
#otu
otu5 <- as.matrix(OTU2[, -1])
rownames(otu5) <- OTU2$sample_name
otu <- otu_table(otu5, taxa_are_rows = FALSE)


#tax
TAX<- read.csv("Data/TT17_3000_summer_16S-pipeline_outputs/raw_consensusClassification_100_taxonomy.csv", colClasses = "character")
tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- tax0$OTU
tax <- tax_table(tax1)

# Read the data into phyloseq
Bac.seqAll = phyloseq(otu, tax,sam) #THIS WORKS
Bac.seqAll

save(Bac.seqAll, file = "Data/Bac.seqAll_phyloseq.RData")

```

Look at the data
```{r}
## check out data
ntaxa(Bac.seqAll)  #num taxa
nsamples(Bac.seqAll)   #num samples
sample_names(Bac.seqAll)[1:300] #samp names
 #sampsy<-sample_names(physeq)[1:300]
#write.csv(sampsy,"sampsy.csv")

# OK Now you need to do the steps in the pipeline.
#1 filter out taxa: get rid of unknown, mitochondria and chloroplast
Bac.seqAll<-subset_taxa(Bac.seqAll, Family !="Mitochondria")
Bac.seqAll<-subset_taxa(Bac.seqAll, Order !="Chloroplast")
Bac.seqAll<-subset_taxa(Bac.seqAll, Taxonomy !="unknown")
Bac.seqAll<-subset_taxa(Bac.seqAll, Taxonomy !="Archaea")

# now get rid of things not represnted 2x or more
#Prune singletons (these are reads that are only found once)
Bac.seqAll.prune <- prune_taxa(taxa_sums(Bac.seqAll) > 2, Bac.seqAll)

  save(Bac.seqAll, file = "Data/Bac.seqAll.prune.RData")
  
```

#16S
Load in your data
```{r}
# Load data (see data_exploration.Rmd)
load("Data/Bac.seqAll.prune.RData") #use RAW data that has been pruned. 
# the df is Bac.seq
rank_names(Bac.seqAll) #ranks are DIV and clade


#remove field samples
Bac.s = subset_samples(Bac.seqAll, Type== "sample" )

# #Bac.seqAll all not 3k use to check who you lose
# # create df of sample data to view 
# sample.dataAll <- as(sample_data(Bac.seqAll), "data.frame") #create sample data frame to view
# sample.dataAll$LibrarySize <- sample_sums(Bac.seqAll)
# #sample.dataAll <- sample.data[order(sample.dataAll$LibrarySize),]
# #sample.dataAll$Index <- seq(nrow(sample.dataAll))  # check 3k rare
# ggplot(data = sample.dataAll, aes(x=Index, y=LibrarySize, color = Species)) +
#   geom_point()
# write.csv(sample.dataAll, "sampscheck.csv")

## check out data
ntaxa(Bac.s)  #num taxa
nsamples(Bac.s)   #num samples
sample_names(Bac.s)[1:5] #samp names
rank_names(Bac.s) #ranks are DIV and clade
sample_variables(Bac.s) #all metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(Bac.s), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.s)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  # check 3k rare
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = Species)) +
  geom_point()



#remove duplicates 
Dups<-subset_samples(Bac.s, Dups=="Yes")

TopNOTUsB = names(sort(taxa_sums(Dups), TRUE)[1:100])
bac50 = prune_taxa(TopNOTUsB, Dups)
bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot
#ggsave("mMcapBarPlot_DIVrelARare.pdf", McapBarPlot, dpi=300, width=8)
Bac.s <- subset_samples(Bac.s, realsample_name != "S1753b" )
Bac.s <- subset_samples(Bac.s, realsample_name != "S208" )
Bac.s <- subset_samples(Bac.s, realsample_name != "S3476" )
Bac.s <- subset_samples(Bac.s, realsample_name != "SN35b" )
Bac.s <- subset_samples(Bac.s, realsample_name != "SN111b" )

Bac.s <- subset_samples(Bac.s, realsample_name != "Field" )


# Subsample to 2K
Bac.s2 <- prune_samples(sample_sums(Bac.s)>=1764, Bac.s) #keep things over 2k
#Rarefied to 2000 reads
Bac.r <- rarefy_even_depth(Bac.s2, sample.size = 1764, rngseed = 711) 


#### Create a duplicate of T0s for the other A and H treatments for comparision: df coralDIV2_rare
#not
controlT0_Amb = subset_samples(Bac.r, Time.Point == "T0"&Treatment == "Ambient")  #copy

#tax table
tax  = tax_table(controlT0_Amb)

  # sample data
  sd = sample_data(controlT0_Amb)
  rownames(sd) <- paste0(rownames(sd), "_dup")
  # otu_table
  OTU = otu_table(controlT0_Amb)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  # sd$Treatment <- as.character(sd$Treatment)
  # sd$ID <- as.character(sd$ID)
  sd$Treatment <- "High"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
  sd$Treatment <- as.factor(sd$Treatment)
  # sd$ID <- as.factor(sd$ID)

  Amb2<-phyloseq(OTU, sd, tax)
#sample.dataAmb2 <- as(sample_data(Amb2), "data.frame") #create sample data frame to view

##High
  controlT0_High = subset_samples(Bac.s, Time.Point == "T0"&Treatment == "High")  #copy

#tax table
tax  = tax_table(controlT0_High)

  # sample data
  sd = sample_data(controlT0_High)
  rownames(sd) <- paste0(rownames(sd), "_dup")
# otu_table
OTU = otu_table(controlT0_High)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  sd$Treatment <- as.character(sd$Treatment)
  # sd$ID <- as.character(sd$ID)
  sd$Treatment <- "Ambient"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
  sd$Treatment <- as.factor(sd$Treatment)
  sd$ID <- as.factor(sd$ID)

  High2<-phyloseq(OTU, sd, tax)
  #sample.dataHigh2 <- as(sample_data(High2), "data.frame") #create sample data frame to view

#Merge control dups H and A  
Controldups = merge_phyloseq(Amb2,High2)
Controldups
    #sample.dataControldups2 <- as(sample_data(Controldups), "data.frame") #create sample data frame to view

#merg them to the original df:

Bac.r2 = merge_phyloseq(Bac.r,Controldups)
Bac.r2
Bac.r2.sd <- as(sample_data(Bac.r2), "data.frame") #create sample data frame to view
Bac.r2.sd

```
Set up, Bac.r2.sd <- this df is rarefied raw counts
```{r}
#make relative abundance df Bac.s2
Bac.r.rel  = transform_sample_counts(Bac.r2, function(x) x / sum(x) )  #save as RELA DIVs
data.Bac.r.rel <- as(sample_data(Bac.r.rel), "data.frame") #look at samp data

#heatmap
  #plot_heatmap(Bac.RelA, method = "NMDS", distance = "bray")# way too much data

```


16 bar plots in ggplot (to make faster than next section)  
#MCAP
PHYLUM
```{r}
#try glom use raw counts Bac.r2
Mcap_Bac.r2<-subset_samples(Bac.r2, Species=="Montipora_capitata")

Mcap_Bac.r2.phy <- Mcap_Bac.r2 %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Mcap_Bac.r2.melt <- psmelt(Mcap_Bac.r2.phy)

# sum.ra.mcap <- ddply(Mcap_Bac.r2.melt, c("Phylum", "Time.Point", "Treatment","Parent.ID"), summarise,
#                 N = length(Abundance),
#                 mean = mean(Abundance),
#                 sd = sd(Abundance), 
#                 se = sd/sqrt(N)
# )
# sum.ra.mcap

Mcap_Bac.r2.melt$Code<-paste(Mcap_Bac.r2.melt$Treatment, Mcap_Bac.r2.melt$Time.Point)
Mcap_Bac.r2.melt$Parent.ID <- factor(Mcap_Bac.r2.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Mcap_Bac.r2.melt$Code <- factor(Mcap_Bac.r2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

mcap.p <- ggplot(Mcap_Bac.r2.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Mcap_Bac.r2.C<-subset_samples(Mcap_Bac.r2.melt, Clade=="C")

mcap.p <- ggplot(Mcap_Bac.r2.C, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap C")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Mcap_Bac.r2.CD<-subset_samples(Mcap_Bac.r2.melt, Clade=="CD")

mcap.p <- ggplot(Mcap_Bac.r2.CD.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

```
CLASS
```{r}
#try glom use raw counts Bac.r2
Mcap_Bac.r2<-subset_samples(Bac.r2, Species=="Montipora_capitata")

Mcap_Bac.r2.Cla <- Mcap_Bac.r2 %>% 
  tax_glom(taxrank = "Class", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Mcap_Bac.r2.melt <- psmelt(Mcap_Bac.r2.Cla)

Mcap_Bac.r2.melt$Code<-paste(Mcap_Bac.r2.melt$Treatment, Mcap_Bac.r2.melt$Time.Point)
Mcap_Bac.r2.melt$Parent.ID <- factor(Mcap_Bac.r2.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Mcap_Bac.r2.melt$Code <- factor(Mcap_Bac.r2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 123
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

mcap.p <- ggplot(Mcap_Bac.r2.melt, aes(x = Parent.ID, y = Abundance, fill = Class)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Mcap_Bac.r2.melt.C<-subset(Mcap_Bac.r2.melt, Clade=="C")

mcap.p <- ggplot(Mcap_Bac.r2.melt.C, aes(x = Parent.ID, y = Abundance, fill = Class)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Mcap_Bac.r2.melt.CD<-subset(Mcap_Bac.r2.melt, Clade=="CD")


mcap.p <- ggplot(Mcap_Bac.r2.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Class)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Mcap CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)



```
Family
```{r}
#try glom use raw counts Bac.r2
Mcap_Bac.r2.Fam <- Mcap_Bac.r2 %>% 
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Mcap_Bac.r2.melt <- psmelt(Mcap_Bac.r2.Fam)

Mcap_Bac.r2.melt$Code<-paste(Mcap_Bac.r2.melt$Treatment, Mcap_Bac.r2.melt$Time.Point)
Mcap_Bac.r2.melt$Parent.ID <- factor(Mcap_Bac.r2.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Mcap_Bac.r2.melt$Code <- factor(Mcap_Bac.r2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 123
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

mcap.p <- ggplot(Mcap_Bac.r2.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Mcap_Bac.r2.melt.C<-subset(Mcap_Bac.r2.melt, Clade=="C")

mcap.p <- ggplot(Mcap_Bac.r2.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Mcap_Bac.r2.melt.CD<-subset(Mcap_Bac.r2.melt, Clade=="CD")


mcap.p <- ggplot(Mcap_Bac.r2.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Mcap CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)



```
#Pcomp
PHYLUM
```{r}
#try glom use raw counts Bac.r2
Pcomp_Bac.r2<-subset_samples(Bac.r2, Species=="Porites_compressa")

Pcomp_Bac.r2.phy <- Pcomp_Bac.r2 %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pcomp_Bac.r2.melt <- psmelt(Pcomp_Bac.r2.phy)

Pcomp_Bac.r2.melt$Code<-paste(Pcomp_Bac.r2.melt$Treatment, Pcomp_Bac.r2.melt$Time.Point)
Pcomp_Bac.r2.melt$Parent.ID <- factor(Pcomp_Bac.r2.melt$Parent.ID, levels = c("313",  "314","315","316", "317","318","319",  "320", "322","323",  "324","321"))

Pcomp_Bac.r2.melt$Code <- factor(Pcomp_Bac.r2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pcomp.p <- ggplot(Pcomp_Bac.r2.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pcomp")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)



### by C15cj or C15n
# C15cj ONLY

Pcomp_Bac.r2.C15cj<-subset(Pcomp_Bac.r2.melt, Clade=="C15cj")

Pcomp.p <- ggplot(Pcomp_Bac.r2.C15cj, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C15cj")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

# C15n ONLY
Pcomp_Bac.r2.C15n<-subset(Pcomp_Bac.r2.melt, Clade=="C15n")

Pcomp.p <- ggplot(Pcomp_Bac.r2.C15n, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C15n")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

```
CLASS
```{r}
#try glom use raw counts Bac.r2

Pcomp_Bac.r2.Cla <- Pcomp_Bac.r2 %>% 
  tax_glom(taxrank = "Class", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pcomp_Bac.r2.melt <- psmelt(Pcomp_Bac.r2.Cla)

Pcomp_Bac.r2.melt$Code<-paste(Pcomp_Bac.r2.melt$Treatment, Pcomp_Bac.r2.melt$Time.Point)
Pcomp_Bac.r2.melt$Parent.ID <- factor(Pcomp_Bac.r2.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Pcomp_Bac.r2.melt$Code <- factor(Pcomp_Bac.r2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 123
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pcomp.p <- ggplot(Pcomp_Bac.r2.melt, aes(x = Parent.ID, y = Abundance, fill = Class)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Pcomp_Bac.r2.melt.C<-subset(Pcomp_Bac.r2.melt, Clade=="C")

Pcomp.p <- ggplot(Pcomp_Bac.r2.melt.C, aes(x = Parent.ID, y = Abundance, fill = Class)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Pcomp_Bac.r2.melt.CD<-subset(Pcomp_Bac.r2.melt, Clade=="CD")


Pcomp.p <- ggplot(Pcomp_Bac.r2.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Class)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Pcomp CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)



```
Family
```{r}
#try glom use raw counts Bac.r2
Pcomp_Bac.r2.Fam <- Pcomp_Bac.r2 %>% 
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pcomp_Bac.r2.melt <- psmelt(Pcomp_Bac.r2.Fam)

Pcomp_Bac.r2.melt$Code<-paste(Pcomp_Bac.r2.melt$Treatment, Pcomp_Bac.r2.melt$Time.Point)
Pcomp_Bac.r2.melt$Parent.ID <- factor(Pcomp_Bac.r2.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Pcomp_Bac.r2.melt$Code <- factor(Pcomp_Bac.r2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 123
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pcomp.p <- ggplot(Pcomp_Bac.r2.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Pcomp_Bac.r2.melt.C<-subset(Pcomp_Bac.r2.melt, Clade=="C")

Pcomp.p <- ggplot(Pcomp_Bac.r2.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Pcomp_Bac.r2.melt.CD<-subset(Pcomp_Bac.r2.melt, Clade=="CD")


Pcomp.p <- ggplot(Pcomp_Bac.r2.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Pcomp CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)



```
#Pvar
PHYLUM
```{r}
#try glom use raw counts Bac.r2
Pvar_Bac.r2<-subset_samples(Bac.r2, Species=="Pavona_varians")

Pvar_Bac.r2.phy <- Pvar_Bac.r2 %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pvar_Bac.r2.melt <- psmelt(Pvar_Bac.r2.phy)

Pvar_Bac.r2.melt$Code<-paste(Pvar_Bac.r2.melt$Treatment, Pvar_Bac.r2.melt$Time.Point)
Pvar_Bac.r2.melt$Parent.ID <- factor(Pvar_Bac.r2.melt$Parent.ID, levels = c("361",  "358","353","356", "325","354",  "359", "357","362", "351", "355","360"))

Pvar_Bac.r2.melt$Code <- factor(Pvar_Bac.r2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pvar.p <- ggplot(Pvar_Bac.r2.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pvar")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)



### by C1 or C127
# C15cj ONLY

Pvar_Bac.r2.C15cj<-subset(Pvar_Bac.r2.melt, Clade=="C1")

Pvar.p <- ggplot(Pvar_Bac.r2.C15cj, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pvar C1")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)

# C15n ONLY
Pvar_Bac.r2.C27<-subset(Pvar_Bac.r2.melt, Clade=="C27")

Pvar.p <- ggplot(Pvar_Bac.r2.C27, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pvar C27")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)

```
#Pacu
PHYLUM
```{r}
#try glom use raw counts Bac.r2
Pacu_Bac.r2<-subset_samples(Bac.r2, Species=="Pocillopora_acuta")

Pacu_Bac.r2.phy <- Pacu_Bac.r2 %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pacu_Bac.r2.melt <- psmelt(Pacu_Bac.r2.phy)

Pacu_Bac.r2.melt$Code<-paste(Pacu_Bac.r2.melt$Treatment, Pacu_Bac.r2.melt$Time.Point)
Pacu_Bac.r2.melt$Parent.ID <- factor(Pacu_Bac.r2.melt$Parent.ID, levels = c("304",  "308","312","303", "306","302",  "310", "301","311", "307", "309","305"))

Pacu_Bac.r2.melt$Code <- factor(Pacu_Bac.r2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pacu.p <- ggplot(Pacu_Bac.r2.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pacu")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)



### by CCC or C42
# CCC ONLY

Pacu_Bac.r2.CCC<-subset(Pacu_Bac.r2.melt, Clade=="CCC")

Pacu.p <- ggplot(Pacu_Bac.r2.CCC, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pacu CCC")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)

# C42 ONLY
Pacu_Bac.r2.C42<-subset(Pacu_Bac.r2.melt, Clade=="C42")

Pacu.p <- ggplot(Pacu_Bac.r2.C42, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pacu C42")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)

```

#16s - bar plots df Bac.s2RelA YOU Don't need now
```{r} 
#ordination time - CUT WHEN YOU HAVE OTHER 4 SPECIES ORDINATION SORTED 
Bac.r.rel.ord <- ordinate(Bac.r.rel, "NMDS", "bray", set.seed(30))
Bac.RelA.ord_plot = plot_ordination(Bac.r.rel, Bac.r.rel.ord, type="samples", shape="Treatment",color="Species", title="species")
print(Bac.RelA.ord_plot)
Bac.RelA.ord_plot + facet_wrap(~Species, 3)

### let's do this again and run the NMDS by species now that we know they are different. 
#subset by species
Mcap_16s_RA <- subset_samples(Bac.r.rel, Species=="Montipora_capitata")
Pcomp_16s_RA <- subset_samples(Bac.r.rel, Species=="Porites_compressa")
Pvar_16s_RA <- subset_samples(Bac.r.rel, Species=="Pavona_varians")
Pacu_16s_RA <- subset_samples(Bac.r.rel, Species=="Pocillopora_acuta")

#top 50 DIVs across by species 

#Mcap
 TopNOTUsMcap = names(sort(taxa_sums(Mcap_16s_RA), TRUE)[1:300])
 Mcapent50 = prune_taxa(TopNOTUsMcap, Mcap_16s_RA)
McapBarPlot<-plot_bar(Mcap_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot
#ggsave("mMcapBarPlot_16srelARare.pdf", McapBarPlot, dpi=300, width=8)


#### try with gg
Mcap_16s_RA.gg<-phyloseq_to_df(Mcap_16s_RA, addtax = T, addtot = F, addmaxrank = F,
  sorting = "abundance")
Mcap_16s_RA.gg.Phylum<-subset(Mcap_16s_RA.gg, select=-c(2:3,5:8))
#convert data frame from a "wide" format to a "long" format
pcm = melt(Mcap_16s_RA.gg, id = c("OTU","Size","Phylum"))

  mx = ggplot(Mcap_16s_RA.gg, aes(x = Parent.ID, fill = Phylum, y = value)) + 
    geom_bar(stat = "identity", colour = "black") + 
    theme(axis.text.x = element_text(angle = 90, size = 14, colour = "black", vjust = 0.5, hjust = 1, face= "bold"), 
    axis.title.y = element_text(size = 16, face = "bold"), legend.title = element_text(size = 16, face = "bold"), 
    legend.text = element_text(size = 12, face = "bold", colour = "black"), 
    axis.text.y = element_text(colour = "black", size = 12, face = "bold")) + 
    scale_y_continuous(expand = c(0,0)) + 
    labs(x = "", y = "Relative Abundance (%)", fill = "OTU") + 
    scale_fill_manual(values = colours)
    
  mx




#Mcap C and CD
#C
Mcap_16s_RA.C<-subset_samples(Mcap_16s_RA, Clade=="C")
TopNOTUsMcap.C = names(sort(taxa_sums(Mcap_16s_RA.C), TRUE)[1:300])
Mcap.50.C = prune_taxa(TopNOTUsMcap.C, Mcap_16s_RA.C)
McapBarPlot.C<-plot_bar(Mcap_16s_RA.C, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot.C
#ggsave("McapBarPlot_16s_C.pdf", McapBarPlot.C, dpi=300, width=8)

#CD
Mcap_16s_RA.CD<-subset_samples(Mcap_16s_RA, Clade=="CD")
TopNOTUsMcap.CD = names(sort(taxa_sums(Mcap_16s_RA.CD), TRUE)[1:300])
 Mcap.50.CD = prune_taxa(TopNOTUsMcap.CD, Mcap_16s_RA.CD)
McapBarPlot.CD<-plot_bar(Mcap_16s_RA.CD, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot.CD
#ggsave("McapBarPlot_16s_CD.pdf", McapBarPlot.CD, dpi=300, width=8)


#Pcomp
TopNOTUsPcomp = names(sort(taxa_sums(Pcomp_16s_RA), TRUE)[1:300])
# Pcompent50 = prune_taxa(TopNOTUsPcomp,Pcomp_16s_RA )
PcompBarPlot<-plot_bar(Pcomp_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot
#ggsave("PcompBarPlot_16srelARare.pdf", PcompBarPlot, dpi=300, width=8)

#C15cj and C15n
  #C15cj
  Pcomp_16s_RA.C15cj<-subset_samples(Pcomp_16s_RA, Clade=="C15cj")
  TopNOTUsPcomp.C15cj = names(sort(taxa_sums(Pcomp_16s_RA.C15cj), TRUE)[1:300])
  Pcomp.50.C15cj = prune_taxa(TopNOTUsPcomp.C15cj, Pcomp_16s_RA.C15cj)
  PcompBarPlot.C15cj<-plot_bar(Pcomp_16s_RA.C15cj, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot.C15cj
  #ggsave("PcompBarPlot_16s_C.pdf", PcompBarPlot.C15cj, dpi=300, width=8)

  #C15n
  Pcomp_16s_RA.C15n<-subset_samples(Pcomp_16s_RA, Clade=="C15n")
  TopNOTUsPcomp.C15n = names(sort(taxa_sums(Pcomp_16s_RA.C15n), TRUE)[1:300])
   Pcomp.50.C15n = prune_taxa(TopNOTUsPcomp.C15n, Pcomp_16s_RA.C15n)
  PcompBarPlot.C15n<-plot_bar(Pcomp_16s_RA.C15n, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot.C15n
#ggsave("PcompBarPlot_16s_CD.pdf", PcompBarPlot.C15n, dpi=300, width=8)


#Pvar
TopNOTUsVar = names(sort(taxa_sums(Pvar_16s_RA), TRUE)[1:300])
Pvarent50 = prune_taxa(TopNOTUsVar, Pvar_16s_RA)
PvarBarPlot<-plot_bar(Pvar_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot
#ggsave("PvarBarPlot_16srelARare.pdf", PvarBarPlot, dpi=300, width=8)


#C1 and C27
  #C15cj
  Pvar_16s_RA.C1<-subset_samples(Pvar_16s_RA, Clade=="C1")
  TopNOTUsPvar.C1 = names(sort(taxa_sums(Pvar_16s_RA.C1), TRUE)[1:300])
  Pvar.50.C1 = prune_taxa(TopNOTUsPvar.C1, Pvar_16s_RA.C1)
  PvarBarPlot.C1<-plot_bar(Pvar_16s_RA.C1, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot.C1
  #ggsave("PvarBarPlot_16s_C.pdf", PvarBarPlot.C15cj, dpi=300, width=8)

  #C27
  Pvar_16s_RA.C27<-subset_samples(Pvar_16s_RA, Clade=="C27")
  TopNOTUsPvar.C27 = names(sort(taxa_sums(Pvar_16s_RA.C27), TRUE)[1:300])
   Pvar.50.C27 = prune_taxa(TopNOTUsPvar.C27, Pvar_16s_RA.C27)
  PvarBarPlot.C27<-plot_bar(Pvar_16s_RA.C27, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot.C27
  #ggsave("PvarBarPlot_16s_CD.pdf", PvarBarPlot.C27, dpi=300, width=8)



#Pacu
# TopNOTUsPacu = names(sort(taxa_sums(Pacu_16s_RA), TRUE)[1:300])
Pacuent50 = prune_taxa(TopNOTUsPacu, Pacu_16s_RA)
PacuBarPlot<-plot_bar(Pacu_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot
#ggsave("PacuBarPlot_16srelARare.pdf", PacuBarPlot, dpi=300, width=8)

#CCC and C42
  #CCC
  Pacu_16s_RA.CCC<-subset_samples(Pacu_16s_RA, Clade=="CCC")
  TopNOTUsPacu.CCC = names(sort(taxa_sums(Pacu_16s_RA.CCC), TRUE)[1:300])
  Pacu.50.CCC = prune_taxa(TopNOTUsPacu.CCC, Pacu_16s_RA.CCC)
  PacuBarPlot.CCC<-plot_bar(Pacu_16s_RA.CCC, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot.CCC
  #ggsave("PacuBarPlot_16s_C.pdf", PacuBarPlotCCC5cj, dpi=300, width=8)

  #C42
  Pacu_16s_RA.C42<-subset_samples(Pacu_16s_RA, Clade=="C42")
  TopNOTUsPacu.C42 = names(sort(taxa_sums(Pacu_16s_RA.C42), TRUE)[1:300])
   Pacu.50.C42 = prune_taxa(TopNOTUsPacu.C42, Pacu_16s_RA.C42)
  PacuBarPlot.C42<-plot_bar(Pacu_16s_RA.C42, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot.C42
  #ggsave("PacuBarPlot_16s_CD.pdf", PacuBarPlot.C42, dpi=300, width=8)
```


#Ordination DIV_RelA <- you might not need this section, code below for each species  
try vegan nmds - its the same, delete when CN says cool, Bac.r.rel
```{r} 
  ##this isn't working, not converging. 
# convert the sample_data() within a phyloseq object to a vegan compatible data object
df16s2veg <- function(Bac.r.rel) { 
  sd <- sample_data(Bac.r.rel)
  return(as(sd,"data.frame"))
}

# convert the otu_table() within a phyloseq object to a vegan compatible data object
df16sotu2veg <- function(Bac.r.rel) {
  OTU <- otu_table(Bac.r.rel)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

# Calculate distance matrix
df16s.spp_distmat <- vegdist(OTU, method = "bray")
# Creating easy to view matrix and writing .csv
df16s.spp_distmat <- 
  as.matrix(df16s.spp_distmat, labels = T)
#write.csv(island.spp_distmat, "island_spp_distmat.csv")

# Running NMDS in vegan (metaMDS)
df16s.spp_NMS <-
  metaMDS(df16s.spp_distmat,
          distance = "bray",
          k =2,
          maxit = 999, 
          trymax = 500,
          wascores = TRUE, set.seed(10))

# Shepards test/goodness of fit
goodness(df16s.spp_NMS) # Produces a results of test statistics for goodness of fit for each point

stressplot(df16s.spp_NMS) # Produces a Shepards diagram

# Plotting points in ordination space
plot(df16s.spp_NMS, "sites")   # Produces distance 
orditorp(df16s.spp_NMS, "sites")   # Gives points labels




```
Heatmap and first crack at ordination, delete when solid
```{r}         
# CUT?? OrdiSpecies<-plot_ordination(DIV_RelA, ordinate(DIV_RelA, "NMDS"), color = "Species", label="ID") + geom_point(size = 5) 
# #ggsave("OrdiSpecies.pdf", OrdiSpecies, dpi=300, width=8)

  
#heat map
plot_heatmap(Bac.r.rel, "NMDS", "bray", "Species", "Phylum")

#by species 
  #mcap
Mcapent50.T1<- subset_samples(Mcapent50, Time.Point=="T1")
Mcapent50.T1.C<- subset_samples(Mcapent50.T1, Clade=="C")
Mcapent50.T1.CD<- subset_samples(Mcapent50.T1, Clade=="CD")

Mcapent50.TF<- subset_samples(Mcapent50, Time.Point=="TF")


  plot_heatmap(Mcapent50.T1, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split
  plot_heatmap(Mcapent50.T1.C, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split
  plot_heatmap(Mcapent50.T1.CD, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split

 #Pcomp
  Pcomp_DIV_RA_T1 <- subset_samples(Pcomp_DIV_RA, Time.Point=="T1")
  plot_heatmap(Pcomp_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
  #Pvar
  Pvar_DIV_RA_T1 <- subset_samples(Pvar_DIV_RA, Time.Point=="T1")
  plot_heatmap(Pvar_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
  plot_heatmap(Pvar_DIV_RA, "NMDS", "bray", "Parent.ID", "DIV")

  
  
  
  
#ordination time - CUT WHEN YOU HAVE OTHER 4 SPECIES ORDINATION SORTED 
DIVrareRel.ord <- ordinate(DIV_RelA, "NMDS", "bray", set.seed(30))
DIVrareRel1 = plot_ordination(DIV_RelA, DIVrareRel.ord, type="samples", shape="Treatment",color="Time.Point", title="species")
print(DIVrareRel1)
DIVrareRel1 + facet_wrap(~Species, 3)

### let's do this again and run the NMDS by species now that we know they are different. 
# mcap
Mcap_DIV_RA.ord <- ordinate(Mcap_DIV_RA, "NMDS", "bray", set.seed(30))
DIVrareRel1_Mcap = plot_ordination(Mcap_DIV_RA, Mcap_DIV_RA.ord, type="samples", shape="Treatment",color="Time.Point", title="Mcap NMDS")
print(DIVrareRel1_Mcap) 
  #add ellipses
  DIVrareRel1_Mcap + 
  stat_ellipse(type = "norm", linetype = 2, alpha=.2) +
  stat_ellipse(type = "t") +
  theme_bw()

# pcomp
Pcomp_DIV_RA.ord <- ordinate(Pcomp_DIV_RA, "NMDS", "bray", set.seed(30))
DIVrareRel1_Pcomp = plot_ordination(Pcomp_DIV_RA, Pcomp_DIV_RA.ord, type="samples", shape="Treatment",color="Time.Point", title="Pcomp NMDS")
print(DIVrareRel1_Pcomp)
#add ellipses
  DIVrareRel1_Pcomp + 
  stat_ellipse(type = "norm",  alpha=.2) +
  theme_bw()
```

#Beta Diversity: df rarefied and relative abudnaces  
All Species Bac.r.rel
```{r}      
#Bac.r.rel 

Bac.r.rel_stats<- Bac.r.rel #make a copy
otu_table(Bac.r.rel_stats)[1:5, 1:5] #check rela worked

#get rid of NA OTUs now!
Bac.r.rel_stats = prune_taxa(taxa_sums(Bac.r.rel_stats) > 0, Bac.r.rel_stats) #this removes any OTU with 0s


#bray curtis presence-absense and abundance
## REMOVE T0 for models
Bac_RelA_stats.T1F<-subset_samples(Bac.r.rel_stats, Time.Point=="T1"|Time.Point=="TF")

#bray curtis presence-absense and abundance

#all speices
set.seed(30)

#make bray curtis data matrix
bc.all <- phyloseq::distance(Bac_RelA_stats.T1F, method = "bray")
  samp.df <- as(sample_data(Bac_RelA_stats.T1F), "data.frame") #sample df
#adonis test
adonis(bc.all ~ Species*Treatment*Time.Point, data = samp.df)
    adonis(bc.all ~ Species*Treatment*Time.Point*Clade, data = samp.df)


# Homogeneity of dispersion test
betaAll <- betadisper(bc.all, samp.df$Species)
permutest(betaAll, pairwise = T) 
(bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
plot(bc.all.HSD)


#Ordination NMDS for all species, T1Tf
set.seed(50)
ord.Bac_RelA_stats <- ordinate(Bac_RelA_stats.T1F, "NMDS", "bray", trymax = 500) 
stressplot(ord.Bac_RelA_stats)
scores.ord.Bac_RelA_stats<-as.data.frame(cbind(vegan::scores(ord.Bac_RelA_stats, display="sites"))) # these are "points" ask hannah 

scores.ord.Bac_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.Bac_RelA_stats$Species <- samp.df$Species


ggplot(scores.ord.Bac_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")

```



#Split data by species 
Beta div
Montipora - T1 and TF together, separet, and then by phylotype. PERMANOVA
```{r}   
# df Bac_RelA_stats.T1F
#Mcap
mcap.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Montipora_capitata")
mcap.bac.T1F = prune_taxa(taxa_sums(mcap.bac.T1F) > 0, mcap.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
mcap.bac.T1 <- subset_samples(mcap.bac.T1F, Time.Point == "T1")
  mcap.bac.T1 = prune_taxa(taxa_sums(mcap.bac.T1) > 0, mcap.bac.T1) #this removes any OTU with 0s

mcap.bac.TF <- subset_samples(mcap.bac.T1F, Time.Point == "TF")
  mcap.bac.TF = prune_taxa(taxa_sums(mcap.bac.TF) > 0, mcap.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.mcap.T1F <- phyloseq::distance(mcap.bac.T1F, method = "bray")
     mcap.samp.df.T1F <- as(sample_data(mcap.bac.T1F), "data.frame") #sample df
    adonis(bc.mcap.T1F ~ Treatment*Time.Point, data = mcap.samp.df.T1F) #adonis test no clade
    adonis(bc.mcap.T1F ~ Clade*Treatment*Time.Point, data = mcap.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(mcap.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
  bc.mcapT1 <- phyloseq::distance(mcap.bac.T1, method = "bray") 
      mcap.samp.df.T1 <- as(sample_data(mcap.bac.T1), "data.frame") 
    adonis(bc.mcapT1 ~ Clade*Treatment, data = mcap.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Clade)  #clade
    permutest(beta.mcap.T1, pairwise = T) 
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Treatment) #treatment
    permutest(beta.mcap.T1, pairwise = T) 

#TF
  bc.mcap.TF <- phyloseq::distance(mcap.bac.TF, method = "bray")
     mcap.samp.df.TF <- as(sample_data(mcap.bac.TF), "data.frame") 
    adonis(bc.mcap.TF ~ Clade*Treatment, data = mcap.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Clade) 
    permutest(beta.mcap.TF, pairwise = T) 
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Treatment) 
    permutest(beta.mcap.TF, pairwise = T) 

######### by phylotype: C vs CD
# C ONLY
  mcap.bac.T1F.C <- subset_samples(mcap.bac.T1F, Clade == "C")
  mcap.bac.T1.C <- subset_samples(mcap.bac.T1, Clade == "C")
  mcap.bac.TF.C <- subset_samples(mcap.bac.TF, Clade == "C")
  
#T1TF.C
  bc.mcap.T1F.C <- phyloseq::distance(mcap.bac.T1F.C, method = "bray")
    mcap.samp.df.T1F.C <- as(sample_data(mcap.bacT1F.C), "data.frame") 
  adonis(bc.mcap.T1F.C ~ Treatment*Time.Point, data = mcap.samp.df.T1F.C)

#T1.C
  bc.mcapT1.C <- phyloseq::distance(mcap.bac.T1.C, method = "bray")
    mcap.samp.df.T1.C <- as(sample_data(mcap.bac.T1.C), "data.frame") #sample df
  adonis(bc.mcapT1.C ~ Treatment, data = mcap.samp.df.T1.C)
  beta.mcap.T1.C <- betadisper(bc.mcapT1.C, mcap.samp.df.T1.C$Treatment)
    permutest(beta.mcap.T1.C, pairwise = T) 

#TF.C
  bc.mcap.TF.C <- phyloseq::distance(mcap.bac.TF.C, method = "bray")
    mcap.samp.df.TF.C <- as(sample_data(mcap.bac.TF.C), "data.frame") #sample df
  adonis(bc.mcap.TF.C ~ Treatment, data = mcap.samp.df.TF.C)
  beta.mcap.TF.C <- betadisper(bc.mcap.TF.C, mcap.samp.df.TF.C$Treatment)
    permutest(beta.mcap.TF.C, pairwise = T) 

# CD ONLY
mcap.bac.T1F.CD <- subset_samples(mcap.bac.T1F, Clade == "CD")
mcap.bac.T1.CD <- subset_samples(mcap.bac.T1, Clade == "CD")
mcap.bac.TF.CD <- subset_samples(mcap.bac.TF, Clade == "CD")
  
#T1TF.CD
  bc.mcap.T1F.CD <- phyloseq::distance(mcap.bac.T1F.CD, method = "bray")
  mcap.samp.df.T1F.CD <- as(sample_data(mcap.bac.T1F.CD), "data.frame") 
  adonis(bc.mcap.T1F.CD ~ Treatment*Time.Point, data = mcap.samp.df.T1F.CD)

#T1.CD
  bc.mcapT1.CD <- phyloseq::distance(mcap.bac.T1.CD, method = "bray")
    mcap.samp.df.T1.CD <- as(sample_data(mcap.bac.T1.CD), "data.frame") 
  adonis(bc.mcapT1.CD ~ Treatment, data = mcap.samp.df.T1.CD)
    beta.mcap.T1.CD <- betadisper(bc.mcapT1.CD, mcap.samp.df.T1.CD$Treatment)
  permutest(beta.mcap.T1.CD, pairwise = T) 

#TF.CD
  bc.mcap.TF.CD <- phyloseq::distance(mcap.bac.TF.CD, method = "bray")
    mcap.samp.df.TF.CD <- as(sample_data(mcap.bac.TF.CD), "data.frame") 
  adonis(bc.mcap.TF.CD ~ Treatment, data = mcap.samp.df.TF.CD)
  beta.mcap.TF.CD <- betadisper(bc.mcap.TF.CD, mcap.samp.df.TF.CD$Treatment)
    permutest(beta.mcap.TF.CD, pairwise = T) 
``` 
Mcap NMDS
```{r}
## create palette for parent.ID and treatments
Mcap.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", Ambient = "Blue", High = "Red","C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred", "Ambient T1"="darkred", "Ambient TF"="darkblue", "High T1"="pink","High TF"="darkred") 

#T1 and TF
ord.mcap2 <- ordinate(mcap.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap2)

scores.mcap.samp.df<-as.data.frame(cbind(vegan::scores(ord.mcap2, display="sites"))) # these are "points" ask hannah 
scores.mcap.samp.df$Time.Point <- mcap.samp.df.T1F$Time.Point
scores.mcap.samp.df$Parent.ID <- mcap.samp.df.T1F$Parent.ID
scores.mcap.samp.df$Treatment <- mcap.samp.df.T1F$Treatment
scores.mcap.samp.df$Clade <- mcap.samp.df.T1F$Clade
scores.mcap.samp.df$Code <- paste(mcap.samp.df.T1F$Parent.ID,mcap.samp.df.T1F$Time.Point)
scores.mcap.samp.df$Code2 <- paste(mcap.samp.df.T1F$Treatment,mcap.samp.df.T1F$Time.Point)
scores.mcap.samp.df$Code3 <- paste(mcap.samp.df.T1F$Treatment,mcap.samp.df.T1F$Time.Point,mcap.samp.df.T1F$Parent.ID)

ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2), linetype = 2) +
  ggtitle("M. capitata T1TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#T1
ord.mcap.T1 <- ordinate(mcap.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap.T1)

ord.mcap.T1.df<-as.data.frame(cbind(vegan::scores(ord.mcap.T1, display="sites"))) # these are "points" ask hannah 
ord.mcap.T1.df$Time.Point <- mcap.samp.df.T1$Time.Point
ord.mcap.T1.df$Parent.ID <- mcap.samp.df.T1$Parent.ID
ord.mcap.T1.df$Treatment <- mcap.samp.df.T1$Treatment
ord.mcap.T1.df$Clade <- mcap.samp.df.T1$Clade
ord.mcap.T1.df$Code <- paste(mcap.samp.df.T1$Parent.ID,mcap.samp.df.T1$Time.Point)

#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-Section-01-Main-Lab.html#procrustes-and-ggplot2 
ggplot(ord.mcap.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
    geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(mcap.bac.TF, ordinate(mcap.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.mcapTF <- ordinate(mcap.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcapTF)
ord.mcap.TF.df<-as.data.frame(cbind(vegan::scores(ord.mcapTF, display="sites"))) # these are "points" ask hannah 
ord.mcap.TF.df$Treatment <- mcap.samp.df.TF$Treatment
ord.mcap.TF.df$Parent.ID <- mcap.samp.df.TF$Parent.ID
ord.mcap.TF.df$Clade <- mcap.samp.df.TF$Clade
ord.mcap.TF.df$CladeTreat <- as.factor(paste(mcap.samp.df.TF$Clade,mcap.samp.df.TF$Treatment))
ord.mcap.TF.df$Code <- as.factor(paste(mcap.samp.df.TF$Parent.ID,mcap.samp.df.TF$Treatment))

ggplot(ord.mcap.TF.df, aes(x = NMDS1, y = NMDS2)) + 
 geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
    scale_color_manual(values=Mcap.Parent.colors)+
    geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")



#### BY CLADE
#ords


#C T1 and TF 
set.seed(30)
ord.mcap_C <- ordinate(mcap.bacT1F.C , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C)
scores.mcap_C<-as.data.frame(cbind(vegan::scores(ord.mcap_C, display="sites"))) # these are "points" ask hannah 
scores.mcap_C$Treatment <- mcap.samp.df.T1F.C$Treatment
scores.mcap_C$Parent.ID <- mcap.samp.df.T1F.C$Parent.ID
scores.mcap_C$Time.Point <- mcap.samp.df.T1F.C$Time.Point

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Cladocopium") +
  theme_classic()
#ggsave("McapTFits_C.pdf")


#C T1
set.seed(30)
ord.mcap_C.T1 <- ordinate(mcap.bac.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.T1)
scores.mcap_C.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_C.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.T1$Treatment <- mcap.samp.df.T1.C$Treatment
scores.mcap_C.T1$Parent.ID <- mcap.samp.df.T1.C$Parent.ID
scores.mcap_C.T1$Time.Point <- mcap.samp.df.T1.C$Time.Point

ggplot(scores.mcap_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("Mcap_T1_its_C.pdf")

#C TF
set.seed(30)
ord.mcap_C.TF <- ordinate(mcap.bac.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.TF)
scores.mcap_C.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_C.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.TF$Treatment <- mcap.samp.df.TF.C$Treatment
scores.mcap_C.TF$Parent.ID <- mcap.samp.df.TF.C$Parent.ID
scores.mcap_C.TF$Time.Point <- mcap.samp.df.TF.C$Time.Point

ggplot(scores.mcap_C.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("Mcap_TF_its_TF_C.pdf")


#CD mcap.bacT1F.CD
set.seed(30)
ord.mcap_CD <- ordinate(mcap.bacT1F.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD)
scores.mcap_CD<-as.data.frame(cbind(vegan::scores(ord.mcap_CD, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD$Treatment <- mcap.samp.df.T1F.CD$Treatment
scores.mcap_CD$Parent.ID <- mcap.samp.df.T1F.CD$Parent.ID
scores.mcap_CD$Time.Point <- mcap.samp.df.T1F.CD$Time.Point

ggplot(scores.mcap_CD, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Mcap.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.mcap_CD.T1 <- ordinate(mcap.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.T1)
scores.mcap_CD.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.T1$Treatment <- mcap.samp.df.T1.CD$Treatment
scores.mcap_CD.T1$Parent.ID <- mcap.samp.df.T1.CD$Parent.ID
scores.mcap_CD.T1$Time.Point <- mcap.samp.df.T1.CD$Time.Point

ggplot(scores.mcap_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_T1.pdf")

#CD TF
set.seed(30)
ord.mcap_CD.TF <- ordinate(mcap.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.TF)
scores.mcap_CD.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.TF$Treatment <- mcap.samp.df.TF.CD$Treatment
scores.mcap_CD.TF$Parent.ID <- mcap.samp.df.TF.CD$Parent.ID
scores.mcap_CD.TF$Time.Point <- mcap.samp.df.TF.CD$Time.Point

ggplot(scores.mcap_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Mcap_ITS2_CD_TF.pdf")



```














##### DELETE BELOW once redon above
Porites - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pcomp
Pcomp.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "porites")
Pcomp.bac.T1 <- subset_samples(Pcomp.bac.T1F, Time.Point == "T1")
Pcomp.bac.TF <- subset_samples(Pcomp.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F <- phyloseq::distance(Pcomp.bac.T1F, method = "bray")
  Pcomp.samp.df.T1F <- as(sample_data(Pcomp.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pcomp.T1F ~ Clade*Treatment*Time.Point, data = Pcomp.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PcompT1 <- phyloseq::distance(Pcomp.bac.T1, method = "bray")
  Pcomp.samp.df.T1 <- as(sample_data(Pcomp.bac.T1), "data.frame") #sample df
  adonis(bc.PcompT1 ~ Clade*Treatment, data = Pcomp.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Clade)
  permutest(beta.Pcomp.T1, pairwise = T) 
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Treatment)
  permutest(beta.Pcomp.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pcomp.TF <- phyloseq::distance(Pcomp.bac.TF, method = "bray")
  Pcomp.samp.df.TF <- as(sample_data(Pcomp.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF ~ Clade*Treatment, data = Pcomp.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Clade)
  permutest(beta.Pcomp.TF, pairwise = T) 
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Treatment)
  permutest(beta.Pcomp.TF, pairwise = T) 


## by clade: C vs CD
  
# C15cj only, new objs
Pcomp.bacT1F.C15cj <- subset_samples(Pcomp.bac.T1F, Clade == "C15cj")
Pcomp.bac.T1.C15cj <- subset_samples(Pcomp.bac.T1, Clade == "C15cj")
Pcomp.bac.TF.C15cj <- subset_samples(Pcomp.bac.TF, Clade == "C15cj")
  
#T1TF.C
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F.C15cj <- phyloseq::distance(Pcomp.bacT1F.C15cj, method = "bray")
  Pcomp.samp.df.T1F.C15cj <- as(sample_data(Pcomp.bacT1F.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.T1F.C15cj ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C15cj)

#T1.C15cj
#make bray curtis data matrix 
  bc.PcompT1.C15cj <- phyloseq::distance(Pcomp.bac.T1.C15cj, method = "bray")
  Pcomp.samp.df.T1.C15cj <- as(sample_data(Pcomp.bac.T1.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.PcompT1.C15cj ~ Treatment, data = Pcomp.samp.df.T1.C15cj)
# Homogeneity of dispersion test  
  beta.Pcomp.T1.C15cj <- betadisper(bc.PcompT1.C15cj, Pcomp.samp.df.T1.C15cj$Treatment)
  permutest(beta.Pcomp.T1.C15cj, pairwise = T) 

#TF.C15cj
#make bray curtis data matrix 
  bc.Pcomp.TF.C15cj <- phyloseq::distance(Pcomp.bac.TF.C15cj, method = "bray")
  Pcomp.samp.df.TF.C15cj <- as(sample_data(Pcomp.bac.TF.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF.C15cj ~ Treatment, data = Pcomp.samp.df.TF.C15cj)
# Homogeneity of dispersion test  
  beta.Pcomp.TF.C15cj <- betadisper(bc.Pcomp.TF.C15cj, Pcomp.samp.df.TF.C15cj$Treatment)
  permutest(beta.Pcomp.TF.C15cj, pairwise = T) 


# C15n only, new objs
Pcomp.bacT1F.C15n <- subset_samples(Pcomp.bac.T1F, Clade == "C15n")
Pcomp.bac.T1.C15n <- subset_samples(Pcomp.bac.T1, Clade == "C15n")
Pcomp.bac.TF.C15n <- subset_samples(Pcomp.bac.TF, Clade == "C15n")
  
#T1TF.C15n
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F.C15n <- phyloseq::distance(Pcomp.bacT1F.C15n, method = "bray")
  Pcomp.samp.df.T1F.C15n <- as(sample_data(Pcomp.bacT1F.C15n), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.T1F.C15n ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C15n)

#T1.C15n
#make bray curtis data matrix 
  bc.PcompT1.C15n <- phyloseq::distance(Pcomp.bac.T1.C15n, method = "bray")
  Pcomp.samp.df.T1.C15n <- as(sample_data(Pcomp.bac.T1.C15n), "data.frame") #sample df
  adonis(bc.PcompT1.C15n ~ Treatment, data = Pcomp.samp.df.T1.C15n)#adonis test
# Homogeneity of dispersion test  
  beta.Pcomp.T1.C15n <- betadisper(bc.PcompT1.C15n, Pcomp.samp.df.T1.C15n$Treatment)
  permutest(beta.Pcomp.T1.C15n, pairwise = T) 

#TF.C15n
#make bray curtis data matrix 
  bc.Pcomp.TF.C15n <- phyloseq::distance(Pcomp.bac.TF.C15n, method = "bray")
  Pcomp.samp.df.TF.C15n <- as(sample_data(Pcomp.bac.TF.C15n), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF.C15n ~ Treatment, data = Pcomp.samp.df.TF.C15n)
# Homogeneity of dispersion test  
  beta.Pcomp.TF.C15n <- betadisper(bc.Pcomp.TF.C15n, Pcomp.samp.df.TF.C15n$Treatment)
  permutest(beta.Pcomp.TF.C15n, pairwise = T) 
``` 
Pcomp NMDS
```{r}
library(RColorBrewer)
library(colorRamps)
library(devtools)
 
## create palette for parent.ID and treatments
Pcomp.Parent.colors <- c("313" = "green4",  "314" = "firebrick3", "315" ="deeppink4", "316" = "orange", "317" = "green1","318" = "gold",  "319"= "hotpink", "320" ="darkslateblue", "321" = "cyan", "322" = "darkorange","323" ="darkorchid1", "324" = "lightblue", "Ambient" = "Blue", "High" = "Red","C15cj Ambient"="blue","C15cj High"="red", "C15n Ambient"="darkblue", "C15n High"="darkred") 

#T1 and TF
ord.Pcomp <- ordinate(Pcomp.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pcomp)

scores.Pcomp.samp.df <- as.data.frame(scores(ord.Pcomp) ) 
scores.Pcomp.samp.df$Time.Point <- Pcomp.samp.df.T1F$Time.Point
scores.Pcomp.samp.df$Parent.ID <- Pcomp.samp.df.T1F$Parent.ID
scores.Pcomp.samp.df$Treatment <- Pcomp.samp.df.T1F$Treatment
scores.Pcomp.samp.df$Clade <- Pcomp.samp.df.T1F$Clade
scores.Pcomp.samp.df$Code <- paste(Pcomp.samp.df.T1F$Parent.ID, Pcomp.samp.df.T1F$Time.Point)


ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P. comp Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_T1TF_its2.pdf")

ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PcompT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pcomp.bac.TF, ordinate(Pcomp.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PcompT1 <- ordinate(Pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompT1)
scores.PcompT1 <- as.data.frame(scores(ord.PcompT1))
scores.PcompT1$Treatment <- Pcomp.samp.df.T1$Treatment
scores.PcompT1$Parent.ID <- Pcomp.samp.df.T1$Parent.ID
scores.PcompT1$Clade <- Pcomp.samp.df.T1$Clade
scores.PcompT1$CladeTreat <- as.factor(paste(Pcomp.samp.df.T1$Clade,Pcomp.samp.df.T1$Treatment))


ggplot(scores.PcompT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. comp T1 Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PcompTF <- ordinate(Pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompTF)
scores.PcompTF <- as.data.frame(scores(ord.PcompTF))
scores.PcompTF$Treatment <- Pcomp.samp.df.TF$Treatment
scores.PcompTF$Parent.ID <- Pcomp.samp.df.TF$Parent.ID
scores.PcompTF$Clade <- Pcomp.samp.df.TF$Clade
scores.PcompTF$CladeTreat <- as.factor(paste(Pcomp.samp.df.TF$Clade,Pcomp.samp.df.TF$Treatment))

ggplot(scores.PcompTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. comp TF Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_TF_its_nmds.pdf")



 Pcomp.bac.T1F
Pcomp.bac.T1
Pcomp.bac.TF. 

#### BY CLADE
#ords
#C T1 and TF
set.seed(30)
ord.pcomp_C15cj  <- ordinate(Pcomp.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj)
scores.pcomp_C15cj <- as.data.frame(scores(ord.pcomp_C15cj))
scores.pcomp_C15cj$Treatment <- Pcomp.samp.df.T1F$Treatment
scores.pcomp_C15cj$Parent.ID <- Pcomp.samp.df.T1F$Parent.ID
scores.pcomp_C15cj$Time.Point <- Pcomp.samp.df.T1F$Time.Point

ggplot(scores.pcomp_C15cj, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P comp C15cj") +
  theme_classic()
#ggsave("pcompTFits_C15cj.pdf")


#C T1
set.seed(30)
ord.pcomp_C15cj.T1 <- ordinate(Pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj.T1)
scores.pcomp_C15cj.T1 <- as.data.frame(scores(ord.pcomp_C15cj.T1))
scores.pcomp_C15cj.T1$Treatment <- pcomp.samp.df.T1$Treatment
scores.pcomp_C15cj.T1$Parent.ID <- pcomp.samp.df.T1$Parent.ID
scores.pcomp_C15cj.T1$Time.Point <- pcomp.samp.df.T1$Time.Point

ggplot(scores.pcomp_C15cj.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp T1 C15cj") +
  theme_classic()
#ggsave("pcomp_T1_its_C15cj.pdf")

#C TF
set.seed(30)
ord.pcomp_C15cj.TF <- ordinate(pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj.TF)
scores.pcomp_C15cj.TF <- as.data.frame(scores(ord.pcomp_C15cj.TF))
scores.pcomp_C15cj.TF$Treatment <- pcomp.samp.df.TF$Treatment
scores.pcomp_C15cj.TF$Parent.ID <- pcomp.samp.df.TF$Parent.ID
scores.pcomp_C15cj.TF$Time.Point <- pcomp.samp.df.TF$Time.Point

ggplot(scores.pcomp_C15cj.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("pcomp_TF_its_TF_C15cj.pdf")


#C15n
set.seed(30)
ord.pcomp_C15n <- ordinate(pcomp_DIV_C15n, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n)
scores.pcomp_C15n <- as.data.frame(scores(ord.pcomp_C15n))
scores.pcomp_C15n$Treatment <- pcomp.samp.df.T1F.CD$Treatment
scores.pcomp_C15n$Parent.ID <- pcomp.samp.df.T1F.CD$Parent.ID
scores.pcomp_C15n$Time.Point <- pcomp.samp.df.T1F.CD$Time.Point

ggplot(scores.pcomp_C15n, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_its_T1TF.pdf")

#C15n T1
set.seed(30)
ord.pcomp_C15n.T1 <- ordinate(pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n.T1)
scores.pcomp_C15n.T1 <- as.data.frame(scores(ord.pcomp_C15n.T1))
scores.pcomp_C15n.T1$Treatment <- pcomp.samp.df.T1$Treatment
scores.pcomp_C15n.T1$Parent.ID <- pcomp.samp.df.T1$Parent.ID
scores.pcomp_C15n.T1$Time.Point <- pcomp.samp.df.T1$Time.Point

ggplot(scores.pcomp_C15n.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P comp T1 C15n") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_T1.pdf")

#CD TF
set.seed(30)
ord.pcomp_C15n.TF <- ordinate(pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n.TF)
scores.pcomp_C15n.TF <- as.data.frame(scores(ord.pcomp_C15n.TF))
scores.pcomp_C15n.TF$Treatment <- pcomp.samp.df.TF$Treatment
scores.pcomp_C15n.TF$Parent.ID <- pcomp.samp.df.TF$Parent.ID
scores.pcomp_C15n.TF$Time.Point <- pcomp.samp.df.TF$Time.Point

ggplot(scores.pcomp_C15n.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp C15n TF") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_TF.pdf")


```
Pavona - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pvar
Pvar.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pavona")
Pvar.bac.T1 <- subset_samples(Pvar.bac.T1F, Time.Point == "T1")
Pvar.bac.TF <- subset_samples(Pvar.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F <- phyloseq::distance(Pvar.bac.T1F, method = "bray")
  Pvar.samp.df.T1F <- as(sample_data(Pvar.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pvar.T1F ~ Clade*Treatment*Time.Point, data = Pvar.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PvarT1 <- phyloseq::distance(Pvar.bac.T1, method = "bray")
  Pvar.samp.df.T1 <- as(sample_data(Pvar.bac.T1), "data.frame") #sample df
  adonis(bc.PvarT1 ~ Clade*Treatment, data = Pvar.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Clade)
  permutest(beta.Pvar.T1, pairwise = T) 
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Treatment)
  permutest(beta.Pvar.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pvar.TF <- phyloseq::distance(Pvar.bac.TF, method = "bray")
  Pvar.samp.df.TF <- as(sample_data(Pvar.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF ~ Clade*Treatment, data = Pvar.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Clade)
  permutest(beta.Pvar.TF, pairwise = T) 
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Treatment)
  permutest(beta.Pvar.TF, pairwise = T) 


## by clade: C1 vs C27
  
# C1 only, new objs
Pvar.bacT1F.C1 <- subset_samples(Pvar.bac.T1F, Clade == "C1")
Pvar.bac.T1.C1 <- subset_samples(Pvar.bac.T1, Clade == "C1")
Pvar.bac.TF.C1 <- subset_samples(Pvar.bac.TF, Clade == "C1")
  
#T1TF.C1
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F.C1 <- phyloseq::distance(Pvar.bacT1F.C1, method = "bray")
  Pvar.samp.df.T1F.C1 <- as(sample_data(Pvar.bacT1F.C1), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.T1F.C1 ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C1)

#T1.C1
#make bray curtis data matrix 
  bc.PvarT1.C1 <- phyloseq::distance(Pvar.bac.T1.C1, method = "bray")
  Pvar.samp.df.T1.C1 <- as(sample_data(Pvar.bac.T1.C1), "data.frame") #sample df
#adonis test
  adonis(bc.PvarT1.C1 ~ Treatment, data = Pvar.samp.df.T1.C1)
# Homogeneity of dispersion test  
  beta.Pvar.T1.C1 <- betadisper(bc.PvarT1.C1, Pvar.samp.df.T1.C1$Treatment)
  permutest(beta.Pvar.T1.C1, pairwise = T) 

#TF.C1
#make bray curtis data matrix 
  bc.Pvar.TF.C1 <- phyloseq::distance(Pvar.bac.TF.C1, method = "bray")
  Pvar.samp.df.TF.C1 <- as(sample_data(Pvar.bac.TF.C1), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF.C1 ~ Treatment, data = Pvar.samp.df.TF.C1)
# Homogeneity of dispersion test  
  beta.Pvar.TF.C1 <- betadisper(bc.Pvar.TF.C1, Pvar.samp.df.TF.C1$Treatment)
  permutest(beta.Pvar.TF.C1, pairwise = T) 


# C27 only, new objs
Pvar.bacT1F.C27 <- subset_samples(Pvar.bac.T1F, Clade == "C27")
Pvar.bac.T1.C27 <- subset_samples(Pvar.bac.T1, Clade == "C27")
Pvar.bac.TF.C27 <- subset_samples(Pvar.bac.TF, Clade == "C27")
  
#T1TF.C27
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F.C27 <- phyloseq::distance(Pvar.bacT1F.C27, method = "bray")
  Pvar.samp.df.T1F.C27 <- as(sample_data(Pvar.bacT1F.C27), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.T1F.C27 ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C27)

#T1.C27
#make bray curtis data matrix 
  bc.PvarT1.C27 <- phyloseq::distance(Pvar.bac.T1.C27, method = "bray")
  Pvar.samp.df.T1.C27 <- as(sample_data(Pvar.bac.T1.C27), "data.frame") #sample df
  adonis(bc.PvarT1.C27 ~ Treatment, data = Pvar.samp.df.T1.C27)#adonis test
# Homogeneity of dispersion test  
  beta.Pvar.T1.C27 <- betadisper(bc.PvarT1.C27, Pvar.samp.df.T1.C27$Treatment)
  permutest(beta.Pvar.T1.C27, pairwise = T) 

#TF.C27
#make bray curtis data matrix 
  bc.Pvar.TF.C27 <- phyloseq::distance(Pvar.bac.TF.C27, method = "bray")
  Pvar.samp.df.TF.C27 <- as(sample_data(Pvar.bac.TF.C27), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF.C27 ~ Treatment, data = Pvar.samp.df.TF.C27)
# Homogeneity of dispersion test  
  beta.Pvar.TF.C27 <- betadisper(bc.Pvar.TF.C27, Pvar.samp.df.TF.C27$Treatment)
  permutest(beta.Pvar.TF.C27, pairwise = T) 
``` 
Pvar NMDS
```{r}
#dfs
# C1 only, new objs
# Pvar.bacT1F.C1 <- subset_samples(Pvar.bac.T1F, Clade == "C1")
# Pvar.bac.T1.C1 <- subset_samples(Pvar.bac.T1, Clade == "C1")
# Pvar.bac.TF.C1 <- subset_samples(Pvar.bac.TF, Clade == "C1")


library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pvar.Parent.colors <- c("325" = "green4",  "351" = "firebrick3", "353" ="deeppink4", "354" = "orange", "355" = "green1","356" = "gold",  "357"= "hotpink", "358" ="darkslateblue", "359" = "honeydew4", "360" = "khaki4","361" ="ivory4", "362" = "lightblue", Ambient = "Blue", High = "Red","C1 Ambient"="blue","C1 High"="red", "C27 Ambient"="darkblue", "C27 High"="darkred") 

#T1 and TF
ord.Pvar <- ordinate(Pvar.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pvar)

scores.Pvar.samp.df <- as.data.frame(scores(ord.Pvar) ) 
scores.Pvar.samp.df$Time.Point <- Pvar.samp.df.T1F$Time.Point
scores.Pvar.samp.df$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar.samp.df$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar.samp.df$Clade <- Pvar.samp.df.T1F$Clade
scores.Pvar.samp.df$Code <- paste(Pvar.samp.df.T1F$Parent.ID, Pvar.samp.df.T1F$Time.Point)


ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P.var Bray Curtis") +
  theme_classic()
#ggsave("Pvar_T1TF_its2.pdf")

ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var Bray Curtis") +
  theme_classic()
#ggsave("PvarT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pvar.bac.TF, ordinate(Pvar.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PvarT1 <- ordinate(Pvar.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PvarT1)
scores.PvarT1 <- as.data.frame(scores(ord.PvarT1))
scores.PvarT1$Treatment <- Pvar.samp.df.T1$Treatment
scores.PvarT1$Parent.ID <- Pvar.samp.df.T1$Parent.ID
scores.PvarT1$Clade <- Pvar.samp.df.T1$Clade
scores.PvarT1$CladeTreat <- as.factor(paste(Pvar.samp.df.T1$Clade,Pvar.samp.df.T1$Treatment))


ggplot(scores.PvarT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var T1 Bray Curtis") +
  theme_classic()
#ggsave("Pvar_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PvarTF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PvarTF)
scores.PvarTF <- as.data.frame(scores(ord.PvarTF))
scores.PvarTF$Treatment <- Pvar.samp.df.TF$Treatment
scores.PvarTF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.PvarTF$Clade <- Pvar.samp.df.TF$Clade
scores.PvarTF$CladeTreat <- as.factor(paste(Pvar.samp.df.TF$Clade,Pvar.samp.df.TF$Treatment))

ggplot(scores.PvarTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. var TF Bray Curtis") +
  theme_classic()
#gsave("Pvar_TF_its_nmds.pdf")



# C1 only, new objs
#Pvar.bacT1F.C1 , Pvar.samp.df.T1F.C1
#Pvar.bac.T1.C1 , Pvar.samp.df.T1.C1
#Pvar.bac.TF.C1 ,Pvar.samp.df.TF.C1

#### BY CLADE

#CCC T1 and TF
set.seed(30)
ord.Pvar_CCC  <- ordinate(Pvar.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC)
scores.Pvar_CCC <- as.data.frame(scores(ord.Pvar_CCC))
scores.Pvar_CCC$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar_CCC$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar_CCC$Time.Point <- Pvar.samp.df.T1F$Time.Point

ggplot(scores.Pvar_CCC, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var CCC") +
  theme_classic()
#ggsave("PvarTFits_CCC.pdf")


#C T1
set.seed(30)
ord.Pvar_CCC.T1 <- ordinate(Pvar.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC.T1)
scores.Pvar_CCC.T1 <- as.data.frame(scores(ord.Pvar_CCC.T1))
scores.Pvar_CCC.T1$Treatment <- Pvar.samp.df.T1$Treatment
scores.Pvar_CCC.T1$Parent.ID <- Pvar.samp.df.T1$Parent.ID
scores.Pvar_CCC.T1$Time.Point <- Pvar.samp.df.T1$Time.Point

ggplot(scores.Pvar_CCC.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var T1 CCC") +
  theme_classic()
#ggsave("Pvar_T1_its_CCC.pdf")

#C TF
set.seed(30)
ord.Pvar_CCC.TF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC.TF)
scores.Pvar_CCC.TF <- as.data.frame(scores(ord.Pvar_CCC.TF))
scores.Pvar_CCC.TF$Treatment <- Pvar.samp.df.TF$Treatment
scores.Pvar_CCC.TF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.Pvar_CCC.TF$Time.Point <- Pvar.samp.df.TF$Time.Point

ggplot(scores.Pvar_CCC.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var CCC TF") +
  theme_classic()
#ggsave("Pvar_TF_its_TF_CCC.pdf")





#C27
set.seed(30)
ord.Pvar_C27 <- ordinate(Pvar.bacT1F.C27, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27)
scores.Pvar_C27 <- as.data.frame(scores(ord.Pvar_C27))
scores.Pvar_C27$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar_C27$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar_C27$Time.Point <- Pvar.samp.df.T1F$Time.Point

ggplot(scores.Pvar_C27, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C&D") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_its_T1TF.pdf")

#C27 T1
set.seed(30)
ord.Pvar_C27.T1 <- ordinate(Pvar.bac.T1.C27 , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27.T1)
scores.Pvar_C27.T1 <- as.data.frame(scores(ord.Pvar_C27.T1))
scores.Pvar_C27.T1$Treatment <- Pvar.samp.df.T1.C27$Treatment
scores.Pvar_C27.T1$Parent.ID <- Pvar.samp.df.T1.C27$Parent.ID
scores.Pvar_C27.T1$Time.Point <- Pvar.samp.df.T1.C27$Time.Point

ggplot(scores.Pvar_C27.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var T1 C27") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_T1.pdf")

#CD TF
set.seed(30)
ord.Pvar_C27.TF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27.TF)
scores.Pvar_C27.TF <- as.data.frame(scores(ord.Pvar_C27.TF))
scores.Pvar_C27.TF$Treatment <- Pvar.samp.df.TF$Treatment
scores.Pvar_C27.TF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.Pvar_C27.TF$Time.Point <- Pvar.samp.df.TF$Time.Point

ggplot(scores.Pvar_C27.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C27 TF") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_TF.pdf")



```

Pocillopora - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pacu
Pacu.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pocillopora")
Pacu.bac.T1 <- subset_samples(Pacu.bac.T1F, Time.Point == "T1")
Pacu.bac.TF <- subset_samples(Pacu.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F <- phyloseq::distance(Pacu.bac.T1F, method = "bray")
  Pacu.samp.df.T1F <- as(sample_data(Pacu.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pacu.T1F ~ Treatment*Time.Point, data = Pacu.samp.df.T1F)      #adonis test
#write.csv(Pacu.samp.df.T1F, "Pacu_its2_samp_df.csv")
#T1
#make bray curtis data matrix 
  bc.PacuT1 <- phyloseq::distance(Pacu.bac.T1, method = "bray")
  Pacu.samp.df.T1 <- as(sample_data(Pacu.bac.T1), "data.frame") #sample df
  adonis(bc.PacuT1 ~ Clade*Treatment, data = Pacu.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Clade)
  permutest(beta.Pacu.T1, pairwise = T) 
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Treatment)
  permutest(beta.Pacu.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pacu.TF <- phyloseq::distance(Pacu.bac.TF, method = "bray")
  Pacu.samp.df.TF <- as(sample_data(Pacu.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF ~ Clade*Treatment, data = Pacu.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Clade)
  permutest(beta.Pacu.TF, pairwise = T) 
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Treatment)
  permutest(beta.Pacu.TF, pairwise = T) 


## by clade: CCC vs C42
  
# C only, new objs
Pacu.bacT1F.CCC <- subset_samples(Pacu.bac.T1F, Clade == "CCC")
Pacu.bac.T1.CCC <- subset_samples(Pacu.bac.T1, Clade == "CCC")
Pacu.bac.TF.CCC <- subset_samples(Pacu.bac.TF, Clade == "CCC")
  
#T1TF.CCC
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F.CCC <- phyloseq::distance(Pacu.bacT1F.CCC, method = "bray")
  Pacu.samp.df.T1F.CCC <- as(sample_data(Pacu.bacT1F.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.T1F.CCC ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.CCC)

#T1.CCC
#make bray curtis data matrix 
  bc.PacuT1.CCC <- phyloseq::distance(Pacu.bac.T1.CCC, method = "bray")
  Pacu.samp.df.T1.CCC <- as(sample_data(Pacu.bac.T1.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.PacuT1.CCC ~ Treatment, data = Pacu.samp.df.T1.CCC)
# Homogeneity of dispersion test  
  beta.Pacu.T1.CCC <- betadisper(bc.PacuT1.CCC, Pacu.samp.df.T1.CCC$Treatment)
  permutest(beta.Pacu.T1.CCC, pairwise = T) 

#TF.CCC
#make bray curtis data matrix 
  bc.Pacu.TF.CCC <- phyloseq::distance(Pacu.bac.TF.CCC, method = "bray")
  Pacu.samp.df.TF.CCC <- as(sample_data(Pacu.bac.TF.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF.CCC ~ Treatment, data = Pacu.samp.df.TF.CCC)
# Homogeneity of dispersion test  
  beta.Pacu.TF.CCC <- betadisper(bc.Pacu.TF.CCC, Pacu.samp.df.TF.CCC$Treatment)
  permutest(beta.Pacu.TF.CCC, pairwise = T) 


# C42 only, new objs
Pacu.bacT1F.C42 <- subset_samples(Pacu.bac.T1F, Clade == "C42")
Pacu.bac.T1.C42 <- subset_samples(Pacu.bac.T1, Clade == "C42")
Pacu.bac.TF.C42 <- subset_samples(Pacu.bac.TF, Clade == "C42")
  
#T1TF.C42
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F.C42 <- phyloseq::distance(Pacu.bacT1F.C42, method = "bray")
  Pacu.samp.df.T1F.C42 <- as(sample_data(Pacu.bacT1F.C42), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.T1F.C42 ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.C42)

#T1.C42
#make bray curtis data matrix 
  bc.PacuT1.C42 <- phyloseq::distance(Pacu.bac.T1.C42, method = "bray")
  Pacu.samp.df.T1.C42 <- as(sample_data(Pacu.bac.T1.C42), "data.frame") #sample df
  adonis(bc.PacuT1.C42 ~ Treatment, data = Pacu.samp.df.T1.C42)#adonis test
# Homogeneity of dispersion test  
  beta.Pacu.T1.C42 <- betadisper(bc.PacuT1.C42, Pacu.samp.df.T1.C42$Treatment)
  permutest(beta.Pacu.T1.C42, pairwise = T) 

#TF.C42
#make bray curtis data matrix 
  bc.Pacu.TF.C42 <- phyloseq::distance(Pacu.bac.TF.C42, method = "bray")
  Pacu.samp.df.TF.C42 <- as(sample_data(Pacu.bac.TF.C42), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF.C42 ~ Treatment, data = Pacu.samp.df.TF.C42)
# Homogeneity of dispersion test  
  beta.Pacu.TF.C42 <- betadisper(bc.Pacu.TF.C42, Pacu.samp.df.TF.C42$Treatment)
  permutest(beta.Pacu.TF.C42, pairwise = T) 
``` 
Pacu NMDS
```{r}
# C only, new objs
Pacu.bacT1F.CCC <- subset_samples(Pacu.bac.T1F, Clade == "CCC")
Pacu.bac.T1.CCC <- subset_samples(Pacu.bac.T1, Clade == "CCC")
Pacu.bac.TF.CCC <- subset_samples(Pacu.bac.TF, Clade == "CCC")
# # pull df sections for craig Pacu.bac.T1F
# OTU_Pacu<-out_table(Pacu.bac.T1F)
# tax  = tax_table(coralDIV2_rare2)
# otu  = otu_table(coralDIV2_rare2)
# 
# OTU1 = as(otu_table(Pacu.bac.T1F), "matrix")
# # transpose if necessary
# if(taxa_are_rows(Pacu.bac.T1F)){OTU1 <- t(OTU1)}
# # Coerce to data.frame
# OTUdf = as.data.frame(OTU1)
# write.csv(OTUdf, "Pacuta_its2_T1TF.csv")
# 


library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pacu.Parent.colors <- c("305" = "green4",  "306" = "firebrick3", "307" ="deeppink4", "308" = "orange", "309" = "green1","310" = "gold",  "311"= "hotpink", "312" ="darkslateblue", "302" = "honeydew4", "303" = "khaki4","304" ="ivory4", "305" = "lightblue", Ambient = "Blue", High = "Red","CCC Ambient"="blue","CCC High"="red", "C42 Ambient"="darkblue", "C42 High"="darkred") 





#T1 and TF
ord.Pacu <- ordinate(Pacu.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pacu)

scores.Pacu.samp.df <- as.data.frame(scores(ord.Pacu) ) 
scores.Pacu.samp.df$Time.Point <- Pacu.samp.df.T1F$Time.Point
scores.Pacu.samp.df$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu.samp.df$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu.samp.df$Clade <- Pacu.samp.df.T1F$Clade
scores.Pacu.samp.df$Code <- paste(Pacu.samp.df.T1F$Parent.ID, Pacu.samp.df.T1F$Time.Point)


ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P.acu Bray Curtis") +
  theme_classic()
#ggsave("Pacu_T1TF_its2.pdf")

ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var Bray Curtis") +
  theme_classic()
#ggsave("PacuT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pacu.bac.TF, ordinate(Pacu.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PacuT1 <- ordinate(Pacu.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuT1)
scores.PacuT1 <- as.data.frame(scores(ord.PacuT1))
scores.PacuT1$Treatment <- Pacu.samp.df.T1$Treatment
scores.PacuT1$Parent.ID <- Pacu.samp.df.T1$Parent.ID
scores.PacuT1$Clade <- Pacu.samp.df.T1$Clade
scores.PacuT1$CladeTreat <- as.factor(paste(Pacu.samp.df.T1$Clade,Pacu.samp.df.T1$Treatment))


ggplot(scores.PacuT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var T1 Bray Curtis") +
  theme_classic()
#ggsave("Pacu_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PacuTF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuTF)
scores.PacuTF <- as.data.frame(scores(ord.PacuTF))
scores.PacuTF$Treatment <- Pacu.samp.df.TF$Treatment
scores.PacuTF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.PacuTF$Clade <- Pacu.samp.df.TF$Clade
scores.PacuTF$CladeTreat <- as.factor(paste(Pacu.samp.df.TF$Clade,Pacu.samp.df.TF$Treatment))

ggplot(scores.PacuTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. var TF Bray Curtis") +
  theme_classic()
#gsave("Pacu_TF_its_nmds.pdf")










#### BY CLADE

#CCC T1 and TF
set.seed(30)
ord.Pacu_CCC  <- ordinate(Pacu.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC)
scores.Pacu_CCC <- as.data.frame(scores(ord.Pacu_CCC))
scores.Pacu_CCC$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu_CCC$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu_CCC$Time.Point <- Pacu.samp.df.T1F$Time.Point

ggplot(scores.Pacu_CCC, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var CCC") +
  theme_classic()
#ggsave("PacuTFits_CCC.pdf")


#C T1
set.seed(30)
ord.Pacu_CCC.T1 <- ordinate(Pacu.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.T1)
scores.Pacu_CCC.T1 <- as.data.frame(scores(ord.Pacu_CCC.T1))
scores.Pacu_CCC.T1$Treatment <- Pacu.samp.df.T1$Treatment
scores.Pacu_CCC.T1$Parent.ID <- Pacu.samp.df.T1$Parent.ID
scores.Pacu_CCC.T1$Time.Point <- Pacu.samp.df.T1$Time.Point

ggplot(scores.Pacu_CCC.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var T1 CCC") +
  theme_classic()
#ggsave("Pacu_T1_its_CCC.pdf")

#C TF
set.seed(30)
ord.Pacu_CCC.TF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.TF)
scores.Pacu_CCC.TF <- as.data.frame(scores(ord.Pacu_CCC.TF))
scores.Pacu_CCC.TF$Treatment <- Pacu.samp.df.TF$Treatment
scores.Pacu_CCC.TF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.Pacu_CCC.TF$Time.Point <- Pacu.samp.df.TF$Time.Point

ggplot(scores.Pacu_CCC.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var CCC TF") +
  theme_classic()
#ggsave("Pacu_TF_its_TF_CCC.pdf")



Pacu.bacT1F.C27  
Pacu.bac.T1.C27 
Pacu.bac.TF.C27 

#C27
set.seed(30)
ord.Pacu_C27 <- ordinate(Pacu.bacT1F.C27, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27)
scores.Pacu_C27 <- as.data.frame(scores(ord.Pacu_C27))
scores.Pacu_C27$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu_C27$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu_C27$Time.Point <- Pacu.samp.df.T1F$Time.Point

ggplot(scores.Pacu_C27, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C&D") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_its_T1TF.pdf")

#C27 T1
set.seed(30)
ord.Pacu_C27.T1 <- ordinate(Pacu.bac.T1.C27 , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27.T1)
scores.Pacu_C27.T1 <- as.data.frame(scores(ord.Pacu_C27.T1))
scores.Pacu_C27.T1$Treatment <- Pacu.samp.df.T1.C27$Treatment
scores.Pacu_C27.T1$Parent.ID <- Pacu.samp.df.T1.C27$Parent.ID
scores.Pacu_C27.T1$Time.Point <- Pacu.samp.df.T1.C27$Time.Point

ggplot(scores.Pacu_C27.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var T1 C27") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_T1.pdf")

#CD TF
set.seed(30)
ord.Pacu_C27.TF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27.TF)
scores.Pacu_C27.TF <- as.data.frame(scores(ord.Pacu_C27.TF))
scores.Pacu_C27.TF$Treatment <- Pacu.samp.df.TF$Treatment
scores.Pacu_C27.TF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.Pacu_C27.TF$Time.Point <- Pacu.samp.df.TF$Time.Point

ggplot(scores.Pacu_C27.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C27 TF") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_TF.pdf")





```









































