---
title: "16S Thermal Tolerance"
html_document:
code_folding: hide
toc:yes
toc_depth:3
toc_float:yes
output: 
chunk_output_type: console
editor_options: 
  chunk_output_type: console
---
Last updated: 27 may 2021


16s data analysis
#Data and libraries
```{r}  
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr) 
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(tidyverse)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(microbiome)
library(vegan)

library(geosphere)
library(ade4)
```


#Load in data and get into phyloseq
```{r}       
 
#load in 3000 rare instead
#read in sample data
MetaData16<-read.csv("Data/Meta16s_3k.csv") #run in physio script first
MetaData16$Parent.ID<-as.factor(as.character(MetaData16$Parent.ID))
MetaData16$Tank.num<-as.factor(as.character(MetaData16$Tank.num))

sam0 <- MetaData16
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))

#load in OUT:raw_abundanceTable_100.shared

OTU3k<-read.table("Data/TT17_3000_summer_16S-pipeline_outputs/abundance_table_100_shared.csv", sep=',', header=T)

#from its  
#otu
otu3k1 <- as.matrix(OTU3k[, -1])
rownames(otu3k1) <- OTU3k$sample_name
otu <- otu_table(otu3k1, taxa_are_rows = FALSE)


#tax table annotations_100.taxonomy.csv
TAX<- read.csv("Data/TT17_3000_summer_16S-pipeline_outputs/annotations_100.taxonomy.csv", colClasses = "character")
tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- TAX$OTU
tax <- tax_table(tax1)


# Read the data into phyloseq
Bac.seq = phyloseq(otu, tax,sam) #THIS WORKS
Bac.seq
Bac.seq.df <- sample_data(Bac.seq)


#load your .tre otu_repr_100.tre
library(ape)
treefile<- read.tree("Data/TT17_3000_summer_16S-pipeline_outputs/Results/postprocessing/unifrac/otu_repr_100.tre")

phy_tree(Bac.seq) <- treefile
Bac.seq

#save(Bac.seq, file = "Data/Bac.seq_phyloseq.RData")



########################################################
# ## THIS IS FOR INCLUDING ALL SAMPLES YOU NEED FOR phylotype analysis.
# #read in sample data
# MetaData16<-read.csv("Data/Meta_16s_new.csv") #run in physio script first
# MetaData16$Parent.ID<-as.factor(as.character(MetaData16$Parent.ID))
# MetaData16$Tank.num<-as.factor(as.character(MetaData16$Tank.num))
# 
# sam0 <- MetaData16
# sam1 <- as.matrix(sam0[, -1])
# rownames(sam1) <- sam0$sample_name
# sam <- sample_data(data.frame(sam1))
# 
# 
# #load in OUT:raw_abundanceTable_100.shared
# OTU2<-read.table("Data/TT17_3000_summer_16S-pipeline_outputs/raw_abundanceTable_100_shared.csv", sep=',', header=T)
# 
# #from its3
# #otu
# otu5 <- as.matrix(OTU2[, -1])
# rownames(otu5) <- OTU2$sample_name
# otu <- otu_table(otu5, taxa_are_rows = FALSE)
# 
# 
# #tax
# TAX<- read.csv("Data/TT17_3000_summer_16S-pipeline_outputs/raw_consensusClassification_100_taxonomy.csv", colClasses = "character")
# tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
# rownames(tax1) <- TAX$OTU
# tax <- tax_table(tax1)
# 
# # Read the data into phyloseq
# Bac.seqAll = phyloseq(otu, tax,sam) #THIS WORKS
# Bac.seqAll
# 
# save(Bac.seqAll, file = "Data/Bac.seqAll_phyloseq.RData")
 
```  
Look at the data, Bac.seq for 3k
```{r}       
## check out data
ntaxa(Bac.seq)  #num taxa
nsamples(Bac.seq)   #num samples
sample_names(Bac.seq)[1:300] #samp names
 #sampsy<-sample_names(physeq)[1:300]
#write.csv(sampsy,"sampsy.csv")

# OK Now you need to do the steps in the pipeline. ONLY for >3K dfs
#1 filter out taxa: get rid of unknown, mitochondria and chloroplast
# Bac.seq<-subset_taxa(Bac.seq, Family !="Mitochondria")
# Bac.seq<-subset_taxa(Bac.seq, Order !="Chloroplast")
# Bac.seq<-subset_taxa(Bac.seq, Domain !="unknown")
# Bac.seq<-subset_taxa(Bac.seq, Domain !="Archaea")

# now get rid of things not represnted 2x or more
#Prune singletons (these are reads that are only found once)
# Bac.seq.prune <- prune_taxa(taxa_sums(Bac.seq) > 2, Bac.seq)
# 
#   save(Bac.seq, file = "Data/Bac.seq.prune.RData")
  
```

#16S
Load in your data
```{r}      
# Load data (see data_exploration.Rmd)
#load("Data/Bac.seq_phyloseq.RData") #use RAW data that has been pruned. 
# the df is Bac.seq
rank_names(Bac.seq) #ranks are DIV and clade




# #Bac.seq  not 3k use to check who you lose
# # create df of sample data to view 
# sample.data <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view
# sample.data$LibrarySize <- sample_sums(Bac.seq)
# #sample.data <- sample.data[order(sample.data$LibrarySize),]
# #sample.data$Index <- seq(nrow(sample.data))  # check 3k rare
# ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = Species)) +
#   geom_point()
# write.csv(sample.data, "sampscheck.csv")

## check out data
ntaxa(Bac.seq)  #num taxa
nsamples(Bac.seq)   #num samples
sample_names(Bac.seq)[1:5] #samp names
rank_names(Bac.seq) #ranks are DIV and clade
sample_variables(Bac.seq) # metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.seq)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  # check 3k rare
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = Species)) +
  geom_point()



#remove duplicates 
Dups<-subset_samples(Bac.seq, Dups=="Yes")

TopNOTUsB = names(sort(taxa_sums(Dups), TRUE)[1:100])
bac50 = prune_taxa(TopNOTUsB, Dups)
bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot
#ggsave("mMcapBarPlot_DIVrelARare.pdf", McapBarPlot, dpi=300, width=8)
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S1753b" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S208" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3476" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "SN35b" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3149" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S3149" )
Bac.seq <- subset_samples(Bac.seq, sample_name.1 != "S1240" )
#Bac.s <- subset_samples(Bac.s, sample_name.1 != "Field" )
Bac.seq <- subset_samples(Bac.seq, Type != "extra" )
Bac.seq <- subset_samples(Bac.seq, Type != " " )

Bac.seq.df <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view


Bac.s = subset_samples(Bac.seq, Type== "sample" )
Bac.field = subset_samples(Bac.seq, Type== "field" )
Bac.seq.all<-Bac.seq


#pull out for CN
Bac.seq.all

c.sd = sample_data(Bac.seq.all)
#   # otu_table
#   OTU = otu_table(controlT0_Amb)







# Subsample to 2K <- only for phylotypes
# Bac.s <- prune_samples(sample_sums(Bac.s)>=1764, Bac.s) #keep things over 2k
#Rarefied to 2000 reads
# Bac.r <- rarefy_even_depth(Bac.s, sample.size = 1764, rngseed = 711) 


#### Create a duplicate of T0s for the other A and H treatments for comparision: df coralDIV2_rare
# #not
# controlT0_Amb = subset_samples(Bac.s, Time.Point == "T0"&Treatment == "Ambient")  #copy
# 
# #tax table
# tax  = tax_table(controlT0_Amb)
# 
#   # sample data
#   sd = sample_data(controlT0_Amb)
#   rownames(sd) <- paste0(rownames(sd), "_dup")
#   # otu_table
#   OTU = otu_table(controlT0_Amb)
#   sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
#   # Avoid conflict between factor and character
#   # sd$Treatment <- as.character(sd$Treatment)
#   # sd$ID <- as.character(sd$ID)
#   sd$Treatment <- "High"    #rename HIGH
#   sd$ID <- paste0(sd$ID, "_dup")
#   sd$Treatment <- as.factor(sd$Treatment)
#   # sd$ID <- as.factor(sd$ID)
# 
#   Amb2<-phyloseq(OTU, sd, tax)
# #sample.dataAmb2 <- as(sample_data(Amb2), "data.frame") #create sample data frame to view
# 
# ##High
#   controlT0_High = subset_samples(Bac.s, Time.Point == "T0"&Treatment == "High")  #copy
# 
# #tax table
# tax  = tax_table(controlT0_High)
# 
#   # sample data
#   sd = sample_data(controlT0_High)
#   rownames(sd) <- paste0(rownames(sd), "_dup")
# # otu_table
# OTU = otu_table(controlT0_High)
#   sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
#   # Avoid conflict between factor and character
#   sd$Treatment <- as.character(sd$Treatment)
#   # sd$ID <- as.character(sd$ID)
#   sd$Treatment <- "Ambient"    #rename HIGH
#   sd$ID <- paste0(sd$ID, "_dup")
#   sd$Treatment <- as.factor(sd$Treatment)
#   sd$ID <- as.factor(sd$ID)
# 
#   High2<-phyloseq(OTU, sd, tax)
#   #sample.dataHigh2 <- as(sample_data(High2), "data.frame") #create sample data frame to view
# 
# #Merge control dups H and A  
# Controldups = merge_phyloseq(Amb2,High2)
# Controldups
#     #sample.dataControldups2 <- as(sample_data(Controldups), "data.frame") #create sample data frame to view
# 
# #merg them to the original df:
# 
# Bac.s = merge_phyloseq(Bac.s,Controldups)
# Bac.s
# Bac.s.sd <- as(sample_data(Bac.s), "data.frame") #create sample data frame to view
# Bac.s.sd
#write.csv(Bac.s.sd,"16sMetadata.csv")

#save(Bac.seq, file = "Data/prune.Bac.seq_phyloseq.RData")
save(Bac.seq,data.Bac.s.rel,erich.tab2, file = "Data/16s_phyloseq4HE.RData")
 
# bacseqOTU = otu_table(Bac.seq)
# 
# #write.csv(bacseqOTU,"bacseqOTU.csv")
# 
# 
# 
# 
# bacseqOTU.tab = as(otu_table(bacseqOTU), "matrix")
#        # transpose if necessary
#           # if(taxa_are_rows(mcap.u.f.bac.T1F.top5P)){OTUfam.v <- t(mcap.u.f.bac.T1F.top5P.OTU)}
#           # Coerce to data.frame
#             bacseqOTU.tab.df = as.data.frame(bacseqOTU.tab)
# 
# 
# mcap.u.f.bac.T01F.top10P.sd <- as(sample_data(mcap.u.f.bac.T01F.top10P), "data.frame") #create sample data frame to view
#          mcap.u.f.bac.T01F.top10P.sd<-mcap.u.f.bac.T01F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
#                                                                    "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
#     
#     mcap.16s.fam.T01f.top10<- merge(mcap.u.f.bac.T01F.top10P.OTU.df,mcap.u.f.bac.T01F.top10P.sd, by="row.names") #merge by rownames
#        mcap.16s.fam.T01f.top10.m <- melt(mcap.16s.fam.T01f.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
#                                                                    "Treatment","Time.Point","Tank.num","Clade"))

```
# look at field samples. after you did this you now imported the field amples normaly
```{r}    
MetaDataFIELD<-read.csv("Data/sample.data_FIELD.csv") #run in physio script first
MetaDataFIELD$Parent.ID<-as.factor(as.character(MetaDataFIELD$Parent.ID))
MetaDataFIELD$Tank.num<-as.factor(as.character(MetaDataFIELD$Tank.num))

sam0 <- MetaDataFIELD
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))

#load in OUT:raw_abundanceTable_100.shared

OTU3k<-read.table("Data/TT17_3000_summer_16S-pipeline_outputs/abundance_table_100_shared.csv", sep=',', header=T)

#from its  
#otu
otu3k1 <- as.matrix(OTU3k[, -1])
rownames(otu3k1) <- OTU3k$sample_name
otu <- otu_table(otu3k1, taxa_are_rows = FALSE)


#tax table annotations_100.taxonomy.csv
TAX<- read.csv("Data/TT17_3000_summer_16S-pipeline_outputs/annotations_100.taxonomy.csv", colClasses = "character")
tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- TAX$OTU
tax <- tax_table(tax1)


# Read the data into phyloseq
Bac.seq.field = phyloseq(otu, tax,sam) #THIS WORKS
Bac.seq.field
Bac.seq.field.df <- sample_data(Bac.seq.field)


#load your .tre otu_repr_100.tre
treefile<- read.tree("Data/TT17_3000_summer_16S-pipeline_outputs/Results/postprocessing/unifrac/otu_repr_100.tre")

phy_tree(Bac.seq.field) <- treefile
Bac.seq.field

Bac.seq.field.rel  = transform_sample_counts(Bac.seq.field, function(x) x / sum(x) )  #save as RELA DIVs


 
#Ordination NMDS for all species, T1Tf
unifrac.field.all <- ordinate(Bac.seq.field.rel, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.field.all)
scores.ord.u.field_RelA_stats<-as.data.frame(cbind(vegan::scores(unifrac.field.all, display="sites")))
     unifrac.all.sd <- as(sample_data(Bac.seq.field.rel), "data.frame") #sample df

scores.ord.u.field_RelA_stats$Parent.ID <- unifrac.all.sd$Parent.ID
scores.ord.u.field_RelA_stats$Species <- unifrac.all.sd$Species
scores.ord.u.field_RelA_stats$Clade <- unifrac.all.sd$Clade
scores.ord.u.field_RelA_stats$Time.Point <- unifrac.all.sd$Time.Point
scores.ord.u.field_RelA_stats$Code <- paste(scores.ord.u.field_RelA_stats$Clade, scores.ord.u.field_RelA_stats$Species)



ggplot(scores.ord.u.field_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species, shape=Time.Point), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS 16s bray") +
  theme_classic()
#ggsave("bc_all.pdf")
 
#by speices ####
#mcap
Bac.seq.field.rel.mcap<-subset_samples(Bac.seq.field.rel, Species=="Montipora_capitata")
#Ordination NMDS for all species, T1Tf
unifrac.field.all.mcap <- ordinate(Bac.seq.field.rel.mcap, method="NMDS", "unifrac",weighted=TRUE, set.seed(50)) 
stressplot(unifrac.field.all.mcap)
scores.ord.u.field_RelA_stats.mcap<-as.data.frame(cbind(vegan::scores(unifrac.field.all.mcap, display="sites")))
     unifrac.all.mcap.sd <- as(sample_data(Bac.seq.field.rel.mcap), "data.frame") #sample df

scores.ord.u.field_RelA_stats.mcap$Parent.ID <- unifrac.all.mcap.sd$Parent.ID
scores.ord.u.field_RelA_stats.mcap$Species <- unifrac.all.mcap.sd$Species
scores.ord.u.field_RelA_stats.mcap$Clade <- unifrac.all.mcap.sd$Clade
scores.ord.u.field_RelA_stats.mcap$Time.Point <- unifrac.all.mcap.sd$Time.Point
scores.ord.u.field_RelA_stats.mcap$Code <- paste(scores.ord.u.field_RelA_stats.mcap$Clade, scores.ord.u.field_RelA_stats.mcap$Species)

ggplot(scores.ord.u.field_RelA_stats.mcap, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape=Time.Point), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("Mcap Field NMDS 16s bray") +
  theme_classic()

#pcomp
Bac.seq.field.rel.pcomp<-subset_samples(Bac.seq.field.rel, Species=="Porites_compressa")
#Ordination NMDS for all species, T1Tf
unifrac.field.all.pcomp <- ordinate(Bac.seq.field.rel.pcomp, method="NMDS", "unifrac",weighted=TRUE, set.seed(50)) 
stressplot(unifrac.field.all.pcomp)
scores.ord.u.field_RelA_stats.pcomp<-as.data.frame(cbind(vegan::scores(unifrac.field.all.pcomp, display="sites")))
     unifrac.all.pcomp.sd <- as(sample_data(Bac.seq.field.rel.pcomp), "data.frame") #sample df

scores.ord.u.field_RelA_stats.pcomp$Parent.ID <- unifrac.all.pcomp.sd$Parent.ID
scores.ord.u.field_RelA_stats.pcomp$Species <- unifrac.all.pcomp.sd$Species
scores.ord.u.field_RelA_stats.pcomp$Clade <- unifrac.all.pcomp.sd$Clade
scores.ord.u.field_RelA_stats.pcomp$Time.Point <- unifrac.all.pcomp.sd$Time.Point
scores.ord.u.field_RelA_stats.pcomp$Code <- paste(scores.ord.u.field_RelA_stats.pcomp$Clade, scores.ord.u.field_RelA_stats.pcomp$Species)

ggplot(scores.ord.u.field_RelA_stats.pcomp, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape=Time.Point), size = 4) +
  #scale_color_manual(values=pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("pcomp Field NMDS 16s bray") +
  theme_classic()

#pvar
Bac.seq.field.rel.pvar<-subset_samples(Bac.seq.field.rel, Species=="Pavona_varians")
#Ordination NMDS for all species, T1Tf
unifrac.field.all.pvar <- ordinate(Bac.seq.field.rel.pvar, method="NMDS", "unifrac",weighted=TRUE, set.seed(50)) 
stressplot(unifrac.field.all.pvar)
scores.ord.u.field_RelA_stats.pvar<-as.data.frame(cbind(vegan::scores(unifrac.field.all.pvar, display="sites")))
     unifrac.all.pvar.sd <- as(sample_data(Bac.seq.field.rel.pvar), "data.frame") #sample df

scores.ord.u.field_RelA_stats.pvar$Parent.ID <- unifrac.all.pvar.sd$Parent.ID
scores.ord.u.field_RelA_stats.pvar$Species <- unifrac.all.pvar.sd$Species
scores.ord.u.field_RelA_stats.pvar$Clade <- unifrac.all.pvar.sd$Clade
scores.ord.u.field_RelA_stats.pvar$Time.Point <- unifrac.all.pvar.sd$Time.Point
scores.ord.u.field_RelA_stats.pvar$Code <- paste(scores.ord.u.field_RelA_stats.pvar$Clade, scores.ord.u.field_RelA_stats.pvar$Species)

ggplot(scores.ord.u.field_RelA_stats.pvar, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape=Time.Point), size = 4) +
  #scale_color_manual(values=pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("pvar Field NMDS 16s bray") +
  theme_classic()


#pacu
Bac.seq.field.rel.pacu<-subset_samples(Bac.seq.field.rel, Species=="Porites_compressa")
#Ordination NMDS for all species, T1Tf
unifrac.field.all.pacu <- ordinate(Bac.seq.field.rel.pacu, method="NMDS", "unifrac",weighted=TRUE, set.seed(50)) 
stressplot(unifrac.field.all.pacu)
scores.ord.u.field_RelA_stats.pacu<-as.data.frame(cbind(vegan::scores(unifrac.field.all.pacu, display="sites")))
     unifrac.all.pacu.sd <- as(sample_data(Bac.seq.field.rel.pacu), "data.frame") #sample df

scores.ord.u.field_RelA_stats.pacu$Parent.ID <- unifrac.all.pacu.sd$Parent.ID
scores.ord.u.field_RelA_stats.pacu$Species <- unifrac.all.pacu.sd$Species
scores.ord.u.field_RelA_stats.pacu$Clade <- unifrac.all.pacu.sd$Clade
scores.ord.u.field_RelA_stats.pacu$Time.Point <- unifrac.all.pacu.sd$Time.Point
scores.ord.u.field_RelA_stats.pacu$Code <- paste(scores.ord.u.field_RelA_stats.pacu$Clade, scores.ord.u.field_RelA_stats.pacu$Species)

ggplot(scores.ord.u.field_RelA_stats.pacu, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape=Time.Point), size = 4) +
  #scale_color_manual(values=pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("pacu Field NMDS 16s bray") +
  theme_classic()
```


#Set up, Bac.s rela
```{r}        
#make rel ativ e abundance df Bac.s
Bac.s.rel  = transform_sample_counts(Bac.s, function(x) x / sum(x) )  #save as RELA DIVs
data.Bac.s.rel <- as(sample_data(Bac.s.rel), "data.frame") #look at samp data

Bac.seq.all<- transform_sample_counts(Bac.seq.all, function(x) x / sum(x) )  
#heatmap
  #plot_heatmap(Bac.RelA, method = "NMDS", distance = "bray")# way too much data

### this isn't running bc you created it farther down.... FIX
#t0
Bac.s.rel.u.T0
#make rel ativ e abundance df Bac.s
Bac.s.rel.u.T0.rel <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #look at samp data
Bac.seq.all.T0<- transform_sample_counts(Bac.s.rel.u.T0, function(x) x / sum(x) )  
#heatmap
  plot_heatmap(Bac.seq.all.T0, method = "NMDS", distance = "bray",sample.label="Species")# way too much data


Bac.s.rel = prune_taxa(taxa_sums(Bac.s.rel) > 0, Bac.s.rel) #this removes any OTU with 0s








Bac.s.rel.p = prune_taxa(taxa_sums(Bac.s.rel) > 0, Bac.s.rel) #this removes any OTU with 0s

#are there ASVs in all samples NOT WORKING
Bac.s.share<- prune_taxa(taxa_sums(Bac.s) > 0, Bac.s) #this removes any OTU with 0s
share<-phyloseq_extract_shared_otus(Bac.s.share, samp_names = sample_names(Bac.s.share))

# remove all OTUs not found at least once in both samples NOT WORKING
phylo2 = filter_taxa(Bac.s.share, function(x) sum(Bac.s.share >= 1) == (192), TRUE)

# display the OTU table
otu_table(phylo2)


#number of ASVs by species
Bac.s.share_Mcap<-subset_samples(Bac.s.share, Species=="Montipora_capitata")
    Bac.s.share_Mcap.p = prune_taxa(taxa_sums(Bac.s.share_Mcap) > 0, Bac.s.share_Mcap) #this removes any OTU with 0s

  Bac.s.share_Pcomp<-subset_samples(Bac.s.share, Species=="Porites_compressa")
    Bac.s.share_Pcomp.p = prune_taxa(taxa_sums(Bac.s.share_Pcomp) > 0, Bac.s.share_Pcomp) #this removes any OTU with 0s

Bac.s.share_Pvar<-subset_samples(Bac.s.share, Species=="Pavona_varians")
    Bac.s.share_Pvar.p = prune_taxa(taxa_sums(Bac.s.share_Pvar) > 0, Bac.s.share_Pvar) #this removes any OTU with 0s
  #by clade for Pvar
  Bac.s.share_Pvar.C1<-subset_samples(Bac.s.share_Pvar.p, Clade=="C1")
    Bac.s.share_Pvar.C1.p = prune_taxa(taxa_sums(Bac.s.share_Pvar.C1) > 0, Bac.s.share_Pvar.C1)
  Bac.s.share_Pvar.C27<-subset_samples(Bac.s.share_Pvar.p, Clade=="C27")
    Bac.s.share_Pvar.C27.p = prune_taxa(taxa_sums(Bac.s.share_Pvar.C27) > 0, Bac.s.share_Pvar.C27)
    
Bac.s.share_Pacu<-subset_samples(Bac.s.share, Species=="Pocillopora_acuta")
    Bac.s.share_Pacu.p = prune_taxa(taxa_sums(Bac.s.share_Pacu) > 0, Bac.s.share_Pacu) #this removes any OTU with 0s

#anything shared between species:
  Bac.s.share_McapPcomp<-subset_samples(Bac.s.share, Species=="Montipora_capitata" | Species=="Porites_compressa")
    Bac.s.share_McapPcomp.p = prune_taxa(taxa_sums(Bac.s.share_McapPcomp) > 0, Bac.s.share_McapPcomp) #this removes any OTU with 0s
  




# export into table
taxCN  = tax_table(Bac.s.rel.p)
taxCN = as(tax_table(Bac.s.rel.p), "matrix")
taxCN2 = as.data.frame(taxCN)
#write.csv(taxCN2, "taxCN_may18.csv")

# taxCN2$OTU <- rownames(taxCN2) #make row names a col
# taxCN3<-taxCN2[,c("OTU","Family")] # keep only columns you need
# rownames(taxCN3) <- c() #remove row namds


OTUCN = otu_table(Bac.s.rel.p)
# Extract abundance matrix from the phyloseq object
OTU.cn = as(otu_table(OTUCN), "matrix")
if(taxa_are_rows(Bac.s.rel.p)){OTU.cn <- t(OTU.cn)}
OTU.cn2 = as.data.frame(OTU.cn)
#write.csv(OTU.cn2, "OTU.cn2_may18.csv")

OTU.cn2 <- tibble::rownames_to_column(OTU.cn2, "sample_name")

MetaData16Keep<- MetaData16[,c("sample_name", "Species","Parent.ID", "Treatment","Time.Point","Tank.num","Clade")] # keep only columns you need

#OK merge
    cn.16s.samples<-merge(MetaData16Keep,OTU.cn2, by="sample_name") #merge on the metadata
write.csv(cn.16s.samples, "cn.16s.samplesMay18.csv")

cn.16s.samples[1:20,1:20]
``` 
For plotting only, subset the data: ok you can't figure this out jump to next chunk
- Families with >0.5% in one or more samples, AND nonzero in 3 or more samples. somehow relax the latter criteria if they are greater than 1% in any one sample. https://bioconductor.riken.jp/packages/3.0/bioc/vignettes/phyloseq/inst/doc/phyloseq-basics.html
```{r}        
library(genefilter)
#df use counts not Rel A
#tax glom should let you glom at Family
# phyFam<-tax_glom(Bac.s, taxrank="Family")  #this isn't doing it by family. wtf

otu_table(phyFam)[1:5, 1:5] 
tax_table(phyFam)[1:5, 1:4]
taxFAM  = tax_table(phyFam)
taxFAM<-taxFAM[!(taxFAM$Size)]
# write.csv(taxFAM,"taxFAM.csv")


# goal: only keep families with >0.5% in one ore more samples and nonzero in 3 or more samples, unless >1% in any sample

#this works for fam level and relA. now we have 543 families
phyFamTest2 <- Bac.s %>% 
  tax_glom(taxrank = "Family", NArm=FALSE) %>% #glom to family and remove anything not classified at family level
  transform_sample_counts(function(x) {x/sum(x)} ) #then turn to REl A
    sum(is.na(otu_table(phyFamTest2))) #check if there are any NA
        otu_table(phyFamTest2)[1:5, 1:5]
        tax_table(phyFamTest2)[1:6, 1:6]


#      #filter taxa not >.5% rela bund per sample, removes 25 taxa <- not exactly sure what this is doing tbh
# phyFamTest2.f = filter_taxa(phyFamTest2, function(x) mean(x) < .005,TRUE) 
    
 
test_function<-function(x){x>0} #keep only taxa with abundance at least 2 in at least 10 samples
#keep only taxa with abundance at least 1 in at least 3 samples
taxa.to.keep<-genefilter_sample(phyFamTest2, test_function, A=3) #trying with skipping the above step
head(taxa.to.keep)

phyFamTest2.fb<-prune_taxa(taxa.to.keep,phyFamTest2 )  #ok now that you did the above, PRUNE them out
    #now we are donw to 334 taxa. This is without the filter step. now there are no empty rows (samps)
    
taxFAM  = tax_table(phyFamTest2.fb) #look at it
tax_table(phyFamTest2.fb)[1:7, 1:6]


# can we change mean to max? wtf did that do?
phyFamTest2TEST = filter_taxa(phyFamTest2, function(x) max(x) > 0.005, TRUE)
tax_table(phyFamTest2TEST)[1:7, 1:6]
otu_table(phyFamTest2TEST)[1:5, 1:5]
#ok now you are down to 257 taxa. this is the 2nd heat
    OTUfam = as(otu_table(phyFamTest2TEST), "matrix")
    # transpose if necessary
    if(taxa_are_rows(phyFamTest2TEST)){OTUfam <- t(OTUfam)}
    # Coerce to data.frame
    OTUfam2.df = as.data.frame(OTUfam)
    write.csv(OTUfam2.df,"phyFamTest2TEST.csv")



# Extract abundance matrix from the phyloseq object< to look at OTU table, why isn't this working below!?
#OK some of your samples now have zero taxa. 
OTUfam = as(otu_table(phyFamTest2.fb), "matrix")
# transpose if necessary
if(taxa_are_rows(phyFamTest2.fb)){OTUfam <- t(OTUfam)}
# Coerce to data.frame
OTUfam.df = as.data.frame(OTUfam)
#write.csv(OTUfam.df,"OTUfam.df.csv")



 #ok so in theory you got rid of stuff sort of how CN wanted. 
#helpful https://www.gdc-docs.ethz.ch/MDA/handouts/MDA20_PhyloseqFormation_Mahendra_Mariadassou.pdf


#try heatmap again
theme_set(theme_bw()) 

#ok....
  plot_heatmap(phyFamTest2TEST, "NMDS", "bray", "Species","Family", sample.order = "Species") #use to show split

#try by speces: 
#Mcap
  Mcap_phyFam<-subset_samples(phyFamTest2TEST, Species=="Montipora_capitata")
  plot_heatmap(Mcap_phyFam, "NMDS", "bray", "Parent.ID","Family") #use to show split

  
  
  Mcap_phyFam.T1<-subset_samples(Mcap_phyFam, Time.Point=="T1")
  plot_heatmap(Mcap_phyFam.T1, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment")
    plot_heatmap(Mcap_phyFam.T1, "NMDS", "bray", "Clade","Family")

  Mcap_phyFam.TF<-subset_samples(Mcap_phyFam, Time.Point=="TF")
  plot_heatmap(Mcap_phyFam.TF, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment") 
  plot_heatmap(Mcap_phyFam.TF, "NMDS", "bray", "Clade","Family") 
  
#Pcomp
  Pcomp_phyFam<-subset_samples(phyFamTest2TEST, Species=="Porites_compressa")
  plot_heatmap(Pcomp_phyFam, "NMDS", "bray", "Parent.ID","Family") #use to show split
  
  Pcomp_phyFam.T1<-subset_samples(Pcomp_phyFam, Time.Point=="T1")
  plot_heatmap(Pcomp_phyFam.T1, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment")
    plot_heatmap(Pcomp_phyFam.T1, "NMDS", "bray", "Clade","Family")

  Pcomp_phyFam.TF<-subset_samples(Pcomp_phyFam, Time.Point=="TF")
  plot_heatmap(Pcomp_phyFam.TF, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment") 
  plot_heatmap(Pcomp_phyFam.TF, "NMDS", "bray", "Clade","Family") 

#Pvar
  Pvar_phyFam<-subset_samples(phyFamTest2TEST, Species=="Pavona_varians")
  plot_heatmap(Pvar_phyFam, "NMDS", "bray", "Parent.ID","Family") #use to show split
  
  Pvar_phyFam.T1<-subset_samples(Pvar_phyFam, Time.Point=="T1")
  plot_heatmap(Pvar_phyFam.T1, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment")
    plot_heatmap(Pvar_phyFam.T1, "NMDS", "bray", "Clade","Family")

  Pvar_phyFam.TF<-subset_samples(Pvar_phyFam, Time.Point=="TF")
  plot_heatmap(Pvar_phyFam.TF, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment") 
  plot_heatmap(Pvar_phyFam.TF, "NMDS", "bray", "Clade","Family") 
  
  
#Pvar
  Pacu_phyFam<-subset_samples(phyFamTest2TEST, Species=="Pocillopora_acuta")
  plot_heatmap(Pacu_phyFam, "NMDS", "bray", "Parent.ID","Family") #use to show split
  
  Pacu_phyFam.T1<-subset_samples(Pacu_phyFam, Time.Point=="T1")
  plot_heatmap(Pacu_phyFam.T1, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment")
    plot_heatmap(Pacu_phyFam.T1, "NMDS", "bray", "Clade","Family")

  Pacu_phyFam.TF<-subset_samples(Pacu_phyFam, Time.Point=="TF")
  plot_heatmap(Pacu_phyFam.TF, "NMDS", "bray", "Treatment","Family",sample.order = "Treatment") 
  plot_heatmap(Pacu_phyFam.TF, "NMDS", "bray", "Clade","Family")
  
  
#lets do some PERMANOVAs at Family level
  
  
```
#Let's try this again with vegan: THIS IS THE WAY to set up you FAMILY df AND Bray models
USE THIS FOR STATS HELP:
https://www.datanovia.com/en/lessons/compute-summary-statistics-in-r/
```{r}   
#from above 
#this works for fam level and relA. now we have 543 families
phyFamTest2 <- Bac.s.rel %>% 
  tax_glom(taxrank = "Family", NArm=FALSE) %>% #glom to family and remove anything not classified at family level
  transform_sample_counts(function(x) {x/sum(x)} ) #then turn to REl A
    sum(is.na(otu_table(phyFamTest2))) #check if there are any NA
        otu_table(phyFamTest2)[1:5, 1:5]
        tax_table(phyFamTest2)[1:6, 1:6]
        
      #phyFamTest2 <- prune_taxa(taxa_sums(phyFamTest2) > 0, phyFamTest2) #prune taxa 0s, this take 543 down to 495
    
      
      
      
#this fucking worked 20210410
# Extract abundance matrix from the phyloseq object< 
#OK some of your samples now have zero taxa. 
OTUfam.v = as(otu_table(phyFamTest2), "matrix")
# transpose if necessary
if(taxa_are_rows(phyFamTest2)){OTUfam.v <- t(OTUfam.v)}
# Coerce to data.frame
OTUfam.v.df = as.data.frame(OTUfam.v)
#write.csv(OTUfam.v.df,"OTUfam_df_20210410_take4.csv")

#ok the above gets you the OTU table, now replace those names with family names

    # taxFAM  = tax_table(phyFamTest2) #look at it
    # tax_table(taxFAM)[1:7, 1:6]
    # # write.csv(taxFAM,"taxFAM.csv")
    # 
    # taxFAM  = tax_table(phyFamTest2) #look at it again
    # tax_table(taxFAM)[1:7, 1:6]
    # #write.csv(taxFAM,"taxFAM_take4.csv")
    # 
    # #reuplaod the OTU tble with family names - YOU DID THIS IN EXCEL, you can prob do in code but will take a hot min to learn how
    # Fam.16s<-read.csv("Data/OTUfam_df_20210410_take4.csv")
    #OTUfam.v.df <- same as above but by OTU name!
      #for vegan. see below for phyloseq


# filter: goal 
#Keep Families with >0.5% in one or more samples, AND nonzero in 3 or more samples. somehow relax the latter criteria if they are greater than 1% in any one sample.
#get summary stats with dplry

# df Fam.16s krissy zoom

Fam.16s.summary <- OTUfam.v.df %>%
  gather(taxa, value, -(1)) %>% #WIDE TO LONG FORMAT, minus sample name in col 1. if you want to include species, add the meta and include here and the next step
  group_by(taxa) %>% #you want summary stat for each unique taxa
  dplyr::summarise(
    Mean = mean(value, na.rm=TRUE),
    Min = min(value, na.rm=TRUE),
    Max = max(value, na.rm=TRUE),
    StDev = sd(value, na.rm = TRUE)
  )

#use this summary df to do filtering
abundFam<-Fam.16s.summary %>%
  filter(Max>0.005) #try diff things to see how many you keep, here this get you to 257 families (0.005). do this by species later

#THIS APPLIES ABOVE TO DF
OTU_name <- abundFam$taxa # a list of all the families that met filter criteria
col.nos <- which(colnames(OTUfam.v.df) %in% OTU_name) #which call nam is in the list of names we are keeping
Fam.abund.f <- cbind(OTUfam.v.df[1], OTUfam.v.df[col.nos]) #if you have metadata you need to deal, this removes those

#sort from most abund to least abund
arrange(abundFam, desc(Mean)) #ie if you want to know the top 20 families across entire dataset. 

    #smash in meta in vegan, multivariate stats
    #MetaData16 metadata
    MetaData16Keep<- MetaData16[,c("sample_name", "Species","Parent.ID", "Treatment","Time.Point","Tank.num","Clade")] # keep only columns     you need
    Fam.filter.16s<-merge(MetaData16Keep,Fam.abund.f) #merge on the metadata


#### OR Fam.abund.f  and smash back into a phyloseq
outfam <- otu_table(Fam.abund.f, taxa_are_rows=F)  

# Read the data into phyloseq
Bac.seqFam = phyloseq(outfam,tax, sam,treefile) #THIS WORKS
Bac.seqFam
Bac.seqFam.df <- sample_data(Bac.seqFam)
#write.csv(Fam.abund.f,"Fam.abund.f.csv")
#write.csv(Bac.seqFam.df,"Bac.seqFam.df.csv")

#do permanova with this new phyloseq obj for FAM that is filetered 


Bac.seqFam_stats<- Bac.seqFam #make a copy
otu_table(Bac.s.rel_stats)[1:5, 1:5] #check rela worked


#bray curtis presence-absense and abundance
## REMOVE T0 for models
Bac.seqFam_stats.T1F<-subset_samples(Bac.seqFam_stats, Time.Point=="T1"|Time.Point=="TF")

#bray curtis presence-absense and abundance
 
#all speices
set.seed(30)

#make bray curtis data matrix
Fam.16s.all <- phyloseq::distance(Bac.seqFam_stats.T1F, method = "bray")
  Fam.samp.df <- as(sample_data(Bac.seqFam_stats.T1F), "data.frame") #sample df
  #adonis test
adonis(Fam.16s.all ~ Species*Treatment*Time.Point, data = Fam.samp.df)
    adonis(Fam.16s.all ~ Species*Treatment*Time.Point*Clade, data = Fam.samp.df)
    
# Homogeneity of dispersion test <-NOT WORKING...why
betaAll <- betadisper(Fam.16s.all, Fam.samp.df$Species)
permutest(betaAll, pairwise = T) 
(bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
plot(bc.all.HSD)


#Ordination NMDS for all species, T1Tf
set.seed(50)
ord.Bac_RelA_stats <- ordinate(Bac.seqFam_stats.T1F, "NMDS", "bray", trymax = 500) 
stressplot(ord.Bac_RelA_stats)
scores.ord.Bac_RelA_stats<-as.data.frame(cbind(vegan::scores(ord.Bac_RelA_stats, display="sites"))) # these are "points" ask hannah 

scores.ord.Bac_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.Bac_RelA_stats$Species <- samp.df$Species


ggplot(scores.ord.Bac_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("Family NMDS 16s Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")

############# mcap


# df Bac.seqFam_stats.T1F

#Mcap
mcap.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Montipora_capitata")
#mcap.f.bac.T1F = prune_taxa(taxa_sums(mcap.f.bac.T1F) > 0, mcap.f.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
mcap.f.bac.T1 <- subset_samples(mcap.f.bac.T1F, Time.Point == "T1")
 # mcap.f.bac.T1 = prune_taxa(taxa_sums(mcap.f.bac.T1) > 0, mcap.f.bac.T1) #this removes any OTU with 0s
mcap.f.bac.TF <- subset_samples(mcap.f.bac.T1F, Time.Point == "TF")
 # mcap.f.bac.TF = prune_taxa(taxa_sums(mcap.f.bac.TF) > 0, mcap.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  set.seed(30)
  bc.mcap.f.T1F <- phyloseq::distance(mcap.f.bac.T1F, method = "bray")
     mcap.f.samp.df.T1F <- as(sample_data(mcap.f.bac.T1F), "data.frame") #sample df
    adonis(bc.mcap.f.T1F ~ Treatment*Time.Point, data = mcap.f.samp.df.T1F) #adonis test no clade
    adonis(bc.mcap.f.T1F ~ Clade*Treatment*Time.Point, data = mcap.f.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(mcap.f.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
    set.seed(30)
  bc.mcapT1 <- phyloseq::distance(mcap.f.bac.T1, method = "bray") 
      mcap.f.samp.df.T1 <- as(sample_data(mcap.f.bac.T1), "data.frame") 
    adonis(bc.mcapT1 ~ Clade*Treatment, data = mcap.f.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.mcap.f.T1 <- betadisper(bc.mcapT1, mcap.f.samp.df.T1$Clade)  #clade
    permutest(beta.mcap.f.T1, pairwise = T) 
  beta.mcap.f.T1 <- betadisper(bc.mcapT1, mcap.f.samp.df.T1$Treatment) #treatment
    permutest(beta.mcap.f.T1, pairwise = T) 

#TF
        set.seed(30)

  bc.mcap.f.TF <- phyloseq::distance(mcap.f.bac.TF, method = "bray")
     mcap.f.samp.df.TF <- as(sample_data(mcap.f.bac.TF), "data.frame") 
    adonis(bc.mcap.f.TF ~ Clade*Treatment, data = mcap.f.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.mcap.f.TF <- betadisper(bc.mcap.f.TF, mcap.f.samp.df.TF$Clade) 
    permutest(beta.mcap.f.TF, pairwise = T) 
  beta.mcap.f.TF <- betadisper(bc.mcap.f.TF, mcap.f.samp.df.TF$Treatment) 
    permutest(beta.mcap.f.TF, pairwise = T) 

############ pcomp
#Pcomp
pcomp.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Porites_compressa")
#pcomp.f.bac.T1F = prune_taxa(taxa_sums(pcomp.f.bac.T1F) > 0, pcomp.f.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
pcomp.f.bac.T1 <- subset_samples(pcomp.f.bac.T1F, Time.Point == "T1")
 # pcomp.f.bac.T1 = prune_taxa(taxa_sums(pcomp.f.bac.T1) > 0, pcomp.f.bac.T1) #this removes any OTU with 0s

pcomp.f.bac.TF <- subset_samples(pcomp.f.bac.T1F, Time.Point == "TF")
 # pcomp.f.bac.TF = prune_taxa(taxa_sums(pcomp.f.bac.TF) > 0, pcomp.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
          set.seed(30)

  bc.pcomp.f.T1F <- phyloseq::distance(pcomp.f.bac.T1F, method = "bray")
     pcomp.f.samp.df.T1F <- as(sample_data(pcomp.f.bac.T1F), "data.frame") #sample df
    adonis(bc.pcomp.f.T1F ~ Treatment*Time.Point, data = pcomp.f.samp.df.T1F) #adonis test no clade
    adonis(bc.pcomp.f.T1F ~ Clade*Treatment*Time.Point, data = pcomp.f.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(pcomp.f.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
              set.seed(30)

  bc.PcompT1 <- phyloseq::distance(pcomp.f.bac.T1, method = "bray") 
      pcomp.f.samp.df.T1 <- as(sample_data(pcomp.f.bac.T1), "data.frame") 
    adonis(bc.PcompT1 ~ Clade*Treatment, data = pcomp.f.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.pcomp.f.T1 <- betadisper(bc.PcompT1, pcomp.f.samp.df.T1$Clade)  #clade
    permutest(beta.pcomp.f.T1, pairwise = T) 
  beta.pcomp.f.T1 <- betadisper(bc.PcompT1, pcomp.f.samp.df.T1$Treatment) #treatment
    permutest(beta.pcomp.f.T1, pairwise = T) 

#TF
              set.seed(30)

  bc.pcomp.f.TF <- phyloseq::distance(pcomp.f.bac.TF, method = "bray")
     pcomp.f.samp.df.TF <- as(sample_data(pcomp.f.bac.TF), "data.frame") 
    adonis(bc.pcomp.f.TF ~ Clade*Treatment, data = pcomp.f.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.pcomp.f.TF <- betadisper(bc.pcomp.f.TF, pcomp.f.samp.df.TF$Clade) 
    permutest(beta.pcomp.f.TF, pairwise = T) 
  beta.pcomp.f.TF <- betadisper(bc.pcomp.f.TF, pcomp.f.samp.df.TF$Treatment) 
    permutest(beta.pcomp.f.TF, pairwise = T) 

#######pvar
    #Pvar
pvar.f.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Pavona_varians")
#pvar.f.f.bac.T1F = prune_taxa(taxa_sums(pvar.f.f.bac.T1F) > 0, pvar.f.f.bac.T1F) #this removes any OTU with 0s
 
#make dfs you need 
pvar.f.f.bac.T1 <- subset_samples(pvar.f.f.bac.T1F, Time.Point == "T1")
 # pvar.f.f.bac.T1 = prune_taxa(taxa_sums(pvar.f.f.bac.T1) > 0, pvar.f.f.bac.T1) #this removes any OTU with 0s

pvar.f.f.bac.TF <- subset_samples(pvar.f.f.bac.T1F, Time.Point == "TF")
#  pvar.f.f.bac.TF = prune_taxa(taxa_sums(pvar.f.f.bac.TF) > 0, pvar.f.f.bac.TF) #this removes any OTU with 0s
 
#T1TF
#make bray curtis data matrix -T1 and TF
                set.seed(30)

  bc.pvar.f.f.T1F <- phyloseq::distance(pvar.f.f.bac.T1F, method = "bray")
     pvar.f.f.samp.df.T1F <- as(sample_data(pvar.f.f.bac.T1F), "data.frame") #sample df
    adonis(bc.pvar.f.f.T1F ~ Treatment*Time.Point, data = pvar.f.f.samp.df.T1F) #adonis test no clade
    adonis(bc.pvar.f.f.T1F ~ Clade*Treatment*Time.Point, data = pvar.f.f.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(pvar.f.f.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
                  set.seed(30)

  bc.PvarT1 <- phyloseq::distance(pvar.f.f.bac.T1, method = "bray") 
      pvar.f.f.samp.df.T1 <- as(sample_data(pvar.f.f.bac.T1), "data.frame") 
    adonis(bc.PvarT1 ~ Treatment, data = pvar.f.f.samp.df.T1) 
        adonis(bc.PvarT1 ~ Clade*Treatment, data = pvar.f.f.samp.df.T1)  

# Homogeneity of dispersion test  
  beta.pvar.f.f.T1 <- betadisper(bc.PvarT1, pvar.f.f.samp.df.T1$Clade)  #clade
    permutest(beta.pvar.f.f.T1, pairwise = T) 
  beta.pvar.f.f.T1 <- betadisper(bc.PvarT1, pvar.f.f.samp.df.T1$Treatment) #treatment
    permutest(beta.pvar.f.f.T1, pairwise = T) 

#TF
                  set.seed(30)

  bc.pvar.f.f.TF <- phyloseq::distance(pvar.f.f.bac.TF, method = "bray")
     pvar.f.f.samp.df.TF <- as(sample_data(pvar.f.f.bac.TF), "data.frame") 
    adonis(bc.pvar.f.f.TF ~ Treatment, data = pvar.f.f.samp.df.TF)
        adonis(bc.pvar.f.f.TF ~ Clade*Treatment, data = pvar.f.f.samp.df.TF)

  # Homogeneity of dispersion test  
  beta.pvar.f.f.TF <- betadisper(bc.pvar.f.f.TF, pvar.f.f.samp.df.TF$Clade) 
    permutest(beta.pvar.f.f.TF, pairwise = T) 
  beta.pvar.f.f.TF <- betadisper(bc.pvar.f.f.TF, pvar.f.f.samp.df.TF$Treatment) 
    permutest(beta.pvar.f.f.TF, pairwise = T) 

####### pacuta

#Pacu
pacu.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Pocillopora_acuta")
#pacu.f.bac.T1F = prune_taxa(taxa_sums(pacu.f.bac.T1F) > 0, pacu.f.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
pacu.f.bac.T1 <- subset_samples(pacu.f.bac.T1F, Time.Point == "T1")
#  pacu.f.bac.T1 = prune_taxa(taxa_sums(pacu.f.bac.T1) > 0, pacu.f.bac.T1) #this removes any OTU with 0s

pacu.f.bac.TF <- subset_samples(pacu.f.bac.T1F, Time.Point == "TF")
 # pacu.f.bac.TF = prune_taxa(taxa_sums(pacu.f.bac.TF) > 0, pacu.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  set.seed(30)
  bc.pacu.f.T1F <- phyloseq::distance(pacu.f.bac.T1F, method = "bray")
     pacu.f.samp.df.T1F <- as(sample_data(pacu.f.bac.T1F), "data.frame") #sample df
    adonis(bc.pacu.f.T1F ~ Treatment*Time.Point, data = pacu.f.samp.df.T1F) #adonis test no clade
    adonis(bc.pacu.f.T1F ~ Clade*Treatment*Time.Point, data = pacu.f.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(pacu.f.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
    set.seed(30)
  bc.PacuT1 <- phyloseq::distance(pacu.f.bac.T1, method = "bray") 
      pacu.f.samp.df.T1 <- as(sample_data(pacu.f.bac.T1), "data.frame") 
    adonis(bc.PacuT1 ~ Clade*Treatment, data = pacu.f.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.pacu.f.T1 <- betadisper(bc.PacuT1, pacu.f.samp.df.T1$Clade)  #clade
    permutest(beta.pacu.f.T1, pairwise = T) 
  beta.pacu.f.T1 <- betadisper(bc.PacuT1, pacu.f.samp.df.T1$Treatment) #treatment
    permutest(beta.pacu.f.T1, pairwise = T) 

    
#TF
    set.seed(30)
  bc.pacu.f.TF <- phyloseq::distance(pacu.f.bac.TF, method = "bray")
     pacu.f.samp.df.TF <- as(sample_data(pacu.f.bac.TF), "data.frame") 
    adonis(bc.pacu.f.TF ~ Clade*Treatment, data = pacu.f.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.pacu.f.TF <- betadisper(bc.pacu.f.TF, pacu.f.samp.df.TF$Clade) 
    permutest(beta.pacu.f.TF, pairwise = T) 
  beta.pacu.f.TF <- betadisper(bc.pacu.f.TF, pacu.f.samp.df.TF$Treatment) 
    permutest(beta.pacu.f.TF, pairwise = T) 
```

# unifrac load in tree above
http://bioconductor.riken.jp/packages/3.5/bioc/vignettes/phyloseq/inst/doc/phyloseq-analysis.html
http://joey711.github.io/phyloseq-demo/unifrac.html
http://joey711.github.io/phyloseq-demo/
 
Weighted UniFrac - which does take into account differences in abundance of taxa between samples, but takes longer to calculate; and

#Unweighted UniFrac FAMILY - which only considers the presence/absence of taxa between sample pairs.
```{r}     
 
#look at your trees
bac.prot <- subset_taxa(Bac.seqFam_stats.T1F, Phylum=="Proteobacteria")

plot_tree(bac.prot, color="Species", label.tips="Genus", size="Abundance")


#what if we try this on T0 just to prove diff?
Bac.seqFam_stats.T0<-subset_samples(Bac.seqFam_stats, Time.Point=="T0")

#exmaple, first pcoa then nmds

GPUF <- UniFrac(Bac.seqFam_stats.T0, weighted=T)
GloPa.pcoa = ordinate(Bac.seqFam_stats.T0, method="PCoA", distance=GPUF)
plot_scree(GloPa.pcoa, "T0 only, UniFrac/PCoA")

GPUF2 <- UniFrac(Bac.seqFam_stats.T1F, weighted=T)
GloPa.pcoa2 = ordinate(Bac.seqFam_stats.T1F, method="PCoA", distance=GPUF2)
plot_scree(GloPa.pcoa2, "T1F only, UniFrac/PCoA")

#PcoA
# df Bac.seqFam_stats.T0
ordu = ordinate(Bac.seqFam_stats.T0, "PCoA", "unifrac", weighted=TRUE)
plot_ordination(Bac.seqFam_stats.T0, ordu, color="Species", shape="Species")
dis<-distance(Bac.seqFam_stats.T0, method="unifrac")
(plot <- plot_ordination(Bac.seqFam_stats.T0, ordu, "samples", color="Species") + 
  geom_point(size=5) + geom_path() + scale_colour_hue(guide = FALSE) )

# df Bac.seqFam_stats.T1F
ordu = ordinate(Bac.seqFam_stats.T1F, "PCoA", "unifrac", weighted=TRUE)
plot_ordination(Bac.seqFam_stats.T1F, ordu, color="Species", shape="Species")
dis<-distance(Bac.seqFam_stats.T1F, method="unifrac")
(plot <- plot_ordination(Bac.seqFam_stats.T1F, ordu, "samples", color="Species") + 
  geom_point(size=5) + geom_path() + scale_colour_hue(guide = FALSE) )

#nmds, T0
GP.NMDS <- ordinate(Bac.seqFam_stats.T0, "NMDS", GPUF)

(p <- plot_ordination(Bac.seqFam_stats.T0, GP.NMDS, "samples", color="Species") +
geom_point(size=5) )

#nmds, T1TF
GP.NMDS2 <- ordinate(Bac.seqFam_stats.T1F, "NMDS", GPUF2)

(p <- plot_ordination(Bac.seqFam_stats.T1F, GP.NMDS2, "samples", color="Species") +
geom_point(size=5) )


#weighted Unifrac (presence absence + abundance + phylogeny) HE code  
wu.mfol <- phyloseq::distance(Bac.seqFam_stats.T1F, method = "wunifrac")
adonis(wu.mfol ~ Species*Treatment*Time.Point, strata = Fam.samp.df$Species, data = Fam.samp.df) #not sig
permutest(betadisper(wu.mfol, Fam.samp.df$Species, type = "centroid")) #sig
```
# Family unifrac stats T1TF and plots. 
```{r}  
  
nb.cols <- 30
mycolors <- colorRampPalette(brewer.pal(11, "Set1"))(nb.cols)

Fam.bac.colors<-c("Hyphomonadaceae" = "#F3C300", 
                  "P30B-42" = "#008856",
                  "Gammaproteobacteria_unclassified" = "#3399FF",
                  "Nodosilineaceae" = "#FF33CC",
                  "Chlorobiaceae" = "#33FFCC",
                  "Cyclobacteriaceae" = "#BE0032",
                  "Amoebophilaceae" = "#C2B280",
                  "Flavobacteriaceae" = "#222222",
                  "Endozoicomonadaceae" = "#99FF00",
                  "Rhodobacteraceae" = "#E68FAC",
                  "Pirellulaceae" = "#A3623A",
                  "Phycispharaceae" = "#F99379",
                  "Saprospiraceae" = "#604E97",
                  "Paraspirulinaceae" = "#F6A600",
                  "Francisellaceae" = "#B3446C",
                  "Bacteria_unclassified_Otu00012" = "#0067A5",
                  "Kiloniellaceae" = "#882D17",
                   "uncultured_Otu00079" = "#8DB600",
                   "Nostocaceae" = "#654522",
                   "uncultured_Otu00270" = "#E25822",
                   "Alteromonadaceae" = "#848482",
                    "uncultured_Otu00533" = "#2B3D26",
                   "uncultured_Otu00563" = "#FF1500",
                  "uncultured_Otu01647" = "#F3C300",
                  "uncultured_Otu00872" = "#875692",
                    "Alphaproteobacteria_unclassified" = "#A1CAF1",

                  "Pseudoalteromonadaceae" = "#F38400",
                  "Montipora_capitata"="red",
                  "Porites_compressa"="purple",
                  "Pavona_varians"="green",
                  "Pocillopora_acuta"="blue"
                  ) 
############################### Family unifrac stats
#make unifrac matrix
Bac.fam.T1F <- phyloseq::distance(Bac.seqFam_stats.T1F, method = "wunifrac") #dist matrix
Fam.samp.df <- as(sample_data(Bac.seqFam_stats.T1F), "data.frame") #sample df
  #adonis 
    set.seed(30)
      adonis(Bac.fam.T1F ~ Species*Treatment*Time.Point, data = Fam.samp.df)
    set.seed(30)    
      adonis(Fam.16s.all ~ Species*Treatment*Time.Point*Clade, data = Fam.samp.df)
  # Homogeneity of dispersion test  
    set.seed(30)    
      permutest(betadisper(Bac.fam.T1F, Fam.samp.df$Clade, type = "centroid")) #put in distance matrix

############# mcap #### 

#Mcap
mcap.u.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Montipora_capitata")
mcap.u.f.bac.T1F = prune_taxa(taxa_sums(mcap.u.f.bac.T1F) > 0, mcap.u.f.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
mcap.u.f.bac.T1 <- subset_samples(mcap.u.f.bac.T1F, Time.Point == "T1")
  mcap.u.f.bac.T1 = prune_taxa(taxa_sums(mcap.u.f.bac.T1) > 0, mcap.u.f.bac.T1) #this removes any OTU with 0s
mcap.u.f.bac.TF <- subset_samples(mcap.u.f.bac.T1F, Time.Point == "TF")
  mcap.u.f.bac.TF = prune_taxa(taxa_sums(mcap.u.f.bac.TF) > 0, mcap.u.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  set.seed(30)
  bc.mcap.u.f.T1F <- phyloseq::distance(mcap.u.f.bac.T1F, method = "wunifrac")
     mcap.u.f.samp.df.T1F <- as(sample_data(mcap.u.f.bac.T1F), "data.frame") #sample df
       set.seed(30)
    adonis(bc.mcap.u.f.T1F ~ Treatment*Time.Point, data = mcap.u.f.samp.df.T1F) #adonis test no clade
      set.seed(30)
    adonis(bc.mcap.u.f.T1F ~ Clade*Treatment*Time.Point, data = mcap.u.f.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(mcap.u.f.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
      # Homogeneity of dispersion test  
set.seed(30)
    permutest(betadisper(bc.mcap.u.f.T1F, mcap.u.f.samp.df.T1F$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(bc.mcap.u.f.T1F, mcap.u.f.samp.df.T1F$Treatment, type = "centroid")) #put in distance matrix
    
#T1
    
  u.mcapT1 <- phyloseq::distance(mcap.u.f.bac.T1, method = "wunifrac") 
      mcap.u.f.samp.df.T1 <- as(sample_data(mcap.u.f.bac.T1), "data.frame") 
          set.seed(30)
    adonis(u.mcapT1 ~ Treatment, data = mcap.u.f.samp.df.T1)  
     set.seed(30)
    adonis(u.mcapT1 ~ Clade*Treatment, data = mcap.u.f.samp.df.T1)  
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.mcapT1, mcap.u.f.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.mcapT1, mcap.u.f.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix

#TF
  u.mcap.u.f.TF <- phyloseq::distance(mcap.u.f.bac.TF, method = "wunifrac")
     mcap.u.f.samp.df.TF <- as(sample_data(mcap.u.f.bac.TF), "data.frame")
    set.seed(30)
    adonis(u.mcap.u.f.TF ~ Treatment, data = mcap.u.f.samp.df.TF)
    set.seed(30)
    adonis(u.mcap.u.f.TF ~ Clade*Treatment, data = mcap.u.f.samp.df.TF)
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.mcap.u.f.TF, mcap.u.f.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.mcap.u.f.TF, mcap.u.f.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix
 
 # plotting only the top 10 FAMs    mcap.u.f.bac.T1F
    #Sort the Phyla by abundance and pick the top 5
    top10P.names.mcap = sort(tapply(taxa_sums(mcap.u.f.bac.T1F), tax_table(mcap.u.f.bac.T1F)[, "Family"], mean), TRUE)[1:10]
      #Cut down the  data to only the top 10 Phyla
      mcap.u.f.bac.T1F.top10P = subset_taxa(mcap.u.f.bac.T1F, Family %in% names(top10P.names.mcap))
      
      #pull out of phyloseq     
        mcap.u.f.bac.T1F.top10P.OTU = as(otu_table(mcap.u.f.bac.T1F.top10P), "matrix")
       # transpose if necessary
          # if(taxa_are_rows(mcap.u.f.bac.T1F.top5P)){OTUfam.v <- t(mcap.u.f.bac.T1F.top5P.OTU)}
          # Coerce to data.frame
            mcap.u.f.bac.T1F.top10P.OTU.df = as.data.frame(mcap.u.f.bac.T1F.top10P.OTU) 
      #rename Fam with taxa
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00104"] <- "Hyphomonadaceae"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00004"] <- "P30B-42"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00058"] <- "Chlorobiaceae"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00030"] <- "Cyclobacteriaceae"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00206"] <- "Flavobacteriaceae"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00015"] <- "Nodosilineaceae"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00062"] <- "Alteromonadaceae"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00001"] <- "Endozoicomonadaceae"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00226"] <- "Gammaproteobacteria_unclassified"
            names(mcap.u.f.bac.T1F.top10P.OTU.df)[names(mcap.u.f.bac.T1F.top10P.OTU.df) == "Otu00025"] <- "Rhodobacteraceae"
    #Add metadata
      mcap.u.f.bac.T1F.top10P.sd <- as(sample_data(mcap.u.f.bac.T1F.top10P), "data.frame") #create sample data frame to view
         mcap.u.f.bac.T1F.top10P.sd<-mcap.u.f.bac.T1F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    mcap.16s.fam.top10<- merge(mcap.u.f.bac.T1F.top10P.OTU.df,mcap.u.f.bac.T1F.top10P.sd, by="row.names") #merge by rownames
       mcap.16s.fam.top10.m <- melt(mcap.16s.fam.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(mcap.16s.fam.top10.m)[names(mcap.16s.fam.top10.m) == "value"] <- "relabund"
            names(mcap.16s.fam.top10.m)[names(mcap.16s.fam.top10.m) == "variable"] <- "family"

#Means
mcap.16s.fam.top10.m_mean <- ddply(mcap.16s.fam.top10.m, c("Treatment", "Clade", "Time.Point","family"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
mcap.16s.fam.top10.m_mean

mcap.16s.fam.top10.m_mean$Code<-paste(mcap.16s.fam.top10.m_mean$Treatment, mcap.16s.fam.top10.m_mean$Time.Point,mcap.16s.fam.top10.m_mean$Clade)

#mcap plots
#plot Mcap
mcap.16s.fam.top10.m_mean_plot<-ggplot(data=mcap.16s.fam.top10.m_mean, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("Mcap top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));mcap.16s.fam.top10.m_mean_plot
mcap.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)

   
    
############ pcomp ####
#Pcomp
pcomp.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Porites_compressa")
pcomp.f.bac.T1F = prune_taxa(taxa_sums(pcomp.f.bac.T1F) > 0, pcomp.f.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
pcomp.f.bac.T1 <- subset_samples(pcomp.f.bac.T1F, Time.Point == "T1")
  pcomp.f.bac.T1 = prune_taxa(taxa_sums(pcomp.f.bac.T1) > 0, pcomp.f.bac.T1) #this removes any OTU with 0s

pcomp.f.bac.TF <- subset_samples(pcomp.f.bac.T1F, Time.Point == "TF")
 pcomp.f.bac.TF = prune_taxa(taxa_sums(pcomp.f.bac.TF) > 0, pcomp.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make wunifrac curtis data matrix -T1 and TF
  u.pcomp.f.T1F <- phyloseq::distance(pcomp.f.bac.T1F, method = "wunifrac")
     pcomp.f.samp.df.T1F <- as(sample_data(pcomp.f.bac.T1F), "data.frame") #sample df
      set.seed(30)
    adonis(u.pcomp.f.T1F ~ Treatment*Time.Point, data = pcomp.f.samp.df.T1F) #adonis test no clade
   set.seed(30)
    adonis(u.pcomp.f.T1F ~ Clade*Treatment*Time.Point, data = pcomp.f.samp.df.T1F)#adonis test with clade
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pcomp.f.T1F, pcomp.f.samp.df.T1F$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pcomp.f.T1F, pcomp.f.samp.df.T1F$Treatment, type = "centroid")) #put in distance matrix
    
#T1
  u.PcompT1 <- phyloseq::distance(pcomp.f.bac.T1, method = "wunifrac") 
      pcomp.f.samp.df.T1 <- as(sample_data(pcomp.f.bac.T1), "data.frame") 
   set.seed(30)
   adonis(u.PcompT1 ~ Treatment, data = pcomp.f.samp.df.T1)  
    set.seed(30)
   adonis(u.PcompT1 ~ Clade*Treatment, data = pcomp.f.samp.df.T1)  
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.PcompT1, pcomp.f.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.PcompT1, pcomp.f.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix

#TF
u.pcomp.f.TF <- phyloseq::distance(pcomp.f.bac.TF, method = "wunifrac")
     pcomp.f.samp.df.TF <- as(sample_data(pcomp.f.bac.TF), "data.frame") 
      set.seed(30)
    adonis(u.pcomp.f.TF ~ Treatment, data = pcomp.f.samp.df.TF)
     set.seed(30)
    adonis(u.pcomp.f.TF ~ Clade*Treatment, data = pcomp.f.samp.df.TF)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pcomp.f.TF, pcomp.f.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pcomp.f.TF, pcomp.f.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix

 #try plotting only the top 10 FAMs     #### 
    #Sort the Phyla by abundance and pick the top 5
    top10P.names.pcomp = sort(tapply(taxa_sums(pcomp.f.bac.T1F), tax_table(pcomp.f.bac.T1F)[, "Family"], sum), TRUE)[1:10]
      #Cut down the  data to only the top 10 Phyla
      pcomp.u.f.bac.T1F.top10P = subset_taxa(pcomp.f.bac.T1F, Family %in% names(top10P.names.pcomp))
      
      #pull out of phyloseq     
        pcomp.u.f.bac.T1F.top10P.OTU = as(otu_table(pcomp.u.f.bac.T1F.top10P), "matrix")
       # transpose if necessary
          # if(taxa_are_rows(pcomp.u.f.bac.T1F.top5P)){OTUfam.v <- t(pcomp.u.f.bac.T1F.top5P.OTU)}
          # Coerce to data.frame
            pcomp.u.f.bac.T1F.top10P.OTU.df = as.data.frame(pcomp.u.f.bac.T1F.top10P.OTU) 
      #rename Fam with taxa
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00068"] <- "Alphaproteobacteria_unclassified"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00274"] <- "Pirellulaceae"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00374"] <- "Phycispharaceae"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00011"] <- "Amoebophilaceae"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00206"] <- "Flavobacteriaceae"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00156"] <- "Saprospiraceae"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00146"] <- "Paraspirulinaceae"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00001"] <- "Endozoicomonadaceae"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00116"] <- "Francisellaceae"
            names(pcomp.u.f.bac.T1F.top10P.OTU.df)[names(pcomp.u.f.bac.T1F.top10P.OTU.df) == "Otu00025"] <- "Rhodobacteraceae"
    #Add metadata
      pcomp.u.f.bac.T1F.top10P.sd <- as(sample_data(pcomp.u.f.bac.T1F.top10P), "data.frame") #create sample data frame to view
         pcomp.u.f.bac.T1F.top10P.sd<-pcomp.u.f.bac.T1F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    pcomp.16s.fam.top10<- merge(pcomp.u.f.bac.T1F.top10P.OTU.df,pcomp.u.f.bac.T1F.top10P.sd, by="row.names") #merge by rownames
       pcomp.16s.fam.top10.m <- melt(pcomp.16s.fam.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(pcomp.16s.fam.top10.m)[names(pcomp.16s.fam.top10.m) == "value"] <- "relabund"
            names(pcomp.16s.fam.top10.m)[names(pcomp.16s.fam.top10.m) == "variable"] <- "family"

#Means
pcomp.16s.fam.top10.m_mean <- ddply(pcomp.16s.fam.top10.m, c("Treatment", "Clade", "Time.Point","family"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
pcomp.16s.fam.top10.m_mean

pcomp.16s.fam.top10.m_mean$Code<-paste(pcomp.16s.fam.top10.m_mean$Treatment, pcomp.16s.fam.top10.m_mean$Time.Point,pcomp.16s.fam.top10.m_mean$Clade)

#pcomp plots
#plot pcomp
pcomp.16s.fam.top10.m_mean_plot<-ggplot(data=pcomp.16s.fam.top10.m_mean, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("pcomp top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));pcomp.16s.fam.top10.m_mean_plot
pcomp.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)
    
    #remove endo pcomp.16s.fam.top10.m_mean
pcomp.16s.fam.top10.m_mean.NoEndo<-subset(pcomp.16s.fam.top10.m_mean, !family=="Endozoicomonadaceae")
    
    #plot pcomp
pcomp.16s.fam.top10.m_mean_plot<-ggplot(data=pcomp.16s.fam.top10.m_mean.NoEndo, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("pcomp top 10 family No Endo")+
  theme(plot.title = element_text(size=20, face = "italic"));pcomp.16s.fam.top10.m_mean_plot
pcomp.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)
    
    
#######pvar ####
    #Pvar
pvar.f.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Pavona_varians")
#pvar.f.f.bac.T1F = prune_taxa(taxa_sums(pvar.f.f.bac.T1F) > 0, pvar.f.f.bac.T1F) #this removes any OTU with 0s
 
#make dfs you need 
pvar.f.f.bac.T1 <- subset_samples(pvar.f.f.bac.T1F, Time.Point == "T1")
 # pvar.f.f.bac.T1 = prune_taxa(taxa_sums(pvar.f.f.bac.T1) > 0, pvar.f.f.bac.T1) #this removes any OTU with 0s

pvar.f.f.bac.TF <- subset_samples(pvar.f.f.bac.T1F, Time.Point == "TF")
#  pvar.f.f.bac.TF = prune_taxa(taxa_sums(pvar.f.f.bac.TF) > 0, pvar.f.f.bac.TF) #this removes any OTU with 0s
 
Bac.seqFam_stats.Pvar<-subset_samples(Bac.seqFam_stats, Species == "Pavona_varians")
Bac.seqFam_stats.Pvar.Ambient<-subset_samples(Bac.seqFam_stats.Pvar, Treatment == "Ambient")



#T1TF
#make wunifrac curtis data matrix -T1 and TF
                set.seed(30)

  u.pvar.f.f.T1F <- phyloseq::distance(pvar.f.f.bac.T1F, method = "wunifrac")
     pvar.f.f.samp.df.T1F <- as(sample_data(pvar.f.f.bac.T1F), "data.frame") #sample df
     set.seed(30)
adonis(u.pvar.f.f.T1F ~ Treatment*Time.Point, data = pvar.f.f.samp.df.T1F) #adonis test no clade
    set.seed(30)
    adonis(u.pvar.f.f.T1F ~ Clade*Treatment*Time.Point, data = pvar.f.f.samp.df.T1F)#adonis test with clade
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pvar.f.f.T1F, pvar.f.f.samp.df.T1F$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pvar.f.f.T1F, pvar.f.f.samp.df.T1F$Treatment, type = "centroid")) #put in distance matrix
    
#T1
                 

  u.PvarT1 <- phyloseq::distance(pvar.f.f.bac.T1, method = "wunifrac") 
      pvar.f.f.samp.df.T1 <- as(sample_data(pvar.f.f.bac.T1), "data.frame") 
   set.seed(30)
   adonis(u.PvarT1 ~ Treatment, data = pvar.f.f.samp.df.T1) 
     set.seed(30)
     adonis(u.PvarT1 ~ Clade*Treatment, data = pvar.f.f.samp.df.T1)  

# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.PvarT1, pvar.f.f.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.PvarT1, pvar.f.f.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix

#TF
  u.pvar.f.f.TF <- phyloseq::distance(pvar.f.f.bac.TF, method = "wunifrac")
     pvar.f.f.samp.df.TF <- as(sample_data(pvar.f.f.bac.TF), "data.frame") 
      set.seed(30)
      adonis(u.pvar.f.f.TF ~ Treatment, data = pvar.f.f.samp.df.TF)
        set.seed(30)
        adonis(u.pvar.f.f.TF ~ Clade*Treatment, data = pvar.f.f.samp.df.TF)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pvar.f.f.TF, pvar.f.f.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pvar.f.f.TF, pvar.f.f.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix

 #try plotting only the top 10 FAMs  ####    
     #Sort the fam by abundance and pick the top 
    top10P.names.Pvar = sort(tapply(taxa_sums(pvar.f.f.bac.T1F), tax_table(pvar.f.f.bac.T1F)[, "Family"], sum), TRUE)[1:11]
    #top10P.names.Pvar2 = sort(tapply(taxa_sums(pvar.f.f.bac.T1F), tax_table(pvar.f.f.bac.T1F)[, "Phylum"], sum), TRUE)[1:11]

    
    #Cut down the  data to only the top 10 fam
      Pvar.u.f.bac.T1F.top10P = subset_taxa(pvar.f.f.bac.T1F, Family %in% names(top10P.names.Pvar))
      
      #pull out of phyloseq     
        Pvar.u.f.bac.T1F.top10P.OTU = as(otu_table(Pvar.u.f.bac.T1F.top10P), "matrix")
        # write.csv(Pvar.u.f.bac.T1F.top10P.OTU, "Pvar.u.f.bac.T1F.top10P.OTU.csv")
       # transpose if necessary
          # if(taxa_are_rows(Pvar.u.f.bac.T1F.top5P)){OTUfam.v <- t(Pvar.u.f.bac.T1F.top5P.OTU)}
          # Coerce to data.frame
            Pvar.u.f.bac.T1F.top10P.OTU.df = as.data.frame(Pvar.u.f.bac.T1F.top10P.OTU) 
      #rename Fam with taxa
            
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00001"] <- "Endozoicomonadaceae"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00011"] <- "Amoebophilaceae"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00012"] <- "Bacteria_unclassified_Otu00012"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00093"] <- "Nostocaceae"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00025"] <- "Rhodobacteraceae"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00027"] <- "Pirellulaceae"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00206"] <- "Flavobacteriaceae"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00030"] <- "Cyclobacteriaceae"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00053"] <- "Kiloniellaceae"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00079"] <- "uncultured_Otu00079"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00872"] <- "Remove1"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00270"] <- "Remove2"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00533"] <- "Remove3"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00563"] <- "Remove4"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu01647"] <- "Remove5"
            names(Pvar.u.f.bac.T1F.top10P.OTU.df)[names(Pvar.u.f.bac.T1F.top10P.OTU.df) == "Otu00062"] <- "Remove6"

Pvar.u.f.bac.T1F.top10P.OTU.df<-subset(Pvar.u.f.bac.T1F.top10P.OTU.df, select=-c(Remove1,Remove2,Remove3, Remove4, Remove5, Remove6)) #because fam names of 'uncultured' don't differentiate OTU, manually see which is it, and remove the rest. 
            
            
    #Add metadata
      Pvar.u.f.bac.T1F.top10P.sd <- as(sample_data(Pvar.u.f.bac.T1F.top10P), "data.frame") #create sample data frame to view
         Pvar.u.f.bac.T1F.top10P.sd<-Pvar.u.f.bac.T1F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    Pvar.16s.fam.top10<- merge(Pvar.u.f.bac.T1F.top10P.OTU.df,Pvar.u.f.bac.T1F.top10P.sd, by="row.names") #merge by rownames
       Pvar.16s.fam.top10.m <- melt(Pvar.16s.fam.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(Pvar.16s.fam.top10.m)[names(Pvar.16s.fam.top10.m) == "value"] <- "relabund"
            names(Pvar.16s.fam.top10.m)[names(Pvar.16s.fam.top10.m) == "variable"] <- "family"

#Means
Pvar.16s.fam.top10.m_mean <- ddply(Pvar.16s.fam.top10.m, c("Treatment", "Clade", "Time.Point","family"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pvar.16s.fam.top10.m_mean


Pvar.16s.fam.top10.m_mean$Code<-paste(Pvar.16s.fam.top10.m_mean$Treatment, Pvar.16s.fam.top10.m_mean$Time.Point,Pvar.16s.fam.top10.m_mean$Clade)

#Pvar plots
#plot Pvar
Pvar.16s.fam.top10.m_mean_plot<-ggplot(data=Pvar.16s.fam.top10.m_mean, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("Pvar top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar.16s.fam.top10.m_mean_plot
Pvar.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)
    

#means by T0 only

Pvar.16s.fam.top10.m_T0<-subset(Pvar.16s.fam.top10.m, Time.Point=="T0")
Pvar.16s.fam.top10.m_T0amb<-subset(Pvar.16s.fam.top10.m_T0, Treatment=="Ambient")

#Means
Pvar.16s.fam.top10.m.T0_mean <- ddply(Pvar.16s.fam.top10.m_T0amb, c("family"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pvar.16s.fam.top10.m.T0_mean


#### means ambient only Bac.seqFam_stats.Pvar.Ambient ####

#pull out of phyloseq     
        Bac.seqFam_stats.Pvar.Ambient.df = as(otu_table(Bac.seqFam_stats.Pvar.Ambient), "matrix")
        # write.csv(Pvar.u.f.bac.T1F.top10P.OTU, "Pvar.u.f.bac.T1F.top10P.OTU.csv")
       # transpose if necessary
          # if(taxa_are_rows(Pvar.u.f.bac.T1F.top5P)){OTUfam.v <- t(Pvar.u.f.bac.T1F.top5P.OTU)}
          # Coerce to data.frame
            Bac.seqFam_stats.Pvar.Ambient.df = as.data.frame(Bac.seqFam_stats.Pvar.Ambient.df) 
 
#Add metadata
      Pvar.u.f.bac.ambient.sd <- as(sample_data(Bac.seqFam_stats.Pvar.Ambient), "data.frame") #create sample data frame to view
         Pvar.u.f.bac.ambient.sd<-Pvar.u.f.bac.ambient.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    Pvar.16s.fam.AMB<- merge(Bac.seqFam_stats.Pvar.Ambient.df,Pvar.u.f.bac.ambient.sd, by="row.names") #merge by rownames
       Pvar.16s.fam.AMB.m <- melt(Pvar.16s.fam.AMB, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(Pvar.16s.fam.AMB.m)[names(Pvar.16s.fam.AMB.m) == "value"] <- "relabund"
            names(Pvar.16s.fam.AMB.m)[names(Pvar.16s.fam.AMB.m) == "variable"] <- "family"

#Means
Pvar.16s.fam.AMB.m_mean <- ddply(Pvar.16s.fam.AMB.m, c("family"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pvar.16s.fam.AMB.m_mean               
     write.csv(Pvar.16s.fam.AMB.m_mean, "pvar_Amb_16s_means.csv")       
            
            
            
            
            
            
            
####### pacuta  ####

#Pacu
pacu.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Pocillopora_acuta")
pacu.f.bac.T1F = prune_taxa(taxa_sums(pacu.f.bac.T1F) > 0, pacu.f.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
pacu.f.bac.T1 <- subset_samples(pacu.f.bac.T1F, Time.Point == "T1")
  pacu.f.bac.T1 = prune_taxa(taxa_sums(pacu.f.bac.T1) > 0, pacu.f.bac.T1) #this removes any OTU with 0s

pacu.f.bac.TF <- subset_samples(pacu.f.bac.T1F, Time.Point == "TF")
  pacu.f.bac.TF = prune_taxa(taxa_sums(pacu.f.bac.TF) > 0, pacu.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make wunifrac curtis data matrix -T1 and TF
  u.pacu.f.T1F <- phyloseq::distance(pacu.f.bac.T1F, method = "wunifrac")
     pacu.f.samp.df.T1F <- as(sample_data(pacu.f.bac.T1F), "data.frame") #sample df
     set.seed(30)
     adonis(u.pacu.f.T1F ~ Treatment*Time.Point, data = pacu.f.samp.df.T1F) #adonis test no clade
     set.seed(30)
     adonis(u.pacu.f.T1F ~ Clade*Treatment*Time.Point, data = pacu.f.samp.df.T1F)#adonis test with clade
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pacu.f.T1F, pacu.f.samp.df.T1F$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pacu.f.T1F, pacu.f.samp.df.T1F$Treatment, type = "centroid")) #put in distance matrix
    
#T1
  u.PacuT1 <- phyloseq::distance(pacu.f.bac.T1, method = "wunifrac") 
      pacu.f.samp.df.T1 <- as(sample_data(pacu.f.bac.T1), "data.frame") 
      set.seed(30)
      adonis(u.PacuT1 ~ Treatment, data = pacu.f.samp.df.T1)  
      set.seed(30)
      adonis(u.PacuT1 ~ Clade*Treatment, data = pacu.f.samp.df.T1)  
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.PacuT1, pacu.f.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.PacuT1, pacu.f.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix

    
#TF
 
  u.pacu.f.TF <- phyloseq::distance(pacu.f.bac.TF, method = "wunifrac")
     pacu.f.samp.df.TF <- as(sample_data(pacu.f.bac.TF), "data.frame") 
      set.seed(30)
      adonis(u.pacu.f.TF ~Treatment, data = pacu.f.samp.df.TF)
      set.seed(30)
      adonis(u.pacu.f.TF ~ Clade*Treatment, data = pacu.f.samp.df.TF)
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pacu.f.TF, pacu.f.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pacu.f.TF, pacu.f.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix

    
  #try plotting only the top 10 FAMs   ####  
    #Sort the Phyla by abundance and pick the top 5
    top10P.names.Pacu = sort(tapply(taxa_sums(pacu.f.bac.T1F), tax_table(pacu.f.bac.T1F)[, "Family"], sum), TRUE)[1:10]
        top10P.names.Pacu.phy = sort(tapply(taxa_sums(pacu.f.bac.T1F), tax_table(pacu.f.bac.T1F)[, "Phylum"], sum), TRUE)[1:10]

      #Cut down the  data to only the top 10 Phyla
      Pacu.u.f.bac.T1F.top10P = subset_taxa(pacu.f.bac.T1F, Family %in% names(top10P.names.Pacu))
      
      #pull out of phyloseq     
        Pacu.u.f.bac.T1F.top10P.OTU = as(otu_table(Pacu.u.f.bac.T1F.top10P), "matrix")
       # transpose if necessary
          # if(taxa_are_rows(Pacu.u.f.bac.T1F.top5P)){OTUfam.v <- t(Pacu.u.f.bac.T1F.top5P.OTU)}
          # Coerce to data.frame
            Pacu.u.f.bac.T1F.top10P.OTU.df = as.data.frame(Pacu.u.f.bac.T1F.top10P.OTU) 
      #rename Fam with taxa
            
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00068"] <- "Alphaproteobacteria_unclassified"
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00011"] <- "Amoebophilaceae"
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00206"] <- "Flavobacteriaceae"
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00156"] <- "Saprospiraceae"

            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00055"] <- "Pseudoalteromonadaceae"
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00062"] <- "Alteromonadaceae"
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00001"] <- "Endozoicomonadaceae"
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00226"] <- "Gammaproteobacteria_unclassified"
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00030"] <- "Cyclobacteriaceae"
            names(Pacu.u.f.bac.T1F.top10P.OTU.df)[names(Pacu.u.f.bac.T1F.top10P.OTU.df) == "Otu00025"] <- "Rhodobacteraceae"
         
            
    #Add metadata
      Pacu.u.f.bac.T1F.top10P.sd <- as(sample_data(Pacu.u.f.bac.T1F.top10P), "data.frame") #create sample data frame to view
         Pacu.u.f.bac.T1F.top10P.sd<-Pacu.u.f.bac.T1F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    Pacu.16s.fam.top10<- merge(Pacu.u.f.bac.T1F.top10P.OTU.df,Pacu.u.f.bac.T1F.top10P.sd, by="row.names") #merge by rownames
       Pacu.16s.fam.top10.m <- melt(Pacu.16s.fam.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(Pacu.16s.fam.top10.m)[names(Pacu.16s.fam.top10.m) == "value"] <- "relabund"
            names(Pacu.16s.fam.top10.m)[names(Pacu.16s.fam.top10.m) == "variable"] <- "family"

#Means
Pacu.16s.fam.top10.m_mean <- ddply(Pacu.16s.fam.top10.m, c("Treatment", "Clade", "Time.Point","family"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pacu.16s.fam.top10.m_mean

Pacu.16s.fam.top10.m_mean$Code<-paste(Pacu.16s.fam.top10.m_mean$Treatment, Pacu.16s.fam.top10.m_mean$Time.Point,Pacu.16s.fam.top10.m_mean$Clade)

#Pacu plots
#plot Pacu
Pacu.16s.fam.top10.m_mean_plot<-ggplot(data=Pacu.16s.fam.top10.m_mean, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("Pacu top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu.16s.fam.top10.m_mean_plot
Pacu.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)
       
    
    
    
    
    
    
    
    
    
```
# Family unifrac stats and plots with T0 Bac.seqFam_stats
 So far only top 10 is updated 
 
First make duplicates of the T0 High and Ambs to all parents get their Ambinet starts
```{r}   
#### Create a duplicate of T0s for the other A and H treatments for comparision: df coralDIV2_rare
#not
control.fam.T0_Amb = subset_samples(Bac.seqFam_stats, Time.Point == "T0"&Treatment == "Ambient")  #copy

#tax table
tax  = tax_table(control.fam.T0_Amb)

  # sample data
  sd = sample_data(control.fam.T0_Amb)
  rownames(sd) <- paste0(rownames(sd), "_dup")
  # otu_table
  OTU = otu_table(control.fam.T0_Amb)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  # sd$Treatment <- as.character(sd$Treatment)
  # sd$ID <- as.character(sd$ID)
  sd$Treatment <- "High"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
  sd$Treatment <- as.factor(sd$Treatment)
  # sd$ID <- as.factor(sd$ID)

  Amb.fam.2<-phyloseq(OTU, sd, tax)
#sample.dataAmb2 <- as(sample_data(Amb2), "data.frame") #create sample data frame to view

##High
  control.fam.T0_High = subset_samples(Bac.seqFam_stats, Time.Point == "T0"&Treatment == "High")  #copy

#tax table
tax  = tax_table(control.fam.T0_High)

  # sample data
  sd = sample_data(control.fam.T0_High)
  rownames(sd) <- paste0(rownames(sd), "_dup")
# otu_table
OTU = otu_table(control.fam.T0_High)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  sd$Treatment <- as.character(sd$Treatment)
  # sd$ID <- as.character(sd$ID)
  sd$Treatment <- "Ambient"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
  sd$Treatment <- as.factor(sd$Treatment)
  sd$ID <- as.factor(sd$ID)

  High.fam.2<-phyloseq(OTU, sd, tax)
  #sample.dataHigh2 <- as(sample_data(High2), "data.frame") #create sample data frame to view

#Merge control dups H and A
Controldups = merge_phyloseq(Amb.fam.2,High.fam.2)
Controldups
    #sample.dataControldups2 <- as(sample_data(Controldups), "data.frame") #create sample data frame to view

#merg them to the original df:

Bac.seqFam_stats.T01F = merge_phyloseq(Bac.seqFam_stats,Controldups)
Bac.seqFam_stats.T01F
Bac.seqFam_stats.T01F.sd <- as(sample_data(Bac.seqFam_stats.T01f), "data.frame") #create sample data frame to view
Bac.seqFam_stats.T01F.sd
```

#Family top ten
 http://colorschemedesigner.com/csd-3.5/ THIS IS COOL BUT ONLY 4
```{r} 
 



############################### Family unifrac stats ####
#make unifrac matrix
Bac.fam.T1F <- phyloseq::distance(Bac.seqFam_stats.T1F, method = "wunifrac") #dist matrix
Fam.samp.df <- as(sample_data(Bac.seqFam_stats.T1F), "data.frame") #sample df
  #adonis 
    set.seed(30)
      adonis(Bac.fam.T1F ~ Species*Treatment*Time.Point, data = Fam.samp.df)
    set.seed(30)    
      adonis(Fam.16s.all ~ Species*Treatment*Time.Point*Clade, data = Fam.samp.df)
  # Homogeneity of dispersion test  
    set.seed(30)    
      permutest(betadisper(Bac.fam.T1F, Fam.samp.df$Clade, type = "centroid")) #put in distance matrix

############# mcap ####not updated

#Mcap
mcap.u.f.bac.T1F <- subset_samples(Bac.seqFam_stats.T1F, Species == "Montipora_capitata")
mcap.u.f.bac.T1F = prune_taxa(taxa_sums(mcap.u.f.bac.T1F) > 0, mcap.u.f.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
mcap.u.f.bac.T1 <- subset_samples(mcap.u.f.bac.T1F, Time.Point == "T1")
  mcap.u.f.bac.T1 = prune_taxa(taxa_sums(mcap.u.f.bac.T1) > 0, mcap.u.f.bac.T1) #this removes any OTU with 0s
mcap.u.f.bac.TF <- subset_samples(mcap.u.f.bac.T1F, Time.Point == "TF")
  mcap.u.f.bac.TF = prune_taxa(taxa_sums(mcap.u.f.bac.TF) > 0, mcap.u.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  set.seed(30)
  bc.mcap.u.f.T1F <- phyloseq::distance(mcap.u.f.bac.T1F, method = "wunifrac")
     mcap.u.f.samp.df.T1F <- as(sample_data(mcap.u.f.bac.T1F), "data.frame") #sample df
       set.seed(30)
    adonis(bc.mcap.u.f.T1F ~ Treatment*Time.Point, data = mcap.u.f.samp.df.T1F) #adonis test no clade
      set.seed(30)
    adonis(bc.mcap.u.f.T1F ~ Clade*Treatment*Time.Point, data = mcap.u.f.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(mcap.u.f.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
      # Homogeneity of dispersion test  
set.seed(30)
    permutest(betadisper(bc.mcap.u.f.T1F, mcap.u.f.samp.df.T1F$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(bc.mcap.u.f.T1F, mcap.u.f.samp.df.T1F$Treatment, type = "centroid")) #put in distance matrix
    
#T1
    
  u.mcapT1 <- phyloseq::distance(mcap.u.f.bac.T1, method = "wunifrac") 
      mcap.u.f.samp.df.T1 <- as(sample_data(mcap.u.f.bac.T1), "data.frame") 
          set.seed(30)
    adonis(u.mcapT1 ~ Treatment, data = mcap.u.f.samp.df.T1)  
     set.seed(30)
    adonis(u.mcapT1 ~ Clade*Treatment, data = mcap.u.f.samp.df.T1)  
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.mcapT1, mcap.u.f.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.mcapT1, mcap.u.f.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix

#TF
  u.mcap.u.f.TF <- phyloseq::distance(mcap.u.f.bac.TF, method = "wunifrac")
     mcap.u.f.samp.df.TF <- as(sample_data(mcap.u.f.bac.TF), "data.frame")
    set.seed(30)
    adonis(u.mcap.u.f.TF ~ Treatment, data = mcap.u.f.samp.df.TF)
    set.seed(30)
    adonis(u.mcap.u.f.TF ~ Clade*Treatment, data = mcap.u.f.samp.df.TF)
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.mcap.u.f.TF, mcap.u.f.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.mcap.u.f.TF, mcap.u.f.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix
    
    
# Mcap top 10 T01F ####
 ##try plotting only the top 10 FAMs    Bac.seqFam_stats has T0 
    Bac.seqFam_stats.T01F.mcap<-subset_samples(Bac.seqFam_stats.T01F, Species=="Montipora_capitata")
    #Sort the Phyla by abundance and pick the top 5
    top10P.names.mcap.T01f = sort(tapply(taxa_sums(Bac.seqFam_stats.T01F.mcap), tax_table(Bac.seqFam_stats.T01F.mcap)[, "Family"], sum), TRUE)[1:10]
      #Cut down the  data to only the top 10 Phyla
      mcap.u.f.bac.T01F.top10P = subset_taxa(Bac.seqFam_stats.T01F.mcap, Family %in% names(top10P.names.mcap.T01f))
      
      #pull out of phyloseq     
        mcap.u.f.bac.T01F.top10P.OTU = as(otu_table(mcap.u.f.bac.T01F.top10P), "matrix")
       # transpose if necessary
          # if(taxa_are_rows(mcap.u.f.bac.T1F.top5P)){OTUfam.v <- t(mcap.u.f.bac.T1F.top5P.OTU)}
          # Coerce to data.frame
            mcap.u.f.bac.T01F.top10P.OTU.df = as.data.frame(mcap.u.f.bac.T01F.top10P.OTU) 
      #rename Fam with taxa
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00104"] <- "Hyphomonadaceae"
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00004"] <- "P30B-42"
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00058"] <- "Chlorobiaceae"
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00030"] <- "Cyclobacteriaceae"
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00011"] <- "Amoebophilaceae"#only one diff from non T0 set
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00206"] <- "Flavobacteriaceae"
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00015"] <- "Nodosilineaceae"
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00001"] <- "Endozoicomonadaceae"
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00226"] <- "Gammaproteobacteria_unclassified"
            names(mcap.u.f.bac.T01F.top10P.OTU.df)[names(mcap.u.f.bac.T01F.top10P.OTU.df) == "Otu00025"] <- "Rhodobacteraceae"
    #Add metadata
      mcap.u.f.bac.T01F.top10P.sd <- as(sample_data(mcap.u.f.bac.T01F.top10P), "data.frame") #create sample data frame to view
         mcap.u.f.bac.T01F.top10P.sd<-mcap.u.f.bac.T01F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    mcap.16s.fam.T01f.top10<- merge(mcap.u.f.bac.T01F.top10P.OTU.df,mcap.u.f.bac.T01F.top10P.sd, by="row.names") #merge by rownames
       mcap.16s.fam.T01f.top10.m <- melt(mcap.16s.fam.T01f.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(mcap.16s.fam.T01f.top10.m)[names(mcap.16s.fam.T01f.top10.m) == "value"] <- "relabund"
            names(mcap.16s.fam.T01f.top10.m)[names(mcap.16s.fam.T01f.top10.m) == "variable"] <- "family"

#Means
mcap.16s.fam.T01F.top10.m_mean <- ddply(mcap.16s.fam.T01f.top10.m, c("Treatment", "Clade", "Time.Point","family","Species"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
mcap.16s.fam.T01F.top10.m_mean

mcap.16s.fam.T01F.top10.m_mean$Code<-paste(mcap.16s.fam.T01F.top10.m_mean$Treatment, mcap.16s.fam.T01F.top10.m_mean$Time.Point,mcap.16s.fam.T01F.top10.m_mean$Clade)

#mcap plots
#plot Mcap
mcap.16s.fam.top10.m_mean_plot<-ggplot(data=mcap.16s.fam.T01F.top10.m_mean, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("Mcap top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));mcap.16s.fam.top10.m_mean_plot
mcap.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)

#mcap no clade
mcap.16s.fam.T01F.top10.m_mean.noC <- ddply(mcap.16s.fam.T01f.top10.m, c("Treatment", "Time.Point","family","Species"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
mcap.16s.fam.T01F.top10.m_mean.noC

mcap.16s.fam.T01F.top10.m_mean.noC$Code<-paste(mcap.16s.fam.T01F.top10.m_mean.noC$Treatment, mcap.16s.fam.T01F.top10.m_mean.noC$Time.Point,mcap.16s.fam.T01F.top10.m_mean.noC$Clade)

#mcap plots
#plot Mcap
mcap.16s.fam.top10.m_mean_plot<-ggplot(data=mcap.16s.fam.T01F.top10.m_mean.noC, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("Mcap top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));mcap.16s.fam.top10.m_mean_plot
mcap.16s.fam.top10.m_mean_plot+facet_wrap(~Treatment)
    
  ############ pcomp #####
#Pcomp
pcomp.f.bac.T01F <- subset_samples(Bac.seqFam_stats.T01F, Species == "Porites_compressa")
#pcomp.f.bac.T01F = prune_taxa(taxa_sums(pcomp.f.bac.T01F) > 0, pcomp.f.bac.T01F) #this removes any OTU with 0s

#make dfs you need 
pcomp.f.bac.T1 <- subset_samples(pcomp.f.bac.T01F, Time.Point == "T1")
 # pcomp.f.bac.T1 = prune_taxa(taxa_sums(pcomp.f.bac.T1) > 0, pcomp.f.bac.T1) #this removes any OTU with 0s

pcomp.f.bac.TF <- subset_samples(pcomp.f.bac.T01F, Time.Point == "TF")
 # pcomp.f.bac.TF = prune_taxa(taxa_sums(pcomp.f.bac.TF) > 0, pcomp.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make wunifrac curtis data matrix -T1 and TF
  u.pcomp.f.T01F <- phyloseq::distance(pcomp.f.bac.T01F, method = "wunifrac")
     pcomp.f.samp.df.T01F <- as(sample_data(pcomp.f.bac.T01F), "data.frame") #sample df
      set.seed(30)
    adonis(u.pcomp.f.T01F ~ Treatment*Time.Point, data = pcomp.f.samp.df.T01F) #adonis test no clade
   set.seed(30)
    adonis(u.pcomp.f.T01F ~ Clade*Treatment*Time.Point, data = pcomp.f.samp.df.T01F)#adonis test with clade
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pcomp.f.T01F, pcomp.f.samp.df.T01F$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pcomp.f.T01F, pcomp.f.samp.df.T01F$Treatment, type = "centroid")) #put in distance matrix
    
#T1
  u.PcompT1 <- phyloseq::distance(pcomp.f.bac.T1, method = "wunifrac") 
      pcomp.f.samp.df.T1 <- as(sample_data(pcomp.f.bac.T1), "data.frame") 
   set.seed(30)
   adonis(u.PcompT1 ~ Treatment, data = pcomp.f.samp.df.T1)  
    set.seed(30)
   adonis(u.PcompT1 ~ Clade*Treatment, data = pcomp.f.samp.df.T1)  
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.PcompT1, pcomp.f.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.PcompT1, pcomp.f.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix

#TF
u.pcomp.f.TF <- phyloseq::distance(pcomp.f.bac.TF, method = "wunifrac")
     pcomp.f.samp.df.TF <- as(sample_data(pcomp.f.bac.TF), "data.frame") 
      set.seed(30)
    adonis(u.pcomp.f.TF ~ Treatment, data = pcomp.f.samp.df.TF)
     set.seed(30)
    adonis(u.pcomp.f.TF ~ Clade*Treatment, data = pcomp.f.samp.df.TF)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pcomp.f.TF, pcomp.f.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pcomp.f.TF, pcomp.f.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix

 #try plotting only the top 10 FAMs     ####
        pcomp.f.bac.T01F<-subset_samples(Bac.seqFam_stats.T01F, Species=="Porites_compressa")

    #Sort the Phyla by abundance and pick the top 5
    top10P.names.pcomp = sort(tapply(taxa_sums(pcomp.f.bac.T01F), tax_table(pcomp.f.bac.T01F)[, "Family"], sum), TRUE)[1:10]
      #Cut down the  data to only the top 10 Phyla
      pcomp.u.f.bac.T01F.top10P = subset_taxa(pcomp.f.bac.T01F, Family %in% names(top10P.names.pcomp))
      
      #pull out of phyloseq     
        pcomp.u.f.bac.T01F.top10P.OTU = as(otu_table(pcomp.u.f.bac.T01F.top10P), "matrix")
       # transpose if necessary
          # if(taxa_are_rows(pcomp.u.f.bac.T01F.top5P)){OTUfam.v <- t(pcomp.u.f.bac.T01F.top5P.OTU)}
          # Coerce to data.frame
            pcomp.u.f.bac.T01F.top10P.OTU.df = as.data.frame(pcomp.u.f.bac.T01F.top10P.OTU) 
      #rename Fam with taxa, btw all the same as without T0
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00068"] <- "Alphaproteobacteria_unclassified"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00274"] <- "Pirellulaceae"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00374"] <- "Phycispharaceae"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00011"] <- "Amoebophilaceae"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00206"] <- "Flavobacteriaceae"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00156"] <- "Saprospiraceae"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00146"] <- "Paraspirulinaceae"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00001"] <- "Endozoicomonadaceae"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00116"] <- "Francisellaceae"
            names(pcomp.u.f.bac.T01F.top10P.OTU.df)[names(pcomp.u.f.bac.T01F.top10P.OTU.df) == "Otu00025"] <- "Rhodobacteraceae"
    #Add metadata
      pcomp.u.f.bac.T01F.top10P.sd <- as(sample_data(pcomp.u.f.bac.T01F.top10P), "data.frame") #create sample data frame to view
         pcomp.u.f.bac.T01F.top10P.sd<-pcomp.u.f.bac.T01F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    pcomp.16s.fam.top10<- merge(pcomp.u.f.bac.T01F.top10P.OTU.df,pcomp.u.f.bac.T01F.top10P.sd, by="row.names") #merge by rownames
       pcomp.16s.fam.top10.m <- melt(pcomp.16s.fam.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(pcomp.16s.fam.top10.m)[names(pcomp.16s.fam.top10.m) == "value"] <- "relabund"
            names(pcomp.16s.fam.top10.m)[names(pcomp.16s.fam.top10.m) == "variable"] <- "family"

#Means
pcomp.16s.fam.top10.m_mean <- ddply(pcomp.16s.fam.top10.m, c("Treatment", "Clade", "Time.Point","family","Species"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
pcomp.16s.fam.top10.m_mean

pcomp.16s.fam.top10.m_mean$Code<-paste(pcomp.16s.fam.top10.m_mean$Treatment, pcomp.16s.fam.top10.m_mean$Time.Point,pcomp.16s.fam.top10.m_mean$Clade)

#pcomp plots
#plot pcomp
pcomp.16s.fam.top10.m_mean_plot<-ggplot(data=pcomp.16s.fam.top10.m_mean, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("pcomp top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));pcomp.16s.fam.top10.m_mean_plot
pcomp.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)
    
    #remove endo pcomp.16s.fam.top10.m_mean
pcomp.16s.fam.top10.m_mean.NoEndo<-subset(pcomp.16s.fam.top10.m_mean, !family=="Endozoicomonadaceae")
    
    #plot pcomp
pcomp.16s.fam.top10.m_mean_plot<-ggplot(data=pcomp.16s.fam.top10.m_mean.NoEndo, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("pcomp top 10 family No Endo")+
  theme(plot.title = element_text(size=20, face = "italic"));pcomp.16s.fam.top10.m_mean_plot
pcomp.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)


#Means no phylotype
pcomp.16s.fam.top10.m_mean.NoC <- ddply(pcomp.16s.fam.top10.m, c("Treatment", "Time.Point","family","Species"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
pcomp.16s.fam.top10.m_mean.NoC

pcomp.16s.fam.top10.m_mean.NoC$Code<-paste(pcomp.16s.fam.top10.m_mean.NoC$Treatment, pcomp.16s.fam.top10.m_mean.NoC$Time.Point,pcomp.16s.fam.top10.m_mean.NoC$Clade)

#pcomp plots
#plot pcomp
pcomp.16s.fam.top10.m_mean_plot<-ggplot(data=pcomp.16s.fam.top10.m_mean.NoC, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("pcomp top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));pcomp.16s.fam.top10.m_mean_plot
pcomp.16s.fam.top10.m_mean_plot+facet_wrap(~Treatment)
    

    #remove endo 
pcomp.16s.fam.top10.m_mean.NoC.noEndo<-subset(pcomp.16s.fam.top10.m_mean.NoC, !family=="Endozoicomonadaceae")
    
    #plot pcomp
pcomp.16s.fam.top10.m_mean_plot<-ggplot(data=pcomp.16s.fam.top10.m_mean.NoC.noEndo, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("pcomp top 10 family No Endo")+
  theme(plot.title = element_text(size=20, face = "italic"));pcomp.16s.fam.top10.m_mean_plot
pcomp.16s.fam.top10.m_mean_plot+facet_wrap(~Treatment)
    
#######pvar ####
    #Pvar
pvar.f.f.bac.T01F <- subset_samples(Bac.seqFam_stats.T01F, Species == "Pavona_varians")
#pvar.f.f.bac.T01F = prune_taxa(taxa_sums(pvar.f.f.bac.T01F) > 0, pvar.f.f.bac.T01F) #this removes any OTU with 0s
 
#make dfs you need 
pvar.f.f.bac.T1 <- subset_samples(pvar.f.f.bac.T01F, Time.Point == "T1")
 # pvar.f.f.bac.T1 = prune_taxa(taxa_sums(pvar.f.f.bac.T1) > 0, pvar.f.f.bac.T1) #this removes any OTU with 0s

pvar.f.f.bac.TF <- subset_samples(pvar.f.f.bac.T01F, Time.Point == "TF")
#  pvar.f.f.bac.TF = prune_taxa(taxa_sums(pvar.f.f.bac.TF) > 0, pvar.f.f.bac.TF) #this removes any OTU with 0s
 
#T1TF
#make wunifrac curtis data matrix -T1 and TF
                set.seed(30)

  u.pvar.f.f.T01F <- phyloseq::distance(pvar.f.f.bac.T01F, method = "wunifrac")
     pvar.f.f.samp.df.T01F <- as(sample_data(pvar.f.f.bac.T01F), "data.frame") #sample df
     set.seed(30)
adonis(u.pvar.f.f.T01F ~ Treatment*Time.Point, data = pvar.f.f.samp.df.T01F) #adonis test no clade
    set.seed(30)
    adonis(u.pvar.f.f.T01F ~ Clade*Treatment*Time.Point, data = pvar.f.f.samp.df.T01F)#adonis test with clade
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pvar.f.f.T01F, pvar.f.f.samp.df.T01F$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pvar.f.f.T01F, pvar.f.f.samp.df.T01F$Treatment, type = "centroid")) #put in distance matrix
    
#T1
                 

  u.PvarT1 <- phyloseq::distance(pvar.f.f.bac.T1, method = "wunifrac") 
      pvar.f.f.samp.df.T1 <- as(sample_data(pvar.f.f.bac.T1), "data.frame") 
   set.seed(30)
   adonis(u.PvarT1 ~ Treatment, data = pvar.f.f.samp.df.T1) 
     set.seed(30)
     adonis(u.PvarT1 ~ Clade*Treatment, data = pvar.f.f.samp.df.T1)  

# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.PvarT1, pvar.f.f.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.PvarT1, pvar.f.f.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix

#TF
  u.pvar.f.f.TF <- phyloseq::distance(pvar.f.f.bac.TF, method = "wunifrac")
     pvar.f.f.samp.df.TF <- as(sample_data(pvar.f.f.bac.TF), "data.frame") 
      set.seed(30)
      adonis(u.pvar.f.f.TF ~ Treatment, data = pvar.f.f.samp.df.TF)
        set.seed(30)
        adonis(u.pvar.f.f.TF ~ Clade*Treatment, data = pvar.f.f.samp.df.TF)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pvar.f.f.TF, pvar.f.f.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pvar.f.f.TF, pvar.f.f.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix

 #try plotting only the top 10 FAMs  ####   
            pvar.f.f.bac.T01F<-subset_samples(Bac.seqFam_stats.T01F, Species=="Pavona_varians")

    #Sort the Phyla by abundance and pick the top 5
    top10P.names.Pvar = sort(tapply(taxa_sums(pvar.f.f.bac.T01F), tax_table(pvar.f.f.bac.T01F)[, "Family"], sum), TRUE)[1:10]
      #Cut down the  data to only the top 10 Phyla
      Pvar.u.f.bac.T01F.top10P = subset_taxa(pvar.f.f.bac.T01F, Family %in% names(top10P.names.Pvar))
      
      #pull out of phyloseq     
        Pvar.u.f.bac.T01F.top10P.OTU = as(otu_table(Pvar.u.f.bac.T01F.top10P), "matrix")
       # transpose if necessary
          # if(taxa_are_rows(Pvar.u.f.bac.T01F.top5P)){OTUfam.v <- t(Pvar.u.f.bac.T01F.top5P.OTU)}
          # Coerce to data.frame
            Pvar.u.f.bac.T01F.top10P.OTU.df = as.data.frame(Pvar.u.f.bac.T01F.top10P.OTU) 
      #rename Fam with taxa, all same as T1F
            
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00001"] <- "Endozoicomonadaceae"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00011"] <- "Amoebophilaceae"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00012"] <- "Bacteria_unclassified_Otu00012"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00025"] <- "Rhodobacteraceae"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00027"] <- "Pirellulaceae"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00206"] <- "Flavobacteriaceae"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00030"] <- "Cyclobacteriaceae"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00053"] <- "Kiloniellaceae"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00079"] <- "uncultured_Otu00079"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00093"] <- "Nostocaceae"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00270"] <- "Remove5"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00533"] <- "Remove4"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00563"] <- "Remove3"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu01647"] <- "Remove2"
            names(Pvar.u.f.bac.T01F.top10P.OTU.df)[names(Pvar.u.f.bac.T01F.top10P.OTU.df) == "Otu00872"] <- "Remove1"

            Pvar.u.f.bac.T01F.top10P.OTU.df<-subset(Pvar.u.f.bac.T01F.top10P.OTU.df, select=-c(Remove1,Remove2,Remove3, Remove4, Remove5)) #because fam names of 'uncultured' don't differentiate OTU, manually see which is it, and remove the rest. 

            
    #Add metadata
      Pvar.u.f.bac.T01F.top10P.sd <- as(sample_data(Pvar.u.f.bac.T01F.top10P), "data.frame") #create sample data frame to view
         Pvar.u.f.bac.T01F.top10P.sd<-Pvar.u.f.bac.T01F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    Pvar.16s.fam.top10<- merge(Pvar.u.f.bac.T01F.top10P.OTU.df, Pvar.u.f.bac.T01F.top10P.sd, by="row.names") #merge by rownames
       Pvar.16s.fam.top10.m <- melt(Pvar.16s.fam.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(Pvar.16s.fam.top10.m)[names(Pvar.16s.fam.top10.m) == "value"] <- "relabund"
            names(Pvar.16s.fam.top10.m)[names(Pvar.16s.fam.top10.m) == "variable"] <- "family"

#Means
Pvar.16s.fam.top10.m_mean <- ddply(Pvar.16s.fam.top10.m, c("Treatment", "Clade", "Time.Point","family","Species"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pvar.16s.fam.top10.m_mean






#Means no phylotype
pvar.16s.fam.top10.m_mean.NoC <- ddply(Pvar.16s.fam.top10.m, c("Treatment", "Time.Point","family","Species"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
pvar.16s.fam.top10.m_mean.NoC


#pvar plots
#plot pvar
pvar.16s.fam.top10.m_mean_plot<-ggplot(data=pvar.16s.fam.top10.m_mean.NoC, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("pvar top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));pvar.16s.fam.top10.m_mean_plot
pvar.16s.fam.top10.m_mean_plot+facet_wrap(~Treatment)

#Pvar plots
#plot Pvar
Pvar.16s.fam.top10.m_mean_plot<-ggplot(data=Pvar.16s.fam.top10.m_mean, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("Pvar top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar.16s.fam.top10.m_mean_plot
Pvar.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)
    

    
####### pacuta  ####

#Pacu
pacu.f.bac.T01F <- subset_samples(Bac.seqFam_stats.T01F, Species == "Pocillopora_acuta")
pacu.f.bac.T01F = prune_taxa(taxa_sums(pacu.f.bac.T01F) > 0, pacu.f.bac.T01F) #this removes any OTU with 0s

#make dfs you need 
pacu.f.bac.T1 <- subset_samples(pacu.f.bac.T01F, Time.Point == "T1")
  pacu.f.bac.T1 = prune_taxa(taxa_sums(pacu.f.bac.T1) > 0, pacu.f.bac.T1) #this removes any OTU with 0s

pacu.f.bac.TF <- subset_samples(pacu.f.bac.T01F, Time.Point == "TF")
  pacu.f.bac.TF = prune_taxa(taxa_sums(pacu.f.bac.TF) > 0, pacu.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make wunifrac curtis data matrix -T1 and TF
  u.pacu.f.T01F <- phyloseq::distance(pacu.f.bac.T01F, method = "wunifrac")
     pacu.f.samp.df.T01F <- as(sample_data(pacu.f.bac.T01F), "data.frame") #sample df
     set.seed(30)
     adonis(u.pacu.f.T01F ~ Treatment*Time.Point, data = pacu.f.samp.df.T01F) #adonis test no clade
     set.seed(30)
     adonis(u.pacu.f.T01F ~ Clade*Treatment*Time.Point, data = pacu.f.samp.df.T01F)#adonis test with clade
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pacu.f.T01F, pacu.f.samp.df.T01F$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pacu.f.T01F, pacu.f.samp.df.T01F$Treatment, type = "centroid")) #put in distance matrix
    
#T1
  u.PacuT1 <- phyloseq::distance(pacu.f.bac.T1, method = "wunifrac") 
      pacu.f.samp.df.T1 <- as(sample_data(pacu.f.bac.T1), "data.frame") 
      set.seed(30)
      adonis(u.PacuT1 ~ Treatment, data = pacu.f.samp.df.T1)  
      set.seed(30)
      adonis(u.PacuT1 ~ Clade*Treatment, data = pacu.f.samp.df.T1)  
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.PacuT1, pacu.f.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.PacuT1, pacu.f.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix

    
#TF
 
  u.pacu.f.TF <- phyloseq::distance(pacu.f.bac.TF, method = "wunifrac")
     pacu.f.samp.df.TF <- as(sample_data(pacu.f.bac.TF), "data.frame") 
      set.seed(30)
      adonis(u.pacu.f.TF ~Treatment, data = pacu.f.samp.df.TF)
      set.seed(30)
      adonis(u.pacu.f.TF ~ Clade*Treatment, data = pacu.f.samp.df.TF)
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.pacu.f.TF, pacu.f.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(u.pacu.f.TF, pacu.f.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix

    
  #try plotting only the top 10 FAMs     ####
                pacu.f.bac.T01F<-subset_samples(Bac.seqFam_stats.T01F, Species=="Pocillopora_acuta")

    #Sort the Phyla by abundance and pick the top 5
    top10P.names.Pacu = sort(tapply(taxa_sums(pacu.f.bac.T01F), tax_table(pacu.f.bac.T01F)[, "Family"], sum), TRUE)[1:10]
      #Cut down the  data to only the top 10 Phyla
      Pacu.u.f.bac.T01F.top10P = subset_taxa(pacu.f.bac.T01F, Family %in% names(top10P.names.Pacu))
      
      #pull out of phyloseq     
        Pacu.u.f.bac.T01F.top10P.OTU = as(otu_table(Pacu.u.f.bac.T01F.top10P), "matrix")
       # transpose if necessary
          # if(taxa_are_rows(Pacu.u.f.bac.T01F.top5P)){OTUfam.v <- t(Pacu.u.f.bac.T01F.top5P.OTU)}
          # Coerce to data.frame
            Pacu.u.f.bac.T01F.top10P.OTU.df = as.data.frame(Pacu.u.f.bac.T01F.top10P.OTU) 
      #rename Fam with taxa, all same but no Otu00084, and yes 30
            
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00068"] <- "Alphaproteobacteria_unclassified"
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00011"] <- "Amoebophilaceae"
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00206"] <- "Flavobacteriaceae"
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00156"] <- "Saprospiraceae"

            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00055"] <- "Pseudoalteromonadaceae"
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00062"] <- "Alteromonadaceae"
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00001"] <- "Endozoicomonadaceae"
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00226"] <- "Gammaproteobacteria_unclassified"
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00030"] <- "Cyclobacteriaceae"
            names(Pacu.u.f.bac.T01F.top10P.OTU.df)[names(Pacu.u.f.bac.T01F.top10P.OTU.df) == "Otu00025"] <- "Rhodobacteraceae"
         
            
    #Add metadata
      Pacu.u.f.bac.T01F.top10P.sd <- as(sample_data(Pacu.u.f.bac.T01F.top10P), "data.frame") #create sample data frame to view
         Pacu.u.f.bac.T01F.top10P.sd<-Pacu.u.f.bac.T01F.top10P.sd[,c("sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade")] # keep cols you want
    
    Pacu.16s.fam.top10<- merge(Pacu.u.f.bac.T01F.top10P.OTU.df,Pacu.u.f.bac.T01F.top10P.sd, by="row.names") #merge by rownames
       Pacu.16s.fam.top10.m <- melt(Pacu.16s.fam.top10, id=c("Row.names","sample_name.1", "Species","Parent.ID",          
                                                                   "Treatment","Time.Point","Tank.num","Clade"))
            names(Pacu.16s.fam.top10.m)[names(Pacu.16s.fam.top10.m) == "value"] <- "relabund"
            names(Pacu.16s.fam.top10.m)[names(Pacu.16s.fam.top10.m) == "variable"] <- "family"

#Means
Pacu.16s.fam.top10.m_mean <- ddply(Pacu.16s.fam.top10.m, c("Treatment", "Clade", "Time.Point","family","Species"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pacu.16s.fam.top10.m_mean

Pacu.16s.fam.top10.m_mean$Code<-paste(Pacu.16s.fam.top10.m_mean$Treatment, Pacu.16s.fam.top10.m_mean$Time.Point,Pacu.16s.fam.top10.m_mean$Clade)

#Pacu plots
#plot Pacu
Pacu.16s.fam.top10.m_mean_plot<-ggplot(data=Pacu.16s.fam.top10.m_mean, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("Pacu top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu.16s.fam.top10.m_mean_plot
Pacu.16s.fam.top10.m_mean_plot+facet_wrap(~Clade*Treatment)
       
    
    
    
#Means no phylotype
pacu.16s.fam.top10.m_mean.NoC <- ddply(Pacu.16s.fam.top10.m, c("Treatment", "Time.Point","family","Species"), summarise, #summarize cell counts
                 N    = length(relabund[!is.na(relabund)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(relabund, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(relabund, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(relabund, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
pacu.16s.fam.top10.m_mean.NoC

pacu.16s.fam.top10.m_mean.NoC$Code<-paste(pacu.16s.fam.top10.m_mean.NoC$Treatment, pacu.16s.fam.top10.m_mean.NoC$Time.Point,pacu.16s.fam.top10.m_mean.NoC$Clade)

#pacu plots
#plot pacu
pacu.16s.fam.top10.m_mean_plot<-ggplot(data=pacu.16s.fam.top10.m_mean.NoC, aes(x=Time.Point, y=mean, group = family, color=family)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=family), size=4, show.legend = T)+
  geom_line(aes(color=family), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("pacu top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));pacu.16s.fam.top10.m_mean_plot
pacu.16s.fam.top10.m_mean_plot+facet_wrap(~Treatment)
    
    
    
    
    
    
```
Plot top 10 families per species all together by taxa
```{r}  
#by clade
Top10fam<-rbind(mcap.16s.fam.T01F.top10.m_mean,pcomp.16s.fam.top10.m_mean,Pvar.16s.fam.top10.m_mean,Pacu.16s.fam.top10.m_mean)
Top10fam$Code<-paste(Top10fam$Species,Top10fam$Clade)
Top10fam.amb<-subset(Top10fam, Treatment=="Ambient")

Top10fam.high<-subset(Top10fam, Treatment=="High")

#amb
b16s.fam.top10.m_mean_plot<-ggplot(data=Top10fam.amb, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Code), size=4, show.legend = T)+
  geom_line(aes(color=Code), linetype=2, show.legend = T)+
  #scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" Ambient top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));b16s.fam.top10.m_mean_plot
b16s.fam.top10.m_mean_plot+facet_wrap(~family)

#high
b16s.fam.top10.m_mean_plotHIGH<-ggplot(data=Top10fam.high, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Code), size=4, show.legend = T)+
  geom_line(aes(color=Code), linetype=2, show.legend = T)+
  #scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("High top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));b16s.fam.top10.m_mean_plotHIGH
b16s.fam.top10.m_mean_plotHIGH+facet_wrap(~family)



#not by clade
Top10fam.noC<-rbind(mcap.16s.fam.T01F.top10.m_mean.noC,pcomp.16s.fam.top10.m_mean.NoC,pvar.16s.fam.top10.m_mean.NoC,pacu.16s.fam.top10.m_mean.NoC)
Top10fam.noC$Code<-paste(Top10fam.noC$Species,Top10fam.noC$Clade)
Top10fam.NoC.amb<-subset(Top10fam.noC, Treatment=="Ambient")
Top10fam.NoC.high<-subset(Top10fam.noC, Treatment=="High")

#amb
b16s.fam.top10.m_mean_plot<-ggplot(data=Top10fam.NoC.amb, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Code), size=4, show.legend = T)+
  geom_line(aes(color=Code), linetype=2, show.legend = T)+
  #scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" Ambient top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));b16s.fam.top10.m_mean_plot
b16s.fam.top10.m_mean_plot+facet_wrap(~family)

#high
b16s.fam.top10.m_mean_plotHIGH<-ggplot(data=Top10fam.NoC.high, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Code), size=4, show.legend = T, position="jitter")+
  # geom_jitter(width=0.1)+
  geom_line(aes(color=Code), linetype=2, show.legend = T)+
  #scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle("High top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));b16s.fam.top10.m_mean_plotHIGH
b16s.fam.top10.m_mean_plotHIGH+facet_wrap(~family)


#JUST endo
Top10fam.noC.ENDO<-subset(Top10fam.noC, family=="Endozoicomonadaceae")
Top10fam.noC.ENDO$Code2<-paste(Top10fam.noC.ENDO$Species, Top10fam.noC.ENDO$Treatment)

Endo.only_plot<-ggplot(data=Top10fam.noC.ENDO, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Species), size=4, show.legend = T)+
  geom_line(aes(color=Species), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Endo.only_plot
Endo.only_plot+facet_wrap(~Treatment)

#JUST endo
Top10fam.noC.ENDO2<-Top10fam.noC.ENDO[!(Top10fam.noC.ENDO$Species=="Porites_compressa"),]
Top10fam.noC.ENDO2$Code2<-paste(Top10fam.noC.ENDO2$Species, Top10fam.noC.ENDO2$Treatment)

Endo.only_plot<-ggplot(data=Top10fam.noC.ENDO2, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Species), size=4, show.legend = T)+
  geom_line(aes(color=Species), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Endo.only_plot
Endo.only_plot+facet_wrap(~Treatment)
```

#Vibrio only 
```{r} 
#JUST vibrio
vibrio<-subset_taxa(Bac.s.rel, Genus=="Vibrio")
vibrio<- prune_taxa(taxa_sums(vibrio) > 0, vibrio) #this removes any OTU with 0s



OTU.v = otu_table(vibrio)
# Extract abundance matrix from the phyloseq object
OTU.v = as(otu_table(OTU.v), "matrix")
if(taxa_are_rows(vibrio)){OTU.v <- t(OTU.v)}
OTU.v2 = as.data.frame(OTU.v)
#write.csv(OTU.cn2, "OTU.cn2_may18.csv")

OTU.v2 <- tibble::rownames_to_column(OTU.v2, "sample_name")

MetaData16Keep<- MetaData16[,c("sample_name", "Species","Parent.ID", "Treatment","Time.Point","Tank.num","Clade")] # keep only columns you need

#OK merge
    avib.16s.samples<-merge(MetaData16Keep,OTU.v2, by="sample_name") #merge on the metadata
write.csv(avib.16s.samples, "avib.16s.samples.csv")



vibrio.only_plot<-ggplot(data=avib.16s.samples, aes(x=Time.Point, y=mean)) +
  # geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
  #               width=0.1, show.legend = F)+
  geom_point(aes(color=Species), size=4, show.legend = T)+
  geom_line(aes(color=Species), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));vibrio.only_plot
vibrio.only_plot+facet_wrap(~Treatment)
```
Ralstonia
```{r}
#JUST Ralstonia
Ralstonia<-subset_taxa(Bac.s.rel, Genus=="Ralstonia")
Ralstonia<- prune_taxa(taxa_sums(Ralstonia) > 0, Ralstonia) #this removes any OTU with 0s



OTU.v = otu_table(Ralstonia)
# Extract abundance matrix from the phyloseq object
OTU.v = as(otu_table(OTU.v), "matrix")
if(taxa_are_rows(Ralstonia)){OTU.v <- t(OTU.v)}
OTU.v2 = as.data.frame(OTU.v)
#write.csv(OTU.cn2, "OTU.cn2_may18.csv")

OTU.v2 <- tibble::rownames_to_column(OTU.v2, "sample_name")

MetaData16Keep<- MetaData16[,c("sample_name", "Species","Parent.ID", "Treatment","Time.Point","Tank.num","Clade")] # keep only columns you need

#OK merge
    Ralst.16s.samples<-merge(MetaData16Keep,OTU.v2, by="sample_name") #merge on the metadata
write.csv(Ralst.16s.samples, "Ralst.16s.samples.csv")



Ralstonia.only_plot<-ggplot(data=avib.16s.samples, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Species), size=4, show.legend = T)+
  geom_line(aes(color=Species), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Ralstonia.only_plot
Ralstonia.only_plot+facet_wrap(~Treatment)
```

Halobacteriovorax
```{r}
#JUST Halobacteriovorax
Halobacteriovorax<-subset_taxa(Bac.s.rel, Genus=="Halobacteriovorax")
Halobacteriovorax<- prune_taxa(taxa_sums(Halobacteriovorax) > 0, Halobacteriovorax) #this removes any OTU with 0s



OTU.v = otu_table(Halobacteriovorax)
# Extract abundance matrix from the phyloseq object
OTU.v = as(otu_table(OTU.v), "matrix")
if(taxa_are_rows(Halobacteriovorax)){OTU.v <- t(OTU.v)}
OTU.v2 = as.data.frame(OTU.v)
#write.csv(OTU.cn2, "OTU.cn2_may18.csv")

OTU.v2 <- tibble::rownames_to_column(OTU.v2, "sample_name")

MetaData16Keep<- MetaData16[,c("sample_name", "Species","Parent.ID", "Treatment","Time.Point","Tank.num","Clade")] # keep only columns you need

#OK merge
    Hal.16s.samples<-merge(MetaData16Keep,OTU.v2, by="sample_name") #merge on the metadata
write.csv(Hal.16s.samples, "hal.16s.samples.csv")



Halobacteriovorax.only_plot<-ggplot(data=avib.16s.samples, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Species), size=4, show.legend = T)+
  geom_line(aes(color=Species), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Halobacteriovorax.only_plot
Halobacteriovorax.only_plot+facet_wrap(~Treatment)
```
Endozoicomonas OTU 002 
```{r}
#JUST Endozoicomonadaceae
Endozoicomonadaceae<-subset_taxa(Bac.s.rel, Family=="Endozoicomonadaceae")
Endozoicomonadaceae<- prune_taxa(taxa_sums(Endozoicomonadaceae) > 0, Endozoicomonadaceae) #this removes any OTU with 0s



OTU.v = otu_table(Endozoicomonadaceae)
# Extract abundance matrix from the phyloseq object
OTU.v = as(otu_table(OTU.v), "matrix")
if(taxa_are_rows(Endozoicomonadaceae)){OTU.v <- t(OTU.v)}
OTU.v2 = as.data.frame(OTU.v)
#write.csv(OTU.cn2, "OTU.cn2_may18.csv")

OTU.v2 <- tibble::rownames_to_column(OTU.v2, "sample_name")

MetaData16Keep<- MetaData16[,c("sample_name", "Species","Parent.ID", "Treatment","Time.Point","Tank.num","Clade")] # keep only columns you need

#OK merge
    Endo.16s.samples<-merge(MetaData16Keep,OTU.v2, by="sample_name") #merge on the metadata
write.csv(Endo.16s.samples, "endo.16s.samples.csv")



Endozoicomonadaceae.only_plot<-ggplot(data=avib.16s.samples, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Species), size=4, show.legend = T)+
  geom_line(aes(color=Species), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Endozoicomonadaceae.only_plot
Endozoicomonadaceae.only_plot+facet_wrap(~Treatment)
```

Francisellaceae OTU 002 
```{r}
#JUST Endozoicomonadaceae
Francisellaceae<-subset_taxa(Bac.s.rel, Family=="Francisellaceae")
Francisellaceae<- prune_taxa(taxa_sums(Francisellaceae) > 0, Francisellaceae) #this removes any OTU with 0s



OTU.v = otu_table(Francisellaceae)
# Extract abundance matrix from the phyloseq object
OTU.v = as(otu_table(OTU.v), "matrix")
if(taxa_are_rows(Francisellaceae)){OTU.v <- t(OTU.v)}
OTU.v2 = as.data.frame(OTU.v)
#write.csv(OTU.cn2, "OTU.cn2_may18.csv")

OTU.v2 <- tibble::rownames_to_column(OTU.v2, "sample_name")

MetaData16Keep<- MetaData16[,c("sample_name", "Species","Parent.ID", "Treatment","Time.Point","Tank.num","Clade")] # keep only columns you need

#OK merge
    Fran.16s.samples<-merge(MetaData16Keep,OTU.v2, by="sample_name") #merge on the metadata
write.csv(Fran.16s.samples, "fran.16s.samples2.csv")



Francisellaceae.only_plot<-ggplot(data=avib.16s.samples, aes(x=Time.Point, y=mean)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Species), size=4, show.legend = T)+
  geom_line(aes(color=Species), linetype=2, show.legend = T)+
  scale_color_manual(values=Fam.bac.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        #axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
       # text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("relative abundance"))) +
  ggtitle(" top 10 family")+
  theme(plot.title = element_text(size=20, face = "italic"));Francisellaceae.only_plot
Francisellaceae.only_plot+facet_wrap(~Treatment)
```
Archaea

```{r}
#JUST ARC
ARC<-subset_taxa(Bac.s.rel, Domain=="Archaea")
ARC<- prune_taxa(taxa_sums(ARC) > 0, ARC) #this removes any OTU with 0s



OTU.v = otu_table(ARC)
# Extract abundance matrix from the phyloseq object
OTU.v = as(otu_table(OTU.v), "matrix")
if(taxa_are_rows(ARC)){OTU.v <- t(OTU.v)}
OTU.v2 = as.data.frame(OTU.v)
#write.csv(OTU.cn2, "OTU.cn2_may18.csv")

OTU.v2 <- tibble::rownames_to_column(OTU.v2, "sample_name")

MetaData16Keep<- MetaData16[,c("sample_name", "Species","Parent.ID", "Treatment","Time.Point","Tank.num","Clade")] # keep only columns you need

#OK merge
    ARC.16s.samples<-merge(MetaData16Keep,OTU.v2, by="sample_name") #merge on the metadata
write.csv(ARC.16s.samples, "Arc.16s.samples.csv")


```

#Permanova UNIFRAC by ASV (no filter) to compare with above
```{r}         
Bac.s.rel.u<-Bac.s.rel #make a copy

saveRDS(Bac.s.rel.u, file = "16S_rel.u.rds")



 #T0 only #### 
Bac.s.rel.u.T0<-subset_samples(Bac.s.rel.u, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s
      sample_sums(Bac.s.rel.u.T0)
#make unifrac matrix
Bac.s.u.T0 <- phyloseq::distance(Bac.s.rel.u.T0, method = "wunifrac")
s.u.samp.df <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df
s.u.samp.df$Code<-paste(s.u.samp.df$Species,s.u.samp.df$Clade)
  #adonis test
set.seed(30)
adonis(Bac.s.u.T0 ~ Species, data = s.u.samp.df)
set.seed(30)    
adonis(Bac.s.u.T0 ~ Species*Clade, data = s.u.samp.df)

# Homogeneity of dispersion test  
    set.seed(30)
    t<-permutest(betadisper(Bac.s.u.T0, s.u.samp.df$Species, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(Bac.s.u.T0, s.u.samp.df$Code, type = "centroid")) 
    
  ball<-  betadisper(Bac.s.u.T0, s.u.samp.df$Species, type = "centroid")
  TukeyHSD(ball,  ordered = FALSE,conf.level = 0.95)  

#T0 by speices
#Mcap  
  #t0 only Bac.s.rel.u.T0
  mcap.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Montipora_capitata")
  mcap.u.bac.T0 = prune_taxa(taxa_sums(mcap.u.bac.T0) > 0, mcap.u.bac.T0) #this removes any OTU with 0s
  sample_sums(mcap.u.bac.T0)
  
  #make bray curtis data matrix -T1 and TF
  u.mcap.u.T0 <- phyloseq::distance(mcap.u.bac.T0, method = "wunifrac")
     mcap.u.samp.df.T0 <- as(sample_data(mcap.u.bac.T0), "data.frame") #sample df
        mcap.u.samp.df.T0$Code<-paste(mcap.u.samp.df.T0$Treatment, mcap.u.samp.df.T0$Time.Point)
        mcap.u.samp.df.T0$Code2<-paste(mcap.u.samp.df.T0$Treatment, mcap.u.samp.df.T0$Time.Point,mcap.u.samp.df.T0$Clade)
set.seed(30)
    adonis(u.mcap.u.T0 ~ Clade, data = mcap.u.samp.df.T0) #adonis test no clade
set.seed(30)
    permutest(betadisper(u.mcap.u.T0, mcap.u.samp.df.T0$Clade, type = "centroid")) 

#Pcomp  
  Pcomp.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Porites_compressa")
  Pcomp.u.bac.T0 = prune_taxa(taxa_sums(Pcomp.u.bac.T0) > 0, Pcomp.u.bac.T0) #this removes any OTU with 0s
  sample_sums(Pcomp.u.bac.T0)
  
  #make bray curtis data matrix -T1 and TF
  u.Pcomp.u.T0 <- phyloseq::distance(Pcomp.u.bac.T0, method = "wunifrac")
     Pcomp.u.samp.df.T0 <- as(sample_data(Pcomp.u.bac.T0), "data.frame") #sample df
        Pcomp.u.samp.df.T0$Code<-paste(Pcomp.u.samp.df.T0$Treatment, Pcomp.u.samp.df.T0$Time.Point)
        Pcomp.u.samp.df.T0$Code2<-paste(Pcomp.u.samp.df.T0$Treatment, Pcomp.u.samp.df.T0$Time.Point,Pcomp.u.samp.df.T0$Clade)
set.seed(30)
    adonis(u.Pcomp.u.T0 ~ Clade, data = Pcomp.u.samp.df.T0) #adonis test no clade
set.seed(30)
    permutest(betadisper(u.Pcomp.u.T0, Pcomp.u.samp.df.T0$Clade, type = "centroid")) 
  
#Pvar  
  #t0 only Bac.s.rel.u.T0
  Pvar.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Pavona_varians")
  Pvar.u.bac.T0 = prune_taxa(taxa_sums(Pvar.u.bac.T0) > 0, Pvar.u.bac.T0) #this removes any OTU with 0s
  sample_sums(Pvar.u.bac.T0)
  
  #make bray curtis data matrix -T1 and TF
  u.Pvar.u.T0 <- phyloseq::distance(Pvar.u.bac.T0, method = "wunifrac")
     Pvar.u.samp.df.T0 <- as(sample_data(Pvar.u.bac.T0), "data.frame") #sample df
        Pvar.u.samp.df.T0$Code<-paste(Pvar.u.samp.df.T0$Treatment, Pvar.u.samp.df.T0$Time.Point)
        Pvar.u.samp.df.T0$Code2<-paste(Pvar.u.samp.df.T0$Treatment, Pvar.u.samp.df.T0$Time.Point,Pvar.u.samp.df.T0$Clade)
set.seed(30)
    adonis(u.Pvar.u.T0 ~ Clade, data = Pvar.u.samp.df.T0) #adonis test no clade
set.seed(30)
    permutest(betadisper(u.Pvar.u.T0, Pvar.u.samp.df.T0$Clade, type = "centroid"))     

    
#Pacu  
  #t0 only Bac.s.rel.u.T0
  Pacu.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Pocillopora_acuta")
  Pacu.u.bac.T0 = prune_taxa(taxa_sums(Pacu.u.bac.T0) > 0, Pacu.u.bac.T0) #this removes any OTU with 0s
  sample_sums(Pacu.u.bac.T0)
  
  #make bray curtis data matrix -T1 and TF
  u.Pacu.u.T0 <- phyloseq::distance(Pacu.u.bac.T0, method = "wunifrac")
     Pacu.u.samp.df.T0 <- as(sample_data(Pacu.u.bac.T0), "data.frame") #sample df
        Pacu.u.samp.df.T0$Code<-paste(Pacu.u.samp.df.T0$Treatment, Pacu.u.samp.df.T0$Time.Point)
        Pacu.u.samp.df.T0$Code2<-paste(Pacu.u.samp.df.T0$Treatment, Pacu.u.samp.df.T0$Time.Point,Pacu.u.samp.df.T0$Clade)
set.seed(30)
    adonis(u.Pacu.u.T0 ~ Clade, data = Pacu.u.samp.df.T0) #adonis test no clade
set.seed(30)
    permutest(betadisper(u.Pacu.u.T0, Pacu.u.samp.df.T0$Clade, type = "centroid")) 
  
    
    
    
    
    
    
    
 #AMBINET only ####
Bac.s.rel.u.Amb<-subset_samples(Bac.s.rel.u, Treatment=="Ambient")
    Bac.s.rel.u.Amb = prune_taxa(taxa_sums(Bac.s.rel.u.Amb) > 0, Bac.s.rel.u.Amb) #this removes any OTU with 0s
      sample_sums(Bac.s.rel.u.Amb)
#make unifrac matrix
Bac.s.u.Amb <- phyloseq::distance(Bac.s.rel.u.Amb, method = "wunifrac")
s.u.samp.df <- as(sample_data(Bac.s.rel.u.Amb), "data.frame") #sample df
s.u.samp.df$Code<-paste(s.u.samp.df$Species,s.u.samp.df$Clade)

  #adonis test
set.seed(30)
adonis(Bac.s.u.Amb ~ Species*Time.Point, data = s.u.samp.df)
set.seed(30)    
adonis(Bac.s.u.Amb ~ Species*Clade, data = s.u.samp.df)

     # Homogeneity of dispersion test  
    set.seed(30)
    t<-permutest(betadisper(Bac.s.u.Amb, s.u.samp.df$Species, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(Bac.s.u.Amb, s.u.samp.df$Treatment, type = "centroid")) 
    
  ball<-  betadisper(Bac.s.u.Amb, s.u.samp.df$Code, type = "centroid")
  TukeyHSD(ball,  ordered = FALSE,conf.level = 0.95)  
 









#T1TF ####
Bac.s.rel.u.T1F<-subset_samples(Bac.s.rel.u, Time.Point=="T1"|Time.Point=="TF")
    Bac.s.rel.u.T1F = prune_taxa(taxa_sums(Bac.s.rel.u.T1F) > 0, Bac.s.rel.u.T1F) #this removes any OTU with 0s
      sample_sums(Bac.s.rel.u.T1F)
#make unifrac matrix
Bac.s.u.T1F <- phyloseq::distance(Bac.s.rel.u.T1F, method = "wunifrac")
s.u.samp.df.T1F <- as(sample_data(Bac.s.rel.u.T1F), "data.frame") #sample df
  #adonis test
set.seed(30)
adonis(Bac.s.u.T1F ~ Species*Treatment*Time.Point, data = s.u.samp.df.T1F)
set.seed(30)    
adonis(Bac.s.u.T1F ~ Species*Treatment*Time.Point*Clade, data = s.u.samp.df.T1F)

     # Homogeneity of dispersion test  
    set.seed(30)
    t<-permutest(betadisper(Bac.s.u.T1F, s.u.samp.df.T1F$Code, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(Bac.s.u.T1F, s.u.samp.df.T1F$Treatment, type = "centroid")) 
    
  ball<-  betadisper(Bac.s.u.T1F, s.u.samp.df.T1F$Species, type = "centroid")
  TukeyHSD(ball,  ordered = FALSE,conf.level = 0.95)  

  
  
      

    
    
    
    
############# mcap #### 
 
   
  
  
  
  
  
#Mcap
mcap.u.bac.T1F <- subset_samples(Bac.s.rel.u.T1F, Species == "Montipora_capitata")
  sample_sums(mcap.u.bac.T1F)

mcap.u.f.bac.T1F = prune_taxa(taxa_sums(mcap.u.f.bac.T1F) > 0, mcap.u.f.bac.T1F) #this removes any OTU with 0s
  sample_sums(mcap.u.f.bac.T1F)



#make dfs you need 
mcap.u.bac.T1 <- subset_samples(mcap.u.bac.T1F, Time.Point == "T1")
  mcap.u.f.bac.T1 = prune_taxa(taxa_sums(mcap.u.f.bac.T1) > 0, mcap.u.f.bac.T1) #this removes any OTU with 0s
mcap.u.bac.TF <- subset_samples(mcap.u.f.bac.T1F, Time.Point == "TF")
  mcap.u.f.bac.TF = prune_taxa(taxa_sums(mcap.u.f.bac.TF) > 0, mcap.u.f.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  u.mcap.u.T1F <- phyloseq::distance(mcap.u.bac.T1F, method = "wunifrac")
     mcap.u.samp.df.T1F <- as(sample_data(mcap.u.bac.T1F), "data.frame") #sample df
        mcap.u.samp.df.T1F$Code<-paste(mcap.u.samp.df.T1F$Treatment, mcap.u.samp.df.T1F$Time.Point)
        mcap.u.samp.df.T1F$Code2<-paste(mcap.u.samp.df.T1F$Treatment, mcap.u.samp.df.T1F$Time.Point,mcap.u.samp.df.T1F$Clade)

       set.seed(30)
    adonis(u.mcap.u.T1F ~ Treatment*Time.Point, data = mcap.u.samp.df.T1F) #adonis test no clade
      set.seed(30)
    adonis(u.mcap.u.T1F ~ Clade*Treatment*Time.Point, data = mcap.u.samp.df.T1F)#adonis test with clade
   # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(u.mcap.u.T1F, mcap.u.samp.df.T1F$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(u.mcap.u.T1F, mcap.u.samp.df.T1F$Treatment, type = "centroid"))
    set.seed(30)
    permutest(betadisper(u.mcap.u.T1F, mcap.u.samp.df.T1F$Time.Point, type = "centroid"))
     set.seed(30)
    permutest(betadisper(u.mcap.u.T1F, mcap.u.samp.df.T1F$Code, type = "centroid"))
    
  
    
    
#T1
  mcap.u.bac.mcapT1 <- phyloseq::distance(mcap.u.bac.T1, method = "wunifrac") 
      mcap.u.samp.df.T1 <- as(sample_data(mcap.u.bac.T1), "data.frame") 
         mcap.u.samp.df.T1$Code<-paste(mcap.u.samp.df.T1$Treatment,mcap.u.samp.df.T1$Clade)
       set.seed(30)
    adonis(mcap.u.bac.mcapT1 ~ Treatment, data = mcap.u.samp.df.T1)  
     set.seed(30)
    adonis(mcap.u.bac.mcapT1 ~ Clade*Treatment, data = mcap.u.samp.df.T1)  
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(mcap.u.bac.mcapT1, mcap.u.samp.df.T1$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(mcap.u.bac.mcapT1, mcap.u.samp.df.T1$Treatment, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(mcap.u.bac.mcapT1, mcap.u.samp.df.T1$Code, type = "centroid")) 
 


#TF
  mcap.u.TF <- phyloseq::distance(mcap.u.bac.TF, method = "wunifrac")
     mcap.u.samp.df.TF <- as(sample_data(mcap.u.bac.TF), "data.frame")
     mcap.u.samp.df.TF$Code<-paste(mcap.u.samp.df.TF$Treatment,mcap.u.samp.df.TF$Clade)
    set.seed(30)
    adonis(mcap.u.TF ~ Treatment, data = mcap.u.samp.df.TF)
    set.seed(30)
    adonis(mcap.u.TF ~ Clade*Treatment, data = mcap.u.samp.df.TF)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(mcap.u.TF, mcap.u.samp.df.TF$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(mcap.u.TF, mcap.u.samp.df.TF$Treatment, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(mcap.u.TF, mcap.u.samp.df.TF$Code, type = "centroid")) 
    
    
#mcap ambient only all three times
    mcap.u.bac.AmbT01F <- subset_samples(Bac.s.rel.u, Species == "Montipora_capitata")
        mcap.u.bac.AmbT01F <- subset_samples(mcap.u.bac.AmbT01F, Treatment == "Ambient")
  sample_sums(mcap.u.bac.AmbT01F)

mcap.u.f.bac.AmbT01F = prune_taxa(taxa_sums(mcap.u.bac.AmbT01F) > 0, mcap.u.bac.AmbT01F) #this removes any OTU with 0s
  sample_sums(mcap.u.f.bac.AmbT01F)
    
  mcap.u.AmbT01F <- phyloseq::distance(mcap.u.f.bac.AmbT01F, method = "wunifrac")
     mcap.u.samp.df.AmbT01F <- as(sample_data(mcap.u.bac.AmbT01F), "data.frame")
     mcap.u.samp.df.AmbT01F$Code<-paste(mcap.u.samp.df.AmbT01F$Treatment,mcap.u.samp.df.AmbT01F$Clade)
    set.seed(30)
    adonis(mcap.u.AmbT01F ~ Time.Point, data = mcap.u.samp.df.AmbT01F)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(mcap.u.AmbT01F, mcap.u.samp.df.AmbT01F$Time.Point, type = "centroid")) 
  
# Mcap Bac.s.rel.u.T0   only
    mcap.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Montipora_capitata")
  sample_sums(mcap.u.bac.T0)

mcap.u.f.bac.T0 = prune_taxa(taxa_sums(mcap.u.bac.T0) > 0, mcap.u.bac.T0) #this removes any OTU with 0s
  sample_sums(mcap.u.f.bac.T0)
    
  mcap.u.T0 <- phyloseq::distance(mcap.u.f.bac.T0, method = "wunifrac")
     mcap.u.samp.df.T0 <- as(sample_data(mcap.u.bac.T0), "data.frame")
     mcap.u.samp.df.T0$Code<-paste(mcap.u.samp.df.T0$Treatment,mcap.u.samp.df.T0$Clade)
    set.seed(30)
    adonis(mcap.u.T0 ~ Clade, data = mcap.u.samp.df.T0)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(mcap.u.T0, mcap.u.samp.df.T0$Clade, type = "centroid")) 
  
  
  
  
  
  
  
############ pcomp #### 
  #what endos are in here?
  Bac.s.rel.u.Endo<-subset_samples(Bac.s.rel.u, Species=="Porites_compressa")
Bac.s.rel.u.Endo.2 = subset_taxa(Bac.s.rel.u.Endo, Family=="Endozoicomonadaceae")
OTUendo1 = as(otu_table(Bac.s.rel.u.Endo.2), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.rel.u.Endo.2)){OTUendo1 <- t(OTUendo1)}
# Coerce to data.frame
OTUendodf = as.data.frame(OTUendo1)  
write.csv(OTUendodf,"OTUendodf.csv")   





    
#Pcomp
pcomp.u.bac.T1F <- subset_samples(Bac.s.rel.u.T1F, Species == "Porites_compressa")
pcomp.u.bac.T1F = prune_taxa(taxa_sums(pcomp.u.bac.T1F) > 0, pcomp.u.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
pcomp.u.bac.T1 <- subset_samples(pcomp.u.bac.T1F, Time.Point == "T1")
  pcomp.u.bac.T1 = prune_taxa(taxa_sums(pcomp.u.bac.T1) > 0, pcomp.u.bac.T1) #this removes any OTU with 0s

pcomp.u.bac.TF <- subset_samples(pcomp.u.bac.T1F, Time.Point == "TF")
 pcomp.u.bac.TF = prune_taxa(taxa_sums(pcomp.u.bac.TF) > 0, pcomp.u.bac.TF) #this removes any OTU with 0s

#T1TF
#make wunifrac curtis data matrix -T1 and TF
  pcomp.u.T1F <- phyloseq::distance(pcomp.u.bac.T1F, method = "wunifrac")
     pcomp.u.samp.df.T1F <- as(sample_data(pcomp.u.bac.T1F), "data.frame") #sample df
     pcomp.u.samp.df.T1F$Code<-paste(pcomp.u.samp.df.T1F$Treatment, pcomp.u.samp.df.T1F$Time.Point)
          pcomp.u.samp.df.T1F$Code2<-paste(pcomp.u.samp.df.T1F$Treatment, pcomp.u.samp.df.T1F$Time.Point,pcomp.u.samp.df.T1F$Clade)

      set.seed(30)
    adonis(pcomp.u.T1F ~ Treatment*Time.Point, data = pcomp.u.samp.df.T1F) #adonis test no clade
   set.seed(30)
    adonis(pcomp.u.T1F ~ Clade*Treatment*Time.Point, data = pcomp.u.samp.df.T1F)#adonis test with clade
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pcomp.u.T1F, pcomp.u.samp.df.T1F$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pcomp.u.T1F, pcomp.u.samp.df.T1F$Time.Point, type = "centroid")) 
     set.seed(30)
    permutest(betadisper(pcomp.u.T1F, pcomp.u.samp.df.T1F$Code, type = "centroid")) 
     set.seed(30)
    permutest(betadisper(pcomp.u.T1F, pcomp.u.samp.df.T1F$Code2, type = "centroid")) 
#T1
  pcomp.u.T1 <- phyloseq::distance(pcomp.u.bac.T1, method = "wunifrac") 
      pcomp.u.samp.df.T1 <- as(sample_data(pcomp.u.bac.T1), "data.frame")
      pcomp.u.samp.df.T1$Code<-paste(pcomp.u.samp.df.T1$Treatment, pcomp.u.samp.df.T1$Clade)
   set.seed(30)
   adonis(pcomp.u.T1 ~ Treatment, data = pcomp.u.samp.df.T1)  
    set.seed(30)
   adonis(pcomp.u.T1 ~ Clade*Treatment, data = pcomp.u.samp.df.T1)  
 # Homogeneity of dispersion test  
   pcomp.u.samp.df.T1$Code<-paste(pcomp.u.samp.df.T1$Treatment,pcomp.u.samp.df.T1$Clade)
    set.seed(30)
    permutest(betadisper(pcomp.u.T1, pcomp.u.samp.df.T1$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pcomp.u.T1, pcomp.u.samp.df.T1$Treatment, type = "centroid")) 
       set.seed(30)
    permutest(betadisper(pcomp.u.T1, pcomp.u.samp.df.T1$Code, type = "centroid")) 
    
     ## Tukey's Honest Significant Differences
  mod<-  (betadisper(pcomp.u.T1, pcomp.u.samp.df.T1$Code, type = "centroid"))
  mod.HSD <- TukeyHSD(mod)
  plot(mod.HSD)
    
    set.seed(30)
    permutest(betadisper(pcomp.u.T1, pcomp.u.samp.df.T1$Code, type = "centroid")) 

#TF
pcomp.u.TF <- phyloseq::distance(pcomp.u.bac.TF, method = "wunifrac")
     pcomp.u.samp.df.TF <- as(sample_data(pcomp.u.bac.TF), "data.frame") 
           pcomp.u.samp.df.TF$Code<-paste(pcomp.u.samp.df.TF$Treatment, pcomp.u.samp.df.TF$Clade)
           set.seed(30)
    adonis(pcomp.u.TF ~ Treatment, data = pcomp.u.samp.df.TF)
     set.seed(30)
    adonis(pcomp.u.TF ~ Clade*Treatment, data = pcomp.u.samp.df.TF)
  # Homogeneity of dispersion test  
       pcomp.u.samp.df.TF$Code<-paste(pcomp.u.samp.df.TF$Treatment,pcomp.u.samp.df.TF$Clade)

    set.seed(30)
    permutest(betadisper(pcomp.u.TF, pcomp.u.samp.df.TF$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pcomp.u.TF, pcomp.u.samp.df.TF$Treatment, type = "centroid")) 
     set.seed(30)
    permutest(betadisper(pcomp.u.TF, pcomp.u.samp.df.TF$Code, type = "centroid")) 
mod<-  (betadisper(pcomp.u.TF, pcomp.u.samp.df.TF$Code, type = "centroid"))
  mod.HSD <- TukeyHSD(mod)
  plot(mod.HSD)
    


# Pcomp Bac.s.rel.u.T0   only
    pcomp.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Porites_compressa")
  sample_sums(pcomp.u.bac.T0)

pcomp.u.f.bac.T0 = prune_taxa(taxa_sums(pcomp.u.bac.T0) > 0, pcomp.u.bac.T0) #this removes any OTU with 0s
  sample_sums(pcomp.u.f.bac.T0)
    
  pcomp.u.T0 <- phyloseq::distance(pcomp.u.f.bac.T0, method = "wunifrac")
     pcomp.u.samp.df.T0 <- as(sample_data(pcomp.u.bac.T0), "data.frame")
     pcomp.u.samp.df.T0$Code<-paste(pcomp.u.samp.df.T0$Treatment,pcomp.u.samp.df.T0$Clade)
    set.seed(30)
    adonis(pcomp.u.T0 ~ Clade, data = pcomp.u.samp.df.T0)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pcomp.u.T0, pcomp.u.samp.df.T0$Clade, type = "centroid")) 
  

#Ambient only !!! ####
    pcomp.u.bac.AllTime <- subset_samples(Bac.s.rel.u, Species == "Porites_compressa") #all pcomp
      
    # pull out otu and tax
    tax  = tax_table(pcomp.u.bac.AllTime)
    OTU = otu_table(pcomp.u.bac.AllTime)
    tree=phy_tree(pcomp.u.bac.AllTime)
  # sample data
  sd = sample_data(pcomp.u.bac.AllTime)
  sd$Treatment<-as.character(sd$Treatment)
  
  sd$Treatment[sd$Treatment == "High" & sd$Time.Point=="T0"] <- "Ambient" 
  sd$Treatment<-as.factor(sd$Treatment)

  pcomp.u.bac.Alltime2<-  phyloseq(sd, tax, OTU,tree) #make phylo ojb
  pcomp.u.bac.Alltime2.sd <- as(sample_data(pcomp.u.bac.Alltime2), "data.frame") #create sample data frame to view

  pcomp.u.bac.Amb <- subset_samples(pcomp.u.bac.Alltime2, Treatment == "Ambient")

  sample_sums(pcomp.u.bac.Amb)

pcomp.u.f.bac.Amb = prune_taxa(taxa_sums(pcomp.u.bac.Amb) > 0, pcomp.u.bac.Amb) #this removes any OTU with 0s
  sample_sums(pcomp.u.f.bac.Amb)
    
  pcomp.u.Amb <- phyloseq::distance(pcomp.u.f.bac.Amb, method = "wunifrac")
     pcomp.u.samp.df.Amb <- as(sample_data(pcomp.u.f.bac.Amb), "data.frame")
     pcomp.u.samp.df.Amb$Code<-paste(pcomp.u.samp.df.Amb$Treatment,pcomp.u.samp.df.Amb$Clade)
    
     set.seed(30)
    adonis(pcomp.u.Amb ~ Clade, data = pcomp.u.samp.df.Amb)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pcomp.u.Amb, pcomp.u.samp.df.Amb$Clade, type = "centroid")) 

# Pcomp remove the two outliers  T1 ####
  pcomp.u.bac.T1.removed <- subset_samples(pcomp.u.bac.T1, sample_name.1 != "SN134" )
    
#T1
  pcomp.u.T1.removed <- phyloseq::distance(pcomp.u.bac.T1.removed, method = "wunifrac") 
      pcomp.u.samp.df.T1.removed <- as(sample_data(pcomp.u.bac.T1.removed), "data.frame")
      pcomp.u.samp.df.T1.removed$Code<-paste(pcomp.u.samp.df.T1.removed$Treatment, pcomp.u.samp.df.T1.removed$Clade)
   set.seed(30)
   adonis(pcomp.u.T1.removed ~ Treatment, data = pcomp.u.samp.df.T1.removed)  
    set.seed(30)
   adonis(pcomp.u.T1.removed ~ Clade*Treatment, data = pcomp.u.samp.df.T1.removed)  
 # Homogeneity of dispersion test  
   pcomp.u.samp.df.T1.removed$Code<-paste(pcomp.u.samp.df.T1.removed$Treatment,pcomp.u.samp.df.T1.removed$Clade)
    set.seed(30)
    permutest(betadisper(pcomp.u.T1.removed, pcomp.u.samp.df.T1.removed$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pcomp.u.T1.removed, pcomp.u.samp.df.T1.removed$Treatment, type = "centroid")) 
       set.seed(30)
    permutest(betadisper(pcomp.u.T1.removed, pcomp.u.samp.df.T1.removed$Code, type = "centroid")) 
    
     ## Tukey's Honest Significant Differences
  mod<-  (betadisper(pcomp.u.T1.removed, pcomp.u.samp.df.T1.removed$Code, type = "centroid"))
  mod.HSD <- TukeyHSD(mod)
  plot(mod.HSD)
    
    set.seed(30)
    permutest(betadisper(pcomp.u.T1.removed, pcomp.u.samp.df.T1.removed$Code, type = "centroid")) 
    
    # Pcomp remove the two outliers  TF ####
  pcomp.u.bac.TF.removed <- subset_samples(pcomp.u.bac.TF, sample_name.1 != "N98" )
    
#TF
  pcomp.u.TF.removed <- phyloseq::distance(pcomp.u.bac.TF.removed, method = "wunifrac") 
      pcomp.u.samp.df.TF.removed <- as(sample_data(pcomp.u.bac.TF.removed), "data.frame")
      pcomp.u.samp.df.TF.removed$Code<-paste(pcomp.u.samp.df.TF.removed$Treatment, pcomp.u.samp.df.TF.removed$Clade)
   set.seed(30)
   adonis(pcomp.u.TF.removed ~ Treatment, data = pcomp.u.samp.df.TF.removed)  
    set.seed(30)
   adonis(pcomp.u.TF.removed ~ Clade*Treatment, data = pcomp.u.samp.df.TF.removed)  
 # Homogeneity of dispersion test  
   pcomp.u.samp.df.TF.removed$Code<-paste(pcomp.u.samp.df.TF.removed$Treatment,pcomp.u.samp.df.TF.removed$Clade)
    set.seed(30)
    permutest(betadisper(pcomp.u.TF.removed, pcomp.u.samp.df.TF.removed$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pcomp.u.TF.removed, pcomp.u.samp.df.TF.removed$Treatment, type = "centroid")) 
       set.seed(30)
    permutest(betadisper(pcomp.u.TF.removed, pcomp.u.samp.df.TF.removed$Code, type = "centroid")) 
    
     ## Tukey's Honest Significant Differences
  mod<-  (betadisper(pcomp.u.TF.removed, pcomp.u.samp.df.TF.removed$Code, type = "centroid"))
  mod.HSD <- TukeyHSD(mod)
  plot(mod.HSD)
    
    set.seed(30)
    permutest(betadisper(pcomp.u.TF.removed, pcomp.u.samp.df.TF.removed$Code, type = "centroid")) 
    
######### pvar ####
pvar.u.u.bac.T1F <- subset_samples(Bac.s.rel.u.T1F, Species == "Pavona_varians")
  pvar.u.u.bac.T1F = prune_taxa(taxa_sums(pvar.u.u.bac.T1F) > 0, pvar.u.u.bac.T1F) #this removes any OTU with 0s
 
#make dfs you need 
pvar.u.u.bac.T1 <- subset_samples(pvar.u.u.bac.T1F, Time.Point == "T1")
  pvar.u.u.bac.T1 = prune_taxa(taxa_sums(pvar.u.u.bac.T1) > 0, pvar.u.u.bac.T1) #this removes any OTU with 0s

pvar.u.u.bac.TF <- subset_samples(pvar.u.u.bac.T1F, Time.Point == "TF")
 pvar.u.u.bac.TF = prune_taxa(taxa_sums(pvar.u.u.bac.TF) > 0, pvar.u.u.bac.TF) #this removes any OTU with 0s
 
#T1TF
#make wunifrac curtis data matrix -T1 and TF

  pvar.u.u.T1F <- phyloseq::distance(pvar.u.u.bac.T1F, method = "wunifrac")
     pvar.u.u.samp.df.T1F <- as(sample_data(pvar.u.u.bac.T1F), "data.frame") #sample df
     pvar.u.u.samp.df.T1F$Code<-paste(pvar.u.u.samp.df.T1F$Clade,pvar.u.u.samp.df.T1F$Treatment)
          pvar.u.u.samp.df.T1F$Code2<-paste(pvar.u.u.samp.df.T1F$Clade,pvar.u.u.samp.df.T1F$Treatment,pvar.u.u.samp.df.T1F$Time.Point)

     set.seed(30)
adonis(pvar.u.u.T1F ~ Treatment*Time.Point, data = pvar.u.u.samp.df.T1F) #adonis test no clade
    set.seed(30)
    adonis(pvar.u.u.T1F ~ Clade*Treatment*Time.Point, data = pvar.u.u.samp.df.T1F)#adonis test with clade
    
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pvar.u.u.T1F, pvar.u.u.samp.df.T1F$Clade, type = "centroid")) #
   set.seed(30)
    permutest(betadisper(pvar.u.u.T1F, pvar.u.u.samp.df.T1F$Treatment, type = "centroid")) #
     set.seed(30)
    permutest(betadisper(pvar.u.u.T1F, pvar.u.u.samp.df.T1F$Time.Point, type = "centroid")) #
    set.seed(30)
    permutest(betadisper(pvar.u.u.T1F, pvar.u.u.samp.df.T1F$Code, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(pvar.u.u.T1F, pvar.u.u.samp.df.T1F$Code2, type = "centroid"))
    
    

#T1
                
  pvar.u.u.T1 <- phyloseq::distance(pvar.u.u.bac.T1, method = "wunifrac") 
      pvar.u.u.samp.df.T1 <- as(sample_data(pvar.u.u.bac.T1), "data.frame") 
      pvar.u.u.samp.df.T1$Code<-paste(pvar.u.u.samp.df.T1$Treatment,pvar.u.u.samp.df.T1$Clade)
   set.seed(30)
   adonis(pvar.u.u.T1 ~ Treatment, data = pvar.u.u.samp.df.T1) 
     set.seed(30)
     adonis(pvar.u.u.T1 ~ Clade*Treatment, data = pvar.u.u.samp.df.T1)  
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pvar.u.u.T1, pvar.u.u.samp.df.T1$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pvar.u.u.T1, pvar.u.u.samp.df.T1$Treatment, type = "centroid")) 
     set.seed(30)
    permutest(betadisper(pvar.u.u.T1, pvar.u.u.samp.df.T1$Code, type = "centroid")) 
    
    

#TF
  pvar.u.u.TF<- phyloseq::distance(pvar.u.u.bac.TF, method = "wunifrac")
     pvar.u.u.samp.df.TF <- as(sample_data(pvar.u.u.bac.TF), "data.frame") 
      pvar.u.u.samp.df.TF$Code<-paste(pvar.u.u.samp.df.TF$Clade,pvar.u.u.samp.df.TF$Treatment)
      set.seed(30)
      adonis(pvar.u.u.TF ~ Treatment, data = pvar.u.u.samp.df.TF)
        set.seed(30)
        adonis(pvar.u.u.TF ~ Clade*Treatment, data = pvar.u.u.samp.df.TF)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pvar.u.u.TF, pvar.u.u.samp.df.TF$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pvar.u.u.TF, pvar.u.u.samp.df.TF$Treatment, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(pvar.u.u.TF, pvar.u.u.samp.df.TF$Code, type = "centroid")) 

    

# Pvar Bac.s.rel.u.T0   only #### 
    pvar.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Pavona_varians")
  sample_sums(pvar.u.bac.T0)

pvar.u.f.bac.T0 = prune_taxa(taxa_sums(pvar.u.bac.T0) > 0, pvar.u.bac.T0) #this removes any OTU with 0s
  sample_sums(pvar.u.f.bac.T0)
    
  pvar.u.T0 <- phyloseq::distance(pvar.u.f.bac.T0, method = "wunifrac")
     pvar.u.samp.df.T0 <- as(sample_data(pvar.u.bac.T0), "data.frame")
     pvar.u.samp.df.T0$Code<-paste(pvar.u.samp.df.T0$Treatment,pvar.u.samp.df.T0$Clade)
    set.seed(30)
    adonis(pvar.u.T0 ~ Clade, data = pvar.u.samp.df.T0)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pvar.u.T0, pvar.u.samp.df.T0$Clade, type = "centroid")) 
      
#Ambient only !!! ####
    pvar.u.bac.AllTime <- subset_samples(Bac.s.rel.u, Species == "Pavona_varians") #all pvar
      
    # pull out otu and tax
    tax  = tax_table(pvar.u.bac.AllTime) 
    OTU = otu_table(pvar.u.bac.AllTime)
    tree=phy_tree(pvar.u.bac.AllTime)
  # sample data
  sd = sample_data(pvar.u.bac.AllTime)
  sd$Treatment<-as.character(sd$Treatment)
  
  sd$Treatment[sd$Treatment == "High" & sd$Time.Point=="T0"] <- "Ambient" 
  sd$Treatment<-as.factor(sd$Treatment)

  pvar.u.bac.Alltime2<-  phyloseq(sd, tax, OTU,tree) #make phylo ojb
  pvar.u.bac.Alltime2.sd <- as(sample_data(pvar.u.bac.Alltime2), "data.frame") #create sample data frame to view

  pvar.u.bac.Amb <- subset_samples(pvar.u.bac.Alltime2, Treatment == "Ambient")

  sample_sums(pvar.u.bac.Amb)

pvar.u.f.bac.Amb = prune_taxa(taxa_sums(pvar.u.bac.Amb) > 0, pvar.u.bac.Amb) #this removes any OTU with 0s
  sample_sums(pvar.u.f.bac.Amb)
    
  pvar.u.Amb <- phyloseq::distance(pvar.u.f.bac.Amb, method = "wunifrac")
     pvar.u.samp.df.Amb <- as(sample_data(pvar.u.f.bac.Amb), "data.frame")
     pvar.u.samp.df.Amb$Code<-paste(pvar.u.samp.df.Amb$Treatment,pvar.u.samp.df.Amb$Clade)
    
     set.seed(30)
    adonis(pvar.u.Amb ~ Clade, data = pvar.u.samp.df.Amb)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pvar.u.Amb, pvar.u.samp.df.Amb$Clade, type = "centroid")) 
      
    
    
    
    
    
    
    
####### pacuta ####

#Pacu ####
pacu.u.bac.T1F <- subset_samples(Bac.s.rel.u.T1F, Species == "Pocillopora_acuta")
pacu.u.bac.T1F = prune_taxa(taxa_sums(pacu.u.bac.T1F) > 0, pacu.u.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
pacu.u.bac.T1 <- subset_samples(pacu.u.bac.T1F, Time.Point == "T1")
pacu.u.bac.T1 = prune_taxa(taxa_sums(pacu.u.bac.T1) > 0, pacu.u.bac.T1) #this removes any OTU with 0s

pacu.u.bac.TF <- subset_samples(pacu.u.bac.T1F, Time.Point == "TF")
pacu.u.bac.TF = prune_taxa(taxa_sums(pacu.u.bac.TF) > 0, pacu.u.bac.TF) #this removes any OTU with 0s

#T1TF
#make wunifrac curtis data matrix -T1 and TF
  pacu.u.T1F <- phyloseq::distance(pacu.u.bac.T1F, method = "wunifrac")
     pacu.u.samp.df.T1F <- as(sample_data(pacu.u.bac.T1F), "data.frame") #sample df
     pacu.u.samp.df.T1F$Code<-paste(pacu.u.samp.df.T1F$Treatment,pacu.u.samp.df.T1F$Time.Point)
          pacu.u.samp.df.T1F$Code2<-paste(pacu.u.samp.df.T1F$Treatment,pacu.u.samp.df.T1F$Time.Point,pacu.u.samp.df.T1F$Clade)

     set.seed(30)
     adonis(pacu.u.T1F ~ Treatment*Time.Point, data = pacu.u.samp.df.T1F) #adonis test no clade
     set.seed(30)
     adonis(pacu.u.T1F ~ Clade*Treatment*Time.Point, data = pacu.u.samp.df.T1F)#adonis test with clade
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pacu.u.T1F, pacu.u.samp.df.T1F$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pacu.u.T1F, pacu.u.samp.df.T1F$Treatment, type = "centroid")) 
     set.seed(30)
    permutest(betadisper(pacu.u.T1F, pacu.u.samp.df.T1F$Code, type = "centroid")) 
     set.seed(30)
    permutest(betadisper(pacu.u.T1F, pacu.u.samp.df.T1F$Code2, type = "centroid")) 
     
     
#T1
  Pacu.u.T1 <- phyloseq::distance(pacu.u.bac.T1, method = "wunifrac") 
      pacu.u.samp.df.T1 <- as(sample_data(pacu.u.bac.T1), "data.frame") 
      pacu.u.samp.df.T1$Code<-paste(pacu.u.samp.df.T1$Treatment, pacu.u.samp.df.T1$Time.Point)
            pacu.u.samp.df.T1$Code2<-paste(pacu.u.samp.df.T1$Treatment, pacu.u.samp.df.T1$Time.Point,pacu.u.samp.df.T1$Clade)

           pacu.u.samp.df.T1$Code<-paste(pacu.u.samp.df.T1$Clade,pacu.u.samp.df.T1$Treatment)

      set.seed(30)
      adonis(Pacu.u.T1 ~ Treatment, data = pacu.u.samp.df.T1)  
      set.seed(30)
      adonis(Pacu.u.T1 ~ Clade*Treatment, data = pacu.u.samp.df.T1)  
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(Pacu.u.T1, pacu.u.samp.df.T1$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(Pacu.u.T1, pacu.u.samp.df.T1$Treatment, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(Pacu.u.T1, pacu.u.samp.df.T1$Code, type = "centroid")) 

     

    
#TF
 
pacu.u.TF <- phyloseq::distance(pacu.u.bac.TF, method = "wunifrac")
     pacu.u.samp.df.TF <- as(sample_data(pacu.u.bac.TF), "data.frame") 
     pacu.u.samp.df.TF$Code<-paste(pacu.u.samp.df.TF$Clade,pacu.u.samp.df.TF$Treatment)
      set.seed(30)
      adonis(pacu.u.TF ~Treatment, data = pacu.u.samp.df.TF)
      set.seed(30)
      adonis(pacu.u.TF ~ Clade*Treatment, data = pacu.u.samp.df.TF)
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pacu.u.TF, pacu.u.samp.df.TF$Clade, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(pacu.u.TF, pacu.u.samp.df.TF$Treatment, type = "centroid")) 
     set.seed(30)
    permutest(betadisper(pacu.u.TF, pacu.u.samp.df.TF$Code, type = "centroid")) 
    
    
 # pacu Bac.s.rel.u.T0   only #### 
    pacu.u.bac.T0 <- subset_samples(Bac.s.rel.u.T0, Species == "Pocillopora_acuta")
  sample_sums(pacu.u.bac.T0)

pacu.u.f.bac.T0 = prune_taxa(taxa_sums(pacu.u.bac.T0) > 0, pacu.u.bac.T0) #this removes any OTU with 0s
  sample_sums(pacu.u.f.bac.T0)
    
  pacu.u.T0 <- phyloseq::distance(pacu.u.f.bac.T0, method = "wunifrac")
     pacu.u.samp.df.T0 <- as(sample_data(pacu.u.bac.T0), "data.frame")
     pacu.u.samp.df.T0$Code<-paste(pacu.u.samp.df.T0$Treatment,pacu.u.samp.df.T0$Clade)
    set.seed(30)
    adonis(pacu.u.T0 ~ Clade, data = pacu.u.samp.df.T0)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pacu.u.T0, pacu.u.samp.df.T0$Clade, type = "centroid"))    
    
#Ambient only !!!
    pacu.u.bac.AllTime <- subset_samples(Bac.s.rel.u, Species == "Pocillopora_acuta") #all pacu
      
    # pull out otu and tax
    tax  = tax_table(pacu.u.bac.AllTime)
    OTU = otu_table(pacu.u.bac.AllTime)
    tree=phy_tree(pacu.u.bac.AllTime)
  # sample data
  sd = sample_data(pacu.u.bac.AllTime)
  sd$Treatment<-as.character(sd$Treatment)
  
  sd$Treatment[sd$Treatment == "High" & sd$Time.Point=="T0"] <- "Ambient" 
  sd$Treatment<-as.factor(sd$Treatment)

  pacu.u.bac.Alltime2<-  phyloseq(sd, tax, OTU,tree) #make phylo ojb
  pacu.u.bac.Alltime2.sd <- as(sample_data(pacu.u.bac.Alltime2), "data.frame") #create sample data frame to view

  pacu.u.bac.Amb <- subset_samples(pacu.u.bac.Alltime2, Treatment == "Ambient")

  sample_sums(pacu.u.bac.Amb)

pacu.u.f.bac.Amb = prune_taxa(taxa_sums(pacu.u.bac.Amb) > 0, pacu.u.bac.Amb) #this removes any OTU with 0s
  sample_sums(pacu.u.f.bac.Amb)
    
  pacu.u.Amb <- phyloseq::distance(pacu.u.f.bac.Amb, method = "wunifrac")
     pacu.u.samp.df.Amb <- as(sample_data(pacu.u.f.bac.Amb), "data.frame")
     pacu.u.samp.df.Amb$Code<-paste(pacu.u.samp.df.Amb$Treatment,pacu.u.samp.df.Amb$Clade)
    
     set.seed(30)
    adonis(pacu.u.Amb ~ Clade, data = pacu.u.samp.df.Amb)

  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(pacu.u.Amb, pacu.u.samp.df.Amb$Clade, type = "centroid")) 
      
        
    
    
    
    
    
    
``` 
#Endo found in what samples and speices
```{r}
 #what endos are in here? all speices
Bac.s.Endo.3 = subset_taxa(Bac.s.rel, Genus=="Endozoicomonas")
OTUendo2 = as(otu_table(Bac.s.Endo.3), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.rel.u.Endo.3)){OTUendo2 <- t(OTUendo2)}
# Coerce to data.frame

OTUendodf2 = as.data.frame(OTUendo2)  
write.csv(OTUendodf2,"OTUendodfMay23.csv")   
    

#by spececies
#mcap
Bac.s.Endo.3Mcap = subset_samples(Bac.s.Endo.3, Species=="Montipora_capitata")
  Bac.s.Endo.3Mcap = prune_taxa(taxa_sums(Bac.s.Endo.3Mcap) > 0, Bac.s.Endo.3Mcap) #this removes any OTU with 0s

OTUendo2Mcap = as(otu_table(Bac.s.Endo.3Mcap), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.Endo.3Mcap)){OTUendo2Mcap <- t(OTUendo2Mcap)}
# Coerce to data.frame

OTUendodf2Mcap = as.data.frame(OTUendo2Mcap)  
write.csv(OTUendodf2Mcap,"OTUendodf2McapMay23.csv") 

#pcomp
Bac.s.Endo.3Pcomp = subset_samples(Bac.s.Endo.3, Species=="Porites_compressa")
  Bac.s.Endo.3Pcomp = prune_taxa(taxa_sums(Bac.s.Endo.3Pcomp) > 0, Bac.s.Endo.3Pcomp) #this removes any OTU with 0s

OTUendo2Pcomp = as(otu_table(Bac.s.Endo.3Pcomp), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.Endo.3Pcomp)){OTUendo2Pcomp <- t(OTUendo2Pcomp)}
# Coerce to data.frame

OTUendodf2Pcomp = as.data.frame(OTUendo2Pcomp)  
write.csv(OTUendodf2Pcomp,"OTUendodf2PcompMay23.csv") 

#Pvar
Bac.s.Endo.3Pvar = subset_samples(Bac.s.Endo.3, Species=="Pavona_varians")
  Bac.s.Endo.3Pvar = prune_taxa(taxa_sums(Bac.s.Endo.3Pvar) > 0, Bac.s.Endo.3Pvar) #this removes any OTU with 0s

OTUendo2Pvar = as(otu_table(Bac.s.Endo.3Pvar), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.Endo.3Pvar)){OTUendo2Pvar <- t(OTUendo2Pvar)}
# Coerce to data.frame

OTUendodf2Pvar = as.data.frame(OTUendo2Pvar)  
write.csv(OTUendodf2Pvar,"OTUendodf2PvarMay23.csv") 

#Pacu
Bac.s.Endo.3Pacu = subset_samples(Bac.s.Endo.3, Species=="Montipora_capitata")
  Bac.s.Endo.3Pacu = prune_taxa(taxa_sums(Bac.s.Endo.3Pacu) > 0, Bac.s.Endo.3Pacu) #this removes any OTU with 0s

OTUendo2Pacu = as(otu_table(Bac.s.Endo.3Pacu), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.Endo.3Pacu)){OTUendo2Pacu <- t(OTUendo2Pacu)}
# Coerce to data.frame

OTUendodf2Pacu = as.data.frame(OTUendo2Pacu)  
write.csv(OTUendodf2Pacu,"OTUendodf2PacuMay23.csv") 
```




Dispersion
```{r}
 
# #dipersion lmer  Bac.s.rel.u
  s.u.samp.df.T01F <- as(sample_data(Bac.s.rel.u), "data.frame") #sample df
  s.u.samp.df.T01F$Code<-paste(s.u.samp.df.T01F$Species,s.u.samp.df.T01F$Clade,s.u.samp.df.T01F$Treatment,s.u.samp.df.T01F$Time.Point)

  Bac.s.u.T01F <- phyloseq::distance(Bac.s.rel.u, method = "wunifrac")

wu.all<-betadisper(Bac.s.u.T01F, s.u.samp.df.T01F$Code, type = "centroid")
 wu.disp<-as.data.frame(wu.all$distances) #extract distances
 names(wu.disp)[names(wu.disp) == "wu.all$distances"] <- "wu.distances" #rename
 Awu.disp.Species<-cbind(wu.disp, s.u.samp.df.T01F) #bind dist to sd
       hist(wu.disp.Species$wu.distances) #none of these are great, do by species
         hist(log10(wu.disp.Species$wu.distances))
        hist(sqrt(wu.disp.Species$wu.distances))
        hist((wu.disp.Species$wu.distances)^1/3)
        hist(1/wu.disp.Species$wu.distances)
#
# wu.disp<-lmer(wu.distances~Species*Treatment*Time.Point+(1|Parent.ID)+(1|Tank.num), data=wu.disp.Species)
# anova(wu.disp)
#   qqPlot(residuals(wu.disp))
#     emm<-emmeans(wu.disp,  ~ Species, adjust="Tukey")
#       cld(emm)
#       pairs(emm)
#
 #T0 make all ambient   
        Awu.disp.Species$Treatment[Awu.disp.Species$Time.Point=="T0"]  <- "Ambient" 

   disp_mean <- ddply(Awu.disp.Species, c("Treatment", "Species", "Time.Point","Clade"), summarise, #summarize cell counts
                   N    = length(wu.distances[!is.na(wu.distances)]), 
                   mean = mean(wu.distances, na.rm=TRUE), 
                   sd   = sd(wu.distances, na.rm=TRUE), 
                   se   = sd / sqrt(N), 
                   max = max(wu.distances, na.rm=TRUE) 
) 
disp_mean #dis 
  disp_mean$Code<-paste(disp_mean$Species,disp_mean$Clade,disp_mean$Treatment,disp_mean$Time.Point)
  disp_mean$Code2<-paste(disp_mean$Species,disp_mean$Clade,disp_mean$Treatment)

   #plot all
disp_mean_plot<-ggplot(data=disp_mean, aes(x=Code, y=mean, group = Code, color=Code)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = T)+
  #scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("Dist to centroid"))) +
  ggtitle("Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));disp_mean_plot 
    disp_mean_plot+facet_wrap(~Species*Clade)

    
#plot by species
#mcap 
    disp_mean.mcap<-subset(disp_mean, Species=="Montipora_capitata")

    mcap_disp_mean_plot<-ggplot(data=disp_mean.mcap, aes(x=Time.Point, y=mean, group = Code2, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = T)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
        ylab("") + #Label the X Axix
  ylim(0, .36) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Dist to centroid"))) +
  # ggtitle("Mcap Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));mcap_disp_mean_plot 
    mcap_disp_mean_plot+facet_wrap(~Clade)
    
 #pcomp
    disp_mean.pcomp<-subset(disp_mean, Species=="Porites_compressa")

    pcomp_disp_mean_plot<-ggplot(data=disp_mean.pcomp, aes(x=Time.Point, y=mean, group = Code2, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = T)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
 xlab("") + #Label the X Axis
        ylab("") + #Label the X Axix
  ylim(0, .36) + #set Y limits  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Dist to centroid"))) +
  # ggtitle("pcomp Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));pcomp_disp_mean_plot 
    pcomp_disp_mean_plot+facet_wrap(~Clade)
    
    
#pvar
    disp_mean.pvar<-subset(disp_mean, Species=="Pavona_varians")

    pvar_disp_mean_plot<-ggplot(data=disp_mean.pvar, aes(x=Time.Point, y=mean, group = Code2, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = T)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
 xlab("") + #Label the X Axis
        ylab("") + #Label the X Axix
  ylim(0, .4) + #set Y limits  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Dist to centroid"))) +
  # ggtitle("pvar Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));pvar_disp_mean_plot 
    pvar_disp_mean_plot+facet_wrap(~Clade)    
    
    #for defense
     disp_mean.pvar.T1<-subset(disp_mean.pvar, Time.Point=="T1")

    disp_mean_plot<-ggplot(data=disp_mean.pvar.T1, aes(x=Time.Point, y=mean, group = Code2, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=6, show.legend = T)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = T)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("Dist to centroid"))) +
  ggtitle("pvar Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));disp_mean_plot 

    
Awu.disp.PvarT1<- subset(Awu.disp.Species, Time.Point=="T1"&Species=="Pavona_varians")

    Rplot1<-ggplot(data=Awu.disp.PvarT1, aes(x=Clade, y=wu.distances, colour=Treatment, group=interaction(Clade, Treatment))) +
  geom_boxplot(position=position_dodge(0.9), lwd=1)+
  geom_jitter(shape=16, position=position_dodge(0.9), size=3)+
  # geom_hline(yintercept=0, linetype="dashed", color="black", size=0.75)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  #scale_y_continuous(limits=c(-0.1, 0.1), labels = scales::number_format(accuracy = 0.01, decimal.mark = '.'))+
  xlab("Time Point")+
      ylim(.2,.45)+
  # ylab(expression(bold(paste("")")))) +
  theme_classic() +
  theme(text = element_text(size = 18, color="black"),
        axis.text = element_text(size = 18, color="black"), 
        legend.position = "top",
        axis.title = element_text(size = 18, color="black", face="bold"), 
        legend.title=element_blank(), 
        legend.text = element_text(size=18), 
        plot.margin=unit(c(1,1,1,1), "cm"), 
        axis.title.y = element_text(margin = margin(t = 0, r = 1, b = 0, l = 1)), 
        axis.title.x = element_text(margin = margin(t = 3, r = 0, b = 0, l = 0)));Rplot1
Rplot2<-Rplot1+theme(legend.position="none")
Rplot3<-Rplot1+facet_wrap(~Species);Rplot3
    
    
#pacu
    disp_mean.pacu<-subset(disp_mean, Species=="Pocillopora_acuta")

    pacu_disp_mean_plot<-ggplot(data=disp_mean.pacu, aes(x=Time.Point, y=mean, group = Code2, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = T)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
 xlab("") + #Label the X Axis
        ylab("") + #Label the X Axix
  ylim(0, .36) + #set Y limits  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Dist to centroid"))) +
  # ggtitle("pacu Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));pacu_disp_mean_plot 
    pacu_disp_mean_plot+facet_wrap(~Clade)
    
    
 
mcap_disp_mean_plot<-mcap_disp_mean_plot+ theme(legend.position = "none") #remove legend
pcomp_disp_mean_plot<-pcomp_disp_mean_plot+ theme(legend.position = "none") #remove legend
pvar_disp_mean_plot<-pvar_disp_mean_plot+ theme(legend.position = "none") #remove legend
pacu_disp_mean_plot<-pacu_disp_mean_plot+ theme(legend.position = "none") #remove legend

grid.arrange(mcap_disp_mean_plot, pcomp_disp_mean_plot, pvar_disp_mean_plot,pacu_disp_mean_plot, nrow = 1)   
    
    
    
    
    
    
    
#T0
    disp_mean.T0<-subset(disp_mean, Time.Point=="T0")

    disp_mean_plot<-ggplot(data=disp_mean.T0, aes(x=Time.Point, y=mean, group = Code2, color=Clade)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Clade), linetype=2, show.legend = T)+
  #scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("Dist to centroid"))) +
  ggtitle("T0 Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));disp_mean_plot 
    disp_mean_plot+facet_wrap(~Species)    
    
    
### Dispersion by treatment only
    
    disp_mean <- ddply(Awu.disp.Species, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                   N    = length(wu.distances[!is.na(wu.distances)]), 
                   mean = mean(wu.distances, na.rm=TRUE), 
                   sd   = sd(wu.distances, na.rm=TRUE), 
                   se   = sd / sqrt(N), 
                   max = max(wu.distances, na.rm=TRUE) 
)
disp_mean #dis 

   #plot all
disp_mean_plot<-ggplot(data=disp_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = T)+
  #scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("Dist to centroid"))) +
  ggtitle("Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));disp_mean_plot 
    disp_mean_plot+facet_wrap(~Species)

    
#plot by species
#mcap
    disp_mean.mcap<-subset(disp_mean, Species=="Montipora_capitata")

  mcap_disp16_plot<-ggplot(data=disp_mean.mcap, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = T)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
 xlab("") + #Label the X Axis
        ylab("") + #Label the X Axix
  ylim(0, .36) + #set Y limits  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Dist to centroid"))) +
  # ggtitle("Mcap Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));mcap_disp16_plot
    
 #pcomp
    disp_mean.pcomp<-subset(disp_mean, Species=="Porites_compressa")

pcomp_disp16_plot<-ggplot(data=disp_mean.pcomp, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
 xlab("") + #Label the X Axis
        ylab("") + #Label the X Axix
  ylim(0, .36) + #set Y limits  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Dist to centroid"))) +
  # ggtitle("pcomp Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));pcomp_disp16_plot
    
    
#pvar
    disp_mean.pvar<-subset(disp_mean, Species=="Pavona_varians")

pvar_disp16_plot<-ggplot(data=disp_mean.pvar, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
 xlab("") + #Label the X Axis
        ylab("") + #Label the X Axix
  ylim(0, .4) + #set Y limits  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Dist to centroid"))) +
  # ggtitle("pvar Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));pvar_disp16_plot
    
    
    
#pacu
    disp_mean.pacu<-subset(disp_mean, Species=="Pocillopora_acuta")

pacu_disp16_plot<- ggplot(data=disp_mean.pacu, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
 xlab("") + #Label the X Axis
        ylab("") + #Label the X Axix
  ylim(0, .36) + #set Y limits  #ylim(0, 11) + #set Y limits
  #scale_y_log10()+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Dist to centroid"))) +
  # ggtitle("pacu Dispersion 16S")+
  theme(plot.title = element_text(size=20, face = "italic"));pacu_disp16_plot
    

 
mcap_disp16_plot<-mcap_disp16_plot+ theme(legend.position = "none") #remove legend
pcomp_disp16_plot<-pcomp_disp16_plot+ theme(legend.position = "none") #remove legend
pvar_disp16_plot<-pvar_disp16_plot+ theme(legend.position = "none") #remove legend
pacu_disp16_plot<-pacu_disp16_plot+ theme(legend.position = "none") #remove legend

grid.arrange(mcap_disp16_plot, pcomp_disp16_plot, pvar_disp16_plot,pacu_disp16_plot, nrow = 1)
 
    
```


#Unifrac NMDS
```{r}      
 Sp.colors <- c("Montipora_capitata"= "#D43F3AFF","Porites_compressa"="#EEA236FF","Pavona_varians"="#5CB85CFF","Pocillopora_acuta"="#46B8DAFF", "High"="darkRed","Ambient"="navyBlue")

#T0 only 
Bac.s.rel.u.T0<-subset_samples(Bac.s.rel.u, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s

unifrac.all.T0 <- ordinate(Bac.s.rel.u.T0, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T0)
scores.ord.u_RelA_stats.T0<-as.data.frame(cbind(vegan::scores(unifrac.all.T0, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.T0 <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df

scores.ord.u_RelA_stats.T0$Treatment <- unifrac.all.sd.T0$Treatment
scores.ord.u_RelA_stats.T0$Species <- unifrac.all.sd.T0$Species
scores.ord.u_RelA_stats.T0$Clade <- unifrac.all.sd.T0$Clade
scores.ord.u_RelA_stats.T0$Code <- paste(scores.ord.u_RelA_stats.T0$Clade, scores.ord.u_RelA_stats.T0$Species)



ggplot(scores.ord.u_RelA_stats.T0, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Species), linetype = 2) +
  ggtitle("NMDS 16s bray T0") +
  theme_classic()
#ggsave("bc_all.pdf")


ggplot(scores.ord.u_RelA_stats.T0, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  #scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
 # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  ggtitle("NMDS 16s unifrac T0") +
  theme_classic()+
  facet_wrap(~Species)

write.csv(scores.ord.u_RelA_stats.T0, "16s_nmds.scores.ord.u_RelA_stats.T0.csv") #export for gauntlet




#T1 only 
Bac.s.rel.u.T1<-subset_samples(Bac.s.rel.u, Time.Point=="T1")
    Bac.s.rel.u.T1 = prune_taxa(taxa_sums(Bac.s.rel.u.T1) > 0, Bac.s.rel.u.T1) #this removes any OTU with 0s

unifrac.all.T1 <- ordinate(Bac.s.rel.u.T1, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T1)
scores.ord.u_RelA_stats.T1<-as.data.frame(cbind(vegan::scores(unifrac.all.T1, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.T1 <- as(sample_data(Bac.s.rel.u.T1), "data.frame") #sample df

scores.ord.u_RelA_stats.T1$Treatment <- unifrac.all.sd.T1$Treatment
scores.ord.u_RelA_stats.T1$Species <- unifrac.all.sd.T1$Species
scores.ord.u_RelA_stats.T1$Clade <- unifrac.all.sd.T1$Clade
scores.ord.u_RelA_stats.T1$Code <- paste(scores.ord.u_RelA_stats.T1$Clade, scores.ord.u_RelA_stats.T1$Species)
scores.ord.u_RelA_stats.T1$Code3 <- paste(scores.ord.u_RelA_stats.T1$Species, scores.ord.u_RelA_stats.T1$Treatment)



p<-ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species, shape=Treatment), size = 4) +
  scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
stat_ellipse(aes(x = NMDS1, y = NMDS2, color=Species, group=Code3,lty = Species)) +
scale_linetype_manual(values=c(1,2,1,2,1,2,1,2))+
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic();p
#ggsave("bc_all.pdf")
p+facet_wrap(~Species)

ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  #scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
 # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic()+
  facet_wrap(~Species)


#for defense

p<-ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2, color=Species,linetype=Treatment)) + 
  geom_point(aes(colour = Species, shape=Treatment), size = 4) +
  scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code3), 
               alpha = 0.25)+
  ggtitle("NMDS 16s  T1") +
  theme_classic()
#ggsave("bc_all.pdf")
p+facet_wrap(~Species)

scores.ord.u_RelA_stats.T1_Pcomp<-subset(scores.ord.u_RelA_stats.T1, Species=="Porites_compressa")
p<-ggplot(scores.ord.u_RelA_stats.T1_Pcomp, aes(x = NMDS1, y = NMDS2, color=Treatment,linetype=Treatment)) + 
  geom_point(aes(colour = Treatment, shape=Treatment), size = 8) +
  scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code3), 
               alpha = 0.25)+
  ggtitle("NMDS 16s  T1 pcomp only") +
  theme_classic()
#ggsave("bc_all.pdf")
p+facet_wrap(~Species)







scores.ord.u_RelA_stats.T1$Code4<-paste(scores.ord.u_RelA_stats.T1$Species,scores.ord.u_RelA_stats.T1$Clade,scores.ord.u_RelA_stats.T1$Treatment)

ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2,color=Clade,linetype=Treatment)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  #scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
 # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code4), 
               alpha = 0.25)+
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic()+
  facet_wrap(~Species)






#TF only 
Bac.s.rel.u.TF<-subset_samples(Bac.s.rel.u, Time.Point=="TF")
    Bac.s.rel.u.TF = prune_taxa(taxa_sums(Bac.s.rel.u.TF) > 0, Bac.s.rel.u.TF) #this removes any OTU with 0s

unifrac.all.TF <- ordinate(Bac.s.rel.u.TF, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.TF)
scores.ord.u_RelA_stats.TF<-as.data.frame(cbind(vegan::scores(unifrac.all.TF, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.TF <- as(sample_data(Bac.s.rel.u.TF), "data.frame") #sample df

scores.ord.u_RelA_stats.TF$Treatment <- unifrac.all.sd.TF$Treatment
scores.ord.u_RelA_stats.TF$Species <- unifrac.all.sd.TF$Species
scores.ord.u_RelA_stats.TF$Clade <- unifrac.all.sd.TF$Clade
scores.ord.u_RelA_stats.TF$Code <- paste(scores.ord.u_RelA_stats.TF$Clade, scores.ord.u_RelA_stats.TF$Species)
scores.ord.u_RelA_stats.TF$Code3 <- paste(scores.ord.u_RelA_stats.TF$Species, scores.ord.u_RelA_stats.TF$Treatment)



ggplot(scores.ord.u_RelA_stats.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species, shape=Treatment), size = 4) +
  scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
stat_ellipse(aes(x = NMDS1, y = NMDS2, color=Species, group=Code3,lty = Code3)) +
scale_linetype_manual(values=c(1,2,1,2,1,2,1,2))+
  ggtitle("NMDS 16s unifrac TF") +
  theme_classic()
#ggsave("bc_all.pdf")


ggplot(scores.ord.u_RelA_stats.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  #scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
 stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  ggtitle("NMDS 16s unifrac TF") +
  theme_classic()+
  facet_wrap(~Species)



#for defense
scores.ord.u_RelA_stats.TF$Code3<-paste(scores.ord.u_RelA_stats.TF$Species, scores.ord.u_RelA_stats.TF$Clade, scores.ord.u_RelA_stats.TF$Clade)
p<-ggplot(scores.ord.u_RelA_stats.TF, aes(x = NMDS1, y = NMDS2, color=Species,linetype=Treatment)) + 
  geom_point(aes(colour = Species, shape=Treatment), size = 4) +
  scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code3), 
               alpha = 0.25)+
  ggtitle("NMDS 16s  TF") +
  theme_classic()
#ggsave("bc_all.pdf")
p+facet_wrap(~Species)


#AMBIENT only 
Bac.s.rel.u.Amb<-subset_samples(Bac.s.rel.u, Treatment=="Ambient")
    Bac.s.rel.u.Amb = prune_taxa(taxa_sums(Bac.s.rel.u.Amb) > 0, Bac.s.rel.u.Amb) #this removes any OTU with 0s

unifrac.all.Amb <- ordinate(Bac.s.rel.u.Amb, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.Amb)
scores.ord.u_RelA_stats.Amb<-as.data.frame(cbind(vegan::scores(unifrac.all.Amb, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd.Amb <- as(sample_data(Bac.s.rel.u.Amb), "data.frame") #sample df
scores.ord.u_RelA_stats.Amb$sample_name.1 <- unifrac.all.sd.Amb$sample_name.1
scores.ord.u_RelA_stats.Amb$Treatment <- unifrac.all.sd.Amb$Treatment
scores.ord.u_RelA_stats.Amb$Species <- unifrac.all.sd.Amb$Species
scores.ord.u_RelA_stats.Amb$Clade <- unifrac.all.sd.Amb$Clade
scores.ord.u_RelA_stats.Amb$Code <- paste(scores.ord.u_RelA_stats.Amb$Clade, scores.ord.u_RelA_stats.Amb$Species)



ggplot(scores.ord.u_RelA_stats.Amb, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  scale_color_manual(values=Sp.colors)+
    geom_text(aes(label=sample_name.1),hjust=-.5, vjust=-.5)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Species), linetype = 2) +
  ggtitle("NMDS 16s bray Ambient only") +
  theme_classic()
#ggsave("bc_all.pdf")


ggplot(scores.ord.u_RelA_stats.Amb, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  #scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
 # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Clade), linetype = 2) +
  ggtitle("NMDS 16s unifrac T0") +
  theme_classic()+
  facet_wrap(~Species)

#Ordination NMDS for all species, T1Tf #### 
unifrac.all <- ordinate(Bac.s.rel.u.T1F, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all)
scores.ord.u_RelA_stats<-as.data.frame(cbind(vegan::scores(unifrac.all, display="sites"))) # these are "points" ask hannah 
     unifrac.all.sd <- as(sample_data(Bac.s.rel.u.T1F), "data.frame") #sample df

scores.ord.u_RelA_stats$Treatment <- unifrac.all.sd$Treatment
scores.ord.u_RelA_stats$Species <- unifrac.all.sd$Species
scores.ord.u_RelA_stats$Clade <- unifrac.all.sd$Clade
scores.ord.u_RelA_stats$Time.Point <- unifrac.all.sd$Time.Point
scores.ord.u_RelA_stats$Code <- paste(scores.ord.u_RelA_stats$Clade, scores.ord.u_RelA_stats$Species)



ggplot(scores.ord.u_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS 16s bray") +
  theme_classic()
#ggsave("bc_all.pdf")

ggplot(scores.ord.u_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code, shape=Species), size = 4) +
  scale_color_manual(values=c("darkblue","orange", "darkred","red","yellow","green","darkgreen","blue"))+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code), linetype = 2) +
  ggtitle("NMDS 16s unifrac") +
  theme_classic()+
  facet_wrap(~Species)
#ggsave("bc_all.pdf")

scores.ord.u_RelA_stats.T1<-subset(scores.ord.u_RelA_stats, Time.Point=="T1")

ggplot(scores.ord.u_RelA_stats.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code, shape=Treatment), size = 4) +
  scale_color_manual(values=c("darkblue","orange", "darkred","red","yellow","green","darkgreen","blue"))+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code), linetype = 2) +
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic()
#ggsave("bc_all.pdf")

scores.ord.u_RelA_stats.PacuPvar<-subset(scores.ord.u_RelA_stats, Species=="Pavona_varians"|Species=="Pocillopora_acuta")
# scores.ord.u_RelA_stats.PacuPvar.T1<-subset(scores.ord.u_RelA_stats.PacuPvar, Time.Point=="T1")
scores.ord.u_RelA_stats.PacuPvar.Amb<-subset(scores.ord.u_RelA_stats.PacuPvar, Treatment=="Ambient")

ggplot(scores.ord.u_RelA_stats.PacuPvar.Amb, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code, shape=Treatment), size = 4) +
  scale_color_manual(values=c("darkblue","orange", "green","purple","yellow","green","darkgreen","blue"))+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code), linetype = 2) +
  ggtitle("NMDS 16s unifrac T1") +
  theme_classic()
#ggsave("bc_all.pdf")









#### mcap #####   
## create palette for parent.ID and treatments
mcap.u.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", 
                        "Ambient" = "#4575B4", "High" = "#D73027",
                        "C Ambient"="#DDCC77","C High"="#999933", "CD Ambient"="#44AA99", "CD High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 



#T1 and TF 
ord.mcap2 <- ordinate(mcap.u.f.bac.T1F, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.mcap2)

scores.mcap.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.mcap2, display="sites"))) # these are "points" ask hannah 
scores.mcap.u.samp.df$Time.Point <- mcap.u.samp.df.T1F$Time.Point
scores.mcap.u.samp.df$Parent.ID <- mcap.u.samp.df.T1F$Parent.ID
scores.mcap.u.samp.df$Treatment <- mcap.u.samp.df.T1F$Treatment
scores.mcap.u.samp.df$Clade <- mcap.u.samp.df.T1F$Clade
scores.mcap.u.samp.df$Code <- paste(mcap.u.samp.df.T1F$Parent.ID,mcap.u.samp.df.T1F$Time.Point)
scores.mcap.u.samp.df$Code2 <- paste(mcap.u.samp.df.T1F$Treatment,mcap.u.samp.df.T1F$Time.Point)
scores.mcap.u.samp.df$Code3 <- paste(mcap.u.samp.df.T1F$Treatment,mcap.u.samp.df.T1F$Time.Point,mcap.u.samp.df.T1F$Parent.ID)

ggplot(scores.mcap.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment, fill=Treatment), size = 5) + 
  #scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=mcap.u.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()

ggplot(scores.mcap.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Treatment, color = Treatment), size = 5) + 
  #scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=mcap.u.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()

#T1
ord.mcap.u.T1 <- ordinate(mcap.u.f.bac.T1, method="NMDS",  "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.mcap.u.T1)

ord.mcap.u.T1.df<-as.data.frame(cbind(vegan::scores(ord.mcap.u.T1, display="sites"))) # these are "points" ask hannah 
ord.mcap.u.T1.df$Time.Point <- mcap.u.samp.df.T1$Time.Point
ord.mcap.u.T1.df$Parent.ID <- mcap.u.samp.df.T1$Parent.ID
ord.mcap.u.T1.df$Treatment <- mcap.u.samp.df.T1$Treatment
ord.mcap.u.T1.df$Clade <- mcap.u.samp.df.T1$Clade
ord.mcap.u.T1.df$Code <- paste(mcap.u.samp.df.T1$Parent.ID,mcap.u.samp.df.T1$Time.Point)
ord.mcap.u.T1.df$Code2 <- paste(mcap.u.samp.df.T1$Clade,mcap.u.samp.df.T1$Treatment)
ord.mcap.u.T1.df$Code2 <- as.factor(ord.mcap.u.T1.df$Code2)


ggplot(ord.mcap.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=mcap.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata T1") +
  theme_classic()

ggplot(ord.mcap.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=mcap.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata T1") +
  theme_classic()



#TF only
set.seed(30)
ord.u.mcapTF <- ordinate(mcap.u.f.bac.TF, "NMDS",  "unifrac",weighted=TRUE,set.seed(30)) 
stressplot(ord.u.mcapTF)
ord.mcap.u.TF.df<-as.data.frame(cbind(vegan::scores(ord.u.mcapTF, display="sites"))) # these are "points" ask hannah 
ord.mcap.u.TF.df$Treatment <- mcap.u.samp.df.TF$Treatment
ord.mcap.u.TF.df$Parent.ID <- mcap.u.samp.df.TF$Parent.ID
ord.mcap.u.TF.df$Clade <- mcap.u.samp.df.TF$Clade
ord.mcap.u.TF.df$CladeTreat <- as.factor(paste(mcap.u.samp.df.TF$Clade,mcap.u.samp.df.TF$Treatment))
ord.mcap.u.TF.df$Code <- as.factor(paste(mcap.u.samp.df.TF$Parent.ID,mcap.u.samp.df.TF$Treatment))
ord.mcap.u.TF.df$Code2 <- paste(mcap.u.samp.df.TF$Clade,mcap.u.samp.df.TF$Treatment)
ord.mcap.u.TF.df$Code2 <- as.factor(ord.mcap.u.TF.df$Code2)

ggplot(ord.mcap.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=mcap.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata TF") +
  theme_classic()

ggplot(ord.mcap.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=mcap.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata TF") +
  theme_classic()




#Ambient only mcap.u.bac.AmbT01F

set.seed(30)
ord.u.mcap.AmbT01F <- ordinate(mcap.u.bac.AmbT01F, "NMDS",  "unifrac",weighted=TRUE,set.seed(30)) 
stressplot(ord.u.mcap.AmbT01F)
ord.mcap.u.AmbT01F.df<-as.data.frame(cbind(vegan::scores(ord.u.mcap.AmbT01F, display="sites"))) # these are "points" ask hannah 
ord.mcap.u.AmbT01F.df$Treatment <- mcap.u.samp.df.AmbT01F$Treatment
ord.mcap.u.AmbT01F.df$Parent.ID <- mcap.u.samp.df.AmbT01F$Parent.ID
ord.mcap.u.AmbT01F.df$Time.Point <- mcap.u.samp.df.AmbT01F$Time.Point
ord.mcap.u.AmbT01F.df$CladeTreat <- as.factor(paste(mcap.u.samp.df.AmbT01F$Clade,mcap.u.samp.df.AmbT01F$Treatment))
ord.mcap.u.AmbT01F.df$Code <- as.factor(paste(mcap.u.samp.df.AmbT01F$Parent.ID,mcap.u.samp.df.AmbT01F$Treatment))
ord.mcap.u.AmbT01F.df$Code2 <- paste(mcap.u.samp.df.AmbT01F$Clade,mcap.u.samp.df.AmbT01F$Treatment)
ord.mcap.u.AmbT01F.df$Code2 <- as.factor(ord.mcap.u.AmbT01F.df$Code2)

ggplot(ord.mcap.u.AmbT01F.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Time.Point, shape=Time.Point,size = 4) )+
  #scale_color_manual(values=mcap.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Time.Point, lty=factor(Time.Point))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata amb only ") +
  theme_classic()



####Pcomp #### 
  
pcomp.u.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue",                         "Ambient" = "#4575B4", "High" = "#D73027","C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred", "Ambient T1"="darkred", "Ambient TF"="darkblue", "High T1"="pink","High TF"="darkred") 




#T1 and TF
ord.pcomp2 <- ordinate(pcomp.u.bac.T1F, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pcomp2)

scores.pcomp.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.pcomp2, display="sites"))) # 
scores.pcomp.u.samp.df$Time.Point <- pcomp.u.samp.df.T1F$Time.Point
scores.pcomp.u.samp.df$Parent.ID <- pcomp.u.samp.df.T1F$Parent.ID
scores.pcomp.u.samp.df$Treatment <- pcomp.u.samp.df.T1F$Treatment
scores.pcomp.u.samp.df$Clade <- pcomp.u.samp.df.T1F$Clade
scores.pcomp.u.samp.df$Code <- paste(pcomp.u.samp.df.T1F$Parent.ID,pcomp.u.samp.df.T1F$Time.Point)
scores.pcomp.u.samp.df$Code2 <- paste(pcomp.u.samp.df.T1F$Treatment,pcomp.u.samp.df.T1F$Time.Point)
scores.pcomp.u.samp.df$Code3 <- paste(pcomp.u.samp.df.T1F$Treatment,pcomp.u.samp.df.T1F$Time.Point,pcomp.u.samp.df.T1F$Parent.ID)

ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    #scale_color_manual(values=pcomp.u.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("Pcomp T1TF") +
  theme_classic()

ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2, shape=Treatment, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("Pcomp T1TF") +
  theme_classic()

ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
      geom_text(aes(label=Parent.ID),hjust=-.5, vjust=-.5)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pcomp T1TF") +
  theme_classic()

#pcomp with 2 problem samples remvoed, 319 T1H and 317 TFA
#scores.pcomp.u.samp.df<-scores.pcomp.u.samp.df[!(scores.pcomp.u.samp.df$Code3=="High T1 319"),]
scores.pcomp.u.samp.df<-scores.pcomp.u.samp.df[!(scores.pcomp.u.samp.df$Code3=="Ambient TF 317"),]

ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
      geom_text(aes(label=Parent.ID),hjust=-.5, vjust=-.5)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pcomp T1TF") +
  theme_classic()



#T1
ord.pcomp.u.T1 <- ordinate(pcomp.u.bac.T1, method="NMDS",  "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pcomp.u.T1)

ord.pcomp.u.T1.df<-as.data.frame(cbind(vegan::scores(ord.pcomp.u.T1, display="sites"))) 
ord.pcomp.u.T1.df$Time.Point <- pcomp.u.samp.df.T1$Time.Point
ord.pcomp.u.T1.df$Parent.ID <- pcomp.u.samp.df.T1$Parent.ID
ord.pcomp.u.T1.df$Treatment <- pcomp.u.samp.df.T1$Treatment
ord.pcomp.u.T1.df$Clade <- pcomp.u.samp.df.T1$Clade
ord.pcomp.u.T1.df$Code <- paste(pcomp.u.samp.df.T1$Parent.ID,pcomp.u.samp.df.T1$Time.Point)
ord.pcomp.u.T1.df$Code2 <- paste(pcomp.u.samp.df.T1$Clade,pcomp.u.samp.df.T1$Treatment)
ord.pcomp.u.T1.df$Code2 <- as.factor(ord.pcomp.u.T1.df$Code2)


ggplot(ord.pcomp.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    # scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pcomp T1") +
  theme_classic()

ggplot(ord.pcomp.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pcomp T1") +
  theme_classic()

#TF only
set.seed(30)
ord.u.pcompTF <- ordinate(pcomp.u.bac.TF, "NMDS",  "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.u.pcompTF)
ord.pcomp.u.TF.df<-as.data.frame(cbind(vegan::scores(ord.u.pcompTF, display="sites"))) # these are "points" ask hannah 
ord.pcomp.u.TF.df$Treatment <- pcomp.u.samp.df.TF$Treatment
ord.pcomp.u.TF.df$Parent.ID <- pcomp.u.samp.df.TF$Parent.ID
ord.pcomp.u.TF.df$Clade <- pcomp.u.samp.df.TF$Clade
ord.pcomp.u.TF.df$CladeTreat <- as.factor(paste(pcomp.u.samp.df.TF$Clade,pcomp.u.samp.df.TF$Treatment))
ord.pcomp.u.TF.df$Code <- as.factor(paste(pcomp.u.samp.df.TF$Parent.ID,pcomp.u.samp.df.TF$Treatment))
ord.pcomp.u.TF.df$Code2 <- paste(pcomp.u.samp.df.TF$Clade,pcomp.u.samp.df.TF$Treatment)
ord.pcomp.u.TF.df$Code2 <- as.factor(ord.pcomp.u.TF.df$Code2)

ggplot(ord.pcomp.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  #scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P comp TF") +
  theme_classic()

ggplot(ord.pcomp.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P comp TF") +
  theme_classic()



####Pvar #### 
pvar.u.Parent.colors <- c("325" = "green4",  "353" = "firebrick3", "356" ="deeppink4", "358" = "orange", "361" = "green1","354" = "gold",  "351"= "hotpink", "355" ="darkslateblue", "357" = "cyan", "359" = "darkorange","360" ="darkorchid1", "362" = "lightblue", 
 "Ambient" = "#4575B4", "High" = "#D73027",                        
 "C1 Ambient"="#DDCC77","C1 High"="#999933", "C27 Ambient"="#44AA99", "C27 High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 


 



#T1 and TF
ord.u.pvar2 <- ordinate(pvar.u.u.bac.T1F, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.u.pvar2)

scores.pvar.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.u.pvar2, display="sites"))) # 
scores.pvar.u.samp.df$Time.Point <- pvar.u.u.samp.df.T1F$Time.Point
scores.pvar.u.samp.df$Parent.ID <- pvar.u.u.samp.df.T1F$Parent.ID
scores.pvar.u.samp.df$Treatment <- pvar.u.u.samp.df.T1F$Treatment
scores.pvar.u.samp.df$Clade <- pvar.u.u.samp.df.T1F$Clade

scores.pvar.u.samp.df$Code <- paste(pvar.u.u.samp.df.T1F$Parent.ID,pvar.u.u.samp.df.T1F$Time.Point)
scores.pvar.u.samp.df$Code2 <- paste(pvar.u.u.samp.df.T1F$Treatment,pvar.u.u.samp.df.T1F$Time.Point)
scores.pvar.u.samp.df$Code3 <- paste(pvar.u.u.samp.df.T1F$Treatment,pvar.u.u.samp.df.T1F$Time.Point,pvar.u.u.samp.df.T1F$Parent.ID)

ggplot(scores.pvar.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=pvar.u.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("pvar T1TF") +
  theme_classic()

ggplot(scores.pvar.u.samp.df, aes(x = NMDS1, y = NMDS2, shape=Treatment, fill=Treatment)) + 
 geom_point(aes(size = 5, color=Treatment) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=pvar.u.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("pvar T1TF") +
  theme_classic()
#ggsave("pvarT1its2.pdf")

ggplot(scores.pvar.u.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("Pvar T1TF") +
  theme_classic()


ggplot(scores.pvar.u.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
    scale_linetype_manual(values=c(2,3))+
  ggtitle("P var T1TF") +
  theme_classic()

#T1
ord.pvar.u.T1 <- ordinate(pvar.u.u.bac.T1, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pvar.u.T1)

ord.pvar.u.T1.df<-as.data.frame(cbind(vegan::scores(ord.pvar.u.T1, display="sites"))) 
ord.pvar.u.T1.df$Time.Point <- pvar.u.u.samp.df.T1$Time.Point
ord.pvar.u.T1.df$Parent.ID <- pvar.u.u.samp.df.T1$Parent.ID
ord.pvar.u.T1.df$Treatment <- pvar.u.u.samp.df.T1$Treatment
ord.pvar.u.T1.df$Clade <- pvar.u.u.samp.df.T1$Clade
ord.pvar.u.T1.df$Code <- paste(pvar.u.u.samp.df.T1$Parent.ID,pvar.u.u.samp.df.T1$Time.Point)
ord.pvar.u.T1.df$Code2 <- paste(pvar.u.u.samp.df.T1$Clade,pvar.u.u.samp.df.T1$Treatment)
ord.pvar.u.T1.df$Code2 <- as.factor(ord.pvar.u.T1.df$Code2)


ggplot(ord.pvar.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour =  Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pvar.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("pvar T1") +
  theme_classic()

  #T1 by clades C1
  pvar.u.u.bac.T1.C1<-subset_samples(pvar.u.u.bac.T1, Clade=="C1")
  pvar.u.u.bac.T1.samp.C1.df <- as(sample_data(pvar.u.u.bac.T1.C1), "data.frame") #sample df

  ord.pvar.u.T1.C1 <- ordinate(pvar.u.u.bac.T1.C1, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
  stressplot(ord.pvar.u.T1.C1)
  
  ord.pvar.u.T1.C1.df<-as.data.frame(cbind(vegan::scores(ord.pvar.u.T1.C1, display="sites"))) 
  ord.pvar.u.T1.C1.df$Time.Point <- pvar.u.u.bac.T1.samp.C1.df$Time.Point
  ord.pvar.u.T1.C1.df$Parent.ID <- pvar.u.u.bac.T1.samp.C1.df$Parent.ID
  ord.pvar.u.T1.C1.df$Treatment <- pvar.u.u.bac.T1.samp.C1.df$Treatment
  ord.pvar.u.T1.C1.df$Clade <- pvar.u.u.bac.T1.samp.C1.df$Clade
  ord.pvar.u.T1.C1.df$Code <- paste(ord.pvar.u.T1.C1.df$Parent.ID,ord.pvar.u.T1.C1.df$Time.Point)
  ord.pvar.u.T1.C1.df$Code2 <- paste(ord.pvar.u.T1.C1.df$Clade,ord.pvar.u.T1.C1.df$Treatment)
  ord.pvar.u.T1.C1.df$Code2 <- as.factor(ord.pvar.u.T1.C1.df$Code2)


  ggplot(ord.pvar.u.T1.C1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pvar.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("pvar T1. C1") +
  theme_classic()
  
#T1 by clades C27
  pvar.u.u.bac.T1.C27<-subset_samples(pvar.u.u.bac.T1, Clade=="C27")
  pvar.u.u.bac.T1.samp.C27.df <- as(sample_data(pvar.u.u.bac.T1.C27), "data.frame") #sample df

  ord.pvar.u.T1.C27 <- ordinate(pvar.u.u.bac.T1.C27, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
  stressplot(ord.pvar.u.T1.C27)
  
  ord.pvar.u.T1.C27.df<-as.data.frame(cbind(vegan::scores(ord.pvar.u.T1.C27, display="sites"))) 
  ord.pvar.u.T1.C27.df$Time.Point <- pvar.u.u.bac.T1.samp.C27.df$Time.Point
  ord.pvar.u.T1.C27.df$Parent.ID <- pvar.u.u.bac.T1.samp.C27.df$Parent.ID
  ord.pvar.u.T1.C27.df$Treatment <- pvar.u.u.bac.T1.samp.C27.df$Treatment
  ord.pvar.u.T1.C27.df$Clade <- pvar.u.u.bac.T1.samp.C27.df$Clade
  ord.pvar.u.T1.C27.df$Code <- paste(ord.pvar.u.T1.C27.df$Parent.ID,ord.pvar.u.T1.C27.df$Time.Point)
  ord.pvar.u.T1.C27.df$Code2 <- paste(ord.pvar.u.T1.C27.df$Clade,ord.pvar.u.T1.C27.df$Treatment)
  ord.pvar.u.T1.C27.df$Code2 <- as.factor(ord.pvar.u.T1.C27.df$Code2)


  ggplot(ord.pvar.u.T1.C27.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pvar.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("pvar T1. C27") +
  theme_classic()

#TF only
set.seed(30)
ord.u.pvarTF <- ordinate(pvar.u.u.bac.TF, "NMDS",  "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.u.pvarTF)
ord.pvar.u.TF.df<-as.data.frame(cbind(vegan::scores(ord.u.pvarTF, display="sites"))) # these are "points" ask hannah 
ord.pvar.u.TF.df$Treatment <- pvar.u.u.samp.df.TF$Treatment
ord.pvar.u.TF.df$Parent.ID <- pvar.u.u.samp.df.TF$Parent.ID
ord.pvar.u.TF.df$Clade <- pvar.u.u.samp.df.TF$Clade
ord.pvar.u.TF.df$Time.Point <- pvar.u.u.samp.df.TF$Time.Point

ord.pvar.u.TF.df$CladeTreat <- as.factor(paste(pvar.u.u.samp.df.TF$Clade,pvar.u.u.samp.df.TF$Treatment))
ord.pvar.u.TF.df$Code <- as.factor(paste(pvar.u.u.samp.df.TF$Parent.ID,pvar.u.u.samp.df.TF$Treatment))
ord.pvar.u.TF.df$Code2 <- paste(pvar.u.u.samp.df.TF$Clade,pvar.u.u.samp.df.TF$Treatment)
ord.pvar.u.TF.df$Code2 <- as.factor(ord.pvar.u.TF.df$Code2)

ggplot(ord.pvar.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pvar.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pvar TF") +
  theme_classic()


ggplot(ord.pvar.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pvar.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("pvar TF") +
  theme_classic()

 #TF by clades C1
  pvar.u.u.bac.TF.C1<-subset_samples(pvar.u.u.bac.TF, Clade=="C1")
  pvar.u.u.bac.TF.samp.C1.df <- as(sample_data(pvar.u.u.bac.TF.C1), "data.frame") #sample df

  ord.pvar.u.TF.C1 <- ordinate(pvar.u.u.bac.TF.C1, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
  stressplot(ord.pvar.u.TF.C1)
  
  ord.pvar.u.TF.C1.df<-as.data.frame(cbind(vegan::scores(ord.pvar.u.TF.C1, display="sites"))) 
  ord.pvar.u.TF.C1.df$Time.Point <- pvar.u.u.bac.TF.samp.C1.df$Time.Point
  ord.pvar.u.TF.C1.df$Parent.ID <- pvar.u.u.bac.TF.samp.C1.df$Parent.ID
  ord.pvar.u.TF.C1.df$Treatment <- pvar.u.u.bac.TF.samp.C1.df$Treatment
  ord.pvar.u.TF.C1.df$Clade <- pvar.u.u.bac.TF.samp.C1.df$Clade
  ord.pvar.u.TF.C1.df$Code <- paste(ord.pvar.u.TF.C1.df$Parent.ID,ord.pvar.u.TF.C1.df$Time.Point)
  ord.pvar.u.TF.C1.df$Code2 <- paste(ord.pvar.u.TF.C1.df$Clade,ord.pvar.u.TF.C1.df$Treatment)
  ord.pvar.u.TF.C1.df$Code2 <- as.factor(ord.pvar.u.TF.C1.df$Code2)


  ggplot(ord.pvar.u.TF.C1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pvar.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("pvar TF. C1") +
  theme_classic()
  
#TF by clades C27
  pvar.u.u.bac.TF.C27<-subset_samples(pvar.u.u.bac.TF, Clade=="C27")
  pvar.u.u.bac.TF.samp.C27.df <- as(sample_data(pvar.u.u.bac.TF.C27), "data.frame") #sample df

  ord.pvar.u.TF.C27 <- ordinate(pvar.u.u.bac.TF.C27, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
  stressplot(ord.pvar.u.TF.C27)
  
  ord.pvar.u.TF.C27.df<-as.data.frame(cbind(vegan::scores(ord.pvar.u.TF.C27, display="sites"))) 
  ord.pvar.u.TF.C27.df$Time.Point <- pvar.u.u.bac.TF.samp.C27.df$Time.Point
  ord.pvar.u.TF.C27.df$Parent.ID <- pvar.u.u.bac.TF.samp.C27.df$Parent.ID
  ord.pvar.u.TF.C27.df$Treatment <- pvar.u.u.bac.TF.samp.C27.df$Treatment
  ord.pvar.u.TF.C27.df$Clade <- pvar.u.u.bac.TF.samp.C27.df$Clade
  ord.pvar.u.TF.C27.df$Code <- paste(ord.pvar.u.TF.C27.df$Parent.ID,ord.pvar.u.TF.C27.df$Time.Point)
  ord.pvar.u.TF.C27.df$Code2 <- paste(ord.pvar.u.TF.C27.df$Clade,ord.pvar.u.TF.C27.df$Treatment)
  ord.pvar.u.TF.C27.df$Code2 <- as.factor(ord.pvar.u.TF.C27.df$Code2)


  ggplot(ord.pvar.u.TF.C27.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pvar.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("pvar TF. C27") +
  theme_classic()


#### Pacu ####
## create palette for parent.ID and treatments
Pacu.u.Parent.colors <- c("305" = "green4",  "307" = "firebrick3", "309" ="deeppink4", "311" = "orange", "301" = "green1","302" = "gold",  "308"= "hotpink", "310" ="darkslateblue", "312" = "cyan", "303" = "darkorange","304" ="darkorchid1", "306" = "lightblue", 
 "Ambient" = "#4575B4", "High" = "#D73027",                        
                        "C42 Ambient"="#DDCC77","C42 High"="#999933", "CCC Ambient"="#44AA99", "CCC High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 



#T1 and TF
ord.u.pacu2 <- ordinate(pacu.u.bac.T1F, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.u.pacu2)

scores.pacu.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.u.pacu2, display="sites"))) # 
scores.pacu.u.samp.df$Time.Point <- pacu.u.samp.df.T1F$Time.Point
scores.pacu.u.samp.df$Parent.ID <- pacu.u.samp.df.T1F$Parent.ID
scores.pacu.u.samp.df$Treatment <- pacu.u.samp.df.T1F$Treatment
scores.pacu.u.samp.df$Clade <- pacu.u.samp.df.T1F$Clade

scores.pacu.u.samp.df$Code <- paste(pacu.u.samp.df.T1F$Parent.ID,pacu.u.samp.df.T1F$Time.Point)
scores.pacu.u.samp.df$Code2 <- paste(pacu.u.samp.df.T1F$Treatment,pacu.u.samp.df.T1F$Time.Point)
scores.pacu.u.samp.df$Code3 <- paste(pacu.u.samp.df.T1F$Treatment,pacu.u.samp.df.T1F$Time.Point,pacu.u.samp.df.T1F$Parent.ID)

ggplot(scores.pacu.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Pacu.u.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("pacu T1TF") +
  theme_classic()

ggplot(scores.pacu.u.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("Pacu T1TF") +
  theme_classic()


#T1
ord.pacu.u.T1 <- ordinate(pacu.u.bac.T1, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pacu.u.T1)

ord.pacu.u.T1.df<-as.data.frame(cbind(vegan::scores(ord.pacu.u.T1, display="sites"))) 
ord.pacu.u.T1.df$Time.Point <- pacu.u.samp.df.T1$Time.Point
ord.pacu.u.T1.df$Parent.ID <- pacu.u.samp.df.T1$Parent.ID
ord.pacu.u.T1.df$Treatment <- pacu.u.samp.df.T1$Treatment
ord.pacu.u.T1.df$Clade <- pacu.u.samp.df.T1$Clade
ord.pacu.u.T1.df$Code <- paste(pacu.u.samp.df.T1$Parent.ID,pacu.u.samp.df.T1$Time.Point)
ord.pacu.u.T1.df$Code2 <- paste(pacu.u.samp.df.T1$Clade,pacu.u.samp.df.T1$Treatment)
ord.pacu.u.T1.df$Code2 <- as.factor(pacu.u.samp.df.T1$Code2)


ggplot(ord.pacu.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("pacu T1") +
  theme_classic()

ggplot(ord.pacu.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Clade, shape=Treatment,size = 4) )+
   # geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  #scale_color_manual(values=Pacu.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
    scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("Pacu T1") +
  theme_classic()

#TF only pacu.u.bac.TF
set.seed(30)
ord.u.pacuTF <- ordinate(pacu.u.bac.TF,"NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.u.pacuTF)
ord.pacu.u.TF.df<-as.data.frame(cbind(vegan::scores(ord.u.pacuTF, display="sites"))) # these are "points" ask hannah 
ord.pacu.u.TF.df$Treatment <- pacu.u.samp.df.TF$Treatment
ord.pacu.u.TF.df$Parent.ID <- pacu.u.samp.df.TF$Parent.ID
ord.pacu.u.TF.df$Clade <- pacu.u.samp.df.TF$Clade
ord.pacu.u.TF.df$Time.Point <- pacu.u.samp.df.TF$Time.Point

ord.pacu.u.TF.df$CladeTreat <- as.factor(paste(pacu.u.samp.df.TF$Clade,pacu.u.samp.df.TF$Treatment))
ord.pacu.u.TF.df$Code <- as.factor(paste(pacu.u.samp.df.TF$Parent.ID,pacu.u.samp.df.TF$Treatment))
ord.pacu.u.TF.df$Code2 <- paste(pacu.u.samp.df.TF$Clade,pacu.u.samp.df.TF$Treatment)
ord.pacu.u.TF.df$Code2 <- as.factor(pacu.u.samp.df.TF$Code2)

ggplot(ord.pacu.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("pacu TF") +
  theme_classic()

ggplot(ord.pacu.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
    scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("Pacu TF") +
  theme_classic()


```
P comp again with Endo removed. havent figured it out yets
```{r}
#T1 and TF
#remove endo
pcomp.u.bac.T1F.NoEndo<-pcomp.u.bac.T1F #make copy
# Define the taxa you don't want like this:
badTaxa = c("Endozoicomonadaceae")
allTaxa = taxa_names(pcomp.u.bac.T1F.NoEndo)
allTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
ex1 = prune_taxa(TAX, pcomp.u.bac.T1F.NoEndo)
# new phyloseq object with just the taxa you kept.
ex1

ord.pcomp2 <- ordinate(pcomp.u.bac.T1F, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pcomp2)

scores.pcomp.u.samp.df<-as.data.frame(cbind(vegan::scores(ord.pcomp2, display="sites"))) # 
scores.pcomp.u.samp.df$Time.Point <- pcomp.u.samp.df.T1F$Time.Point
scores.pcomp.u.samp.df$Parent.ID <- pcomp.u.samp.df.T1F$Parent.ID
scores.pcomp.u.samp.df$Treatment <- pcomp.u.samp.df.T1F$Treatment
scores.pcomp.u.samp.df$Clade <- pcomp.u.samp.df.T1F$Clade
scores.pcomp.u.samp.df$Code <- paste(pcomp.u.samp.df.T1F$Parent.ID,pcomp.u.samp.df.T1F$Time.Point)
scores.pcomp.u.samp.df$Code2 <- paste(pcomp.u.samp.df.T1F$Treatment,pcomp.u.samp.df.T1F$Time.Point)
scores.pcomp.u.samp.df$Code3 <- paste(pcomp.u.samp.df.T1F$Treatment,pcomp.u.samp.df.T1F$Time.Point,pcomp.u.samp.df.T1F$Parent.ID)

ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    #scale_color_manual(values=pcomp.u.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("Pcomp T1TF") +
  theme_classic()

ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2, shape=Treatment, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("Pcomp T1TF") +
  theme_classic()

ggplot(scores.pcomp.u.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pcomp T1TF") +
  theme_classic()

#T1
ord.pcomp.u.T1 <- ordinate(pcomp.u.bac.T1, method="NMDS",  "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.pcomp.u.T1)

ord.pcomp.u.T1.df<-as.data.frame(cbind(vegan::scores(ord.pcomp.u.T1, display="sites"))) 
ord.pcomp.u.T1.df$Time.Point <- pcomp.u.samp.df.T1$Time.Point
ord.pcomp.u.T1.df$Parent.ID <- pcomp.u.samp.df.T1$Parent.ID
ord.pcomp.u.T1.df$Treatment <- pcomp.u.samp.df.T1$Treatment
ord.pcomp.u.T1.df$Clade <- pcomp.u.samp.df.T1$Clade
ord.pcomp.u.T1.df$Code <- paste(pcomp.u.samp.df.T1$Parent.ID,pcomp.u.samp.df.T1$Time.Point)
ord.pcomp.u.T1.df$Code2 <- paste(pcomp.u.samp.df.T1$Clade,pcomp.u.samp.df.T1$Treatment)
ord.pcomp.u.T1.df$Code2 <- as.factor(ord.pcomp.u.T1.df$Code2)


ggplot(ord.pcomp.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pcomp T1") +
  theme_classic()

ggplot(ord.pcomp.u.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pcomp T1") +
  theme_classic()

#TF only
set.seed(30)
ord.u.pcompTF <- ordinate(pcomp.u.bac.TF, "NMDS",  "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(ord.u.pcompTF)
ord.pcomp.u.TF.df<-as.data.frame(cbind(vegan::scores(ord.u.pcompTF, display="sites"))) # these are "points" ask hannah 
ord.pcomp.u.TF.df$Treatment <- pcomp.u.samp.df.TF$Treatment
ord.pcomp.u.TF.df$Parent.ID <- pcomp.u.samp.df.TF$Parent.ID
ord.pcomp.u.TF.df$Clade <- pcomp.u.samp.df.TF$Clade
ord.pcomp.u.TF.df$CladeTreat <- as.factor(paste(pcomp.u.samp.df.TF$Clade,pcomp.u.samp.df.TF$Treatment))
ord.pcomp.u.TF.df$Code <- as.factor(paste(pcomp.u.samp.df.TF$Parent.ID,pcomp.u.samp.df.TF$Treatment))
ord.pcomp.u.TF.df$Code2 <- paste(pcomp.u.samp.df.TF$Clade,pcomp.u.samp.df.TF$Treatment)
ord.pcomp.u.TF.df$Code2 <- as.factor(ord.pcomp.u.TF.df$Code2)

ggplot(ord.pcomp.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  #scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P comp TF") +
  theme_classic()

ggplot(ord.pcomp.u.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P comp TF") +
  theme_classic()
```
Unifrac with field samps
```{r} 
 # Overhead view including field ####  
unifrac.all.field <- ordinate(Bac.seq.all, method="NMDS", "unifrac",weighted=TRUE, set.seed(40)) 
stressplot(unifrac.all.field)
scores.ord.u_ALLfield_stats<-as.data.frame(cbind(vegan::scores(unifrac.all.field, display="sites"))) 
     unifrac.samp.field.sd <- as(sample_data(Bac.seq.all), "data.frame") #sample df

scores.ord.u_ALLfield_stats$Treatment <- unifrac.samp.field.sd$Treatment
scores.ord.u_ALLfield_stats$Species <- unifrac.samp.field.sd$Species
scores.ord.u_ALLfield_stats$Type <- unifrac.samp.field.sd$Type
scores.ord.u_ALLfield_stats$Time.Point <- unifrac.samp.field.sd$Time.Point
scores.ord.u_ALLfield_stats$Code <- paste(scores.ord.u_ALLfield_stats$Species, scores.ord.u_ALLfield_stats$Type)


scores.ord.u_ALLfield_stats$Time.Point[scores.ord.u_ALLfield_stats$Time.Point == "F1"] <- "T1"          # Replace b by XXX
scores.ord.u_ALLfield_stats$Time.Point[scores.ord.u_ALLfield_stats$Time.Point == "TC"] <- "T0"          # Replace b by XXX


pf<-ggplot(scores.ord.u_ALLfield_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species, shape=Type), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code), linetype = 2) +
  ggtitle("NMDS 16s ") +
  theme_classic()
#ggsave("bc_all.pdf")
pf+facet_wrap(~Time.Point*Species)



#PERMANOVA

#make unifrac matrix
s.u.samp.df.all <- as(sample_data(Bac.seq.all), "data.frame") #sample df
s.u.samp.df.all$Code<-paste(s.u.samp.df.all$Species,s.u.samp.df.all$Clade)

s.u.samp.df.all$Time.Point[s.u.samp.df.all$Time.Point == "F1"] <- "T1"          # Replace b by XXX
s.u.samp.df.all$Time.Point[s.u.samp.df.all$Time.Point == "TC"] <- "T0"          # Replace b by XXX



 FieldSamps <- phyloseq::distance(Bac.seq.all, method = "wunifrac")
     FieldSamps.df <- as(sample_data(Bac.seq.all), "data.frame")
     FieldSamps.df$Code<-paste(FieldSamps.df$Treatment,FieldSamps.df$Clade)
    set.seed(30)
    adonis(FieldSamps ~ Species*Type, data = FieldSamps.df)
    FieldSamps.df$Code<-paste(FieldSamps.df$Species, FieldSamps.df$Type)
    ball<-  betadisper(FieldSamps, FieldSamps.df$Code, type = "centroid")
    TukeyHSD(ball,  ordered = FALSE,conf.level = 0.95)  
    
# Homogeneity of dispersion test  
    set.seed(30)
    t<-permutest(betadisper(Bac.s.u.T0, s.u.samp.df$Species, type = "centroid")) 
   set.seed(30)
    permutest(betadisper(Bac.s.u.T0, s.u.samp.df$Code, type = "centroid")) 
    
  ball<-  betadisper(Bac.s.u.T0, s.u.samp.df$Species, type = "centroid")
  TukeyHSD(ball,  ordered = FALSE,conf.level = 0.95)  


``` 





#For lmers of each taxa: do on full dataset, rel abund Bac.s.rel
```{r} 
#adonis in vegan - you have the otu table and the taxa df, you need to add in the metadata to do this
#for regressions (uni) lmtest, KR code, start with metatdata in otu df, with cols as taxa, lme4 or lmertest

#df Bac.s.rel is rel a. Split this by species, then remove T0, then pull out of phyloseq
Bac.s.rel.mcap<-subset_samples(Bac.s.rel, Species=="Montipora_capitata")
Bac.s.rel.mcap.df <- sample_data(Bac.s.rel.mcap)
Bac.s.rel.mcap.df<-as.data.frame(Bac.s.rel.mcap.df)


#take out of phyloseq
OTUfamMcap.v = as(otu_table(Bac.s.rel.mcap), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.rel.mcap)){OTUfamMcap.v <- t(OTUfamMcap.v)}
# Coerce to data.frame
OTUfamMcap.v.df = as.data.frame(OTUfamMcap.v)

OTU<-abundFam$taxa #create a list of all the taxa names 

Mod2_output <- data.frame() #create df 
for(i in OTU){
  lmFAM = lmer(Fam.abund.f[,i] ~ Species, data = Fam.abund.f, REML = TRUE) #otu matrix
  output = anova(lmFAM, Fam.abund.f, type = 3)  #run anova on the output
  Mod2_output = rbind(Mod2_output, data.frame("OTU" = i,     #records ouput
                                                # "Temp P-Value" = output$`Pr(>F)`[which(rownames(output) == "Temp")],
                                                "Speices P-value" = output$`Pr(>F)`[which(rownames(output) == "Speices")] #,
                                               # "Temp:Speices P-value" = output$`Pr(>F)`[which(rownames(output)== "Temp:Species")]
                                              ))
}
#package indexspecies hannah or deseq2 only 2 teratments

#creates df with pval for all effects by taxa, then I pull out ones that are significant (do lines 508 and 509 to pull those out)
```

All species by family df phy,FamTest2TEST relA, this is phyloseq
```{r}        
###### ADONIS PERMANOVA,Fam.filter.16s, [1:6] are meta

Fam.filter.16s.T1TF<-subset(Fam.filter.16s, Time.Point=="T1"|Time.Point=="TF")

# Fam.filter.16s you need sample data and data matrix
Fam.filter.16s.T1TF.m<-Fam.filter.16s.T1TF[7:263]
Fam.filter.16s.T1TF.sd<-Fam.filter.16s.T1TF[1:6]

# Calculate distance matrix 
Fam.distmat <- (Fam.filter.16s.m, method = "bray")
# Creating easy to view matrix and writing .csv
Fam.distmat2 <- as.matrix(Fam.distmat, labels = T)

#bray curtis presence-absense and abundance
#all speices
set.seed(30)


#adonis test
adonis(Fam.distmat ~ Species*Treatment*Time.Point, data = Fam.filter.16s.T1TF.sd)
    

adonis(bc.all ~ Species*Treatment*Time.Point*Clade, data = samp.df)












# Running NMDS in vegan (metaMDS)
Fam_NMDS <-
  metaMDS(Fam.distmat,
          distance = "bray",
          k =2,
          maxit = 999, 
          trymax = 500,
          wascores = TRUE, set.seed(10))

# Shepards test/goodness of fit
goodness(df16s.spp_NMS) # Produces a results of test statistics for goodness of fit for each point

stressplot(df16s.spp_NMS) # Produces a Shepards diagram

# Plotting points in ordination space
plot(df16s.spp_NMS, "sites")   # Produces distance 
orditorp(df16s.spp_NMS, "sites")   # Gives points labels

 

#older
#phyFamTest2TEST 

phyFamTest2TEST_stats<- phyFamTest2TEST #make a copy
otu_table(phyFamTest2TEST_stats)[1:5, 1:5] #check rela worked

#get rid of NA OTUs now!
phyFamTest2TEST_stats = prune_taxa(taxa_sums(phyFamTest2TEST_stats) > 0, phyFamTest2TEST_stats) #this removes any OTU with 0s






    
  
# Homogeneity of dispersion test
betaAll <- betadisper(bc.all, samp.df$Species)
permutest(betaAll, pairwise = T) 
(bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
plot(bc.all.HSD)


#Ordination NMDS for all species, T1Tf
set.seed(50)
ord.Bac_RelA_stats <- ordinate(Bac_RelA_stats.T1F, "NMDS", "bray", trymax = 500) 
stressplot(ord.Bac_RelA_stats)
scores.ord.Bac_RelA_stats<-as.data.frame(cbind(vegan::scores(ord.Bac_RelA_stats, display="sites"))) # these are "points" ask hannah 

scores.ord.Bac_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.Bac_RelA_stats$Species <- samp.df$Species
scores.ord.Bac_RelA_stats$Code <- paste(samp.df$Species,samp.df$Treatment, samp.df$Time.Point)


ggplot(scores.ord.Bac_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Species), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")
 

```

Crap you were working on for above
```{r} 
#everything below here is shit that doesn't work

# phyFamTest2.f2 = filter_taxa(phyFamTest2.f, function(x) sum(x > 3) > (0.013*length(x)), TRUE)

# 
# #This function will remove taxa (OTUs) with low prevalence, where prevalence is the fraction of total samples in which an OTU is observed.
# phyloseq_filter_prevalence(GlobalPatterns, prev.trh = 0.05, abund.trh = NULL)  # 15389 taxa
# 
# 
# 
# 
# library(metagMisc)
# #https://rdrr.io/github/vmikk/metagMisc/man/phyloseq_filter_prevalence.html
# #then non zero in 3 or more samples unless >1% in a sample
# phyFamTest2.f3<-phyloseq_filter_prevalence(phyFamTest2.f, prev.trh = 0.05, abund.trh = .01, threshold_condition = "OR")  # 15639 taxa
# 
# 
# tax_tab <- phyloseq::tax_table(phyFam.REL)
# tax_tab[1:7,1:7] # for my table show me [1 to 5 otu ids, 1 to 5 first five ranks]
# 
# 
# 
# 
# 
# #remove taxa not seen more than 3 times in at least 3 samples (do do this by %, 3/230 is 1.3%)
# phyFam.rel.f = filter_taxa(phyFam, function(x) sum(x > 3) > (0.013*length(x)), TRUE)
# sum(is.na(otu_table(phyFam))) #check if there are any NA
# 
# 
# #aparently you have to use count not rel a
# #first set the genefilter, then send to prune_taxa
# filter <- phyloseq::genefilter_sample(phyFam.REL, filterfun_sample(function(x) x >= 3), 
#                                        A = 0.013*nsamples(phyFam.REL))
# ps_filtered <- prune_taxa(filter, phyFam.REL)
# 
# 
# f <- filter_taxa(phyFam.REL, function(x) var(x) > 1e-05, TRUE) #this worked. this is for variance wtf
# 
# #global test
# otu_table(GlobalPatterns)[1:5, 1:5]
# tax_table(GlobalPatterns)[1:5, 1:4]
# 
# 
# require("genefilter")
#  flist    <- filterfun(kOverA(3, 2e-05))
#  ent.logi <- filter_taxa(phyFamTest2, flist)
#  ent.trim <- filter_taxa(phyFamTest2, flist, TRUE)
#  identical(ent.trim, prune_taxa(ent.logi, phyFamTest2)) 
#  identical(sum(ent.logi), ntaxa(ent.trim))
#  filter_taxa(phyFamTest2, flist, TRUE)   
# 
# 
# 
# 
# 
# 
# data("enterotype")
# library("genefilter")
# flist<- filterfun(kOverA(5, 2e-05))
# ent.logi <- filter_taxa(enterotype, flist)
# ent.trim <- filter_taxa(enterotype, flist, TRUE)
# identical(ent.trim, prune_taxa(ent.logi, enterotype)) 

```

16 bar plots in ggplot (to make faster than next section)  
#MCAP bar plots
PHYLUM
```{r}           
#try glom use raw counts Bac.s
Mcap_Bac.s<-subset_samples(Bac.s, Species=="Montipora_capitata")

Mcap_Bac.s.phy <- Mcap_Bac.s %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Mcap_Bac.s.melt <- psmelt(Mcap_Bac.s.phy)

# sum.ra.mcap <- ddply(Mcap_Bac.s.melt, c("Phylum", "Time.Point", "Treatment","Parent.ID"), summarise,
#                 N = length(Abundance),
#                 mean = mean(Abundance),
#                 sd = sd(Abundance), 
#                 se = sd/sqrt(N)
# )
# sum.ra.mcap

Mcap_Bac.s.melt$Code<-paste(Mcap_Bac.s.melt$Treatment, Mcap_Bac.s.melt$Time.Point)
Mcap_Bac.s.melt$Parent.ID <- factor(Mcap_Bac.s.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Mcap_Bac.s.melt$Code <- factor(Mcap_Bac.s.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

mcap.p <- ggplot(Mcap_Bac.s.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)


################################## 
#by C or CD
# C ONLY

Mcap_Bac.s.C<-subset_samples(Mcap_Bac.s.melt, Clade=="C")

mcap.p <- ggplot(Mcap_Bac.s.C, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap C")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Mcap_Bac.s.CD<-subset_samples(Mcap_Bac.s.melt, Clade=="CD")

mcap.p <- ggplot(Mcap_Bac.s.CD.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

```

Family
```{r}      
library(tidyr)

Bac.s.Mcap<-subset_samples(Bac.s.rel, Species=="Montipora_capitata") #subset mcap
Bac.s.Mcap <- prune_taxa(taxa_sums(Bac.s.Mcap) > 0, Bac.s.Mcap) #prune taxa 0s

Mcap_Bac.s.fam <- Bac.s.Mcap %>% 
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Mcap_Bac.s.fam.melt <- psmelt(Mcap_Bac.s.fam)


#set 
Mcap_Bac.s.fam.melt$Code<-paste(Mcap_Bac.s.fam.melt$Treatment, Mcap_Bac.s.fam.melt$Time.Point)
Mcap_Bac.s.fam.melt$Parent.ID <- factor(Mcap_Bac.s.fam.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365",
                                                                    "366","368",  "369","374"))
Mcap_Bac.s.fam.melt$Code <- factor(Mcap_Bac.s.fam.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
#FAMILY COLORS
nb.cols <- 252
nb.cols <- 353

mycolors <- colorRampPalette(brewer.pal(11, "BrBG"))(nb.cols)

mcap.p <- ggplot(Mcap_Bac.s.fam.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Mcap_Bac.s.melt.C<-subset(Mcap_Bac.s.melt, Clade=="C")

mcap.p <- ggplot(Mcap_Bac.s.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Mcap C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Mcap_Bac.s.melt.CD<-subset(Mcap_Bac.s.melt, Clade=="CD")


mcap.p <- ggplot(Mcap_Bac.s.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Mcap CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ Code, nrow=2)



```
#Pcomp
ASV
```{r}    
#try glom use raw counts Bac.s
Pcomp_Bac.s<-subset_samples(Bac.s, Species=="Porites_compressa")

Pcomp_Bac.s.phy <- Pcomp_Bac.s %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pcomp_Bac.s.melt <- psmelt(Pcomp_Bac.s.phy)

Pcomp_Bac.s.melt$Code<-paste(Pcomp_Bac.s.melt$Treatment, Pcomp_Bac.s.melt$Time.Point)
Pcomp_Bac.s.melt$Parent.ID <- factor(Pcomp_Bac.s.melt$Parent.ID, levels = c("313",  "314","315","316", "317","318","319",  "320", "322","323",  "324","321"))

Pcomp_Bac.s.melt$Code <- factor(Pcomp_Bac.s.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pcomp.p <- ggplot(Pcomp_Bac.s.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pcomp")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)



### by C15cj or C15n
# C15cj ONLY

Pcomp_Bac.s.C15cj<-subset(Pcomp_Bac.s.melt, Clade=="C15cj")

Pcomp.p <- ggplot(Pcomp_Bac.s.C15cj, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C15cj")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

# C15n ONLY
Pcomp_Bac.s.C15n<-subset(Pcomp_Bac.s.melt, Clade=="C15n")

Pcomp.p <- ggplot(Pcomp_Bac.s.C15n, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C15n")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

```
Family
```{r}   

Bac.s.Pcomp<-subset_samples(Bac.s.rel, Species=="Porites_compressa") #subset Pcomp
Bac.s.Pcomp <- prune_taxa(taxa_sums(Bac.s.Pcomp) > 0, Bac.s.Pcomp) #prune taxa 0s

Pcomp_Bac.s.fam <- Bac.s.Pcomp %>% 
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pcomp_Bac.s.fam.melt <- psmelt(Pcomp_Bac.s.fam)

#set 
Pcomp_Bac.s.fam.melt$Code<-paste(Pcomp_Bac.s.fam.melt$Treatment, Pcomp_Bac.s.fam.melt$Time.Point)
Pcomp_Bac.s.fam.melt$Parent.ID <- factor(Pcomp_Bac.s.fam.melt$Parent.ID, levels = c("313",  "314","315","316", "317","318","319",  "320", "322","323",  "324","321"))
Pcomp_Bac.s.fam.melt$Code <- factor(Pcomp_Bac.s.fam.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
#FAMILY COLORS
nb.cols <- 252
mycolors <- colorRampPalette(brewer.pal(11, "BrBG"))(nb.cols)

Pcomp.p <- ggplot(Pcomp_Bac.s.fam.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Pcomp_Bac.s.melt.C<-subset(Pcomp_Bac.s.melt, Clade=="C")

Pcomp.p <- ggplot(Pcomp_Bac.s.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pcomp C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Pcomp_Bac.s.melt.CD<-subset(Pcomp_Bac.s.melt, Clade=="CD")


Pcomp.p <- ggplot(Pcomp_Bac.s.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Pcomp CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pcomp.p +  facet_wrap(~ Code, nrow=2)



```
#Pvar
PHYLUM
```{r}
#try glom use raw counts Bac.s
Pvar_Bac.s<-subset_samples(Bac.s, Species=="Pavona_varians")

Pvar_Bac.s.phy <- Pvar_Bac.s %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pvar_Bac.s.melt <- psmelt(Pvar_Bac.s.phy)

Pvar_Bac.s.melt$Code<-paste(Pvar_Bac.s.melt$Treatment, Pvar_Bac.s.melt$Time.Point)
Pvar_Bac.s.melt$Parent.ID <- factor(Pvar_Bac.s.melt$Parent.ID, levels = c("361",  "358","353","356", "325","354",  "359", "357","362", "351", "355","360"))

Pvar_Bac.s.melt$Code <- factor(Pvar_Bac.s.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pvar.p <- ggplot(Pvar_Bac.s.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pvar")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)



### by C1 or C127
# C15cj ONLY

Pvar_Bac.s.C15cj<-subset(Pvar_Bac.s.melt, Clade=="C1")

Pvar.p <- ggplot(Pvar_Bac.s.C15cj, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pvar C1")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)

# C15n ONLY
Pvar_Bac.s.C27<-subset(Pvar_Bac.s.melt, Clade=="C27")

Pvar.p <- ggplot(Pvar_Bac.s.C27, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pvar C27")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)

```
Family
```{r}  
 

Bac.s.Pvar<-subset_samples(Bac.s.rel, Species=="Pavona_varians") #subset Pvar
Bac.s.Pvar <- prune_taxa(taxa_sums(Bac.s.Pvar) > 0, Bac.s.Pvar) #prune taxa 0s

Pvar_Bac.s.fam <- Bac.s.Pvar %>% 
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pvar_Bac.s.fam.melt <- psmelt(Pvar_Bac.s.fam)

#set 
Pvar_Bac.s.fam.melt$Code<-paste(Pvar_Bac.s.fam.melt$Treatment, Pvar_Bac.s.fam.melt$Time.Point)
Pvar_Bac.s.fam.melt$Parent.ID <- factor(Pvar_Bac.s.fam.melt$Parent.ID, levels = c("361",  "358","353","356", "325","354",  "359", "357","362", "351", "355","360"))

Pvar_Bac.s.fam.melt$Code <- factor(Pvar_Bac.s.fam.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
#FAMILY COLORS
nb.cols <- 365
mycolors <- colorRampPalette(brewer.pal(11, "BrBG"))(nb.cols)

Pvar.p <- ggplot(Pvar_Bac.s.fam.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Pvar_Bac.s.melt.C<-subset(Pvar_Bac.s.melt, Clade=="C")

Pvar.p <- ggplot(Pvar_Bac.s.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pvar C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Pvar_Bac.s.melt.CD<-subset(Pvar_Bac.s.melt, Clade=="CD")


Pvar.p <- ggplot(Pvar_Bac.s.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Pvar CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pvar.p +  facet_wrap(~ Code, nrow=2)



```


#Pacu
PHYLUM
```{r}
#try glom use raw counts Bac.s
Pacu_Bac.s<-subset_samples(Bac.s, Species=="Pocillopora_acuta")

Pacu_Bac.s.phy <- Pacu_Bac.s %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pacu_Bac.s.melt <- psmelt(Pacu_Bac.s.phy)

Pacu_Bac.s.melt$Code<-paste(Pacu_Bac.s.melt$Treatment, Pacu_Bac.s.melt$Time.Point)
Pacu_Bac.s.melt$Parent.ID <- factor(Pacu_Bac.s.melt$Parent.ID, levels = c("304",  "308","312","303", "306","302",  "310", "301","311", "307", "309","305"))

Pacu_Bac.s.melt$Code <- factor(Pacu_Bac.s.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pacu.p <- ggplot(Pacu_Bac.s.melt, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Parent.ID") +
    #theme(legend.position = "none")+
    ggtitle("16S Pacu")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)



### by CCC or C42
# CCC ONLY

Pacu_Bac.s.CCC<-subset(Pacu_Bac.s.melt, Clade=="CCC")

Pacu.p <- ggplot(Pacu_Bac.s.CCC, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pacu CCC")+
    theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)

# C42 ONLY
Pacu_Bac.s.C42<-subset(Pacu_Bac.s.melt, Clade=="C42")

Pacu.p <- ggplot(Pacu_Bac.s.C42, aes(x = Parent.ID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pacu C42")+
      theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)

```
Family
```{r}  

Bac.s.Pacu<-subset_samples(Bac.s.rel, Species=="Pocillopora_acuta") #subset Pacu
Bac.s.Pacu <- prune_taxa(taxa_sums(Bac.s.Pacu) > 0, Bac.s.Pacu) #prune taxa 0s

Pacu_Bac.s.fam <- Bac.s.Pacu %>% 
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pacu_Bac.s.fam.melt <- psmelt(Pacu_Bac.s.fam)



#set 
Pacu_Bac.s.fam.melt$Code<-paste(Pacu_Bac.s.fam.melt$Treatment, Pacu_Bac.s.fam.melt$Time.Point)
Pacu_Bac.s.fam.melt$Parent.ID <- factor(Pacu_Bac.s.fam.melt$Parent.ID, levels = c("304",  "308","312","303", "306","302",  "310", "301","311", "307", "309","305"))
Pacu_Bac.s.fam.melt$Code <- factor(Pacu_Bac.s.fam.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
#FAMILY COLORS
nb.cols <- 365
mycolors <- colorRampPalette(brewer.pal(11, "BrBG"))(nb.cols)

Pacu.p <- ggplot(Pacu_Bac.s.fam.melt, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)


### by C or CD
# C ONLY

Pacu_Bac.s.melt.C<-subset(Pacu_Bac.s.melt, Clade=="C")

Pacu.p <- ggplot(Pacu_Bac.s.melt.C, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  ggtitle("16S Pacu C")+
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)

# CD ONLY
Pacu_Bac.s.melt.CD<-subset(Pacu_Bac.s.melt, Clade=="CD")


Pacu.p <- ggplot(Pacu_Bac.s.melt.CD, aes(x = Parent.ID, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(legend.position = "none")+
  ggtitle("16S Pacu CD")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Pacu.p +  facet_wrap(~ Code, nrow=2)



```




#16s - bar plots df Bac.sRelA YOU Don't need now
```{r}       
#ordination time - CUT WHEN YOU HAVE OTHER 4 SPECIES ORDINATION SORTED 
Bac.s.rel.ord <- ordinate(Bac.s.rel, "NMDS", "bray", set.seed(30))
Bac.RelA.ord_plot = plot_ordination(Bac.s.rel, Bac.s.rel.ord, type="samples", shape="Treatment",color="Species", title="species")
print(Bac.RelA.ord_plot)
Bac.RelA.ord_plot + facet_wrap(~Species, 3)

### let's do this again and run the NMDS by species now that we know they are different. 
#subset by species
Mcap_16s_RA <- subset_samples(Bac.s.rel, Species=="Montipora_capitata")
Pcomp_16s_RA <- subset_samples(Bac.s.rel, Species=="Porites_compressa")
Pvar_16s_RA <- subset_samples(Bac.s.rel, Species=="Pavona_varians")
Pacu_16s_RA <- subset_samples(Bac.s.rel, Species=="Pocillopora_acuta")




#top 50 ASVs across by species 

#Mcap
 TopNOTUsMcap = names(sort(taxa_sums(Mcap_16s_RA), TRUE)[1:300])
 Mcapent50 = prune_taxa(TopNOTUsMcap, Mcap_16s_RA)
McapBarPlot<-plot_bar(Mcap_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot
#ggsave("mMcapBarPlot_16srelARare.pdf", McapBarPlot, dpi=300, width=8)


#### try with gg
Mcap_16s_RA.gg<-phyloseq_to_df(Mcap_16s_RA, addtax = T, addtot = F, addmaxrank = F,
  sorting = "abundance")
Mcap_16s_RA.gg.Phylum<-subset(Mcap_16s_RA.gg, select=-c(2:3,5:8))
#convert data frame from a "wide" format to a "long" format
pcm = melt(Mcap_16s_RA.gg, id = c("OTU","Size","Phylum"))

  mx = ggplot(Mcap_16s_RA.gg, aes(x = Parent.ID, fill = Phylum, y = value)) + 
    geom_bar(stat = "identity", colour = "black") + 
    theme(axis.text.x = element_text(angle = 90, size = 14, colour = "black", vjust = 0.5, hjust = 1, face= "bold"), 
    axis.title.y = element_text(size = 16, face = "bold"), legend.title = element_text(size = 16, face = "bold"), 
    legend.text = element_text(size = 12, face = "bold", colour = "black"), 
    axis.text.y = element_text(colour = "black", size = 12, face = "bold")) + 
    scale_y_continuous(expand = c(0,0)) + 
    labs(x = "", y = "Relative Abundance (%)", fill = "OTU") + 
    scale_fill_manual(values = colours)
    
  mx




#Mcap C and CD
#C
Mcap_16s_RA.C<-subset_samples(Mcap_16s_RA, Clade=="C")
TopNOTUsMcap.C = names(sort(taxa_sums(Mcap_16s_RA.C), TRUE)[1:300])
Mcap.50.C = prune_taxa(TopNOTUsMcap.C, Mcap_16s_RA.C)
McapBarPlot.C<-plot_bar(Mcap_16s_RA.C, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot.C
#ggsave("McapBarPlot_16s_C.pdf", McapBarPlot.C, dpi=300, width=8)

#CD
Mcap_16s_RA.CD<-subset_samples(Mcap_16s_RA, Clade=="CD")
TopNOTUsMcap.CD = names(sort(taxa_sums(Mcap_16s_RA.CD), TRUE)[1:300])
 Mcap.50.CD = prune_taxa(TopNOTUsMcap.CD, Mcap_16s_RA.CD)
McapBarPlot.CD<-plot_bar(Mcap_16s_RA.CD, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");McapBarPlot.CD
#ggsave("McapBarPlot_16s_CD.pdf", McapBarPlot.CD, dpi=300, width=8)


#Pcomp
TopNOTUsPcomp = names(sort(taxa_sums(Pcomp_16s_RA), TRUE)[1:300])
# Pcompent50 = prune_taxa(TopNOTUsPcomp,Pcomp_16s_RA )
PcompBarPlot<-plot_bar(Pcomp_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot
#ggsave("PcompBarPlot_16srelARare.pdf", PcompBarPlot, dpi=300, width=8)

#C15cj and C15n
  #C15cj
  Pcomp_16s_RA.C15cj<-subset_samples(Pcomp_16s_RA, Clade=="C15cj")
  TopNOTUsPcomp.C15cj = names(sort(taxa_sums(Pcomp_16s_RA.C15cj), TRUE)[1:300])
  Pcomp.50.C15cj = prune_taxa(TopNOTUsPcomp.C15cj, Pcomp_16s_RA.C15cj)
  PcompBarPlot.C15cj<-plot_bar(Pcomp_16s_RA.C15cj, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot.C15cj
  #ggsave("PcompBarPlot_16s_C.pdf", PcompBarPlot.C15cj, dpi=300, width=8)

  #C15n
  Pcomp_16s_RA.C15n<-subset_samples(Pcomp_16s_RA, Clade=="C15n")
  TopNOTUsPcomp.C15n = names(sort(taxa_sums(Pcomp_16s_RA.C15n), TRUE)[1:300])
   Pcomp.50.C15n = prune_taxa(TopNOTUsPcomp.C15n, Pcomp_16s_RA.C15n)
  PcompBarPlot.C15n<-plot_bar(Pcomp_16s_RA.C15n, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PcompBarPlot.C15n
#ggsave("PcompBarPlot_16s_CD.pdf", PcompBarPlot.C15n, dpi=300, width=8)


#Pvar
TopNOTUsVar = names(sort(taxa_sums(Pvar_16s_RA), TRUE)[1:300])
Pvarent50 = prune_taxa(TopNOTUsVar, Pvar_16s_RA)
PvarBarPlot<-plot_bar(Pvar_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot
#ggsave("PvarBarPlot_16srelARare.pdf", PvarBarPlot, dpi=300, width=8)


#C1 and C27
  #C15cj
  Pvar_16s_RA.C1<-subset_samples(Pvar_16s_RA, Clade=="C1")
  TopNOTUsPvar.C1 = names(sort(taxa_sums(Pvar_16s_RA.C1), TRUE)[1:300])
  Pvar.50.C1 = prune_taxa(TopNOTUsPvar.C1, Pvar_16s_RA.C1)
  PvarBarPlot.C1<-plot_bar(Pvar_16s_RA.C1, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot.C1
  #ggsave("PvarBarPlot_16s_C.pdf", PvarBarPlot.C15cj, dpi=300, width=8)

  #C27
  Pvar_16s_RA.C27<-subset_samples(Pvar_16s_RA, Clade=="C27")
  TopNOTUsPvar.C27 = names(sort(taxa_sums(Pvar_16s_RA.C27), TRUE)[1:300])
   Pvar.50.C27 = prune_taxa(TopNOTUsPvar.C27, Pvar_16s_RA.C27)
  PvarBarPlot.C27<-plot_bar(Pvar_16s_RA.C27, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PvarBarPlot.C27
  #ggsave("PvarBarPlot_16s_CD.pdf", PvarBarPlot.C27, dpi=300, width=8)



#Pacu
# TopNOTUsPacu = names(sort(taxa_sums(Pacu_16s_RA), TRUE)[1:300])
Pacuent50 = prune_taxa(TopNOTUsPacu, Pacu_16s_RA)
PacuBarPlot<-plot_bar(Pacu_16s_RA, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot
#ggsave("PacuBarPlot_16srelARare.pdf", PacuBarPlot, dpi=300, width=8)

#CCC and C42
  #CCC
  Pacu_16s_RA.CCC<-subset_samples(Pacu_16s_RA, Clade=="CCC")
  TopNOTUsPacu.CCC = names(sort(taxa_sums(Pacu_16s_RA.CCC), TRUE)[1:300])
  Pacu.50.CCC = prune_taxa(TopNOTUsPacu.CCC, Pacu_16s_RA.CCC)
  PacuBarPlot.CCC<-plot_bar(Pacu_16s_RA.CCC, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot.CCC
  #ggsave("PacuBarPlot_16s_C.pdf", PacuBarPlotCCC5cj, dpi=300, width=8)

  #C42
  Pacu_16s_RA.C42<-subset_samples(Pacu_16s_RA, Clade=="C42")
  TopNOTUsPacu.C42 = names(sort(taxa_sums(Pacu_16s_RA.C42), TRUE)[1:300])
   Pacu.50.C42 = prune_taxa(TopNOTUsPacu.C42, Pacu_16s_RA.C42)
  PacuBarPlot.C42<-plot_bar(Pacu_16s_RA.C42, x="Parent.ID", fill="Phylum", facet_grid = "Time.Point~Treatment");PacuBarPlot.C42
  #ggsave("PacuBarPlot_16s_CD.pdf", PacuBarPlot.C42, dpi=300, width=8)
```


#Ordination DIV_RelA <- you might not need this section, code below for each species  
try vegan nmds - its the same, delete when CN says cool, Bac.s.rel
```{r}     
  ##this isn't working, not converging. 
# convert the sample_data() within a phyloseq object to a vegan compatible data object
df16s2veg <- function(Bac.s.rel) {  
  sd <- sample_data(Bac.s.rel)
  return(as(sd,"data.frame"))
}

# convert the otu_table() within a phyloseq object to a vegan compatible data object
df16sotu2veg <- function(Bac.s.rel) {
  OTU <- otu_table(Bac.s.rel)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

# Calculate distance matrix 
df16s.spp_distmat <- (OTU, method = "bray")
# Creating easy to view matrix and writing .csv
df16s.spp_distmat <- 
  as.matrix(df16s.spp_distmat, labels = T)
#write.csv(island.spp_distmat, "island_spp_distmat.csv")

# Running NMDS in vegan (metaMDS)
df16s.spp_NMS <-
  metaMDS(df16s.spp_distmat,
          distance = "bray",
          k =2,
          maxit = 999, 
          trymax = 500,
          wascores = TRUE, set.seed(10))

# Shepards test/goodness of fit
goodness(df16s.spp_NMS) # Produces a results of test statistics for goodness of fit for each point

stressplot(df16s.spp_NMS) # Produces a Shepards diagram

# Plotting points in ordination space
plot(df16s.spp_NMS, "sites")   # Produces distance 
orditorp(df16s.spp_NMS, "sites")   # Gives points labels




```
Heatmap and first crack at ordination, delete when solid
```{r }            
# CUT?? OrdiSpecies<-plot_ordination(DIV_RelA, ordinate(DIV_RelA, "NMDS"), color = "Species", label="ID") + geom_point(size = 5) 
# #ggsave("OrdiSpecies.pdf", OrdiSpecies, dpi=300, width=8)

  
#heat map
plot_heatmap(Bac.s.rel, "NMDS", "bray", "Species", "Phylum")

#by species 
  #mcap
Mcapent50.T1<- subset_samples(Mcapent50, Time.Point=="T1")
Mcapent50.T1.C<- subset_samples(Mcapent50.T1, Clade=="C")
Mcapent50.T1.CD<- subset_samples(Mcapent50.T1, Clade=="CD")

Mcapent50.TF<- subset_samples(Mcapent50, Time.Point=="TF")


  plot_heatmap(Mcapent50.T1, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split
  plot_heatmap(Mcapent50.T1.C, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split
  plot_heatmap(Mcapent50.T1.CD, "NMDS", "bray", "Treatment","Phylum", low="#000033", high="#CCFF66") #use to show split

 #Pcomp
  Pcomp_DIV_RA_T1 <- subset_samples(Pcomp_DIV_RA, Time.Point=="T1")
  plot_heatmap(Pcomp_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
  #Pvar
  Pvar_DIV_RA_T1 <- subset_samples(Pvar_DIV_RA, Time.Point=="T1")
  plot_heatmap(Pvar_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
  plot_heatmap(Pvar_DIV_RA, "NMDS", "bray", "Parent.ID", "DIV")

  
  
  
  
#ordination time - CUT WHEN YOU HAVE OTHER 4 SPECIES ORDINATION SORTED 
DIVrareRel.ord <- ordinate(DIV_RelA, "NMDS", "bray", set.seed(30))
DIVrareRel1 = plot_ordination(DIV_RelA, DIVrareRel.ord, type="samples", shape="Treatment",color="Time.Point", title="species")
print(DIVrareRel1)
DIVrareRel1 + facet_wrap(~Species, 3)

### let's do this again and run the NMDS by species now that we know they are different. 
# mcap
Mcap_DIV_RA.ord <- ordinate(Mcap_DIV_RA, "NMDS", "bray", set.seed(30))
DIVrareRel1_Mcap = plot_ordination(Mcap_DIV_RA, Mcap_DIV_RA.ord, type="samples", shape="Treatment",color="Time.Point", title="Mcap NMDS")
print(DIVrareRel1_Mcap) 
  #add ellipses
  DIVrareRel1_Mcap + 
  stat_ellipse(type = "norm", linetype = 2, alpha=.2) +
  stat_ellipse(type = "t") +
  theme_bw()

# pcomp
Pcomp_DIV_RA.ord <- ordinate(Pcomp_DIV_RA, "NMDS", "bray", set.seed(30))
DIVrareRel1_Pcomp = plot_ordination(Pcomp_DIV_RA, Pcomp_DIV_RA.ord, type="samples", shape="Treatment",color="Time.Point", title="Pcomp NMDS")
print(DIVrareRel1_Pcomp)
#add ellipses
  DIVrareRel1_Pcomp + 
  stat_ellipse(type = "norm",  alpha=.2) +
  theme_bw()
```

#Beta Diversity: df rarefied and relative abudnaces  
All Species Bac.s.rel .  relative abundances  BRAY
```{r}                
#Bac.s.rel  

Bac.s.rel_stats<- Bac.s.rel #make a copy
otu_table(Bac.s.rel_stats)[1:5, 1:5] #check rela worked

#get rid of NA OTUs now!
Bac.s.rel_stats = prune_taxa(taxa_sums(Bac.s.rel_stats) > 0, Bac.s.rel_stats) #this removes any OTU with 0s ~~~ this really drops stuff
      sample_sums(Bac.s.rel_stats) #check we didn't lose anything important

#bray curtis presence-absense and abundance
## REMOVE T0 for models
Bac_RelA_stats.T1F<-subset_samples(Bac.s.rel_stats, Time.Point=="T1"|Time.Point=="TF")

#bray curtis presence-absense and abundance
 
#all speices


#make bray curtis data matrix

bc.all <- phyloseq::distance(Bac_RelA_stats.T1F, method = "bray",trymax=300)
  samp.df <- as(sample_data(Bac_RelA_stats.T1F), "data.frame") #sample df
#adonis test
  set.seed(30)
adonis(bc.all ~ Species*Treatment*Time.Point, data = samp.df)
set.seed(30)
    adonis(bc.all ~ Species*Treatment*Time.Point*Clade, data = samp.df)
    
set.seed(30)
    permutest(betadisper(bc.all, samp.df$Species, type = "centroid")) #put in distance matrix
  

    
  
#Ordination NMDS for all species, T1Tf
set.seed(50)
ord.Bac_RelA_stats <- ordinate(Bac_RelA_stats.T1F, "NMDS", "bray", trymax = 500) 
stressplot(ord.Bac_RelA_stats)
scores.ord.Bac_RelA_stats<-as.data.frame(cbind(vegan::scores(ord.Bac_RelA_stats, display="sites"))) # these are "points" ask hannah 

scores.ord.Bac_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.Bac_RelA_stats$Species <- samp.df$Species


ggplot(scores.ord.Bac_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")


#Ordination NMDS for all species, T1
Bac_RelA_stats.T1<-subset_samples(Bac.s.rel_stats, Time.Point=="T1")
Bac_RelA_stats.T1 = prune_taxa(taxa_sums(Bac_RelA_stats.T1) > 0, Bac_RelA_stats.T1)

set.seed(30)
ord.Bac_RelA_stats <- ordinate(Bac_RelA_stats.T1, "NMDS", "bray", trymax = 500) 
stressplot(ord.Bac_RelA_stats)
scores.ord.Bac_RelA_stats<-as.data.frame(cbind(vegan::scores(ord.Bac_RelA_stats, display="sites"))) # these are "points" ask hannah 

scores.ord.Bac_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.Bac_RelA_stats$Species <- samp.df$Species


ggplot(scores.ord.Bac_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")


```

# Beta Split data by species  bray curtis

Beta div BRAY 
Montipora - T1 and TF together, separet, and then by phylotype. PERMANOVA
```{r}              
# df Bac_RelA_stats.T1F
#Mcap 
mcap.g.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Montipora_capitata")
 mcap.g.bac.T1F = prune_taxa(taxa_sums(mcap.g.bac.T1F) > 0, mcap.g.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
mcap.g.bac.T1 <- subset_samples(mcap.g.bac.T1F, Time.Point == "T1")
  mcap.g.bac.T1 = prune_taxa(taxa_sums(mcap.g.bac.T1) > 0, mcap.g.bac.T1) #this removes any OTU with 0s

mcap.g.bac.TF <- subset_samples(mcap.g.bac.T1F, Time.Point == "TF")
  mcap.g.bac.TF = prune_taxa(taxa_sums(mcap.g.bac.TF) > 0, mcap.g.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.mcap.g.T1F <- phyloseq::distance(mcap.g.bac.T1F, method = "bray")
     mcap.g.samp.df.T1F <- as(sample_data(mcap.g.bac.T1F), "data.frame") #sample df
    
     mcap.g.samp.df.T1F$Code<-paste(mcap.g.samp.df.T1F$Treatment,mcap.g.samp.df.T1F$Time.Point)
       set.seed(30)
          mcap.g.samp.df.T1F$Code2<-paste(mcap.g.samp.df.T1F$Treatment,mcap.g.samp.df.T1F$Time.Point,mcap.g.samp.df.T1F$Clade)
       set.seed(30)
    adonis(bc.mcap.g.T1F ~ Treatment*Time.Point, data = mcap.g.samp.df.T1F) #adonis test no clade
     set.seed(30)
 adonis(bc.mcap.g.T1F ~ Clade*Treatment*Time.Point, data = mcap.g.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(mcap.g.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.mcap.g.T1F, mcap.g.samp.df.T1F$Code, type = "centroid")) #put in distance matrix
      set.seed(30)
    permutest(betadisper(bc.mcap.g.T1F, mcap.g.samp.df.T1F$Clade, type = "centroid"))
   set.seed(30)
    permutest(betadisper(bc.mcap.g.T1F, mcap.g.samp.df.T1F$Treatment, type = "centroid"))
    set.seed(30)
    permutest(betadisper(bc.mcap.g.T1F, mcap.g.samp.df.T1F$Time.Point, type = "centroid"))
     set.seed(30)
    permutest(betadisper(bc.mcap.g.T1F, mcap.g.samp.df.T1F$Code2, type = "centroid"))
    
#T1
  bc.mcapT1 <- phyloseq::distance(mcap.g.bac.T1, method = "bray") 
      mcap.g.samp.df.T1 <- as(sample_data(mcap.g.bac.T1), "data.frame") 
            mcap.g.samp.df.T1$Code<-paste(mcap.g.samp.df.T1$Treatment,mcap.g.samp.df.T1$Clade)
            set.seed(30)
    adonis(bc.mcapT1 ~ Treatment, data = mcap.g.samp.df.T1)
          set.seed(30)
        adonis(bc.mcapT1 ~ Clade*Treatment, data = mcap.g.samp.df.T1)  

# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.mcapT1, mcap.g.samp.df.T1$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(bc.mcapT1, mcap.g.samp.df.T1$Treatment, type = "centroid")) #put in distance matrix
    set.seed(30)
    permutest(betadisper(bc.mcapT1, mcap.g.samp.df.T1$Code, type = "centroid")) #put in distance matrix

#TF
  bc.mcap.g.TF <- phyloseq::distance(mcap.g.bac.TF, method = "bray")
     mcap.g.samp.df.TF <- as(sample_data(mcap.g.bac.TF), "data.frame") 
     mcap.g.samp.df.TF$Code<-paste(mcap.g.samp.df.TF$Treatment,mcap.g.samp.df.TF$Clade)
        set.seed(30)
adonis(bc.mcap.g.TF ~ Treatment, data = mcap.g.samp.df.TF)
           set.seed(30)
 adonis(bc.mcap.g.TF ~ Clade*Treatment, data = mcap.g.samp.df.TF)
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.mcap.g.TF, mcap.g.samp.df.TF$Clade, type = "centroid")) #put in distance matrix
   set.seed(30)
    permutest(betadisper(bc.mcap.g.TF, mcap.g.samp.df.TF$Treatment, type = "centroid")) #put in distance matrix
    set.seed(30)
    permutest(betadisper(bc.mcap.g.TF, mcap.g.samp.df.TF$Code, type = "centroid")) #put in distance matrix
    
  

######### by phylotype: C vs CD
# C ONLY
  mcap.g.bac.T1F.C <- subset_samples(mcap.g.bac.T1F, Clade == "C")
  mcap.g.bac.T1.C <- subset_samples(mcap.g.bac.T1, Clade == "C")
  mcap.g.bac.TF.C <- subset_samples(mcap.g.bac.TF, Clade == "C")
  
#T1TF.C
  bc.mcap.g.T1F.C <- phyloseq::distance(mcap.g.bac.T1F.C, method = "bray")
    mcap.g.samp.df.T1F.C <- as(sample_data(mcap.g.bac.T1F.C), "data.frame") 
  adonis(bc.mcap.g.T1F.C ~ Treatment*Time.Point, data = mcap.g.samp.df.T1F.C)

#T1.C
  bc.mcapT1.C <- phyloseq::distance(mcap.g.bac.T1.C, method = "bray")
    mcap.g.samp.df.T1.C <- as(sample_data(mcap.g.bac.T1.C), "data.frame") #sample df
  adonis(bc.mcapT1.C ~ Treatment, data = mcap.g.samp.df.T1.C)
  beta.mcap.g.T1.C <- betadisper(bc.mcapT1.C, mcap.g.samp.df.T1.C$Treatment)
    permutest(beta.mcap.g.T1.C, pairwise = T) 

#TF.C
  bc.mcap.g.TF.C <- phyloseq::distance(mcap.g.bac.TF.C, method = "bray")
    mcap.g.samp.df.TF.C <- as(sample_data(mcap.g.bac.TF.C), "data.frame") #sample df
  adonis(bc.mcap.g.TF.C ~ Treatment, data = mcap.g.samp.df.TF.C)
  beta.mcap.g.TF.C <- betadisper(bc.mcap.g.TF.C, mcap.g.samp.df.TF.C$Treatment)
    permutest(beta.mcap.g.TF.C, pairwise = T) 

# CD ONLY
mcap.g.bac.T1F.CD <- subset_samples(mcap.g.bac.T1F, Clade == "CD")
mcap.g.bac.T1.CD <- subset_samples(mcap.g.bac.T1, Clade == "CD")
mcap.g.bac.TF.CD <- subset_samples(mcap.g.bac.TF, Clade == "CD")
  
#T1TF.CD
  bc.mcap.g.T1F.CD <- phyloseq::distance(mcap.g.bac.T1F.CD, method = "bray")
  mcap.g.samp.df.T1F.CD <- as(sample_data(mcap.g.bac.T1F.CD), "data.frame") 
  adonis(bc.mcap.g.T1F.CD ~ Treatment*Time.Point, data = mcap.g.samp.df.T1F.CD)

#T1.CD
  bc.mcapT1.CD <- phyloseq::distance(mcap.g.bac.T1.CD, method = "bray")
    mcap.g.samp.df.T1.CD <- as(sample_data(mcap.g.bac.T1.CD), "data.frame") 
  adonis(bc.mcapT1.CD ~ Treatment, data = mcap.g.samp.df.T1.CD)
    beta.mcap.g.T1.CD <- betadisper(bc.mcapT1.CD, mcap.g.samp.df.T1.CD$Treatment)
  permutest(beta.mcap.g.T1.CD, pairwise = T) 

#TF.CD
  bc.mcap.g.TF.CD <- phyloseq::distance(mcap.g.bac.TF.CD, method = "bray")
    mcap.g.samp.df.TF.CD <- as(sample_data(mcap.g.bac.TF.CD), "data.frame") 
  adonis(bc.mcap.g.TF.CD ~ Treatment, data = mcap.g.samp.df.TF.CD)
  beta.mcap.g.TF.CD <- betadisper(bc.mcap.g.TF.CD, mcap.g.samp.df.TF.CD$Treatment)
    permutest(beta.mcap.g.TF.CD, pairwise = T) 
``` 
Mcap NMDS
```{r}      
## create palette for parent.ID and treatments
mcap.g.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", 
 "Ambient" = "#4575B4", "High" = "#D73027",                        
                        "C Ambient"="#DDCC77","C High"="#999933", "CD Ambient"="#44AA99", "CD High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 

#T1 and TF
ord.mcap2 <- ordinate(mcap.g.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap2)

scores.mcap.g.samp.df<-as.data.frame(cbind(vegan::scores(ord.mcap2, display="sites"))) # these are "points" ask hannah 
scores.mcap.g.samp.df$Time.Point <- mcap.g.samp.df.T1F$Time.Point
scores.mcap.g.samp.df$Parent.ID <- mcap.g.samp.df.T1F$Parent.ID
scores.mcap.g.samp.df$Treatment <- mcap.g.samp.df.T1F$Treatment
scores.mcap.g.samp.df$Clade <- mcap.g.samp.df.T1F$Clade
scores.mcap.g.samp.df$Code <- paste(mcap.g.samp.df.T1F$Parent.ID,mcap.g.samp.df.T1F$Time.Point)
scores.mcap.g.samp.df$Code2 <- paste(mcap.g.samp.df.T1F$Treatment,mcap.g.samp.df.T1F$Time.Point)
scores.mcap.g.samp.df$Code3 <- paste(mcap.g.samp.df.T1F$Treatment,mcap.g.samp.df.T1F$Time.Point,mcap.g.samp.df.T1F$Parent.ID)


ggplot(scores.mcap.g.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,3,3,3))+
  ggtitle("Mcap bc TF") +
  theme_classic()

ggplot(scores.mcap.g.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, fill = Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=mcap.g.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()

ggplot(scores.mcap.g.samp.df, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=mcap.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#T1
ord.mcap.g.T1 <- ordinate(mcap.g.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap.g.T1)

ord.mcap.g.T1.df<-as.data.frame(cbind(vegan::scores(ord.mcap.g.T1, display="sites"))) # these are "points" ask hannah 
ord.mcap.g.T1.df$Time.Point <- mcap.g.samp.df.T1$Time.Point
ord.mcap.g.T1.df$Parent.ID <- mcap.g.samp.df.T1$Parent.ID
ord.mcap.g.T1.df$Treatment <- mcap.g.samp.df.T1$Treatment
ord.mcap.g.T1.df$Clade <- mcap.g.samp.df.T1$Clade
ord.mcap.g.T1.df$Code <- paste(mcap.g.samp.df.T1$Parent.ID,mcap.g.samp.df.T1$Time.Point)
ord.mcap.g.T1.df$Code2 <- paste(mcap.g.samp.df.T1$Clade,mcap.g.samp.df.T1$Treatment)
ord.mcap.g.T1.df$Code2 <- as.factor(ord.mcap.g.T1.df$Code2)

ggplot(ord.mcap.g.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,3,3,3))+
  ggtitle("Mcap bc T1") +
  theme_classic()

ggplot(ord.mcap.g.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=mcap.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata T1") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(mcap.g.bac.TF, ordinate(mcap.g.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.mcapTF <- ordinate(mcap.g.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcapTF)
ord.mcap.g.TF.df<-as.data.frame(cbind(vegan::scores(ord.mcapTF, display="sites"))) # these are "points" ask hannah 
ord.mcap.g.TF.df$Treatment <- mcap.g.samp.df.TF$Treatment
ord.mcap.g.TF.df$Parent.ID <- mcap.g.samp.df.TF$Parent.ID
ord.mcap.g.TF.df$Clade <- mcap.g.samp.df.TF$Clade
ord.mcap.g.TF.df$CladeTreat <- as.factor(paste(mcap.g.samp.df.TF$Clade,mcap.g.samp.df.TF$Treatment))
ord.mcap.g.TF.df$Code <- as.factor(paste(mcap.g.samp.df.TF$Parent.ID,mcap.g.samp.df.TF$Treatment))
ord.mcap.g.TF.df$Code2 <- paste(mcap.g.samp.df.TF$Clade,mcap.g.samp.df.TF$Treatment)
ord.mcap.g.TF.df$Code2 <- as.factor(ord.mcap.g.TF.df$Code2)

ggplot(ord.mcap.g.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=mcap.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")



#### BY CLADE
#ords


#C T1 and TF 
set.seed(30)
ord.mcap_C <- ordinate(mcap.g.bac.T1F.C , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C)
scores.mcap_C<-as.data.frame(cbind(vegan::scores(ord.mcap_C, display="sites"))) # these are "points" ask hannah 
scores.mcap_C$Treatment <- mcap.g.samp.df.T1F.C$Treatment
scores.mcap_C$Parent.ID <- mcap.g.samp.df.T1F.C$Parent.ID
scores.mcap_C$Time.Point <- mcap.g.samp.df.T1F.C$Time.Point
scores.mcap_C$Code2<-as.factor(paste(scores.mcap_C$Treatment, scores.mcap_C$Time.Point))

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape =Code2 , color = Parent.ID, fill=Code2), size = 5,stroke = 1) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=mcap.g.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=mcap.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=mcap.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata C only T1F") +
  theme_classic()


#C T1
set.seed(30)
ord.mcap_C.T1 <- ordinate(mcap.g.bac.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.T1)
scores.mcap_C.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_C.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.T1$Treatment <- mcap.g.samp.df.T1.C$Treatment
scores.mcap_C.T1$Parent.ID <- mcap.g.samp.df.T1.C$Parent.ID
scores.mcap_C.T1$Time.Point <- mcap.g.samp.df.T1.C$Time.Point

ggplot(scores.mcap_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=mcap.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("Mcap_T1_its_C.pdf")

#C TF
set.seed(30)
ord.mcap_C.TF <- ordinate(mcap.g.bac.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.TF)
scores.mcap_C.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_C.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.TF$Treatment <- mcap.g.samp.df.TF.C$Treatment
scores.mcap_C.TF$Parent.ID <- mcap.g.samp.df.TF.C$Parent.ID
scores.mcap_C.TF$Time.Point <- mcap.g.samp.df.TF.C$Time.Point

ggplot(scores.mcap_C.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=mcap.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("Mcap_TF_its_TF_C.pdf")


#CD mcap.g.bacT1F.CD
set.seed(30)
ord.mcap_CD <- ordinate(mcap.g.bac.T1F.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD)
scores.mcap_CD<-as.data.frame(cbind(vegan::scores(ord.mcap_CD, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD$Treatment <- mcap.g.samp.df.T1F.CD$Treatment
scores.mcap_CD$Parent.ID <- mcap.g.samp.df.T1F.CD$Parent.ID
scores.mcap_CD$Time.Point <- mcap.g.samp.df.T1F.CD$Time.Point

ggplot(scores.mcap_CD, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=mcap.g.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.mcap_CD.T1 <- ordinate(mcap.g.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.T1)
scores.mcap_CD.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.T1$Treatment <- mcap.g.samp.df.T1.CD$Treatment
scores.mcap_CD.T1$Parent.ID <- mcap.g.samp.df.T1.CD$Parent.ID
scores.mcap_CD.T1$Time.Point <- mcap.g.samp.df.T1.CD$Time.Point

ggplot(scores.mcap_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=mcap.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_T1.pdf")

#CD TF
set.seed(30)
ord.mcap_CD.TF <- ordinate(mcap.g.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.TF)
scores.mcap_CD.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.TF$Treatment <- mcap.g.samp.df.TF.CD$Treatment
scores.mcap_CD.TF$Parent.ID <- mcap.g.samp.df.TF.CD$Parent.ID
scores.mcap_CD.TF$Time.Point <- mcap.g.samp.df.TF.CD$Time.Point

ggplot(scores.mcap_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=mcap.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Mcap_ITS2_CD_TF.pdf")



```

Montipora - T1 and TF together, separet, and then by phylotype. PERMANOVA  
Family good to go
```{r}     
# df   Bac_RelA_stats.T1F
#Mcap
mcap.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Montipora_capitata")
mcap.bac.T1F = prune_taxa(taxa_sums(mcap.bac.T1F) > 0, mcap.bac.T1F) #this removes any OTU with 0s




#make dfs you need 
mcap.bac.T1 <- subset_samples(mcap.bac.T1F, Time.Point == "T1")
  mcap.bac.T1 = prune_taxa(taxa_sums(mcap.bac.T1) > 0, mcap.bac.T1) #this removes any OTU with 0s

mcap.bac.TF <- subset_samples(mcap.bac.T1F, Time.Point == "TF")
  mcap.bac.TF = prune_taxa(taxa_sums(mcap.bac.TF) > 0, mcap.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.mcap.T1F <- phyloseq::distance(mcap.bac.T1F, method = "bray")
     mcap.samp.df.T1F <- as(sample_data(mcap.bac.T1F), "data.frame") #sample df
    adonis(bc.mcap.T1F ~ Treatment*Time.Point, data = mcap.samp.df.T1F) #adonis test no clade
    adonis(bc.mcap.T1F ~ Clade*Treatment*Time.Point, data = mcap.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(mcap.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    
#T1
  bc.mcapT1 <- phyloseq::distance(mcap.bac.T1, method = "bray") 
      mcap.samp.df.T1 <- as(sample_data(mcap.bac.T1), "data.frame") 
    adonis(bc.mcapT1 ~ Clade*Treatment, data = mcap.samp.df.T1)  
# Homogeneity of dispersion test  
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Clade)  #clade
    permutest(beta.mcap.T1, pairwise = T) 
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Treatment) #treatment
    permutest(beta.mcap.T1, pairwise = T) 

#TF
  bc.mcap.TF <- phyloseq::distance(mcap.bac.TF, method = "bray")
     mcap.samp.df.TF <- as(sample_data(mcap.bac.TF), "data.frame") 
    adonis(bc.mcap.TF ~ Clade*Treatment, data = mcap.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Clade) 
    permutest(beta.mcap.TF, pairwise = T) 
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Treatment) 
    permutest(beta.mcap.TF, pairwise = T) 

######### by phylotype: C vs CD
# C ONLY
  mcap.bac.T1F.C <- subset_samples(mcap.bac.T1F, Clade == "C")
  mcap.bac.T1.C <- subset_samples(mcap.bac.T1, Clade == "C")
  mcap.bac.TF.C <- subset_samples(mcap.bac.TF, Clade == "C")
  
#T1TF.C
  bc.mcap.T1F.C <- phyloseq::distance(mcap.bac.T1F.C, method = "bray")
    mcap.samp.df.T1F.C <- as(sample_data(mcap.bac.T1F.C), "data.frame") 
  adonis(bc.mcap.T1F.C ~ Treatment*Time.Point, data = mcap.samp.df.T1F.C)

#T1.C
  bc.mcapT1.C <- phyloseq::distance(mcap.bac.T1.C, method = "bray")
    mcap.samp.df.T1.C <- as(sample_data(mcap.bac.T1.C), "data.frame") #sample df
  adonis(bc.mcapT1.C ~ Treatment, data = mcap.samp.df.T1.C)
  beta.mcap.T1.C <- betadisper(bc.mcapT1.C, mcap.samp.df.T1.C$Treatment)
    permutest(beta.mcap.T1.C, pairwise = T) 

#TF.C
  bc.mcap.TF.C <- phyloseq::distance(mcap.bac.TF.C, method = "bray")
    mcap.samp.df.TF.C <- as(sample_data(mcap.bac.TF.C), "data.frame") #sample df
  adonis(bc.mcap.TF.C ~ Treatment, data = mcap.samp.df.TF.C)
  beta.mcap.TF.C <- betadisper(bc.mcap.TF.C, mcap.samp.df.TF.C$Treatment)
    permutest(beta.mcap.TF.C, pairwise = T) 

# CD ONLY
mcap.bac.T1F.CD <- subset_samples(mcap.bac.T1F, Clade == "CD")
mcap.bac.T1.CD <- subset_samples(mcap.bac.T1, Clade == "CD")
mcap.bac.TF.CD <- subset_samples(mcap.bac.TF, Clade == "CD")
  
#T1TF.CD
  bc.mcap.T1F.CD <- phyloseq::distance(mcap.bac.T1F.CD, method = "bray")
  mcap.samp.df.T1F.CD <- as(sample_data(mcap.bac.T1F.CD), "data.frame") 
  adonis(bc.mcap.T1F.CD ~ Treatment*Time.Point, data = mcap.samp.df.T1F.CD)

#T1.CD
  bc.mcapT1.CD <- phyloseq::distance(mcap.bac.T1.CD, method = "bray")
    mcap.samp.df.T1.CD <- as(sample_data(mcap.bac.T1.CD), "data.frame") 
  adonis(bc.mcapT1.CD ~ Treatment, data = mcap.samp.df.T1.CD)
    beta.mcap.T1.CD <- betadisper(bc.mcapT1.CD, mcap.samp.df.T1.CD$Treatment)
  permutest(beta.mcap.T1.CD, pairwise = T) 

#TF.CD
  bc.mcap.TF.CD <- phyloseq::distance(mcap.bac.TF.CD, method = "bray")
    mcap.samp.df.TF.CD <- as(sample_data(mcap.bac.TF.CD), "data.frame") 
  adonis(bc.mcap.TF.CD ~ Treatment, data = mcap.samp.df.TF.CD)
  beta.mcap.TF.CD <- betadisper(bc.mcap.TF.CD, mcap.samp.df.TF.CD$Treatment)
    permutest(beta.mcap.TF.CD, pairwise = T) 
``` 
Mcap NMDS Family good 
```{r} 
## create palette for parent.ID and treatments
Mcap.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", 
                        "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#DDCC77","C High"="#999933", "CD Ambient"="#44AA99", "CD High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 

#T1 and TF
ord.mcap2 <- ordinate(mcap.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap2)

scores.mcap.samp.df<-as.data.frame(cbind(vegan::scores(ord.mcap2, display="sites"))) # these are "points" ask hannah 
scores.mcap.samp.df$Time.Point <- mcap.samp.df.T1F$Time.Point
scores.mcap.samp.df$Parent.ID <- mcap.samp.df.T1F$Parent.ID
scores.mcap.samp.df$Treatment <- mcap.samp.df.T1F$Treatment
scores.mcap.samp.df$Clade <- mcap.samp.df.T1F$Clade
scores.mcap.samp.df$Code <- paste(mcap.samp.df.T1F$Parent.ID,mcap.samp.df.T1F$Time.Point)
scores.mcap.samp.df$Code2 <- paste(mcap.samp.df.T1F$Treatment,mcap.samp.df.T1F$Time.Point)
scores.mcap.samp.df$Code3 <- paste(mcap.samp.df.T1F$Treatment,mcap.samp.df.T1F$Time.Point,mcap.samp.df.T1F$Parent.ID)

ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Mcap.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()

ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata T1TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#T1
ord.mcap.T1 <- ordinate(mcap.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.mcap.T1)

ord.mcap.T1.df<-as.data.frame(cbind(vegan::scores(ord.mcap.T1, display="sites"))) # these are "points" ask hannah 
ord.mcap.T1.df$Time.Point <- mcap.samp.df.T1$Time.Point
ord.mcap.T1.df$Parent.ID <- mcap.samp.df.T1$Parent.ID
ord.mcap.T1.df$Treatment <- mcap.samp.df.T1$Treatment
ord.mcap.T1.df$Clade <- mcap.samp.df.T1$Clade
ord.mcap.T1.df$Code <- paste(mcap.samp.df.T1$Parent.ID,mcap.samp.df.T1$Time.Point)
ord.mcap.T1.df$Code2 <- paste(mcap.samp.df.T1$Clade,mcap.samp.df.T1$Treatment)
ord.mcap.T1.df$Code2 <- as.factor(ord.mcap.T1.df$Code2)


ggplot(ord.mcap.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata T1") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(mcap.bac.TF, ordinate(mcap.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.mcapTF <- ordinate(mcap.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcapTF)
ord.mcap.TF.df<-as.data.frame(cbind(vegan::scores(ord.mcapTF, display="sites"))) # these are "points" ask hannah 
ord.mcap.TF.df$Treatment <- mcap.samp.df.TF$Treatment
ord.mcap.TF.df$Parent.ID <- mcap.samp.df.TF$Parent.ID
ord.mcap.TF.df$Clade <- mcap.samp.df.TF$Clade
ord.mcap.TF.df$CladeTreat <- as.factor(paste(mcap.samp.df.TF$Clade,mcap.samp.df.TF$Treatment))
ord.mcap.TF.df$Code <- as.factor(paste(mcap.samp.df.TF$Parent.ID,mcap.samp.df.TF$Treatment))
ord.mcap.TF.df$Code2 <- paste(mcap.samp.df.TF$Clade,mcap.samp.df.TF$Treatment)
ord.mcap.TF.df$Code2 <- as.factor(ord.mcap.TF.df$Code2)

ggplot(ord.mcap.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")



#### BY CLADE
#ords


#C T1 and TF 
set.seed(30)
ord.mcap_C <- ordinate(mcap.bac.T1F.C , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C)
scores.mcap_C<-as.data.frame(cbind(vegan::scores(ord.mcap_C, display="sites"))) # these are "points" ask hannah 
scores.mcap_C$Treatment <- mcap.samp.df.T1F.C$Treatment
scores.mcap_C$Parent.ID <- mcap.samp.df.T1F.C$Parent.ID
scores.mcap_C$Time.Point <- mcap.samp.df.T1F.C$Time.Point
scores.mcap_C$Code2<-as.factor(paste(scores.mcap_C$Treatment, scores.mcap_C$Time.Point))

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape =Code2 , color = Parent.ID, fill=Code2), size = 5,stroke = 1) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Mcap.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata C only T1F") +
  theme_classic()


#C T1
set.seed(30)
ord.mcap_C.T1 <- ordinate(mcap.bac.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.T1)
scores.mcap_C.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_C.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.T1$Treatment <- mcap.samp.df.T1.C$Treatment
scores.mcap_C.T1$Parent.ID <- mcap.samp.df.T1.C$Parent.ID
scores.mcap_C.T1$Time.Point <- mcap.samp.df.T1.C$Time.Point

ggplot(scores.mcap_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("Mcap_T1_its_C.pdf")

#C TF
set.seed(30)
ord.mcap_C.TF <- ordinate(mcap.bac.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.TF)
scores.mcap_C.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_C.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.TF$Treatment <- mcap.samp.df.TF.C$Treatment
scores.mcap_C.TF$Parent.ID <- mcap.samp.df.TF.C$Parent.ID
scores.mcap_C.TF$Time.Point <- mcap.samp.df.TF.C$Time.Point

ggplot(scores.mcap_C.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("Mcap_TF_its_TF_C.pdf")


#CD mcap.bacT1F.CD
set.seed(30)
ord.mcap_CD <- ordinate(mcap.bac.T1F.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD)
scores.mcap_CD<-as.data.frame(cbind(vegan::scores(ord.mcap_CD, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD$Treatment <- mcap.samp.df.T1F.CD$Treatment
scores.mcap_CD$Parent.ID <- mcap.samp.df.T1F.CD$Parent.ID
scores.mcap_CD$Time.Point <- mcap.samp.df.T1F.CD$Time.Point

ggplot(scores.mcap_CD, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Mcap.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.mcap_CD.T1 <- ordinate(mcap.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.T1)
scores.mcap_CD.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.T1$Treatment <- mcap.samp.df.T1.CD$Treatment
scores.mcap_CD.T1$Parent.ID <- mcap.samp.df.T1.CD$Parent.ID
scores.mcap_CD.T1$Time.Point <- mcap.samp.df.T1.CD$Time.Point

ggplot(scores.mcap_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("Mcap_ITS2_CD_T1.pdf")

#CD TF
set.seed(30)
ord.mcap_CD.TF <- ordinate(mcap.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.TF)
scores.mcap_CD.TF<-as.data.frame(cbind(vegan::scores(ord.mcap_CD.TF, display="sites"))) # these are "points" ask hannah 
scores.mcap_CD.TF$Treatment <- mcap.samp.df.TF.CD$Treatment
scores.mcap_CD.TF$Parent.ID <- mcap.samp.df.TF.CD$Parent.ID
scores.mcap_CD.TF$Time.Point <- mcap.samp.df.TF.CD$Time.Point

ggplot(scores.mcap_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Mcap_ITS2_CD_TF.pdf")



```


Porites - T1 and TF together, separet, and then by phylotype. PERMANOVA
```{r}         
# df Bac_RelA_stats.T1F
#Pcomp
pcomp.g.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Porites_compressa")
pcomp.g.bac.T1F = prune_taxa(taxa_sums(pcomp.g.bac.T1F) > 0, pcomp.g.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
pcomp.g.bac.T1 <- subset_samples(pcomp.g.bac.T1F, Time.Point == "T1")
  pcomp.g.bac.T1 = prune_taxa(taxa_sums(pcomp.g.bac.T1) > 0, pcomp.g.bac.T1) #this removes any OTU with 0s

pcomp.g.bac.TF <- subset_samples(pcomp.g.bac.T1F, Time.Point == "TF")
  pcomp.g.bac.TF = prune_taxa(taxa_sums(pcomp.g.bac.TF) > 0, pcomp.g.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.pcomp.g.T1F <- phyloseq::distance(pcomp.g.bac.T1F, method = "bray")
     pcomp.g.samp.df.T1F <- as(sample_data(pcomp.g.bac.T1F), "data.frame") #sample df
     pcomp.g.samp.df.T1F$Code<-paste(pcomp.g.samp.df.T1F$Treatment, pcomp.g.samp.df.T1F$Time.Point)
          pcomp.g.samp.df.T1F$Code2<-paste(pcomp.g.samp.df.T1F$Treatment, pcomp.g.samp.df.T1F$Time.Point,pcomp.g.samp.df.T1F$Clade)

          set.seed(30)
adonis(bc.pcomp.g.T1F ~ Treatment*Time.Point, data = pcomp.g.samp.df.T1F) #adonis test no clade
          set.seed(30)
adonis(bc.pcomp.g.T1F ~ Clade*Treatment*Time.Point, data = pcomp.g.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(pcomp.g.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
    # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.pcomp.g.T1F, pcomp.g.samp.df.T1F$Time.Point, type = "centroid"))
   set.seed(30)
    permutest(betadisper(bc.pcomp.g.T1F, pcomp.g.samp.df.T1F$Treatment, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(bc.pcomp.g.T1F, pcomp.g.samp.df.T1F$Code, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(bc.pcomp.g.T1F, pcomp.g.samp.df.T1F$Code2, type = "centroid")) 


#T1

  bc.PcompT1 <- phyloseq::distance(pcomp.g.bac.T1, method = "bray") 
      pcomp.g.samp.df.T1 <- as(sample_data(pcomp.g.bac.T1), "data.frame") 
      pcomp.g.samp.df.T1$Code<-paste(pcomp.g.samp.df.T1$Treatment, pcomp.g.samp.df.T1$Clade)
       set.seed(30)
 adonis(bc.PcompT1 ~ Treatment, data = pcomp.g.samp.df.T1)  
           set.seed(30)
 adonis(bc.PcompT1 ~ Clade*Treatment, data = pcomp.g.samp.df.T1)  
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.PcompT1, pcomp.g.samp.df.T1$Clade, type = "centroid"))
   set.seed(30)
    permutest(betadisper(bc.PcompT1, pcomp.g.samp.df.T1$Treatment, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(bc.PcompT1, pcomp.g.samp.df.T1$Code, type = "centroid")) 

#TF

  bc.pcomp.g.TF <- phyloseq::distance(pcomp.g.bac.TF, method = "bray")
     pcomp.g.samp.df.TF <- as(sample_data(pcomp.g.bac.TF), "data.frame") 
           pcomp.g.samp.df.TF$Code<-paste(pcomp.g.samp.df.TF$Treatment, pcomp.g.samp.df.TF$Clade)

           set.seed(30)
 adonis(bc.pcomp.g.TF ~ Treatment, data = pcomp.g.samp.df.TF)
           set.seed(30)
 adonis(bc.pcomp.g.TF ~ Clade*Treatment, data = pcomp.g.samp.df.TF)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.pcomp.g.TF, pcomp.g.samp.df.TF$Clade, type = "centroid"))
   set.seed(30)
    permutest(betadisper(bc.pcomp.g.TF, pcomp.g.samp.df.TF$Treatment, type = "centroid")) 
     set.seed(30)
    permutest(betadisper(bc.pcomp.g.TF, pcomp.g.samp.df.TF$Code, type = "centroid"))


######### by phylotype: C15cj vs C15n
# C ONLY
  pcomp.g.bac.T1F.C15cj <- subset_samples(pcomp.g.bac.T1F, Clade == "C15cj")
  pcomp.g.bac.T1F.C15cj = prune_taxa(taxa_sums(pcomp.g.bac.T1F.C15cj) > 0, pcomp.g.bac.T1F.C15cj) #this removes any OTU with 0s

  pcomp.g.bac.T1.C15cj <- subset_samples(pcomp.g.bac.T1, Clade == "C15cj")
  pcomp.g.bac.T1.C15cj = prune_taxa(taxa_sums(pcomp.g.bac.T1.C15cj) > 0, pcomp.g.bac.T1.C15cj) #this removes any OTU with 0s

  pcomp.g.bac.TF.C15cj <- subset_samples(pcomp.g.bac.TF, Clade == "C15cj")
  pcomp.g.bac.TF.C15cj = prune_taxa(taxa_sums(pcomp.g.bac.TF.C15cj) > 0, pcomp.g.bac.TF.C15cj) #this removes any OTU with 0s

  
#T1TF.C15cj
  bc.pcomp.g.T1F.C15cj <- phyloseq::distance(pcomp.g.bac.T1F.C15cj, method = "bray")
    pcomp.g.samp.df.T1F.C15cj <- as(sample_data(pcomp.g.bac.T1F.C15cj), "data.frame") 
  set.seed(30)
  adonis(bc.pcomp.g.T1F.C15cj ~ Treatment*Time.Point, data = pcomp.g.samp.df.T1F.C15cj)

#T1.C15cj
  bc.PcompT1.C15cj <- phyloseq::distance(pcomp.g.bac.T1.C15cj, method = "bray")
    pcomp.g.samp.df.T1.C15cj <- as(sample_data(pcomp.g.bac.T1.C15cj), "data.frame") #sample df
      set.seed(30)
  adonis(bc.PcompT1.C15cj ~ Treatment, data = pcomp.g.samp.df.T1.C15cj)
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.PcompT1.C15cj, pcomp.g.samp.df.T1.C15cj$Treatment, type = "centroid"))

#TF.C15cj
  bc.pcomp.g.TF.C15cj <- phyloseq::distance(pcomp.g.bac.TF.C15cj, method = "bray")
    pcomp.g.samp.df.TF.C15cj <- as(sample_data(pcomp.g.bac.TF.C15cj), "data.frame") #sample df
  set.seed(30)
  adonis(bc.pcomp.g.TF.C15cj ~ Treatment, data = pcomp.g.samp.df.TF.C15cj)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.pcomp.g.TF.C15cj, pcomp.g.samp.df.TF.C15cj$Treatment, type = "centroid"))

# CD ONLY
pcomp.g.bac.T1F.C15n <- subset_samples(pcomp.g.bac.T1F, Clade == "C15n")
pcomp.g.bac.T1.C15n <- subset_samples(pcomp.g.bac.T1, Clade == "C15n")
pcomp.g.bac.TF.C15n <- subset_samples(pcomp.g.bac.TF, Clade == "C15n")
  
#T1TF.C15n
  bc.pcomp.g.T1F.C15n <- phyloseq::distance(pcomp.g.bac.T1F.C15n, method = "bray")
  pcomp.g.samp.df.T1F.C15n <- as(sample_data(pcomp.g.bac.T1F.C15n), "data.frame") 
  adonis(bc.pcomp.g.T1F.C15n ~ Treatment*Time.Point, data = pcomp.g.samp.df.T1F.C15n)

#T1.C15n
  bc.PcompT1.C15n <- phyloseq::distance(pcomp.g.bac.T1.C15n, method = "bray")
    pcomp.g.samp.df.T1.C15n <- as(sample_data(pcomp.g.bac.T1.C15n), "data.frame") 
  adonis(bc.PcompT1.C15n ~ Treatment, data = pcomp.g.samp.df.T1.C15n)
    beta.pcomp.g.T1.C15n <- betadisper(bc.PcompT1.C15n, pcomp.g.samp.df.T1.C15n$Treatment)
  permutest(beta.pcomp.g.T1.C15n, pairwise = T) 

#TF.C15n
  bc.pcomp.g.TF.C15n <- phyloseq::distance(pcomp.g.bac.TF.C15n, method = "bray")
    pcomp.g.samp.df.TF.C15n <- as(sample_data(pcomp.g.bac.TF.C15n), "data.frame") 
  adonis(bc.pcomp.g.TF.C15n ~ Treatment, data = pcomp.g.samp.df.TF.C15n)
  beta.pcomp.g.TF.C15n <- betadisper(bc.pcomp.g.TF.C15n, pcomp.g.samp.df.TF.C15n$Treatment)
    permutest(beta.pcomp.g.TF.C15n, pairwise = T) 
``` 
Pcomp NMDS bray
```{r}  
## create palette for parent.ID and treatments
pcomp.g.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", 
 "Ambient" = "#4575B4", "High" = "#D73027",                        
                           "C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred", "Ambient T1"="darkred", "Ambient TF"="darkblue", "High T1"="pink","High TF"="darkred") 

#T1 and TF
ord.Pcomp2 <- ordinate(pcomp.g.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pcomp2)

scores.pcomp.g.samp.df<-as.data.frame(cbind(vegan::scores(ord.Pcomp2, display="sites"))) # these are "points" ask hannah 
scores.pcomp.g.samp.df$Time.Point <- pcomp.g.samp.df.T1F$Time.Point
scores.pcomp.g.samp.df$Parent.ID <- pcomp.g.samp.df.T1F$Parent.ID
scores.pcomp.g.samp.df$Treatment <- pcomp.g.samp.df.T1F$Treatment
scores.pcomp.g.samp.df$Clade <- pcomp.g.samp.df.T1F$Clade
scores.pcomp.g.samp.df$Code <- paste(pcomp.g.samp.df.T1F$Parent.ID,pcomp.g.samp.df.T1F$Time.Point)
scores.pcomp.g.samp.df$Code2 <- paste(pcomp.g.samp.df.T1F$Treatment,pcomp.g.samp.df.T1F$Time.Point)
scores.pcomp.g.samp.df$Code3 <- paste(pcomp.g.samp.df.T1F$Treatment,pcomp.g.samp.df.T1F$Time.Point,pcomp.g.samp.df.T1F$Parent.ID)

ggplot(scores.pcomp.g.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Pcomp.g.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. com T1TF") +
  theme_classic()

ggplot(scores.pcomp.g.samp.df, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=pcomp.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. comp T1TF") +
  theme_classic()
#ggsave("McapT1its2.pdf")





ggplot(scores.pcomp.g.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=pcomp.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2), linetype = 2) +
  ggtitle("Pcmop T1TF") +
  theme_classic()
#ggsave("PcompT1its2.pdf")



#T1
ord.pcomp.g.T1 <- ordinate(pcomp.g.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.pcomp.g.T1)

ord.pcomp.g.T1.df<-as.data.frame(cbind(vegan::scores(ord.pcomp.g.T1, display="sites"))) # these are "points" ask hannah 
ord.pcomp.g.T1.df$Time.Point <- pcomp.g.samp.df.T1$Time.Point
ord.pcomp.g.T1.df$Parent.ID <- pcomp.g.samp.df.T1$Parent.ID
ord.pcomp.g.T1.df$Treatment <- pcomp.g.samp.df.T1$Treatment
ord.pcomp.g.T1.df$Clade <- pcomp.g.samp.df.T1$Clade
ord.pcomp.g.T1.df$Code <- paste(pcomp.g.samp.df.T1$Parent.ID,pcomp.g.samp.df.T1$Time.Point)

#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-Section-01-Main-Lab.html#procrustes-and-ggplot2 
ggplot(ord.pcomp.g.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.g.Parent.colors)+
    #geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("Pcomp T1") +
  theme_classic()
#ggsave("PcompT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(pcomp.g.bac.TF, ordinate(pcomp.g.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.PcompTF <- ordinate(pcomp.g.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompTF)
ord.pcomp.g.TF.df<-as.data.frame(cbind(vegan::scores(ord.PcompTF, display="sites"))) # these are "points" ask hannah 
ord.pcomp.g.TF.df$Treatment <- pcomp.g.samp.df.TF$Treatment
ord.pcomp.g.TF.df$Parent.ID <- pcomp.g.samp.df.TF$Parent.ID
ord.pcomp.g.TF.df$Clade <- pcomp.g.samp.df.TF$Clade
ord.pcomp.g.TF.df$CladeTreat <- as.factor(paste(pcomp.g.samp.df.TF$Clade,pcomp.g.samp.df.TF$Treatment))
ord.pcomp.g.TF.df$Code <- as.factor(paste(pcomp.g.samp.df.TF$Parent.ID,pcomp.g.samp.df.TF$Treatment))

ggplot(ord.pcomp.g.TF.df, aes(x = NMDS1, y = NMDS2)) + 
 geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
    scale_color_manual(values=pcomp.g.Parent.colors)+
    #geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp TF") +
  theme_classic()
#ggsave("PcompT1its2.pdf")



#### BY CLADE
#ords


#C15cj T1 and TF 
set.seed(30)
ord.pcomp.g.C15cj <- ordinate(pcomp.g.bac.T1F.C15cj , "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp.g.C15cj)
scores.pcomp.g.C15cj<-as.data.frame(cbind(vegan::scores(ord.pcomp.g.C15cj, display="sites"))) # these are "points" ask hannah
scores.pcomp.g.C15cj$Treatment <- pcomp.g.samp.df.T1F.C15cj$Treatment
scores.pcomp.g.C15cj$Parent.ID <- pcomp.g.samp.df.T1F.C15cj$Parent.ID
scores.pcomp.g.C15cj$Time.Point <- pcomp.g.samp.df.T1F.C15cj$Time.Point

ggplot(scores.pcomp.g.C15cj, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.g.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P.comp C15cj") +
  theme_classic()
#ggsave("PcompTFits_C.pdf")


#C15cj T1
set.seed(30)
ord.Pcomp_C.T1 <- ordinate(pcomp.g.bac.T1.C15cj, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_C.T1)
scores.Pcomp_C.T1<-as.data.frame(cbind(vegan::scores(ord.Pcomp_C.T1, display="sites"))) # these are "points" ask hannah 
scores.Pcomp_C.T1$Treatment <- pcomp.g.samp.df.T1.C15cj$Treatment
scores.Pcomp_C.T1$Parent.ID <- pcomp.g.samp.df.T1.C15cj$Parent.ID
scores.Pcomp_C.T1$Time.Point <- pcomp.g.samp.df.T1.C15cj$Time.Point

ggplot(scores.Pcomp_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=pcomp.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P comp T1 Cladocopium") +
  theme_classic()
#ggsave("Pcomp_T1_its_C.pdf")

#C TF
set.seed(30)
ord.Pcomp_C15n.TF <- ordinate(pcomp.g.bac.TF.C15cj, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_C15n.TF)
scores.Pcomp_C15n.TF<-as.data.frame(cbind(vegan::scores(ord.Pcomp_C15n.TF, display="sites"))) # 
scores.Pcomp_C15n.TF$Treatment <- pcomp.g.samp.df.TF.C15cj$Treatment
scores.Pcomp_C15n.TF$Parent.ID <- pcomp.g.samp.df.TF.C15cj$Parent.ID
scores.Pcomp_C15n.TF$Time.Point <- pcomp.g.samp.df.TF.C15cj$Time.Point

ggplot(scores.Pcomp_C15n.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P.comp C15 TF") +
  theme_classic()
#ggsave("Pcomp_TF_its_TF_C.pdf")


#CD pcomp.g.bacT1F.C15n
set.seed(30)
ord.pcomp.g.C15n <- ordinate(pcomp.g.bac.T1F.C15n, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp.g.C15n)
scores.pcomp.g.C15n<-as.data.frame(cbind(vegan::scores(ord.pcomp.g.C15n, display="sites"))) # these are "points" ask hannah 
scores.pcomp.g.C15n$Treatment <- pcomp.g.samp.df.T1F.CD$Treatment
scores.pcomp.g.C15n$Parent.ID <- pcomp.g.samp.df.T1F.CD$Parent.ID
scores.pcomp.g.C15n$Time.Point <- pcomp.g.samp.df.T1F.CD$Time.Point

ggplot(scores.pcomp.g.C15n, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=pcomp.g.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("Pcomp_ITS2.C15n_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.pcomp.g.C15n.T1 <- ordinate(pcomp.g.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp.g.C15n.T1)
scores.pcomp.g.C15n.T1<-as.data.frame(cbind(vegan::scores(ord.pcomp.g.C15n.T1, display="sites"))) # these are "points" ask hannah 
scores.pcomp.g.C15n.T1$Treatment <- pcomp.g.samp.df.T1.CD$Treatment
scores.pcomp.g.C15n.T1$Parent.ID <- pcomp.g.samp.df.T1.CD$Parent.ID
scores.pcomp.g.C15n.T1$Time.Point <- pcomp.g.samp.df.T1.CD$Time.Point

ggplot(scores.pcomp.g.C15n.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=pcomp.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("Pcomp_ITS2.C15n_T1.pdf")

#CD TF
set.seed(30)
ord.pcomp.g.C15n.TF <- ordinate(pcomp.g.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp.g.C15n.TF)
scores.pcomp.g.C15n.TF<-as.data.frame(cbind(vegan::scores(ord.pcomp.g.C15n.TF, display="sites"))) # these are "points" ask hannah 
scores.pcomp.g.C15n.TF$Treatment <- pcomp.g.samp.df.TF.CD$Treatment
scores.pcomp.g.C15n.TF$Parent.ID <- pcomp.g.samp.df.TF.CD$Parent.ID
scores.pcomp.g.C15n.TF$Time.Point <- pcomp.g.samp.df.TF.CD$Time.Point

ggplot(scores.pcomp.g.C15n.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=pcomp.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Pcomp_ITS2.C15n_TF.pdf")



```

Pavona - T1 and TF together, separet, and then by phylotype. PERMANOVA
CHANGE PHYLOTYPES

```{r}         
# df Bac_RelA_stats.T1F
#Pvar
Pvar.g.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Pavona_varians")
Pvar.g.bac.T1F = prune_taxa(taxa_sums(Pvar.g.bac.T1F) > 0, Pvar.g.bac.T1F) #this removes any OTU with 0s
 
#make dfs you need 
Pvar.g.bac.T1 <- subset_samples(Pvar.g.bac.T1F, Time.Point == "T1")
  Pvar.g.bac.T1 = prune_taxa(taxa_sums(Pvar.g.bac.T1) > 0, Pvar.g.bac.T1) #this removes any OTU with 0s

Pvar.g.bac.TF <- subset_samples(Pvar.g.bac.T1F, Time.Point == "TF")
  Pvar.g.bac.TF = prune_taxa(taxa_sums(Pvar.g.bac.TF) > 0, Pvar.g.bac.TF) #this removes any OTU with 0s
 
#T1TF
#make bray curtis data matrix -T1 and TF

  bc.Pvar.g.T1F <- phyloseq::distance(Pvar.g.bac.T1F, method = "bray")
     Pvar.g.samp.df.T1F <- as(sample_data(Pvar.g.bac.T1F), "data.frame") #sample df
     Pvar.g.samp.df.T1F$Code<-paste(Pvar.g.samp.df.T1F$Treatment,Pvar.g.samp.df.T1F$Time.Point)
          Pvar.g.samp.df.T1F$Code2<-paste(Pvar.g.samp.df.T1F$Treatment,Pvar.g.samp.df.T1F$Time.Point,Pvar.g.samp.df.T1F$Clade)

             set.seed(30)
 adonis(bc.Pvar.g.T1F ~ Treatment*Time.Point, data = Pvar.g.samp.df.T1F) #adonis test no clade
             set.seed(30)
 adonis(bc.Pvar.g.T1F ~ Clade*Treatment*Time.Point, data = Pvar.g.samp.df.T1F)#adonis test with clade
set.seed(30)
    permutest(betadisper(bc.Pvar.g.T1F, Pvar.g.samp.df.T1F$Treatment, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(bc.Pvar.g.T1F, Pvar.g.samp.df.T1F$Time.Point, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(bc.Pvar.g.T1F, Pvar.g.samp.df.T1F$Code, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(bc.Pvar.g.T1F, Pvar.g.samp.df.T1F$Code2, type = "centroid")) 
    
#T1

  bc.PvarT1 <- phyloseq::distance(Pvar.g.bac.T1, method = "bray") 
      Pvar.g.samp.df.T1 <- as(sample_data(Pvar.g.bac.T1), "data.frame") 
      Pvar.g.samp.df.T1$Code2<-paste(Pvar.g.samp.df.T1$Clade,Pvar.g.samp.df.T1$Treatment)
               set.seed(30)
  adonis(bc.PvarT1 ~ Treatment, data = Pvar.g.samp.df.T1)  
  set.seed(30)
  adonis(bc.PvarT1 ~ Clade*Treatment, data = Pvar.g.samp.df.T1)  
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.PvarT1, Pvar.g.samp.df.T1$Clade, type = "centroid"))
   set.seed(30)
    permutest(betadisper(bc.PvarT1, Pvar.g.samp.df.T1$Treatment, type = "centroid"))
    set.seed(30)
    permutest(betadisper(bc.PvarT1, Pvar.g.samp.df.T1$Code2, type = "centroid"))

#TF

  bc.Pvar.g.TF <- phyloseq::distance(Pvar.g.bac.TF, method = "bray")
     Pvar.g.samp.df.TF <- as(sample_data(Pvar.g.bac.TF), "data.frame") 
           Pvar.g.samp.df.TF$Code2<-paste(Pvar.g.samp.df.TF$Clade,Pvar.g.samp.df.TF$Treatment)

      set.seed(30)
adonis(bc.Pvar.g.TF ~Treatment, data = Pvar.g.samp.df.TF)
                set.seed(30)
adonis(bc.Pvar.g.TF ~ Clade*Treatment, data = Pvar.g.samp.df.TF)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.Pvar.g.TF, Pvar.g.samp.df.TF$Clade, type = "centroid"))
   set.seed(30)
    permutest(betadisper(bc.Pvar.g.TF, Pvar.g.samp.df.TF$Treatment, type = "centroid")) 
    set.seed(30)
    permutest(betadisper(bc.Pvar.g.TF, Pvar.g.samp.df.TF$Code2, type = "centroid")) 

######### by phylotype: C vs CD
# C ONLY
  Pvar.g.bac.T1F.C27 <- subset_samples(Pvar.g.bac.T1F, Clade == "C27")
    Pvar.g.bac.T1F.C27 = prune_taxa(taxa_sums(Pvar.g.bac.T1F.C27) > 0, Pvar.g.bac.T1F.C27) #this removes any OTU with 0s
  Pvar.g.bac.T1.C27 <- subset_samples(Pvar.g.bac.T1, Clade == "C27")
    Pvar.g.bac.T1.C27 = prune_taxa(taxa_sums(Pvar.g.bac.T1.C27) > 0, Pvar.g.bac.T1.C27) #this removes any OTU with 0s
  Pvar.g.bac.TF.C27 <- subset_samples(Pvar.g.bac.TF, Clade == "C27")
    Pvar.g.bac.TF.C27 = prune_taxa(taxa_sums(Pvar.g.bac.TF.C27) > 0, Pvar.g.bac.TF.C27) #this removes any OTU with 0s

#T1TF.C27
  bc.Pvar.g.T1F.C27 <- phyloseq::distance(Pvar.g.bac.T1F.C27, method = "bray")
    Pvar.g.samp.df.T1F.C27 <- as(sample_data(Pvar.g.bac.T1F.C27), "data.frame") 
  set.seed(30)
  adonis(bc.Pvar.g.T1F.C27 ~ Treatment*Time.Point, data = Pvar.g.samp.df.T1F.C27)

#T1.C27
  bc.PvarT1.C27 <- phyloseq::distance(Pvar.g.bac.T1.C27, method = "bray")
    Pvar.g.samp.df.T1.C27 <- as(sample_data(Pvar.g.bac.T1.C27), "data.frame") #sample df
  set.seed(30)
  adonis(bc.PvarT1.C27 ~ Treatment, data = Pvar.g.samp.df.T1.C27)
    # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.PvarT1.C27, Pvar.g.samp.df.T1.C27$Treatment, type = "centroid"))

#TF.C27
  bc.Pvar.g.TF.C27 <- phyloseq::distance(Pvar.g.bac.TF.C27, method = "bray")
    Pvar.g.samp.df.TF.C27 <- as(sample_data(Pvar.g.bac.TF.C27), "data.frame") #sample df
     set.seed(30)
 adonis(bc.Pvar.g.TF.C27 ~ Treatment, data = Pvar.g.samp.df.TF.C27)
    # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.Pvar.g.TF.C27, Pvar.g.samp.df.TF.C27$Treatment, type = "centroid"))
 

# C1 ONLY
Pvar.g.bac.T1F.C1 <- subset_samples(Pvar.g.bac.T1F, Clade == "C1")
    Pvar.g.bac.T1F.C1 = prune_taxa(taxa_sums(Pvar.g.bac.T1F.C1) > 0, Pvar.g.bac.T1F.C1) #this removes any OTU with 0s
Pvar.g.bac.T1.C1 <- subset_samples(Pvar.g.bac.T1, Clade == "C1")   
Pvar.g.bac.T1.C1 = prune_taxa(taxa_sums(Pvar.g.bac.T1.C1) > 0, Pvar.g.bac.T1.C1) #this removes any OTU with 0s
Pvar.g.bac.TF.C1 <- subset_samples(Pvar.g.bac.TF, Clade == "C1")
      Pvar.g.bac.TF.C1 = prune_taxa(taxa_sums(Pvar.g.bac.TF.C1) > 0, Pvar.g.bac.TF.C1) #this removes any OTU with 0s

#T1TF.C1 
  bc.Pvar.g.T1F.C1 <- phyloseq::distance(Pvar.g.bac.T1F.C1, method = "bray")
  Pvar.g.samp.df.T1F.C1 <- as(sample_data(Pvar.g.bac.T1F.C1), "data.frame") 
      set.seed(30)
adonis(bc.Pvar.g.T1F.C1 ~ Treatment*Time.Point, data = Pvar.g.samp.df.T1F.C1)

#T1.C1
  bc.PvarT1.C1 <- phyloseq::distance(Pvar.g.bac.T1.C1, method = "bray")
    Pvar.g.samp.df.T1.C1 <- as(sample_data(Pvar.g.bac.T1.C1), "data.frame") 
      set.seed(30)
adonis(bc.PvarT1.C1 ~ Treatment, data = Pvar.g.samp.df.T1.C1)
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.PvarT1.C1, Pvar.g.samp.df.T1.C1$Treatment, type = "centroid"))

#TF.C1
  bc.Pvar.g.TF.C1 <- phyloseq::distance(Pvar.g.bac.TF.C1, method = "bray")
    Pvar.g.samp.df.TF.C1 <- as(sample_data(Pvar.g.bac.TF.C1), "data.frame") 
      set.seed(30)
adonis(bc.Pvar.g.TF.C1 ~ Treatment, data = Pvar.g.samp.df.TF.C1)
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.Pvar.g.TF.C27, Pvar.g.samp.df.TF.C27$Treatment, type = "centroid"))
``` 
Pvar NMDS
```{r}  
## create palette for parent.ID and treatments
Pvar.g.Parent.colors <- c("325" = "green4",  "353" = "firebrick3", "356" ="deeppink4", "358" = "orange", "361" = "green1","354" = "gold",  "351"= "hotpink", "355" ="darkslateblue", "357" = "cyan", "359" = "darkorange","360" ="darkorchid1", "362" = "lightblue", 
 "Ambient" = "#4575B4", "High" = "#D73027",                        
                        "C1 Ambient"="#DDCC77","C1 High"="#999933", "C27 Ambient"="#44AA99", "C27 High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 


 
#T1 and TF
ord.Pvar2 <- ordinate(Pvar.g.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pvar2)

scores.Pvar.g.samp.df<-as.data.frame(cbind(vegan::scores(ord.Pvar2, display="sites"))) # these are "points" ask hannah 
scores.Pvar.g.samp.df$Time.Point <- Pvar.g.samp.df.T1F$Time.Point
scores.Pvar.g.samp.df$Parent.ID <- Pvar.g.samp.df.T1F$Parent.ID
scores.Pvar.g.samp.df$Treatment <- Pvar.g.samp.df.T1F$Treatment
scores.Pvar.g.samp.df$Clade <- Pvar.g.samp.df.T1F$Clade
scores.Pvar.g.samp.df$Code <- paste(Pvar.g.samp.df.T1F$Parent.ID,Pvar.g.samp.df.T1F$Time.Point)
scores.Pvar.g.samp.df$Code2 <- paste(Pvar.g.samp.df.T1F$Treatment,Pvar.g.samp.df.T1F$Time.Point)
scores.Pvar.g.samp.df$Code3 <- paste(Pvar.g.samp.df.T1F$Treatment,Pvar.g.samp.df.T1F$Time.Point,Pvar.g.samp.df.T1F$Parent.ID)

ggplot(scores.Pvar.g.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pvar.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. varians T1TF") +
  theme_classic()
#ggsave("PvarT1its2.pdf")

ggplot(scores.Pvar.g.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,3,3,3))+
  ggtitle("Pvar bc TF") +
  theme_classic()

ggplot(scores.Pvar.g.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Pvar.g.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. var T1TF") +
  theme_classic()

ggplot(scores.Pvar.g.samp.df, aes(x = NMDS1, y = NMDS2, shape=Treatment, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Pvar.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. var T1TF") +
  theme_classic()

#T1
ord.Pvar.g.T1 <- ordinate(Pvar.g.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pvar.g.T1)

ord.Pvar.g.T1.df<-as.data.frame(cbind(vegan::scores(ord.Pvar.g.T1, display="sites"))) # these are "points" ask hannah 
ord.Pvar.g.T1.df$Time.Point <- Pvar.g.samp.df.T1$Time.Point
ord.Pvar.g.T1.df$Parent.ID <- Pvar.g.samp.df.T1$Parent.ID
ord.Pvar.g.T1.df$Treatment <- Pvar.g.samp.df.T1$Treatment
ord.Pvar.g.T1.df$Clade <- Pvar.g.samp.df.T1$Clade
ord.Pvar.g.T1.df$Code <- paste(Pvar.g.samp.df.T1$Parent.ID,Pvar.g.samp.df.T1$Time.Point)
ord.Pvar.g.T1.df$Code2 <- paste(Pvar.g.samp.df.T1$Clade,Pvar.g.samp.df.T1$Treatment)
ord.Pvar.g.T1.df$Code2 <- as.factor(ord.Pvar.g.T1.df$Code2)


ggplot(ord.Pvar.g.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. var T1") +
  theme_classic()
#ggsave("pvarT1its2.pdf")



#workaround-for when this randomly breaks
# plot_ordination(Pvar.g.bac.TF, ordinate(Pvar.g.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)


#TF only
set.seed(30)
ord.pvarTF <- ordinate(Pvar.g.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvarTF)
ord.Pvar.g.TF.df<-as.data.frame(cbind(vegan::scores(ord.pvarTF, display="sites"))) # these are "points" ask hannah 
ord.Pvar.g.TF.df$Treatment <- Pvar.g.samp.df.TF$Treatment
ord.Pvar.g.TF.df$Parent.ID <- Pvar.g.samp.df.TF$Parent.ID
ord.Pvar.g.TF.df$Clade <- Pvar.g.samp.df.TF$Clade
ord.Pvar.g.TF.df$CladeTreat <- as.factor(paste(Pvar.g.samp.df.TF$Clade,Pvar.g.samp.df.TF$Treatment))
ord.Pvar.g.TF.df$Code <- as.factor(paste(Pvar.g.samp.df.TF$Parent.ID,Pvar.g.samp.df.TF$Treatment))
ord.Pvar.g.TF.df$Code2 <- paste(Pvar.g.samp.df.TF$Clade,Pvar.g.samp.df.TF$Treatment)
ord.Pvar.g.TF.df$Code2 <- as.factor(ord.Pvar.g.TF.df$Code2)

ggplot(ord.Pvar.g.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Treatment))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. var TF") +
  theme_classic()
#ggsave("pvarT1its2.pdf")



#### BY CLADE
#ords


#C T1 and TF 
set.seed(30)
ord.pvar_C <- ordinate(Pvar.g.bac.T1F.C , "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_C)
scores.pvar_C<-as.data.frame(cbind(vegan::scores(ord.pvar_C, display="sites"))) # these are "points" ask hannah 
scores.pvar_C$Treatment <- Pvar.g.samp.df.T1F.C$Treatment
scores.pvar_C$Parent.ID <- Pvar.g.samp.df.T1F.C$Parent.ID
scores.pvar_C$Time.Point <- Pvar.g.samp.df.T1F.C$Time.Point
scores.pvar_C$Code2<-as.factor(paste(scores.pvar_C$Treatment, scores.pvar_C$Time.Point))

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape =Code2 , color = Parent.ID, fill=Code2), size = 5,stroke = 1) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Mcap.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("M. capitata C only T1TF") +
  theme_classic()

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Mcap.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("M. capitata C only T1F") +
  theme_classic()






#C T1
set.seed(30)
ord.mcap_C.T1 <- ordinate(mcap.bac.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.T1)
scores.mcap_C.T1<-as.data.frame(cbind(vegan::scores(ord.mcap_C.T1, display="sites"))) # these are "points" ask hannah 
scores.mcap_C.T1$Treatment <- mcap.samp.df.T1.C$Treatment
scores.mcap_C.T1$Parent.ID <- mcap.samp.df.T1.C$Parent.ID
scores.mcap_C.T1$Time.Point <- mcap.samp.df.T1.C$Time.Point

ggplot(scores.pvar_C.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pvar.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("Pvar T1") +
  theme_classic()
#ggsave("pvar_T1_its_C.pdf")

#C TF
set.seed(30)
ord.pvar_C.TF <- ordinate(Pvar.g.bac.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_C.TF)
scores.pvar_C.TF<-as.data.frame(cbind(vegan::scores(ord.pvar_C.TF, display="sites"))) # these are "points" ask hannah 
scores.pvar_C.TF$Treatment <- Pvar.g.samp.df.TF.C$Treatment
scores.pvar_C.TF$Parent.ID <- Pvar.g.samp.df.TF.C$Parent.ID
scores.pvar_C.TF$Time.Point <- Pvar.g.samp.df.TF.C$Time.Point

ggplot(scores.pvar_C.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("pvar_TF_its_TF_C.pdf")


#CD Pvar.g.bacT1F.CD
set.seed(30)
ord.pvar_CD <- ordinate(Pvar.g.bac.T1F.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_CD)
scores.pvar_CD<-as.data.frame(cbind(vegan::scores(ord.pvar_CD, display="sites"))) # these are "points" ask hannah 
scores.pvar_CD$Treatment <- Pvar.g.samp.df.T1F.CD$Treatment
scores.pvar_CD$Parent.ID <- Pvar.g.samp.df.T1F.CD$Parent.ID
scores.pvar_CD$Time.Point <- Pvar.g.samp.df.T1F.CD$Time.Point

ggplot(scores.pvar_CD, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Pvar.g.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D T1 TF") +
  theme_classic()
#ggsave("pvar_ITS2_CD_its_T1TF.pdf")

#CD T1
set.seed(30)
ord.pvar_CD.T1 <- ordinate(Pvar.g.bac.T1.CD , "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_CD.T1)
scores.pvar_CD.T1<-as.data.frame(cbind(vegan::scores(ord.pvar_CD.T1, display="sites"))) # these are "points" ask hannah 
scores.pvar_CD.T1$Treatment <- Pvar.g.samp.df.T1.CD$Treatment
scores.pvar_CD.T1$Parent.ID <- Pvar.g.samp.df.T1.CD$Parent.ID
scores.pvar_CD.T1$Time.Point <- Pvar.g.samp.df.T1.CD$Time.Point

ggplot(scores.pvar_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pvar.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
#ggsave("pvar_ITS2_CD_T1.pdf")

#CD TF
set.seed(30)
ord.pvar_CD.TF <- ordinate(Pvar.g.bac.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvar_CD.TF)
scores.pvar_CD.TF<-as.data.frame(cbind(vegan::scores(ord.pvar_CD.TF, display="sites"))) # these are "points" ask hannah 
scores.pvar_CD.TF$Treatment <- Pvar.g.samp.df.TF.CD$Treatment
scores.pvar_CD.TF$Parent.ID <- Pvar.g.samp.df.TF.CD$Parent.ID
scores.pvar_CD.TF$Time.Point <- Pvar.g.samp.df.TF.CD$Time.Point

ggplot(scores.pvar_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pvar.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("pvar_ITS2_CD_TF.pdf")


```

Pocillopora - T1 and TF together, separet, and then by phylotype. PERMANOVA
```{r}     
# df Bac_RelA_stats.T1F 
#Pacu
Pacu.g.bac.T1F <- subset_samples(Bac_RelA_stats.T1F, Species == "Pocillopora_acuta")
Pacu.g.bac.T1F = prune_taxa(taxa_sums(Pacu.g.bac.T1F) > 0, Pacu.g.bac.T1F) #this removes any OTU with 0s

#make dfs you need 
Pacu.g.bac.T1 <- subset_samples(Pacu.g.bac.T1F, Time.Point == "T1")
  Pacu.g.bac.T1 = prune_taxa(taxa_sums(Pacu.g.bac.T1) > 0, Pacu.g.bac.T1) #this removes any OTU with 0s

Pacu.g.bac.TF <- subset_samples(Pacu.g.bac.T1F, Time.Point == "TF")
  Pacu.g.bac.TF = prune_taxa(taxa_sums(Pacu.g.bac.TF) > 0, Pacu.g.bac.TF) #this removes any OTU with 0s

#T1TF
#make bray curtis data matrix -T1 and TF

  bc.Pacu.g.T1F <- phyloseq::distance(Pacu.g.bac.T1F, method = "bray")
     Pacu.g.samp.df.T1F <- as(sample_data(Pacu.g.bac.T1F), "data.frame") #sample df
     Pacu.g.samp.df.T1F$Code<-paste(Pacu.g.samp.df.T1F$Treatment, Pacu.g.samp.df.T1F$Time.Point)
          Pacu.g.samp.df.T1F$Code2<-paste(Pacu.g.samp.df.T1F$Treatment, Pacu.g.samp.df.T1F$Time.Point,Pacu.g.samp.df.T1F$Clade)

                  set.seed(30)
adonis(bc.Pacu.g.T1F ~ Treatment*Time.Point, data = Pacu.g.samp.df.T1F) #adonis test no clade
                  set.seed(30)
adonis(bc.Pacu.g.T1F ~ Clade*Treatment*Time.Point, data = Pacu.g.samp.df.T1F)#adonis test with clade
 #heatmap
    #plot_heatmap(Pacu.g.bac.T1F, "NMDS", "bray", "Parent.ID", "Phylum")
     set.seed(30)
    permutest(betadisper(bc.Pacu.g.T1F, Pacu.g.samp.df.T1F$Treatment, type = "centroid"))
 set.seed(30)
    permutest(betadisper(bc.Pacu.g.T1F, Pacu.g.samp.df.T1F$Time.Point, type = "centroid"))
     set.seed(30)
    permutest(betadisper(bc.Pacu.g.T1F, Pacu.g.samp.df.T1F$Code, type = "centroid"))
       set.seed(30)
    permutest(betadisper(bc.Pacu.g.T1F, Pacu.g.samp.df.T1F$Code2, type = "centroid"))
    
    




#T1

  bc.PacuT1 <- phyloseq::distance(Pacu.g.bac.T1, method = "bray") 
      Pacu.g.samp.df.T1 <- as(sample_data(Pacu.g.bac.T1), "data.frame") 
    
      Pacu.g.samp.df.T1$Code<-paste(Pacu.g.samp.df.T1$Clade, Pacu.g.samp.df.T1$Treatment)
                          set.seed(30)
adonis(bc.PacuT1 ~ Treatment, data = Pacu.g.samp.df.T1)  
                   set.seed(30)
 adonis(bc.PacuT1 ~ Clade*Treatment, data = Pacu.g.samp.df.T1)  
 # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.PacuT1, Pacu.g.samp.df.T1$Clade, type = "centroid"))
   set.seed(30)
    permutest(betadisper(bc.PacuT1, Pacu.g.samp.df.T1$Treatment, type = "centroid"))
     set.seed(30)
    permutest(betadisper(bc.PacuT1, Pacu.g.samp.df.T1$Code, type = "centroid"))

#TF

  bc.Pacu.g.TF <- phyloseq::distance(Pacu.g.bac.TF, method = "bray")
     Pacu.g.samp.df.TF <- as(sample_data(Pacu.g.bac.TF), "data.frame")
           Pacu.g.samp.df.TF$Code<-paste(Pacu.g.samp.df.TF$Clade, Pacu.g.samp.df.TF$Treatment)

                       set.seed(30)
  adonis(bc.Pacu.g.TF ~ Treatment, data = Pacu.g.samp.df.TF)
                  set.seed(30)
  adonis(bc.Pacu.g.TF ~ Clade*Treatment, data = Pacu.g.samp.df.TF)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.Pacu.g.TF, Pacu.g.samp.df.TF$Clade, type = "centroid"))
   set.seed(30)
    permutest(betadisper(bc.Pacu.g.TF, Pacu.g.samp.df.TF$Treatment, type = "centroid")) 

     set.seed(30)
    permutest(betadisper(bc.Pacu.g.TF, Pacu.g.samp.df.TF$Code, type = "centroid")) 
######### by phylotype: CCC vs C42
# C ONLY
  Pacu.g.bac.T1F.CCC <- subset_samples(Pacu.g.bac.T1F, Clade == "CCC")
    Pacu.g.bac.T1F.CCC = prune_taxa(taxa_sums(Pacu.g.bac.T1F.CCC) > 0, Pacu.g.bac.T1F.CCC) #this removes any OTU with 0s

  Pacu.g.bac.T1.CCC <- subset_samples(Pacu.g.bac.T1, Clade == "CCC")
    Pacu.g.bac.T1.CCC = prune_taxa(taxa_sums(Pacu.g.bac.T1.CCC) > 0, Pacu.g.bac.T1.CCC) #this removes any OTU with 0s

  Pacu.g.bac.TF.CCC <- subset_samples(Pacu.g.bac.TF, Clade == "CCC")
    Pacu.g.bac.TF.CCC = prune_taxa(taxa_sums(Pacu.g.bac.TF.CCC) > 0, Pacu.g.bac.TF.CCC) #this removes any OTU with 0s

  
#T1TF.CCC
  bc.Pacu.g.T1F.CCC <- phyloseq::distance(Pacu.g.bac.T1F.CCC, method = "bray")
    Pacu.g.samp.df.T1F.CCC <- as(sample_data(Pacu.g.bac.T1F.CCC), "data.frame") 
  set.seed(30)
  adonis(bc.Pacu.g.T1F.CCC ~ Treatment*Time.Point, data = Pacu.g.samp.df.T1F.CCC)

#T1.CCC
  bc.PacuT1.CCC <- phyloseq::distance(Pacu.g.bac.T1.CCC, method = "bray")
    Pacu.g.samp.df.T1.CCC <- as(sample_data(Pacu.g.bac.T1.CCC), "data.frame") #sample df
  set.seed(30)
  adonis(bc.PacuT1.CCC ~ Treatment, data = Pacu.g.samp.df.T1.CCC)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.PacuT1.CCC, Pacu.g.samp.df.T1.CCC$Treatment, type = "centroid"))

#TF.CCC
  bc.Pacu.g.TF.CCC <- phyloseq::distance(Pacu.g.bac.TF.CCC, method = "bray")
    Pacu.g.samp.df.TF.CCC <- as(sample_data(Pacu.g.bac.TF.CCC), "data.frame") #sample df
   set.seed(30)
 adonis(bc.Pacu.g.TF.CCC ~ Treatment, data = Pacu.g.samp.df.TF.CCC)
  # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.Pacu.g.TF.CCC, Pacu.g.samp.df.TF.CCC$Treatment, type = "centroid"))
    
# C42 ONLY
Pacu.g.bac.T1F.C42 <- subset_samples(Pacu.g.bac.T1F, Clade == "C42")
    Pacu.g.bac.T1F.C42 = prune_taxa(taxa_sums(Pacu.g.bac.T1F.C42) > 0, Pacu.g.bac.T1F.C42) #this removes any OTU with 0s

Pacu.g.bac.T1.C42 <- subset_samples(Pacu.g.bac.T1, Clade == "C42")
    Pacu.g.bac.T1.C42 = prune_taxa(taxa_sums(Pacu.g.bac.T1.C42) > 0, Pacu.g.bac.T1.C42) #this removes any OTU with 0s

Pacu.g.bac.TF.C42 <- subset_samples(Pacu.g.bac.TF, Clade == "C42")
    Pacu.g.bac.TF.C42 = prune_taxa(taxa_sums(Pacu.g.bac.TF.C42) > 0, Pacu.g.bac.TF.C42) #this removes any OTU with 0s

  
#T1TF.C42
  bc.Pacu.g.T1F.C42 <- phyloseq::distance(Pacu.g.bac.T1F.C42, method = "bray")
  Pacu.g.samp.df.T1F.C42 <- as(sample_data(Pacu.g.bac.T1F.C42), "data.frame") 
     set.seed(30)
      adonis(bc.Pacu.g.T1F.C42 ~ Treatment, data = Pacu.g.samp.df.T1F.C42)


    
#T1.C42
  bc.PacuT1.C42 <- phyloseq::distance(Pacu.g.bac.T1.C42, method = "bray")
    Pacu.g.samp.df.T1.C42 <- as(sample_data(Pacu.g.bac.T1.C42), "data.frame") 
    set.seed(30)
 adonis(bc.PacuT1.C42 ~ Treatment, data = Pacu.g.samp.df.T1.C42)
    # Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.PacuT1.C42, Pacu.g.samp.df.T1.C42$Treatment, type = "centroid"))

#TF.C42
  bc.Pacu.g.TF.C42 <- phyloseq::distance(Pacu.g.bac.TF.C42, method = "bray")
    Pacu.g.samp.df.TF.C42 <- as(sample_data(Pacu.g.bac.TF.C42), "data.frame") 
    set.seed(30)
 adonis(bc.Pacu.g.TF.C42 ~ Treatment, data = Pacu.g.samp.df.TF.C42)
# Homogeneity of dispersion test  
    set.seed(30)
    permutest(betadisper(bc.Pacu.g.TF.C42, Pacu.g.samp.df.TF.C42$Treatment, type = "centroid"))
``` 
Pacu NMDS
```{r}
## create palette for parent.ID and treatments
Pacu.g.Parent.colors <- c("305" = "green4",  "307" = "firebrick3", "309" ="deeppink4", "311" = "orange", "301" = "green1","302" = "gold",  "308"= "hotpink", "310" ="darkslateblue", "312" = "cyan", "303" = "darkorange","304" ="darkorchid1", "306" = "lightblue", 
 "Ambient" = "#4575B4", "High" = "#D73027",                        
                        "C42 Ambient"="#DDCC77","C42 High"="#999933", "CCC Ambient"="#44AA99", "CCC High"="#117733", 
                        "Ambient T1"="darkblue", "Ambient TF"="skyblue", "High T1"="darkred","High TF"="hotpink") 



#T1 and TF
ord.Pacu2 <- ordinate(Pacu.g.bac.T1F, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pacu2)

scores.Pacu.g.samp.df<-as.data.frame(cbind(vegan::scores(ord.Pacu2, display="sites"))) # these are "points" ask hannah 
scores.Pacu.g.samp.df$Time.Point <- Pacu.g.samp.df.T1F$Time.Point
scores.Pacu.g.samp.df$Parent.ID <- Pacu.g.samp.df.T1F$Parent.ID
scores.Pacu.g.samp.df$Treatment <- Pacu.g.samp.df.T1F$Treatment
scores.Pacu.g.samp.df$Clade <- Pacu.g.samp.df.T1F$Clade
scores.Pacu.g.samp.df$Code <- paste(Pacu.g.samp.df.T1F$Parent.ID,Pacu.g.samp.df.T1F$Time.Point)
scores.Pacu.g.samp.df$Code2 <- paste(Pacu.g.samp.df.T1F$Treatment,Pacu.g.samp.df.T1F$Time.Point)
scores.Pacu.g.samp.df$Code3 <- paste(Pacu.g.samp.df.T1F$Treatment,Pacu.g.samp.df.T1F$Time.Point,Pacu.g.samp.df.T1F$Parent.ID)



ggplot(scores.Pacu.g.samp.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pacu bc T1TF") +
  theme_classic()



ggplot(scores.Pacu.g.samp.df, aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = Code2, color = Parent.ID, fill=Treatment), size = 5) + 
  scale_shape_manual(values = c(21, 22,24,25)) + 
    scale_color_manual(values=Pacu.g.Parent.colors)+
stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. acuta T1TF") +
  theme_classic()

ggplot(scores.Pacu.g.samp.df, aes(x = NMDS1, y = NMDS2, shape=Treatment, fill=Treatment)) + 
 geom_point(aes(size = 5, color=Treatment) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  #scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. acuta T1TF") +
  theme_classic()
#ggsave("PacuT1its2.pdf")

#T1
ord.Pacu.g.T1 <- ordinate(Pacu.g.bac.T1, method="NMDS", "bray", trymax = 300, set.seed(30)) 
stressplot(ord.Pacu.g.T1)

ord.Pacu.g.T1.df<-as.data.frame(cbind(vegan::scores(ord.Pacu.g.T1, display="sites"))) # these are "points" ask hannah 
ord.Pacu.g.T1.df$Time.Point <- Pacu.g.samp.df.T1$Time.Point
ord.Pacu.g.T1.df$Parent.ID <- Pacu.g.samp.df.T1$Parent.ID
ord.Pacu.g.T1.df$Treatment <- Pacu.g.samp.df.T1$Treatment
ord.Pacu.g.T1.df$Time.Point <- Pacu.g.samp.df.T1$Time.Point

ord.Pacu.g.T1.df$Clade <- Pacu.g.samp.df.T1$Clade
ord.Pacu.g.T1.df$Code <- paste(Pacu.g.samp.df.T1$Parent.ID,Pacu.g.samp.df.T1$Time.Point)
ord.Pacu.g.T1.df$Code2 <- paste(Pacu.g.samp.df.T1$Clade,Pacu.g.samp.df.T1$Treatment)
ord.Pacu.g.T1.df$Code2 <- as.factor(ord.Pacu.g.T1.df$Code2)


ggplot(ord.Pacu.g.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pacu bc T1") +
  theme_classic()



ggplot(ord.Pacu.g.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta T1") +
  theme_classic()
#ggsave("PacuT1its2.pdf")

ggplot(ord.Pacu.g.T1.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(colour = Treatment, shape=Treatment,size = 4) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Time.Point))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta T1") +
  theme_classic()


ggplot(ord.Pacu.g.T1.df, aes(x = NMDS1, y = NMDS2, shape=Treatment, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. acuta T1") +
  theme_classic()


#TF only
set.seed(30)
ord.PacuTF <- ordinate(Pacu.g.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuTF)
ord.Pacu.g.TF.df<-as.data.frame(cbind(vegan::scores(ord.PacuTF, display="sites"))) # these are "points" ask hannah 
ord.Pacu.g.TF.df$Treatment <- Pacu.g.samp.df.TF$Treatment
ord.Pacu.g.TF.df$Parent.ID <- Pacu.g.samp.df.TF$Parent.ID
ord.Pacu.g.TF.df$Clade <- Pacu.g.samp.df.TF$Clade
ord.Pacu.g.TF.df$CladeTreat <- as.factor(paste(Pacu.g.samp.df.TF$Clade,Pacu.g.samp.df.TF$Treatment))
ord.Pacu.g.TF.df$Code <- as.factor(paste(Pacu.g.samp.df.TF$Parent.ID,Pacu.g.samp.df.TF$Treatment))
ord.Pacu.g.TF.df$Code2 <- paste(Pacu.g.samp.df.TF$Clade,Pacu.g.samp.df.TF$Treatment)
ord.Pacu.g.TF.df$Code2 <- as.factor(ord.Pacu.g.TF.df$Code2)

ggplot(ord.Pacu.g.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Treatment, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=pcomp.u.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("Pacu bc TF") +
  theme_classic()





ggplot(ord.Pacu.g.TF.df, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Code2, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta TF") +
  theme_classic()
#ggsave("PacuT1its2.pdf")



#### BY CLADE
#ords


#CCC T1 and TF 
set.seed(30)
ord.Pacu_CCC <- ordinate(Pacu.g.bac.T1F.CCC , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC)
scores.Pacu_CCC<-as.data.frame(cbind(vegan::scores(ord.Pacu_CCC, display="sites"))) # these are "points" ask hannah 
scores.Pacu_CCC$Treatment <- Pacu.g.samp.df.T1F.CCC$Treatment
scores.Pacu_CCC$Parent.ID <- Pacu.g.samp.df.T1F.CCC$Parent.ID
scores.Pacu_CCC$Time.Point <- Pacu.g.samp.df.T1F.CCC$Time.Point
scores.Pacu_CCC$Code2<-as.factor(paste(scores.Pacu_CCC$Treatment, scores.Pacu_CCC$Time.Point))


ggplot(scores.Pacu_CCC, aes(x = NMDS1, y = NMDS2, shape=Code2, fill=Treatment)) + 
 geom_point(aes(size = 5) )+
    #geom_point(aes(colour = Treatment, size = 4, shape=Code2)) +
    scale_shape_manual(values = c(21, 22,24,25)) + 
  scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Time.Point))) +
  scale_linetype_manual(values=c(2,3,2,3))+
  ggtitle("P. acuta C only T1TF") +
  theme_classic()

ggplot(scores.Pacu_CCC, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta C only T1F") +
  theme_classic()


#C T1
set.seed(30)
ord.Pacu_CCC.T1 <- ordinate(Pacu.g.bac.T1.CCC, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.T1)
scores.Pacu_CCC.T1<-as.data.frame(cbind(vegan::scores(ord.Pacu_CCC.T1, display="sites"))) # these are "points" ask hannah 
scores.Pacu_CCC.T1$Treatment <- Pacu.g.samp.df.T1.CCC$Treatment
scores.Pacu_CCC.T1$Parent.ID <- Pacu.g.samp.df.T1.CCC$Parent.ID
scores.Pacu_CCC.T1$Time.Point <- Pacu.g.samp.df.T1.CCC$Time.Point

ggplot(scores.Pacu_CCC.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pacu.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. acuta T1 CCC") +
  theme_classic()
#ggsave("Pacu_T1_its_CCC.pdf")

#C TF
set.seed(30)
ord.Pacu_CCC.TF <- ordinate(Pacu.g.bac.TF.CCC, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.TF)
scores.Pacu_CCC.TF<-as.data.frame(cbind(vegan::scores(ord.Pacu_CCC.TF, display="sites"))) # these are "points" ask hannah 
scores.Pacu_CCC.TF$Treatment <- Pacu.g.samp.df.TF.CCC$Treatment
scores.Pacu_CCC.TF$Parent.ID <- Pacu.g.samp.df.TF.CCC$Parent.ID
scores.Pacu_CCC.TF$Time.Point <- Pacu.g.samp.df.TF.CCC$Time.Point

ggplot(scores.Pacu_C42, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta 42  T1F") +
  theme_classic()
#ggsave("Pacu_TF_its_TF_CCC.pdf")


#C42 Pacu.g.bacT1F C42
set.seed(30)
ord.Pacu_C42 <- ordinate(Pacu.g.bac.T1F.C42, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C42)
scores.Pacu_C42<-as.data.frame(cbind(vegan::scores(ord.Pacu_C42, display="sites"))) # these are "points" ask hannah 
scores.Pacu_C42$Treatment <- Pacu.g.samp.df.T1F.C42$Treatment
scores.Pacu_C42$Parent.ID <- Pacu.g.samp.df.T1F.C42$Parent.ID
scores.Pacu_C42$Time.Point <- Pacu.g.samp.df.T1F.C42$Time.Point
scores.Pacu_C42$Time.Point <- Pacu.g.samp.df.T1F.C42$Time.Point
scores.Pacu_C42$Code2<-as.factor(paste(scores.Pacu_C42$Treatment, scores.Pacu_C42$Time.Point))

ggplot(scores.Pacu_C42, aes(x = NMDS1, y = NMDS2)) + 
   geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.g.Parent.colors)+
   # geom_line(aes(group = Parent.ID), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code2, lty=factor(Code2))) +
    scale_linetype_manual(values=c(2,2,3,3))+
  ggtitle("P. acuta CC42 T1F") +
  theme_classic()

#CD T1
set.seed(30)
ord.Pacu_C42.T1 <- ordinate(Pacu.g.bac.T1.C42 , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C42.T1)
scores.Pacu_C42.T1<-as.data.frame(cbind(vegan::scores(ord.Pacu_C42.T1, display="sites"))) # these are "points" ask hannah 
scores.Pacu_C42.T1$Treatment <- Pacu.g.samp.df.T1.C42$Treatment
scores.Pacu_C42.T1$Parent.ID <- Pacu.g.samp.df.T1.C42$Parent.ID
scores.Pacu_C42.T1$Time.Point <- Pacu.g.samp.df.T1.C42$Time.Point

ggplot(scores.Pacu_C42.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. acuta T1 C42") +
  theme_classic()
#ggsave("Pacu_ITS2_C42_T1.pdf")

#CD TF
set.seed(30)
ord.Pacu_C42.TF <- ordinate(Pacu.g.bac.TF.C42, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C42.TF)
scores.Pacu_C42.TF<-as.data.frame(cbind(vegan::scores(ord.Pacu_C42.TF, display="sites"))) # these are "points" ask hannah 
scores.Pacu_C42.TF$Treatment <- Pacu.g.samp.df.TF.C42$Treatment
scores.Pacu_C42.TF$Parent.ID <- Pacu.g.samp.df.TF.C42$Parent.ID
scores.Pacu_C42.TF$Time.Point <- Pacu.g.samp.df.TF.C42$Time.Point

ggplot(scores.Pacu_C42.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.g.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. acuta CD TF") +
  theme_classic()
ggsave("Pacu_ITS2_C42_TF.pdf")




```









##### DELETE BELOW once redon above
Porites - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pcomp
Pcomp.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "porites")
Pcomp.bac.T1 <- subset_samples(Pcomp.bac.T1F, Time.Point == "T1")
Pcomp.bac.TF <- subset_samples(Pcomp.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F <- phyloseq::distance(Pcomp.bac.T1F, method = "bray")
  Pcomp.samp.df.T1F <- as(sample_data(Pcomp.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pcomp.T1F ~ Clade*Treatment*Time.Point, data = Pcomp.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PcompT1 <- phyloseq::distance(Pcomp.bac.T1, method = "bray")
  Pcomp.samp.df.T1 <- as(sample_data(Pcomp.bac.T1), "data.frame") #sample df
  adonis(bc.PcompT1 ~ Clade*Treatment, data = Pcomp.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Clade)
  permutest(beta.Pcomp.T1, pairwise = T) 
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Treatment)
  permutest(beta.Pcomp.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pcomp.TF <- phyloseq::distance(Pcomp.bac.TF, method = "bray")
  Pcomp.samp.df.TF <- as(sample_data(Pcomp.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF ~ Clade*Treatment, data = Pcomp.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Clade)
  permutest(beta.Pcomp.TF, pairwise = T) 
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Treatment)
  permutest(beta.Pcomp.TF, pairwise = T) 


## by clade: C vs CD
  
# C15cj only, new objs
Pcomp.bacT1F.C15cj <- subset_samples(Pcomp.bac.T1F, Clade == "C15cj")
Pcomp.bac.T1.C15cj <- subset_samples(Pcomp.bac.T1, Clade == "C15cj")
Pcomp.bac.TF.C15cj <- subset_samples(Pcomp.bac.TF, Clade == "C15cj")
  
#T1TF.C
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F.C15cj <- phyloseq::distance(Pcomp.bacT1F.C15cj, method = "bray")
  Pcomp.samp.df.T1F.C15cj <- as(sample_data(Pcomp.bacT1F.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.T1F.C15cj ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C15cj)

#T1.C15cj
#make bray curtis data matrix 
  bc.PcompT1.C15cj <- phyloseq::distance(Pcomp.bac.T1.C15cj, method = "bray")
  Pcomp.samp.df.T1.C15cj <- as(sample_data(Pcomp.bac.T1.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.PcompT1.C15cj ~ Treatment, data = Pcomp.samp.df.T1.C15cj)
# Homogeneity of dispersion test  
  beta.Pcomp.T1.C15cj <- betadisper(bc.PcompT1.C15cj, Pcomp.samp.df.T1.C15cj$Treatment)
  permutest(beta.Pcomp.T1.C15cj, pairwise = T) 

#TF.C15cj
#make bray curtis data matrix 
  bc.Pcomp.TF.C15cj <- phyloseq::distance(Pcomp.bac.TF.C15cj, method = "bray")
  Pcomp.samp.df.TF.C15cj <- as(sample_data(Pcomp.bac.TF.C15cj), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF.C15cj ~ Treatment, data = Pcomp.samp.df.TF.C15cj)
# Homogeneity of dispersion test  
  beta.Pcomp.TF.C15cj <- betadisper(bc.Pcomp.TF.C15cj, Pcomp.samp.df.TF.C15cj$Treatment)
  permutest(beta.Pcomp.TF.C15cj, pairwise = T) 


# C15n only, new objs
Pcomp.bacT1F.C15n <- subset_samples(Pcomp.bac.T1F, Clade == "C15n")
Pcomp.bac.T1.C15n <- subset_samples(Pcomp.bac.T1, Clade == "C15n")
Pcomp.bac.TF.C15n <- subset_samples(Pcomp.bac.TF, Clade == "C15n")
  
#T1TF.C15n
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F.C15n <- phyloseq::distance(Pcomp.bacT1F.C15n, method = "bray")
  Pcomp.samp.df.T1F.C15n <- as(sample_data(Pcomp.bacT1F.C15n), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.T1F.C15n ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C15n)

#T1.C15n
#make bray curtis data matrix 
  bc.PcompT1.C15n <- phyloseq::distance(Pcomp.bac.T1.C15n, method = "bray")
  Pcomp.samp.df.T1.C15n <- as(sample_data(Pcomp.bac.T1.C15n), "data.frame") #sample df
  adonis(bc.PcompT1.C15n ~ Treatment, data = Pcomp.samp.df.T1.C15n)#adonis test
# Homogeneity of dispersion test  
  beta.Pcomp.T1.C15n <- betadisper(bc.PcompT1.C15n, Pcomp.samp.df.T1.C15n$Treatment)
  permutest(beta.Pcomp.T1.C15n, pairwise = T) 

#TF.C15n
#make bray curtis data matrix 
  bc.Pcomp.TF.C15n <- phyloseq::distance(Pcomp.bac.TF.C15n, method = "bray")
  Pcomp.samp.df.TF.C15n <- as(sample_data(Pcomp.bac.TF.C15n), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF.C15n ~ Treatment, data = Pcomp.samp.df.TF.C15n)
# Homogeneity of dispersion test  
  beta.Pcomp.TF.C15n <- betadisper(bc.Pcomp.TF.C15n, Pcomp.samp.df.TF.C15n$Treatment)
  permutest(beta.Pcomp.TF.C15n, pairwise = T) 
``` 
Pcomp NMDS
```{r}
library(RColorBrewer)
library(colorRamps)
library(devtools)
 
## create palette for parent.ID and treatments
Pcomp.Parent.colors <- c("313" = "green4",  "314" = "firebrick3", "315" ="deeppink4", "316" = "orange", "317" = "green1","318" = "gold",  "319"= "hotpink", "320" ="darkslateblue", "321" = "cyan", "322" = "darkorange","323" ="darkorchid1", "324" = "lightblue", "Ambient" = "Blue", "High" = "Red","C15cj Ambient"="blue","C15cj High"="red", "C15n Ambient"="darkblue", "C15n High"="darkred") 

#T1 and TF
ord.Pcomp <- ordinate(Pcomp.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pcomp)

scores.Pcomp.samp.df <- as.data.frame(scores(ord.Pcomp) ) 
scores.Pcomp.samp.df$Time.Point <- Pcomp.samp.df.T1F$Time.Point
scores.Pcomp.samp.df$Parent.ID <- Pcomp.samp.df.T1F$Parent.ID
scores.Pcomp.samp.df$Treatment <- Pcomp.samp.df.T1F$Treatment
scores.Pcomp.samp.df$Clade <- Pcomp.samp.df.T1F$Clade
scores.Pcomp.samp.df$Code <- paste(Pcomp.samp.df.T1F$Parent.ID, Pcomp.samp.df.T1F$Time.Point)


ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P. comp Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_T1TF_its2.pdf")

ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PcompT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pcomp.bac.TF, ordinate(Pcomp.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PcompT1 <- ordinate(Pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompT1)
scores.PcompT1 <- as.data.frame(scores(ord.PcompT1))
scores.PcompT1$Treatment <- Pcomp.samp.df.T1$Treatment
scores.PcompT1$Parent.ID <- Pcomp.samp.df.T1$Parent.ID
scores.PcompT1$Clade <- Pcomp.samp.df.T1$Clade
scores.PcompT1$CladeTreat <- as.factor(paste(Pcomp.samp.df.T1$Clade,Pcomp.samp.df.T1$Treatment))


ggplot(scores.PcompT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. comp T1 Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PcompTF <- ordinate(Pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompTF)
scores.PcompTF <- as.data.frame(scores(ord.PcompTF))
scores.PcompTF$Treatment <- Pcomp.samp.df.TF$Treatment
scores.PcompTF$Parent.ID <- Pcomp.samp.df.TF$Parent.ID
scores.PcompTF$Clade <- Pcomp.samp.df.TF$Clade
scores.PcompTF$CladeTreat <- as.factor(paste(Pcomp.samp.df.TF$Clade,Pcomp.samp.df.TF$Treatment))

ggplot(scores.PcompTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. comp TF Bray Curtis") +
  theme_classic()
#ggsave("Pcomp_TF_its_nmds.pdf")



 Pcomp.bac.T1F
Pcomp.bac.T1
Pcomp.bac.TF. 

#### BY CLADE
#ords
#C T1 and TF
set.seed(30)
ord.pcomp_C15cj  <- ordinate(Pcomp.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj)
scores.pcomp_C15cj <- as.data.frame(scores(ord.pcomp_C15cj))
scores.pcomp_C15cj$Treatment <- Pcomp.samp.df.T1F$Treatment
scores.pcomp_C15cj$Parent.ID <- Pcomp.samp.df.T1F$Parent.ID
scores.pcomp_C15cj$Time.Point <- Pcomp.samp.df.T1F$Time.Point

ggplot(scores.pcomp_C15cj, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P comp C15cj") +
  theme_classic()
#ggsave("pcompTFits_C15cj.pdf")


#C T1
set.seed(30)
ord.pcomp_C15cj.T1 <- ordinate(Pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj.T1)
scores.pcomp_C15cj.T1 <- as.data.frame(scores(ord.pcomp_C15cj.T1))
scores.pcomp_C15cj.T1$Treatment <- pcomp.samp.df.T1$Treatment
scores.pcomp_C15cj.T1$Parent.ID <- pcomp.samp.df.T1$Parent.ID
scores.pcomp_C15cj.T1$Time.Point <- pcomp.samp.df.T1$Time.Point

ggplot(scores.pcomp_C15cj.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp T1 C15cj") +
  theme_classic()
#ggsave("pcomp_T1_its_C15cj.pdf")

#C TF
set.seed(30)
ord.pcomp_C15cj.TF <- ordinate(pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15cj.TF)
scores.pcomp_C15cj.TF <- as.data.frame(scores(ord.pcomp_C15cj.TF))
scores.pcomp_C15cj.TF$Treatment <- pcomp.samp.df.TF$Treatment
scores.pcomp_C15cj.TF$Parent.ID <- pcomp.samp.df.TF$Parent.ID
scores.pcomp_C15cj.TF$Time.Point <- pcomp.samp.df.TF$Time.Point

ggplot(scores.pcomp_C15cj.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("pcomp_TF_its_TF_C15cj.pdf")


#C15n
set.seed(30)
ord.pcomp_C15n <- ordinate(pcomp_DIV_C15n, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n)
scores.pcomp_C15n <- as.data.frame(scores(ord.pcomp_C15n))
scores.pcomp_C15n$Treatment <- pcomp.samp.df.T1F.CD$Treatment
scores.pcomp_C15n$Parent.ID <- pcomp.samp.df.T1F.CD$Parent.ID
scores.pcomp_C15n$Time.Point <- pcomp.samp.df.T1F.CD$Time.Point

ggplot(scores.pcomp_C15n, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_its_T1TF.pdf")

#C15n T1
set.seed(30)
ord.pcomp_C15n.T1 <- ordinate(pcomp.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n.T1)
scores.pcomp_C15n.T1 <- as.data.frame(scores(ord.pcomp_C15n.T1))
scores.pcomp_C15n.T1$Treatment <- pcomp.samp.df.T1$Treatment
scores.pcomp_C15n.T1$Parent.ID <- pcomp.samp.df.T1$Parent.ID
scores.pcomp_C15n.T1$Time.Point <- pcomp.samp.df.T1$Time.Point

ggplot(scores.pcomp_C15n.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P comp T1 C15n") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_T1.pdf")

#CD TF
set.seed(30)
ord.pcomp_C15n.TF <- ordinate(pcomp.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcomp_C15n.TF)
scores.pcomp_C15n.TF <- as.data.frame(scores(ord.pcomp_C15n.TF))
scores.pcomp_C15n.TF$Treatment <- pcomp.samp.df.TF$Treatment
scores.pcomp_C15n.TF$Parent.ID <- pcomp.samp.df.TF$Parent.ID
scores.pcomp_C15n.TF$Time.Point <- pcomp.samp.df.TF$Time.Point

ggplot(scores.pcomp_C15n.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp C15n TF") +
  theme_classic()
#ggsave("pcomp_ITS2_C15n_TF.pdf")


```
Pavona - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pvar
Pvar.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pavona")
Pvar.bac.T1 <- subset_samples(Pvar.bac.T1F, Time.Point == "T1")
Pvar.bac.TF <- subset_samples(Pvar.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F <- phyloseq::distance(Pvar.bac.T1F, method = "bray")
  Pvar.samp.df.T1F <- as(sample_data(Pvar.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pvar.T1F ~ Clade*Treatment*Time.Point, data = Pvar.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PvarT1 <- phyloseq::distance(Pvar.bac.T1, method = "bray")
  Pvar.samp.df.T1 <- as(sample_data(Pvar.bac.T1), "data.frame") #sample df
  adonis(bc.PvarT1 ~ Clade*Treatment, data = Pvar.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Clade)
  permutest(beta.Pvar.T1, pairwise = T) 
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Treatment)
  permutest(beta.Pvar.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pvar.TF <- phyloseq::distance(Pvar.bac.TF, method = "bray")
  Pvar.samp.df.TF <- as(sample_data(Pvar.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF ~ Clade*Treatment, data = Pvar.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Clade)
  permutest(beta.Pvar.TF, pairwise = T) 
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Treatment)
  permutest(beta.Pvar.TF, pairwise = T) 


## by clade: C1 vs C27
  
# C1 only, new objs
Pvar.bacT1F.C1 <- subset_samples(Pvar.bac.T1F, Clade == "C1")
Pvar.bac.T1.C1 <- subset_samples(Pvar.bac.T1, Clade == "C1")
Pvar.bac.TF.C1 <- subset_samples(Pvar.bac.TF, Clade == "C1")
  
#T1TF.C1
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F.C1 <- phyloseq::distance(Pvar.bacT1F.C1, method = "bray")
  Pvar.samp.df.T1F.C1 <- as(sample_data(Pvar.bacT1F.C1), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.T1F.C1 ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C1)

#T1.C1
#make bray curtis data matrix 
  bc.PvarT1.C1 <- phyloseq::distance(Pvar.bac.T1.C1, method = "bray")
  Pvar.samp.df.T1.C1 <- as(sample_data(Pvar.bac.T1.C1), "data.frame") #sample df
#adonis test
  adonis(bc.PvarT1.C1 ~ Treatment, data = Pvar.samp.df.T1.C1)
# Homogeneity of dispersion test  
  beta.Pvar.T1.C1 <- betadisper(bc.PvarT1.C1, Pvar.samp.df.T1.C1$Treatment)
  permutest(beta.Pvar.T1.C1, pairwise = T) 

#TF.C1
#make bray curtis data matrix 
  bc.Pvar.TF.C1 <- phyloseq::distance(Pvar.bac.TF.C1, method = "bray")
  Pvar.samp.df.TF.C1 <- as(sample_data(Pvar.bac.TF.C1), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF.C1 ~ Treatment, data = Pvar.samp.df.TF.C1)
# Homogeneity of dispersion test  
  beta.Pvar.TF.C1 <- betadisper(bc.Pvar.TF.C1, Pvar.samp.df.TF.C1$Treatment)
  permutest(beta.Pvar.TF.C1, pairwise = T) 


# C27 only, new objs
Pvar.bacT1F.C27 <- subset_samples(Pvar.bac.T1F, Clade == "C27")
Pvar.bac.T1.C27 <- subset_samples(Pvar.bac.T1, Clade == "C27")
Pvar.bac.TF.C27 <- subset_samples(Pvar.bac.TF, Clade == "C27")
  
#T1TF.C27
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F.C27 <- phyloseq::distance(Pvar.bacT1F.C27, method = "bray")
  Pvar.samp.df.T1F.C27 <- as(sample_data(Pvar.bacT1F.C27), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.T1F.C27 ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C27)

#T1.C27
#make bray curtis data matrix 
  bc.PvarT1.C27 <- phyloseq::distance(Pvar.bac.T1.C27, method = "bray")
  Pvar.samp.df.T1.C27 <- as(sample_data(Pvar.bac.T1.C27), "data.frame") #sample df
  adonis(bc.PvarT1.C27 ~ Treatment, data = Pvar.samp.df.T1.C27)#adonis test
# Homogeneity of dispersion test  
  beta.Pvar.T1.C27 <- betadisper(bc.PvarT1.C27, Pvar.samp.df.T1.C27$Treatment)
  permutest(beta.Pvar.T1.C27, pairwise = T) 

#TF.C27
#make bray curtis data matrix 
  bc.Pvar.TF.C27 <- phyloseq::distance(Pvar.bac.TF.C27, method = "bray")
  Pvar.samp.df.TF.C27 <- as(sample_data(Pvar.bac.TF.C27), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF.C27 ~ Treatment, data = Pvar.samp.df.TF.C27)
# Homogeneity of dispersion test  
  beta.Pvar.TF.C27 <- betadisper(bc.Pvar.TF.C27, Pvar.samp.df.TF.C27$Treatment)
  permutest(beta.Pvar.TF.C27, pairwise = T) 
``` 
Pvar NMDS
```{r}
#dfs
# C1 only, new objs
# Pvar.bacT1F.C1 <- subset_samples(Pvar.bac.T1F, Clade == "C1")
# Pvar.bac.T1.C1 <- subset_samples(Pvar.bac.T1, Clade == "C1")
# Pvar.bac.TF.C1 <- subset_samples(Pvar.bac.TF, Clade == "C1")


library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pvar.Parent.colors <- c("325" = "green4",  "351" = "firebrick3", "353" ="deeppink4", "354" = "orange", "355" = "green1","356" = "gold",  "357"= "hotpink", "358" ="darkslateblue", "359" = "honeydew4", "360" = "khaki4","361" ="ivory4", "362" = "lightblue", Ambient = "Blue", High = "Red","C1 Ambient"="blue","C1 High"="red", "C27 Ambient"="darkblue", "C27 High"="darkred") 

#T1 and TF
ord.Pvar <- ordinate(Pvar.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pvar)

scores.Pvar.samp.df <- as.data.frame(scores(ord.Pvar) ) 
scores.Pvar.samp.df$Time.Point <- Pvar.samp.df.T1F$Time.Point
scores.Pvar.samp.df$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar.samp.df$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar.samp.df$Clade <- Pvar.samp.df.T1F$Clade
scores.Pvar.samp.df$Code <- paste(Pvar.samp.df.T1F$Parent.ID, Pvar.samp.df.T1F$Time.Point)


ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P.var Bray Curtis") +
  theme_classic()
#ggsave("Pvar_T1TF_its2.pdf")

ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var Bray Curtis") +
  theme_classic()
#ggsave("PvarT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pvar.bac.TF, ordinate(Pvar.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PvarT1 <- ordinate(Pvar.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PvarT1)
scores.PvarT1 <- as.data.frame(scores(ord.PvarT1))
scores.PvarT1$Treatment <- Pvar.samp.df.T1$Treatment
scores.PvarT1$Parent.ID <- Pvar.samp.df.T1$Parent.ID
scores.PvarT1$Clade <- Pvar.samp.df.T1$Clade
scores.PvarT1$CladeTreat <- as.factor(paste(Pvar.samp.df.T1$Clade,Pvar.samp.df.T1$Treatment))


ggplot(scores.PvarT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var T1 Bray Curtis") +
  theme_classic()
#ggsave("Pvar_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PvarTF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PvarTF)
scores.PvarTF <- as.data.frame(scores(ord.PvarTF))
scores.PvarTF$Treatment <- Pvar.samp.df.TF$Treatment
scores.PvarTF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.PvarTF$Clade <- Pvar.samp.df.TF$Clade
scores.PvarTF$CladeTreat <- as.factor(paste(Pvar.samp.df.TF$Clade,Pvar.samp.df.TF$Treatment))

ggplot(scores.PvarTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. var TF Bray Curtis") +
  theme_classic()
#gsave("Pvar_TF_its_nmds.pdf")



# C1 only, new objs
#Pvar.bacT1F.C1 , Pvar.samp.df.T1F.C1
#Pvar.bac.T1.C1 , Pvar.samp.df.T1.C1
#Pvar.bac.TF.C1 ,Pvar.samp.df.TF.C1

#### BY CLADE

#CCC T1 and TF
set.seed(30)
ord.Pvar_CCC  <- ordinate(Pvar.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC)
scores.Pvar_CCC <- as.data.frame(scores(ord.Pvar_CCC))
scores.Pvar_CCC$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar_CCC$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar_CCC$Time.Point <- Pvar.samp.df.T1F$Time.Point

ggplot(scores.Pvar_CCC, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var CCC") +
  theme_classic()
#ggsave("PvarTFits_CCC.pdf")


#C T1
set.seed(30)
ord.Pvar_CCC.T1 <- ordinate(Pvar.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC.T1)
scores.Pvar_CCC.T1 <- as.data.frame(scores(ord.Pvar_CCC.T1))
scores.Pvar_CCC.T1$Treatment <- Pvar.samp.df.T1$Treatment
scores.Pvar_CCC.T1$Parent.ID <- Pvar.samp.df.T1$Parent.ID
scores.Pvar_CCC.T1$Time.Point <- Pvar.samp.df.T1$Time.Point

ggplot(scores.Pvar_CCC.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var T1 CCC") +
  theme_classic()
#ggsave("Pvar_T1_its_CCC.pdf")

#C TF
set.seed(30)
ord.Pvar_CCC.TF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CCC.TF)
scores.Pvar_CCC.TF <- as.data.frame(scores(ord.Pvar_CCC.TF))
scores.Pvar_CCC.TF$Treatment <- Pvar.samp.df.TF$Treatment
scores.Pvar_CCC.TF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.Pvar_CCC.TF$Time.Point <- Pvar.samp.df.TF$Time.Point

ggplot(scores.Pvar_CCC.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var CCC TF") +
  theme_classic()
#ggsave("Pvar_TF_its_TF_CCC.pdf")





#C27
set.seed(30)
ord.Pvar_C27 <- ordinate(Pvar.bacT1F.C27, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27)
scores.Pvar_C27 <- as.data.frame(scores(ord.Pvar_C27))
scores.Pvar_C27$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar_C27$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar_C27$Time.Point <- Pvar.samp.df.T1F$Time.Point

ggplot(scores.Pvar_C27, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C&D") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_its_T1TF.pdf")

#C27 T1
set.seed(30)
ord.Pvar_C27.T1 <- ordinate(Pvar.bac.T1.C27 , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27.T1)
scores.Pvar_C27.T1 <- as.data.frame(scores(ord.Pvar_C27.T1))
scores.Pvar_C27.T1$Treatment <- Pvar.samp.df.T1.C27$Treatment
scores.Pvar_C27.T1$Parent.ID <- Pvar.samp.df.T1.C27$Parent.ID
scores.Pvar_C27.T1$Time.Point <- Pvar.samp.df.T1.C27$Time.Point

ggplot(scores.Pvar_C27.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var T1 C27") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_T1.pdf")

#CD TF
set.seed(30)
ord.Pvar_C27.TF <- ordinate(Pvar.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C27.TF)
scores.Pvar_C27.TF <- as.data.frame(scores(ord.Pvar_C27.TF))
scores.Pvar_C27.TF$Treatment <- Pvar.samp.df.TF$Treatment
scores.Pvar_C27.TF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.Pvar_C27.TF$Time.Point <- Pvar.samp.df.TF$Time.Point

ggplot(scores.Pvar_C27.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C27 TF") +
  theme_classic()
#ggsave("Pvar_ITS2_C27_TF.pdf")



```

Pocillopora - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pacu
Pacu.bac.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pocillopora")
Pacu.bac.T1 <- subset_samples(Pacu.bac.T1F, Time.Point == "T1")
Pacu.bac.TF <- subset_samples(Pacu.bac.T1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F <- phyloseq::distance(Pacu.bac.T1F, method = "bray")
  Pacu.samp.df.T1F <- as(sample_data(Pacu.bac.T1F), "data.frame")            #sample df
  adonis(bc.Pacu.T1F ~ Treatment*Time.Point, data = Pacu.samp.df.T1F)      #adonis test
#write.csv(Pacu.samp.df.T1F, "Pacu_its2_samp_df.csv")
#T1
#make bray curtis data matrix 
  bc.PacuT1 <- phyloseq::distance(Pacu.bac.T1, method = "bray")
  Pacu.samp.df.T1 <- as(sample_data(Pacu.bac.T1), "data.frame") #sample df
  adonis(bc.PacuT1 ~ Clade*Treatment, data = Pacu.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Clade)
  permutest(beta.Pacu.T1, pairwise = T) 
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Treatment)
  permutest(beta.Pacu.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pacu.TF <- phyloseq::distance(Pacu.bac.TF, method = "bray")
  Pacu.samp.df.TF <- as(sample_data(Pacu.bac.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF ~ Clade*Treatment, data = Pacu.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Clade)
  permutest(beta.Pacu.TF, pairwise = T) 
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Treatment)
  permutest(beta.Pacu.TF, pairwise = T) 


## by clade: CCC vs C42
  
# C only, new objs
Pacu.bacT1F.CCC <- subset_samples(Pacu.bac.T1F, Clade == "CCC")
Pacu.bac.T1.CCC <- subset_samples(Pacu.bac.T1, Clade == "CCC")
Pacu.bac.TF.CCC <- subset_samples(Pacu.bac.TF, Clade == "CCC")
  
#T1TF.CCC
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F.CCC <- phyloseq::distance(Pacu.bacT1F.CCC, method = "bray")
  Pacu.samp.df.T1F.CCC <- as(sample_data(Pacu.bacT1F.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.T1F.CCC ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.CCC)

#T1.CCC
#make bray curtis data matrix 
  bc.PacuT1.CCC <- phyloseq::distance(Pacu.bac.T1.CCC, method = "bray")
  Pacu.samp.df.T1.CCC <- as(sample_data(Pacu.bac.T1.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.PacuT1.CCC ~ Treatment, data = Pacu.samp.df.T1.CCC)
# Homogeneity of dispersion test  
  beta.Pacu.T1.CCC <- betadisper(bc.PacuT1.CCC, Pacu.samp.df.T1.CCC$Treatment)
  permutest(beta.Pacu.T1.CCC, pairwise = T) 

#TF.CCC
#make bray curtis data matrix 
  bc.Pacu.TF.CCC <- phyloseq::distance(Pacu.bac.TF.CCC, method = "bray")
  Pacu.samp.df.TF.CCC <- as(sample_data(Pacu.bac.TF.CCC), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF.CCC ~ Treatment, data = Pacu.samp.df.TF.CCC)
# Homogeneity of dispersion test  
  beta.Pacu.TF.CCC <- betadisper(bc.Pacu.TF.CCC, Pacu.samp.df.TF.CCC$Treatment)
  permutest(beta.Pacu.TF.CCC, pairwise = T) 


# C42 only, new objs
Pacu.bacT1F.C42 <- subset_samples(Pacu.bac.T1F, Clade == "C42")
Pacu.bac.T1.C42 <- subset_samples(Pacu.bac.T1, Clade == "C42")
Pacu.bac.TF.C42 <- subset_samples(Pacu.bac.TF, Clade == "C42")
  
#T1TF.C42
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F.C42 <- phyloseq::distance(Pacu.bacT1F.C42, method = "bray")
  Pacu.samp.df.T1F.C42 <- as(sample_data(Pacu.bacT1F.C42), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.T1F.C42 ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.C42)

#T1.C42
#make bray curtis data matrix 
  bc.PacuT1.C42 <- phyloseq::distance(Pacu.bac.T1.C42, method = "bray")
  Pacu.samp.df.T1.C42 <- as(sample_data(Pacu.bac.T1.C42), "data.frame") #sample df
  adonis(bc.PacuT1.C42 ~ Treatment, data = Pacu.samp.df.T1.C42)#adonis test
# Homogeneity of dispersion test  
  beta.Pacu.T1.C42 <- betadisper(bc.PacuT1.C42, Pacu.samp.df.T1.C42$Treatment)
  permutest(beta.Pacu.T1.C42, pairwise = T) 

#TF.C42
#make bray curtis data matrix 
  bc.Pacu.TF.C42 <- phyloseq::distance(Pacu.bac.TF.C42, method = "bray")
  Pacu.samp.df.TF.C42 <- as(sample_data(Pacu.bac.TF.C42), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF.C42 ~ Treatment, data = Pacu.samp.df.TF.C42)
# Homogeneity of dispersion test  
  beta.Pacu.TF.C42 <- betadisper(bc.Pacu.TF.C42, Pacu.samp.df.TF.C42$Treatment)
  permutest(beta.Pacu.TF.C42, pairwise = T) 
``` 
Pacu NMDS
```{r}
# C only, new objs
Pacu.bacT1F.CCC <- subset_samples(Pacu.bac.T1F, Clade == "CCC")
Pacu.bac.T1.CCC <- subset_samples(Pacu.bac.T1, Clade == "CCC")
Pacu.bac.TF.CCC <- subset_samples(Pacu.bac.TF, Clade == "CCC")
# # pull df sections for craig Pacu.bac.T1F
# OTU_Pacu<-out_table(Pacu.bac.T1F)
# tax  = tax_table(coralDIV2_rare2)
# otu  = otu_table(coralDIV2_rare2)
# 
# OTU1 = as(otu_table(Pacu.bac.T1F), "matrix")
# # transpose if necessary
# if(taxa_are_rows(Pacu.bac.T1F)){OTU1 <- t(OTU1)}
# # Coerce to data.frame
# OTUdf = as.data.frame(OTU1)
# write.csv(OTUdf, "Pacuta_its2_T1TF.csv")
# 


library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pacu.Parent.colors <- c("305" = "green4",  "306" = "firebrick3", "307" ="deeppink4", "308" = "orange", "309" = "green1","310" = "gold",  "311"= "hotpink", "312" ="darkslateblue", "302" = "honeydew4", "303" = "khaki4","304" ="ivory4", "305" = "lightblue", Ambient = "Blue", High = "Red","CCC Ambient"="blue","CCC High"="red", "C42 Ambient"="darkblue", "C42 High"="darkred") 





#T1 and TF
ord.Pacu <- ordinate(Pacu.bac.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pacu)

scores.Pacu.samp.df <- as.data.frame(scores(ord.Pacu) ) 
scores.Pacu.samp.df$Time.Point <- Pacu.samp.df.T1F$Time.Point
scores.Pacu.samp.df$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu.samp.df$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu.samp.df$Clade <- Pacu.samp.df.T1F$Clade
scores.Pacu.samp.df$Code <- paste(Pacu.samp.df.T1F$Parent.ID, Pacu.samp.df.T1F$Time.Point)


ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
   # geom_line(aes(group = Code), color = "black") + 
  ggtitle("P.acu Bray Curtis") +
  theme_classic()
#ggsave("Pacu_T1TF_its2.pdf")

ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
     # geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var Bray Curtis") +
  theme_classic()
#ggsave("PacuT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pacu.bac.TF, ordinate(Pacu.bac, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PacuT1 <- ordinate(Pacu.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuT1)
scores.PacuT1 <- as.data.frame(scores(ord.PacuT1))
scores.PacuT1$Treatment <- Pacu.samp.df.T1$Treatment
scores.PacuT1$Parent.ID <- Pacu.samp.df.T1$Parent.ID
scores.PacuT1$Clade <- Pacu.samp.df.T1$Clade
scores.PacuT1$CladeTreat <- as.factor(paste(Pacu.samp.df.T1$Clade,Pacu.samp.df.T1$Treatment))


ggplot(scores.PacuT1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  #    geom_line(aes(group = Parent.ID), color = "black") + 
  ggtitle("P. var T1 Bray Curtis") +
  theme_classic()
#ggsave("Pacu_T1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PacuTF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuTF)
scores.PacuTF <- as.data.frame(scores(ord.PacuTF))
scores.PacuTF$Treatment <- Pacu.samp.df.TF$Treatment
scores.PacuTF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.PacuTF$Clade <- Pacu.samp.df.TF$Clade
scores.PacuTF$CladeTreat <- as.factor(paste(Pacu.samp.df.TF$Clade,Pacu.samp.df.TF$Treatment))

ggplot(scores.PacuTF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Clade, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Clade)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("P. var TF Bray Curtis") +
  theme_classic()
#gsave("Pacu_TF_its_nmds.pdf")










#### BY CLADE

#CCC T1 and TF
set.seed(30)
ord.Pacu_CCC  <- ordinate(Pacu.bac.T1F, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC)
scores.Pacu_CCC <- as.data.frame(scores(ord.Pacu_CCC))
scores.Pacu_CCC$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu_CCC$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu_CCC$Time.Point <- Pacu.samp.df.T1F$Time.Point

ggplot(scores.Pacu_CCC, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var CCC") +
  theme_classic()
#ggsave("PacuTFits_CCC.pdf")


#C T1
set.seed(30)
ord.Pacu_CCC.T1 <- ordinate(Pacu.bac.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.T1)
scores.Pacu_CCC.T1 <- as.data.frame(scores(ord.Pacu_CCC.T1))
scores.Pacu_CCC.T1$Treatment <- Pacu.samp.df.T1$Treatment
scores.Pacu_CCC.T1$Parent.ID <- Pacu.samp.df.T1$Parent.ID
scores.Pacu_CCC.T1$Time.Point <- Pacu.samp.df.T1$Time.Point

ggplot(scores.Pacu_CCC.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=2, colour = Parent.ID,shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4,shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var T1 CCC") +
  theme_classic()
#ggsave("Pacu_T1_its_CCC.pdf")

#C TF
set.seed(30)
ord.Pacu_CCC.TF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CCC.TF)
scores.Pacu_CCC.TF <- as.data.frame(scores(ord.Pacu_CCC.TF))
scores.Pacu_CCC.TF$Treatment <- Pacu.samp.df.TF$Treatment
scores.Pacu_CCC.TF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.Pacu_CCC.TF$Time.Point <- Pacu.samp.df.TF$Time.Point

ggplot(scores.Pacu_CCC.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var CCC TF") +
  theme_classic()
#ggsave("Pacu_TF_its_TF_CCC.pdf")



Pacu.bacT1F.C27  
Pacu.bac.T1.C27 
Pacu.bac.TF.C27 

#C27
set.seed(30)
ord.Pacu_C27 <- ordinate(Pacu.bacT1F.C27, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27)
scores.Pacu_C27 <- as.data.frame(scores(ord.Pacu_C27))
scores.Pacu_C27$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu_C27$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu_C27$Time.Point <- Pacu.samp.df.T1F$Time.Point

ggplot(scores.Pacu_C27, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Time.Point,size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Time.Point)) +  
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C&D") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_its_T1TF.pdf")

#C27 T1
set.seed(30)
ord.Pacu_C27.T1 <- ordinate(Pacu.bac.T1.C27 , "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27.T1)
scores.Pacu_C27.T1 <- as.data.frame(scores(ord.Pacu_C27.T1))
scores.Pacu_C27.T1$Treatment <- Pacu.samp.df.T1.C27$Treatment
scores.Pacu_C27.T1$Parent.ID <- Pacu.samp.df.T1.C27$Parent.ID
scores.Pacu_C27.T1$Time.Point <- Pacu.samp.df.T1.C27$Time.Point

ggplot(scores.Pacu_C27.T1, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P var T1 C27") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_T1.pdf")

#CD TF
set.seed(30)
ord.Pacu_C27.TF <- ordinate(Pacu.bac.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C27.TF)
scores.Pacu_C27.TF <- as.data.frame(scores(ord.Pacu_C27.TF))
scores.Pacu_C27.TF$Treatment <- Pacu.samp.df.TF$Treatment
scores.Pacu_C27.TF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.Pacu_C27.TF$Time.Point <- Pacu.samp.df.TF$Time.Point

ggplot(scores.Pacu_C27.TF, aes(x = NMDS1, y = NMDS2)) + 
geom_point(aes(stroke=3, colour = Parent.ID, shape=Treatment, size = 4) )+
    geom_point(aes(colour = Treatment, size = 4, shape=Treatment)) + 
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. var C27 TF") +
  theme_classic()
#ggsave("Pacu_ITS2_C27_TF.pdf")





```












































#dispersion, figuring out how to do
```{r}
    
# Homogeneity of dispersion test bray curtis
  #   bc.all <- phyloseq::distance(Bac_RelA_stats.T1F, method = "bray")
  # samp.df <- as(sample_data(Bac_RelA_stats.T1F), "data.frame") #sample df
  
bc.disp.all <- betadisper(bc.all, samp.df$Species, type = "centroid") #by species
bc.disp<-as.data.frame(bc.disp.all$distances) #extract distances
names(bc.disp)[names(bc.disp) == "bc.disp.all$distances"] <- "bc.distances" #rename
bc.disp.Species<-cbind(bc.disp, samp.df) #bind dist to sd
hist(bc.disp$bc.distances) #none of these are great
hist(log10(bc.disp$bc.distances))
hist(sqrt(bc.disp$bc.distances))
hist((bc.disp$bc.distances)^1/3)
hist(1/bc.disp$bc.distances)

bc.disp<-lmer(bc.distances~Species*Treatment*Time.Point+(1|Parent.ID)+(1|Tank.num), data=bc.disp.Species)
anova(bc.disp)

qqPlot(residuals(bc.disp)) 

  emm<-emmeans(bc.disp,  ~ Species, adjust="Tukey")
cld(emm)
pairs(emm)
```

#Alpha diversit: observed richness, Shannon (+evenness) - Hannah  
Use rarefied data, not relative abunances   
df:Bac.s and
```{r}           
#trying to install microbiome on old R version: https://microbiome.github.io/tutorials/Installation.html https://bioconductor.org/install/
Sp.colors <- c("Montipora_capitata"= "#D43F3AFF","Porites_compressa"="#EEA236FF","Pavona_varians"="#5CB85CFF","Pocillopora_acuta"="#46B8DAFF", "High"="darkRed","Ambient"="navyBlue")

Bac.al.alpha<-Bac.s #make a copy of df, use COUNTS

#Bac.al.alpha<-subset_samples(Bac.al.alpha, Time.Point=="T1" |Time.Point=="TF")  # ONLY KEEP T1 and TF!
Bac.al.alpha2 <- prune_taxa(taxa_sums(Bac.al.alpha) > 0, Bac.al.alpha) #prune taxa 0s
Bac.al.alpha2.sd <- as(sample_data(Bac.al.alpha2), "data.frame") #create sample data frame to view


#####try microbiome package for evenness and alpha diversity. use raw counts
library(microbiome)  
library(knitr)

erich.tab <-microbiome::alpha(Bac.al.alpha2, index = "all") # pull out: observed, diversity_shannon, evenness_pielou

erich.tab2<-erich.tab[-c(2:4,6:8,10:22)] #keep the cols you need

 erich.tab2$Treatment <- Bac.al.alpha2.sd$Treatment       #add back in meta
  erich.tab2$Time.Point <- Bac.al.alpha2.sd$Time.Point
  erich.tab2$Parent.ID <- Bac.al.alpha2.sd$Parent.ID
  erich.tab2$ID <- Bac.al.alpha2.sd$Frag.ID
  erich.tab2$Species <- Bac.al.alpha2.sd$Species
    erich.tab2$Clade <- Bac.al.alpha2.sd$Clade
    erich.tab2$Tank.num <- Bac.al.alpha2.sd$Tank.num

  erich.tab2$Time.Point = factor(erich.tab2$Time.Point, levels = c("T0", "T1", "TF"))


#all species - to show significance and to subset
  erich.tab2T0 <- subset(erich.tab2, Time.Point=="T0")#make T0 to use below

erich.tab2 <- subset(erich.tab2, Time.Point=="T1"|Time.Point=="TF")

#ANOVA
#shannon
  aov.shannon.species = aov(diversity_shannon ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.shannon.species) # sig diff between species
  TukeyHSD(aov.shannon.species)
#observed  
  aov.observed.species = aov(observed ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.observed.species) # sig diff between species
  TukeyHSD(aov.observed.species)
#evenness
  aov.evenness.species = aov(evenness_pielou ~ Species*Time.Point*Treatment, data=erich.tab2)
  summary(aov.evenness.species) # sig diff between species
  TukeyHSD(aov.evenness.species)  
#Plot
  boxplot(diversity_shannon ~ Species, data=erich.tab2, ylab="Shannon's diversity") 
  boxplot(observed ~ Species, data=erich.tab2, ylab="Observed") 
  boxplot(evenness_pielou ~ Species, data=erich.tab2, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.species <- levels(erich.tab2$Species) # get the variables

# make a pairwise list that we want to compare.
  al.species.pairs <- combn(seq_along(al.species), 2, simplify = FALSE, FUN = function(i)al.species[i])
  print(al.species.pairs)

 erich.tab2$Code<-paste(erich.tab2$Species,erich.tab2$Clade)
  
#observed
p1 <- ggviolin(erich.tab2, x = "Species", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

    p1 <- ggviolin(erich.tab2, x = "Code", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

  #Shannon
p1 <- ggviolin(erich.tab2, x = "Species", y = "diversity_shannon",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.species.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2, x = "Species", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.species.pairs) #non-parametric test (Wilcoxon test)
  print(p1)

p1 <- ggviolin(erich.tab2, x = "Code", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

#by species:
erich.tab2.mcap<-subset(erich.tab2, Species=="Montipora_capitata")
erich.tab2.pcomp<-subset(erich.tab2, Species=="Porites_compressa")
erich.tab2.pvar<-subset(erich.tab2, Species=="Pavona_varians")
erich.tab2.pacu<-subset(erich.tab2, Species=="Pocillopora_acuta")


#look at data
#observed
hist(erich.tab2$observed)
ggqqplot(erich.tab2$observed)
hist(log(erich.tab2$observed))
ggqqplot(log10(erich.tab2$observed))
erich.tab2$observed_log<-log10(erich.tab2$observed)

hist(erich.tab2.mcap$observed_log)
ggqqplot(erich.tab2.mcap$observed_log)
 hist(erich.tab2.pcomp$observed_log)
ggqqplot(erich.tab2.pcomp$observed_log)
   hist(erich.tab2.pvar$observed_log)
ggqqplot(erich.tab2.pvar$observed_log)
        hist(erich.tab2.pacu$observed_log)
ggqqplot(erich.tab2.pacu$observed_log)



#shannon
hist(erich.tab2$diversity_shannon)
ggqqplot(erich.tab2$diversity_shannon)
hist(log10(erich.tab2$diversity_shannon))
ggqqplot(log10(erich.tab2$diversity_shannon))
hist(sqrt(erich.tab2$diversity_shannon))
ggqqplot(sqrt(erich.tab2$diversity_shannon))
hist((erich.tab2$diversity_shannon)^(1/4))
ggqqplot(sqrt(erich.tab2$diversity_shannon))


#evenness
hist(erich.tab2$evenness_pielou) #best
ggqqplot(erich.tab2$evenness_pielou)
hist(log10(erich.tab2$evenness_pielou))#worse
ggqqplot(erich.tab2$evenness_pielou)
erich.tab2$even.t<-sqrt(max(erich.tab2$evenness_pielou+1)-erich.tab2$evenness_pielou)
hist(sqrt(max(erich.tab2$evenness_pielou+1)-erich.tab2$evenness_pielou))
ggqqplot(sqrt(max(erich.tab2$evenness_pielou+1)-erich.tab2$evenness_pielou))

hist(erich.tab2.mcap$evenness_pielou)
ggqqplot(erich.tab2.mcap$evenness_pielou)
 hist(erich.tab2.pcomp$evenness_pielou)
ggqqplot(erich.tab2.pcomp$evenness_pielou)
   hist(erich.tab2.pvar$evenness_pielou)
ggqqplot(erich.tab2.pvar$evenness_pielou)
        hist(erich.tab2.pacu$evenness_pielou)
ggqqplot(erich.tab2.pacu$evenness_pielou)

hist(erich.tab2.mcap$even.t)
ggqqplot(erich.tab2.mcap$even.t)
 hist(erich.tab2.pcomp$even.t)
ggqqplot(erich.tab2.pcomp$even.t)
   hist(erich.tab2.pvar$even.t)
ggqqplot(erich.tab2.pvar$even.t)
        hist(erich.tab2.pacu$even.t)
ggqqplot(erich.tab2.pacu$even.t)






#by species:
erich.tab2.mcap<-subset(erich.tab2, Species=="Montipora_capitata")
erich.tab2.pcomp<-subset(erich.tab2, Species=="Porites_compressa")
erich.tab2.pvar<-subset(erich.tab2, Species=="Pavona_varians")
erich.tab2.pacu<-subset(erich.tab2, Species=="Pocillopora_acuta")

hist(erich.tab2.mcap$observed_log)
hist(erich.tab2.mcap$observed) #better
hist(erich.tab2.pcomp$observed_log)#better
hist(erich.tab2.pcomp$observed) 
hist(erich.tab2.pvar$observed_log)#better
hist(erich.tab2.pvar$observed) 
hist(erich.tab2.pacu$observed_log)#better
hist(erich.tab2.pacu$observed) 





```
#Observed Richness 
```{r}      
#observed    
 #log it
#T0
hist(erich.tab2T0$observed)
ggqqplot(erich.tab2T0$observed)
erich.tab2T0$observed_log<-log10(erich.tab2T0$observed)
 hist(erich.tab2T0$observed_log)

observed_alpha_modT0<-lmer(observed_log ~ Species+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  observed_alpha_modT0<-lmer(observed_log ~ Species*Clade+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  
#Richness
p1 <- ggviolin(erich.tab2T0, x = "Species", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
#Evenness
p1 <- ggviolin(erich.tab2T0, x = "Species", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

 
 
 
 
 erich.tab2.T1F<-subset(erich.tab2, Time.Point=="T1"|Time.Point=="TF")
 
 #T1TF
observed_alpha_mod<-lmer(observed_log ~ Species*Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.T1F) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_mod) 
  qqPlot(residuals(observed_alpha_mod)) 
  emm<-emmeans(observed_alpha_mod,  ~ Species*Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  observed_alpha_mod<-lmer(observed_log ~ Species*Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2.T1F) #adding tank is singularity
  anova(observed_alpha_mod) 
  qqPlot(residuals(observed_alpha_mod)) 
  emm<-emmeans(observed_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)



################Mcap only
  hist(erich.tab2.mcap$observed_log)
ggqqplot(erich.tab2.mcap$observed_log)
  
 #t1TF
erich.tab2.mcap.T1F<-subset(erich.tab2.mcap, Time.Point=="T1"|Time.Point=="TF")
  Mcap_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment+(1|Parent.ID), data=erich.tab2.mcap.T1F) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.T1F) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.mcap.T1<-subset(erich.tab2.mcap.T1F, Time.Point=="T1")

  Mcap_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.mcap.TF<-subset(erich.tab2.mcap.T1F, Time.Point=="TF")

   Mcap_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)

 #t0
  erich.tab2.mcap.T0<-subset(erich.tab2T0, Species=="Montipora_capitata")

  Mcap_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.mcap.T0) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)

  
  
####### PCOMP only ####
   erich.tab2.pcomp.T1F<-subset(erich.tab2.pcomp, Time.Point=="T1"|Time.Point=="TF")

  #t1TF
  pcomp_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp.T1F) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp.T1F) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pcomp.T1<-subset(erich.tab2.pcomp, Time.Point=="T1")

  pcomp_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Clade*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pcomp.TF<-subset(erich.tab2.pcomp, Time.Point=="TF")

   pcomp_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

 #t0
  erich.tab2.pcomp.T0<-subset(erich.tab2T0, Species=="Porites_compressa")

  pcomp_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.pcomp.T0) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
####### pvar only####
   
  #t1TF
  pvar_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pvar.T1<-subset(erich.tab2.pvar, Time.Point=="T1")

  pvar_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pvar.TF<-subset(erich.tab2.pvar, Time.Point=="TF")

   pvar_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #t0
  erich.tab2.pvar.T0<-subset(erich.tab2T0, Species=="Pavona_varians")

  pvar_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.pvar.T0) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  ####### pacu only####

  #t1TF
  pacu_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed_log ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pacu.T1<-subset(erich.tab2.pacu, Time.Point=="T1")

  pacu_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID), data=erich.tab2.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pacu.TF<-subset(erich.tab2.pacu, Time.Point=="TF")

   pacu_alpha_mod<-lmer(observed_log ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed_log ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #t0
  erich.tab2.pacu.T0<-subset(erich.tab2T0, Species=="Pocilopora_acuta")

  pacu_alpha_mod<-lmer(observed_log ~ Clade+(1|Tank.num), data=erich.tab2.pacu.T0) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  
```
alpha Observed Plots, no clade
```{r}  
#Means   
observed_mean <- ddply(erich.tab2, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                 N    = length(observed[!is.na(observed)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(observed, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(observed, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(observed, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
observed_mean

#Mcap 
Mcap_observed_mean<-subset(observed_mean, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_observed_mean<-subset(observed_mean, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_observed_mean<-subset(observed_mean, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_observed_mean<-subset(observed_mean, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_observed_plot<-ggplot(data=Mcap_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_observed_plot


#plot Pcomp
Pcomp_observed_plot<-ggplot(data=Pcomp_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_observed_plot

#Pvar plots
#plot Pvar
Pvar_observed_plot<-ggplot(data=Pvar_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
    xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_observed_plot


#Pacu plots
#plot Pacu
Pacu_observed_plot<-ggplot(data=Pacu_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_observed_plot

###### All plots together
Mcap_observed_plot<-Mcap_observed_plot+ theme(legend.position = "none") #remove legend
Pcomp_observed_plot<-Pcomp_observed_plot+ theme(legend.position = "none") #remove legend
Pvar_observed_plot<-Pvar_observed_plot+ theme(legend.position = "none") #remove legend
Pacu_observed_plot<-Pacu_observed_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_observed_plot, Pcomp_observed_plot, Pvar_observed_plot,Pacu_observed_plot, nrow = 1)
 
```  

alpha Observed Plots, clade  
```{r}   
# color pal from eta mcap.g.Parent.colors
alpha.colors <- c( "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#1F78B4","C High"="#6A3D9A", "CD Ambient"="#E31A1C", "CD High"="#FF7F00") 

#Means
observed_mean.C <- ddply(erich.tab2, c("Treatment", "Species", "Time.Point","Clade"), summarise, #summarize cell counts
                 N    = length(observed[!is.na(observed)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(observed, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(observed, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(observed, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
observed_mean.C
observed_mean.C$Code<-paste(observed_mean.C$Clade, observed_mean.C$Treatment)

#Mcap 
Mcap_observed_mean.C<-subset(observed_mean.C, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_observed_mean.C<-subset(observed_mean.C, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_observed_mean.C<-subset(observed_mean.C, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_observed_mean.C<-subset(observed_mean.C, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_observed_plot<-ggplot(data=Mcap_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
    xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  # ylim(0, 190) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_observed_plot

#boxplot

Rich16.plot<-ggplot(erich.tab2.mcap.T1, aes(x=Treatment, y=mean, color=Treatment)) + 
  scale_color_manual(values=c("blue", "red"))+
     # coord_cartesian(ylim = c(0, 10))+
   theme_classic()+
 theme(legend.position="none") +# Remove legend
  geom_boxplot(notch=F);Rich16.plot

library(tidyverse)
library(hrbrthemes)
library(viridis)

erich.tab2.mcap.T1$code2<-paste(erich.tab2.mcap.T1$Code, erich.tab2.mcap.T1$Treatment)


erich.tab2.mcap.T1 %>%
  ggplot( aes(x=code2, y=observed, fill=Treatment)) +
    geom_boxplot() +
    scale_color_manual(values=c("red", "blue"))+
    # scale_fill_viridis(discrete = TRUE, alpha=0.6, option="A") +
    # theme_ipsum() +
    theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
    ggtitle("Mcap Rich T1 ") +
    xlab("")

#Pcomp plots
#plot Pcomp
Pcomp_observed_plot<-ggplot(data=Pcomp_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
   xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_observed_plot

#Pvar plots
#plot Pvar
Pvar_observed_plot<-ggplot(data=Pvar_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
   xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_observed_plot


#plot Pacu
Pacu_observed_plot<-ggplot(data=Pacu_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 180) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_observed_plot

###### All plots together
Mcap_observed_plot<-Mcap_observed_plot+ theme(legend.position = "none") #remove legend
Pcomp_observed_plot<-Pcomp_observed_plot+ theme(legend.position = "none") #remove legend
Pvar_observed_plot<-Pvar_observed_plot+ theme(legend.position = "none") #remove legend
Pacu_observed_plot<-Pacu_observed_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_observed_plot, Pcomp_observed_plot, Pvar_observed_plot,Pacu_observed_plot, nrow = 1)
 
```
 
#evenness_pielou Richness 
```{r}   
#evenness_pielou  
#not transformeds

evenness_pielou_alpha_mod<-lmer(evenness_pielou ~ Species*Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2) #adding tank is singularity
  anova(evenness_pielou_alpha_mod) 
  qqPlot(residuals(evenness_pielou_alpha_mod)) 
  emm<-emmeans(evenness_pielou_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  evenness_pielou_alpha_mod<-lmer(evenness_pielou ~ Species*Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2) #adding tank is singularity
  anova(evenness_pielou_alpha_mod) 
  qqPlot(residuals(evenness_pielou_alpha_mod)) 
  emm<-emmeans(evenness_pielou_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

#T0
hist(erich.tab2T0$evenness_pielou)
ggqqplot(erich.tab2T0$evenness_pielou)


observed_alpha_modT0<-lmer(evenness_pielou ~ Species+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #this is nesting so not sure you can do this this way. 
  observed_alpha_modT0<-lmer(evenness_pielou ~ Species*Clade+(1|Tank.num), data=erich.tab2T0) #adding tank is singularity
set.seed(30)
  anova(observed_alpha_modT0) 
  qqPlot(residuals(observed_alpha_modT0)) 
  emm<-emmeans(observed_alpha_modT0,  ~ Species*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  

################Mcap only
 #t1TF
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID), data=erich.tab2.mcap) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.mcap.T1<-subset(erich.tab2.mcap, Time.Point=="T1")

  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Clade*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.mcap.TF<-subset(erich.tab2.mcap, Time.Point=="TF")

   Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)

#T0
  #t1
  erich.tab2.mcap.T0<-subset(erich.tab2.mcap, Time.Point=="T0")
hist(erich.tab2.mcap.T0$evenness_pielou)
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.mcap.T0) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
####### PCOMP only
  #t1TF
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pcomp.T1<-subset(erich.tab2.pcomp, Time.Point=="T1")

  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pcomp.TF<-subset(erich.tab2.pcomp, Time.Point=="TF")

   pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

#t0
  #t1
  erich.tab2.pcomp.T0<-subset(erich.tab2.pcomp, Time.Point=="T0")
hist(erich.tab2.pcomp.T0$evenness_pielou)
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.pcomp.T0) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
####### pvar only
  #t1TF
  pvar_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pvar.T1<-subset(erich.tab2.pvar, Time.Point=="T1")

  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pvar.TF<-subset(erich.tab2.pvar, Time.Point=="TF")

   pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
 
  
    #t0
  erich.tab2.pvar.T0<-subset(erich.tab2.pvar, Time.Point=="T0")
hist(erich.tab2.pvar.T0$evenness_pielou)
   pvar_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.pvar.T0) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 
  
  ####### pacu only
  #t1TF
  pacu_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2.pacu.T1<-subset(erich.tab2.pacu, Time.Point=="T1")

  pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2.pacu.TF<-subset(erich.tab2.pacu, Time.Point=="TF")

   pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #t0
  erich.tab2.pacu.T0<-subset(erich.tab2.pacu, Time.Point=="T0")
hist(erich.tab2.pacu.T0$evenness_pielou)#medium
   pacu_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2.pacu.T0) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
```
alpha evenness_pielou Plots, no clade
```{r} 
#Means
evenness_pielou_mean <- ddply(erich.tab2, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
evenness_pielou_mean

#Mcap 
Mcap_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_evenness_pielou_plot<-ggplot(data=Mcap_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_evenness_pielou_plot


#plot Pcomp
Pcomp_evenness_pielou_plot<-ggplot(data=Pcomp_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_evenness_pielou_plot

#Pvar plots
#plot Pvar
Pvar_evenness_pielou_plot<-ggplot(data=Pvar_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_evenness_pielou_plot


#Pacu plots
#plot Pacu
Pacu_evenness_pielou_plot<-ggplot(data=Pacu_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_evenness_pielou_plot

###### All plots together
Mcap_evenness_pielou_plot<-Mcap_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pcomp_evenness_pielou_plot<-Pcomp_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pvar_evenness_pielou_plot<-Pvar_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pacu_evenness_pielou_plot<-Pacu_evenness_pielou_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_evenness_pielou_plot, Pcomp_evenness_pielou_plot, Pvar_evenness_pielou_plot,Pacu_evenness_pielou_plot, nrow = 1)
 
```  

alpha evenness_pielou Plots, clade  
```{r}
# color pal from eta mcap.g.Parent.colors
alpha.colors <- c( "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#1F78B4","C High"="#6A3D9A", "CD Ambient"="#E31A1C", "CD High"="#FF7F00") 

#Means
evenness_pielou_mean.C <- ddply(erich.tab2, c("Treatment", "Species", "Time.Point","Clade"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
evenness_pielou_mean.C
evenness_pielou_mean.C$Code<-paste(evenness_pielou_mean.C$Clade, evenness_pielou_mean.C$Treatment)

#Mcap 
Mcap_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_evenness_pielou_plot<-ggplot(data=Mcap_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_evenness_pielou_plot

#Pcomp plots
#plot Pcomp
Pcomp_evenness_pielou_plot<-ggplot(data=Pcomp_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
 xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_evenness_pielou_plot

#Pvar plots
#plot Pvar
Pvar_evenness_pielou_plot<-ggplot(data=Pvar_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
 xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_evenness_pielou_plot


#plot Pacu
Pacu_evenness_pielou_plot<-ggplot(data=Pacu_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
 xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_evenness_pielou_plot

###### All plots together
Mcap_evenness_pielou_plot<-Mcap_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pcomp_evenness_pielou_plot<-Pcomp_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pvar_evenness_pielou_plot<-Pvar_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pacu_evenness_pielou_plot<-Pacu_evenness_pielou_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_evenness_pielou_plot, Pcomp_evenness_pielou_plot, Pvar_evenness_pielou_plot,Pacu_evenness_pielou_plot, nrow = 1)
 
```
 




















#Observed Richness OLD ANOVA CODE DELETE WHEN UPDATED ABOVE
```{r}
#observed  


################Mcap only


  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Time.Point*Treatment, data=erich.tab2.mcap)
  summary(aov.diversity_shannon.mcap) # sig diff between species
  TukeyHSD(aov.diversity_shannon.mcap)
  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Time.Point*Treatment*Clade, data=erich.tab2.mcap)
  summary(aov.diversity_shannon.mcap) # sig diff between species
  
  #lmer
shannon_Mcap_model<-lmer(diversity_shannon~Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), erich.tab2.mcap)
  anova(shannon_Mcap_model) 
  qqPlot(residuals(shannon_Mcap_model)) 
  emm<-emmeans(shannon_Mcap_model,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  
  
  
 #t1
  erich.tab2.mcap.T1<-subset(erich.tab2.mcap, Time.Point=="T1")

  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Treatment, data=erich.tab2.mcap.T1)
  summary(aov.diversity_shannon.mcap) # sig diff between species
  TukeyHSD(aov.diversity_shannon.mcap)
  
  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.mcap.T1)
  summary(aov.diversity_shannon.mcap) # sig diff between species 
  
  #tf
  erich.tab2.mcap.TF<-subset(erich.tab2.mcap, Time.Point=="TF")

  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Treatment, data=erich.tab2.mcap.TF)
  summary(aov.diversity_shannon.mcap) # sig diff between species
  TukeyHSD(aov.diversity_shannon.mcap)
  
  aov.diversity_shannon.mcap = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.mcap.TF)
  summary(aov.diversity_shannon.mcap) # sig diff between species 

  
####### PCOMP only
 #t1TF
  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Time.Point*Treatment, data=erich.tab2.pcomp)
  summary(aov.diversity_shannon.pcomp) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pcomp)
  
  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Time.Point*Treatment*Clade, data=erich.tab2.pcomp)
  summary(aov.diversity_shannon.pcomp) # sig diff between species
  
 #t1
  erich.tab2.pcomp.T1<-subset(erich.tab2.pcomp, Time.Point=="T1")

  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Treatment, data=erich.tab2.pcomp.T1)
  summary(aov.diversity_shannon.pcomp) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pcomp)
  
  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.pcomp.T1)
  summary(aov.diversity_shannon.pcomp) # sig diff between species 
  
  #tf
  erich.tab2.pcomp.TF<-subset(erich.tab2.pcomp, Time.Point=="TF")

  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Treatment, data=erich.tab2.pcomp.TF)
  summary(aov.diversity_shannon.pcomp) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pcomp)
  
  aov.diversity_shannon.pcomp = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.pcomp.TF)
  summary(aov.diversity_shannon.pcomp) # sig diff between species 
  
################pacu only
 #t1TF
  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Time.Point*Treatment, data=erich.tab2.pacu)
  summary(aov.diversity_shannon.pacu) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pacu)
  
  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Time.Point*Treatment*Clade, data=erich.tab2.pacu)
  summary(aov.diversity_shannon.pacu) # sig diff between species
  
 #t1
  erich.tab2.pacu.T1<-subset(erich.tab2.pacu, Time.Point=="T1")

  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Treatment, data=erich.tab2.pacu.T1)
  summary(aov.diversity_shannon.pacu) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pacu)
  
  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.pacu.T1)
  summary(aov.diversity_shannon.pacu) # sig diff between species 
  
  #tf
  erich.tab2.pacu.TF<-subset(erich.tab2.pacu, Time.Point=="TF")

  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Treatment, data=erich.tab2.pacu.TF)
  summary(aov.diversity_shannon.pacu) # sig diff between species
  TukeyHSD(aov.diversity_shannon.pacu)
  
  aov.diversity_shannon.pacu = aov(diversity_shannon ~ Treatment*Clade, data=erich.tab2.pacu.TF)
  summary(aov.diversity_shannon.pacu) # sig diff between species 

```
OLDER alpha, delete when ready
```{r}
#### older alpha

#########OK you are removing T0 here and then you need to make a new samp data frame and then do below. 

 


erich <- estimate_richness(Bac.r2.alpha2, measures = c("Observed", "Shannon")) #select which estimates and estimate
  erich$Treatment <- Bac.r2.sd$Treatment       #add back in meta
  erich$Time.Point <- Bac.r2.sd$Time.Point
  erich$Parent.ID <- Bac.r2.sd$Parent.ID
  erich$ID <- Bac.r2.sd$ID
  erich$Species <- Bac.r2.sd$Species
  erich$Time.Point = factor(erich$Time.Point, levels = c("T0", "T1", "TF"))

  
  
 
#stats ALL THE DATA
 library(ggpubr)
  #look at data
  
#observed
# hist(erich$Observed) 
# ggqqplot(erich$Observed)
# erich$Observed_log<-log10(erich$Observed)
# hist(erich$Observed_log) 

## subset by coral species for OBSERVED
erich.mcap<-subset(erich, Species=="Montipora_capitata")
  hist(erich.mcap$Observed) 
  # hist(erich.mcap$Observed_log) 

erich.pcomp<-subset(erich, Species=="Porites_compressa")
  hist(erich.pcomp$Observed) #meh
  #   hist(erich.pcomp$Observed_log) #Better 
  # ggqqplot(erich.pcomp$Observed_log) 
 
erich.pvar<-subset(erich, Species=="Pavona_varians")
 hist(erich.pvar$Observed) #
 # hist(erich.pvar$Observed_log) #better
 #  ggqqplot(erich.pvar$Observed_log) #

erich.pacu<-subset(erich, Species=="Pocillopora_acuta")
 hist(erich.pacu$Observed) 
  hist(erich.pacu$Observed_log) 
  ggqqplot(erich.pacu$Observed_log) 

  
## subset for Shannon
#shannon
hist(erich$Shannon)
ggqqplot(erich$Shannon)
# erich$Shannon_log<-log10(erich$Shannon)
# hist(erich$Shannon_log)
# ggqqplot(erich$Shannon_log)
# erich$Shannon_sq<-sqrt(erich$Shannon)
# hist(erich$Shannon_sq)
# ggqqplot(erich$Shannon_sq)


erich.mcap<-subset(erich, Species=="Montipora_capitata")
  hist(erich.mcap$Shannon) # 
  ggqqplot(erich.mcap$Shannon) #
  #   hist(erich.mcap$Shannon_log) # 
  # ggqqplot(erich.mcap$Shannon) #
 
erich.pcomp<-subset(erich, Species=="Porites_compressa")
  hist(erich.pcomp$Shannon) # normal 
  ggqqplot(erich.pcomp$Shannon) 
  # hist(erich.pcomp$Shannon_log) # normal 
  # ggqqplot(erich.pcomp$Shannon) 

erich.pvar<-subset(erich, Species=="Pavona_varians")
 hist(erich.pvar$Shannon) #not normal 
  ggqqplot(erich.pvar$Shannon) #
  
erich.pacu<-subset(erich, Species=="Pocillopora_acuta")
 hist(erich.pacu$Shannon) #not awfuk 
  ggqqplot(erich.pacu$Shannon) #not bad
  

### ALL SPECIES
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.species = aov(Shannon ~ Species, data=erich)

#Call for the summary of that ANOVA, which will include P-values
  summary(aov.shannon.species) # sig diff between species
  TukeyHSD(aov.shannon.species)

#Plot
  boxplot(Shannon ~ Species, data=erich, ylab="Shannon's diversity") 

  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.species = aov(Observed ~ Species*Treatment*Time.Point, data=erich)
  #aov.observed.species = aov(Observed ~ Species, data=erich)

#Call for the summary of that ANOVA, which will include P-values
  summary(aov.observed.species) # sig diff between species
  TukeyHSD(aov.observed.species)

#Plot
  boxplot(Observed ~ Species, data=erich, ylab="observed's diversity") 
  
#### LHK paper code
  
alphaplot <- plot_richness(Bac.r2.alpha, x="Species", measures=c("Observed","Shannon", "Simpson"), color="Time.Point") + 
  labs(x="Genet") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(strip.text.x = element_text(size=10, face="bold"),
        strip.background = element_rect(fill="#FFFFFF"))


alphaplot


#stats
erich <- estimate_richness(Bac.r2.alpha, measures=c("Observed","Shannon", "Simpson"))
#make dataframe
SampleID <- row.names(erich) 
# makes a new dataframe with the specified columnes
alpha <- data.frame(SampleID, erich) 
s <- data.frame(sample_data(Bac.r2.alpha)) 
# Change first column title to SampleID to match distances dataframe
colnames(s)[1] <- "SampleID" 
# merges metadata df and distances df
alphaGC <- merge(alpha, s, by = "SampleID") 

```

### MCAP alpha, df: erich.tab2.mcap
```{r} 
#ANOVA
erich.tab2.mcap$treat.time<-paste(erich.tab2.mcap$Treatment,erich.tab2.mcap$Time.Point)
erich.tab2.mcap$treat.time<-as.factor(erich.tab2.mcap$treat.time)

#shannon
  aov.shannon.treat.time = aov(diversity_shannon ~ treat.time, data=erich.tab2.mcap)
  summary(aov.shannon.treat.time) # sig diff between treat.time
  TukeyHSD(aov.shannon.treat.time)
#observed  
  aov.observed.treat.time = aov(observed ~ treat.time, data=erich.tab2.mcap)
  summary(aov.observed.treat.time) # sig diff between treat.time
  TukeyHSD(aov.observed.treat.time)
#evenness
  aov.evenness.treat.time = aov(evenness_pielou ~ treat.time, data=erich.tab2.mcap)
  summary(aov.evenness.treat.time) # sig diff between treat.time
  TukeyHSD(aov.evenness.treat.time)  
#Plot
  boxplot(diversity_shannon ~ treat.time, data=erich.tab2.mcap, ylab="Shannon's diversity") 
  boxplot(observed ~ treat.time, data=erich.tab2.mcap, ylab="Observed") 
  boxplot(evenness_pielou ~ treat.time, data=erich.tab2.mcap, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.treat.time <- levels(erich.tab2.mcap$treat.time) # get the variables

# make a pairwise list that we want to compare.
  al.treat.time.pairs <- combn(seq_along(al.treat.time), 2, simplify = FALSE, FUN = function(i)al.treat.time[i])
  print(al.treat.time.pairs)

#observed
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "observed",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
    
#Shannon
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "diversity_shannon",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2.mcap, x = "treat.time", y = "evenness_pielou",
   add = "boxplot", fill = "treat.time", palette = c(Mcap.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)






#OLD

#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesMcap = aov(Shannon ~ Treatment*Time.Point, data=erich.mcap)
  summary(aov.shannon.speciesMcap) # sig diff between species
  TukeyHSD(aov.shannon.speciesMcap)

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.mcap, ylab="Shannon's diversity") 

  
  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesMcap = aov(Observed ~ Treatment*Time.Point, data=erich.mcap)
  summary(aov.observed.speciesMcap) # sig diff between species
  TukeyHSD(aov.observed.speciesMcap)
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.mcap, ylab="observed's diversity") 
```
#### Do by C and CD
```{r}
#mcap  erich.mcap


  # by parent ID set clade dom
  erich.mcap$Clade[erich.mcap$Parent.ID==363]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==364]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==365]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==366]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==367]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==368]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==369]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==370]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==371]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==372]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==373]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==374]<-"CD"
  erich.mcap$Clade<-as.factor(erich.mcap$Clade) #change to factor

## let look at C vs CD
  #Run the ANOVA and save it as an object
  aov.shannon.speciesMcap_CCD = aov(Shannon ~ Clade*Treatment*Time.Point, data=erich.mcap)
  summary(aov.shannon.speciesMcap_CCD) # sig diff between species
  TukeyHSD(aov.shannon.speciesMcap_CCD)

#Plot
  boxplot(Shannon ~ Clade*Treatment*Time.Point, data=erich.mcap, ylab="Shannon's diversity") 

#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesMcap_CCD = aov(Observed ~ Clade*Treatment*Time.Point, data=erich.mcap)
  summary(aov.observed.speciesMcap_CCD) # sig diff between species
  TukeyHSD(aov.observed.speciesMcap_CCD)
#Plot
  boxplot(Observed ~ Clade*Treatment*Time.Point, data=erich.mcap, ylab="observed's diversity") 
  
``` 
# Pcomp alpha
```{r} 
#ANOVA
erich.tab2.pcomp$treat.time<-paste(erich.tab2.pcomp$Treatment,erich.tab2.pcomp$Time.Point)
erich.tab2.pcomp$treat.time<-as.factor(erich.tab2.pcomp$treat.time)

#shannon
  aov.shannon.treat.time = aov(diversity_shannon ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.shannon.treat.time) # sig diff between treat.time
  TukeyHSD(aov.shannon.treat.time)
#observed  
  aov.observed.treat.time = aov(observed ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.observed.treat.time) # sig diff between treat.time
  TukeyHSD(aov.observed.treat.time)
#evenness
  aov.evenness.treat.time = aov(evenness_pielou ~ treat.time, data=erich.tab2.pcomp)
  summary(aov.evenness.treat.time) # sig diff between treat.time
  TukeyHSD(aov.evenness.treat.time)  
#Plot
  boxplot(diversity_shannon ~ treat.time, data=erich.tab2.pcomp, ylab="Shannon's diversity") 
  boxplot(observed ~ treat.time, data=erich.tab2.pcomp, ylab="Observed") 
  boxplot(evenness_pielou ~ treat.time, data=erich.tab2.pcomp, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.treat.time <- levels(erich.tab2.pcomp$treat.time) # get the variables

# make a pairwise list that we want to compare.
  al.treat.time.pairs <- combn(seq_along(al.treat.time), 2, simplify = FALSE, FUN = function(i)al.treat.time[i])
  print(al.treat.time.pairs)

#observed
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "observed",
   add = "boxplot", fill = "treat.time", palette = c(Pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
    
#Shannon
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "diversity_shannon",
   add = "boxplot", fill = "treat.time", palette = c(pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2.pcomp, x = "treat.time", y = "evenness_pielou",
   add = "boxplot", fill = "treat.time", palette = c(pcomp.Parent.colors)) 
  print(p1)
p1 <- p1 + stat_compare_means(comparisons = al.treat.time.pairs) #non-parametric test (Wilcoxon test)
  print(p1)















#old
#Shannon 
#Run the ANOVA and save it as an object
  aov.shannon.speciesPcomp = aov(Shannon ~ Treatment*Time.Point, data=erich.pcomp)
  summary(aov.shannon.speciesPcomp) # sig diff between species
  TukeyHSD(aov.shannon.speciesPcomp)
  
 
 

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pcomp, ylab="Shannon's diversity") 

#Pcomp  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPcomp = aov(Observed ~ Treatment*Time.Point, data=erich.pcomp)
  summary(aov.observed.speciesPcomp) # sig diff between species
  TukeyHSD(aov.observed.speciesPcomp)
  


  
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.pcomp, ylab="observed's diversity") 


  
``` 
# Pvar
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesPvar = aov(Shannon ~ Treatment*Time.Point, data=erich.pvar)
  summary(aov.shannon.speciesPvar) # sig diff between species
  TukeyHSD(aov.shannon.speciesPvar)

  

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pvar, ylab="Shannon's diversity") 
  
 
  
  
  
  
  
  

#Pvar  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPvar = aov(Observed ~ Treatment*Time.Point, data=erich.pvar)
  summary(aov.observed.speciesPvar) # sig diff between species
  TukeyHSD(aov.observed.speciesPvar)


  
#Plot
  boxplot(log(Observed) ~ Treatment*Time.Point, data=erich.pvar, ylab="observed's diversity") 
  
  
``` 

# Pacu
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesPacu = aov(Shannon ~ Treatment*Time.Point, data=erich.pacu)
  summary(aov.shannon.speciesPacu) # sig diff between species
  TukeyHSD(aov.shannon.speciesPacu)


#Plot
  aov.shannon.speciesPacu$Time.Point = factor(aov.shannon.speciesPacu$Time.Point, levels = c("T0", "T1", "TF"))

  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pacu, ylab="Shannon's diversity") 
  
 
  
  
  
  
  
  

#Pvar  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPacu = aov(Observed ~ Treatment*Time.Point, data=erich.pacu)
  summary(aov.observed.speciesPacu) # sig diff between species
  TukeyHSD(aov.observed.speciesPacu)


  
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.pacu, ylab="observed's diversity") 

# Just T1
  erich.pacuT1<-subset(erich.pacu, Time.Point=="T1")
  hist(log10(erich.pacuT1$Observed))
  aov.observed.speciesPacuT1 = aov(log10(Observed) ~ Treatment, data=erich.pacuT1)
  summary(aov.observed.speciesPacuT1) # sig diff between species
  TukeyHSD(aov.observed.speciesPacuT1)
  
  hist((erich.pacuT1$Shannon))
  aov.shannon.speciesPacuT1 = aov(log10(Shannon) ~ Treatment, data=erich.pacuT1)
  summary(aov.shannon.speciesPacuT1) # sig diff between species
  TukeyHSD(aov.shannon.speciesPacuT1)
  
  
``` 
 





 
```{r}
#for non normal Kruskal wallis rank sum  
  kruskal.test(x ~ AgeGroup, data=meta)

  
#testing
  
  coral <- get_variable(coralDIV2_rare.alpha2, "host_genus") %in% c("montipora",  "pocillopora")
sample_data(coralDIV2_rare.alpha2)$human <- factor(coral)

 #estimate_richness(coralDIV2_rare, "coral","host_genus", measures = c("Observed", "Shannon")) #select which estimates and estimate CANNOT GET TO WORK with X
  
## from tutorial can't get it to wokr in my code.   
  GP <- prune_taxa(taxa_sums(GlobalPatterns) > 0, GlobalPatterns)

human <- get_variable(GP, "SampleType") %in% c("Feces", "Mock", "Skin", "Tongue")
sample_data(GP)$human <- factor(human)

alpha_meas = c("Observed", "Shannon")
(p <- plot_richness(GP, "human", "SampleType", measures=alpha_meas))






##
## make bar plot of rank abundance of all DIVs
par(mar = c(10, 4, 4, 2) + 0.1) # make more room on bottom margin
N <- 50
barplot(sort(taxa_sums(coralDIV2_rare.alpha2), TRUE)[1:N]/nsamples(coralDIV2_rare.alpha2), las=2)

sample_variables(coralDIV2_rare.alpha2) #check sample variables

#plot_bar(coralDIV2_rare.alpha2, "Species", fill="DIV", facet_grid=~DIV)


#make network - not helpful for ITS2 here

ig <- make_network(coralDIV2_rare.alpha2, max.dist=0.3)
plot_network(ig, coralDIV2_rare.alpha2, color="Parent.ID", shape="Species", line_weight=0.4, label=NULL)






```






# for defense NMDS
```{r}
AA_16S<-Bac.s.rel.u.T1F #copy df

fullNMDS<-AA_16S
fullNMDS = prune_taxa(taxa_sums(fullNMDS) > 0, fullNMDS) #this removes any OTU with 0s


# extract OTU table with metadata. 

# Extract abundance matrix from the phyloseq object< 
#OK some of your samples now have zero taxa. 
OTU.16S.v = as(otu_table(fullNMDS), "matrix")
# transpose if necessary
if(taxa_are_rows(fullNMDS)){OTU.16S.v <- t(OTU.16S.v)}
# Coerce to data.frame
OTU.16S.v.df = as.data.frame(OTU.16S.v)
    sample_data<-rownames(OTU.16S.v.df)  #make row names a column
  rownames(OTU.16S.v.df) <- NULL
  OTU.16S.v.df2 <- cbind(sample_data,OTU.16S.v.df)



#smash in meta in vegan, multivariate stats
Asample.data16S <- as(sample_data(fullNMDS), "data.frame") #create sample data frame to view

sample_data<-rownames(Asample.data16S)  #get meta organized, row names a column
  rownames(Asample.data16S) <- NULL
  Asample.data16S2 <- cbind(sample_data,Asample.data16S)
  

B16S.data<-merge(Asample.data16S2,OTU.16S.v.df2) #merge on the metadata to OTU table. Bam

B16S.data<-subset(B16S.data, Time.Point=="T1"|Time.Point=="TF")
#CW code
# cleaned data for full dataframe NMDS (Site, Period, dom)

B16S.data$SpTrTi<-interaction(B16S.data$Species, B16S.data$Treatment, B16S.data$Time.Point)

 fac.B16S.data<-B16S.data[,c("Treatment","Species", "Time.Point", "SpTrTi")] #  sans responses

resp.B16S.data<-B16S.data[, 13:6461] # just responses

# resp.B16S.data <- scale(resp.B16S.data, center = TRUE, scale = TRUE) 
# 
# ##  Run MANOVA 
# # all.Manova<-adonis2(resp.B16S.data~Species*Treatment*Time.Point, data=B16S.data, permutations=1000,  method="euclidian")
# # all.Manova
# 

allNMDS<-metaMDS(resp.B16S.data, distance='bray', k=3, trymax=300)
# 
# stressplot(allNMDS)
# 
# ##### next  ####
# 
allNMDS.df<-data.frame(x=allNMDS$point[,1], y=allNMDS$point[,2],
                            Species=as.factor(fac.B16S.data[,2]),
                            Treatment=as.factor(fac.B16S.data[,1]),
                            Time.Point=as.factor(fac.B16S.data[,3]),
                            SpTrTi=as.factor(fac.B16S.data[,4]))

colnames(allNMDS.df)[1:2]<-c("MDS1", "MDS2")


# scores.ord.B16S_RelA_stats  df from above nmds you can't get it to work here. 

#centroid means scores.ord.B16S_RelA_stats
NMDS.mean=aggregate(allNMDS.df[,1:2],list(group=allNMDS.df$SpTrTi), mean) # mean by species*treatment*time

############### All groups NMDS
# B16S.data$Code<-droplevels(B16S.data$Code)
groups<-B16S.data$Code
groups2<-c("Montipora_capitata Ambient T1", 
           "Montipora_capitata Ambient TF", 
           "Montipora_capitata High T1",
           "Montipora_capitata High TF", 
           "Porites_compressa Ambient T1", 
           "Porites_compressa Ambient TF", 
           "Porites_compressa High T1",
           "Porites_compressa High TF",
           "Pavona_varians Ambient T1", 
           "Pavona_varians Ambient TF", 
           "Pavona_varians High T1",
           "Pavona_varians High TF",
           "Pocillopora_acuta Ambient T1", 
           "Pocillopora_acuta Ambient TF", 
           "Pocillopora_acutaa High T1",
           "Pocillopora_acuta High TF")
colors=c(
  "#F8766D", #Mcap 
  "#C77CFF", #Porites
  "#7CAE00", # Pvar
  "#00BFC4" # Pacu
   ) 
plot<-ordiplot(allNMDS, type="n", cex.main=1, display="sites", cex.lab=0.8, cex.axis=0.8) #ylim=c(-0.3, 0.3))
axis(side = 1, labels = FALSE)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
ordihull(plot, groups=B16S.data$SpTrTi, draw="polygon", alpha=20, col=c("#F8766D", "#7CAE00","#00BFC4","#C77CFF"), border=F)

# points for start and end
with(plot, points(NMDS.mean[5,2], NMDS.mean[5,3], bg="white", col=colors[1], pch=21, cex=2, lwd=1)) #mcap T1 H
with(plot, points(NMDS.mean[1,2], NMDS.mean[1,3], col=colors[1], pch=16, cex=2, lwd=1)) # Mcap A T1
with(plot, points(NMDS.mean[13,2], NMDS.mean[13,3], bg="white",col=colors[1], pch=24, cex=2, lwd=1)) #mcap TF H
with(plot, points(NMDS.mean[9,2], NMDS.mean[9,3], col=colors[1], pch=17, cex=2, lwd=1)) # Mcap A TF

with(plot, points(NMDS.mean[8,2], NMDS.mean[8,3], bg="white", col=colors[2], pch=21, cex=2, lwd=1)) # Pcomp H T1
with(plot, points(NMDS.mean[4,2], NMDS.mean[4,3], col=colors[2], pch=16, cex=2, lwd=1)) # Pcomp  A T1
with(plot, points(NMDS.mean[16,2], NMDS.mean[16,3], bg="white", col=colors[2], pch=24, cex=2, lwd=1)) # Pcomp H TF
with(plot, points(NMDS.mean[12,2], NMDS.mean[12,3], col=colors[2], pch=17, cex=2, lwd=1)) # PcompA TF

with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], bg="white", col=colors[3], pch=21, cex=2, lwd=1)) # Pvar H T1
with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[3], pch=16, cex=2, lwd=1)) # Pvar A T1
with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], bg="white", col=colors[3], pch=24, cex=2, lwd=1)) # Pvar H TF
with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=17, cex=2, lwd=1)) # Pvar A TF

with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], bg="white", col=colors[4], pch=21, cex=2, lwd=1)) # Pacu H T1
with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[4], pch=16, cex=2, lwd=1)) # Pac A Ta
with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], bg="white", col=colors[4], pch=24, cex=2, lwd=1)) # Pacu H TF
with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[4], pch=17, cex=2, lwd=1)) # Pac A TF



# # points for others (if want)
# with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D
# with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D

# # lines for start and end
# with(plot, lines(NMDS.mean[c(1,9),2], NMDS.mean[c(1,9),3], lwd=1.5, col =colors[1], lty=1))# Mcap A
# with(plot, lines(NMDS.mean[c(5,13),2], NMDS.mean[c(5,13),3], lwd=1.5, col =colors[1], lty=2))# Mcap H
# with(plot, lines(NMDS.mean[c(8,16),2], NMDS.mean[c(8,16),3], lwd=1.5, col =colors[2], lty=2))# Pcomp H
# with(plot, lines(NMDS.mean[c(4,12),2], NMDS.mean[c(4,12),3], lwd=1.5, col =colors[2], lty=1))# Pcomp A
# with(plot, lines(NMDS.mean[c(6, 14),2], NMDS.mean[c(6,14),3], lwd=1.5, col =colors[3], lty=2))# Pvar H
# with(plot, lines(NMDS.mean[c(2,10),2], NMDS.mean[c(2,10),3], lwd=1.5, col =colors[3], lty=1))# Pvar A
# with(plot, lines(NMDS.mean[c(7,15),2], NMDS.mean[c(7,15),3], lwd=1.5, col =colors[4], lty=2))# Pacu H
# with(plot, lines(NMDS.mean[c(3,11),2], NMDS.mean[c(3,11),3], lwd=1.5, col =colors[4], lty=1))# Pacu A

# add labels
# text(0, -4, "Montipora capitata", cex=0.8, col=colors[1])
# text(-5, -3, "Porites compressa", cex=0.8, col=colors[2])
# text(4, -3, "Pavona varians", cex=0.8, col=colors[3])
# text(4, 2, "Pocillopora acuta", cex=0.8, col=colors[4])

text(-3, -1.8, "2D Stress = 0.15", cex=0.8, col="black")
legend("topright", legend=groups2, inset=c(0.03, -0.01), x.intersp=0.6, y.intersp=0.9, cex=0.7, pch=16, col=colors, lty=c(1,3,1,3), seg.len=2, pt.cex=1, bty="n")

dev.copy(pdf, "ITSNMDS.pdf", height=5.5, width=6, useDingbats=FALSE)
dev.off() 

```


#try again with weighted unifrac _ You are stucks....
```{r}
Bac.s.rel.u.T1TF2<-subset_samples(Bac.s.rel.u, Time.Point=="T1"|Time.Point=="TF")
    Bac.s.rel.u.T1TF2 = prune_taxa(taxa_sums(Bac.s.rel.u.T1TF2) > 0, Bac.s.rel.u.T1TF2) #this removes any OTU with 0s

unifrac.all.T1TF <- ordinate(Bac.s.rel.u.T1TF2, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T1TF)
scores.ord.unifrac.all.T1TF<-as.data.frame(cbind(vegan::scores(unifrac.all.T1TF, display="sites"))) #  
     sd.unifrac.T1TF2 <- as(sample_data(Bac.s.rel.u.T1TF2), "data.frame") #sample df

scores.ord.unifrac.all.T1TF$Treatment <- sd.unifrac.T1TF2$Treatment
scores.ord.unifrac.all.T1TF$Species <- sd.unifrac.T1TF2$Species
scores.ord.unifrac.all.T1TF$Clade <- sd.unifrac.T1TF2$Clade
scores.ord.unifrac.all.T1TF$Time.Point <- sd.unifrac.T1TF2$Time.Point
scores.ord.unifrac.all.T1TF$Code <- paste(sd.unifrac.T1TF2$Species, sd.unifrac.T1TF2$Treatment,sd.unifrac.T1TF2$Time.Point)



OTU.16.v = as(otu_table(Bac.s.rel.u.T1TF2), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.rel.u.T1TF2)){OTU.16.v <- t(OTU.16.v)}
# Coerce to data.frame
OTU.16.v.df = as.data.frame(OTU.16.v)
    sample_data16<-rownames(OTU.16.v.df)  #make row names a column
  rownames(OTU.16.v.df) <- NULL
  OTU.16.v.df2 <- cbind(sample_data16,OTU.16.v.df)



#smash in meta in vegan, multivariate stats
Asample.data16 <- as(sample_data(Bac.s.rel.u.T1TF2), "data.frame") #create sample data frame to view

sample_data<-rownames(Asample.data16)  #get meta organized, row names a column
  rownames(Asample.data16) <- NULL
  Asample.data162 <- cbind(sample_data,Asample.data16)
  

B16S.data<-merge(Asample.data162,OTU.16.v.df2) #merge on the metadata to OTU table. Bam

B16S.data<-subset(B16S.data, Time.Point=="T1"|Time.Point=="TF")
#CW code
# cleaned data for full dataframe NMDS (Site, Period, dom)

B16S.data$SpTrTi<-interaction(B16S.data$Species, B16S.data$Treatment, B16S.data$Time.Point)




# ##### next  ####
# 
# allNMDS.df<-data.frame(x=allNMDS$point[,1], y=allNMDS$point[,2],
#                             Species=as.factor(fac.B16S.data[,2]),
#                             Treatment=as.factor(fac.B16S.data[,1]),
#                             Time.Point=as.factor(fac.B16S.data[,3]),
#                             SpTrTi=as.factor(fac.B16S.data[,4]))

# colnames(allNMDS.df)[1:2]<-c("MDS1", "MDS2")


# scores.ord.B16S_RelA_stats  df from above nmds you can't get it to work here. 
#centroid means scores.ord.B16S_RelA_stats
NMDS.mean=aggregate(scores.ord.unifrac.all.T1TF[,1:2],list(group=B16S.data$Code), mean) # mean by species*treatment*time

############### All groups NMDS
# B16S.data$Code<-droplevels(B16S.data$Code)
groups<-B16S.data$Code
groups2<-c("Montipora_capitata Ambient T1", 
           "Montipora_capitata Ambient TF", 
           "Montipora_capitata High T1",
           "Montipora_capitata High TF", 
           "Porites_compressa Ambient T1", 
           "Porites_compressa Ambient TF", 
           "Porites_compressa High T1",
           "Porites_compressa High TF",
           "Pavona_varians Ambient T1", 
           "Pavona_varians Ambient TF", 
           "Pavona_varians High T1",
           "Pavona_varians High TF",
           "Pocillopora_acuta Ambient T1", 
           "Pocillopora_acuta Ambient TF", 
           "Pocillopora_acutaa High T1",
           "Pocillopora_acuta High TF")
colors=c(
  "#F8766D", #Mcap 
  "#C77CFF", #Porites
  "#7CAE00", # Pvar
  "#00BFC4" # Pacu
   ) 
plot<-ordiplot(unifrac.all.T1TF, type="n", cex.main=1, display="sites", cex.lab=0.8, cex.axis=0.8) #ylim=c(-0.3, 0.3))
axis(side = 1, labels = FALSE)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
ordihull(plot, groups=unifrac.all.T1TF$Code, draw="polygon", alpha=20, col=c("#F8766D", "#7CAE00","#00BFC4","#C77CFF"), border=F)

# points for start and end
with(plot, points(NMDS.mean[5,2], NMDS.mean[5,3], bg="white", col=colors[1], pch=21, cex=2, lwd=1)) #mcap T1 H
with(plot, points(NMDS.mean[1,2], NMDS.mean[1,3], col=colors[1], pch=16, cex=2, lwd=1)) # Mcap A T1
with(plot, points(NMDS.mean[13,2], NMDS.mean[13,3], bg="white",col=colors[1], pch=24, cex=2, lwd=1)) #mcap TF H
with(plot, points(NMDS.mean[9,2], NMDS.mean[9,3], col=colors[1], pch=17, cex=2, lwd=1)) # Mcap A TF

with(plot, points(NMDS.mean[8,2], NMDS.mean[8,3], bg="white", col=colors[2], pch=21, cex=2, lwd=1)) # Pcomp H T1
with(plot, points(NMDS.mean[4,2], NMDS.mean[4,3], col=colors[2], pch=16, cex=2, lwd=1)) # Pcomp  A T1
with(plot, points(NMDS.mean[16,2], NMDS.mean[16,3], bg="white", col=colors[2], pch=24, cex=2, lwd=1)) # Pcomp H TF
with(plot, points(NMDS.mean[12,2], NMDS.mean[12,3], col=colors[2], pch=17, cex=2, lwd=1)) # PcompA TF

with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], bg="white", col=colors[3], pch=21, cex=2, lwd=1)) # Pvar H T1
with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[3], pch=16, cex=2, lwd=1)) # Pvar A T1
with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], bg="white", col=colors[3], pch=24, cex=2, lwd=1)) # Pvar H TF
with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=17, cex=2, lwd=1)) # Pvar A TF

with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], bg="white", col=colors[4], pch=21, cex=2, lwd=1)) # Pacu H T1
with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[4], pch=16, cex=2, lwd=1)) # Pac A Ta
with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], bg="white", col=colors[4], pch=24, cex=2, lwd=1)) # Pacu H TF
with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[4], pch=17, cex=2, lwd=1)) # Pac A TF



# # points for others (if want)
# with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D
# with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D

# lines for start and end
with(plot, lines(NMDS.mean[c(1,9),2], NMDS.mean[c(1,9),3], lwd=1.5, col =colors[1], lty=1))# Mcap A
with(plot, lines(NMDS.mean[c(5,13),2], NMDS.mean[c(5,13),3], lwd=1.5, col =colors[1], lty=2))# Mcap H
with(plot, lines(NMDS.mean[c(8,16),2], NMDS.mean[c(8,16),3], lwd=1.5, col =colors[2], lty=2))# Pcomp H
with(plot, lines(NMDS.mean[c(4,12),2], NMDS.mean[c(4,12),3], lwd=1.5, col =colors[2], lty=1))# Pcomp A
with(plot, lines(NMDS.mean[c(6, 14),2], NMDS.mean[c(6,14),3], lwd=1.5, col =colors[3], lty=2))# Pvar H
with(plot, lines(NMDS.mean[c(2,10),2], NMDS.mean[c(2,10),3], lwd=1.5, col =colors[3], lty=1))# Pvar A
with(plot, lines(NMDS.mean[c(7,15),2], NMDS.mean[c(7,15),3], lwd=1.5, col =colors[4], lty=2))# Pacu H
with(plot, lines(NMDS.mean[c(3,11),2], NMDS.mean[c(3,11),3], lwd=1.5, col =colors[4], lty=1))# Pacu A

# add labels
# text(0, -4, "Montipora capitata", cex=0.8, col=colors[1])
# text(-5, -3, "Porites compressa", cex=0.8, col=colors[2])
# text(4, -3, "Pavona varians", cex=0.8, col=colors[3])
# text(4, 2, "Pocillopora acuta", cex=0.8, col=colors[4])

text(-30, -22, "2D Stress = 0.15", cex=0.8, col="black")
legend("topright", legend=groups2, inset=c(0.03, -0.01), x.intersp=0.6, y.intersp=0.9, cex=0.7, pch=16, col=colors, lty=c(1,3,1,3), seg.len=2, pt.cex=1, bty="n")

dev.copy(pdf, "ITSNMDS.pdf", height=5.5, width=6, useDingbats=FALSE)
dev.off() 

```

```{r}
Bac.s.rel.u.T0<-subset_samples(Bac.s.rel.u, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s

unifrac.all.T0 <- ordinate(Bac.s.rel.u.T0, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.all.T0)
scores.ord.unifrac.all.T0<-as.data.frame(cbind(vegan::scores(unifrac.all.T0, display="sites"))) #  
     sd.unifrac.T0 <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df

scores.ord.unifrac.all.T0$Treatment <- sd.unifrac.T0$Treatment
scores.ord.unifrac.all.T0$Species <- sd.unifrac.T0$Species
scores.ord.unifrac.all.T0$Clade <- sd.unifrac.T0$Clade
scores.ord.unifrac.all.T0$Time.Point <- sd.unifrac.T0$Time.Point
scores.ord.unifrac.all.T0$Code <- paste(sd.unifrac.T0$Species, sd.unifrac.T0$Treatment,sd.unifrac.T0$Time.Point)



OTU.16.v = as(otu_table(Bac.s.rel.u.T0), "matrix")
# transpose if necessary
if(taxa_are_rows(Bac.s.rel.u.T0)){OTU.16.v <- t(OTU.16.v)}
# Coerce to data.frame
OTU.16.v.df = as.data.frame(OTU.16.v)
    sample_data16<-rownames(OTU.16.v.df)  #make row names a column
  rownames(OTU.16.v.df) <- NULL
  OTU.16.v.df2 <- cbind(sample_data16,OTU.16.v.df)



#smash in meta in vegan, multivariate stats
Asample.data16 <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #create sample data frame to view

sample_data<-rownames(Asample.data16)  #get meta organized, row names a column
  rownames(Asample.data16) <- NULL
  Asample.data162 <- cbind(sample_data,Asample.data16)
  

B16S.data<-merge(Asample.data162,OTU.16.v.df2) #merge on the metadata to OTU table. Bam

B16S.data<-subset(B16S.data, Time.Point=="T1"|Time.Point=="TF")
#CW code
# cleaned data for full dataframe NMDS (Site, Period, dom)

B16S.data$SpTrTi<-interaction(B16S.data$Species, B16S.data$Treatment, B16S.data$Time.Point)




# ##### next  ####
# 
# allNMDS.df<-data.frame(x=allNMDS$point[,1], y=allNMDS$point[,2],
#                             Species=as.factor(fac.B16S.data[,2]),
#                             Treatment=as.factor(fac.B16S.data[,1]),
#                             Time.Point=as.factor(fac.B16S.data[,3]),
#                             SpTrTi=as.factor(fac.B16S.data[,4]))

# colnames(allNMDS.df)[1:2]<-c("MDS1", "MDS2")


# scores.ord.B16S_RelA_stats  df from above nmds you can't get it to work here. 
#centroid means scores.ord.B16S_RelA_stats
NMDS.mean=aggregate(scores.ord.unifrac.all.T0[,1:2],list(group=B16S.data$Code), mean) # mean by species*treatment*time

############### All groups NMDS
# B16S.data$Code<-droplevels(B16S.data$Code)
groups<-B16S.data$Code
groups2<-c("Montipora_capitata Ambient T1", 
           "Montipora_capitata Ambient TF", 
           "Montipora_capitata High T1",
           "Montipora_capitata High TF", 
           "Porites_compressa Ambient T1", 
           "Porites_compressa Ambient TF", 
           "Porites_compressa High T1",
           "Porites_compressa High TF",
           "Pavona_varians Ambient T1", 
           "Pavona_varians Ambient TF", 
           "Pavona_varians High T1",
           "Pavona_varians High TF",
           "Pocillopora_acuta Ambient T1", 
           "Pocillopora_acuta Ambient TF", 
           "Pocillopora_acutaa High T1",
           "Pocillopora_acuta High TF")
colors=c(
  "#F8766D", #Mcap 
  "#C77CFF", #Porites
  "#7CAE00", # Pvar
  "#00BFC4" # Pacu
   ) 
plot<-ordiplot(unifrac.all.T0, type="n", cex.main=1, display="sites", cex.lab=0.8, cex.axis=0.8) #ylim=c(-0.3, 0.3))
axis(side = 1, labels = FALSE)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
ordihull(plot, groups=unifrac.all.T0$Code, draw="polygon", alpha=20, col=c("#F8766D", "#7CAE00","#00BFC4","#C77CFF"), border=F)

# points for start and end
with(plot, points(NMDS.mean[5,2], NMDS.mean[5,3], bg="white", col=colors[1], pch=21, cex=2, lwd=1)) #mcap T1 H
with(plot, points(NMDS.mean[1,2], NMDS.mean[1,3], col=colors[1], pch=16, cex=2, lwd=1)) # Mcap A T1
with(plot, points(NMDS.mean[13,2], NMDS.mean[13,3], bg="white",col=colors[1], pch=24, cex=2, lwd=1)) #mcap TF H
with(plot, points(NMDS.mean[9,2], NMDS.mean[9,3], col=colors[1], pch=17, cex=2, lwd=1)) # Mcap A TF

with(plot, points(NMDS.mean[8,2], NMDS.mean[8,3], bg="white", col=colors[2], pch=21, cex=2, lwd=1)) # Pcomp H T1
with(plot, points(NMDS.mean[4,2], NMDS.mean[4,3], col=colors[2], pch=16, cex=2, lwd=1)) # Pcomp  A T1
with(plot, points(NMDS.mean[16,2], NMDS.mean[16,3], bg="white", col=colors[2], pch=24, cex=2, lwd=1)) # Pcomp H TF
with(plot, points(NMDS.mean[12,2], NMDS.mean[12,3], col=colors[2], pch=17, cex=2, lwd=1)) # PcompA TF

with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], bg="white", col=colors[3], pch=21, cex=2, lwd=1)) # Pvar H T1
with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[3], pch=16, cex=2, lwd=1)) # Pvar A T1
with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], bg="white", col=colors[3], pch=24, cex=2, lwd=1)) # Pvar H TF
with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=17, cex=2, lwd=1)) # Pvar A TF

with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], bg="white", col=colors[4], pch=21, cex=2, lwd=1)) # Pacu H T1
with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[4], pch=16, cex=2, lwd=1)) # Pac A Ta
with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], bg="white", col=colors[4], pch=24, cex=2, lwd=1)) # Pacu H TF
with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[4], pch=17, cex=2, lwd=1)) # Pac A TF



# # points for others (if want)
# with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D
# with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D

# lines for start and end
with(plot, lines(NMDS.mean[c(1,9),2], NMDS.mean[c(1,9),3], lwd=1.5, col =colors[1], lty=1))# Mcap A
with(plot, lines(NMDS.mean[c(5,13),2], NMDS.mean[c(5,13),3], lwd=1.5, col =colors[1], lty=2))# Mcap H
with(plot, lines(NMDS.mean[c(8,16),2], NMDS.mean[c(8,16),3], lwd=1.5, col =colors[2], lty=2))# Pcomp H
with(plot, lines(NMDS.mean[c(4,12),2], NMDS.mean[c(4,12),3], lwd=1.5, col =colors[2], lty=1))# Pcomp A
with(plot, lines(NMDS.mean[c(6, 14),2], NMDS.mean[c(6,14),3], lwd=1.5, col =colors[3], lty=2))# Pvar H
with(plot, lines(NMDS.mean[c(2,10),2], NMDS.mean[c(2,10),3], lwd=1.5, col =colors[3], lty=1))# Pvar A
with(plot, lines(NMDS.mean[c(7,15),2], NMDS.mean[c(7,15),3], lwd=1.5, col =colors[4], lty=2))# Pacu H
with(plot, lines(NMDS.mean[c(3,11),2], NMDS.mean[c(3,11),3], lwd=1.5, col =colors[4], lty=1))# Pacu A

# add labels
# text(0, -4, "Montipora capitata", cex=0.8, col=colors[1])
# text(-5, -3, "Porites compressa", cex=0.8, col=colors[2])
# text(4, -3, "Pavona varians", cex=0.8, col=colors[3])
# text(4, 2, "Pocillopora acuta", cex=0.8, col=colors[4])

text(-30, -22, "2D Stress = 0.15", cex=0.8, col="black")
legend("topright", legend=groups2, inset=c(0.03, -0.01), x.intersp=0.6, y.intersp=0.9, cex=0.7, pch=16, col=colors, lty=c(1,3,1,3), seg.len=2, pt.cex=1, bty="n")

dev.copy(pdf, "ITSNMDS.pdf", height=5.5, width=6, useDingbats=FALSE)
dev.off() 

```


T0 bray bc it works
```{r}
AA_16ST0<-Bac.s.rel.u.T0 #copy df

fullNMDS<-AA_16ST0
fullNMDS = prune_taxa(taxa_sums(fullNMDS) > 0, fullNMDS) #this removes any OTU with 0s


# extract OTU table with metadata. 

# Extract abundance matrix from the phyloseq object< 
#OK some of your samples now have zero taxa. 
OTU.16S.v = as(otu_table(fullNMDS), "matrix")
# transpose if necessary
if(taxa_are_rows(fullNMDS)){OTU.16S.v <- t(OTU.16S.v)}
# Coerce to data.frame
OTU.16S.v.df = as.data.frame(OTU.16S.v)
    sample_data<-rownames(OTU.16S.v.df)  #make row names a column
  rownames(OTU.16S.v.df) <- NULL
  OTU.16S.v.df2 <- cbind(sample_data,OTU.16S.v.df)



#smash in meta in vegan, multivariate stats
Asample.data16S <- as(sample_data(fullNMDS), "data.frame") #create sample data frame to view

sample_data<-rownames(Asample.data16S)  #get meta organized, row names a column
  rownames(Asample.data16S) <- NULL
  Asample.data16S2 <- cbind(sample_data,Asample.data16S)
  

B16S.data<-merge(Asample.data16S2,OTU.16S.v.df2) #merge on the metadata to OTU table. Bam

B16S.data<-subset(B16S.data, Time.Point=="T0")
#CW code
# cleaned data for full dataframe NMDS (Site, Period, dom)


 fac.B16S.data<-B16S.data[,c("Species","Clade")] #  sans responses

resp.B16S.data<-B16S.data[, 13:2449] # just responses

# resp.B16S.data <- scale(resp.B16S.data, center = TRUE, scale = TRUE) 
# 
# ##  Run MANOVA 
# # all.Manova<-adonis2(resp.B16S.data~Species*Treatment*Time.Point, data=B16S.data, permutations=1000,  method="euclidian")
# # all.Manova
# 
 
allNMDS<-metaMDS(resp.B16S.data, distance='bray', k=3, trymax=300)
# 
# stressplot(allNMDS)
# 
# ##### next  #### 
# 
allNMDS.df<-data.frame(x=allNMDS$point[,1], y=allNMDS$point[,2],
                            Species=as.factor(fac.B16S.data[,2]))

colnames(allNMDS.df)[1:2]<-c("MDS1", "MDS2")


# scores.ord.B16S_RelA_stats  df from above nmds you can't get it to work here. 

#centroid means scores.ord.B16S_RelA_stats
NMDS.mean=aggregate(allNMDS.df[,1:2],list(group=allNMDS.df$Species), mean) # mean by species*treatment*time

############### All groups NMDS
# B16S.data$Code<-droplevels(B16S.data$Code)
groups<-B16S.data$Species
groups2<-c("Montipora_capitata",
           "Porites_compressa",
           "Pavona_varians", 
           "Pocillopora_acuta")
colors=c(
  "#F8766D", #Mcap 
  "#C77CFF", #Porites
  "#7CAE00", # Pvar
  "#00BFC4" # Pacu
   ) 
plot<-ordiplot(allNMDS, type="n", cex.main=1, display="sites", cex.lab=0.8, cex.axis=0.8) #ylim=c(-0.3, 0.3))
axis(side = 1, labels = FALSE)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
ordihull(plot, groups=B16S.data$Species, draw="polygon", alpha=40, col=c("#F8766D", "#7CAE00","#00BFC4","#C77CFF"), border=F)

# points for start and end
with(plot, points(NMDS.mean[1,2], NMDS.mean[1,3],  col=colors[1], pch=18, cex=6, lwd=1)) #mcap C
with(plot, points(NMDS.mean[8,2], NMDS.mean[8,3], col=colors[1], pch=23, cex=5, lwd=2)) # McapCD
with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[3], pch=18, cex=6, lwd=1)) #pvar c1
with(plot, points(NMDS.mean[5,2], NMDS.mean[5,3], col=colors[3], pch=23, cex=5, lwd=2)) # pvar c27

with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3],  col=colors[2], pch=18, cex=6, lwd=1)) # pcomp cj
with(plot, points(NMDS.mean[4,2], NMDS.mean[4,3], col=colors[2], pch=23, cex=5, lwd=2)) # Pcomp  n
with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3],  col=colors[4], pch=18, cex=6, lwd=1)) # pacu c42
with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[4], pch=23, cex=5, lwd=2)) # Pacu ccc




# # points for others (if want)
# with(plot, points(NMDS.mean[2,2], NMDS.mean[2,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[3,2], NMDS.mean[3,3], col=colors[1], pch=16, cex=1, lwd=1)) # Lil C
# with(plot, points(NMDS.mean[6,2], NMDS.mean[6,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[7,2], NMDS.mean[7,3], col=colors[2], pch=16, cex=1, lwd=1)) # R14 C
# with(plot, points(NMDS.mean[10,2], NMDS.mean[10,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[11,2], NMDS.mean[11,3], col=colors[3], pch=16, cex=1, lwd=1)) # Lil D
# with(plot, points(NMDS.mean[14,2], NMDS.mean[14,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D
# with(plot, points(NMDS.mean[15,2], NMDS.mean[15,3], col=colors[4], pch=16, cex=1, lwd=1)) # R14 D

# # lines for start and end
# with(plot, lines(NMDS.mean[c(1,9),2], NMDS.mean[c(1,9),3], lwd=1.5, col =colors[1], lty=1))# Mcap A
# with(plot, lines(NMDS.mean[c(5,13),2], NMDS.mean[c(5,13),3], lwd=1.5, col =colors[1], lty=2))# Mcap H
# with(plot, lines(NMDS.mean[c(8,16),2], NMDS.mean[c(8,16),3], lwd=1.5, col =colors[2], lty=2))# Pcomp H
# with(plot, lines(NMDS.mean[c(4,12),2], NMDS.mean[c(4,12),3], lwd=1.5, col =colors[2], lty=1))# Pcomp A
# with(plot, lines(NMDS.mean[c(6, 14),2], NMDS.mean[c(6,14),3], lwd=1.5, col =colors[3], lty=2))# Pvar H
# with(plot, lines(NMDS.mean[c(2,10),2], NMDS.mean[c(2,10),3], lwd=1.5, col =colors[3], lty=1))# Pvar A
# with(plot, lines(NMDS.mean[c(7,15),2], NMDS.mean[c(7,15),3], lwd=1.5, col =colors[4], lty=2))# Pacu H
# with(plot, lines(NMDS.mean[c(3,11),2], NMDS.mean[c(3,11),3], lwd=1.5, col =colors[4], lty=1))# Pacu A

# add labels
# text(0, -4, "Montipora capitata", cex=0.8, col=colors[1])
# text(-5, -3, "Porites compressa", cex=0.8, col=colors[2])
# text(4, -3, "Pavona varians", cex=0.8, col=colors[3])
# text(4, 2, "Pocillopora acuta", cex=0.8, col=colors[4])

text(-3, -1.8, "2D Stress = 0.15", cex=0.8, col="black")
legend("topright", legend=groups2, inset=c(0.03, -0.01), x.intersp=0.6, y.intersp=0.9, cex=0.7, pch=16, col=colors, lty=c(1,3,1,3), seg.len=2, pt.cex=1, bty="n")

dev.copy(pdf, "ITSNMDS.pdf", height=5.5, width=6, useDingbats=FALSE)
dev.off() 

```
 
#Mantel Test
import the its2 matrix from the ITS script

T0 only
```{r}
# figure out which samples are represented in the its2 and 16s dfs

# T0 only  #####

#set up 16s
    Bac.s.rel.u.T0<-subset_samples(Bac.s.rel.u, Time.Point=="T0")
    Bac.s.rel.u.T0 = prune_taxa(taxa_sums(Bac.s.rel.u.T0) > 0, Bac.s.rel.u.T0) #this removes any OTU with 0s
  # look at sample names
    bac.T0.sd <- as(sample_data(Bac.s.rel.u.T0), "data.frame") #sample df
    #save just sample names
      bac.T0.sd.samps<-bac.T0.sd[['sample_name.1']]

#set up ITS2 T0, import  (already pruned)
    ITS2_T0_mantel <- readRDS("ITS2_T0_phyloseq.rds")
  # look at sample names
    ITS.T0.sd <- as(sample_data(ITS2_T0_mantel), "data.frame") #sample df
  #save just sample names
      ITS.T0.sd.samps<-ITS.T0.sd[['ID']]

# compare ITS to 16s at T0
  #identify missing samples btw datasets     
      setdiff(bac.T0.sd.samps,ITS.T0.sd.samps) #samples in 16s but not ITS: S208b, SN70b
      setdiff(ITS.T0.sd.samps,bac.T0.sd.samps) #samples in ITS but not 16s: "S208"  "SN60"  "SN70"  "S49"   "SN11"  "SN145" "SN40"  "SN111" "SN6"   "SN56"  "SN109"
        
      # in 16S change sample S208b name to S208

  # 16s changes: rename sample S208b = S208

    # pull OTU table
     OTU.16sT0 = as(otu_table(Bac.s.rel.u.T0), "matrix") #pull otu
       OTU.16sT0<-as.data.frame(OTU.16sT0) #make into df
        row.names(OTU.16sT0)[5] <- "S208"   #change row name by row number
                row.names(OTU.16sT0)[31] <- "SN70"   #change row name by row number

    #pull TAX table
     aTax.16sT0 = as(tax_table(Bac.s.rel.u.T0), "matrix") #pull tax
     
     #pull phy tree table
     tree.seq = phy_tree(Bac.s.rel.u.T0) #pull out tree

    #pull SAMPLE DATA
     aSD.16sT0 = as(sample_data(Bac.s.rel.u.T0), "matrix") #pull samp.data
     aSD.16sT0<-as.data.frame(aSD.16sT0) #change to df
        row.names(aSD.16sT0)[5] <- "S208"  #replace name S208b
        row.names(aSD.16sT0)[31] <- "SN70"  #replace name S208b
            

    #remake the phyloseq obj
      samdata = sample_data(aSD.16sT0)
        seqtab = otu_table(OTU.16sT0, taxa_are_rows = FALSE)
        taxtab = tax_table(aTax.16sT0)
        phy_tree = read_tree(phyTree.16sT0)
          Bac_T0_mantel = phyloseq(otu_table(seqtab), tax_table(taxtab), sample_data(samdata),phy_tree(tree.seq))
          
          
   #ITS2 - remove samps not in the 16s df - "SN60"  "S49"   "SN11"  "SN145" "SN40"  "SN111" "SN6"   "SN56"  "SN109" 
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN60" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "S49" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN11" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN145" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN40" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN111" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN6" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN56" )
        ITS2_T0_mantel <- subset_samples(ITS2_T0_mantel, ID != "SN109" )

        # both have 39 samples - Bac_T0_mantel ITS2_T0_mantel
      
        # reorder ps object so samples are in the same order
        library(microbiome)
        library(microViz)
        
        Bac_T0_mantel <- Bac_T0_mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac_T0_mantel) %>% head(8)
        
        ITS2_T0_mantel <- ITS2_T0_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS2_T0_mantel) %>% head(8)
        
  # MAKE data matrix
       library(vegan)
  
  #16s T0        
        unifrac.Bac.T0.mantel <- phyloseq::distance(Bac_T0_mantel, method = "wunifrac")
        
  #ITS2 T0
     bray.ITS2.T0.mantel <- phyloseq::distance(ITS2_T0_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T0.mantel, bray.ITS2.T0.mantel,  permutations=1000)
    
        
#### T0 Mcap only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for mcap
    Bac_T0_mantel.Mcap <- subset_samples(Bac_T0_mantel, Species == "Montipora_capitata")
    Bac_T0_mantel.Mcap = prune_taxa(taxa_sums(Bac_T0_mantel.Mcap) > 0, Bac_T0_mantel.Mcap) 
    
    ITS2_T0_mantel.Mcap  <- subset_samples(ITS2_T0_mantel, Species == "Montipora_capitata")
    ITS2_T0_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T0_mantel.Mcap) > 0, ITS2_T0_mantel.Mcap)
        
    #16s T0 mcap dist        
      unifrac.Bac_T0_mantel.Mcap <- phyloseq::distance(Bac_T0_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T0 mcap dist
     bray.ITS2.T0.mantel.Mcap <- phyloseq::distance(ITS2_T0_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Mcap, bray.ITS2.T0.mantel.Mcap,  permutations=1000)
    
#### T0 Pcomp only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pcomp
    Bac_T0_mantel.Pcomp <- subset_samples(Bac_T0_mantel, Species == "Porites_compressa")
    Bac_T0_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T0_mantel.Pcomp) > 0, Bac_T0_mantel.Pcomp) 
    
    ITS2_T0_mantel.Pcomp  <- subset_samples(ITS2_T0_mantel, Species == "Porites_compressa")
    ITS2_T0_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T0_mantel.Pcomp) > 0, ITS2_T0_mantel.Pcomp)
        
    #16s T0 Pcomp dist        
      unifrac.Bac_T0_mantel.Pcomp <- phyloseq::distance(Bac_T0_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T0 Pcomp dist
     bray.ITS2.T0.mantel.Pcomp <- phyloseq::distance(ITS2_T0_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pcomp, bray.ITS2.T0.mantel.Pcomp,  permutations=1000)        
        
        
  #### T0 Pvar only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pvar
    Bac_T0_mantel.Pvar <- subset_samples(Bac_T0_mantel, Species == "Pavona_varians")
    Bac_T0_mantel.Pvar = prune_taxa(taxa_sums(Bac_T0_mantel.Pvar) > 0, Bac_T0_mantel.Pvar) 
    
    ITS2_T0_mantel.Pvar  <- subset_samples(ITS2_T0_mantel, Species == "Pavona_varians")
    ITS2_T0_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T0_mantel.Pvar) > 0, ITS2_T0_mantel.Pvar)
        
    #16s T0 Pvar dist        
      unifrac.Bac_T0_mantel.Pvar <- phyloseq::distance(Bac_T0_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T0 Pvar dist
     bray.ITS2.T0.mantel.Pvar <- phyloseq::distance(ITS2_T0_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pvar, bray.ITS2.T0.mantel.Pvar,  permutations=1000)            
        
        
     #### T0 Pacu only ####        
 # Bac_T0_mantel ITS2_T0_mantel
    
    #subset df for Pacu
    Bac_T0_mantel.Pacu <- subset_samples(Bac_T0_mantel, Species == "Pavona_varians")
    Bac_T0_mantel.Pacu = prune_taxa(taxa_sums(Bac_T0_mantel.Pacu) > 0, Bac_T0_mantel.Pacu) 
    
    ITS2_T0_mantel.Pacu  <- subset_samples(ITS2_T0_mantel, Species == "Pavona_varians")
    ITS2_T0_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T0_mantel.Pacu) > 0, ITS2_T0_mantel.Pacu)
        
    #16s T0 Pacu dist        
      unifrac.Bac_T0_mantel.Pacu <- phyloseq::distance(Bac_T0_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T0 Pacu dist
     bray.ITS2.T0.mantel.Pacu <- phyloseq::distance(ITS2_T0_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T0_mantel.Pacu, bray.ITS2.T0.mantel.Pacu,  permutations=1000)            
        
        
    


```
T1 High only
```{r}
# figure out which samples are represented in the its2 and 16s dfs

# T1 only  #####

#high only 

#set up 16s
    Bac.s.rel.u.T1<-subset_samples(Bac.s.rel.u, Time.Point=="T1")
    Bac.s.rel.u.T1.High<-subset_samples(Bac.s.rel.u.T1, Treatment=="High")

    Bac.s.rel.u.T1.High = prune_taxa(taxa_sums(Bac.s.rel.u.T1.High) > 0, Bac.s.rel.u.T1.High) #this removes any OTU with 0s
  # look at sample names
    bac.T1.sd <- as(sample_data(Bac.s.rel.u.T1.High), "data.frame") #sample df
    #save just sample names
      bac.T1.High.sd.samps<-bac.T1.sd[['sample_name.1']]

#set up ITS2 T1, import  T1TF
    ITS2_T1TF_mantel <- readRDS("ITS2_T1TF_phyloseq.rds") 
     ITS2_T1_mantel<-subset_samples(ITS2_T1TF_mantel, Time.Point=="T1")
    ITS2_T1_mantel.High<-subset_samples(ITS2_T1_mantel, Treatment=="High")
      ITS2_T1_mantel.High = prune_taxa(taxa_sums(ITS2_T1_mantel.High) > 0, ITS2_T1_mantel.High) #this removes any OTU with 0s

    
  # look at sample names
    ITS.T1.High.sd <- as(sample_data(ITS2_T1_mantel.High), "data.frame") #sample df
  #save just sample names
      ITS.T1.High.sd.samps<-ITS.T1.High.sd[['ID']]

# compare ITS to 16s at T1
  #identify missing samples btw datasets     
      setdiff(bac.T1.High.sd.samps,ITS.T1.High.sd.samps) #samples in 16s but not ITS: "S1857" "S3469" "SN134"
      setdiff(ITS.T1.High.sd.samps,bac.T1.High.sd.samps) #samples in ITS but not 16s: "S1857b" "S3468"  "SN134b" "S670"   "S3025"  "S1388a"
        
      # in 16S remove S3469
      #in ITS change S1857b to S1857 and SN134b to SN134. Remove S3468, S670, S3025, S1388a

  # 16s changes: rename sample S3469
      
    Bac.T1.High.mantel <- subset_samples(Bac.s.rel.u.T1.High, sample_name.1 != "S3469" )
        #40 samples
          
   #ITS2 - remove samps not in the 16s df - " S3468, S670, S3025, S1388a
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S3468" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S670" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S3025" )
        ITS2_T1_mantel.High <- subset_samples(ITS2_T1_mantel.High, ID != "S1388a" )

        
    #change names ITS2 S1857b to S1857 and SN134b to SN134

    # pull OTU table
     OTU.ITS2T1.High = as(otu_table(ITS2_T1_mantel.High), "matrix") #pull otu
       OTU.ITS2T1.High<-as.data.frame(OTU.ITS2T1.High) #make into df
        row.names(OTU.ITS2T1.High)[7] <- "S1857"   #change row name by row number
                row.names(OTU.ITS2T1.High)[17] <- "SN134"   #change row name by row number

    #pull TAX table
     aTax.ITS2T1.High = as(tax_table(ITS2_T1_mantel.High), "matrix") #pull tax

    #pull SAMPLE DATA
     aSD.ITS2T1.High = as(sample_data(ITS2_T1_mantel.High), "matrix") #pull samp.data
     aSD.ITS2T1.High<-as.data.frame(aSD.ITS2T1.High) #change to df
        row.names(aSD.ITS2T1.High)[7] <- "S1857"  #replace name S208b
        row.names(aSD.ITS2T1.High)[17] <- "SN134"  #replace name S208b
            

    #remake the phyloseq obj
      samdata.High = sample_data(aSD.ITS2T1.High)
        seqtab.High = otu_table(OTU.ITS2T1.High, taxa_are_rows = FALSE)
        taxtab.High = tax_table(aTax.ITS2T1.High)
          ITS_T1.High_mantel = phyloseq(otu_table(seqtab.High), tax_table(taxtab.High), sample_data(samdata.High))
             
        # both have 40 samples - 
      
        # reorder ps object so samples are in the same order
        
        Bac.T1.High.mantel <- Bac.T1.High.mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac.T1.High.mantel) %>% head(8)
        
        ITS_T1.High_mantel <- ITS_T1.High_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS_T1.High_mantel) %>% head(8)
        
  # MAKE data matrix
  
  #16s T1 High        
        unifrac.Bac.T1.High.mantel <- phyloseq::distance(Bac.T1.High.mantel, method = "wunifrac")
        
  #ITS2 T1 High
     bray.ITS2.T1.High.mantel <- phyloseq::distance(ITS_T1.High_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T1.High.mantel, bray.ITS2.T1.High.mantel,  permutations=1000)
    
        
#### T1.High Mcap only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for mcap
    Bac_T1.High_mantel.Mcap <- subset_samples(Bac.T1.High.mantel, Species == "Montipora_capitata")
    Bac_T1.High_mantel.Mcap = prune_taxa(taxa_sums(Bac_T1.High_mantel.Mcap) > 0, Bac_T1.High_mantel.Mcap) 
    
    ITS2_T1.High_mantel.Mcap  <- subset_samples(ITS_T1.High_mantel, Species == "Montipora_capitata")
    ITS2_T1.High_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Mcap) > 0, ITS2_T1.High_mantel.Mcap)
        
    #16s T1.High mcap dist        
      unifrac.Bac_T1.High_mantel.Mcap <- phyloseq::distance(Bac_T1.High_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T1.High mcap dist
     bray.ITS2.T1.High.mantel.Mcap <- phyloseq::distance(ITS2_T1.High_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Mcap, bray.ITS2.T1.High.mantel.Mcap,  permutations=1000)
    
#### T1.High Pcomp only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pcomp
    Bac_T1.High_mantel.Pcomp <- subset_samples(Bac.T1.High.mantel, Species == "Porites_compressa")
    Bac_T1.High_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pcomp) > 0, Bac_T1.High_mantel.Pcomp) 
    
    ITS2_T1.High_mantel.Pcomp  <- subset_samples(ITS_T1.High_mantel, Species == "Porites_compressa")
    ITS2_T1.High_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pcomp) > 0, ITS2_T1.High_mantel.Pcomp)
        
    #16s T1.High Pcomp dist        
      unifrac.Bac_T1.High_mantel.Pcomp <- phyloseq::distance(Bac_T1.High_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T1.High Pcomp dist
     bray.ITS2.T1.High.mantel.Pcomp <- phyloseq::distance(ITS2_T1.High_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pcomp, bray.ITS2.T1.High.mantel.Pcomp,  permutations=1000)        
        
    
#### T1.High Pvar only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pvar
    Bac_T1.High_mantel.Pvar <- subset_samples(Bac.T1.High.mantel, Species == "Pavona_varians")
    Bac_T1.High_mantel.Pvar = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pvar) > 0, Bac_T1.High_mantel.Pvar) 
    
    ITS2_T1.High_mantel.Pvar  <- subset_samples(ITS_T1.High_mantel, Species == "Pavona_varians")
    ITS2_T1.High_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pvar) > 0, ITS2_T1.High_mantel.Pvar)
        
    #16s T1.High Pvar dist        
      unifrac.Bac_T1.High_mantel.Pvar <- phyloseq::distance(Bac_T1.High_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T1.High Pvar dist
     bray.ITS2.T1.High.mantel.Pvar <- phyloseq::distance(ITS2_T1.High_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pvar, bray.ITS2.T1.High.mantel.Pvar,  permutations=1000)       
      
      #### T1.High Pacu only ####        
 # Bac_T1.High_mantel ITS2_T1.High_mantel
    
    #subset df for Pacu
    Bac_T1.High_mantel.Pacu <- subset_samples(Bac.T1.High.mantel, Species == "Pocillopora_acuta")
    Bac_T1.High_mantel.Pacu = prune_taxa(taxa_sums(Bac_T1.High_mantel.Pacu) > 0, Bac_T1.High_mantel.Pacu) 
    
    ITS2_T1.High_mantel.Pacu  <- subset_samples(ITS_T1.High_mantel, Species == "Pocillopora_acuta")
    ITS2_T1.High_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T1.High_mantel.Pacu) > 0, ITS2_T1.High_mantel.Pacu)
        
    #16s T1.High Pacu dist        
      unifrac.Bac_T1.High_mantel.Pacu <- phyloseq::distance(Bac_T1.High_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T1.High Pacu dist
     bray.ITS2.T1.High.mantel.Pacu <- phyloseq::distance(ITS2_T1.High_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.High_mantel.Pacu, bray.ITS2.T1.High.mantel.Pacu,  permutations=1000)  

```
T1 Ambient only
```{r}
# figure out which samples are represented in the its2 and 16s dfs

# T1 only  #####

#Ambient only 

#set up 16s
    Bac.s.rel.u.T1<-subset_samples(Bac.s.rel.u, Time.Point=="T1")
    Bac.s.rel.u.T1.Ambient<-subset_samples(Bac.s.rel.u.T1, Treatment=="Ambient")

    Bac.s.rel.u.T1.Ambient = prune_taxa(taxa_sums(Bac.s.rel.u.T1.Ambient) > 0, Bac.s.rel.u.T1.Ambient) #this removes any OTU with 0s
  # look at sample names
    bac.T1.sd <- as(sample_data(Bac.s.rel.u.T1.Ambient), "data.frame") #sample df
    #save just sample names
      bac.T1.Ambient.sd.samps<-bac.T1.sd[['sample_name.1']]

#set up ITS2 T1, import  T1TF
    ITS2_T1TF_mantel <- readRDS("ITS2_T1TF_phyloseq.rds") 
     ITS2_T1_mantel<-subset_samples(ITS2_T1TF_mantel, Time.Point=="T1")
    ITS2_T1_mantel.Ambient<-subset_samples(ITS2_T1_mantel, Treatment=="Ambient")
      ITS2_T1_mantel.Ambient = prune_taxa(taxa_sums(ITS2_T1_mantel.Ambient) > 0, ITS2_T1_mantel.Ambient) #this removes any OTU with 0s

    
  # look at sample names
    ITS.T1.Ambient.sd <- as(sample_data(ITS2_T1_mantel.Ambient), "data.frame") #sample df
  #save just sample names
      ITS.T1.Ambient.sd.samps<-ITS.T1.Ambient.sd[['ID']]

# compare ITS to 16s at T1
  #identify missing samples btw datasets     
      setdiff(bac.T1.Ambient.sd.samps,ITS.T1.Ambient.sd.samps) #samples in 16s but not ITS: "S1443b" "SN108b" "SN197b" "SN222"
      setdiff(ITS.T1.Ambient.sd.samps,bac.T1.Ambient.sd.samps) #samples in ITS but not 16s: "S1573" "S2424" "S3035" "S1443" "S1895" "SN209" "SN63"  "S1418" "SN197" "SN126" "S785"  "SN108"
        
      # in 16S remove SN222,remove the bs on the rest "S1443b" "SN108b" "SN197b"
      #in ITS  Remove "S1573" "S2424" "S3035" "S1895" "SN209" "SN63"  "S1418" " "SN126" "S785"  

      
    #######stoped here#########  
      
      
      
      
      
  # 16s changes: rename sample S3469
      
    Bac.T1.Ambient.mantel <- subset_samples(Bac.s.rel.u.T1.Ambient, sample_name.1 != "S3469" )
        #40 samples
          
   #ITS2 - remove samps not in the 16s df - " S3468, S670, S3025, S1388a
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S3468" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S670" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S3025" )
        ITS2_T1_mantel.Ambient <- subset_samples(ITS2_T1_mantel.Ambient, ID != "S1388a" )

        
    #change names ITS2 S1857b to S1857 and SN134b to SN134

    # pull OTU table
     OTU.ITS2T1.Ambient = as(otu_table(ITS2_T1_mantel.Ambient), "matrix") #pull otu
       OTU.ITS2T1.Ambient<-as.data.frame(OTU.ITS2T1.Ambient) #make into df
        row.names(OTU.ITS2T1.Ambient)[7] <- "S1857"   #change row name by row number
                row.names(OTU.ITS2T1.Ambient)[17] <- "SN134"   #change row name by row number

    #pull TAX table
     aTax.ITS2T1.Ambient = as(tax_table(ITS2_T1_mantel.Ambient), "matrix") #pull tax

    #pull SAMPLE DATA
     aSD.ITS2T1.Ambient = as(sample_data(ITS2_T1_mantel.Ambient), "matrix") #pull samp.data
     aSD.ITS2T1.Ambient<-as.data.frame(aSD.ITS2T1.Ambient) #change to df
        row.names(aSD.ITS2T1.Ambient)[7] <- "S1857"  #replace name S208b
        row.names(aSD.ITS2T1.Ambient)[17] <- "SN134"  #replace name S208b
            

    #remake the phyloseq obj
      samdata.Ambient = sample_data(aSD.ITS2T1.Ambient)
        seqtab.Ambient = otu_table(OTU.ITS2T1.Ambient, taxa_are_rows = FALSE)
        taxtab.Ambient = tax_table(aTax.ITS2T1.Ambient)
          ITS_T1.Ambient_mantel = phyloseq(otu_table(seqtab.Ambient), tax_table(taxtab.Ambient), sample_data(samdata.Ambient))
             
        # both have 40 samples - 
      
        # reorder ps object so samples are in the same order
        
        Bac.T1.Ambient.mantel <- Bac.T1.Ambient.mantel %>% ps_arrange(sample_name.1, desc(sample_name.1))
        phyloseq::sample_data(Bac.T1.Ambient.mantel) %>% head(8)
        
        ITS_T1.Ambient_mantel <- ITS_T1.Ambient_mantel %>% ps_arrange(ID, desc(ID))
        phyloseq::sample_data(ITS_T1.Ambient_mantel) %>% head(8)
        
  # MAKE data matrix
  
  #16s T1 Ambient        
        unifrac.Bac.T1.Ambient.mantel <- phyloseq::distance(Bac.T1.Ambient.mantel, method = "wunifrac")
        
  #ITS2 T1 Ambient
     bray.ITS2.T1.Ambient.mantel <- phyloseq::distance(ITS_T1.Ambient_mantel, method = "bray") 

  # Run mantel test
      mantel(unifrac.Bac.T1.Ambient.mantel, bray.ITS2.T1.Ambient.mantel,  permutations=1000)
    
        
#### T1.Ambient Mcap only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for mcap
    Bac_T1.Ambient_mantel.Mcap <- subset_samples(Bac.T1.Ambient.mantel, Species == "Montipora_capitata")
    Bac_T1.Ambient_mantel.Mcap = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Mcap) > 0, Bac_T1.Ambient_mantel.Mcap) 
    
    ITS2_T1.Ambient_mantel.Mcap  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Montipora_capitata")
    ITS2_T1.Ambient_mantel.Mcap = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Mcap) > 0, ITS2_T1.Ambient_mantel.Mcap)
        
    #16s T1.Ambient mcap dist        
      unifrac.Bac_T1.Ambient_mantel.Mcap <- phyloseq::distance(Bac_T1.Ambient_mantel.Mcap, method = "wunifrac")
        
   #ITS2 T1.Ambient mcap dist
     bray.ITS2.T1.Ambient.mantel.Mcap <- phyloseq::distance(ITS2_T1.Ambient_mantel.Mcap, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Mcap, bray.ITS2.T1.Ambient.mantel.Mcap,  permutations=1000)
    
#### T1.Ambient Pcomp only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pcomp
    Bac_T1.Ambient_mantel.Pcomp <- subset_samples(Bac.T1.Ambient.mantel, Species == "Porites_compressa")
    Bac_T1.Ambient_mantel.Pcomp = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pcomp) > 0, Bac_T1.Ambient_mantel.Pcomp) 
    
    ITS2_T1.Ambient_mantel.Pcomp  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Porites_compressa")
    ITS2_T1.Ambient_mantel.Pcomp = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pcomp) > 0, ITS2_T1.Ambient_mantel.Pcomp)
        
    #16s T1.Ambient Pcomp dist        
      unifrac.Bac_T1.Ambient_mantel.Pcomp <- phyloseq::distance(Bac_T1.Ambient_mantel.Pcomp, method = "wunifrac")
        
   #ITS2 T1.Ambient Pcomp dist
     bray.ITS2.T1.Ambient.mantel.Pcomp <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pcomp, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pcomp, bray.ITS2.T1.Ambient.mantel.Pcomp,  permutations=1000)        
        
    
#### T1.Ambient Pvar only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pvar
    Bac_T1.Ambient_mantel.Pvar <- subset_samples(Bac.T1.Ambient.mantel, Species == "Pavona_varians")
    Bac_T1.Ambient_mantel.Pvar = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pvar) > 0, Bac_T1.Ambient_mantel.Pvar) 
    
    ITS2_T1.Ambient_mantel.Pvar  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Pavona_varians")
    ITS2_T1.Ambient_mantel.Pvar = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pvar) > 0, ITS2_T1.Ambient_mantel.Pvar)
        
    #16s T1.Ambient Pvar dist        
      unifrac.Bac_T1.Ambient_mantel.Pvar <- phyloseq::distance(Bac_T1.Ambient_mantel.Pvar, method = "wunifrac")
        
   #ITS2 T1.Ambient Pvar dist
     bray.ITS2.T1.Ambient.mantel.Pvar <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pvar, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pvar, bray.ITS2.T1.Ambient.mantel.Pvar,  permutations=1000)       
      
      #### T1.Ambient Pacu only ####        
 # Bac_T1.Ambient_mantel ITS2_T1.Ambient_mantel
    
    #subset df for Pacu
    Bac_T1.Ambient_mantel.Pacu <- subset_samples(Bac.T1.Ambient.mantel, Species == "Pocillopora_acuta")
    Bac_T1.Ambient_mantel.Pacu = prune_taxa(taxa_sums(Bac_T1.Ambient_mantel.Pacu) > 0, Bac_T1.Ambient_mantel.Pacu) 
    
    ITS2_T1.Ambient_mantel.Pacu  <- subset_samples(ITS_T1.Ambient_mantel, Species == "Pocillopora_acuta")
    ITS2_T1.Ambient_mantel.Pacu = prune_taxa(taxa_sums(ITS2_T1.Ambient_mantel.Pacu) > 0, ITS2_T1.Ambient_mantel.Pacu)
        
    #16s T1.Ambient Pacu dist        
      unifrac.Bac_T1.Ambient_mantel.Pacu <- phyloseq::distance(Bac_T1.Ambient_mantel.Pacu, method = "wunifrac")
        
   #ITS2 T1.Ambient Pacu dist
     bray.ITS2.T1.Ambient.mantel.Pacu <- phyloseq::distance(ITS2_T1.Ambient_mantel.Pacu, method = "bray") 

   # Run mantel test
      mantel(unifrac.Bac_T1.Ambient_mantel.Pacu, bray.ITS2.T1.Ambient.mantel.Pacu,  permutations=1000)  

```
