---
title: "R Notebook"
html_document:
code_folding: hide
toc:yes
toc_depth:3
toc_float:yes
output: 
chunk_output_type: console
editor_options: 
  chunk_output_type: console
---

Data analysis for the All Coral Experiment #2021. Physiological variables:  
AFDW
Flow Cytometry (cell counts)
Protien
Biomass
BW
PAM


#Data and libraries
```{r} 
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(tidyverse)
library(factoextra)
library(reshape2)
library(vegan)
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
```

Load meta
```{r}     
meta  <- read.csv("Data/Master_Metadata_20190402.csv")
#remove frags that re duplicated or T3
meta<-meta[!(meta$X=="REMOVE"),]
meta<- meta[!(meta$Frag.ID == "3149"),] # dup
#write.csv(meta,"ACE21meta.csv")
# load in datasheets 
#Ashfreedryweight, Wax dipping, Total airbrush volume, metadata

#respiration check

RespCheck  <- read.csv("Data/RespCheck.csv")
RespCheckC<-cast(RespCheck, Frag.ID~Time.Point)

metaResp<-meta
metaResp<-metaResp[,c("Frag.ID","Time.Point")] # keep only columns you need
Respcheck3<-merge(metaResp, RespCheckC)
write.csv(Respcheck3,"Respcheck3.csv")


AFDW <- read.csv("Data/ACE21_AFDW_20210308.csv")
#AFDW <- AFDW[!(AFDW$Frag.ID == "1195"),]
#AFDW <- AFDW[!(AFDW$Frag.ID == "N49"),]
#AFDW <- AFDW[!(AFDW$Frag.ID == "3149"),]
AFDWtest<-merge(meta, AFDW, all = T)
#write.csv(AFDWtest,"AFDWtest.csv")

WaxDipping <- read.csv("Data/WaxDipping_Results_20190403.csv")
#WaxDipping <- WaxDipping[!(WaxDipping$Frag.ID == "1195"),]
#WaxDipping <- WaxDipping[!(WaxDipping$Frag.ID == "N49"),]
#WaxDipping <- WaxDipping[!(WaxDipping$Frag.ID == "3149"),]
#WaxDipping <- WaxDipping[!(WaxDipping$Frag.ID == "634"),]

Slurry <- read.csv("Data/TT17_airbrush_totalVol_20190420.csv")
#Slurry <- Slurry[!(Slurry$Frag.ID == "1195"),]
#Slurry <- Slurry[!(Slurry$Frag.ID == "N49"),]
#Slurry <- Slurry[!(Slurry$Frag.ID == "3149"),]
#Slurry <- Slurry[!(WaxDipping$Frag.ID == "634"),]
Slurrytest<-merge(meta, Slurry, all.x = T)

#merge 
testAFDW<-merge(AFDW, meta, all=T)
#testSlurry<-merge(Slurry, meta, all.y=T)
#testflow<-merge(tData, meta, all.y=T)
#SurfaceArea<-merge(SA, meta, all.y=T)
#write.csv(SurfaceArea, "SurfaceAreaACE21.csv")



```


#Biomass (AFDW and SA)
```{r}    
 
SA<-WaxDipping 
SA<- SA[,c("Frag.ID","Surface.area")] # keep only columns you need
#WaxDippingtest<-merge(meta, SA, all.x = T)

# merge AFDW and slurry
AFDW.slurry <- merge(AFDW, Slurry)
AFDW.slurry$AFDW.mg.ml <- AFDW.slurry$AFDW.g.mL*1000  # grams to mg
AFDW.slurry$AFDWmg <-AFDW.slurry$AFDW.mg.ml*AFDW.slurry$TotalVolume # now we have mg of AFDW


DUP2<-duplicated(AFDW.slurry$Frag.ID) #look for duplicate entries

# SA cm2 and AFDW g/mL --> mg/cm2

# merge
biomass <- merge(AFDW.slurry, SA)
biomass$Surface.area<-as.numeric(as.character(biomass$Surface.area))
biomass<- biomass[,c("Frag.ID", "AFDWmg", "Surface.area")] # keep only columns you need
biomass$AFDWmg.cm2 <-biomass$AFDWmg/biomass$Surface.area # mg/cm2

# merge with metadata to plot by treatments
biomass2 <- merge(biomass, meta)
#write.csv(biomass2, "biomass2.csv")

# Mcap_biomass
biomass2_mean <- ddply(biomass2, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                 N    = length(AFDWmg.cm2[!is.na(AFDWmg.cm2)]), #calculate the length of the data frame, excluding NAâ€™s *this part is important*
                 mean = mean(AFDWmg.cm2, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(AFDWmg.cm2, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(AFDWmg.cm2, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
biomass2_mean #display table
#Mcap_biomass_mean <-Mcap_biomass_mean[-3,] # Remove T0
#biomass2_mean

# Make "BOTH" into Ambeint and High...THIS IS UGLY CODE BUT IT WORKS
# Mcap 
#Mcap_biomass_mean<-subset(biomass2_mean, Species=="Montipora_capitata") #Subset Mcap

#Mcap_biomass_mean_Both<-Mcap_biomass_mean #copy df
#row.names.remove <- c("1","2","13","14") #remove high and amb by row numbers
#Mcap_biomass_mean_Both<-Mcap_biomass_mean_Both[!(row.names(Mcap_biomass_mean_Both) %in% row.names.remove), ] #remove

#Mcap_biomass_mean_Both2<-Mcap_biomass_mean_Both # make a copy of the BOTH df
#Mcap_biomass_mean_Both[Mcap_biomass_mean_Both=="Both"]<-"Ambient"  #rename "Both" to "Ambient"
#Mcap_biomass_mean_Both2[Mcap_biomass_mean_Both2=="Both"]<-"High"  #rename "Both" to "High"

#row.names.remove <- c("9") #remove "high and amb by row numbers "both""
#Mcap_biomass_mean<-Mcap_biomass_mean[!(row.names(Mcap_biomass_mean) %in% row.names.remove), ]
#Mcap_biomass_mean<-rbind(Mcap_biomass_mean,Mcap_biomass_mean_Both,Mcap_biomass_mean_Both2) #bind together

#Mcap 
Mcap_biomass_mean<-subset(biomass2_mean, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_biomass_mean<-subset(biomass2_mean, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_biomass_mean<-subset(biomass2_mean, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_biomass_mean<-subset(biomass2_mean, Species=="Pavona_varians") #Subset Pvar


```
Stats - T0 is n=6, not n=12
```{r}
Biomass_stats<-biomass2

hist(Biomass_stats$AFDWmg.cm2) 
Biomass_stats$AFDWmg.cm2_log<-log10(Biomass_stats$AFDWmg.cm2+1)
hist(Biomass_stats$AFDWmg.cm2_log) #normal (use)

Biomass_model<-lmer(AFDWmg.cm2_log~Time.Point*Treatment*Species+(1|Parent.ID)+(1|Tank.num), Biomass_stats)
anova(Biomass_model) 
qqPlot(residuals(Biomass_model)) 

  emm<-emmeans(Biomass_model,  ~ Time.Point*Treatment|Species, adjust="Tukey")
cld(emm)
pairs(emm)

```
PLOTS
```{r}  
#plot Mcap
Mcap_biomass_plot<-ggplot(data=Mcap_biomass_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  ylim(0, 10) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("Tissue biomass mg cm"^"-2"))) +
  ggtitle("Montipora capitata")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_biomass_plot

# Pcomp 

Pcomp_biomass_plot<-ggplot(data=Pcomp_biomass_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  ylab("") + #Label the X Axis
  #ylim(0, 10) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  #ylab(expression(paste("Tissue biomass mg cm"^"-2"))) +
  ggtitle("Porites compressa")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_biomass_plot

# Pvar 

Pvar_biomass_mean_plot<-ggplot(data=Pvar_biomass_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  ylim(2.5, 6) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("Tissue biomass mg cm"^"-2"))) +
  ggtitle("Pavona varians")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_biomass_mean_plot

# Pacu 

Pacu_biomass_mean_plot<-ggplot(data=Pacu_biomass_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  ylab("") + #Label the Y Axis
  ylim(0, 10) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background =element_blank(), #Set the plot background
        text = element_text(size=18),  # set element text
        legend.key = element_blank()) + #Set plot legend key
  #ylab(expression(paste("Tissue biomass mg cm"^"-2"))) +
  ggtitle("Pocillopora acuta")+
  theme(plot.title = element_text(size=20,face = "italic"));Pacu_biomass_mean_plot


###### All plots together
Mcap_biomass_plot_plot<-Mcap_biomass_plot+ theme(legend.position = "none") #remove legend
Pcomp_biomass_plot<-Pcomp_biomass_plot+ theme(legend.position = "none") #remove legend
Pvar_biomass_mean_plot<-Pvar_biomass_mean_plot+ theme(legend.position = "none") #remove legend
Pacu_biomass_mean_plot<-Pacu_biomass_mean_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_biomass_plot, Pcomp_biomass_plot, Pvar_biomass_mean_plot,Pacu_biomass_mean_plot, nrow = 2)
 
```

# Flow cytometry  cell counts
```{r}   
#import data
Std.curve <- read.csv("Data/standard_curve_fcm20181312.csv")

###notes: 2289 isn't anywhere. 

PlateA <- read.csv("Data/Symbio.Counts.Dec.14.2018.PlateA.csv")
PlateA <- subset(PlateA, Gate.Name == "Symbiodinium")       # keep only Symbiodinium rows
keepA <- c("Sample.Name", "Concentration") # keep only Sample.name and concentration
PlateA <- PlateA[keepA]
A_meta <- read.csv("Data/TT17_FC_PlateA.csv") # load Plate A location data
PlateA <- merge(PlateA, A_meta)   # merge with location data
PlateA <-merge(PlateA, meta, all.x = TRUE) # merge with Sample metadata
PlateA <- na.omit(PlateA) # omit NAs
PlateA<- PlateA[!(PlateA$Frag.ID == "1133"),] # redid in plate E
PlateA<- PlateA[!(PlateA$Frag.ID == "N172"),] # redid in plate E


PlateB <- read.csv("Data/Symbio.Counts.Dec.14.2018.PlateB1.csv")
PlateB$Concentration<-as.numeric(as.character(PlateB$Concentration)) # this is loading in as a factor, not sure why
PlateB <- subset(PlateB, Gate.Name == "Symbiodinium")
keepB <- c("Sample.Name", "Concentration")
PlateB <- PlateB[keepB]
B_meta <- read.csv("Data/TT17_FC_PlateB.csv") # lood Plate B location data
PlateB <- merge(PlateB, B_meta)
PlateB <-merge(PlateB, meta, all.x = TRUE)
PlateB <- na.omit(PlateB)
PlateB<- PlateB[!(PlateB$Frag.ID == "428"),] # redid in plate E

PlateC <- read.csv("Data/Symbio.Counts.Dec.14.2018.PlateCredo.csv")
PlateC$Concentration<-as.numeric(as.character(PlateC$Concentration)) # this is loading in as a factor, not sure why
PlateC <- subset(PlateC, Gate.Name == "Symbiodinium")
keepC <- c("Sample.Name", "Concentration")
PlateC <- PlateC[keepC]
C_meta <- read.csv("Data/TT17_FC_PlateCredo.csv") # lood Plate C location data
PlateC <- merge(PlateC, C_meta)
PlateC <-merge(PlateC, meta, all.x = TRUE)
PlateC <- na.omit(PlateC)
PlateC<- PlateC[!(PlateC$Frag.ID == "3481"),] # redid in plate E
PlateC<- PlateC[!(PlateC$Frag.ID == "49"),] # redid in plate E
PlateC<- PlateC[!(PlateC$Frag.ID == "N49"),] # redid in plate E

#note: Frag.ID 403 Pacu seems abnormally high

PlateE <- read.csv("Data/Symbio.Count.apr.02.2019.PlateE.csv")
PlateE$Concentration<-as.numeric(as.character(PlateE$Concentration)) # this is loading in as a factor, not sure why
PlateE <- subset(PlateE, Gate.Name == "Symbiodinium")
keepE <- c("Sample.Name", "Concentration")
PlateE <- PlateE[keepE]
E_meta <- read.csv("Data/TT17_FC_PlateE.csv") # lood Plate A location data
PlateE <- merge(PlateE, E_meta)
PlateE <-merge(PlateE, meta, all.x = TRUE)
PlateE <- na.omit(PlateE)

# Bind all runs together
tData <- rbind(PlateA, PlateB, PlateC, PlateE)
tData$Parent.ID<- as.factor(tData$Parent.ID) #for some reason, "Concentration" was a chr not numeric. Check by calling str()

# Create column to relate to standard curve
tData$corrected <- tData$Concentration*3.2959 # <- also confirm you did this correctly. 

# Use Slurry
#merge with tdata
tData <- merge(tData, Slurry)
tData$TotalVolume <- tData$TotalVolume * 1000 # change total volume to from mL to uL

write.csv(tData,"FLOW.csv")
#use SA for slurry

#merge with tData, make numeric
tData <- merge(tData, SA)
tData$Surface.area<- as.numeric(as.character(tData$Surface.area))
duplicated(tData$Frag.ID) #check 428 Pacu TF, N172mcap T1,1133 mcap t1


#write.csv(tData, "tdataCHECK.csv")
#Subset by Species and correct for 10% dilutions in Mcap and Pcomp
Mcap_FC <- subset(tData, Species == "Montipora_capitata")
Mcap_FC$corrected <- Mcap_FC$corrected*10 # to compensate for the 10% dultion

Pcomp_FC <- subset(tData, Species == "Porites_compressa")
Pcomp_FC$corrected <- Pcomp_FC$corrected*10 # to compensate for the 10% dultion

Pvar_FC <- subset(tData, Species == "Pavona_varians") #not diluted
Pacu_FC <- subset(tData, Species == "Pocillopora_acuta") #not diluted

# Create new column for total cells in total slurry
Mcap_FC$Tot.zoox <- Mcap_FC$corrected*(Mcap_FC$TotalVolume)
Pcomp_FC$Tot.zoox <- Pcomp_FC$corrected*(Pcomp_FC$TotalVolume)
Pvar_FC$Tot.zoox <- Pvar_FC$corrected*(Pvar_FC$TotalVolume)
Pacu_FC$Tot.zoox <- Pacu_FC$corrected*(Pacu_FC$TotalVolume)

# Calculate zoox/cm2
Mcap_FC$zoox.cm2<-Mcap_FC$Tot.zoox/Mcap_FC$Surface.area
Pcomp_FC$zoox.cm2<-Pcomp_FC$Tot.zoox/Pcomp_FC$Surface.area
Pvar_FC$zoox.cm2<-Pvar_FC$Tot.zoox/Pvar_FC$Surface.area
Pacu_FC$zoox.cm2<-Pacu_FC$Tot.zoox/Pacu_FC$Surface.area

FC_Data <- rbind(Mcap_FC, Pcomp_FC, Pvar_FC, Pacu_FC) #combine

#mean tables
# Mcap
Mcap_FC_Sum <- ddply(Mcap_FC, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                   N    = length(zoox.cm2[!is.na(zoox.cm2)]), #calculate the length of the data frame, excluding NAâ€™s *this part is important*
                   mean = mean(zoox.cm2, na.rm=TRUE), #calculate mean of response variable, removing NA's
                   sd   = sd(zoox.cm2, na.rm=TRUE), #calculate standard deviation
                   se   = sd / sqrt(N), #calculate standard error
                   max = max(zoox.cm2, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Mcap_FC_Sum #display table

# Pcomp
Pcomp_FC_Sum <- ddply(Pcomp_FC, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                   N    = length(zoox.cm2[!is.na(zoox.cm2)]), #calculate the length of the data frame, excluding NAâ€™s *this part is important*
                   mean = mean(zoox.cm2, na.rm=TRUE), #calculate mean of response variable, removing NA's
                   sd   = sd(zoox.cm2, na.rm=TRUE), #calculate standard deviation
                   se   = sd / sqrt(N), #calculate standard error
                   max = max(zoox.cm2, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pcomp_FC_Sum #display table


# Pvar
Pvar_FC_Sum <- ddply(Pvar_FC, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                   N    = length(zoox.cm2[!is.na(zoox.cm2)]), #calculate the length of the data frame, excluding NAâ€™s *this part is important*
                   mean = mean(zoox.cm2, na.rm=TRUE), #calculate mean of response variable, removing NA's
                   sd   = sd(zoox.cm2, na.rm=TRUE), #calculate standard deviation
                   se   = sd / sqrt(N), #calculate standard error
                   max = max(zoox.cm2, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pvar_FC_Sum #display table

# Pacu
Pacu_FC_Sum <- ddply(Pacu_FC, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                   N    = length(zoox.cm2[!is.na(zoox.cm2)]), #calculate the length of the data frame, excluding NAâ€™s *this part is important*
                   mean = mean(zoox.cm2, na.rm=TRUE), #calculate mean of response variable, removing NA's
                   sd   = sd(zoox.cm2, na.rm=TRUE), #calculate standard deviation
                   se   = sd / sqrt(N), #calculate standard error
                   max = max(zoox.cm2, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Pacu_FC_Sum #display table

```
Stats
```{r}



hist(FC_Data$zoox.cm2) 
FC_Data$zoox.cm2_log<-log10(FC_Data$zoox.cm2+1)
hist(FC_Data$zoox.cm2_log) #normal (use)

CellCount_model<-lmer(zoox.cm2_log~Time.Point*Treatment*Species+(1|Parent.ID)+(1|Tank.num), FC_Data)
anova(CellCount_model) 
qqPlot(residuals(CellCount_model)) 

  emm<-emmeans(CellCount_model,  ~ Time.Point*Treatment|Species, adjust="Tukey")
cld(emm)
pairs(emm)
```

Line plots
```{r} 
# Mcap  

Mcap_FC_lineplot<-ggplot(data=Mcap_FC_Sum, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
       scale_y_log10()+
    #  ylim(0, 4000000)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("cells"))) +
  ggtitle("Mcap Flow Cytometry")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Mcap_FC_lineplot

# Pcomp 

Pcomp_FC_lineplot<-ggplot(data=Pcomp_FC_Sum, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
       scale_y_log10()+
   # ylim(0, 4000000)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("cells"))) +
  ggtitle("Pcomp Flow Cytometry")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pcomp_FC_lineplot


# Pvar 

Pvar_FC_lineplot<-ggplot(data=Pvar_FC_Sum, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
       scale_y_log10()+
   # ylim(0, 4000000)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("cells"))) +
  ggtitle("Pvar Flow Cytometry")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pvar_FC_lineplot

# Pacu 

Pacu_FC_lineplot<-ggplot(data=Pacu_FC_Sum, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
       scale_y_log10()+
  #ylim(0, 4000000)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("cells"))) +
  ggtitle("Pacu Flow Cytometry")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pacu_FC_lineplot

###### All plots together
Mcap_FC_lineplot<-Mcap_FC_lineplot+ theme(legend.position = "none") #remove legend
Pcomp_FC_lineplot<-Pcomp_FC_lineplot+ theme(legend.position = "none") #remove legend
Pvar_FC_lineplot<-Pvar_FC_lineplot+ theme(legend.position = "none") #remove legend
Pacu_FC_lineplot<-Pacu_FC_lineplot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_FC_lineplot, Pcomp_FC_lineplot, Pvar_FC_lineplot,Pacu_FC_lineplot, nrow = 2)

```
#Buoyant Weight/Calcification

calibration (not sure what this does)
```{r}     
BWCalData <- read.csv('Data/Bouyant_weight_calibration_curve.csv')
#Data with Standard weights in grams in deionized water, saltwater, and the temp that they were measured at

### Standard Cruve
#plot relationship between temp and Standard in fresh and salt water
plot(BWCalData$Temp, BWCalData$StandardSalt, col = 'red', ylab = 'Salt Weight (g)', xlab = 'Temp C')
par(new=TRUE)
plot(BWCalData$Temp, BWCalData$StandardFresh, col = 'blue',xaxt="n",yaxt="n",xlab="",ylab="")
axis(4)
mtext("'Fresh Weight (g)'",side=4,line=3)
legend("topleft",col=c("red","blue"),lty=1,legend=c("Salt","Fresh"), bty = "n")

#linear regression between temp and Standard
#Salt
StandardSaltModel <- lm(BWCalData$StandardSalt~BWCalData$Temp)
summary(StandardSaltModel)
summary(StandardSaltModel)$r.squared
StandardSaltModel$coef

#Fresh
StandardFreshModel <- lm(BWCalData$StandardFresh~BWCalData$Temp)
summary(StandardFreshModel)
summary(StandardFreshModel)$r.squared
StandardFreshModel$coef

```
Analysis 
```{r}  

#load coral weight data
BW.data <-read.csv('Data/BW_TT17.csv') # this is a file I compiled with all dates
# merge to sample data for sorting
AllBWdata<-merge(BW.data, meta, by = "Frag.ID")

## Split data by species, change to numeric
Mcap_BW<- subset(AllBWdata, Species =="Montipora_capitata")
Mcap_BW$Temp.C <- as.numeric(as.character(Mcap_BW$Temp.C))
BW_Mcap <- Mcap_BW$Mass.g
Temp_Mcap <- Mcap_BW$Temp.C 

Pcomp_BW<- subset(AllBWdata, Species =="Porites_compressa")
Pcomp_BW$Temp.C <- as.numeric(as.character(Pcomp_BW$Temp.C))
BW_Pcomp <- Pcomp_BW$Mass.g
Temp_Pcomp <- Pcomp_BW$Temp.C 

Pvar_BW<- subset(AllBWdata, Species =="Pavona_varians")
Pvar_BW$Temp.C <- as.numeric(as.character(Pvar_BW$Temp.C))
BW_Pvar <- Pvar_BW$Mass.g
Temp_Pvar <- Pvar_BW$Temp.C 

Pacu_BW<- subset(AllBWdata, Species =="Pocillopora_acuta")
BW_Pacu <- Pacu_BW$Mass.g
Temp_Pacu <- Pacu_BW$Temp.C  


```
Montipora capitata
```{r} 
BWCalc_Mcap <- function(StandardAir= 39.092, Temp, BW, CoralDensity = 2.03){
  #Parameters---------------------------------------------------------------
  # StandardAir is the weight of the Standard in air (Default set at 39.092 grams weighed on balance)
  # Temp is the temperature of the water during the coral measurement
  # BW is the buoywant weight of the coral
  # CoralDensity <- 2.03 #g cm-3, set aragonite density for Montipora from literature 
  #Montipora Arag = 2.03 g cm-3 average from table 2 in Anthony 2003 Functional Ecology 17:246-259
  # Porites asteroides T Hughes 1987 1.69 at 5m (KB got 1.82)
  # below: Medellin-Maldonado et al 2016
  # Pavona varians 1.04 +-0.07
  # panama 8.5380Â° N, 80.7821Â° W, 
  # coasta rica 9.7489Â° N, 83.7534Â° W
  # mexico la entrega 15.7424Â° N, 96.1282Â° W
  # hawaii 19.8968Â° N, 155.5828Â° W
  # Pocillopora 1.78 +- 0.31
  
  
  # You can change the density to literatrue values for the specific coral species of interest in the function. 
  
 #Calculation------------------------------------------------------------
  #Step 1: correct the Standard weights for temperature
  # StandardFresh is the weight of the Standard in  fresh water at temp x
  # StandardSalt is the weight of the Standard in salt water at temp x
  
  # This is based on a calibration curve for Standards weighed in fresh and salt water at many temps
StandardFresh <- StandardFreshModel$coef[-1]*Temp + StandardFreshModel$coef[1] 
StandardSalt <- StandardSaltModel$coef[-1]*Temp + StandardSaltModel$coef[1] 
  
  # Step 2: Use weight in air and freshwater of the glass Standard to calculate
  # the density of the Standard (Davies eq. 4)
FreshDensity <- 1 #Fresh water has a density of 1 g/cm3
StandardDensity <- FreshDensity/(1-(StandardFresh/StandardAir)) 
  
  # Step 3: Calculate the density of seawater using the density of the Standard
  # (Spencer Davies eq. 3)
SWDensity <- StandardDensity*(1-(StandardSalt/StandardAir))
  
  # Step 4: Calculate the dry weight of the coral (Davies eq. 1)
CoralWeight <- BW_Mcap/(1-(SWDensity/CoralDensity))
  
  return(CoralWeight) #returns coral dry weights in g
}
  
Mcap_BW$Dry.Weigh.g <- BWCalc_Mcap(BW=BW_Mcap, Temp=Temp_Mcap) #use function to calculate dry weight
hist(Mcap_BW$Dry.Weigh.g) #not normal
Mcap_BW$Dry.Weigh.g_log<-log10(Mcap_BW$Dry.Weigh.g+1)
hist(Mcap_BW$Dry.Weigh.g_log) #better still tail
boxplot(Mcap_BW$Dry.Weigh.g) #examine data
Mcap_BW <- Mcap_BW[!(Mcap_BW$Frag.ID == "797"),] # remove sample 797, error in BW record (impossible value)

# below, we are getting rid of columns we don't need anymore, and reorganizing the data by Fragment and clade/color and treatment. This way our time points wil be in the same row)
data_McapBW <- reshape(Mcap_BW, timevar = "TimePoint", drop = c("Mass.g","Temp.C", "Date","Tank", "X",  "Parent.ID", "Species","Time.Point", "Tank.num"), idvar=c("Frag.ID", "Treatment"), direction="wide")

# This calculates the growth rate, if you look it takes Timepoint1 subtracts timepoint0 and divides by timepoint 0 and then divides by number of days , times 100
data_McapBW$time1.growth <- (((data_McapBW$Dry.Weigh.g.Time1 - data_McapBW$Dry.Weigh.g.Time0)/data_McapBW$Dry.Weigh.g.Time0)*100)/data_McapBW$Days.Time1 #calculate growth rate for time1
data_McapBW$time2.growth <- (((data_McapBW$Dry.Weigh.g.Time2 - data_McapBW$Dry.Weigh.g.Time1)/data_McapBW$Dry.Weigh.g.Time1)*100)/data_McapBW$Days.Time2 #calculate growth rate for time2





# mean, standard dev, and sem
time1_McapBW <-ddply(data_McapBW, .( Treatment), summarize, #For each subset of a data frame, apply function then combine results into a data frame.
              N=length(na.omit(time1.growth)), # the sample size of the light column summarized by Tank number
              mean = (mean(time1.growth, na.rm=TRUE)) ,  #take the average of the light column summarized by Tank number
              sd = sd(time1.growth, na.rm=TRUE) ,# the standard deviation of the temp column summarized by Tank number
              sem = (sd(time1.growth,na.rm=T)/sqrt(N))) #calculate the SEM as the sd/sqrt of the count or data length
time1_McapBW$Time <- "Time1"  # this puts these data in our "time1" dataframe

time2_McapBW <-ddply(data_McapBW, .( Treatment), summarize, #For each subset of a data frame, apply function then combine results into a data frame.
              N=length(na.omit(time2.growth)), # the sample size of the light column summarized by Tank number
              mean = (mean(time2.growth, na.rm=TRUE)) ,  #take the average of the light column summarized by Tank number
              sd = sd(time2.growth, na.rm=TRUE) ,# the standard deviation of the temp column summarized by Tank number
              sem = (sd(time2.growth,na.rm=T)/sqrt(N))) #calculate the SEM as the sd/sqrt of the count or data length
time2_McapBW$Time <- "Time2"  # this puts these data in our "time1" dataframe

#save this dataframe as "G"
G_McapA <- rbind(time1_McapBW[1,],time2_McapBW[1,])# Greate a new column that we rename each row by treatment
G_McapH <- rbind(time1_McapBW[2,],time2_McapBW[2,])# Greate a new column that we rename each row by treatment
G_Mcap<-rbind(G_McapA, G_McapH)
# Stress.Recovery column 
 # make new column that renames "time" to Stress or Recovery
G_Mcap$newtime = ifelse(G_Mcap$Time == "Time1", "Post-Stress", "Post-Recovery")
G_Mcap <- within(G_Mcap, newtime <- factor(newtime, levels = c("Post-Stress", "Post-Recovery"))) # Set order of levels so plots in right order



Mcap_BW_plot<-ggplot(data=G_Mcap, aes(x=newtime, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
     # ylim(0, .7)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("% Change Dry.Weigh.g"))) +
  ggtitle("Montipora capitata")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.25));Mcap_BW_plot


```
Porites compressa
```{r} 
 
################################     PORITES COMPRESSA
BWCalc_Pcomp <- function(StandardAir= 39.092, Temp, BW, CoralDensity = 1.82){
  #Parameters---------------------------------------------------------------
  # StandardAir is the weight of the Standard in air (Default set at 39.092 grams weighed on balance)
  # Temp is the temperature of the water during the coral measurement
  # BW is the buoywant weight of the coral
  # CoralDensity <- 2.03 #g cm-3, set aragonite density for Montipora from literature 
  # Montipora Arag = 2.03 g cm-3 average from table 2 in Anthony 2003 Functional Ecology 17:246-259
  
  # Porites asteroides (T Hughes 1987 1.69 at 5m) (KB got 1.82)
  
  # below: Medellin-Maldonado et al 2016
  # Pavona varians 1.04 +-0.07
  # panama 8.5380Â° N, 80.7821Â° W, 
  # coasta rica 9.7489Â° N, 83.7534Â° W
  # mexico la entrega 15.7424Â° N, 96.1282Â° W
  # hawaii 19.8968Â° N, 155.5828Â° W
  # Pocillopora 1.78 +- 0.31
  
  
  # You can change the density to literatrue values for the specific coral species of interest in the function. 
  
  #Calculation------------------------------------------------------------
  #Step 1: correct the Standard weights for temperature
  # StandardFresh is the weight of the Standard in  fresh water at temp x
  # StandardSalt is the weight of the Standard in salt water at temp x
  
  # This is based on a calibration curve for Standards weighed in fresh and salt water at many temps
  StandardFresh <- StandardFreshModel$coef[-1]*Temp + StandardFreshModel$coef[1] 
  StandardSalt <- StandardSaltModel$coef[-1]*Temp + StandardSaltModel$coef[1] 
  
  # Step 2: Use weight in air and freshwater of the glass Standard to calculate
  # the density of the Standard (Davies eq. 4)
  FreshDensity <- 1 #Fresh water has a density of 1 g/cm3
  StandardDensity <- FreshDensity/(1-(StandardFresh/StandardAir)) 
  
  # Step 3: Calculate the density of seawater using the density of the Standard
  # (Spencer Davies eq. 3)
  SWDensity <- StandardDensity*(1-(StandardSalt/StandardAir))
  
  # Step 4: Calculate the dry weight of the coral (Davies eq. 1)
  CoralWeight <- BW_Pcomp/(1-(SWDensity/CoralDensity))
  
  return(CoralWeight) #returns coral dry weights in g
}


Pcomp_BW$Dry.Weigh.g <- BWCalc_Pcomp(BW=BW_Pcomp, Temp=Temp_Pcomp) #use function to calculate dry weight
hist(Pcomp_BW$Dry.Weigh.g) 
Pcomp_BW$Dry.Weigh.g_log<-log10(Pcomp_BW$Dry.Weigh.g+1)
hist(Pcomp_BW$Dry.Weigh.g_log) # normal
boxplot(Pcomp_BW$Dry.Weigh.g_log) #examine data

# below, we are getting rid of columns we don't need anymore, and reorganizing the data by Fragment and clade/color and treatment. This way our time points wil be in the same row)
data_PcompBW <- reshape(Pcomp_BW, timevar = "TimePoint", drop = c("Mass.g","Temp.C", "Date","Tank", "X", "Time.Point", "Parent.ID", "Species", "Tank.num"), idvar=c("Frag.ID", "Treatment"), direction="wide")

# This calculates the growth rate, if you look it takes Timepoint1 subtracts timepoint0 and divides by timepoint 0, times 100
data_PcompBW$time1.growth <- (((data_PcompBW$Dry.Weigh.g.Time1 - data_PcompBW$Dry.Weigh.g.Time0)/data_PcompBW$Dry.Weigh.g.Time0)*100)/data_PcompBW$Days.Time1 #calculate growth rate for time1
data_PcompBW$time2.growth <- (((data_PcompBW$Dry.Weigh.g.Time2 - data_PcompBW$Dry.Weigh.g.Time1)/data_PcompBW$Dry.Weigh.g.Time1)*100)/data_PcompBW$Days.Time2 #calculate growth rate for time2

data_PcompBW$time1.growth[data_PcompBW$time1.growth < 0] <- 0 
data_PcompBW$time2.growth[data_PcompBW$time2.growth < 0] <- 0 

# mean, standard dev, and sem
time1_PcompBW <-ddply(data_PcompBW, .( Treatment), summarize, #For each subset of a data frame, apply function then combine results into a data frame.
              N=length(na.omit(time1.growth)), # the sample size of the light column summarized by Tank number
              mean = (mean(time1.growth, na.rm=TRUE)) ,  #take the average of the light column summarized by Tank number
              sd = sd(time1.growth, na.rm=TRUE) ,# the standard deviation of the temp column summarized by Tank number
              sem = (sd(time1.growth,na.rm=T)/sqrt(N))) #calculate the SEM as the sd/sqrt of the count or data length
time1_PcompBW$Time <- "Time1"  # this puts these data in our "time1" dataframe

time2_PcompBW <-ddply(data_PcompBW, .( Treatment), summarize, #For each subset of a data frame, apply function then combine results into a data frame.
              N=length(na.omit(time2.growth)), # the sample size of the light column summarized by Tank number
              mean = (mean(time2.growth, na.rm=TRUE)) ,  #take the average of the light column summarized by Tank number
              sd = sd(time2.growth, na.rm=TRUE) ,# the standard deviation of the temp column summarized by Tank number
              sem = (sd(time2.growth,na.rm=T)/sqrt(N))) #calculate the SEM as the sd/sqrt of the count or data length
time2_PcompBW$Time <- "Time2"  # this puts these data in our "time1" dataframe

#save this dataframe as "G"
G_PcompA <- rbind(time1_PcompBW[1,],time2_PcompBW[1,])# Greate a new column that we rename each row by treatment
G_PcompH <- rbind(time1_PcompBW[2,],time2_PcompBW[2,])# Greate a new column that we rename each row by treatment
G_Pcomp<-rbind(G_PcompA, G_PcompH)
# Stress.Recovery column 
 # make new column that renames "time" to Stress or Recovery
G_Pcomp$newtime = ifelse(G_Pcomp$Time == "Time1", "Post-Stress", "Post-Recovery")
G_Pcomp <- within(G_Pcomp, newtime <- factor(newtime, levels = c("Post-Stress", "Post-Recovery"))) # Set order of levels so plots in right order



Pcomp_BW_plot<-ggplot(data=G_Pcomp, aes(x=newtime, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, .7)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
 ylab(expression(paste(""))) +
  ggtitle("Porites compressa")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.25));Pcomp_BW_plot

```
Pavona varians
```{r}
BWCalc_Pvar <- function(StandardAir= 39.092, Temp, BW, CoralDensity = 1.04){
  #Parameters---------------------------------------------------------------
  # StandardAir is the weight of the Standard in air (Default set at 39.092 grams weighed on balance)
  # Temp is the temperature of the water during the coral measurement
  # BW is the buoywant weight of the coral
  # CoralDensity <- 2.03 #g cm-3, set aragonite density for Montipora from literature 
  # Montipora Arag = 2.03 g cm-3 average from table 2 in Anthony 2003 Functional Ecology 17:246-259
  
  # Porites asteroides (T Hughes 1987 1.69 at 5m) (KB got 1.82)
  
  # below: Medellin-Maldonado et al 2016
  # Pavona varians 1.04 +-0.07
  # panama 8.5380Â° N, 80.7821Â° W, 
  # coasta rica 9.7489Â° N, 83.7534Â° W
  # mexico la entrega 15.7424Â° N, 96.1282Â° W
  # hawaii 19.8968Â° N, 155.5828Â° W
  # Pocillopora 1.78 +- 0.31
  
  
  # You can change the density to literatrue values for the specific coral species of interest in the function. 
  
  #Calculation------------------------------------------------------------
  #Step 1: correct the Standard weights for temperature
  # StandardFresh is the weight of the Standard in  fresh water at temp x
  # StandardSalt is the weight of the Standard in salt water at temp x
  
  # This is based on a calibration curve for Standards weighed in fresh and salt water at many temps
  StandardFresh <- StandardFreshModel$coef[-1]*Temp + StandardFreshModel$coef[1] 
  StandardSalt <- StandardSaltModel$coef[-1]*Temp + StandardSaltModel$coef[1] 
  
  # Step 2: Use weight in air and freshwater of the glass Standard to calculate
  # the density of the Standard (Davies eq. 4)
  FreshDensity <- 1 #Fresh water has a density of 1 g/cm3
  StandardDensity <- FreshDensity/(1-(StandardFresh/StandardAir)) 
  
  # Step 3: Calculate the density of seawater using the density of the Standard
  # (Spencer Davies eq. 3)
  SWDensity <- StandardDensity*(1-(StandardSalt/StandardAir))
  
  # Step 4: Calculate the dry weight of the coral (Davies eq. 1)
  CoralWeight <- BW_Pvar/(1-(SWDensity/CoralDensity))
  
  return(CoralWeight) #returns coral dry weights in g
}

Pvar_BW$Dry.Weigh.g <- BWCalc_Pvar(BW=BW_Pvar, Temp=Temp_Pvar) #use function to calculate dry weight
hist(Pvar_BW$Dry.Weigh.g) #not normal

boxplot(Pvar_BW$Dry.Weigh.g) #examine data
Pvar_BW$Dry.Weigh.g_log<-log10(Pvar_BW$Dry.Weigh.g+1)
hist(Pvar_BW$Dry.Weigh.g_log) #not normal


# below, we are getting rid of columns we don't need anymore, and reorganizing the data by Fragment and clade/color and treatment. This way our time points wil be in the same row)
data_PvarBW <- reshape(Pvar_BW, timevar = "TimePoint", drop = c("Mass.g","Temp.C", "Date","Tank", "X",  "Parent.ID", "Species","Time.Point", "Tank.num"), idvar=c("Frag.ID", "Treatment"), direction="wide")

# This calculates the growth rate, if you look it takes Timepoint1 subtracts timepoint0 and divides by timepoint 0, times 100
data_PvarBW$time1.growth <- (((data_PvarBW$Dry.Weigh.g.Time1 - data_PvarBW$Dry.Weigh.g.Time0)/data_PvarBW$Dry.Weigh.g.Time0)*100)/data_PvarBW$Days.Time1 #calculate growth rate for time1
data_PvarBW$time2.growth <- (((data_PvarBW$Dry.Weigh.g.Time2 - data_PvarBW$Dry.Weigh.g.Time1)/data_PvarBW$Dry.Weigh.g.Time1)*100)/data_PvarBW$Days.Time2 #calculate growth rate for time2

data_PvarBW$time1.growth[data_PvarBW$time1.growth < 0] <- 0 # neg values to 0 (chipped frags)
data_PvarBW$time2.growth[data_PvarBW$time2.growth < 0] <- 0 


# mean, standard dev, and sem
time1_PvarBW <-ddply(data_PvarBW, .( Treatment), summarize, #For each subset of a data frame, apply function then combine results into a data frame.
              N=length(na.omit(time1.growth)), # the sample size of the light column summarized by Tank number
              mean = (mean(time1.growth, na.rm=TRUE)) ,  #take the average of the light column summarized by Tank number
              sd = sd(time1.growth, na.rm=TRUE) ,# the standard deviation of the temp column summarized by Tank number
              sem = (sd(time1.growth,na.rm=T)/sqrt(N))) #calculate the SEM as the sd/sqrt of the count or data length
time1_PvarBW$Time <- "Time1"  # this puts these data in our "time1" dataframe

time2_PvarBW <-ddply(data_PvarBW, .( Treatment), summarize, #For each subset of a data frame, apply function then combine results into a data frame.
              N=length(na.omit(time2.growth)), # the sample size of the light column summarized by Tank number
              mean = (mean(time2.growth, na.rm=TRUE)) ,  #take the average of the light column summarized by Tank number
              sd = sd(time2.growth, na.rm=TRUE) ,# the standard deviation of the temp column summarized by Tank number
              sem = (sd(time2.growth,na.rm=T)/sqrt(N))) #calculate the SEM as the sd/sqrt of the count or data length
time2_PvarBW$Time <- "Time2"  # this puts these data in our "time1" dataframe

#save this dataframe as "G"
G_PvarA <- rbind(time1_PvarBW[1,],time2_PvarBW[1,])# Greate a new column that we rename each row by treatment
G_PvarH <- rbind(time1_PvarBW[2,],time2_PvarBW[2,])# Greate a new column that we rename each row by treatment
G_Pvar<-rbind(G_PvarA, G_PvarH)
# Stress.Recovery column 
 # make new column that renames "time" to Stress or Recovery
G_Pvar$newtime = ifelse(G_Pvar$Time == "Time1", "Post-Stress", "Post-Recovery")
G_Pvar <- within(G_Pvar, newtime <- factor(newtime, levels = c("Post-Stress", "Post-Recovery"))) # Set order of levels so plots in right order



Pvar_BW_plot<-ggplot(data=G_Pvar, aes(x=newtime, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, .7)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("% Change Dry.Weigh.g"))) +
  ggtitle("Pavona varians")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.25));Pvar_BW_plot

```
P acuta
```{r}
BWCalc_Pacu <- function(StandardAir= 39.092, Temp, BW, CoralDensity = 1.78){
  #Parameters---------------------------------------------------------------
  # StandardAir is the weight of the Standard in air (Default set at 39.092 grams weighed on balance)
  # Temp is the temperature of the water during the coral measurement
  # BW is the buoywant weight of the coral
  # CoralDensity <- 2.03 #g cm-3, set aragonite density for Montipora from literature 
  # Montipora Arag = 2.03 g cm-3 average from table 2 in Anthony 2003 Functional Ecology 17:246-259
  
  # Porites asteroides (T Hughes 1987 1.69 at 5m) (KB got 1.82)
  
  # below: Medellin-Maldonado et al 2016
  # Pavona varians 1.04 +-0.07
  # panama 8.5380Â° N, 80.7821Â° W, 
  # coasta rica 9.7489Â° N, 83.7534Â° W
  # mexico la entrega 15.7424Â° N, 96.1282Â° W
  # hawaii 19.8968Â° N, 155.5828Â° W
  # Pocillopora 1.78 +- 0.31
  
  
  # You can change the density to literatrue values for the specific coral species of interest in the function. 
  
  #Calculation------------------------------------------------------------
  #Step 1: correct the Standard weights for temperature
  # StandardFresh is the weight of the Standard in  fresh water at temp x
  # StandardSalt is the weight of the Standard in salt water at temp x
  
  # This is based on a calibration curve for Standards weighed in fresh and salt water at many temps
  StandardFresh <- StandardFreshModel$coef[-1]*Temp + StandardFreshModel$coef[1] 
  StandardSalt <- StandardSaltModel$coef[-1]*Temp + StandardSaltModel$coef[1] 
  
  # Step 2: Use weight in air and freshwater of the glass Standard to calculate
  # the density of the Standard (Davies eq. 4)
  FreshDensity <- 1 #Fresh water has a density of 1 g/cm3
  StandardDensity <- FreshDensity/(1-(StandardFresh/StandardAir)) 
  
  # Step 3: Calculate the density of seawater using the density of the Standard
  # (Spencer Davies eq. 3)
  SWDensity <- StandardDensity*(1-(StandardSalt/StandardAir))
  
  # Step 4: Calculate the dry weight of the coral (Davies eq. 1)
  CoralWeight <- BW_Pacu/(1-(SWDensity/CoralDensity))
  
  return(CoralWeight) #returns coral dry weights in g
}

Pacu_BW$Dry.Weigh.g <- BWCalc_Pacu(BW=BW_Pacu, Temp=Temp_Pacu) #use function to calculate dry weight
hist(Pacu_BW$Dry.Weigh.g) #not normal
Pacu_BW$Dry.Weigh.g_log<-log10(Pacu_BW$Dry.Weigh.g+1)
hist(Pacu_BW$Dry.Weigh.g_log) #better still tail
boxplot(Pacu_BW$Dry.Weigh.g) #examine data

# below, we are getting rid of columns we don't need anymore, and reorganizing the data by Fragment and clade/color and treatment. This way our time points wil be in the same row)
data_PacuBW <- reshape(Pacu_BW, timevar = "TimePoint", drop = c("Mass.g","Temp.C", "Date","Tank", "X",  "Parent.ID", "Species","Time.Point", "Tank.num"), idvar=c("Frag.ID", "Treatment"), direction="wide")

# This calculates the growth rate, if you look it takes Timepoint1 subtracts timepoint0 and divides by timepoint 0, times 100
data_PacuBW$time1.growth <- (((data_PacuBW$Dry.Weigh.g.Time1 - data_PacuBW$Dry.Weigh.g.Time0)/data_PacuBW$Dry.Weigh.g.Time0)*100)/data_PacuBW$Days.Time1 #calculate growth rate for time1
data_PacuBW$time2.growth <- (((data_PacuBW$Dry.Weigh.g.Time2 - data_PacuBW$Dry.Weigh.g.Time1)/data_PacuBW$Dry.Weigh.g.Time1)*100)/data_PacuBW$Days.Time2 #calculate growth rate for time2
data_PacuBW$time1.growth[data_PacuBW$time1.growth < 0] <- 0 # neg values to 0 (chipped frags)
data_PacuBW$time2.growth[data_PacuBW$time2.growth < 0] <- 0 



# mean, standard dev, and sem
time1_PacuBW <-ddply(data_PacuBW, .( Treatment), summarize, #For each subset of a data frame, apply function then combine results into a data frame.
              N=length(na.omit(time1.growth)), # the sample size of the light column summarized by Tank number
              mean = (mean(time1.growth, na.rm=TRUE)) ,  #take the average of the light column summarized by Tank number
              sd = sd(time1.growth, na.rm=TRUE) ,# the standard deviation of the temp column summarized by Tank number
              sem = (sd(time1.growth,na.rm=T)/sqrt(N))) #calculate the SEM as the sd/sqrt of the count or data length
time1_PacuBW$Time <- "Time1"  # this puts these data in our "time1" dataframe

time2_PacuBW <-ddply(data_PacuBW, .( Treatment), summarize, #For each subset of a data frame, apply function then combine results into a data frame.
              N=length(na.omit(time2.growth)), # the sample size of the light column summarized by Tank number
              mean = (mean(time2.growth, na.rm=TRUE)) ,  #take the average of the light column summarized by Tank number
              sd = sd(time2.growth, na.rm=TRUE) ,# the standard deviation of the temp column summarized by Tank number
              sem = (sd(time2.growth,na.rm=T)/sqrt(N))) #calculate the SEM as the sd/sqrt of the count or data length
time2_PacuBW$Time <- "Time2"  # this puts these data in our "time1" dataframe

#save this dataframe as "G"
G_PacuA <- rbind(time1_PacuBW[1,],time2_PacuBW[1,])# Greate a new column that we rename each row by treatment
G_PacuH <- rbind(time1_PacuBW[2,],time2_PacuBW[2,])# Greate a new column that we rename each row by treatment
G_Pacu<-rbind(G_PacuA, G_PacuH)
# Stress.Recovery column 
 # make new column that renames "time" to Stress or Recovery
G_Pacu$newtime = ifelse(G_Pacu$Time == "Time1", "Post-Stress", "Post-Recovery")
G_Pacu <- within(G_Pacu, newtime <- factor(newtime, levels = c("Post-Stress", "Post-Recovery"))) # Set order of levels so plots in right order



Pacu_BW_plot<-ggplot(data=G_Pacu, aes(x=newtime, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, .7)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste(" "))) + 
  ggtitle("Pocillopora acuta")+ 
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.25));Pacu_BW_plot
 
```
Plots together
```{r}
###### All plots together 
Mcap_BW_plot<-Mcap_BW_plot+ theme(legend.position = "none") #remove legend
Pcomp_BW_plot<-Pcomp_BW_plot+ theme(legend.position = "none") #remove legend
Pvar_BW_plot<-Pvar_BW_plot+ theme(legend.position = "none") #remove legend
Pacu_BW_plot<-Pacu_BW_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_BW_plot, Pcomp_BW_plot, Pvar_BW_plot,Pacu_BW_plot, nrow = 2)
``` 

stats
```{r}   
Growth<- rbind(data_McapBW, data_PcompBW, data_PvarBW, data_PacuBW)
GrowthKeep<- c("Frag.ID","time1.growth","time2.growth", "Treatment")
Growth <- Growth[GrowthKeep]

Grow_m <- melt(Growth, id=c("Frag.ID","Treatment")) #melt
names(Grow_m)[names(Grow_m) == "value"] <- "percent.change"
names(Grow_m)[names(Grow_m) == "variable"] <- "Time.point"

Growth2<-merge(Grow_m, meta) #merge metadata
GrowthKeep2<- c("Frag.ID","Time.point", "Treatment", "Tank.num","Parent.ID","Species","percent.change")
Growth3 <- Growth2[GrowthKeep2]
Growth3<-na.omit(Growth3)

hist(Growth3$percent.change) 
Growth3$percent.change_log<-log10(Growth3$percent.change+1)
hist(Growth3$percent.change_log) #normal 

BW_model<-lmer(percent.change_log~Time.point*Treatment*Species+(1|Parent.ID)+(1|Tank.num), Growth3)
anova(BW_model) 
qqPlot(residuals(BW_model)) 

  emm<-emmeans(BW_model,  ~ Time.point*Treatment|Species, adjust="Tukey")
cld(emm)
pairs(emm)
 

# make DF for BW for Physio, Cast so 2 columns 

Growth.Percent.day<-merge(Growth, meta, all=T)

```



#Chl-a
dfs: Slurry=airbrush, SA=wax dip, 
```{r}   

 # Plate 1

Chla_plate1<- read.csv("Data/Chla_plate1.csv")

#merge 
ChlaData1 <- merge(meta, Chla_plate1)
ChlaData1 <- merge(ChlaData1, Slurry)
ChlaData1 <- merge(ChlaData1, SA)
ChlaData1$Parent.ID<-as.factor(as.character(ChlaData1$Parent.ID))
ChlaData1$Tank.num<-as.factor(as.character(ChlaData1$Tank.num))
ChlaData1$Surface.area<-as.numeric(as.character(ChlaData1$Surface.area))


ChlaData1$totChla1<-ChlaData1$Rep1*ChlaData1$TotalVolume
ChlaData1$chlaSA1 <- ChlaData1$totChla1/ChlaData1$Surface.area

ChlaData1$totChla2<-ChlaData1$Rep2*ChlaData1$TotalVolume
ChlaData1$chlaSA2 <- ChlaData1$totChla2/ChlaData1$Surface.area

plate1<-plot(ChlaData1$chlaSA1,ChlaData1$chlaSA2)
ChlaData1<-ChlaData1[!(ChlaData1$Frag.ID=="314"),] #redid plate 8

# Plate 2
Chla_plate2<- read.csv("Data/Chla_plate2.csv")

#merge 
ChlaData2 <- merge(meta, Chla_plate2)
ChlaData2 <- merge(ChlaData2, Slurry)
ChlaData2 <- merge(ChlaData2, SA)
ChlaData2$Parent.ID<-as.factor(as.character(ChlaData2$Parent.ID))
ChlaData2$Tank.num<-as.factor(as.character(ChlaData2$Tank.num))
ChlaData2$Surface.area<-as.numeric(as.character(ChlaData2$Surface.area))

ChlaData2$totChla1<-ChlaData2$Rep1*ChlaData2$TotalVolume
ChlaData2$chlaSA1 <- ChlaData2$totChla1/ChlaData2$Surface.area

ChlaData2$totChla2<-ChlaData2$Rep2*ChlaData2$TotalVolume
ChlaData2$chlaSA2 <- ChlaData2$totChla2/ChlaData2$Surface.area

plate2<-plot(ChlaData2$chlaSA1,ChlaData2$chlaSA2)
ChlaData2<-ChlaData2[!(ChlaData2$Frag.ID=="3476"),] #redid plate 8
ChlaData2<-ChlaData2[!(ChlaData2$Frag.ID=="N211"),] #redid plate 8

# Plate 3
Chla_plate3<- read.csv("Data/Chla_plate3.csv")

#merge 
ChlaData3 <- merge(meta, Chla_plate3)
ChlaData3 <- merge(ChlaData3, Slurry)
ChlaData3 <- merge(ChlaData3, SA)
ChlaData3$Parent.ID<-as.factor(as.character(ChlaData3$Parent.ID))
ChlaData3$Tank.num<-as.factor(as.character(ChlaData3$Tank.num))
ChlaData3$Surface.area<-as.numeric(as.character(ChlaData3$Surface.area))

ChlaData3$totChla1<-ChlaData3$Rep1*ChlaData3$TotalVolume
ChlaData3$chlaSA1 <- ChlaData3$totChla1/ChlaData3$Surface.area

ChlaData3$totChla2<-ChlaData3$Rep2*ChlaData3$TotalVolume
ChlaData3$chlaSA2 <- ChlaData3$totChla2/ChlaData3$Surface.area

plate3<-plot(ChlaData3$chlaSA1,ChlaData3$chlaSA2) #check
ChlaData3<-ChlaData3[!(ChlaData3$Frag.ID=="N158"),] #redid plate 8

# Plate 4 
Chla_plate4<- read.csv("Data/test_plate4.csv")

#merge 
ChlaData4 <- merge(meta, Chla_plate4)
ChlaData4 <- merge(ChlaData4, Slurry)
ChlaData4 <- merge(ChlaData4, SA)
ChlaData4$Parent.ID<-as.factor(as.character(ChlaData4$Parent.ID))
ChlaData4$Tank.num<-as.factor(as.character(ChlaData4$Tank.num))
ChlaData4$Surface.area<-as.numeric(as.character(ChlaData4$Surface.area))


ChlaData4$totChla1<-ChlaData4$Rep1*ChlaData4$TotalVolume
ChlaData4$chlaSA1 <- ChlaData4$totChla1/ChlaData4$Surface.area

ChlaData4$totChla2<-ChlaData4$Rep2*ChlaData4$TotalVolume
ChlaData4$chlaSA2 <- ChlaData4$totChla2/ChlaData4$Surface.area

plate4<-plot(ChlaData4$chlaSA1,ChlaData4$chlaSA2)
ChlaData4<-ChlaData4[!(ChlaData4$Frag.ID=="N30"),] #redid plate 8

# Plate 5 20190319
Chla_plate5<- read.csv("Data/Chla_20190319_plate5.csv")

#merge 
ChlaData5 <- merge(meta, Chla_plate5)
ChlaData5 <- merge(ChlaData5, Slurry)
ChlaData5 <- merge(ChlaData5, SA)
ChlaData5$Parent.ID<-as.factor(as.character(ChlaData5$Parent.ID))
ChlaData5$Tank.num<-as.factor(as.character(ChlaData5$Tank.num))
ChlaData5$Surface.area<-as.numeric(as.character(ChlaData5$Surface.area))


ChlaData5$totChla1<-ChlaData5$Rep1*ChlaData5$TotalVolume
ChlaData5$chlaSA1 <- ChlaData5$totChla1/ChlaData5$Surface.area

ChlaData5$totChla2<-ChlaData5$Rep2*ChlaData5$TotalVolume
ChlaData5$chlaSA2 <- ChlaData5$totChla2/ChlaData5$Surface.area

plate5<-plot(ChlaData5$chlaSA1,ChlaData5$chlaSA2) #some not close, check 



# Plate 6 20190321
Chla_plate6<- read.csv("Data/Chla_plate6.csv")

#merge 
ChlaData6 <- merge(meta, Chla_plate6)
ChlaData6 <- merge(ChlaData6, Slurry)
ChlaData6 <- merge(ChlaData6, SA)
ChlaData6$Parent.ID<-as.factor(as.character(ChlaData6$Parent.ID))
ChlaData6$Tank.num<-as.factor(as.character(ChlaData6$Tank.num))
ChlaData6$Surface.area<-as.numeric(as.character(ChlaData6$Surface.area))

 
ChlaData6$totChla1<-ChlaData6$Rep1*ChlaData6$TotalVolume
ChlaData6$chlaSA1 <- ChlaData6$totChla1/ChlaData6$Surface.area

ChlaData6$totChla2<-ChlaData6$Rep2*ChlaData6$TotalVolume
ChlaData6$chlaSA2 <- ChlaData6$totChla2/ChlaData6$Surface.area

plate6<-plot(ChlaData6$chlaSA1,ChlaData6$chlaSA2) #some not close, check 
ChlaData6<-ChlaData6[!(ChlaData6$Frag.ID=="3484"),] #redid plate 8

# Plate 8 20190321
Chla_plate8<- read.csv("Data/Chla_plate8.csv")

#merge 
ChlaData8 <- merge(meta, Chla_plate8)
ChlaData8 <- merge(ChlaData8, Slurry)
ChlaData8 <- merge(ChlaData8, SA)
ChlaData8$Parent.ID<-as.factor(as.character(ChlaData8$Parent.ID))
ChlaData8$Tank.num<-as.factor(as.character(ChlaData8$Tank.num))
ChlaData8$Surface.area<-as.numeric(as.character(ChlaData8$Surface.area))

 
ChlaData8$totChla1<-ChlaData8$Rep1*ChlaData8$TotalVolume
ChlaData8$chlaSA1 <- ChlaData8$totChla1/ChlaData8$Surface.area

ChlaData8$totChla2<-ChlaData8$Rep2*ChlaData8$TotalVolume
ChlaData8$chlaSA2 <- ChlaData8$totChla2/ChlaData8$Surface.area

plate8<-plot(ChlaData8$chlaSA1,ChlaData8$chlaSA2) #some not close, check 
  
Chla<-rbind(ChlaData1, ChlaData2, ChlaData3, ChlaData4, ChlaData5, ChlaData6,ChlaData8)

chlatest<-merge(meta, Chla, all = T) 
  
```

means for plots
```{r}  

chlatest<-merge(meta, Chla, all.x = T)

duplicated(Chla) #look for duplicate entries
Chla<-unique(Chla) #remove any duplicate entries 

Chla$Chla_RepDiff<-Chla$chlaSA1-Chla$chlaSA2
Chla$ChlaSA<-(Chla$chlaSA1+Chla$chlaSA2)/2
checkChla<-merge(Chla,meta, all.y=T)
duplicated(checkChla)
#write.csv(checkChla,"checkChla.csv")

#2285, 3491, N141 Remove sample where reps are >2 apart
#Chla<-Chla[!(Chla$Frag.ID=="N138"),]


#make slightly neg values 0 
Chla$ChlaSA[Chla$ChlaSA < 0] <- 0 

# 
Chla_mean <- ddply(Chla, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                   N    = length(ChlaSA[!is.na(ChlaSA)]), #calculate the length of the data frame, excluding NAâ€™s *this part is important*
                   mean = mean(ChlaSA, na.rm=TRUE), #calculate mean of response variable, removing NA's
                   sd   = sd(ChlaSA, na.rm=TRUE), #calculate standard deviation
                   se   = sd / sqrt(N), #calculate standard error
                   max = max(ChlaSA, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Chla_mean #display table

# divide by species
Mcap_chla<-subset(Chla_mean, Species =="Montipora_capitata")
Pcomp_chla<-subset(Chla_mean, Species =="Porites_compressa")
Pvar_chla<-subset(Chla_mean, Species =="Pavona_varians")
Pacu_chla<-subset(Chla_mean, Species =="Pocillopora_acuta")

#Mcap plot

Mcap_Chla_plot<-ggplot(data=Mcap_chla, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, 8)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("(Âµg chl-a cm" ^ "-2"*")"))) +
  ggtitle("Montipora capitata")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Mcap_Chla_plot

#Pcomp plot

Pcomp_Chla_plot<-ggplot(data=Pcomp_chla, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, 8)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("(Âµg chl-a cm" ^ "-2"*")"))) +
  ggtitle("Porites compressa")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pcomp_Chla_plot

#Pvar plot

Pvar_Chla_plot<-ggplot(data=Pvar_chla, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, 8)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("(Âµg chl-a cm" ^ "-2"*")"))) +
  ggtitle("Pavona varians")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pvar_Chla_plot


#Pacu plot

Pacu_Chla_plot<-ggplot(data=Pacu_chla, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, 5)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("(Âµg chl-a cm" ^ "-2"*")"))) +
  ggtitle("Pocillopora acuta")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pacu_Chla_plot



```

Plots together
```{r}
###### All plots together
Mcap_Chla_plot<-Mcap_Chla_plot+ theme(legend.position = "none") #remove legend
Pcomp_Chla_plot<-Pcomp_Chla_plot+ theme(legend.position = "none") #remove legend
Pvar_Chla_plot<-Pvar_Chla_plot+ theme(legend.position = "none") #remove legend
Pacu_Chla_plot<-Pacu_Chla_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_Chla_plot, Pcomp_Chla_plot, Pvar_Chla_plot,Pacu_Chla_plot, nrow = 2)
```
 
stats  
```{r}  
 
hist (Chla$ChlaSA) 
Chla$ChlaSA_log<-log10(Chla$ChlaSA+1)
hist(Chla$ChlaSA_log) #pos skew this ok?
Chla$ChlaSA_sqrt<-sqrt(Chla$ChlaSA) 
hist(Chla$ChlaSA_sqrt) #nope

Chla$ChlaSA_cube<-Chla$ChlaSA^(1/3)
hist(Chla$ChlaSA_cube) #better but no
shapiro.test(Chla$ChlaSA_cube)

Chla$ChlaSA_quart<-Chla$ChlaSA^(1/4)
hist(Chla$ChlaSA_quart) #best
shapiro.test(Chla$ChlaSA_quart) # p=0.08

Chla_model<-lmer(ChlaSA_quart~Time.Point*Treatment*Species+(1|Parent.ID)+(1|Tank.num), Chla)
anova(Chla_model) 
qqPlot(residuals(Chla_model)) 


emm<-emmeans(Chla_model,  ~ Time.Point*Treatment|Species, adjust="Tukey")
cld(emm)
pairs(emm)

```


#PAM 
```{r}     
#Load PAM data
Time0 <- read.csv("Data/PAM_20170320.csv", header=T) #load data
Time0<-merge(Time0, meta)
Time1 <- read.csv("Data/PAM_20170321.csv", header=T) #load data
Time1<-merge(Time1, meta)
Time2 <- read.csv("Data/PAM_20170322.csv", header=T) #load data
Time2<-merge(Time2, meta)
Time3 <- read.csv("Data/PAM_20170323.csv", header=T) #load data
Time3<-merge(Time3, meta)
#Time3b <- read.csv("Data/PAM_new_20170323b.csv", header=T) #load data
Time4 <- read.csv("Data/PAM_20170406.csv", header=T) #load data
Time4<-merge(Time4, meta)
Time5 <- read.csv("Data/PAM_20170413.csv", header=T) #load data
Time5<-merge(Time5, meta)
Time6 <- read.csv("Data/PAM_6test.csv", header=T) #load data had to remove rows with no data bc wouldn't upload
Time6<-merge(Time6, meta)
Time7 <- read.csv("Data/PAM_20170427.csv", header=T) #load data
Time7<-merge(Time7, meta)
Time8 <- read.csv("Data/PAM_20170504.csv", header=T) #load data
Time8<-merge(Time8, meta)
Time9 <- read.csv("Data/PAM_20170511.csv", header=T) #load data
Time9<-merge(Time9, meta)
Time10 <- read.csv("Data/PAM_20170517.csv", header=T) #load data
Time10<-merge(Time10, meta)

# combine all data files
PAMdata<-rbind(Time0, Time1, Time2, Time3, Time4, Time5, Time6, Time7, Time8, Time9, Time10)
# F0/FM: change F0, yield, and FM to numeric
PAMdata$F0<-as.numeric(as.character(PAMdata$F0))
PAMdata$FM<-as.numeric(as.character(PAMdata$FM))
PAMdata$Tank<-as.factor(as.character(PAMdata$Tank))
PAMdata$Tank.num<-as.factor(as.character(PAMdata$Tank.num))

# Date - change date to R format and then to numeric
PAMdata$Date <- as.Date(PAMdata$Date, format="%e-%b-%y")
#PAMdata$Date[PAMdata$Date<"2017-03-24"]<-"2017-03-20" # make all T0 same starte date to get average
PAMdata$Days <- as.numeric(PAMdata$Date - as.Date("2017-03-20"))
PAMdata$Days[PAMdata$Days<4]<- 0  #If you want to make T0 all the same start time do this


# make yield ratio
PAMdata <- transform(PAMdata, Yield = PAMdata$Yield/1000)
str(PAMdata) # check

#check yield calculations versus records and build dataframe
checks <- (PAMdata[,(1:9)]) #add samples to dataframe
checks$Yield.Check <- PAMdata$Yield - ((PAMdata$FM-PAMdata$F0)/PAMdata$FM)
checks$checks <- ifelse(abs(checks$Yield.Check-checks$Yield) < 1, TRUE, FALSE) # check to see if Yield matches check
checks[checks$checks==FALSE,]#pull out FALSE

# check for outliers
#boxplot(checks$Yield.Check, main="Times") #check F0r outliers

#Check notes manulaly and remove data points as needed
#check for duplicate rows
DUP<-duplicated(PAMdata) #look for duplicate entries
PAMdata<-unique(PAMdata) #remove any duplicate entries 
#write.csv(PAMdata,"PAMdatacheck.csv")

# if NA in YIELD remove
PAMdata<-PAMdata[!is.na(PAMdata$Yield), ]




# Many frags were too low to PAM (no signal) but corals were alive. Change fv.fm to 0.1, a minimum baseline to indicate the lack of photosynthetic yield. 0.1 is below the threshold of any PAM signal in a readable coral 

LowSig<-read.csv("Data/PAM_lowSignalSamps.csv") #read in df with low sig samps

#remove samples in 'extra;
#testmer<-merge(meta, LowSig, all.y=T)
LowSig$Date <- as.Date(LowSig$Date, format="%e-%b-%y") #update date
LowSig<-LowSig[!(LowSig$REMOVE=="REMOVE"),] #remove the REMOVES

PAMdata2<- left_join(PAMdata, LowSig, by=c("Date","Frag.ID")) %>%  #merge df based on frag * date
  rowwise() 
# Change Yields in lowsigs to 0.1
PAMdata2 <- within(PAMdata2, Yield[Low.Sig == 'Yes' ] <- '0.1')

PAMdata2$Yield<-as.numeric(PAMdata2$Yield)
PAMdata2$Parent.ID<-as.factor(as.character(PAMdata2$Parent.ID))


```
Stats
```{r}    
#remove notes

PAMdata2_stats<-PAMdata2[,c("Frag.ID","Date","Yield","Parent.ID","Treatment","Species","Tank.num","Days")] # keep only columns you need
PAMdata2_stats$Days<-as.factor(as.character(PAMdata2_stats$Days))
PAMdata2_stats$Yield<-as.numeric(PAMdata2_stats$Yield)
PAMdata2_stats$Parent.ID<-as.factor(as.character(PAMdata2_stats$Parent.ID))

# Historgam
hist(PAMdata2_stats$Yield)

#transform
PAMdata2_stats$Yield_sq<-(PAMdata2_stats$Yield)^2
hist(PAMdata2_stats$Yield_sq)

PAMdata2_stats$Yield_cu<-(PAMdata2_stats$Yield)^3
hist(PAMdata2_stats$Yield_cu) #beter


#lmer with ^3
PAMdata2_stats_model<-lmer(Yield_cu~Days*Treatment*Species+(1|Parent.ID:Frag.ID)+(1|Tank.num), PAMdata2_stats)
anova(PAMdata2_stats_model) 
qqPlot(residuals(PAMdata2_stats_model)) 

emm<-emmeans(PAMdata2_stats_model,  ~ Days*Treatment|Species, adjust="Tukey")
cld(emm)
pairs(emm)



#is the data still not normal by species

Mcap_PAM<-subset(PAMdata2_stats, Species=="Montipora_capitata")
hist(Mcap_PAM$Yield_cu)
Mcap_PAMdata2_stats_model<-lmer(Yield_cu~Days*Treatment+(1|Parent.ID)+(1|Tank.num), Mcap_PAM)
anova(Mcap_PAMdata2_stats_model) 
qqPlot(residuals(Mcap_PAMdata2_stats_model))

Pcomp_PAM<-subset(PAMdata2_stats, Species=="Porites_compressa")
hist(Pcomp_PAM$Yield_cu)
Pcomp_PAMdata2_stats_model<-lmer(Yield_cu~Days*Treatment+(1|Parent.ID)+(1|Tank.num), Pcomp_PAM)
anova(Pcomp_PAMdata2_stats_model) 
qqPlot(residuals(Pcomp_PAMdata2_stats_model))

Pvar_PAM<-subset(PAMdata2_stats, Species=="Pavona_varians") #normal AF
hist(Pvar_PAM$Yield_cu)
Pvar_PAMdata2_stats_model<-lmer(Yield_cu~Days*Treatment+(1|Parent.ID)+(1|Tank.num), Pvar_PAM)
anova(Pvar_PAMdata2_stats_model) 
qqPlot(residuals(Pvar_PAMdata2_stats_model))

Pacu_PAM<-subset(PAMdata2_stats, Species=="Pocillopora_acuta")
hist(Pacu_PAM$Yield_cu)
Pacu_PAMdata2_stats_model<-lmer(Yield_cu~Days*Treatment+(1|Parent.ID)+(1|Tank.num), Pacu_PAM)
anova(Pacu_PAMdata2_stats_model) 
qqPlot(residuals(Pacu_PAMdata2_stats_model))


#set up df for physio\
  keepPAM<-c("Frag.ID", "Parent.ID","Species","Tank.num","Treatment","Days","Yield_cu") #keep cols you need
PAMdata2_stats2<-PAMdata2_stats[keepPAM] #keep cols you want
PAMdata2_stats2$Parent.ID<-as.factor(as.character(PAMdata2_stats2$Parent.ID))
#PAMdata2_physio<-dcast(PAMdata2_stats, Frag.ID+Parent.ID+Species+Tank.num+Treatment~Days,fun.aggregate = mean) #cast

# names(PAMdata2_physio)[names(PAMdata2_physio) == "0"] <- "T0.PAM"
# names(PAMdata2_physio)[names(PAMdata2_physio) == "24"] <- "T1.PAM"
# names(PAMdata2_physio)[names(PAMdata2_physio) == "52"] <- "TF.PAM"




```
Plots
```{r}  
PAMdata2_mean <- ddply(PAMdata2, c("Treatment", "Species", "Days"), summarise, #summarize cell counts
                   N    = length(Yield[!is.na(Yield)]), #calculate the length of the data frame, excluding NAâ€™s *this part is important*
                   mean = mean(Yield, na.rm=TRUE), #calculate mean of response variable, removing NA's
                   sd   = sd(Yield, na.rm=TRUE), #calculate standard deviation
                   se   = sd / sqrt(N), #calculate standard error
                   max = max(Yield, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
PAMdata2_mean #display table

# divide by species
Mcap_PAM_mean<-subset(PAMdata2_mean, Species =="Montipora_capitata")
Pcomp_PAM_mean<-subset(PAMdata2_mean, Species =="Porites_compressa")
Pvar_PAM_mean<-subset(PAMdata2_mean, Species =="Pavona_varians")
Pacu_PAM_mean<-subset(PAMdata2_mean, Species =="Pocillopora_acuta")

#Mcap plot

Mcap_PAM_plot<-ggplot(data=Mcap_PAM_mean, aes(x=Days, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, 1)+
   scale_x_continuous(name="Days", breaks=c(0,10,20,30,40,50,60))+

  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("fv/fm"))) +
  ggtitle("Mcap PAM")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Mcap_PAM_plot

#Pcomp plot

Pcomp_PAM_plot<-ggplot(data=Pcomp_PAM_mean, aes(x=Days, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, 1)+
   scale_x_continuous(name="Days", breaks=c(0,10,20,30,40,50,60))+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("fv/fm"))) +
  ggtitle("Pcomp PAM")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pcomp_PAM_plot

#Pvar plot

Pvar_PAM_plot<-ggplot(data=Pvar_PAM_mean, aes(x=Days, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, 1)+
   scale_x_continuous(name="Days", breaks=c(0,10,20,30,40,50,60))+

  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("fv/fm"))) +
  ggtitle("Pvar PAM")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pvar_PAM_plot

#Pacu plot
Pacu_PAM_mean<-Pacu_PAM_mean[!(Pacu_PAM_mean$N<5),] #remove time points if n<5

Pacu_PAM_plot<-ggplot(data=Pacu_PAM_mean, aes(x=Days, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, 1)+
 scale_x_continuous(name="Days", breaks=c(0,10,20,30,40,50,60))+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("fv/fm"))) +
  ggtitle("Pacu PAM")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pacu_PAM_plot

```
Plots together
```{r}
###### All plots together
Mcap_PAM_plot<-Mcap_PAM_plot+ theme(legend.position = "none") #remove legend
Pcomp_PAM_plot<-Pcomp_PAM_plot+ theme(legend.position = "none") #remove legend
Pvar_PAM_plot<-Pvar_PAM_plot+ theme(legend.position = "none") #remove legend
Pacu_PAM_plot<-Pacu_PAM_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_PAM_plot, Pcomp_PAM_plot, Pvar_PAM_plot,Pacu_PAM_plot, nrow = 2)
```

#Protein
```{r}   
Protein <- read.csv("Data/Protein_TT17_20190214.csv")
prottest<-merge(meta, Protein, all=T)
keepPr<-c("Frag.ID", "mg.prot.ml") #keep cols you need
Protein<-Protein[keepPr]

Protein<-merge(Protein, Slurry) #merge with slurry
Protein<-merge(Protein, SA) # merge with SA
Protein$Surface.area<-as.numeric(as.character(Protein$Surface.area))
Protein<-merge(Protein, meta) #merge with metadata
Protein$Parent.ID<-as.factor(as.character(Protein$Parent.ID))
Protein$Tank.num<-as.factor(as.character(Protein$Tank.num))

# calculate total prot per total vol of slurry
Protein$prot.ml<- Protein$mg.prot.ml*Protein$TotalVolume

# create a new column to calculate per surface area (mg/mm2)
Protein$mg.mm2 <- Protein$prot.ml/Protein$Surface.area

PRotcheck<-merge(Protein, meta, all=T)
#write.csv(PRotcheck, "Proteincheck.csv")

#calc means
Protein_mean <- ddply(Protein, c("Treatment", "Species", "Time.Point"), summarise, #summarize prot
                   N    = length(mg.mm2[!is.na(mg.mm2)]), #calculate the length of the data frame, excluding NAâ€™s *this part is important*
                   mean = mean(mg.mm2, na.rm=TRUE), #calculate mean of response variable, removing NA's
                   sd   = sd(mg.mm2, na.rm=TRUE), #calculate standard deviation
                   se   = sd / sqrt(N), #calculate standard error
                   max = max(mg.mm2, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
Protein_mean

#stats

hist(Protein$mg.mm2)
Protein$mg.mm2_log<-log10(Protein$mg.mm2+1)
hist(Protein$mg.mm2_log)
Protein$mg.mm2_sqrt<-sqrt(Protein$mg.mm2)
hist(Protein$mg.mm2_sqrt) #better

 
```
Stats
```{r}
Protein_model<-lmer(mg.mm2_sqrt~Time.Point*Treatment*Species+(1|Parent.ID:Frag.ID)+(1|Tank.num), Protein)
anova(Protein_model) 
qqPlot(residuals(Protein_model)) #not awful

  emm<-emmeans(Protein_model,  ~ Time.Point*Treatment|Species, adjust="Tukey")
cld(emm)
pairs(emm)

  emm<-emmeans(Protein_model,  ~ Species*Treatment|Time.Point, adjust="Tukey")
cld(emm)
pairs(emm)


```
plots
```{r}
Mcap_Protein_mean<-subset(Protein_mean, Species=="Montipora_capitata")
Pcomp_Protein_mean<-subset(Protein_mean, Species=="Porites_compressa")
Pvar_Protein_mean<-subset(Protein_mean, Species=="Pavona_varians")
Pacu_Protein_mean<-subset(Protein_mean, Species=="Pocillopora_acuta")


#Mcap
Mcap_Protein_plot<-ggplot(data=Mcap_Protein_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, .7)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("mg/mm2"))) +
  ggtitle("Montipora capitata")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Mcap_Protein_plot

#Pcomp
Pcomp_Protein_plot<-ggplot(data=Pcomp_Protein_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
     # ylim(0, .7)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("mg/mm2"))) +
  ggtitle("Porites compressa")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pcomp_Protein_plot

#Pvar
Pvar_Protein_plot<-ggplot(data=Pvar_Protein_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
     # ylim(0, .7)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("mg/mm2"))) +
  ggtitle("Pavona varians")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pvar_Protein_plot


#Pacu
Pacu_Protein_plot<-ggplot(data=Pacu_Protein_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = T)+
  geom_point(aes(color=Treatment), size=4, show.legend = T)+  
  geom_line(aes(color=Treatment),linetype="dotted")+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   geom_point(shape = 1,size = 4,colour = "black")+
  xlab("Time Point") + #Label the X Axis
      ylim(0, .7)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border 
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  ylab(expression(paste("mg/mm2"))) +
  ggtitle("Pocillopora acuta")+
  theme(plot.title = element_text(size=20, face = "italic", hjust=0.5));Pacu_Protein_plot


###### All plots together
Mcap_Protein_plot<-Mcap_Protein_plot+ theme(legend.position = "none") #remove legend
Pcomp_Protein_plot<-Pcomp_Protein_plot+ theme(legend.position = "none") #remove legend
Pvar_Protein_plot<-Pvar_Protein_plot+ theme(legend.position = "none") #remove legend
Pacu_Protein_plot<-Pacu_Protein_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_Protein_plot, Pcomp_Protein_plot, Pvar_Protein_plot,Pacu_Protein_plot, nrow = 2)
```







#COMBINE all physio dfs
Set these up so they are by PARENT-TREATMENT-TIMEPOINT and average across frag reps
```{r}    
#biomass: keep Frag.ID and AFDW.cm2_log
Biomass_psy<- Biomass_stats[,c("Frag.ID","Parent.ID" ,"Species","Treatment","Tank.num","Time.Point","AFDWmg.cm2_log")] # keep only columns you need
  #create duplicate T0 for H or A that isn't there for each parent. id
  Biomass_psyT0<-Biomass_psy #copy
  Biomass_psyT0<-subset(Biomass_psyT0, Time.Point=="T0")
  Biomass_psyT0Amb<-Biomass_psyT0
  Biomass_psyT0Amb$Treatment[Biomass_psyT0Amb$Treatment == "High"] <- "Ambient" #change 
  Biomass_psyT0High<-Biomass_psyT0
  Biomass_psyT0High$Treatment[Biomass_psyT0High$Treatment == "Ambient"] <- "High" #change 
  Biomass_psyT02<-rbind(Biomass_psyT0Amb,Biomass_psyT0High) #now oyu have T0 for Amb and High

  Biomass_psy<-Biomass_psy[!(Biomass_psy$Time.Point=="T0"),] #remove T0s
  Biomass_psy<-rbind(Biomass_psy,Biomass_psyT02) # add the full T0 dataset
duplicated(Biomass_psy)
  Biomass_psy$Code<-paste(Biomass_psy$Species,Biomass_psy$Treatment,Biomass_psy$Parent.ID, Biomass_psy$Time.Point)

  Biomass_psy<- Biomass_psy[,c("Code" ,"AFDWmg.cm2_log")] # keep only columns you need
duplicated(Biomass_psy) #check
Biomass_psy<-unique(Biomass_psy) # remove dups

#cell counts:FC_Data df and keep  Frag.ID and zoox.cm2_log
CellCounts_psy<- FC_Data[,c("Frag.ID","Parent.ID" ,"Species","Treatment","Tank.num","Time.Point","zoox.cm2_log")] # keep only columns you need

#create duplicate T0 for H or A that isn't there for each parent. id
  CellCounts_psyT0<-CellCounts_psy #copy
  CellCounts_psyT0<-subset(CellCounts_psyT0, Time.Point=="T0")
  CellCounts_psyT0Amb<-CellCounts_psyT0
  CellCounts_psyT0Amb$Treatment[CellCounts_psyT0Amb$Treatment == "High"] <- "Ambient" #change 
  CellCounts_psyT0High<-CellCounts_psyT0
  CellCounts_psyT0High$Treatment[CellCounts_psyT0High$Treatment == "Ambient"] <- "High" #change 
  CellCounts_psyT02<-rbind(CellCounts_psyT0Amb,CellCounts_psyT0High) #now oyu have T0 for Amb and High

  CellCounts_psy<-CellCounts_psy[!(CellCounts_psy$Time.Point=="T0"),] #remove T0s
  CellCounts_psy<-rbind(CellCounts_psy,CellCounts_psyT02) # add the full T0 dataset

  CellCounts_psy$Code<-paste(CellCounts_psy$Species,CellCounts_psy$Treatment,CellCounts_psy$Parent.ID, CellCounts_psy$Time.Point)

  CellCounts_psy<- CellCounts_psy[,c("Code" ,"zoox.cm2_log")] # keep only columns you need

  duplicated(CellCounts_psy)
  CellCounts_psy<-unique(CellCounts_psy) #keep uniques


#Chla Chla df ChlaSA_quart
Chla_psy<- Chla[,c("Parent.ID" ,"Species","Treatment","Time.Point","ChlaSA_quart")] # keep only columns you need
names(Chla_psy)[names(Chla_psy) == "ChlaSA_quart"] <- "Chla_ug.chla.cm2"

#create duplicate T0 for H or A that isn't there for each parent. id
  Chla_psyT0<-Chla_psy #copy
  Chla_psyT0<-subset(Chla_psyT0, Time.Point=="T0")
  Chla_psyT0Amb<-Chla_psyT0
  Chla_psyT0Amb$Treatment[Chla_psyT0Amb$Treatment == "High"] <- "Ambient" #change 
  Chla_psyT0High<-Chla_psyT0
  Chla_psyT0High$Treatment[Chla_psyT0High$Treatment == "Ambient"] <- "High" #change 
  Chla_psyT02<-rbind(Chla_psyT0Amb,Chla_psyT0High) #now oyu have T0 for Amb and High

  Chla_psy<-Chla_psy[!(Chla_psy$Time.Point=="T0"),] #remove T0s
  Chla_psy<-rbind(Chla_psy,Chla_psyT02) # add the full T0 dataset

  Chla_psy$Code<-paste(Chla_psy$Species,Chla_psy$Treatment,Chla_psy$Parent.ID, Chla_psy$Time.Point)
Chla_psy<- Chla_psy[,c("Code","Chla_ug.chla.cm2")] # keep only columns you need

duplicated(Chla_psy)


#Protein Protein$mg.mm2_sqrt
Protein_psy<- Protein[,c("Parent.ID" ,"Species","Treatment","Time.Point","mg.mm2_sqrt")] # keep only columns you need
names(Protein_psy)[names(Protein_psy) == "mg.mm2_sqrt"] <- "Protein_mg.mm2_sqrt"

#create duplicate T0 for H or A that isn't there for each parent. id
  Protein_psyT0<-Protein_psy #copy
  Protein_psyT0<-subset(Protein_psyT0, Time.Point=="T0")
  Protein_psyAmb<-Protein_psyT0
  Protein_psyAmb$Treatment[Protein_psyAmb$Treatment == "High"] <- "Ambient" #change 
  Protein_psyHigh<-Protein_psyT0
  Protein_psyHigh$Treatment[Protein_psyHigh$Treatment == "Ambient"] <- "High" #change 
  Protein_psy2<-rbind(Protein_psyAmb,Protein_psyHigh) #now oyu have T0 for Amb and High

  Protein_psy<-Protein_psy[!(Protein_psy$Time.Point=="T0"),] #remove T0s
  Protein_psy<-rbind(Protein_psy,Protein_psy2) # add the full T0 dataset

  Protein_psy$Code<-paste(Protein_psy$Species,Protein_psy$Treatment,Protein_psy$Parent.ID, Protein_psy$Time.Point)
Protein_psy<- Protein_psy[,c("Code","Protein_mg.mm2_sqrt")] # keep only columns you need
duplicated(Protein_psy)

#Repeate measures
#Calcification, Growth and percent.change_log do for each time point

Calcification_psy<- Growth3 #copy df

Calcification_psy$Time.point<-as.character(Calcification_psy$Time.point) #change to character to replace
Calcification_psy$Time.point[Calcification_psy$Time.point == "time1.growth"] <- "T1" #change 
Calcification_psy$Time.point[Calcification_psy$Time.point == "time2.growth"] <- "TF" #change 
Calcification_psy$Time.point<-as.factor(Calcification_psy$Time.point)
names(Calcification_psy)[names(Calcification_psy) == "Time.point"] <- "Time.Point" #change name

Calcification_psy$Code<-paste(Calcification_psy$Species,Calcification_psy$Treatment,Calcification_psy$Parent.ID,Calcification_psy$Time.Point) #make code

Calcification_psy<- Calcification_psy[,c("Code","percent.change_log","Parent.ID", "Treatment","Species","Time.Point")] # keep only columns you need
Calcification_psy<-Calcification_psy %>% group_by(Code) %>% mutate(mean=mean(percent.change_log)) # mean by code, this worked
names(Calcification_psy)[names(Calcification_psy) == "mean"] <- "calc.percentchange" #change name
Calcification_psy<-Calcification_psy[,c("Code","Parent.ID", "Treatment","Species","calc.percentchange")] #remove col
Calcification_psy<-unique(Calcification_psy) #remove duplicates, now you have 

Calcification_psy<-merge(metaPhysio, Calcification_psy, all.x = T)
Calcification_psy$calc.percentchange[is.na(Calcification_psy$calc.percentchange)] <- 0 #change T0 growth to 0s


#PAM df PAMdata_stats make for T0, T1, TF
PAM_psy<- PAMdata2_stats2

PAM_psy$Days<-as.character(PAM_psy$Days)
PAM_psy$Days[PAM_psy$Days == "0"] <- "T0" #change 
PAM_psy$Days[PAM_psy$Days == "24"] <- "T1" #change 
PAM_psy$Days[PAM_psy$Days == "58"] <- "TF" #change 

PAM_psy<-PAM_psy[!(PAM_psy$Days=="17"),] #remove other days by name
PAM_psy<-PAM_psy[!(PAM_psy$Days=="31"),] #remove other days by name
PAM_psy<-PAM_psy[!(PAM_psy$Days=="38"),] #remove other days by name
PAM_psy<-PAM_psy[!(PAM_psy$Days=="45"),] #remove other days by name
PAM_psy<-PAM_psy[!(PAM_psy$Days=="52"),] #remove other days by name

#now you need to figure out the sample 'day' for each parent.id, and subset out only those. 
names(PAM_psy)[names(PAM_psy) == "Days"] <- "Time.Point" #change name

PAM_psy$Code<-paste(PAM_psy$Species,PAM_psy$Treatment,PAM_psy$Parent.ID,PAM_psy$Time.Point) #make code
PAM_psy$Code<-as.factor(PAM_psy$Code)

PAM_psy<- PAM_psy[,c("Code","Yield_cu","Parent.ID", "Treatment","Species","Time.Point")] # keep only columns you need
PAM_psy<-PAM_psy %>% group_by(Code) %>% mutate(mean=mean(Yield_cu)) # mean by code, this worked
names(PAM_psy)[names(PAM_psy) == "mean"] <- "PAM.Yield_cu" #change name
PAM_psy<-PAM_psy[,c("Code","Parent.ID", "Treatment","Species","PAM.Yield_cu","Time.Point")] #remove col
PAM_psy<-unique(PAM_psy) #remove duplicates, now you have 

 
```
#merging full physio 
```{r}
#merg: this is EVERYTHING! PAM and BW subsetted to ONLY T0, T1 and TF time points, and all data is already transformed! 

#prep metadata for by Parent.ID df for ordinations
metaPhysio<-subset(meta, select= -X)#remove X from df
metaPhysio<-subset(metaPhysio, select= -Tank.num)#remove tank from df
metaPhysio<-subset(metaPhysio, select= -Frag.ID)#remove tank from df

#create duplicate T0 for H or A that isn't there for each parent. id
MetaT0<-metaPhysio #copy
MetaT0<-subset(MetaT0, Time.Point=="T0")
MetaT0Amb<-MetaT0
MetaT0Amb$Treatment[MetaT0Amb$Treatment == "High"] <- "Ambient" #change 
MetaT0High<-MetaT0
MetaT0High$Treatment[MetaT0High$Treatment == "Ambient"] <- "High" #change 
MetaT02<-rbind(MetaT0Amb,MetaT0High) #now oyu have T0 for Amb and High
metaPhysio<-metaPhysio[!(metaPhysio$Time.Point=="T0"),] #remove T0s 
metaPhysio<-rbind(metaPhysio,MetaT02) # add the full T0 dataset
metaPhysio$Code<-paste(metaPhysio$Species,metaPhysio$Treatment,metaPhysio$Parent.ID,metaPhysio$Time.Point) #make code


#MERGE
PhysioData<-merge(metaPhysio,Biomass_psy, all.x=T)
PhysioData<-merge(PhysioData,CellCounts_psy,all.x=T)
PhysioData<-merge(PhysioData,Calcification_psy,all.x=T)
PhysioData<-merge(PhysioData,Chla_psy,all.x=T)
PhysioData<-merge(PhysioData,Protein_psy,all.x=T)
PhysioData<-merge(PhysioData,PAM_psy,all.x=T)

#check
duplicated(PhysioData$Code)
PhysioData$Code[duplicated(PhysioData$Code)]
PhysioData<-unique(PhysioData) #this is the df. 

PhysioData$Parent.ID<-as.factor(as.character(PhysioData$Parent.ID))
PhysioData$Code<-as.factor(as.character(PhysioData$Code))

duplicated(PhysioData$Code)
PhysioData$Code[duplicated(PhysioData$Code)]

PhysioData_noMissing<-na.omit(PhysioData) #this removes 4 samples (partent/time/species)
#write.csv(PhysioData,"PhysioData_20210310.csv")
```

#nMDS of physio, PhysioData df, by Parent.ID, all data
```{r}   
Phy_nmds<-PhysioData_noMissing
 
  
library(vegan)  #hannah code below, edited
set.seed(30)

Phy_nmds.MDS<-metaMDS(PhysioData_noMissing[,7:12], distance = "bray", k = 2, trymax = 200, engine = c("monoMDS", "isoMDS"))
stressplot(Phy_nmds.MDS)
Phy_nmds.MDS.scores <- as.data.frame(scores(Phy_nmds.MDS)) #make the nMDS scores into a dataframe
Phy_nmds.MDS #get summary info, stress is 0.096 great

##Add variables that you want to test to your scores data sheet
Phy_nmds.MDS.scores$Parent.ID <- Phy_nmds$Parent.ID
Phy_nmds.MDS.scores$Treatment <- Phy_nmds$Treatment
Phy_nmds.MDS.scores$Species <- Phy_nmds$Species
Phy_nmds.MDS.scores$Time.Point <- Phy_nmds$Time.Point
Phy_nmds.MDS.scores$Code <- Phy_nmds$Code # move the code 
Phy_nmds.MDS.scores$Code2 <- paste(Phy_nmds.MDS.scores$Species,Phy_nmds.MDS.scores$Treatment,Phy_nmds.MDS.scores$Time.Point) # create new code species, time point treatment 
Phy_nmds.MDS.scores$Code2<-as.factor(Phy_nmds.MDS.scores$Code2)

Phy_nmds.MDS.scores$Code3 <- paste(Phy_nmds.MDS.scores$Species,Phy_nmds.MDS.scores$Treatment) # create new code species, time point treatment 
Phy_nmds.MDS.scores$Code3<-as.factor(Phy_nmds.MDS.scores$Code3)

##You can make hulls for your data clouds to group them by a variable
Phy_MY.hull <- Phy_nmds.MDS.scores %>% group_by(Species) %>%
  do({
    x = .
    data.frame(x[chull(x$NMDS1, x$NMDS2), ])
  })

##or you can make 95% ellipses using the following commands:

#first make the function veganCovEllipse:
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

#Next, make sure that your variable you are after is a factor:
str(Phy_nmds.MDS.scores)

#Make your ellipse:
phy_ell <- data.frame()
for(g in levels(Phy_nmds.MDS.scores$Species)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds.MDS.scores[Phy_nmds.MDS.scores$Species==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}

#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##Next we plot the nMDS:

##With hulls:
#ggplot(Phy_nmds.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Species)) +
#  geom_polygon(data = Phy_MY.hull, aes(x = NMDS1, y = NMDS2, colour= Code2, fill = Species), alpha = 0.1) #+
#  theme_classic() 

##With ellipses:
library("ggsci")
#use locuszoom pal
# Mcap   #D43F3AFF  red
# Pcomp  #EEA236FF   yellow
# Pvar  #5CB85CFF  green
# Pacu #46B8DAFF lt blue
#   scale_color_locuszoom()+


Sp.colors <- c("Montipora_capitata"= "#D43F3AFF","Porites_compressa"="#EEA236FF","Pavona_varians"="#5CB85CFF","Pocillopora_acuta"="#46B8DAFF", "High"="darkRed","Ambient"="navyBlue")

  
ggplot(Phy_nmds.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Species)) +
    scale_color_manual(values=Sp.colors)+
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
  ggtitle("NMDS Speices+Time+Treatment")


# DO BY TREATMENT
#Make elippses by TREATMENT

phy_ell <- data.frame()
for(g in levels(Phy_nmds.MDS.scores$Treatment)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds.MDS.scores[Phy_nmds.MDS.scores$Treatment==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}

ggplot(Phy_nmds.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Treatment)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
      scale_color_manual(values=Sp.colors)+
  ggtitle("NMDS Speices+Time+Treatment")

# DO BY CODE
#Make elippses by CODE

phy_ell <- data.frame()
for(g in levels(Phy_nmds.MDS.scores$Code2)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds.MDS.scores[Phy_nmds.MDS.scores$Code2==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}

ggplot(Phy_nmds.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Code2)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
  ggtitle("NMDS Speices+Time+Treatment")

#################### Subset PLOTS (do not rerun nmds) for T1 and TF visualizations


Phy_nmds.MDS.scores_T1<-subset(Phy_nmds.MDS.scores, Time.Point=="T1")



#first make the function veganCovEllipse:
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

#Next, make sure that your variable you are after is a factor:
str(Phy_nmds.MDS.scores)

#Make your ellipse:
phy_ell <- data.frame()
for(g in levels(Phy_nmds.MDS.scores_T1$Code3)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds.MDS.scores_T1[Phy_nmds.MDS.scores_T1$Code3==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}

#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##Next we plot the nMDS:

##With hulls:
#ggplot(Phy_nmds.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Species)) +
#  geom_polygon(data = Phy_MY.hull, aes(x = NMDS1, y = NMDS2, colour= Code2, fill = Species), alpha = 0.1) #+
#  theme_classic() 

##With ellipses:
cbp1 <- c("blue", "red", "green4", "green",
          "darkviolet", "hotpink", "darkorange3", "gold","turquoise4", "turquoise")

ggplot(Phy_nmds.MDS.scores_T1, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Treatment, shape=Treatment)) +
   scale_color_manual(values = cbp1)+
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
  ggtitle("NMDS Speices T1")


#################### Subset PLOTS (do not rerun nmds) for TF visualizations


Phy_nmds.MDS.scores_TF<-subset(Phy_nmds.MDS.scores, Time.Point=="TF")



#first make the function veganCovEllipse:
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

#Next, make sure that your variable you are after is a factor:
str(Phy_nmds.MDS.scores)

#Make your ellipse:
phy_ell <- data.frame()
for(g in levels(Phy_nmds.MDS.scores_TF$Code3)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds.MDS.scores_TF[Phy_nmds.MDS.scores_TF$Code3==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}

#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##Next we plot the nMDS:

##With hulls:
#ggplot(Phy_nmds.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Species)) +
#  geom_polygon(data = Phy_MY.hull, aes(x = NMDS1, y = NMDS2, colour= Code2, fill = Species), alpha = 0.1) #+
#  theme_classic() 

##With ellipses:
cbp1 <- c("blue", "red", "green4", "green",
          "darkviolet", "hotpink", "darkorange3", "gold","turquoise4", "turquoise")

ggplot(Phy_nmds.MDS.scores_TF, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Treatment, shape=Treatment)) +
   scale_color_manual(values = cbp1)+
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
  ggtitle("NMDS Speices TF")

```
#### By SPECIES. Make a NEW ordination for each species. 
Mcap
```{r}
Phy_nmds_Mcap<-PhysioData_noMissing
  

Phy_nmds_Mcap<-subset(Phy_nmds_Mcap, Species=="Montipora_capitata")

set.seed(30)

Phy_nmds_Mcap.MDS<-metaMDS(Phy_nmds_Mcap[,7:12], distance = "bray", k = 2, trymax = 200, engine = c("monoMDS", "isoMDS"))
stressplot(Phy_nmds_Mcap.MDS)
Phy_nmds_Mcap.MDS.scores <- as.data.frame(scores(Phy_nmds_Mcap.MDS)) #make the nMDS scores into a dataframe
Phy_nmds_Mcap.MDS #get summary info, stress is 0.096 great

##Add variables that you want to test to your scores data sheet
Phy_nmds_Mcap.MDS.scores$Parent.ID <- Phy_nmds_Mcap$Parent.ID
Phy_nmds_Mcap.MDS.scores$Treatment <- Phy_nmds_Mcap$Treatment
Phy_nmds_Mcap.MDS.scores$Species <- Phy_nmds_Mcap$Species
Phy_nmds_Mcap.MDS.scores$Time.Point <- Phy_nmds_Mcap$Time.Point
Phy_nmds_Mcap.MDS.scores$Code <- Phy_nmds_Mcap$Code # move the code 
Phy_nmds_Mcap.MDS.scores$Code2 <- paste(Phy_nmds_Mcap.MDS.scores$Treatment,Phy_nmds_Mcap.MDS.scores$Time.Point) # create new code species, time point treatment 
Phy_nmds_Mcap.MDS.scores$Code2<-as.factor(Phy_nmds_Mcap.MDS.scores$Code2)


#### BY Ttreatment 
##or you can make 95% ellipses using the following commands:

#first make the function veganCovEllipse:
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_Mcap.MDS.scores)

#Make your ellipse:
phy_ell <- data.frame()
for(g in levels(Phy_nmds_Mcap.MDS.scores$Code2)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_Mcap.MDS.scores[Phy_nmds_Mcap.MDS.scores$Code2==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}

#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)

#colors
Time.colors <- c("T0"= "#e8aa38","T1"="#666698","TF"="#76a6b6", "High"="#bd0212","Ambient"="#0e2ffb")
Code2.colors <- c("High"="","Ambient"="#c0cbd0",
                  "Ambient T0"= "#c9c9c9","High T0"= "#c9c9c9",
                  "Ambient T1"="#001eff","High T1"="#cc0000",
                  "Ambient TF"="#091f92","High TF"="#990000")
##Next we plot the nMDS:


ggplot(Phy_nmds_Mcap.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Code2)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
      scale_color_manual(values=Code2.colors)+
  ggtitle("NMDS Mcap Time+Treatment")


#### BY Time and Treat:

#first make the function veganCovEllipse:
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_Mcap.MDS.scores)

#Make your ellipse:
phy_ell <- data.frame()
for(g in levels(Phy_nmds_Mcap.MDS.scores$Code)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_Mcap.MDS.scores[Phy_nmds_Mcap.MDS.scores$Code==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}

#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)

#colors
Time.colors <- c("T0"= "#e8aa38","T1"="#666698","TF"="#76a6b6", "High"="#bd0212","Ambient"="#0e2ffb")
Code2.colors <- c( "Ambient T0"= "#c9c9c9","High T0"= "#c9c9c9",
                  "Ambient T1"="#001eff","High T1"="#cc0000",
                  "Ambient TF"="#091f92","High TF"="#990000")
##Next we plot the nMDS:


ggplot(Phy_nmds_Mcap.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Time.Point, shape =Time.Point, size=2)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
      scale_color_manual(values=Time.colors)+
  ggtitle("NMDS Mcap Time+Treatment")




```
Pcomp
```{r} 
Phy_nmds_Pcomp<-PhysioData_noMissing
Phy_nmds_Pcomp<-subset(Phy_nmds_Pcomp, Species=="Porites_compressa")

set.seed(30)

Phy_nmds_Pcomp.MDS<-metaMDS(Phy_nmds_Pcomp[,7:12], distance = "bray", k = 2, trymax = 200, engine = c("monoMDS", "isoMDS"))
stressplot(Phy_nmds_Pcomp.MDS)
Phy_nmds_Pcomp.MDS.scores <- as.data.frame(scores(Phy_nmds_Pcomp.MDS)) #make the nMDS scores into a dataframe
Phy_nmds_Pcomp.MDS #get summary info, stress is 0.096 great

##Add variables that you want to test to your scores data sheet
Phy_nmds_Pcomp.MDS.scores$Parent.ID <- Phy_nmds_Pcomp$Parent.ID
Phy_nmds_Pcomp.MDS.scores$Treatment <- Phy_nmds_Pcomp$Treatment
Phy_nmds_Pcomp.MDS.scores$Species <- Phy_nmds_Pcomp$Species
Phy_nmds_Pcomp.MDS.scores$Time.Point <- Phy_nmds_Pcomp$Time.Point
Phy_nmds_Pcomp.MDS.scores$Code <- Phy_nmds_Pcomp$Code # move the code 
Phy_nmds_Pcomp.MDS.scores$Code2 <- paste(Phy_nmds_Pcomp.MDS.scores$Treatment,Phy_nmds_Pcomp.MDS.scores$Time.Point) # create new code species, time point treatment 
Phy_nmds_Pcomp.MDS.scores$Code2<-as.factor(Phy_nmds_Pcomp.MDS.scores$Code2)

##You can make hulls for your data clouds to group them by a variable
#Phy_MY.hull <- Phy_nmds_Pcomp.MDS.scores %>% group_by(Treatment) %>%
#  do({
#    x = .
#    data.frame(x[chull(x$NMDS1, x$NMDS2), ])
#  })



############################################# BY TREATMENT


#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_Pcomp.MDS.scores)

## make 95% ellipses using the following commands: CODE2
phy_ell <- data.frame()
for(g in levels(Phy_nmds_Pcomp.MDS.scores$Code2)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_Pcomp.MDS.scores[Phy_nmds_Pcomp.MDS.scores$Code2==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}
#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##plot the nMDS:
##With ellipses:
Time.colors <- c("T0"= "#e8aa38","T1"="#666698","TF"="#76a6b6", "High"="#bd0212","Ambient"="#0e2ffb")
Code2.colors <- c( "Ambient T0"= "#c9c9c9","High T0"= "#c9c9c9",
                  "Ambient T1"="#001eff","High T1"="#cc0000",
                  "Ambient TF"="#091f92","High TF"="#990000")

ggplot(Phy_nmds_Pcomp.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Code2)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
        scale_color_manual(values=Code2.colors)+
  ggtitle("NMDS Pcomp Time+Treatment")



```
Pvar
```{r}
Phy_nmds_Pvar<-PhysioData_noMissing
Phy_nmds_Pvar<-subset(Phy_nmds_Pvar, Species=="Pavona_varians")

set.seed(30)

Phy_nmds_Pvar.MDS<-metaMDS(Phy_nmds_Pvar[,7:12], distance = "bray", k = 2, trymax = 200, engine = c("monoMDS", "isoMDS"))
stressplot(Phy_nmds_Pvar.MDS)
Phy_nmds_Pvar.MDS.scores <- as.data.frame(scores(Phy_nmds_Pvar.MDS)) #make the nMDS scores into a dataframe
Phy_nmds_Pvar.MDS #get summary info, stress is 0.096 great

##Add variables that you want to test to your scores data sheet
Phy_nmds_Pvar.MDS.scores$Parent.ID <- Phy_nmds_Pvar$Parent.ID
Phy_nmds_Pvar.MDS.scores$Treatment <- Phy_nmds_Pvar$Treatment
Phy_nmds_Pvar.MDS.scores$Species <- Phy_nmds_Pvar$Species
Phy_nmds_Pvar.MDS.scores$Time.Point <- Phy_nmds_Pvar$Time.Point
Phy_nmds_Pvar.MDS.scores$Code <- Phy_nmds_Pvar$Code # move the code 
Phy_nmds_Pvar.MDS.scores$Code2 <- paste(Phy_nmds_Pvar.MDS.scores$Treatment,Phy_nmds_Pvar.MDS.scores$Time.Point) # create new code species, time point treatment 
Phy_nmds_Pvar.MDS.scores$Code2<-as.factor(Phy_nmds_Pvar.MDS.scores$Code2)


############################################# BY TREATMENT * Time


#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_Pvar.MDS.scores)

## make 95% ellipses using the following commands: CODE2
phy_ell <- data.frame()
for(g in levels(Phy_nmds_Pvar.MDS.scores$Code2)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_Pvar.MDS.scores[Phy_nmds_Pvar.MDS.scores$Code2==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}
#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##plot the nMDS:
##With ellipses: 
#https://www.color-hex.com/color-palette/106923
Code2.colors <- c("High"="","Ambient"="#c0cbd0",
                  "Ambient T0"= "#c9c9c9","High T0"= "#c9c9c9",
                  "Ambient T1"="#001eff","High T1"="#cc0000",
                  "Ambient TF"="#091f92","High TF"="#990000")

ggplot(Phy_nmds_Pvar.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Code2)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
          scale_color_manual(values=Code2.colors)+
  ggtitle("NMDS Pvar Time+Treatment")

############################################# BY TIME
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_Pvar.MDS.scores)

## make 95% ellipses using the following commands: CODE2
phy_ell <- data.frame()
for(g in levels(Phy_nmds_Pvar.MDS.scores$Code2)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_Pvar.MDS.scores[Phy_nmds_Pvar.MDS.scores$Code2==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}
#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##plot the nMDS:
##With ellipses:

ggplot(Phy_nmds_Pvar.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Code2, shape =Code2)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
  ggtitle("NMDS Pvar Time+Treatment")


```
P acuta
```{r}

Phy_nmds_Pacu<-PhysioData_noMissing
Phy_nmds_Pacu<-subset(Phy_nmds_Pacu, Species=="Pocillopora_acuta")

set.seed(30)

Phy_nmds_Pacu.MDS<-metaMDS(Phy_nmds_Pacu[,7:12], distance = "bray", k = 2, trymax = 200, engine = c("monoMDS", "isoMDS"))
stressplot(Phy_nmds_Pacu.MDS)
Phy_nmds_Pacu.MDS.scores <- as.data.frame(scores(Phy_nmds_Pacu.MDS)) #make the nMDS scores into a dataframe
Phy_nmds_Pacu.MDS #get summary info, stress is 0.096 great

##Add variables that you want to test to your scores data sheet
Phy_nmds_Pacu.MDS.scores$Parent.ID <- Phy_nmds_Pacu$Parent.ID
Phy_nmds_Pacu.MDS.scores$Treatment <- Phy_nmds_Pacu$Treatment
Phy_nmds_Pacu.MDS.scores$Species <- Phy_nmds_Pacu$Species
Phy_nmds_Pacu.MDS.scores$Time.Point <- Phy_nmds_Pacu$Time.Point
Phy_nmds_Pacu.MDS.scores$Code <- Phy_nmds_Pacu$Code # move the code 
Phy_nmds_Pacu.MDS.scores$Code2 <- paste(Phy_nmds_Pacu.MDS.scores$Treatment,Phy_nmds_Pacu.MDS.scores$Time.Point) # create new code species, time point treatment 
Phy_nmds_Pacu.MDS.scores$Code2<-as.factor(Phy_nmds_Pacu.MDS.scores$Code2)

##You can make hulls for your data clouds to group them by a variable
#Phy_MY.hull <- Phy_nmds_Pacu.MDS.scores %>% group_by(Treatment) %>%
#  do({
#    x = .
#    data.frame(x[chull(x$NMDS1, x$NMDS2), ])
#  })



############################################# BY TREATMENT


#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_Pacu.MDS.scores)

## make 95% ellipses using the following commands: CODE2
phy_ell <- data.frame()
for(g in levels(Phy_nmds_Pacu.MDS.scores$Code2)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_Pacu.MDS.scores[Phy_nmds_Pacu.MDS.scores$Code2==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}
#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##plot the nMDS:
##With ellipses:

ggplot(Phy_nmds_Pacu.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Code2)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
            scale_color_manual(values=Code2.colors)+
  ggtitle("NMDS Pacu Treatment")

############################################# BY TIME
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_Pacu.MDS.scores)

## make 95% ellipses using the following commands: CODE2
phy_ell <- data.frame()
for(g in levels(Phy_nmds_Pacu.MDS.scores$Code2)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_Pacu.MDS.scores[Phy_nmds_Pacu.MDS.scores$Code2==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}
#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##plot the nMDS:
##With ellipses:

ggplot(Phy_nmds_Pacu.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Code2, shape =Code2)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
  ggtitle("NMDS Pacu Time+Treatment")

```

T1- only all species: RERUN NMDS
```{r}

Phy_nmds_T1only<-PhysioData_noMissing
Phy_nmds_T1only<-subset(Phy_nmds_T1only, Time.Point=="T1")

set.seed(30)

Phy_nmds_T1only.MDS<-metaMDS(Phy_nmds_T1only[,7:12], distance = "bray", k = 2, trymax = 200, engine = c("monoMDS", "isoMDS"))
stressplot(Phy_nmds_T1only.MDS)
Phy_nmds_T1only.MDS.scores <- as.data.frame(scores(Phy_nmds_T1only.MDS)) #make the nMDS scores into a dataframe
Phy_nmds_T1only.MDS #get summary info, stress is 0.096 great

##Add variables that you want to test to your scores data sheet
Phy_nmds_T1only.MDS.scores$Parent.ID <- Phy_nmds_T1only$Parent.ID
Phy_nmds_T1only.MDS.scores$Treatment <- Phy_nmds_T1only$Treatment
Phy_nmds_T1only.MDS.scores$Species <- Phy_nmds_T1only$Species
Phy_nmds_T1only.MDS.scores$Time.Point <- Phy_nmds_T1only$Time.Point
Phy_nmds_T1only.MDS.scores$Code <- Phy_nmds_T1only$Code # move the code 
Phy_nmds_T1only.MDS.scores$Code2 <- paste(Phy_nmds_T1only.MDS.scores$Treatment,Phy_nmds_T1only.MDS.scores$Species) # create new code species, time point treatment 
Phy_nmds_T1only.MDS.scores$Code2<-as.factor(Phy_nmds_T1only.MDS.scores$Code2)

##You can make hulls for your data clouds to group them by a variable
#Phy_MY.hull <- Phy_nmds_T1only.MDS.scores %>% group_by(Treatment) %>%
#  do({
#    x = .
#    data.frame(x[chull(x$NMDS1, x$NMDS2), ])
#  })



############################################# BY TREATMENT


#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_T1only.MDS.scores)

## make 95% ellipses using the following commands: CODE2
phy_ell <- data.frame()
for(g in levels(Phy_nmds_T1only.MDS.scores$Treatment)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_T1only.MDS.scores[Phy_nmds_T1only.MDS.scores$Treatment==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}
#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##plot the nMDS:
##With ellipses:

ggplot(Phy_nmds_T1only.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Treatment, shape =Time.Point)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
  ggtitle("NMDS T1only Treatment")

############################################# BY TIME
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

#Next, make sure that your variable you are after is a factor:
str(Phy_nmds_T1only.MDS.scores)

## make 95% ellipses using the following commands: CODE2
phy_ell <- data.frame()
for(g in levels(Phy_nmds_T1only.MDS.scores$Code2)){
  phy_ell <- rbind(phy_ell, cbind(as.data.frame(with(Phy_nmds_T1only.MDS.scores[Phy_nmds_T1only.MDS.scores$Code2==g,],
                                                   veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                ,group=g))
}
#Check that the command worked - often I run into issues if the variable is not a factor (important to check as above!)
str(phy_ell)


##plot the nMDS:
##With ellipses:

ggplot(Phy_nmds_T1only.MDS.scores, aes(x = NMDS1, y= NMDS2)) + geom_point(aes(colour = Treatment, shape =Treatment)) +
  geom_path(data = phy_ell, aes(x = NMDS1, y = NMDS2, colour= group),size = 0.5, linetype = 1) +
  theme_classic()+
  ggtitle("NMDS T1only Species and Treatment")

```

#PERMANOVA on physio only

```{r}   
  
#ALL THE DATA

phy.dist <- vegdist(PhysioData_noMissing [7:12], method="bray") #make dist matrix
phy.div <- adonis2(PhysioData_noMissing [7:12] ~ Species*Treatment*Time.Point, data = PhysioData_noMissing [1:6], permutations = 999, method="bray")
phy.div #view, everything is significant

##Check for homogeneity of dispersions (you want this to be insignificant)
phy.dist.disp <- betadisper(phy.dist, phy.dist$Treatment, type = "centroid")
#par(pty = 's')
#boxplot(phy.dist.disp) #visualise your dispersion
#permutest(phy.dist.disp) #this is your PERMDISP test

#this runs but it's only univariate so not helpful. 
pairwise.adonis(PhysioData_noMissing [7:12], PhysioData_noMissing$Species)


#krissy code not running 
adonis(PhysioData_noMissing [7:12]~ Species + Treatment+ Time.Point+ Species:Treatment:Time.Point, data=PhysioData_noMissing [1:6], contr.unordered = "contr.treatment", by = "margin", distance = "bray", set.seed(30))
#all significant. 





```


##By SPECIES
```{r}   
#Mcap, Phy_nmds_Mcap
phy_Mcap.dist <- vegdist(Phy_nmds_Mcap [7:12], method="bray") #make dist matrix
phy_Mcap.div <- adonis2(Phy_nmds_Mcap [7:12] ~ Treatment*Time.Point, data = Phy_nmds_Mcap [1:6], permutations = 999, method="bray")
phy_Mcap.div #view, Time is significant

#Pcomp, Phy_nmds_Pcomp
phy_Pcomp.dist <- vegdist(Phy_nmds_Pcomp [7:12], method="bray") #make dist matrix
phy_Pcomp.div <- adonis2(Phy_nmds_Pcomp [7:12] ~ Treatment*Time.Point, data = Phy_nmds_Pcomp [1:6], permutations = 999, method="bray")
phy_Pcomp.div #view, all is significant


#Pvar, Phy_nmds_Pvar
phy_Pvar.dist <- vegdist(Phy_nmds_Pvar [7:12], method="bray") #make dist matrix
phy_Pvar.div <- adonis2(Phy_nmds_Pvar [7:12] ~ Treatment*Time.Point, data = Phy_nmds_Pvar [1:6], permutations = 999, method="bray")
phy_Pvar.div #view, all is significant

#Pacu, Phy_nmds_Pacu
phy_Pacu.dist <- vegdist(Phy_nmds_Pacu [7:12], method="bray") #make dist matrix
phy_Pacu.div <- adonis2(Phy_nmds_Pacu [7:12] ~ Treatment*Time.Point, data = Phy_nmds_Pacu [1:6], permutations = 999, method="bray")
phy_Pacu.div #view, all is significant


```






####ITS2
## ITS2
#Analysis (RCunning Script) REL Abundance -  
```{r}
library(tidyverse)
library(janitor)
library(phyloseq)  # source('http://bioconductor.org/biocLite.R'); biocLite('phyloseq')
library(zoo)
library(stringr)
library(vegan)
library(multcompView)
library("ggpubr")

library(phyloseq)
library(ggplot2)
library(dplyr)
library(plyr)
library(tidyverse)
library(gridExtra)
library(vegan)
library(decontam)
library(RColorBrewer)

# Load data (see data_exploration.Rmd)
load("data/Raw_coral_phyloseq.RData") #use RAW data
```
Hey what does this do?
```{r}  
coral2 <- coral_RAW %>%
  subset_samples(!is.na(host_species))
coralDIV2 <- coralDIV_RAW %>% 
  subset_samples(!is.na(host_species))

# USE IF YOU WANT TO MAKE RELA DATA ACTUALLY OUT OF 100% 
# Transform to relative abundance
#coral2 <- transform_sample_counts(coral2, function(x) 100 * x/sum(x))
#coralDIV2 <- transform_sample_counts(coralDIV2, function(x) 100 * x/sum(x))

```

Monster all data vis (not helpful)
```{r, fig.width = 10, fig.height = 40} 
# Plot ITS2 profiles for each coral sample by species and site
plot_bar(coral, fill = "its2_type_profile") +
  facet_wrap(~ host_genus + host_species, ncol = 6, scales = "free") +
  #theme(legend.position = "none") +
  geom_bar(stat = "identity")
# Plot all post-MED sequences
plot_bar(coralDIV2, fill = "DIV") +
  facet_wrap(~ host_genus + host_species, scales = "free") +
  #theme(legend.position = "none") +
  geom_bar(stat = "identity")
# Glom all post-MED sequences by clade:
divclades <- coralDIV2 %>%
  tax_glom(taxrank = "clade") %>%                # agglomerate at clade level
  psmelt() %>%                                         # Melt to long format
  arrange(clade)                                      # Sort data frame alphabetically by clade
ggplot(divclades, aes(x = Sample, y = Abundance, fill = clade)) +
  facet_wrap(~ host_genus + host_species, scales = "free") +
  geom_bar(stat = "identity") +
  labs(title=NULL, x = NULL, y = "Relative abundance")
```
Look at low samps  
```{r}  
#lets see what is up with the low samps
LowSamps<-readxl::read_xlsx("Data/ITS2/post_med_seqs/133_20201216_DBV_20201216T020209.seqs.absolute.meta_only.xlsx")

LowSamps_dups<-merge(LowSamps, Metadata, all.x=T)
LowSamps_dups2<-subset(LowSamps_dups, Notes=="Dup")
#write.csv(LowSamps_dups2,"LowSamps_dups.csv")

LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs_log<-log10(LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs)
hist(LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs_log, breaks=20)
```

Data wrangle: remove samples with under 5k, remove duplicated samps, save RDS Rarefied and non: coralDIV2_nonrare and coralDIV2_rare
hannah code!

#FOR DIVS set up and clean up
```{r}    
## check out data
ntaxa(coralDIV2)  #num taxa
nsamples(coralDIV2)   #num samples
sample_names(coralDIV2)[1:5] #samp names
rank_names(coralDIV2) #ranks are DIV and clade
sample_variables(coralDIV2) #all metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(coralDIV2), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(coralDIV2)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = sample_type)) +
  geom_point()

#Remove all samples with <5000 reads and create sample data
#These data also useful for beta diversity analyses (non-rarefied)
coralDIV2_nonrare <- prune_samples(sample_sums(coralDIV2)>=5000, coralDIV2) # remove samps <5000, save as new obj
coralDIV2_data.nonrare <- as(sample_data(coralDIV2_nonrare), "data.frame")

##Sample filtering and rarefaction
#remove duplicates that failed
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1110B")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1166")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1388b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1857")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3085")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3149b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3149")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3477a")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S761b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S775b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN134")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN198b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN209b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3085b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1857")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S761b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S198b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN127")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN98b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S2893")

coralDIV2_data.nonrare <- as(sample_data(coralDIV2_nonrare), "data.frame")

#saveRDS(coralDIV2_nonrare, file = "coralDIV2_nonrare.RDS", compress = TRUE) #s

#Make a rarefied phyloseq object for alpha diversity analyses
coralDIV2_rare <- rarefy_even_depth(coralDIV2_nonrare, sample.size = 7000, rngseed = 711)   #this removed 17 DIVs
sample_sums(coralDIV2_rare) #Double check that the sub-sampling worked, this should report 1000 for each sample
coralDIV2.rare <- as(sample_data(coralDIV2_rare), "data.frame")
coralDIV2_rare
#saveRDS(coralDIV2_rare, file = "coralDIV2_rare.RDS", compress = TRUE) #save!

#### Create a duplicate of T0s for the other A and H treatments for comparision: df coralDIV2_rare

controlT0_Amb = subset_samples(coralDIV2_rare, Time.Point == "T0"&Treatment == "Ambient")  #copy

#tax table
tax  = tax_table(controlT0_Amb)

  # sample data
  sd = sample_data(controlT0_Amb)
  rownames(sd) <- paste0(rownames(sd), "_dup")
  # otu_table
  OTU = otu_table(controlT0_Amb)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  sd$Treatment <- as.character(sd$Treatment)
  sd$ID <- as.character(sd$ID)
  sd$Treatment <- "High"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
  sd$Treatment <- as.factor(sd$Treatment)
  sd$ID <- as.factor(sd$ID)

  Amb2<-phyloseq(OTU, sd, tax)

##High
  controlT0_High = subset_samples(coralDIV2_rare, Time.Point == "T0"&Treatment == "High")  #copy

#tax table
tax  = tax_table(controlT0_High)

  # sample data
  sd = sample_data(controlT0_High)
  rownames(sd) <- paste0(rownames(sd), "_dup")
  # otu_table
  OTU = otu_table(controlT0_High)
  sample_names(OTU) <- paste0(sample_names(OTU), "_dup")
  # Avoid conflict between factor and character
  sd$Treatment <- as.character(sd$Treatment)
  sd$ID <- as.character(sd$ID)
  sd$Treatment <- "Ambient"    #rename HIGH
  sd$ID <- paste0(sd$ID, "_dup")
  sd$Treatment <- as.factor(sd$Treatment)
  sd$ID <- as.factor(sd$ID)

  High2<-phyloseq(OTU, sd, tax)
  
#Merge control dups H and A  
Controldups = merge_phyloseq(Amb2,High2)
Controldups
  
#merg them to the original df:

coralDIV2_rare2 = merge_phyloseq(coralDIV2_rare,Controldups)
coralDIV2_rare2
coralDIV2_rare2.sd <- as(sample_data(coralDIV2_rare2), "data.frame") #create sample data frame to view
coralDIV2_rare2.sd

# add a Clade column to the metadata, fill as you devide species (may become profile)

# # pull out tax table
# tax  = tax_table(coralDIV2_rare2)
# # pull out otu_table
# OTU = otu_table(coralDIV2_rare2)
# 
# # EDIT sample data
#   sd = sample_data(coralDIV2_rare2)
#   sd$Clade[sd$Parent.ID==363]<-"C"
#   sd$Clade[sd$Parent.ID==364]<-"CD"
#   sd$Clade[sd$Parent.ID==365]<-"CD"
#   sd$Clade[sd$Parent.ID==366]<-"CD"
#   sd$Clade[sd$Parent.ID==367]<-"C"
#   sd$Clade[sd$Parent.ID==368]<-"CD"
#   sd$Clade[sd$Parent.ID==369]<-"CD"
#   sd$Clade[sd$Parent.ID==370]<-"C"
#   sd$Clade[sd$Parent.ID==371]<-"C"
#   sd$Clade[sd$Parent.ID==372]<-"C"
#   sd$Clade[sd$Parent.ID==373]<-"C"
#   sd$Clade[sd$Parent.ID==374]<-"CD"
#   sd$Clade<-as.factor(sd$Clade) #change to factor
#   sd$Clade[sd$Parent.ID==363]<-"C"
#   # Avoid conflict between factor and character
#   sd$Clade <- as.factor(sd$Clade)
# 
#   coralDIV2_rare2<-phyloseq(OTU, sd, tax)
#   coralDIV2_rare2
# coralDIV2_rare2.sd <- as(sample_data(coralDIV2_rare2), "data.frame") #create sample data frame to view


  
```

# DIV analysis and plots, use rarefied coralDIV2_rare2
```{r}   
#RELA
DIV_RelA  = transform_sample_counts(coralDIV2_rare2, function(x) x / sum(x) )  #save as RELA DIVs
data.ra <- as(sample_data(DIV_RelA), "data.frame") #look at samp data

# there are >300 taxa, let's prune to top 100
topN = 100
most_abundant_taxa = sort(taxa_sums(DIV_RelA), TRUE)[1:topN]
print(most_abundant_taxa)

#top100 DIVs across all data
TopNOTUs = names(sort(taxa_sums(DIV_RelA), TRUE)[1:100])
ent100 = prune_taxa(TopNOTUs, DIV_RelA)
#plot_bar(ent100, "Species", fill = "DIV", facet_grid = ~DIV)

#subset by species
Mcap_DIV_RA <- subset_samples(DIV_RelA, host_genus=="montipora")
Pcomp_DIV_RA <- subset_samples(DIV_RelA, host_genus=="porites")
Pvar_DIV_RA <- subset_samples(DIV_RelA, host_genus=="pavona")
Pacu_DIV_RA <- subset_samples(DIV_RelA, host_genus=="pocillopora")

#top 50 DIVs across by species 

#Mcap
TopNOTUsMcap = names(sort(taxa_sums(Mcap_DIV_RA), TRUE)[1:50])
Mcapent50 = prune_taxa(TopNOTUsMcap, Mcap_DIV_RA)
McapBarPlot<-plot_bar(Mcapent50, x="Parent.ID", fill="DIV", facet_grid = "Time.Point~Treatment");McapBarPlot
#ggsave("mMcapBarPlot_DIVrelARare.pdf", McapBarPlot, dpi=300, width=8)

#Mcap C and CD
#C
Mcap_DIV_RA.C<-subset_samples(Mcap_DIV_RA, Clade=="C")
TopNOTUsMcap.C = names(sort(taxa_sums(Mcap_DIV_RA.C), TRUE)[1:50])
Mcap.50.C = prune_taxa(TopNOTUsMcap.C, Mcap_DIV_RA.C)
McapBarPlot.C<-plot_bar(Mcap.50.C, x="Parent.ID", fill="DIV", facet_grid = "Time.Point~Treatment");McapBarPlot.C
#ggsave("McapBarPlot_DIV_C.pdf", McapBarPlot.C, dpi=300, width=8)

#CD
Mcap_DIV_RA.CD<-subset_samples(Mcap_DIV_RA, Clade=="CD")
TopNOTUsMcap.CD = names(sort(taxa_sums(Mcap_DIV_RA.CD), TRUE)[1:50])
Mcap.50.CD = prune_taxa(TopNOTUsMcap.CD, Mcap_DIV_RA.CD)
McapBarPlot.CD<-plot_bar(Mcap.50.CD, x="Parent.ID", fill="DIV", facet_grid = "Time.Point~Treatment");McapBarPlot.CD
#ggsave("McapBarPlot_DIV_CD.pdf", McapBarPlot.CD, dpi=300, width=8)


#Pcomp
TopNOTUsPcomp = names(sort(taxa_sums(Pcomp_DIV_RA), TRUE)[1:50])
Pcompent50 = prune_taxa(TopNOTUsPcomp,Pcomp_DIV_RA )
PcompBarPlot<-plot_bar(Pcompent50, x="Parent.ID", fill="DIV", facet_grid = "Time.Point~Treatment");PcompBarPlot
#ggsave("PcompBarPlot_DIVrelARare.pdf", PcompBarPlot, dpi=300, width=8)

#Pvar
TopNOTUsVar = names(sort(taxa_sums(Pvar_DIV_RA), TRUE)[1:50])
Pvarent50 = prune_taxa(TopNOTUsVar, Pvar_DIV_RA)
PvarBarPlot<-plot_bar(Pvarent50, x="Parent.ID", fill="DIV", facet_grid = "Time.Point~Treatment");PvarBarPlot
#ggsave("PvarBarPlot_DIVrelARare.pdf", PvarBarPlot, dpi=300, width=8)

#Pacu
TopNOTUsPacu = names(sort(taxa_sums(Pacu_DIV_RA), TRUE)[1:50])
Pacuent50 = prune_taxa(TopNOTUsPacu, Pacu_DIV_RA)
PacuBarPlot<-plot_bar(Pacuent50, x="Parent.ID", fill="DIV", facet_grid = "Time.Point~Treatment");PacuBarPlot
#ggsave("PacuBarPlot_DIVrelARare.pdf", PacuBarPlot, dpi=300, width=8)

```

#Ordination DIV_RelA <- you might not need this section, code below for each species  
try vegan nmds - its the same, delete when CN says cool
```{r} 
  
# convert the sample_data() within a phyloseq object to a vegan compatible data object
ITS22veg <- function(DIV_RelA) { 
  sd <- sample_data(DIV_RelA)
  return(as(sd,"data.frame"))
}

# convert the otu_table() within a phyloseq object to a vegan compatible data object
ITS2otu2veg <- function(DIV_RelA) {
  OTU <- otu_table(DIV_RelA)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

# Calculate distance matrix
ITS2.spp_distmat <- vegdist(OTU, method = "bray")
# Creating easy to view matrix and writing .csv
its2.spp_distmat <- 
  as.matrix(ITS2.spp_distmat, labels = T)
#write.csv(island.spp_distmat, "island_spp_distmat.csv")

# Running NMDS in vegan (metaMDS)
its2.spp_NMS <-
  metaMDS(ITS2.spp_distmat,
          distance = "bray",
          k =2,
          maxit = 999, 
          trymax = 500,
          wascores = TRUE, set.seed(30))

# Shepards test/goodness of fit
goodness(its2.spp_NMS) # Produces a results of test statistics for goodness of fit for each point

stressplot(its2.spp_NMS) # Produces a Shepards diagram

# Plotting points in ordination space
plot(its2.spp_NMS, "sites")   # Produces distance 
orditorp(its2.spp_NMS, "sites")   # Gives points labels




```
Heatmap and first crack at ordination, delete when solid
```{r}      
# CUT?? OrdiSpecies<-plot_ordination(DIV_RelA, ordinate(DIV_RelA, "NMDS"), color = "Species", label="ID") + geom_point(size = 5) 
# #ggsave("OrdiSpecies.pdf", OrdiSpecies, dpi=300, width=8)


#heat map
plot_heatmap(DIV_RelA, "NMDS", "bray", "Species", "DIV")

#by species NOT INTERESTING
  #mcap
  Mcap_DIV_RA_T1 <- subset_samples(Mcap_DIV_RA, Time.Point=="T1")
  plot_heatmap(Mcap_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
 #Pcomp
  Pcomp_DIV_RA_T1 <- subset_samples(Pcomp_DIV_RA, Time.Point=="T1")
  plot_heatmap(Pcomp_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
  #Pvar
  Pvar_DIV_RA_T1 <- subset_samples(Pvar_DIV_RA, Time.Point=="T1")
  plot_heatmap(Pvar_DIV_RA_T1, "NMDS", "bray", "Treatment", "DIV") #nothing great
  plot_heatmap(Pvar_DIV_RA, "NMDS", "bray", "Parent.ID", "DIV")

#ordination time - CUT WHEN YOU HAVE OTHER 4 SPECIES ORDINATION SORTED 
DIVrareRel.ord <- ordinate(DIV_RelA, "NMDS", "bray", set.seed(30))
DIVrareRel1 = plot_ordination(DIV_RelA, DIVrareRel.ord, type="samples", shape="Treatment",color="Time.Point", title="species")
print(DIVrareRel1)
DIVrareRel1 + facet_wrap(~Species, 3)

### let's do this again and run the NMDS by species now that we know they are different. 
# mcap
Mcap_DIV_RA.ord <- ordinate(Mcap_DIV_RA, "NMDS", "bray", set.seed(30))
DIVrareRel1_Mcap = plot_ordination(Mcap_DIV_RA, Mcap_DIV_RA.ord, type="samples", shape="Treatment",color="Time.Point", title="Mcap NMDS")
print(DIVrareRel1_Mcap)
  #add ellipses
  DIVrareRel1_Mcap + 
  stat_ellipse(type = "norm", linetype = 2, alpha=.2) +
  stat_ellipse(type = "t") +
  theme_bw()

# pcomp
Pcomp_DIV_RA.ord <- ordinate(Pcomp_DIV_RA, "NMDS", "bray", set.seed(30))
DIVrareRel1_Pcomp = plot_ordination(Pcomp_DIV_RA, Pcomp_DIV_RA.ord, type="samples", shape="Treatment",color="Time.Point", title="Pcomp NMDS")
print(DIVrareRel1_Pcomp)
#add ellipses
  DIVrareRel1_Pcomp + 
  stat_ellipse(type = "norm",  alpha=.2) +
  theme_bw()
```
#Beta Diversity: df rarefied and relative abudnaces  
All Species
```{r}   
#DIV_RelA is coralDIV2_rare2 with RELA

DIV_RelA_stats  = DIV_RelA #make a copy
otu_table(DIV_RelA_stats)[1:5, 1:5] #check rela worked

#bray curtis presence-absense and abundance
## REMOVE T0 for models
DIV_RelA_stats.T1F<-subset_samples(DIV_RelA_stats, Time.Point=="T1"|Time.Point=="TF")

#bray curtis presence-absense and abundance

#all speices
set.seed(30)

#make bray curtis data matrix
bc.all <- phyloseq::distance(DIV_RelA_stats, method = "bray")
  samp.df <- as(sample_data(DIV_RelA_stats), "data.frame") #sample df
#adonis test
adonis(bc.all ~ Species*Treatment*Time.Point, data = samp.df)
    

# Homogeneity of dispersion test
betaAll <- betadisper(bc.all, samp.df$Species)
permutest(betaAll, pairwise = T) 
(bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
plot(bc.all.HSD)


#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats <- ordinate(DIV_RelA_stats, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats)
scores.ord.DIV_RelA_stats <- as.data.frame(scores(ord.DIV_RelA_stats)) #make df
scores.ord.DIV_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.DIV_RelA_stats$Species <- samp.df$Species

ggplot(scores.ord.DIV_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
ggsave("bc_all.pdf")

```

#Split data by species 
Beta
Montipora - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#mcap
mcap.RelRare.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "montipora")
mcap.RelRare.T1 <- subset_samples(mcap.RelRareT1F, Time.Point == "T1")
mcap.RelRare.TF <- subset_samples(mcap.RelRareT1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.mcap.T1F <- phyloseq::distance(mcap.RelRare.T1F, method = "bray")
  mcap.samp.df.T1F <- as(sample_data(mcap.RelRare.T1F), "data.frame")            #sample df
  adonis(bc.mcap.T1F ~ Clade*Treatment*Time.Point, data = mcap.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.mcapT1 <- phyloseq::distance(mcap.RelRare.T1, method = "bray")
  mcap.samp.df.T1 <- as(sample_data(mcap.RelRare.T1), "data.frame") #sample df
  adonis(bc.mcapT1 ~ Clade*Treatment, data = mcap.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Clade)
  permutest(beta.mcap.T1, pairwise = T) 
  beta.mcap.T1 <- betadisper(bc.mcapT1, mcap.samp.df.T1$Treatment)
  permutest(beta.mcap.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.mcap.TF <- phyloseq::distance(mcap.RelRare.TF, method = "bray")
  mcap.samp.df.TF <- as(sample_data(mcap.RelRare.TF), "data.frame") #sample df
#adonis test
  adonis(bc.mcap.TF ~ Clade*Treatment, data = mcap.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Clade)
  permutest(beta.mcap.TF, pairwise = T) 
  beta.mcap.TF <- betadisper(bc.mcap.TF, mcap.samp.df.TF$Treatment)
  permutest(beta.mcap.TF, pairwise = T) 


## by clade: C vs CD
  
# C only, new objs
mcap.RelRareT1F.C <- subset_samples(mcap.RelRareT1F, Clade == "C")
mcap.RelRare.T1.C <- subset_samples(mcap.RelRare.T1, Clade == "C")
mcap.RelRare.TF.C <- subset_samples(mcap.RelRare.TF, Clade == "C")
  
#T1TF.C
#make bray curtis data matrix -T1 and TF
  bc.mcap.T1F.C <- phyloseq::distance(mcap.RelRareT1F.C, method = "bray")
  mcap.samp.df.T1F.C <- as(sample_data(mcap.RelRareT1F.C), "data.frame") #sample df
#adonis test
  adonis(bc.mcap.T1F.C ~ Treatment*Time.Point, data = mcap.samp.df.T1F.C)

#T1.C
#make bray curtis data matrix 
  bc.mcapT1.C <- phyloseq::distance(mcap.RelRare.T1.C, method = "bray")
  mcap.samp.df.T1.C <- as(sample_data(mcap.RelRare.T1.C), "data.frame") #sample df
#adonis test
  adonis(bc.mcapT1.C ~ Treatment, data = mcap.samp.df.T1.C)
# Homogeneity of dispersion test  
  beta.mcap.T1.C <- betadisper(bc.mcapT1.C, mcap.samp.df.T1.C$Treatment)
  permutest(beta.mcap.T1.C, pairwise = T) 

#TF.C
#make bray curtis data matrix 
  bc.mcap.TF.C <- phyloseq::distance(mcap.RelRare.TF.C, method = "bray")
  mcap.samp.df.TF.C <- as(sample_data(mcap.RelRare.TF.C), "data.frame") #sample df
#adonis test
  adonis(bc.mcap.TF.C ~ Treatment, data = mcap.samp.df.TF.C)
# Homogeneity of dispersion test  
  beta.mcap.TF.C <- betadisper(bc.mcap.TF.C, mcap.samp.df.TF.C$Treatment)
  permutest(beta.mcap.TF.C, pairwise = T) 


# CD only, new objs
mcap.RelRareT1F.CD <- subset_samples(mcap.RelRareT1F, Clade == "CD")
mcap.RelRare.T1.CD <- subset_samples(mcap.RelRare.T1, Clade == "CD")
mcap.RelRare.TF.CD <- subset_samples(mcap.RelRare.TF, Clade == "CD")
  
#T1TF.CD
#make bray curtis data matrix -T1 and TF
  bc.mcap.T1F.CD <- phyloseq::distance(mcap.RelRareT1F.CD, method = "bray")
  mcap.samp.df.T1F.CD <- as(sample_data(mcap.RelRareT1F.CD), "data.frame") #sample df
#adonis test
  adonis(bc.mcap.T1F.CD ~ Treatment*Time.Point, data = mcap.samp.df.T1F.CD)

#T1.CD
#make bray curtis data matrix 
  bc.mcapT1.CD <- phyloseq::distance(mcap.RelRare.T1.CD, method = "bray")
  mcap.samp.df.T1.CD <- as(sample_data(mcap.RelRare.T1.CD), "data.frame") #sample df
  adonis(bc.mcapT1.CD ~ Treatment, data = mcap.samp.df.T1.CD)#adonis test
# Homogeneity of dispersion test  
  beta.mcap.T1.CD <- betadisper(bc.mcapT1.CD, mcap.samp.df.T1.CD$Treatment)
  permutest(beta.mcap.T1.CD, pairwise = T) 

#TF.CD
#make bray curtis data matrix 
  bc.mcap.TF.CD <- phyloseq::distance(mcap.RelRare.TF.CD, method = "bray")
  mcap.samp.df.TF.CD <- as(sample_data(mcap.RelRare.TF.CD), "data.frame") #sample df
#adonis test
  adonis(bc.mcap.TF.CD ~ Treatment, data = mcap.samp.df.TF.CD)
# Homogeneity of dispersion test  
  beta.mcap.TF.CD <- betadisper(bc.mcap.TF.CD, mcap.samp.df.TF.CD$Treatment)
  permutest(beta.mcap.TF.CD, pairwise = T) 
``` 
Mcap NMDS
```{r}
library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Mcap.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "honeydew4", "372" = "khaki4","373" ="ivory4", "374" = "lightblue", Ambient = "Blue", High = "Red","C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred") 

#T1 and TF
ord.mcap <- ordinate(mcap.RelRareT1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.mcap)

scores.mcap.samp.df <- as.data.frame(scores(ord.mcap) ) 
scores.mcap.samp.df$Time.Point <- mcap.samp.df.T1F$Time.Point
scores.mcap.samp.df$Parent.ID <- mcap.samp.df.T1F$Parent.ID
scores.mcap.samp.df$Treatment <- mcap.samp.df.T1F$Treatment
scores.mcap.samp.df$Clade <- mcap.samp.df.T1F$Clade


ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Clade), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("McapT1its2.pdf")
ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("McapT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(mcap.RelRare.TF, ordinate(mcap.RelRare, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.mcapT1 <- ordinate(mcap.RelRare.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcapT1)
scores.mcapT1 <- as.data.frame(scores(ord.mcapT1))
scores.mcapT1$Treatment <- mcap.samp.df.T1$Treatment
scores.mcapT1$Parent.ID <- mcap.samp.df.T1$Parent.ID
scores.mcapT1$Clade <- mcap.samp.df.T1$Clade
scores.mcapT1$CladeTreat <- as.factor(paste(mcap.samp.df.T1$Clade,mcap.samp.df.T1$Treatment))


ggplot(scores.mcapT1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("M. capitata T1 Bray Curtis") +
  theme_classic()
ggsave("McapT1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.mcapTF <- ordinate(mcap.RelRare.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcapTF)
scores.mcapTF <- as.data.frame(scores(ord.mcapTF))
scores.mcapTF$Treatment <- mcap.samp.df.TF$Treatment
scores.mcapTF$Parent.ID <- mcap.samp.df.TF$Parent.ID
scores.mcapTF$Clade <- mcap.samp.df.TF$Clade
scores.mcapTF$CladeTreat <- as.factor(paste(mcap.samp.df.TF$Clade,mcap.samp.df.TF$Treatment))

ggplot(scores.mcapTF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("M. capitata TF Bray Curtis") +
  theme_classic()
ggsave("Mcap_TF_its_nmds.pdf")


#ords
#C T1 and TF
set.seed(30)
ord.mcap_C <- ordinate(Mcap_DIV_C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C)
scores.mcap_C <- as.data.frame(scores(ord.mcap_C))
scores.mcap_C$Treatment <- mcap.samp.df.T1F.C$Treatment
scores.mcap_C$Parent.ID <- mcap.samp.df.T1F.C$Parent.ID
scores.mcap_C$Time.Point <- mcap.samp.df.T1F.C$Time.Point

ggplot(scores.mcap_C, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Cladocopium") +
  theme_classic()
#ggsave("McapTFits.pdf")


#C T1
set.seed(30)
ord.mcap_C.T1 <- ordinate(mcap.RelRare.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.T1)
scores.mcap_C.T1 <- as.data.frame(scores(ord.mcap_C.T1))
scores.mcap_C.T1$Treatment <- mcap.samp.df.T1.C$Treatment
scores.mcap_C.T1$Parent.ID <- mcap.samp.df.T1.C$Parent.ID
scores.mcap_C.T1$Time.Point <- mcap.samp.df.T1.C$Time.Point

ggplot(scores.mcap_C.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("McapTFits.pdf")

#C TF
set.seed(30)
ord.mcap_C.TF <- ordinate(mcap.RelRare.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_C.TF)
scores.mcap_C.TF <- as.data.frame(scores(ord.mcap_C.TF))
scores.mcap_C.TF$Treatment <- mcap.samp.df.TF.C$Treatment
scores.mcap_C.TF$Parent.ID <- mcap.samp.df.TF.C$Parent.ID
scores.mcap_C.TF$Time.Point <- mcap.samp.df.TF.C$Time.Point

ggplot(scores.mcap_C.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("McapTFits.pdf")


#CD
set.seed(30)
ord.mcap_CD <- ordinate(Mcap_DIV_CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD)
scores.mcap_CD <- as.data.frame(scores(ord.mcap_CD))
scores.mcap_CD$Treatment <- mcap.samp.df.T1F.CD$Treatment
scores.mcap_CD$Parent.ID <- mcap.samp.df.T1F.CD$Parent.ID
scores.mcap_CD$Time.Point <- mcap.samp.df.T1F.CD$Time.Point

ggplot(scores.mcap_CD, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D") +
  theme_classic()
#ggsave("Mcap_ITS2_CCD_its.pdf")

#C T1
set.seed(30)
ord.mcap_CD.T1 <- ordinate(mcap.RelRare.T1.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.T1)
scores.mcap_CD.T1 <- as.data.frame(scores(ord.mcap_CD.T1))
scores.mcap_CD.T1$Treatment <- mcap.samp.df.T1.CD$Treatment
scores.mcap_CD.T1$Parent.ID <- mcap.samp.df.T1.CD$Parent.ID
scores.mcap_CD.T1$Time.Point <- mcap.samp.df.T1.CD$Time.Point

ggplot(scores.mcap_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
ggsave("Mcap_ITS2_C_TF.pdf")

#CD TF
set.seed(30)
ord.mcap_CD.TF <- ordinate(mcap.RelRare.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.mcap_CD.TF)
scores.mcap_CD.TF <- as.data.frame(scores(ord.mcap_CD.TF))
scores.mcap_CD.TF$Treatment <- mcap.samp.df.TF.CD$Treatment
scores.mcap_CD.TF$Parent.ID <- mcap.samp.df.TF.CD$Parent.ID
scores.mcap_CD.TF$Time.Point <- mcap.samp.df.TF.CD$Time.Point

ggplot(scores.mcap_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Mcap_ITS2_CD_TF.pdf")



```

Porites - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pcomp
Pcomp.RelRare.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "montipora")
Pcomp.RelRare.T1 <- subset_samples(Pcomp.RelRareT1F, Time.Point == "T1")
Pcomp.RelRare.TF <- subset_samples(Pcomp.RelRareT1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F <- phyloseq::distance(Pcomp.RelRare.T1F, method = "bray")
  Pcomp.samp.df.T1F <- as(sample_data(Pcomp.RelRare.T1F), "data.frame")            #sample df
  adonis(bc.Pcomp.T1F ~ Clade*Treatment*Time.Point, data = Pcomp.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PcompT1 <- phyloseq::distance(Pcomp.RelRare.T1, method = "bray")
  Pcomp.samp.df.T1 <- as(sample_data(Pcomp.RelRare.T1), "data.frame") #sample df
  adonis(bc.PcompT1 ~ Clade*Treatment, data = Pcomp.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Clade)
  permutest(beta.Pcomp.T1, pairwise = T) 
  beta.Pcomp.T1 <- betadisper(bc.PcompT1, Pcomp.samp.df.T1$Treatment)
  permutest(beta.Pcomp.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pcomp.TF <- phyloseq::distance(Pcomp.RelRare.TF, method = "bray")
  Pcomp.samp.df.TF <- as(sample_data(Pcomp.RelRare.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF ~ Clade*Treatment, data = Pcomp.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Clade)
  permutest(beta.Pcomp.TF, pairwise = T) 
  beta.Pcomp.TF <- betadisper(bc.Pcomp.TF, Pcomp.samp.df.TF$Treatment)
  permutest(beta.Pcomp.TF, pairwise = T) 


## by clade: C vs CD
  
# C only, new objs
Pcomp.RelRareT1F.C <- subset_samples(Pcomp.RelRareT1F, Clade == "C")
Pcomp.RelRare.T1.C <- subset_samples(Pcomp.RelRare.T1, Clade == "C")
Pcomp.RelRare.TF.C <- subset_samples(Pcomp.RelRare.TF, Clade == "C")
  
#T1TF.C
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F.C <- phyloseq::distance(Pcomp.RelRareT1F.C, method = "bray")
  Pcomp.samp.df.T1F.C <- as(sample_data(Pcomp.RelRareT1F.C), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.T1F.C ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.C)

#T1.C
#make bray curtis data matrix 
  bc.PcompT1.C <- phyloseq::distance(Pcomp.RelRare.T1.C, method = "bray")
  Pcomp.samp.df.T1.C <- as(sample_data(Pcomp.RelRare.T1.C), "data.frame") #sample df
#adonis test
  adonis(bc.PcompT1.C ~ Treatment, data = Pcomp.samp.df.T1.C)
# Homogeneity of dispersion test  
  beta.Pcomp.T1.C <- betadisper(bc.PcompT1.C, Pcomp.samp.df.T1.C$Treatment)
  permutest(beta.Pcomp.T1.C, pairwise = T) 

#TF.C
#make bray curtis data matrix 
  bc.Pcomp.TF.C <- phyloseq::distance(Pcomp.RelRare.TF.C, method = "bray")
  Pcomp.samp.df.TF.C <- as(sample_data(Pcomp.RelRare.TF.C), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF.C ~ Treatment, data = Pcomp.samp.df.TF.C)
# Homogeneity of dispersion test  
  beta.Pcomp.TF.C <- betadisper(bc.Pcomp.TF.C, Pcomp.samp.df.TF.C$Treatment)
  permutest(beta.Pcomp.TF.C, pairwise = T) 


# CD only, new objs
Pcomp.RelRareT1F.CD <- subset_samples(Pcomp.RelRareT1F, Clade == "CD")
Pcomp.RelRare.T1.CD <- subset_samples(Pcomp.RelRare.T1, Clade == "CD")
Pcomp.RelRare.TF.CD <- subset_samples(Pcomp.RelRare.TF, Clade == "CD")
  
#T1TF.CD
#make bray curtis data matrix -T1 and TF
  bc.Pcomp.T1F.CD <- phyloseq::distance(Pcomp.RelRareT1F.CD, method = "bray")
  Pcomp.samp.df.T1F.CD <- as(sample_data(Pcomp.RelRareT1F.CD), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.T1F.CD ~ Treatment*Time.Point, data = Pcomp.samp.df.T1F.CD)

#T1.CD
#make bray curtis data matrix 
  bc.PcompT1.CD <- phyloseq::distance(Pcomp.RelRare.T1.CD, method = "bray")
  Pcomp.samp.df.T1.CD <- as(sample_data(Pcomp.RelRare.T1.CD), "data.frame") #sample df
  adonis(bc.PcompT1.CD ~ Treatment, data = Pcomp.samp.df.T1.CD)#adonis test
# Homogeneity of dispersion test  
  beta.Pcomp.T1.CD <- betadisper(bc.PcompT1.CD, Pcomp.samp.df.T1.CD$Treatment)
  permutest(beta.Pcomp.T1.CD, pairwise = T) 

#TF.CD
#make bray curtis data matrix 
  bc.Pcomp.TF.CD <- phyloseq::distance(Pcomp.RelRare.TF.CD, method = "bray")
  Pcomp.samp.df.TF.CD <- as(sample_data(Pcomp.RelRare.TF.CD), "data.frame") #sample df
#adonis test
  adonis(bc.Pcomp.TF.CD ~ Treatment, data = Pcomp.samp.df.TF.CD)
# Homogeneity of dispersion test  
  beta.Pcomp.TF.CD <- betadisper(bc.Pcomp.TF.CD, Pcomp.samp.df.TF.CD$Treatment)
  permutest(beta.Pcomp.TF.CD, pairwise = T) 
``` 
Pcomp NMDS
```{r}
library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pcomp.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "honeydew4", "372" = "khaki4","373" ="ivory4", "374" = "lightblue", Ambient = "Blue", High = "Red","C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred") 

#T1 and TF
ord.Pcomp <- ordinate(Pcomp.RelRareT1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pcomp)

scores.Pcomp.samp.df <- as.data.frame(scores(ord.Pcomp) ) 
scores.Pcomp.samp.df$Time.Point <- Pcomp.samp.df.T1F$Time.Point
scores.Pcomp.samp.df$Parent.ID <- Pcomp.samp.df.T1F$Parent.ID
scores.Pcomp.samp.df$Treatment <- Pcomp.samp.df.T1F$Treatment
scores.Pcomp.samp.df$Clade <- Pcomp.samp.df.T1F$Clade


ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Clade), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PcompT1its2.pdf")
ggplot(scores.Pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PcompT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pcomp.RelRare.TF, ordinate(Pcomp.RelRare, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PcompT1 <- ordinate(Pcomp.RelRare.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompT1)
scores.PcompT1 <- as.data.frame(scores(ord.PcompT1))
scores.PcompT1$Treatment <- Pcomp.samp.df.T1$Treatment
scores.PcompT1$Parent.ID <- Pcomp.samp.df.T1$Parent.ID
scores.PcompT1$Clade <- Pcomp.samp.df.T1$Clade
scores.PcompT1$CladeTreat <- as.factor(paste(Pcomp.samp.df.T1$Clade,Pcomp.samp.df.T1$Treatment))


ggplot(scores.PcompT1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("M. capitata T1 Bray Curtis") +
  theme_classic()
ggsave("PcompT1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PcompTF <- ordinate(Pcomp.RelRare.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PcompTF)
scores.PcompTF <- as.data.frame(scores(ord.PcompTF))
scores.PcompTF$Treatment <- Pcomp.samp.df.TF$Treatment
scores.PcompTF$Parent.ID <- Pcomp.samp.df.TF$Parent.ID
scores.PcompTF$Clade <- Pcomp.samp.df.TF$Clade
scores.PcompTF$CladeTreat <- as.factor(paste(Pcomp.samp.df.TF$Clade,Pcomp.samp.df.TF$Treatment))

ggplot(scores.PcompTF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("M. capitata TF Bray Curtis") +
  theme_classic()
ggsave("Pcomp_TF_its_nmds.pdf")


#ords
#C T1 and TF
set.seed(30)
ord.Pcomp_C <- ordinate(Pcomp_DIV_C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_C)
scores.Pcomp_C <- as.data.frame(scores(ord.Pcomp_C))
scores.Pcomp_C$Treatment <- Pcomp.samp.df.T1F.C$Treatment
scores.Pcomp_C$Parent.ID <- Pcomp.samp.df.T1F.C$Parent.ID
scores.Pcomp_C$Time.Point <- Pcomp.samp.df.T1F.C$Time.Point

ggplot(scores.Pcomp_C, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Cladocopium") +
  theme_classic()
#ggsave("PcompTFits.pdf")


#C T1
set.seed(30)
ord.Pcomp_C.T1 <- ordinate(Pcomp.RelRare.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_C.T1)
scores.Pcomp_C.T1 <- as.data.frame(scores(ord.Pcomp_C.T1))
scores.Pcomp_C.T1$Treatment <- Pcomp.samp.df.T1.C$Treatment
scores.Pcomp_C.T1$Parent.ID <- Pcomp.samp.df.T1.C$Parent.ID
scores.Pcomp_C.T1$Time.Point <- Pcomp.samp.df.T1.C$Time.Point

ggplot(scores.Pcomp_C.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("PcompTFits.pdf")

#C TF
set.seed(30)
ord.Pcomp_C.TF <- ordinate(Pcomp.RelRare.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_C.TF)
scores.Pcomp_C.TF <- as.data.frame(scores(ord.Pcomp_C.TF))
scores.Pcomp_C.TF$Treatment <- Pcomp.samp.df.TF.C$Treatment
scores.Pcomp_C.TF$Parent.ID <- Pcomp.samp.df.TF.C$Parent.ID
scores.Pcomp_C.TF$Time.Point <- Pcomp.samp.df.TF.C$Time.Point

ggplot(scores.Pcomp_C.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("PcompTFits.pdf")


#CD
set.seed(30)
ord.Pcomp_CD <- ordinate(Pcomp_DIV_CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_CD)
scores.Pcomp_CD <- as.data.frame(scores(ord.Pcomp_CD))
scores.Pcomp_CD$Treatment <- Pcomp.samp.df.T1F.CD$Treatment
scores.Pcomp_CD$Parent.ID <- Pcomp.samp.df.T1F.CD$Parent.ID
scores.Pcomp_CD$Time.Point <- Pcomp.samp.df.T1F.CD$Time.Point

ggplot(scores.Pcomp_CD, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D") +
  theme_classic()
#ggsave("Pcomp_ITS2_CCD_its.pdf")

#C T1
set.seed(30)
ord.Pcomp_CD.T1 <- ordinate(Pcomp.RelRare.T1.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_CD.T1)
scores.Pcomp_CD.T1 <- as.data.frame(scores(ord.Pcomp_CD.T1))
scores.Pcomp_CD.T1$Treatment <- Pcomp.samp.df.T1.CD$Treatment
scores.Pcomp_CD.T1$Parent.ID <- Pcomp.samp.df.T1.CD$Parent.ID
scores.Pcomp_CD.T1$Time.Point <- Pcomp.samp.df.T1.CD$Time.Point

ggplot(scores.Pcomp_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
ggsave("Pcomp_ITS2_C_TF.pdf")

#CD TF
set.seed(30)
ord.Pcomp_CD.TF <- ordinate(Pcomp.RelRare.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pcomp_CD.TF)
scores.Pcomp_CD.TF <- as.data.frame(scores(ord.Pcomp_CD.TF))
scores.Pcomp_CD.TF$Treatment <- Pcomp.samp.df.TF.CD$Treatment
scores.Pcomp_CD.TF$Parent.ID <- Pcomp.samp.df.TF.CD$Parent.ID
scores.Pcomp_CD.TF$Time.Point <- Pcomp.samp.df.TF.CD$Time.Point

ggplot(scores.Pcomp_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Pcomp_ITS2_CD_TF.pdf")



```
Pavona - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pvar
Pvar.RelRare.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "montipora")
Pvar.RelRare.T1 <- subset_samples(Pvar.RelRareT1F, Time.Point == "T1")
Pvar.RelRare.TF <- subset_samples(Pvar.RelRareT1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F <- phyloseq::distance(Pvar.RelRare.T1F, method = "bray")
  Pvar.samp.df.T1F <- as(sample_data(Pvar.RelRare.T1F), "data.frame")            #sample df
  adonis(bc.Pvar.T1F ~ Clade*Treatment*Time.Point, data = Pvar.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PvarT1 <- phyloseq::distance(Pvar.RelRare.T1, method = "bray")
  Pvar.samp.df.T1 <- as(sample_data(Pvar.RelRare.T1), "data.frame") #sample df
  adonis(bc.PvarT1 ~ Clade*Treatment, data = Pvar.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Clade)
  permutest(beta.Pvar.T1, pairwise = T) 
  beta.Pvar.T1 <- betadisper(bc.PvarT1, Pvar.samp.df.T1$Treatment)
  permutest(beta.Pvar.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pvar.TF <- phyloseq::distance(Pvar.RelRare.TF, method = "bray")
  Pvar.samp.df.TF <- as(sample_data(Pvar.RelRare.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF ~ Clade*Treatment, data = Pvar.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Clade)
  permutest(beta.Pvar.TF, pairwise = T) 
  beta.Pvar.TF <- betadisper(bc.Pvar.TF, Pvar.samp.df.TF$Treatment)
  permutest(beta.Pvar.TF, pairwise = T) 


## by clade: C vs CD
  
# C only, new objs
Pvar.RelRareT1F.C <- subset_samples(Pvar.RelRareT1F, Clade == "C")
Pvar.RelRare.T1.C <- subset_samples(Pvar.RelRare.T1, Clade == "C")
Pvar.RelRare.TF.C <- subset_samples(Pvar.RelRare.TF, Clade == "C")
  
#T1TF.C
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F.C <- phyloseq::distance(Pvar.RelRareT1F.C, method = "bray")
  Pvar.samp.df.T1F.C <- as(sample_data(Pvar.RelRareT1F.C), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.T1F.C ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.C)

#T1.C
#make bray curtis data matrix 
  bc.PvarT1.C <- phyloseq::distance(Pvar.RelRare.T1.C, method = "bray")
  Pvar.samp.df.T1.C <- as(sample_data(Pvar.RelRare.T1.C), "data.frame") #sample df
#adonis test
  adonis(bc.PvarT1.C ~ Treatment, data = Pvar.samp.df.T1.C)
# Homogeneity of dispersion test  
  beta.Pvar.T1.C <- betadisper(bc.PvarT1.C, Pvar.samp.df.T1.C$Treatment)
  permutest(beta.Pvar.T1.C, pairwise = T) 

#TF.C
#make bray curtis data matrix 
  bc.Pvar.TF.C <- phyloseq::distance(Pvar.RelRare.TF.C, method = "bray")
  Pvar.samp.df.TF.C <- as(sample_data(Pvar.RelRare.TF.C), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF.C ~ Treatment, data = Pvar.samp.df.TF.C)
# Homogeneity of dispersion test  
  beta.Pvar.TF.C <- betadisper(bc.Pvar.TF.C, Pvar.samp.df.TF.C$Treatment)
  permutest(beta.Pvar.TF.C, pairwise = T) 


# CD only, new objs
Pvar.RelRareT1F.CD <- subset_samples(Pvar.RelRareT1F, Clade == "CD")
Pvar.RelRare.T1.CD <- subset_samples(Pvar.RelRare.T1, Clade == "CD")
Pvar.RelRare.TF.CD <- subset_samples(Pvar.RelRare.TF, Clade == "CD")
  
#T1TF.CD
#make bray curtis data matrix -T1 and TF
  bc.Pvar.T1F.CD <- phyloseq::distance(Pvar.RelRareT1F.CD, method = "bray")
  Pvar.samp.df.T1F.CD <- as(sample_data(Pvar.RelRareT1F.CD), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.T1F.CD ~ Treatment*Time.Point, data = Pvar.samp.df.T1F.CD)

#T1.CD
#make bray curtis data matrix 
  bc.PvarT1.CD <- phyloseq::distance(Pvar.RelRare.T1.CD, method = "bray")
  Pvar.samp.df.T1.CD <- as(sample_data(Pvar.RelRare.T1.CD), "data.frame") #sample df
  adonis(bc.PvarT1.CD ~ Treatment, data = Pvar.samp.df.T1.CD)#adonis test
# Homogeneity of dispersion test  
  beta.Pvar.T1.CD <- betadisper(bc.PvarT1.CD, Pvar.samp.df.T1.CD$Treatment)
  permutest(beta.Pvar.T1.CD, pairwise = T) 

#TF.CD
#make bray curtis data matrix 
  bc.Pvar.TF.CD <- phyloseq::distance(Pvar.RelRare.TF.CD, method = "bray")
  Pvar.samp.df.TF.CD <- as(sample_data(Pvar.RelRare.TF.CD), "data.frame") #sample df
#adonis test
  adonis(bc.Pvar.TF.CD ~ Treatment, data = Pvar.samp.df.TF.CD)
# Homogeneity of dispersion test  
  beta.Pvar.TF.CD <- betadisper(bc.Pvar.TF.CD, Pvar.samp.df.TF.CD$Treatment)
  permutest(beta.Pvar.TF.CD, pairwise = T) 
``` 
Pvar NMDS
```{r}
library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pvar.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "honeydew4", "372" = "khaki4","373" ="ivory4", "374" = "lightblue", Ambient = "Blue", High = "Red","C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred") 

#T1 and TF
ord.Pvar <- ordinate(Pvar.RelRareT1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pvar)

scores.Pvar.samp.df <- as.data.frame(scores(ord.Pvar) ) 
scores.Pvar.samp.df$Time.Point <- Pvar.samp.df.T1F$Time.Point
scores.Pvar.samp.df$Parent.ID <- Pvar.samp.df.T1F$Parent.ID
scores.Pvar.samp.df$Treatment <- Pvar.samp.df.T1F$Treatment
scores.Pvar.samp.df$Clade <- Pvar.samp.df.T1F$Clade


ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Clade), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PvarT1its2.pdf")
ggplot(scores.Pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PvarT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pvar.RelRare.TF, ordinate(Pvar.RelRare, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PvarT1 <- ordinate(Pvar.RelRare.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PvarT1)
scores.PvarT1 <- as.data.frame(scores(ord.PvarT1))
scores.PvarT1$Treatment <- Pvar.samp.df.T1$Treatment
scores.PvarT1$Parent.ID <- Pvar.samp.df.T1$Parent.ID
scores.PvarT1$Clade <- Pvar.samp.df.T1$Clade
scores.PvarT1$CladeTreat <- as.factor(paste(Pvar.samp.df.T1$Clade,Pvar.samp.df.T1$Treatment))


ggplot(scores.PvarT1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("M. capitata T1 Bray Curtis") +
  theme_classic()
ggsave("PvarT1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PvarTF <- ordinate(Pvar.RelRare.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PvarTF)
scores.PvarTF <- as.data.frame(scores(ord.PvarTF))
scores.PvarTF$Treatment <- Pvar.samp.df.TF$Treatment
scores.PvarTF$Parent.ID <- Pvar.samp.df.TF$Parent.ID
scores.PvarTF$Clade <- Pvar.samp.df.TF$Clade
scores.PvarTF$CladeTreat <- as.factor(paste(Pvar.samp.df.TF$Clade,Pvar.samp.df.TF$Treatment))

ggplot(scores.PvarTF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("M. capitata TF Bray Curtis") +
  theme_classic()
ggsave("Pvar_TF_its_nmds.pdf")


#ords
#C T1 and TF
set.seed(30)
ord.Pvar_C <- ordinate(Pvar_DIV_C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C)
scores.Pvar_C <- as.data.frame(scores(ord.Pvar_C))
scores.Pvar_C$Treatment <- Pvar.samp.df.T1F.C$Treatment
scores.Pvar_C$Parent.ID <- Pvar.samp.df.T1F.C$Parent.ID
scores.Pvar_C$Time.Point <- Pvar.samp.df.T1F.C$Time.Point

ggplot(scores.Pvar_C, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Cladocopium") +
  theme_classic()
#ggsave("PvarTFits.pdf")


#C T1
set.seed(30)
ord.Pvar_C.T1 <- ordinate(Pvar.RelRare.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C.T1)
scores.Pvar_C.T1 <- as.data.frame(scores(ord.Pvar_C.T1))
scores.Pvar_C.T1$Treatment <- Pvar.samp.df.T1.C$Treatment
scores.Pvar_C.T1$Parent.ID <- Pvar.samp.df.T1.C$Parent.ID
scores.Pvar_C.T1$Time.Point <- Pvar.samp.df.T1.C$Time.Point

ggplot(scores.Pvar_C.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("PvarTFits.pdf")

#C TF
set.seed(30)
ord.Pvar_C.TF <- ordinate(Pvar.RelRare.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_C.TF)
scores.Pvar_C.TF <- as.data.frame(scores(ord.Pvar_C.TF))
scores.Pvar_C.TF$Treatment <- Pvar.samp.df.TF.C$Treatment
scores.Pvar_C.TF$Parent.ID <- Pvar.samp.df.TF.C$Parent.ID
scores.Pvar_C.TF$Time.Point <- Pvar.samp.df.TF.C$Time.Point

ggplot(scores.Pvar_C.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("PvarTFits.pdf")


#CD
set.seed(30)
ord.Pvar_CD <- ordinate(Pvar_DIV_CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CD)
scores.Pvar_CD <- as.data.frame(scores(ord.Pvar_CD))
scores.Pvar_CD$Treatment <- Pvar.samp.df.T1F.CD$Treatment
scores.Pvar_CD$Parent.ID <- Pvar.samp.df.T1F.CD$Parent.ID
scores.Pvar_CD$Time.Point <- Pvar.samp.df.T1F.CD$Time.Point

ggplot(scores.Pvar_CD, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D") +
  theme_classic()
#ggsave("Pvar_ITS2_CCD_its.pdf")

#C T1
set.seed(30)
ord.Pvar_CD.T1 <- ordinate(Pvar.RelRare.T1.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CD.T1)
scores.Pvar_CD.T1 <- as.data.frame(scores(ord.Pvar_CD.T1))
scores.Pvar_CD.T1$Treatment <- Pvar.samp.df.T1.CD$Treatment
scores.Pvar_CD.T1$Parent.ID <- Pvar.samp.df.T1.CD$Parent.ID
scores.Pvar_CD.T1$Time.Point <- Pvar.samp.df.T1.CD$Time.Point

ggplot(scores.Pvar_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
ggsave("Pvar_ITS2_C_TF.pdf")

#CD TF
set.seed(30)
ord.Pvar_CD.TF <- ordinate(Pvar.RelRare.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pvar_CD.TF)
scores.Pvar_CD.TF <- as.data.frame(scores(ord.Pvar_CD.TF))
scores.Pvar_CD.TF$Treatment <- Pvar.samp.df.TF.CD$Treatment
scores.Pvar_CD.TF$Parent.ID <- Pvar.samp.df.TF.CD$Parent.ID
scores.Pvar_CD.TF$Time.Point <- Pvar.samp.df.TF.CD$Time.Point

ggplot(scores.Pvar_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Pvar_ITS2_CD_TF.pdf")



```

Pocillopora - T1 and TF together, then independent. PERMANOVA
```{r}   
#DIV_RelA_stats.T1F
#Pacu
Pacu.RelRare.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pocillopora")
Pacu.RelRare.T1 <- subset_samples(Pacu.RelRareT1F, Time.Point == "T1")
Pacu.RelRare.TF <- subset_samples(Pacu.RelRareT1F, Time.Point == "TF")

#T1TF
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F <- phyloseq::distance(Pacu.RelRare.T1F, method = "bray")
  Pacu.samp.df.T1F <- as(sample_data(Pacu.RelRare.T1F), "data.frame")            #sample df
  adonis(bc.Pacu.T1F ~ Treatment*Time.Point, data = Pacu.samp.df.T1F)      #adonis test

#T1
#make bray curtis data matrix 
  bc.PacuT1 <- phyloseq::distance(Pacu.RelRare.T1, method = "bray")
  Pacu.samp.df.T1 <- as(sample_data(Pacu.RelRare.T1), "data.frame") #sample df
  adonis(bc.PacuT1 ~ Clade*Treatment, data = Pacu.samp.df.T1)   #adonis test
# Homogeneity of dispersion test  
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Clade)
  permutest(beta.Pacu.T1, pairwise = T) 
  beta.Pacu.T1 <- betadisper(bc.PacuT1, Pacu.samp.df.T1$Treatment)
  permutest(beta.Pacu.T1, pairwise = T) 

#TF
#make bray curtis data matrix 
  bc.Pacu.TF <- phyloseq::distance(Pacu.RelRare.TF, method = "bray")
  Pacu.samp.df.TF <- as(sample_data(Pacu.RelRare.TF), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF ~ Clade*Treatment, data = Pacu.samp.df.TF)
  # Homogeneity of dispersion test  
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Clade)
  permutest(beta.Pacu.TF, pairwise = T) 
  beta.Pacu.TF <- betadisper(bc.Pacu.TF, Pacu.samp.df.TF$Treatment)
  permutest(beta.Pacu.TF, pairwise = T) 


## by clade: C vs CD
  
# C only, new objs
Pacu.RelRareT1F.C <- subset_samples(Pacu.RelRareT1F, Clade == "C")
Pacu.RelRare.T1.C <- subset_samples(Pacu.RelRare.T1, Clade == "C")
Pacu.RelRare.TF.C <- subset_samples(Pacu.RelRare.TF, Clade == "C")
  
#T1TF.C
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F.C <- phyloseq::distance(Pacu.RelRareT1F.C, method = "bray")
  Pacu.samp.df.T1F.C <- as(sample_data(Pacu.RelRareT1F.C), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.T1F.C ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.C)

#T1.C
#make bray curtis data matrix 
  bc.PacuT1.C <- phyloseq::distance(Pacu.RelRare.T1.C, method = "bray")
  Pacu.samp.df.T1.C <- as(sample_data(Pacu.RelRare.T1.C), "data.frame") #sample df
#adonis test
  adonis(bc.PacuT1.C ~ Treatment, data = Pacu.samp.df.T1.C)
# Homogeneity of dispersion test  
  beta.Pacu.T1.C <- betadisper(bc.PacuT1.C, Pacu.samp.df.T1.C$Treatment)
  permutest(beta.Pacu.T1.C, pairwise = T) 

#TF.C
#make bray curtis data matrix 
  bc.Pacu.TF.C <- phyloseq::distance(Pacu.RelRare.TF.C, method = "bray")
  Pacu.samp.df.TF.C <- as(sample_data(Pacu.RelRare.TF.C), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF.C ~ Treatment, data = Pacu.samp.df.TF.C)
# Homogeneity of dispersion test  
  beta.Pacu.TF.C <- betadisper(bc.Pacu.TF.C, Pacu.samp.df.TF.C$Treatment)
  permutest(beta.Pacu.TF.C, pairwise = T) 


# CD only, new objs
Pacu.RelRareT1F.CD <- subset_samples(Pacu.RelRareT1F, Clade == "CD")
Pacu.RelRare.T1.CD <- subset_samples(Pacu.RelRare.T1, Clade == "CD")
Pacu.RelRare.TF.CD <- subset_samples(Pacu.RelRare.TF, Clade == "CD")
  
#T1TF.CD
#make bray curtis data matrix -T1 and TF
  bc.Pacu.T1F.CD <- phyloseq::distance(Pacu.RelRareT1F.CD, method = "bray")
  Pacu.samp.df.T1F.CD <- as(sample_data(Pacu.RelRareT1F.CD), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.T1F.CD ~ Treatment*Time.Point, data = Pacu.samp.df.T1F.CD)

#T1.CD
#make bray curtis data matrix 
  bc.PacuT1.CD <- phyloseq::distance(Pacu.RelRare.T1.CD, method = "bray")
  Pacu.samp.df.T1.CD <- as(sample_data(Pacu.RelRare.T1.CD), "data.frame") #sample df
  adonis(bc.PacuT1.CD ~ Treatment, data = Pacu.samp.df.T1.CD)#adonis test
# Homogeneity of dispersion test  
  beta.Pacu.T1.CD <- betadisper(bc.PacuT1.CD, Pacu.samp.df.T1.CD$Treatment)
  permutest(beta.Pacu.T1.CD, pairwise = T) 

#TF.CD
#make bray curtis data matrix 
  bc.Pacu.TF.CD <- phyloseq::distance(Pacu.RelRare.TF.CD, method = "bray")
  Pacu.samp.df.TF.CD <- as(sample_data(Pacu.RelRare.TF.CD), "data.frame") #sample df
#adonis test
  adonis(bc.Pacu.TF.CD ~ Treatment, data = Pacu.samp.df.TF.CD)
# Homogeneity of dispersion test  
  beta.Pacu.TF.CD <- betadisper(bc.Pacu.TF.CD, Pacu.samp.df.TF.CD$Treatment)
  permutest(beta.Pacu.TF.CD, pairwise = T) 
``` 
Pacu NMDS
```{r}
library(RColorBrewer)
library(colorRamps)
library(devtools)

## create palette for parent.ID and treatments
Pacu.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "honeydew4", "372" = "khaki4","373" ="ivory4", "374" = "lightblue", Ambient = "Blue", High = "Red","C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred") 

#T1 and TF
ord.Pacu <- ordinate(Pacu.RelRareT1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.Pacu)

scores.Pacu.samp.df <- as.data.frame(scores(ord.Pacu) ) 
scores.Pacu.samp.df$Time.Point <- Pacu.samp.df.T1F$Time.Point
scores.Pacu.samp.df$Parent.ID <- Pacu.samp.df.T1F$Parent.ID
scores.Pacu.samp.df$Treatment <- Pacu.samp.df.T1F$Treatment
scores.Pacu.samp.df$Clade <- Pacu.samp.df.T1F$Clade


ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Clade), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PacuT1its2.pdf")
ggplot(scores.Pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Bray Curtis") +
  theme_classic()
#ggsave("PacuT1its2.pdf")

#workaround-for when this randomly breaks
# plot_ordination(Pacu.RelRare.TF, ordinate(Pacu.RelRare, "MDS", set.seed=(30)), color = "Treatment", shape="Clade") + geom_point(size = 5)



#T1 only
set.seed(30)
ord.PacuT1 <- ordinate(Pacu.RelRare.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuT1)
scores.PacuT1 <- as.data.frame(scores(ord.PacuT1))
scores.PacuT1$Treatment <- Pacu.samp.df.T1$Treatment
scores.PacuT1$Parent.ID <- Pacu.samp.df.T1$Parent.ID
scores.PacuT1$Clade <- Pacu.samp.df.T1$Clade
scores.PacuT1$CladeTreat <- as.factor(paste(Pacu.samp.df.T1$Clade,Pacu.samp.df.T1$Treatment))


ggplot(scores.PacuT1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("M. capitata T1 Bray Curtis") +
  theme_classic()
ggsave("PacuT1_its2_nmds_clade.pdf")

#TF only
set.seed(30)
ord.PacuTF <- ordinate(Pacu.RelRare.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.PacuTF)
scores.PacuTF <- as.data.frame(scores(ord.PacuTF))
scores.PacuTF$Treatment <- Pacu.samp.df.TF$Treatment
scores.PacuTF$Parent.ID <- Pacu.samp.df.TF$Parent.ID
scores.PacuTF$Clade <- Pacu.samp.df.TF$Clade
scores.PacuTF$CladeTreat <- as.factor(paste(Pacu.samp.df.TF$Clade,Pacu.samp.df.TF$Treatment))

ggplot(scores.PacuTF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = CladeTreat), linetype = 2) +
  ggtitle("M. capitata TF Bray Curtis") +
  theme_classic()
ggsave("Pacu_TF_its_nmds.pdf")


#ords
#C T1 and TF
set.seed(30)
ord.Pacu_C <- ordinate(Pacu_DIV_C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C)
scores.Pacu_C <- as.data.frame(scores(ord.Pacu_C))
scores.Pacu_C$Treatment <- Pacu.samp.df.T1F.C$Treatment
scores.Pacu_C$Parent.ID <- Pacu.samp.df.T1F.C$Parent.ID
scores.Pacu_C$Time.Point <- Pacu.samp.df.T1F.C$Time.Point

ggplot(scores.Pacu_C, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata Cladocopium") +
  theme_classic()
#ggsave("PacuTFits.pdf")


#C T1
set.seed(30)
ord.Pacu_C.T1 <- ordinate(Pacu.RelRare.T1.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C.T1)
scores.Pacu_C.T1 <- as.data.frame(scores(ord.Pacu_C.T1))
scores.Pacu_C.T1$Treatment <- Pacu.samp.df.T1.C$Treatment
scores.Pacu_C.T1$Parent.ID <- Pacu.samp.df.T1.C$Parent.ID
scores.Pacu_C.T1$Time.Point <- Pacu.samp.df.T1.C$Time.Point

ggplot(scores.Pacu_C.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 Cladocopium") +
  theme_classic()
#ggsave("PacuTFits.pdf")

#C TF
set.seed(30)
ord.Pacu_C.TF <- ordinate(Pacu.RelRare.TF.C, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_C.TF)
scores.Pacu_C.TF <- as.data.frame(scores(ord.Pacu_C.TF))
scores.Pacu_C.TF$Treatment <- Pacu.samp.df.TF.C$Treatment
scores.Pacu_C.TF$Parent.ID <- Pacu.samp.df.TF.C$Parent.ID
scores.Pacu_C.TF$Time.Point <- Pacu.samp.df.TF.C$Time.Point

ggplot(scores.Pacu_C.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C TF") +
  theme_classic()
#ggsave("PacuTFits.pdf")


#CD
set.seed(30)
ord.Pacu_CD <- ordinate(Pacu_DIV_CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CD)
scores.Pacu_CD <- as.data.frame(scores(ord.Pacu_CD))
scores.Pacu_CD$Treatment <- Pacu.samp.df.T1F.CD$Treatment
scores.Pacu_CD$Parent.ID <- Pacu.samp.df.T1F.CD$Parent.ID
scores.Pacu_CD$Time.Point <- Pacu.samp.df.T1F.CD$Time.Point

ggplot(scores.Pacu_CD, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Time.Point), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata C&D") +
  theme_classic()
#ggsave("Pacu_ITS2_CCD_its.pdf")

#C T1
set.seed(30)
ord.Pacu_CD.T1 <- ordinate(Pacu.RelRare.T1.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CD.T1)
scores.Pacu_CD.T1 <- as.data.frame(scores(ord.Pacu_CD.T1))
scores.Pacu_CD.T1$Treatment <- Pacu.samp.df.T1.CD$Treatment
scores.Pacu_CD.T1$Parent.ID <- Pacu.samp.df.T1.CD$Parent.ID
scores.Pacu_CD.T1$Time.Point <- Pacu.samp.df.T1.CD$Time.Point

ggplot(scores.Pacu_CD.T1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata T1 CD") +
  theme_classic()
ggsave("Pacu_ITS2_C_TF.pdf")

#CD TF
set.seed(30)
ord.Pacu_CD.TF <- ordinate(Pacu.RelRare.TF.CD, "NMDS", "bray", trymax = 100) 
stressplot(ord.Pacu_CD.TF)
scores.Pacu_CD.TF <- as.data.frame(scores(ord.Pacu_CD.TF))
scores.Pacu_CD.TF$Treatment <- Pacu.samp.df.TF.CD$Treatment
scores.Pacu_CD.TF$Parent.ID <- Pacu.samp.df.TF.CD$Parent.ID
scores.Pacu_CD.TF$Time.Point <- Pacu.samp.df.TF.CD$Time.Point

ggplot(scores.Pacu_CD.TF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata CD TF") +
  theme_classic()
ggsave("Pacu_ITS2_CD_TF.pdf")



```

CUT when above updated 
Porites its2 OLD not by 'clade'
```{r} 
#pcomp 
pcomp.RelRare <- subset_samples(DIV_RelA_stats, host_genus == "porites")

#all speices
set.seed(30) 

#make bray curtis data matrix
bc.pcomp <- phyloseq::distance(pcomp.RelRare, method = "bray")
  pcomp.samp.df <- as(sample_data(pcomp.RelRare), "data.frame") #sample df
#adonis test
adonis(bc.pcomp ~ Treatment*Time.Point, data = pcomp.samp.df)

# Homogeneity of dispersion test  
beta.pcomp <- betadisper(bc.pcomp, pcomp.samp.df$Time.Point)
permutest(beta.pcomp)  #p = 0.996
permutest(beta.pcomp, pairwise = T) #.991

(bc.pcomp.HSD <- TukeyHSD(beta.pcomp)) #do you need?
plot(bc.pcomp.HSD)


# pcomp by timepoint: T1
pcomp.RelRare.T1 <- subset_samples(pcomp.RelRare, Time.Point == "T1")

set.seed(30)

#make bray curtis data matrix
bc.pcomp.T1 <- phyloseq::distance(pcomp.RelRare.T1, method = "bray")
  pcomp.samp.df.T1 <- as(sample_data(pcomp.RelRare.T1), "data.frame") #sample df
#adonis test
adonis(bc.pcomp.T1 ~ Treatment, data = pcomp.samp.df.T1) # p=.647
# Homogeneity of dispersion test 
beta.pcomp <- betadisper(bc.pcomp.T1, pcomp.samp.df.T1$Treatment)
permutest(beta.pcomp)  
permutest(beta.pcomp, pairwise = T) 

# pcomp by timepoint: TF
pcomp.RelRare.TF <- subset_samples(pcomp.RelRare, Time.Point == "TF")

set.seed(30)

#make bray curtis data matrix
bc.pcomp.TF <- phyloseq::distance(pcomp.RelRare.TF, method = "bray")
  pcomp.samp.df.TF <- as(sample_data(pcomp.RelRare.TF), "data.frame") #sample df
#adonis test
adonis(bc.pcomp.TF ~ Treatment, data = pcomp.samp.df.TF) # p=.907
# Homogeneity of dispersion test 
beta.pcomp <- betadisper(bc.pcomp.TF, pcomp.samp.df.TF$Treatment)
permutest(beta.pcomp)  
permutest(beta.pcomp, pairwise = T) 



#make bray curtis data matrix
bc.pcomp.High <- phyloseq::distance(pcomp.RelRare.High, method = "bray")
  pcomp.samp.df.High <- as(sample_data(pcomp.RelRare.High), "data.frame") #sample df
#adonis test
adonis(bc.pcomp.High ~ Time.Point, data = pcomp.samp.df.High) # p=.414
 


## create palette for parent.ID and treatments
pcomp.Parent.colors <- c("313" = "green4",  "314" = "firebrick3", "315" ="deeppink4", "316" = "orange", "317" = "green1","318" = "gold",  "319"= "hotpink", "320" ="darkslateblue", "321" = "honeydew4", "322" = "khaki4","323" ="ivory4", "324" = "lightblue", Ambient = "Blue", High = "Red") 



#T1 only
set.seed(30)
ord.pcompT1 <- ordinate(pcomp.RelRare.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcompT1)
scores.pcompT1 <- as.data.frame(scores(ord.pcompT1))
scores.pcompT1$Treatment <- pcomp.samp.df.T1$Treatment
scores.pcompT1$Parent.ID <- pcomp.samp.df.T1$Parent.ID

ggplot(scores.pcompT1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. compressa T1 Bray Curtis") +
  theme_classic()
ggsave("pcompT1its2.pdf")

#TF only
set.seed(30)
ord.pcompTF <- ordinate(pcomp.RelRare.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pcompTF)
scores.pcompTF <- as.data.frame(scores(ord.pcompTF))
scores.pcompTF$Treatment <- pcomp.samp.df.TF$Treatment
scores.pcompTF$Parent.ID <- pcomp.samp.df.TF$Parent.ID

ggplot(scores.pcompTF, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. comp TF Bray Curtis") +
  theme_classic()
ggsave("pcompTFits.pdf")


```
Pavona its2 OLD not by 'clade'
```{r}
#pvar 
pvar.RelRare <- subset_samples(DIV_RelA_stats, host_genus == "pavona")

#all speices
set.seed(30)

#make bray curtis data matrix
bc.pvar <- phyloseq::distance(pvar.RelRare, method = "bray")
  pvar.samp.df <- as(sample_data(pvar.RelRare), "data.frame") #sample df
#adonis test
adonis(bc.pvar ~ Treatment*Time.Point, data = pvar.samp.df)

#                     Df SumsOfSqs  MeanSqs F.Model      R2 Pr(>F)
#Treatment             1    0.0850 0.084965 0.33204 0.00625  0.659
#Time.Point            2    0.0220 0.011013 0.04304 0.00162  0.994
#Treatment:Time.Point  2    0.1781 0.089055 0.34802 0.01310  0.791
#Residuals            52   13.3062 0.255889         0.97902       
#Total                57   13.5913                  1.00000       

# Homogeneity of dispersion test  <- do I need to do this if ^^ not significant?
beta.pvar <- betadisper(bc.pvar, pvar.samp.df$Treatment)
permutest(beta.pvar)  #p = 0.921
permutest(beta.pvar, pairwise = T)  

(bc.pvar.HSD <- TukeyHSD(beta.pvar)) #do you need?
plot(bc.pvar.HSD)


# pvar by timepoint: T1
pvar.RelRare.T1 <- subset_samples(pvar.RelRare, Time.Point == "T1")

set.seed(30)

#make bray curtis data matrix
bc.pvar.T1 <- phyloseq::distance(pvar.RelRare.T1, method = "bray")
  pvar.samp.df.T1 <- as(sample_data(pvar.RelRare.T1), "data.frame") #sample df
#adonis test
adonis(bc.pvar.T1 ~ Treatment, data = pvar.samp.df.T1) # p=.937

# pvar by timepoint: TF
pvar.RelRare.TF <- subset_samples(pvar.RelRare, Time.Point == "TF")

set.seed(30)

#make bray curtis data matrix
bc.pvar.TF <- phyloseq::distance(pvar.RelRare.TF, method = "bray")
  pvar.samp.df.TF <- as(sample_data(pvar.RelRare.TF), "data.frame") #sample df
#adonis test
adonis(bc.pvar.TF ~ Treatment, data = pvar.samp.df.TF) # p=.948


# High treatment over time
pvar.RelRare.High <- subset_samples(pvar.RelRare, Treatment == "High")

set.seed(30)

#make bray curtis data matrix
bc.pvar.High <- phyloseq::distance(pvar.RelRare.High, method = "bray")
  pvar.samp.df.High <- as(sample_data(pvar.RelRare.High), "data.frame") #sample df
#adonis test
adonis(bc.pvar.High ~ Time.Point, data = pvar.samp.df.High) # p=.955

## create palette for parent.ID and treatments
pvar.Parent.colors <- c("325" = "green4",  "351" = "firebrick3", "353" ="deeppink4", "354" = "orange", "355" = "green1","356" = "gold",  "357"= "hotpink", "358" ="darkslateblue", "359" = "honeydew4", "360" = "khaki4","361" ="ivory4", "362" = "lightblue", Ambient = "Blue", High = "Red") 



#T1 only
set.seed(30)
ord.pvarT1 <- ordinate(pvar.RelRare.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvarT1)
scores.pvarT1 <- as.data.frame(scores(ord.pvarT1))
scores.pvarT1$Treatment <- pvar.samp.df.T1$Treatment
scores.pvarT1$Parent.ID <- pvar.samp.df.T1$Parent.ID

ggplot(scores.pvarT1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. varians T1 Bray Curtis") +
  theme_classic()
ggsave("pvarT1its2.pdf")

#TF only
set.seed(30)
ord.pvarTF <- ordinate(pvar.RelRare.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pvarTF)
scores.pvarTF <- as.data.frame(scores(ord.pvarTF))
scores.pvarTF$Treatment <- pvar.samp.df.TF$Treatment
scores.pvarTF$Parent.ID <- pvar.samp.df.TF$Parent.ID

ggplot(scores.pvarTF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=pvar.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("M. capitata TF Bray Curtis") +
  theme_classic()
ggsave("pvarTFits.pdf")



```
Pocillopora its2 OLD not by 'clade'
```{r}
#pacu
pacu.RelRare <- subset_samples(DIV_RelA_stats, host_genus == "pocillopora")

#all speices
set.seed(30)
 
#make bray curtis data matrix
bc.pacu <- phyloseq::distance(pacu.RelRare, method = "bray")
  pacu.samp.df <- as(sample_data(pacu.RelRare), "data.frame") #sample df
#adonis test
adonis(bc.pacu ~ Treatment*Time.Point, data = pacu.samp.df)

#                     Df SumsOfSqs  MeanSqs F.Model      R2 Pr(>F)
#Treatment             1    0.0565 0.056546 0.38979 0.00828  0.563
#Time.Point            2    0.0407 0.020345 0.14024 0.00596  0.955
#Treatment:Time.Point  2    0.0598 0.029897 0.20609 0.00875  0.908
#Residuals            46    6.6732 0.145069         0.97701       
#Total                51    6.8302                  1.00000       

# Homogeneity of dispersion test  <- do I need to do this if ^^ not significant?
beta.pacu <- betadisper(bc.pacu, pacu.samp.df$Treatment)
permutest(beta.pacu)  #p = 0.59
permutest(beta.pacu, pairwise = T) 

(bc.pacu.HSD <- TukeyHSD(beta.pacu)) #do you need?
plot(bc.pacu.HSD)


# pacu by timepoint: T1
pacu.RelRare.T1 <- subset_samples(pacu.RelRare, Time.Point == "T1")

set.seed(30)

#make bray curtis data matrix
bc.pacu.T1 <- phyloseq::distance(pacu.RelRare.T1, method = "bray")
  pacu.samp.df.T1 <- as(sample_data(pacu.RelRare.T1), "data.frame") #sample df
#adonis test
adonis(bc.pacu.T1 ~ Treatment, data = pacu.samp.df.T1) # p=.978


# pacu by timepoint: TF
pacu.RelRare.TF <- subset_samples(pacu.RelRare, Time.Point == "TF")

set.seed(30)

#make bray curtis data matrix
bc.pacu.TF <- phyloseq::distance(pacu.RelRare.TF, method = "bray")
  pacu.samp.df.TF <- as(sample_data(pacu.RelRare.TF), "data.frame") #sample df
#adonis test
adonis(bc.pacu.TF ~ Treatment, data = pacu.samp.df.TF) # p=.609



# High treatment over time
pacu.RelRare.High <- subset_samples(pacu.RelRare, Treatment == "High")

set.seed(30)

#make bray curtis data matrix
bc.pacu.High <- phyloseq::distance(pacu.RelRare.High, method = "bray")
  pacu.samp.df.High <- as(sample_data(pacu.RelRare.High), "data.frame") #sample df
#adonis test
adonis(bc.pacu.High ~ Time.Point, data = pacu.samp.df.High) # p=.609
#0.788 No diff in High over time (BUT you did not remake the data to include all genotypes at T0)


#High only
set.seed(30)
ord.bc.pacu.High <- ordinate(pacu.RelRare.High, "NMDS", "bray", trymax = 100) 
stressplot(ord.bc.pacu.High)
scores.pacuHigh <- as.data.frame(scores(ord.bc.pacu.High))
scores.pacuHigh$Time.Point <- pacu.samp.df.High$Time.Point
scores.pacuHigh$Parent.ID <- pacu.samp.df.High$Parent.ID

ggplot(scores.pacuHigh, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Time.Point, shape = Treatment), size = 4) +
 # scale_color_manual(values=pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Time.Point), linetype = 2) +
  ggtitle("P. acuta High Bray Curtis") +
  theme_classic()
ggsave("pacuHighits2.pdf")


## create palette for parent.ID and treatments
pacu.Parent.colors <- c("301" = "green4",  "302" = "firebrick3", "303" ="deeppink4", "304" = "orange", "305" = "green1","306" = "gold",  "307"= "hotpink", "308" ="darkslateblue", "309" = "honeydew4", "310" = "khaki4","311" ="ivory4", "312" = "lightblue", Ambient = "Blue", High = "Red") 

#T1 only
set.seed(30)
ord.pacuT1 <- ordinate(pacu.RelRare.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.pacuT1)
scores.pacuT1 <- as.data.frame(scores(ord.pacuT1))
scores.pacuT1$Treatment <- pacu.samp.df.T1$Treatment
scores.pacuT1$Parent.ID <- pacu.samp.df.T1$Parent.ID

ggplot(scores.pacuT1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. acuta T1 Bray Curtis") +
  theme_classic()
ggsave("pacuT1its2.pdf")

#TF only
set.seed(30)
ord.pacuTF <- ordinate(pacu.RelRare.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.pacuTF)
scores.pacuTF <- as.data.frame(scores(ord.pacuTF))
scores.pacuTF$Treatment <- pacu.samp.df.TF$Treatment
scores.pacuTF$Parent.ID <- pacu.samp.df.TF$Parent.ID

ggplot(scores.pacuTF, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Parent.ID, shape = Treatment), size = 4) +
  scale_color_manual(values=pacu.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("P. acuta TF Bray Curtis") +
  theme_classic()
ggsave("pacuTFits.pdf")

```


#Alpha diversit: observed richness, Shannon (+evenness) - Hannah  
Use rarefied data, not relative abunances  
coralDIV2_rare2 df phyloseq
coralDIV2_rare2.sd df sample data
```{r}  
coralDIV2_rare.alpha<-coralDIV2_rare2 #make a copy of df


coralDIV2_rare.alpha2 <- prune_taxa(taxa_sums(coralDIV2_rare.alpha) > 0, coralDIV2_rare.alpha) #prune taxa 0s

erich <- estimate_richness(coralDIV2_rare.alpha2, measures = c("Observed", "Shannon")) #select which estimates and estimate
  erich$Treatment <- coralDIV2_rare2.sd$Treatment       #add back in meta
  erich$Time.Point <- coralDIV2_rare2.sd$Time.Point
  erich$Parent.ID <- coralDIV2_rare2.sd$Parent.ID
  erich$ID <- coralDIV2_rare2.sd$ID
  erich$Species <- coralDIV2_rare2.sd$Species
  erich$Time.Point = factor(erich$Time.Point, levels = c("T0", "T1", "TF"))

 
#stats ALL THE DATA
 library(ggpubr)
  #look at data
#observed
hist(erich$Observed)   
ggqqplot(erich$Observed)

#shannon
hist(erich$Shannon) #not normal  <- bimodal,due to species? 
ggqqplot(erich$Shannon)
  
## subset by coral species for OBSERVED
erich.mcap<-subset(erich, Species=="Montipora_capitata")
  hist(erich.mcap$Observed) #not normal  <- bimodal 
  
erich.pcomp<-subset(erich, Species=="Porites_compressa")
  hist(erich.pcomp$Observed) #not normal 
  ggqqplot(erich.pcomp$Observed) #it's just one sample that is pulling it. ideas? 
 
erich.pvar<-subset(erich, Species=="Pavona_varians")
 hist(erich.pvar$Observed) #not normal 
  ggqqplot(erich.pvar$Observed) #looks bad

erich.pacu<-subset(erich, Species=="Pocillopora_acuta")
 hist(erich.pacu$Observed) #not awfuk 
  ggqqplot(erich.pacu$Observed) #not bad

  
## subset for Shannon
  
erich.mcap<-subset(erich, Species=="Montipora_capitata")
  hist(erich.mcap$Shannon) #not normal  
  ggqqplot(erich.mcap$Shannon) #looks good
 
erich.pcomp<-subset(erich, Species=="Porites_compressa")
  hist(erich.pcomp$Shannon) # normal 
  ggqqplot(erich.pcomp$Shannon) #it's just one sample that is pulling it. ideas? 

erich.pvar<-subset(erich, Species=="Pavona_varians")
 hist(erich.pvar$Shannon) #not normal 
  ggqqplot(erich.pvar$Shannon) #
  
erich.pacu<-subset(erich, Species=="Pocillopora_acuta")
 hist(erich.pacu$Shannon) #not awfuk 
  ggqqplot(erich.pacu$Shannon) #not bad
  
    #use kruskal-wallis, if not normal (https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html#alpha-diversity)
  
#If it is normal 'enough'  

### ALL SPECIES
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.species = aov(Shannon ~ Species, data=erich)
    aov.shannon.species = aov(log(Shannon) ~ Species*Treatment*Time.Point, data=erich)

#Call for the summary of that ANOVA, which will include P-values
  summary(aov.shannon.species) # sig diff between species
  TukeyHSD(aov.shannon.species)

#Plot
  boxplot(Shannon ~ Species, data=erich, ylab="Shannon's diversity") 

  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.species = aov(Observed ~ Species*Treatment*Time.Point, data=erich)
  #aov.observed.species = aov(Observed ~ Species, data=erich)

#Call for the summary of that ANOVA, which will include P-values
  summary(aov.observed.species) # sig diff between species
  TukeyHSD(aov.observed.species)

#Plot
  boxplot(Observed ~ Species, data=erich, ylab="observed's diversity") 
```

### MCAP alpha
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesMcap = aov(Shannon ~ Treatment*Time.Point, data=erich.mcap)
  summary(aov.shannon.speciesMcap) # sig diff between species
  TukeyHSD(aov.shannon.speciesMcap)

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.mcap, ylab="Shannon's diversity") 

#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesMcap = aov(Observed ~ Treatment*Time.Point, data=erich.mcap)
  summary(aov.observed.speciesMcap) # sig diff between species
  TukeyHSD(aov.observed.speciesMcap)
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.mcap, ylab="observed's diversity") 
```
#### Do by C and CD
```{r}
#mcap  erich.mcap


  # by parent ID set clade dom
  erich.mcap$Clade[erich.mcap$Parent.ID==363]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==364]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==365]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==366]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==367]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==368]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==369]<-"CD"
  erich.mcap$Clade[erich.mcap$Parent.ID==370]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==371]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==372]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==373]<-"C"
  erich.mcap$Clade[erich.mcap$Parent.ID==374]<-"CD"
  erich.mcap$Clade<-as.factor(erich.mcap$Clade) #change to factor

## let look at C vs CD
  #Run the ANOVA and save it as an object
  aov.shannon.speciesMcap_CCD = aov(Shannon ~ Clade*Treatment*Time.Point, data=erich.mcap)
  summary(aov.shannon.speciesMcap_CCD) # sig diff between species
  TukeyHSD(aov.shannon.speciesMcap_CCD)

#Plot
  boxplot(Shannon ~ Clade*Treatment*Time.Point, data=erich.mcap, ylab="Shannon's diversity") 

#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesMcap_CCD = aov(Observed ~ Clade*Treatment*Time.Point, data=erich.mcap)
  summary(aov.observed.speciesMcap_CCD) # sig diff between species
  TukeyHSD(aov.observed.speciesMcap_CCD)
#Plot
  boxplot(Observed ~ Clade*Treatment*Time.Point, data=erich.mcap, ylab="observed's diversity") 
  
``` 
# Pcomp alpha
```{r}
#Shannon 
#Run the ANOVA and save it as an object
  aov.shannon.speciesPcomp = aov(Shannon ~ Treatment*Time.Point, data=erich.pcomp)
  summary(aov.shannon.speciesPcomp) # sig diff between species
  TukeyHSD(aov.shannon.speciesPcomp)
  
 
 

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pcomp, ylab="Shannon's diversity") 

#Pcomp  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPcomp = aov(Observed ~ Treatment*Time.Point, data=erich.pcomp)
  summary(aov.observed.speciesPcomp) # sig diff between species
  TukeyHSD(aov.observed.speciesPcomp)
  


  
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.pcomp, ylab="observed's diversity") 


  
``` 
# Pvar
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesPvar = aov(Shannon ~ Treatment*Time.Point, data=erich.pvar)
  summary(aov.shannon.speciesPvar) # sig diff between species
  TukeyHSD(aov.shannon.speciesPvar)

  

#Plot
  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pvar, ylab="Shannon's diversity") 
  
 
  
  
  
  
  
  

#Pvar  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPvar = aov(Observed ~ Treatment*Time.Point, data=erich.pvar)
  summary(aov.observed.speciesPvar) # sig diff between species
  TukeyHSD(aov.observed.speciesPvar)


  
#Plot
  boxplot(log(Observed) ~ Treatment*Time.Point, data=erich.pvar, ylab="observed's diversity") 
  
  
``` 

# Pacu
```{r}
#Shannon
#Run the ANOVA and save it as an object
  aov.shannon.speciesPacu = aov(Shannon ~ Treatment*Time.Point, data=erich.pacu)
  summary(aov.shannon.speciesPacu) # sig diff between species
  TukeyHSD(aov.shannon.speciesPacu)


#Plot
  aov.shannon.speciesPacu$Time.Point = factor(aov.shannon.speciesPacu$Time.Point, levels = c("T0", "T1", "TF"))

  boxplot(Shannon ~ Treatment*Time.Point, data=erich.pacu, ylab="Shannon's diversity") 
  
 
  
  
  
  
  
  

#Pvar  
#Observed
#Run the ANOVA and save it as an object
  aov.observed.speciesPacu = aov(Observed ~ Treatment*Time.Point, data=erich.pacu)
  summary(aov.observed.speciesPacu) # sig diff between species
  TukeyHSD(aov.observed.speciesPacu)


  
#Plot
  boxplot(Observed ~ Treatment*Time.Point, data=erich.pacu, ylab="observed's diversity") 

# Just T1
  erich.pacuT1<-subset(erich.pacu, Time.Point=="T1")
  hist(log10(erich.pacuT1$Observed))
  aov.observed.speciesPacuT1 = aov(log10(Observed) ~ Treatment, data=erich.pacuT1)
  summary(aov.observed.speciesPacuT1) # sig diff between species
  TukeyHSD(aov.observed.speciesPacuT1)
  
  hist((erich.pacuT1$Shannon))
  aov.shannon.speciesPacuT1 = aov(log10(Shannon) ~ Treatment, data=erich.pacuT1)
  summary(aov.shannon.speciesPacuT1) # sig diff between species
  TukeyHSD(aov.shannon.speciesPacuT1)
  
  
``` 
 





 
```{r}
#for non normal Kruskal wallis rank sum  
  kruskal.test(x ~ AgeGroup, data=meta)

  
#testing
  
  coral <- get_variable(coralDIV2_rare.alpha2, "host_genus") %in% c("montipora",  "pocillopora")
sample_data(coralDIV2_rare.alpha2)$human <- factor(coral)

 #estimate_richness(coralDIV2_rare, "coral","host_genus", measures = c("Observed", "Shannon")) #select which estimates and estimate CANNOT GET TO WORK with X
  
## from tutorial can't get it to wokr in my code.   
  GP <- prune_taxa(taxa_sums(GlobalPatterns) > 0, GlobalPatterns)

human <- get_variable(GP, "SampleType") %in% c("Feces", "Mock", "Skin", "Tongue")
sample_data(GP)$human <- factor(human)

alpha_meas = c("Observed", "Shannon")
(p <- plot_richness(GP, "human", "SampleType", measures=alpha_meas))






##
## make bar plot of rank abundance of all DIVs
par(mar = c(10, 4, 4, 2) + 0.1) # make more room on bottom margin
N <- 50
barplot(sort(taxa_sums(coralDIV2_rare.alpha2), TRUE)[1:N]/nsamples(coralDIV2_rare.alpha2), las=2)

sample_variables(coralDIV2_rare.alpha2) #check sample variables

#plot_bar(coralDIV2_rare.alpha2, "Species", fill="DIV", facet_grid=~DIV)


#make network - not helpful for ITS2 here

ig <- make_network(coralDIV2_rare.alpha2, max.dist=0.3)
plot_network(ig, coralDIV2_rare.alpha2, color="Parent.ID", shape="Species", line_weight=0.4, label=NULL)






```





#For PRO (maybe you do not do)
```{r}  
## check out data
ntaxa(coral2)  #num taxa
nsamples(coral2)   #num samples
sample_names(coral2)[1:5] #samp names
rank_names(coral2) #ranks are  and clade
sample_variables(coral2) #all metadata cats

# create df of sample data to view 
sample.dataPRO <- as(sample_data(coral2), "data.frame") #create sample data frame to view
sample.dataPRO$LibrarySize <- sample_sums(coral2)
sample.dataPRO <- sample.data[order(sample.dataPRO$LibrarySize),]
sample.dataPRO$Index <- seq(nrow(sample.dataPRO))
ggplot(data = sample.dataPRO, aes(x=Index, y=LibrarySize, color = sample_type)) +
  geom_point()

#Remove all samples with <5000 reads and create sample data
#These data also useful for beta ersity analyses (non-rarefied)
coral2_nonrare <- prune_samples(sample_sums(coral2)>=5000, coral2) # remove samps <5000, save as new obj
coral2_data.nonrare <- as(sample_data(coral2_nonrare), "data.frame")

##Sample filtering and rarefaction
#remove duplicates that failed
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S1110B")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S1166")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S1388b")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S1857")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S3085")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S3149b")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S3477a")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S761b")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "S775b")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "SN134")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "SN198b")
coral2_nonrare <- subset_samples(coral2_nonrare, ID != "SN209b")

saveRDS(coral2_nonrare, file = "coral2_nonrare.RDS", compress = TRUE) #save this for beta


#Make a rarefied phyloseq object for alpha ersity analyses
coral2_rare <- rarefy_even_depth(coral2_nonrare, sample.size = 7000, rngseed = 711)   #this removed 19 s
sample_sums(coral2_rare) #Double check that the sub-sampling worked, this should report 1000 for each sample
coral2.rare <- as(sample_data(coral2_rare), "data.frame")
saveRDS(coral2_rare, file = "coral2_rare.RDS", compress = TRUE) #save!


coral2_rareView <- as(sample_data(coral2.rare), "data.frame") #create sample data frame to view


```

#FROM Eckard, wish this would work but you are fucking stuck
```{r}
if (!require("pacman")) install.packages("pacman")
pacman::p_load("dplyr", "edgeR", "ggplot2", "MCMC.OTU", "pairwiseAdonis", "rcartocolor",
               "RColorBrewer", "Redmonder", "reshape2", "vegan")

pacman::p_load_gh("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")


if (!require("edgeR")){BiocManager::install("edgeR", update = FALSE) 
  library(edgeR)}

options("scipen" = 10)

its2Seq = read.delim("Data/ITS2/post_med_seqs/133_20201216_DBV_20201216T020209.seqs.absolute.abund_and_meta.txt", header = TRUE, check.names = FALSE)
# load in data like they do
head(its2Seq)
its2Seq <- its2Seq[ -c(1,3:13) ] #drop cols you dont need 
its2Seq <- its2Seq[ -c(11:21) ] #drop more cols you dont need 

its2MetaData<-read.csv("Data/ACE21_Metadata_Molec_20210317.csv")
#its2MetaData<-its2MetaData[!(its2MetaData$Notes=="REMOVE"),]
#its2MetaData<- its2MetaData[!(its2MetaData$sample_name == "3149"),] # dup
#names(its2MetaData)[names(its2MetaData) == "Frag.ID"] <- "sample_name"
#rownames(its2MetaData) <- its2MetaData[,1] #make first col row names *(but keep them as a sep col too)
head(its2MetaData)

#write.csv(its2MetaData,"its2MetaData.csv")
#write.csv(its2Seq,"its2Seq.csv")

its2Seq = cbind(its2Seq[1], its2MetaData[,2:7], its2Seq[,c(2:length(its2Seq))])
its2Seq$Parent.ID<-as.factor(as.character(its2Seq$Parent.ID))

its2Seq$Time.Point = factor(its2Seq$Time.Point, levels = c("T0", "T1", "TF"))
levels(its2Seq$Time.Point)


#above works

#first they purge outliers
goods = purgeOutliers(its2Seq, count.columns = 8:338, sampleZcut=(-5), otu.cut = 0.0001)
its2SeqTransposed = t(its2Seq[, 8:length(its2Seq[1, ])])
its2SeqList = DGEList(counts = its2SeqTransposed)
head(its2SeqList$samples)
## yo ucan't get this to work





```



#16S

```{r}

```












































# Physio PCA
```{r} 
PhysioData4.pr <- prcomp(PhysioData4[c(8:11)], center = TRUE, scale = TRUE)
summary(PhysioData.pr)
 
library(VIM)
library(FactoMineR)
library(missMDA) #lets you do with incomplete dataset
library(naniar)
gg_miss_var(PhysioData)
res<-summary(aggr(PhysioData, sortVar=TRUE))$combinations


nb <- estim_ncpPCA(PhysioData,method.cv = "Kfold", verbose = FALSE) # estimate the number of components from incomplete data
#(available methods include GCV to approximate CV)
nb$ncp #2

res <- cor(PhysioData)
round(res, 2)


library("PerformanceAnalytics")
my_data <- PhysioData[, c(8:16)]
chart.Correlation(my_data, histogram=TRUE, pch=19)


#Display Pearson correlation: testing physio (normal data) correlations

cor.test(PhysioData3$AFDWmg.cm2_log, PhysioData3$Cellcounts_zoox.cm2_log, method=c("pearson")) #

PhysioData3$Treatment_level <- factor(PhysioData3$Treatment, levels = c("Ambient", "High")) #make new col

Cor1<-ggplot(PhysioData3, aes(x=AFDWmg.cm2_log, y=Cellcounts_zoox.cm2_log)) + 
  geom_point(aes(color=Treatment, shape=Treatment), size=6) +
   #scale_color_manual(name="Nutrition",
        #            values=c("orange", "purple"), 
          #          labels=c("Fed", "Unfed"))+
  scale_shape_manual(values = c(19,21))+
  theme_classic()+
  geom_text(x=800, y=20, label="r=0.50, p<0.001", size=9, color="black") +
  theme(text = element_text(size = 20))+
  theme(axis.text = element_text(size = 20, color="black"))+
  theme(axis.title = element_text(size = 20, face="bold"))+
  theme(legend.title = element_text(size = 20, face="bold"))+
  theme(legend.text = element_text(size = 20, color="black"))+
  ylab(expression(bold(paste("AFDWmg.cm2_log")))) +
  xlab(expression(bold(paste("Cellcounts_zoox.cm2_log"))));Cor1

corrModel<-lmer(AFDWmg.cm2_log~Cellcounts_zoox.cm2_log * Treatment * Species + (1|Tank.num) + (1|Parent.ID), data=PhysioData3)
summary(corrModel)
anova(corrModel, type=2)


##### trying correlations with only complete data (no PAM or BW)
correlation<-PhysioData4
correlation<-correlation[complete.cases(correlation), ] #only full data (2 samps no chla)

correlation_Mcap<-subset(correlation, Species=="Montipora_capitata")
correlation_Mcap_T1<-subset(correlation_Mcap, Time.Point=="T1")

#not workin
corrplots<-ggplot(correlation_Mcap_T1, aes(x=Treatment, y=Frag.ID))+
  geom_point()+
  facet_wrap(~compound, scales="free")+
  geom_smooth(method='lm', color='red')+
  theme_classic()+
  theme(text=element_text(size=14, face="bold"))
corrplots


#back to PCA with the shorter coor dataset that is full
PhysioData5<-PhysioData4
PhysioData5<-PhysioData5[complete.cases(PhysioData5), ] #only full data (2 samps no chla)

PhysioData5.pr <- prcomp(PhysioData5[c(8:11)], center = TRUE, scale = TRUE)
summary(PhysioData5.pr)

#amount of variance explained
screeplot(PhysioData5.pr, type = "l", npcs = 15, main = "Screeplot of the first 10 PCs")
abline(h = 1, col="red", lty=5)
legend("topright", legend=c("Eigenvalue = 1"),
       col=c("red"), lty=5, cex=0.6)
cumpro <- cumsum(PhysioData5.pr$sdev^2 / sum(PhysioData5.pr$sdev^2))
plot(cumpro[0:15], xlab = "PC #", ylab = "Amount of explained variance", main = "Cumulative variance plot")
abline(v = 6, col="blue", lty=5)
abline(h = 0.88759, col="blue", lty=5)
legend("topleft", legend=c("Cut-off @ PC6"),
       col=c("blue"), lty=5, cex=0.6)


#try plotting PC1 and 2 (how to get the percents?????)
plot(PhysioData5.pr$x[,1],PhysioData5.pr$x[,2], xlab="PC1 ()", ylab = "PC2 ()", main = "PC1 / PC2 - plot")


library("factoextra")
fviz_pca_ind(PhysioData5.pr, geom.ind = "point", pointshape = 21, 
             pointsize = 2, 
             fill.ind = PhysioData5$Treatment, 
             col.ind = "black", 
             palette = "jco", 
             addEllipses = TRUE,
             label = "var",
             col.var = "black",
             repel = TRUE,
             legend.title = "Treatment") +
  ggtitle("") +
  theme(plot.title = element_text(hjust = 0.5))


PhysioData5_mat<-PhysioData5[c(8:11)]
cor.res <- cor(PhysioData5_mat)
corrplot.mixed(cor.res, tl.cex = 0.7, tl.col = "black", addCoefasPercent = TRUE)
 
```



PCA hannah use physio data
```{r}
##PCA with imputation using PPCA (Probabilistic PCA, can handle ~15% missing data)

##Use the pcaMethods package for imputing
#BiocManager::install("pcaMethods")
library(pcaMethods)
sum(is.na(PhysioData)) #385
data<-PhysioData #make copy
View(data)

data$time1.growth[data$time1.growth == "NaN"] <- NA #change NaN to NA
data$time1.growth[data$Chla_ug.chla.cm2 == "NaN"] <- NA #change NaN to NA

data$T1.PAM[data$T1.PAM == "NaN"] <- NA #change NaN to NA
data$TF.PAM[data$TF.PAM == "NaN"] <- NA #change NaN to NA
data$T0.PAM[data$T0.PAM == "NaN"] <- NA #change NaN to NA

#Now we get the estimated data set by using PPCA and two principal components
#But you can use nPCs = 3 or however many principal components you want to use.

pc <- pca(data, nPcs=2, method="ppca") #probabilistic PCA
imputed <- completeObs(pc) #This creates a new dataframe with imputed values based on your probability pca

#Then you can run a pca on the imputed datafram
pca <- prcomp(imputed, center = TRUE, scale. = TRUE)
summary(pca)
View(pca)

#you can view it this way
library(devtools)
#install_github("vqv/ggbiplot")
library(ggbiplot)
ggbiplot(pca)

#Or extract the scores and make a ggplot
#Then add your metadata
library(ggplot2)
Miss_scores <- as.data.frame(pca$x)
Miss_scores$frag.id <- data$Frag.ID
Miss_scores$parent.id <- data$Parent.ID
Miss_scores$treatment <- data$Treatment
Miss_scores$species <- data$Species
Miss_scores$time.point <- data$Time.Point

#Plot
ggplot(Miss_scores, aes(x = PC1, y= PC2)) + geom_point(aes(colour = species, shape = time.point), size = 4) +
  stat_ellipse(aes(x = PC1, y = PC2, colour = species), linetype = 2) +
  theme_classic()

#plot by time and species
Miss_scores$treatSpecies<-paste0(Miss_scores$species, Miss_scores$treatment )
Miss_scores$treatTime<-paste0(Miss_scores$treatment, Miss_scores$time.point )


#subset by T0
Miss_scores_T0<-subset(Miss_scores, time.point=="T0")

Miss_scores_T0_plot<-ggplot(Miss_scores_T0, aes(x = PC1, y= PC2)) + geom_point(aes(colour = species, shape = treatment), size = 4) +
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatSpecies), linetype = 2, level=0.25) +
  theme_classic();Miss_scores_T0_plot

#subset by T1
Miss_scores_T1<-subset(Miss_scores, time.point=="T1")
Miss_scores_T1_plot<-ggplot(Miss_scores_T1, aes(x = PC1, y= PC2)) + geom_point(aes(colour = species, shape = treatment), size = 4) +
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatSpecies), linetype = 2, level=.25) +
  theme_classic();Miss_scores_T1_plot

#subset by TF
Miss_scores_TF<-subset(Miss_scores, time.point=="TF")
Miss_scores_TF_plot<-ggplot(Miss_scores_TF, aes(x = PC1, y= PC2)) + geom_point(aes(colour = species, shape = treatment), size = 4) +
  #stat_ellipse(aes(x = PC1, y = PC2, colour = treatSpecies), linetype = 2) +
  theme_classic();Miss_scores_TF_plot

#plot all time plots
Miss_scores_T0_plot<-Miss_scores_T0_plot+ theme(legend.position = "none") #remove legend
Miss_scores_T1_plot<-Miss_scores_T1_plot+ theme(legend.position = "none") #remove legend
Miss_scores_TF_plot<-Miss_scores_TF_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Miss_scores_T0_plot, Miss_scores_T1_plot, Miss_scores_TF_plot, nrow = 1)

#Subset MCAP
Miss_scores_Mcap<-subset(Miss_scores, species=="Montipora_capitata")

Miss_Mcap_pca<-ggplot(Miss_scores_Mcap, aes(x = PC1, y= PC2)) + geom_point(aes(colour = treatment, shape = time.point), size = 4)+ 
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatTime), linetype = 2, level=.25) +
   ggtitle("Mcap") +
  theme_classic();Miss_Mcap_pca

#Subset Pcomp
Miss_scores_Pcomp<-subset(Miss_scores, species=="Porites_compressa")
Miss_Pcomp_pca<-ggplot(Miss_scores_Pcomp, aes(x = PC1, y= PC2)) + geom_point(aes(colour = treatment, shape = time.point), size = 4)+
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatTime), linetype = 2, level=.25) +
      ggtitle("Pcomp") +
  theme_classic();Miss_Pcomp_pca

#Subset Pvar
Miss_scores_Pvar<-subset(Miss_scores, species=="Pavona_varians")
Miss_Pvar_pca<-ggplot(Miss_scores_Pvar, aes(x = PC1, y= PC2)) + geom_point(aes(colour = treatment, shape = time.point), size = 4)+
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatTime), linetype = 2, level=.25) +
    ggtitle("Pvar") +
  theme_classic();Miss_Pvar_pca

#Subset Pacu
Miss_scores_Pacu<-subset(Miss_scores, species=="Pocillopora_acuta")
Miss_Pacu_pca<-ggplot(Miss_scores_Pacu, aes(x = PC1, y= PC2)) + geom_point(aes(colour = treatment, shape = time.point), size = 4)+
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatTime), linetype = 2, level=.25) +
  ggtitle("Pacuta") +
  theme_classic();Miss_Pacu_pca

#plot all species plots
Miss_Mcap_pca<-Miss_Mcap_pca+ theme(legend.position = "none") #remove legend
Miss_Pcomp_pca<-Miss_Pcomp_pca+ theme(legend.position = "none") #remove legend
Miss_Pvar_pca<-Miss_Pvar_pca+ theme(legend.position = "none") #remove legend
Miss_Pacu_pca<-Miss_Pacu_pca+ theme(legend.position = "none") #remove legend

grid.arrange(Miss_Mcap_pca, Miss_Pcomp_pca, Miss_Pvar_pca,Miss_Pacu_pca, nrow = 2)

```


PCA not alive measurements' not PAM.BW
```{r}
#PhysioData5 

PhysioData5.pca <- prcomp(PhysioData5[,c(8:11)], center = TRUE,scale. = TRUE)
summary(PhysioData5.pca)
str(PhysioData5.pca)


scores <- as.data.frame(PhysioData5.pca$x)
scores$frag.id <- data$Frag.ID
scores$parent.id <- data$Parent.ID
scores$treatment <- data$Treatment
scores$species <- data$Species
scores$time.point <- data$Time.Point
scores$treatSpecies<-paste0(scores$species, scores$treatment )
scores$treatTime<-paste0(scores$treatment, scores$time.point )


ggplot(scores, aes(x = PC1, y= PC2)) + geom_point(aes(colour = species, shape = time.point), size = 4) +
  stat_ellipse(aes(x = PC1, y = PC2, colour = species), linetype = 2) +
  theme_classic()


#subset by T0
scores_T0<-subset(scores, time.point=="T0")
scores_T0_plot<-ggplot(scores_T0, aes(x = PC1, y= PC2)) + geom_point(aes(colour = species, shape = treatment), size = 4) +
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatSpecies), linetype = 2, level=0.25) +
  theme_classic();scores_T0_plot

#subset by T1
scores_T1<-subset(scores, time.point=="T1")
scores_T1_plot<-ggplot(scores_T1, aes(x = PC1, y= PC2)) + geom_point(aes(colour = species, shape = treatment), size = 4) +
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatSpecies), linetype = 2, level=0.25) +
  theme_classic();scores_T1_plot

#subset by TFs
scores_TF<-subset(scores, time.point=="TF")
scores_TF_plot<-ggplot(scores_TF, aes(x = PC1, y= PC2)) + geom_point(aes(colour = species, shape = treatment), size = 4) +
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatSpecies), linetype = 2, level=.25) +
  theme_classic();scores_TF_plot

#plot all time plots
scores_T0_plot<-scores_T0_plot+ theme(legend.position = "none") #remove legend
scores_T1_plot<-scores_T1_plot+ theme(legend.position = "none") #remove legend
scores_TF_plot<-scores_TF_plot+ theme(legend.position = "none") #remove legend

grid.arrange(scores_T0_plot, scores_T1_plot, scores_TF_plot, nrow = 1)


#Subset MCAP
scores_Mcap<-subset(scores, species=="Montipora_capitata")

Mcap_pca<-ggplot(scores_Mcap, aes(x = PC1, y= PC2)) + geom_point(aes(colour = treatment, shape = time.point), size = 4)+ 
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatTime), linetype = 2, level=.25) +
   ggtitle("Mcap") +
  theme_classic();Mcap_pca

#Subset Pcomp
scores_Pcomp<-subset(scores, species=="Porites_compressa")
Pcomp_pca<-ggplot(scores_Pcomp, aes(x = PC1, y= PC2)) + geom_point(aes(colour = treatment, shape = time.point), size = 4)+
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatTime), linetype = 2, level=.25) +
      ggtitle("Pcomp") +
  theme_classic();Pcomp_pca

#Subset Pvar
scores_Pvar<-subset(scores, species=="Pavona_varians")
Pvar_pca<-ggplot(scores_Pvar, aes(x = PC1, y= PC2)) + geom_point(aes(colour = treatment, shape = time.point), size = 4)+
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatTime), linetype = 2, level=.25) +
    ggtitle("Pvar") +
  theme_classic();Pvar_pca

#Subset Pacu
scores_Pacu<-subset(scores, species=="Pocillopora_acuta")
Pacu_pca<-ggplot(scores_Pacu, aes(x = PC1, y= PC2)) + geom_point(aes(colour = treatment, shape = time.point), size = 4)+
  stat_ellipse(aes(x = PC1, y = PC2, colour = treatTime), linetype = 2, level=.25) +
  ggtitle("Pacuta") +
  theme_classic();Pacu_pca

#plot all species plots
Mcap_pca<-Mcap_pca+ theme(legend.position = "none") #remove legend
Pcomp_pca<-Pcomp_pca+ theme(legend.position = "none") #remove legend
Pvar_pca<-Pvar_pca+ theme(legend.position = "none") #remove legend
Pacu_pca<-Pacu_pca+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_pca, Pcomp_pca, Pvar_pca,Pacu_pca, nrow = 2)
```
Lets try the PCA by Parent.ID, collapse the df by parent. 
```{r} 
 PhysioData6<-PhysioData #make a copy


Testing <- cbind(
  aggregate(PhysioData6$AFDWmg.cm2_log, by = list(PhysioData6$ParentID, PhysioData6$Treatment, PhysioData6$Species,PhysioData6$Time.Point,PhysioData6$Depth,PhysioData6$Tank.num), FUN=mean, na.rm = F))
  
  
  
  




```

