---
title: "ITS2  data analysis"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

Use this one. other one for code pull refs. 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
rm(list = ls())

```
#Load library
```{r}   
# rm(list=ls())
library(car)
library(readxl)
library(phyloseq)
library(janitor)
library(phyloseq)  # source('http://bioconductor.org/biocLite.R'); biocLite('phyloseq')
library(zoo)
library(stringr)
library(vegan)
library(multcompView)
library("ggpubr")
library(dplyr)
library(plyr)
library(gridExtra)
library(vegan)
#library(decontam)
library(RColorBrewer)
library(ggforce)
library(cowplot)  
library(ggplot2)
library(grid)
library(lmerTest)
library(RColorBrewer)
library(colorRamps) 
library(devtools)
library(phyloseq)
library(tidyverse)

#set colors
species_colors <- c("Montipora_capitata" = "#E69F00", "Porites_compressa" = "#56B4E9", "Pavona_varians" = "#CC79A7", "Pocillopora_acuta" = "#009E73")

clade_colors <- c("C31" = "goldenrod1", "C31D4" = "darkorange3", "C15cj" = "powderblue", "C15n" = "#0072B2", "C1"="pink1", "C27"="firebrick", "C1d"="palegreen2", "C42_2"="darkgreen")
```

Load phyloseq objects (if starting at analysis)
```{r}
# Load phyloseq objects - not subsampled or relative abundance yet
load("data/Raw_coral_phyloseq.RData") #coral_profile and coralDIV
```

## Data Wrangle - Build phyloseq dataframes using Symportal outputs for Profiles and DIVs
Read in coral ITS2 profiles: "coral_Profiles"
```{r}    
library(janitor)
library(phyloseq)  # BiocManager::install("phyloseq")
library(cowplot)
library(ggrepel)
library(scales)
library(RColorBrewer)
library(MASS)
library(lme4)
library(emmeans)
library(tidyverse)
library(vegan)
library(ggh4x)
library(stringi)
#add metadata to symportal submission data
Symp_sub<-readxl::read_xlsx("Data/ITS2/ITS2_TT17_Matsuda_metadata_2020_update.xlsx", skip = 1)
  Metadata<-read.csv("Data/ACE21_Metadata_Molec_20210317.csv")
    Metadata$Treatment<-as.factor(Metadata$Treatment)
    Metadata$Species<-as.factor(Metadata$Species)
    Metadata$Parent.ID<-as.factor(Metadata$Parent.ID)
    Metadata$Time.Point<-as.factor(Metadata$Time.Point)
# EDIT sample data, add CLADE COL ####
  #Mcap
  Metadata$Clade[Metadata$Parent.ID==363]<-"C31"
  Metadata$Clade[Metadata$Parent.ID==364]<-"C31D4"
  Metadata$Clade[Metadata$Parent.ID==365]<-"C31D4"
  Metadata$Clade[Metadata$Parent.ID==366]<-"C31D4"
  Metadata$Clade[Metadata$Parent.ID==367]<-"C31"
  Metadata$Clade[Metadata$Parent.ID==368]<-"C31D4"
  Metadata$Clade[Metadata$Parent.ID==369]<-"C31D4"
  Metadata$Clade[Metadata$Parent.ID==370]<-"C31"
  Metadata$Clade[Metadata$Parent.ID==371]<-"C31"
  Metadata$Clade[Metadata$Parent.ID==372]<-"C31"
  Metadata$Clade[Metadata$Parent.ID==373]<-"C31"
  Metadata$Clade[Metadata$Parent.ID==374]<-"C31D4"
 #Pcomp
  Metadata$Clade[Metadata$Parent.ID==314]<-"C15cj"
  Metadata$Clade[Metadata$Parent.ID==317]<-"C15cj"
  Metadata$Clade[Metadata$Parent.ID==319]<-"C15cj"
  Metadata$Clade[Metadata$Parent.ID==320]<-"C15cj"
  Metadata$Clade[Metadata$Parent.ID==321]<-"C15cj"
  Metadata$Clade[Metadata$Parent.ID==322]<-"C15cj"
  Metadata$Clade[Metadata$Parent.ID==313]<-"C15n"
  Metadata$Clade[Metadata$Parent.ID==315]<-"C15n"
  Metadata$Clade[Metadata$Parent.ID==316]<-"C15n"
  Metadata$Clade[Metadata$Parent.ID==318]<-"C15n"
  Metadata$Clade[Metadata$Parent.ID==323]<-"C15n"
  Metadata$Clade[Metadata$Parent.ID==324]<-"C15n"
  
  #Pvar
  Metadata$Clade[Metadata$Parent.ID==353]<-"C1"
  Metadata$Clade[Metadata$Parent.ID==358]<-"C1"
  Metadata$Clade[Metadata$Parent.ID==361]<-"C1"
  Metadata$Clade[Metadata$Parent.ID==325]<-"C1"
  Metadata$Clade[Metadata$Parent.ID==356]<-"C1"
  Metadata$Clade[Metadata$Parent.ID==351]<-"C27"
  Metadata$Clade[Metadata$Parent.ID==354]<-"C27"
  Metadata$Clade[Metadata$Parent.ID==355]<-"C27"
  Metadata$Clade[Metadata$Parent.ID==357]<-"C27"
  Metadata$Clade[Metadata$Parent.ID==359]<-"C27"
  Metadata$Clade[Metadata$Parent.ID==360]<-"C27"
  Metadata$Clade[Metadata$Parent.ID==362]<-"C27"
  
#Pacu
  Metadata$Clade[Metadata$Parent.ID==311]<-"C42_2"
  Metadata$Clade[Metadata$Parent.ID==301]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==304]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==306]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==303]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==307]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==308]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==310]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==312]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==302]<-"C1d"
  Metadata$Clade[Metadata$Parent.ID==305]<-"C42_2"
  Metadata$Clade[Metadata$Parent.ID==309]<-"C42_2"
  
Metadata$Clade<-as.factor(Metadata$Clade)
Metadata$Clade <- factor(Metadata$Clade, levels = c("C31", "C31D4", "C15cj","C15n","C1","C27","C1d","C42_2"))
  Metadata$Species <- factor(Metadata$Species, levels = c("Montipora_capitata", "Porites_compressa", "Pavona_varians","Pocillopora_acuta"))

# merge profile info
 AsymportalPro<-read.csv("Data/Symportal_profile.csv")  
  AsymportalPro2<-merge(Metadata,AsymportalPro, by="Frag.ID") 
  #write.csv(AsymportalPro2,"AsymportalPro2.csv") 

  names(Metadata)[names(Metadata) == "Frag.ID"] <- "sample_name" #change to same ID to merge
Symp_sub_Meta<-left_join(Symp_sub, Metadata, by="sample_name") # merge
Symp_sub_Meta$Parent.ID<-as.factor(as.character(Symp_sub_Meta$Parent.ID))

#Phyloseq object
sam0 <- Symp_sub_Meta
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))

tax0 <- read_tsv(
  file  = "Data/ITS2/its2_type_profiles/133_20201216_DBV_20201216T020209.profiles.absolute.abund_and_meta.txt",
  n_max = 6) %>%
  dplyr::select(-2) %>% 
  gather(UID, value, -1) %>% 
  spread(1, value) %>%
  clean_names()

tax1 <- as.matrix(tax0[, -1], dimnames = list(tax0$uid, colnames(tax0[-1])))
rownames(tax1) <- tax0$uid
tax <- tax_table(tax1)

library(tidyverse)

otuVIEW<-read_tsv(
  file  = "Data/ITS2/its2_type_profiles/133_20201216_DBV_20201216T020209.profiles.absolute.abund_and_meta.txt") 
otu0 <- read_tsv(
  file  = "Data/ITS2/its2_type_profiles/133_20201216_DBV_20201216T020209.profiles.absolute.abund_and_meta.txt") %>% 
  dplyr::rename(sample_name = 2) %>%
  dplyr::select(-1) %>%
  slice(7:(n() - 2)) %>%               # Keep rows 7 through (n - 2) 
  mutate_at(2:ncol(.), as.numeric)
otu1 <- as.matrix(otu0[, -1])
rownames(otu1) <- otu0$sample_name
otu <- otu_table(otu1, taxa_are_rows = FALSE)

coral_profile <- phyloseq(otu, tax, sam)
```
Read in coral post-QC sequence variants: "coralDIV_RELA"
```{r}

taxnames <- read_tsv(file  = "Data/ITS2/post_med_seqs/133_20201216_DBV_20201216T020209.seqs.absolute.abund_only.txt",
  n_max = 0) %>%
  dplyr::select(-1) %>%
  names(.)
tax0 <- data_frame(
  DIV = taxnames,
  clade = str_extract(DIV, "[A-Z]")
)
tax1 <- as.matrix(tax0)
rownames(tax1) <- tax0$DIV

tax <- tax_table(tax1)
otu0 <- read_tsv(
  file  = "Data/ITS2/post_med_seqs/133_20201216_DBV_20201216T020209.seqs.absolute.abund_and_meta.txt") %>%
  dplyr::select(-1, -(3:33))
otu1 <- as.matrix(otu0[, -1])
rownames(otu1) <- otu0$sample_name
otu <- otu_table(otu1, taxa_are_rows = FALSE)
coralDIV <- phyloseq(otu, tax, sam)
```
Save phyloseq objects
```{r}
coral_profile <- coral_profile %>%
  subset_samples(!is.na(host_species))

coralDIV2 <- coralDIV %>% 
  subset_samples(!is.na(host_species))

#save(coral_profile, coralDIV2, file = "Data/coral_phyloseq.RData")
```

#Quality Check
Look at low sequence samples and duplicates
```{r}    
LowSamps<-readxl::read_xlsx("Data/ITS2/post_med_seqs/133_20201216_DBV_20201216T020209.seqs.absolute.meta_only.xlsx")
LowSamps_dups<-merge(LowSamps, Metadata, all.x=T)
LowSamps_dups2<-subset(LowSamps_dups, Notes=="Dup")
LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs_log<-log10(LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs)
hist(LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs_log, breaks=20)
```
Data wrangle: 
Profiles
```{r}
# check out data coral_profile
ntaxa(coral_profile)  #num taxa
nsamples(coral_profile)   #num samples
sample_names(coral_profile)[1:5] #samp names
rank_names(coral_profile) #ranks are DIV and clade
sample_variables(coral_profile) #all metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(coral_profile), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(coral_profile)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = sample_type)) +
  geom_point()

##Sample filtering and rarefaction
coral_profile2<-coral_profile #make copy
#remove duplicates that failed
coral_profile2 <- subset_samples(coral_profile2, ID != "S1110B")
coral_profile2 <- subset_samples(coral_profile2, ID != "S1166")
coral_profile2 <- subset_samples(coral_profile2, ID != "S1388b")
coral_profile2 <- subset_samples(coral_profile2, ID != "S1857")
coral_profile2 <- subset_samples(coral_profile2, ID != "S3085")
coral_profile2 <- subset_samples(coral_profile2, ID != "S3149b")
coral_profile2 <- subset_samples(coral_profile2, ID != "S3149")
coral_profile2 <- subset_samples(coral_profile2, ID != "S3477a")
coral_profile2 <- subset_samples(coral_profile2, ID != "S761b")
coral_profile2 <- subset_samples(coral_profile2, ID != "S775b")
coral_profile2 <- subset_samples(coral_profile2, ID != "SN134")
coral_profile2 <- subset_samples(coral_profile2, ID != "SN198b")
coral_profile2 <- subset_samples(coral_profile2, ID != "SN209b")
coral_profile2 <- subset_samples(coral_profile2, ID != "S3085b")
coral_profile2 <- subset_samples(coral_profile2, ID != "S1857")
coral_profile2 <- subset_samples(coral_profile2, ID != "S761b")
coral_profile2 <- subset_samples(coral_profile2, ID != "S198b")
coral_profile2 <- subset_samples(coral_profile2, ID != "SN127")
coral_profile2 <- subset_samples(coral_profile2, ID != "SN98b")
coral_profile2 <- subset_samples(coral_profile2, ID != "S2893")

saveRDS(coral_profile2, file = "coral_profile2_cleaned.RDS", compress = TRUE) #save!
```
DIVs: remove samples with under 7k, remove duplicated samples
```{r}         
## check out data
ntaxa(coralDIV2)  #num taxa
nsamples(coralDIV2)   #num samples
sample_names(coralDIV2)[1:5] #samp names
rank_names(coralDIV2) #ranks are DIV and clade
sample_variables(coralDIV2) #all metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(coralDIV2), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(coralDIV2)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = sample_type)) +
  geom_point()

#Remove all samples with <7000 reads and create sample data
#These data also useful for beta diversity analyses (non-rarefied)
coralDIV2_nonrare <- prune_samples(sample_sums(coralDIV2)>=7000, coralDIV2) # remove samps <7000, save as new obj
coralDIV2_data.nonrare <- as(sample_data(coralDIV2_nonrare), "data.frame")

##Sample filtering and rarefaction
#remove duplicates that failed
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1110B")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1166")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1388b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1857")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3085")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3149b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3149")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3477a")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S761b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S775b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN134")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN198b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN209b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S3085b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S1857")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S761b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S198b")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN127")
coralDIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "SN98b")
coral_DIV2_nonrare <- subset_samples(coralDIV2_nonrare, ID != "S2893")
coral_DIV2_data.nonrare <- as(sample_data(coral_DIV2_nonrare), "data.frame")


#Make a rarefied phyloseq object 
coral_DIV2_rare <- rarefy_even_depth(coral_DIV2_nonrare, sample.size = 7000, rngseed = 711)   #this removed 17 DIVs
sample_sums(coral_DIV2_rare) #Double check that the sub-sampling worked, this should report 7000 for each sample
coral_DIV2rare_sd <- as(sample_data(coral_DIV2_rare), "data.frame")

saveRDS(coral_DIV2_rare, file = "coral_DIV2_rare.RDS", compress = TRUE)


#Make relative abundance df
coral_DIV2_rare_Rel <- transform_sample_counts(coral_DIV2_rare, function(x) 100 * x/sum(x))

saveRDS(coral_DIV2_rare_Rel, file = "coral_DIV2_rare_Rel.RDS", compress = TRUE)
```
#### Data Analysis Load RDS
Load RDS
```{r}

coral_profile2 <- readRDS("coral_profile2_cleaned.RDS")
coral_DIV2_rare <- readRDS("coral_DIV2_rare.RDS")
coral_DIV2_rare_Rel <- readRDS("coral_DIV2_rare_Rel.RDS")

```



Data frames for analysis:
coral_DIV2_nonrare: DIV, not subsampled
coral_DIV2_rare: DIV, subsampled
coral_DIV2rare_sd: sample data
coral_DIV2_rare_Rel: DIV, subsampled, Relative Abundance
coral_profile2: Profiles

# Visualizations 
DIVs: Bar plots for high level visualization
```{r}  
#try glom use raw counts its2.s2
Mcap_its2.s2<-subset_samples(coral_DIV2_rare, Species=="Montipora_capitata") #use raw counts here

Mcap_its2 <- Mcap_its2.s2 %>% 
  tax_glom(taxrank = "DIV", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Mcap_its2.s2.melt <- psmelt(Mcap_its2)

Mcap_its2.s2.melt$Code<-paste(Mcap_its2.s2.melt$Treatment, Mcap_its2.s2.melt$Time.Point)
Mcap_its2.s2.melt$Parent.ID <- factor(Mcap_its2.s2.melt$Parent.ID, levels = c("363",  "372","373","371", "367","370","364",  "365","366","368",  "369","374"))
Mcap_its2.s2.melt$Code <- factor(Mcap_its2.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 400
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

mcap.p <- ggplot(Mcap_its2.s2.melt, aes(x = Parent.ID, y = Abundance, fill = DIV)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  theme(legend.position = "none")
mcap.p +  facet_wrap(~ Code, nrow=2)


#try glom use raw counts its2.s2
Pcomp_its2.s2<-subset_samples(coral_DIV2_rare, Species=="Porites_compressa") #use raw counts here

Pcomp_its2 <- Pcomp_its2.s2 %>% 
  tax_glom(taxrank = "DIV", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pcomp_its2.s2.melt <- psmelt(Pcomp_its2)

Pcomp_its2.s2.melt$Code<-paste(Pcomp_its2.s2.melt$Treatment, Pcomp_its2.s2.melt$Time.Point)
Pcomp_its2.s2.melt$Parent.ID <- factor(Pcomp_its2.s2.melt$Parent.ID, levels = c("313",  "314","315","316", "317","318","319",  "320", "322","323",  "324","321"))
Pcomp_its2.s2.melt$Code <- factor(Pcomp_its2.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 400
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pcomp.p <- ggplot(Pcomp_its2.s2.melt, aes(x = Parent.ID, y = Abundance, fill = DIV)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  theme(legend.position = "none")
Pcomp.p +  facet_wrap(~ Code, nrow=2)

#try glom use raw counts its2.s2
Pvar_its2.s2<-subset_samples(coral_DIV2_rare, Species=="Pavona_varians") #use raw counts here

Pvar_its2 <- Pvar_its2.s2 %>% 
  tax_glom(taxrank = "DIV", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pvar_its2.s2.melt <- psmelt(Pvar_its2)

Pvar_its2.s2.melt$Code<-paste(Pvar_its2.s2.melt$Treatment, Pvar_its2.s2.melt$Time.Point)
Pvar_its2.s2.melt$Parent.ID <- factor(Pvar_its2.s2.melt$Parent.ID, levels = c("361",  "358","353","356", "325","354",  "359", "357","362", "351", "355","360"))
Pvar_its2.s2.melt$Code <- factor(Pvar_its2.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 400
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pvar.p <- ggplot(Pvar_its2.s2.melt, aes(x = Parent.ID, y = Abundance, fill = DIV)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  theme(legend.position = "none")
Pvar.p +  facet_wrap(~ Code, nrow=2)


#try glom use raw counts its2.s2
Pacu_its2.s2<-subset_samples(coral_DIV2_rare, Species=="Pocillopora_acuta") #use raw counts here

Pacu_its2 <- Pacu_its2.s2 %>% 
  tax_glom(taxrank = "DIV", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Pacu_its2.s2.melt <- psmelt(Pacu_its2)

Pacu_its2.s2.melt$Code<-paste(Pacu_its2.s2.melt$Treatment, Pacu_its2.s2.melt$Time.Point)
Pacu_its2.s2.melt$Parent.ID <- factor(Pacu_its2.s2.melt$Parent.ID, levels = c("304",  "308","312","303", "306","302",  "310", "301","311", "307", "309","305"))
Pacu_its2.s2.melt$Code <- factor(Pacu_its2.s2.melt$Code, levels = c("Ambient T0",  "Ambient T1",
                                                                    "Ambient TF","High T0",  "High T1","High TF"))
nb.cols <- 400
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

Pacu.p <- ggplot(Pacu_its2.s2.melt, aes(x = Parent.ID, y = Abundance, fill = DIV)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values=mycolors) +
  ylab("Relative Abundance") +
  xlab("Sample Label") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  theme(legend.position = "none")
Pacu.p +  facet_wrap(~ Code, nrow=2)
```

#T0 only bar plots: Figure 3A
```{r}
#PROFILES coral_profile2 df
#try glom use raw counts 
profiles_T0<-subset_samples(coral_profile2, Time.Point=="T0") #use raw counts df here

T0_its2 <- profiles_T0 %>%  #make relative abundance
  tax_glom(taxrank = "its2_type_profile", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
T0_its2.melt <- psmelt(T0_its2)

#Change so all Cs are shades of Blue and all Ds are shades of orange
# Define a color palette with 30 shades of blue
blue_palette <- colorRampPalette(c("lightblue", "blue", "darkblue"))(30)
orange_palette <- colorRampPalette(c("lightyellow", "orange", "darkorange"))(3)

# Display the colors
barplot(rep(1, 30), col = blue_palette, border = NA, space = 0, main = "30 Shades of Blue")
barplot(rep(1, 3), col = orange_palette, border = NA, space = 0, main = "3 Shades of Orange")
# Combine palettes
mycolors <- c(blue_palette, orange_palette)


# Arrange and create an ordered factor for Sample based on Clade within each Species
T0_its2.melt1 <- T0_its2.melt %>%
  arrange(Species, Clade, Sample) %>%
  mutate(Sample = factor(Sample, levels = unique(Sample)))

T0_its2.melt1 <- T0_its2.melt1 %>%
  mutate(Species = factor(Species, levels = c("Montipora_capitata", "Porites_compressa", "Pavona_varians", "Pocillopora_acuta")))  # Adjust order as needed

# Ensure colors are assigned based on Clade groups in `its2_type_profile`
T0.p <- ggplot(T0_its2.melt1, aes(x = Sample, y = Abundance, fill = its2_type_profile)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = mycolors) +
  ylab("Relative Abundance") +
  xlab("") +
  theme_minimal() +  # Apply minimal theme (white background, no grid)
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1),  # Rotate x-axis labels
    panel.grid = element_blank(),  # Remove grid lines
    plot.background = element_blank(),
     # legend.position = "none" ,# Remove the outer background
    axis.line = element_line(color = "black")  # Add black axis lines
  )

T0_ITS2_samp <- T0.p + facet_wrap(~Species, scales = "free_x", nrow = 1)
T0_ITS2_samp

#ggsave("ITS2_T0_samp_leg.pdf", plot = T0_ITS2_samp, device = "pdf", width = 20, height = 6)


#Make T0 with DIVs ####
#try glom use raw counts 
DIVs_T0<-subset_samples(coralDIV2, Time.Point=="T0") #use raw counts df here

T0_its2DIV <- DIVs_T0 %>%  #make relative abundance
  tax_glom(taxrank = "DIV", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
T0_its2DIV.melt <- psmelt(T0_its2DIV)

# Extract the sample order (should now work) from Clade plot
sample_order <- levels(T0_its2.melt1$Sample)

# Check if sample_order is not NULL
print(sample_order)  # Should print a list of sample names in order
T0_its2DIV.melt$Sample
T0_its2DIV.melt <- T0_its2DIV.melt %>%
  mutate(Sample = factor(Sample, levels = sample_order))

T0_its2DIV.melt <- T0_its2DIV.melt %>%
  mutate(Species = factor(Species, levels = c("Montipora_capitata", "Porites_compressa", "Pavona_varians", "Pocillopora_acuta")))  # Adjust order as needed

# Ensure colors are assigned based on Clade groups in `its2_type_profile`
T0DIV.p  <- ggplot(T0_its2DIV.melt, aes(x = Sample, y = Abundance, fill = DIV)) +
  geom_bar(stat = "identity", color = "black", size = 0.2) +  # Add black outlines
  ylab("Relative Abundance") +
  xlab("") +
  theme_minimal() +  # Apply minimal theme (white background, no grid)
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1),  # Rotate x-axis labels
    panel.grid = element_blank(),  # Remove grid lines
    plot.background = element_blank(),  # Remove the outer background
    axis.line = element_line(color = "black"),  # Add black axis lines
    legend.position = "none"  # Remove the legend
  );T0DIV.p


T0_ITS2_DIV_samp <- T0DIV.p + facet_wrap(~Species, scales = "free_x", nrow = 1)
T0_ITS2_DIV_samp

#ggsave("ITS2_T0_DIV_samp.pdf", plot = T0_ITS2_DIV_samp, device = "pdf", width = 20, height = 6)

```

#Beta Diversity - All Species
coral_DIV2_rare_Rel df
```{r}              
DIV_RelA_stats<- coral_DIV2_rare_Rel #make a copy
otu_table(DIV_RelA_stats)[1:5, 1:5] #check rela worked


# Pull out sample data
meta <- sample_data(DIV_RelA_stats)

# Convert character columns back to factors
meta[] <- lapply(meta, function(x) if (is.character(x)) as.factor(x) else x)

# Reassign
sample_data(DIV_RelA_stats) <- meta




#bray curtis presence-absense and abundance
```
#### T0 only ######
Figure 2C
```{r}
DIV_RelA_stats.T0<-subset_samples(DIV_RelA_stats, Time.Point=="T0")
  DIV_RelA_stats.T0 = prune_taxa(taxa_sums(DIV_RelA_stats.T0) > 0, DIV_RelA_stats.T0) #this removes any OTU with 0s

  #save as object for mantel test in the 16s rmd
  #saveRDS(DIV_RelA_stats.T0,"ITS2_T0_phyloseq.rds")
  
#bray curtis presence-absense and abundance
#all speices
set.seed(30)

#make bray curtis data matrix
bc.all.T0 <- phyloseq::distance(DIV_RelA_stats.T0, method = "bray")
  samp.df.T0 <- as(sample_data(DIV_RelA_stats.T0), "data.frame") #sample df
#adonis test
adonis2(bc.all.T0 ~ Species, data = samp.df.T0, by="margin")
    
# Homogeneity of dispersion test
betaAll <- betadisper(bc.all.T0, samp.df.T0$Species)
perm<-permutest(betaAll, pairwise = T) 
(bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
plot(bc.all.HSD)

#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.T0 <- ordinate(DIV_RelA_stats.T0, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.T0)
scores.ord.DIV_RelA_stats.T0 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.T0, display="sites"))) #make df
scores.ord.DIV_RelA_stats.T0$Treatment <- samp.df.T0$Treatment
scores.ord.DIV_RelA_stats.T0$Species <- samp.df.T0$Species
scores.ord.DIV_RelA_stats.T0$Clade <- samp.df.T0$Clade
scores.ord.DIV_RelA_stats.T0$Code <- paste(samp.df.T0$Species,samp.df.T0$Clade)

# FIGURE 2C ITS2 T0 - shown by Profile group  ####

its2plot<-ggplot(scores.ord.DIV_RelA_stats.T0, aes(x = NMDS1, y = NMDS2, color = Code)) + 
  geom_point(size = 2) +
    stat_ellipse(aes(fill = Code), level = 0.95, alpha = 0.2, linetype = 1, size = 0.5, geom = "polygon") + 
scale_color_manual(values=c("goldenrod1","darkorange3", "powderblue","#0072B2" ,"pink1","firebrick",   "palegreen2","darkgreen")) +  # Set colors for points
   scale_fill_manual(values=c("goldenrod1","darkorange3", "powderblue","#0072B2" ,"pink1","firebrick",   "palegreen2","darkgreen"))+   
  xlim(-3, 3) +  # Set lower bound of x-axis to -2.5
    ylim(-3, 3) +  # Set lower bound of x-axis to -2.5
  ggtitle("NMDS - Algal Symbiont Communities") +
   theme_classic() +
  theme(
    axis.text = element_text(size = 9),        # Axis text size
    axis.title = element_text(size = 9),       # Axis labels size
    #legend.position = "none",                   # Remove legend
    plot.title = element_text(size = 12)        # Title size
  )+
  coord_fixed(ratio = 1);its2plot  # Ensure x and y axes have the same scaling

#ggsave("Figures/NMDS_ITS2_T0clade.pdf", plot = its2plot, device = "pdf")
```
#T1TF #### 
```{r}
# Subset T1 TF for models
DIV_RelA_stats.T1F<-subset_samples(DIV_RelA_stats, Time.Point=="T1"|Time.Point=="TF")
  DIV_RelA_stats.T1F = prune_taxa(taxa_sums(DIV_RelA_stats.T1F) > 0, DIV_RelA_stats.T1F) #this removes any OTU with 0s

  #save as object for mantel test in the 16s rmd
  #saveRDS(DIV_RelA_stats.T1F,"ITS2_T1TF_phyloseq.rds")

#T1 
DIV_RelA_stats.T1<-subset_samples(DIV_RelA_stats, Time.Point=="T1")
  DIV_RelA_stats.T1 = prune_taxa(taxa_sums(DIV_RelA_stats.T1) > 0, DIV_RelA_stats.T1) #this removes any OTU with 0s

#bray curtis presence-absense and abundance
#all speices
set.seed(30)
bc.all <- phyloseq::distance(DIV_RelA_stats.T1, method = "bray")#make bray curtis data matrix
  samp.df <- as(sample_data(DIV_RelA_stats.T1), "data.frame") #sample df
#adonis test
adonis2(bc.all ~ Species*Treatment, data = samp.df)
    
# Homogeneity of dispersion test
betaAll <- betadisper(bc.all, samp.df$Species)
permutest(betaAll, pairwise = T) 

#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.T1 <- ordinate(DIV_RelA_stats.T1, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.T1)
scores.ord.DIV_RelA_statsT1 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.T1, display="sites"))) #make df
scores.ord.DIV_RelA_statsT1$Treatment <- samp.df$Treatment
scores.ord.DIV_RelA_statsT1$Species <- samp.df$Species
scores.ord.DIV_RelA_statsT1$Clade <- samp.df$Clade
scores.ord.DIV_RelA_statsT1$Code3<-paste(scores.ord.DIV_RelA_statsT1$Species,scores.ord.DIV_RelA_statsT1$Treatment)
scores.ord.DIV_RelA_statsT1$Code3<-paste(scores.ord.DIV_RelA_statsT1$Species,scores.ord.DIV_RelA_statsT1$Treatment)

ggplot(scores.ord.DIV_RelA_statsT1, aes(x = NMDS1, y = NMDS2, color = Species, linetype=Treatment)) +
  geom_point(aes(shape=Treatment), size=4) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code3), 
               alpha = 0.25)+
    theme_classic()

ggplot(scores.ord.DIV_RelA_statsT1, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Treatment), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(color=Clade, fill=4, alpha=0.25)) +
scale_linetype_manual(values=c(1,2,1,2,1,2,1,2))+
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()

#TF only

DIV_RelA_stats.TF<-subset_samples(DIV_RelA_stats, Time.Point=="TF")
  DIV_RelA_stats.TF = prune_taxa(taxa_sums(DIV_RelA_stats.TF) > 0, DIV_RelA_stats.TF) #this removes any OTU with 0s

#all speices
set.seed(30)
#make bray curtis data matrix
bc.all <- phyloseq::distance(DIV_RelA_stats.TF, method = "bray")
  samp.df <- as(sample_data(DIV_RelA_stats.TF), "data.frame") #sample df
#adonis test
adonis2(bc.all ~ Species*Treatment, data = samp.df)
    
# Homogeneity of dispersion test
betaAll <- betadisper(bc.all, samp.df$Species)
permutest(betaAll, pairwise = T) 

#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.TF <- ordinate(DIV_RelA_stats.TF, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.TF)
scores.ord.DIV_RelA_stats <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.TF, display="sites"))) #make df
scores.ord.DIV_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.DIV_RelA_stats$Species <- samp.df$Species
scores.ord.DIV_RelA_stats$Clade <- samp.df$Clade
scores.ord.DIV_RelA_stats$Code3<-paste(scores.ord.DIV_RelA_stats$Species,scores.ord.DIV_RelA_stats$Treatment)

ggplot(scores.ord.DIV_RelA_stats, aes(x = NMDS1, y = NMDS2, color = Species, linetype=Treatment)) +
  geom_point(aes(shape=Treatment), size=4) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code3), 
               alpha = 0.25)+
    theme_classic()
```

#T0 Beta diversiry and dispersion, T0
Table S17
```{r} 
#T0 only DIV_RelA_stats.T0

#Mcap
mcap.RelRare.T0 <- subset_samples(DIV_RelA_stats.T0, host_genus == "montipora")
    mcap.RelRare.T0 = prune_taxa(taxa_sums(mcap.RelRare.T0) > 0, mcap.RelRare.T0) 

#make bray curtis data matrix
bc.mcap.T0 <- phyloseq::distance(mcap.RelRare.T0, method = "bray") 
      mcap.DIV.samp.df.T0 <- as(sample_data(mcap.RelRare.T0), "data.frame") 
            mcap.DIV.samp.df.T0$Code<-paste(mcap.DIV.samp.df.T0$Treatment,mcap.DIV.samp.df.T0$Clade)    
    set.seed(30)
    adonis2(bc.mcap.T0 ~ Clade, data = mcap.DIV.samp.df.T0, by="margin") #adonis test no clade
    
     set.seed(30)
    permutest(betadisper(bc.mcap.T0, mcap.DIV.samp.df.T0$Clade, type = "centroid")) #put in distance matrix
    
  #Pcomp
Pcomp.RelRare.T0 <- subset_samples(DIV_RelA_stats.T0, host_genus == "porites")
    Pcomp.RelRare.T0 = prune_taxa(taxa_sums(Pcomp.RelRare.T0) > 0, Pcomp.RelRare.T0) 

#make bray curtis data matrix
bc.Pcomp.T0 <- phyloseq::distance(Pcomp.RelRare.T0, method = "bray") 
      Pcomp.DIV.samp.df.T0 <- as(sample_data(Pcomp.RelRare.T0), "data.frame") 
            Pcomp.DIV.samp.df.T0$Code<-paste(Pcomp.DIV.samp.df.T0$Treatment,Pcomp.DIV.samp.df.T0$Clade)    
    set.seed(30)
    adonis2(bc.Pcomp.T0 ~ Clade, data = Pcomp.DIV.samp.df.T0, by="margin") #adonis test no clade
    
     set.seed(30)
    permutest(betadisper(bc.Pcomp.T0, Pcomp.DIV.samp.df.T0$Clade, type = "centroid")) #put in distance matrix 
    
 #Pvar
Pvar.RelRare.T0 <- subset_samples(DIV_RelA_stats.T0, host_genus == "pavona")
    Pvar.RelRare.T0 = prune_taxa(taxa_sums(Pvar.RelRare.T0) > 0, Pvar.RelRare.T0) 

#make bray curtis data matrix
bc.Pvar.T0 <- phyloseq::distance(Pvar.RelRare.T0, method = "bray") 
      Pvar.DIV.samp.df.T0 <- as(sample_data(Pvar.RelRare.T0), "data.frame") 
            Pvar.DIV.samp.df.T0$Code<-paste(Pvar.DIV.samp.df.T0$Treatment,Pvar.DIV.samp.df.T0$Clade)    
    set.seed(30)
    adonis2(bc.Pvar.T0 ~ Clade, data = Pvar.DIV.samp.df.T0, by="margin") #adonis test no clade
    
     set.seed(30)
    permutest(betadisper(bc.Pvar.T0, Pvar.DIV.samp.df.T0$Clade, type = "centroid")) #put in distance matrix
       
#Pacu
Pacu.RelRare.T0 <- subset_samples(DIV_RelA_stats.T0, host_genus == "pocillopora")
    Pacu.RelRare.T0 = prune_taxa(taxa_sums(Pacu.RelRare.T0) > 0, Pacu.RelRare.T0) 

#make bray curtis data matrix
bc.Pacu.T0 <- phyloseq::distance(Pacu.RelRare.T0, method = "bray") 
      Pacu.DIV.samp.df.T0 <- as(sample_data(Pacu.RelRare.T0), "data.frame") 
            Pacu.DIV.samp.df.T0$Code<-paste(Pacu.DIV.samp.df.T0$Treatment,Pacu.DIV.samp.df.T0$Clade)    
    set.seed(30)
    adonis2(bc.Pacu.T0 ~ Clade, data = Pacu.DIV.samp.df.T0, by="margin") #adonis test no clade
    
     set.seed(30)
    permutest(betadisper(bc.Pacu.T0, Pacu.DIV.samp.df.T0$Clade, type = "centroid")) #put in distance matrix   
```
#Beta T1, TF, by treatment: 
Tables S9, S12
Figure S19
Montipora
```{r}       
#DIV_RelA_stats.T1F ####
#mcap
mcap.RelRare.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "montipora")
    mcap.RelRare.T1F = prune_taxa(taxa_sums(mcap.RelRare.T1F) > 0, mcap.RelRare.T1F) 
  mcap.RelRare.T1F.samp.df <- as(sample_data(mcap.RelRare.T1F), "data.frame") #sample df

mcap.RelRare.T1 <- subset_samples(mcap.RelRare.T1F, Time.Point == "T1")
    mcap.RelRare.T1 = prune_taxa(taxa_sums(mcap.RelRare.T1) > 0, mcap.RelRare.T1) 
mcap.RelRare.T1.samp.df <- as(sample_data(mcap.RelRare.T1), "data.frame") #sample df

mcap.RelRare.TF <- subset_samples(mcap.RelRare.T1F, Time.Point == "TF")
    mcap.RelRare.TF = prune_taxa(taxa_sums(mcap.RelRare.TF) > 0, mcap.RelRare.TF) 
  mcap.RelRare.TF.samp.df <- as(sample_data(mcap.RelRare.TF), "data.frame") #sample df

#T1
  bc.mcap.T1 <- phyloseq::distance(mcap.RelRare.T1, method = "bray") 
      mcap.DIV.samp.df.T1 <- as(sample_data(mcap.RelRare.T1), "data.frame")
      mcap.DIV.samp.df.T1 <- mcap.DIV.samp.df.T1 %>%
  mutate(
    Clade = as.factor(Clade),
    Treatment = as.factor(Treatment)
  )
    mcap.DIV.samp.df.T1$Code<-paste(mcap.DIV.samp.df.T1$Treatment, mcap.DIV.samp.df.T1$Clade)
  
set.seed(30)
    adonis2(bc.mcap.T1 ~ Treatment, data = mcap.DIV.samp.df.T1, by="terms")
set.seed(30)
        adonis2(bc.mcap.T1 ~ Clade*Treatment, data = mcap.DIV.samp.df.T1, by="terms")  

# Homogeneity of dispersion test  
  set.seed(30)
    permutest(betadisper(bc.mcap.T1, mcap.DIV.samp.df.T1$Treatment, type = "centroid"))
  set.seed(30)
    permutest(betadisper(bc.mcap.T1, mcap.DIV.samp.df.T1$Code, type = "centroid"))

#TF
  bc.mcap.TF <- phyloseq::distance(mcap.RelRare.TF, method = "bray") 
      mcap.DIV.samp.df.TF <- as(sample_data(mcap.RelRare.TF), "data.frame")
      mcap.DIV.samp.df.TF <- mcap.DIV.samp.df.TF %>%
  mutate(
    Clade = as.factor(Clade),
    Treatment = as.factor(Treatment)
  )
    mcap.DIV.samp.df.TF$Code<-paste(mcap.DIV.samp.df.TF$Treatment, mcap.DIV.samp.df.TF$Clade)
  
set.seed(30)
    adonis2(bc.mcap.TF ~ Treatment, data = mcap.DIV.samp.df.TF, by="terms")
set.seed(30)
        adonis2(bc.mcap.TF ~ Clade*Treatment, data = mcap.DIV.samp.df.TF, by="terms")  

# Homogeneity of dispersion test  
  set.seed(30)
    permutest(betadisper(bc.mcap.TF, mcap.DIV.samp.df.TF$Treatment, type = "centroid"))
  set.seed(30)
    permutest(betadisper(bc.mcap.TF, mcap.DIV.samp.df.TF$Code, type = "centroid"))

# NMDS ####
## create palette for parent.ID and treatments
Mcap.Parent.colors <- c("363" = "green4",  "364" = "firebrick3", "365" ="deeppink4", "366" = "orange", "367" = "green1","368" = "gold",  "369"= "hotpink", "370" ="darkslateblue", "371" = "cyan", "372" = "darkorange","373" ="darkorchid1", "374" = "lightblue", "Ambient" = "#4575B4", "High" = "#D73027","C Ambient"="blue","C High"="red", "CD Ambient"="darkblue", "CD High"="darkred") 

#T1 and TF
ord.mcap <- ordinate(mcap.RelRare.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.mcap)
scores.mcap.samp.df <- as.data.frame(cbind(vegan::scores(ord.mcap, display="sites"))) #make df
scores.mcap.samp.df$Time.Point <- mcap.RelRare.T1F.samp.df$Time.Point
scores.mcap.samp.df$Parent.ID <- mcap.RelRare.T1F.samp.df$Parent.ID
scores.mcap.samp.df$Treatment <- mcap.RelRare.T1F.samp.df$Treatment
scores.mcap.samp.df$ID <- mcap.RelRare.T1F.samp.df$ID
scores.mcap.samp.df$Clade <- mcap.RelRare.T1F.samp.df$Clade
scores.mcap.samp.df$Code <- paste(mcap.RelRare.T1F.samp.df$Parent.ID,mcap.RelRare.T1F.samp.df$Time.Point)
scores.mcap.samp.df$Code2 <- paste(mcap.RelRare.T1F.samp.df$Treatment,mcap.RelRare.T1F.samp.df$Time.Point)

#Check if Parent.ID shifts within time point
#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-Section-01-Main-Lab.html#procrustes-and-ggplot2 
McapITS2_T1TF<-ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Mcap.Parent.colors)+
    geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle(expression(italic("M. capitata")~"ITS2")) +
  theme_classic()+
  facet_wrap(~Time.Point);McapITS2_T1TF
#ggsave("McapT1its2.pdf")

#NMDS Mcap T1 and TF
 
 McapITS2_T1TFplot<-ggplot(scores.mcap.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Treatment), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle(expression(italic("M. capitata")~"ITS2")) +
  theme_classic()+
  facet_wrap(~Time.Point);McapITS2_T1TFplot

```
Porites compressa
```{r}       
#DIV_RelA_stats.T1F ####
#pcomp
pcomp.RelRare.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "porites")
    pcomp.RelRare.T1F = prune_taxa(taxa_sums(pcomp.RelRare.T1F) > 0, pcomp.RelRare.T1F) 
  pcomp.RelRare.T1F.samp.df <- as(sample_data(pcomp.RelRare.T1F), "data.frame") #sample df

pcomp.RelRare.T1 <- subset_samples(pcomp.RelRare.T1F, Time.Point == "T1")
    pcomp.RelRare.T1 = prune_taxa(taxa_sums(pcomp.RelRare.T1) > 0, pcomp.RelRare.T1) 
pcomp.RelRare.T1.samp.df <- as(sample_data(pcomp.RelRare.T1), "data.frame") #sample df

pcomp.RelRare.TF <- subset_samples(pcomp.RelRare.T1F, Time.Point == "TF")
    pcomp.RelRare.TF = prune_taxa(taxa_sums(pcomp.RelRare.TF) > 0, pcomp.RelRare.TF) 
  pcomp.RelRare.TF.samp.df <- as(sample_data(pcomp.RelRare.TF), "data.frame") #sample df

#T1
  bc.pcomp.T1 <- phyloseq::distance(pcomp.RelRare.T1, method = "bray") 
      pcomp.DIV.samp.df.T1 <- as(sample_data(pcomp.RelRare.T1), "data.frame")
      pcomp.DIV.samp.df.T1 <- pcomp.DIV.samp.df.T1 %>%
  mutate(
    Clade = as.factor(Clade),
    Treatment = as.factor(Treatment)
  )
    pcomp.DIV.samp.df.T1$Code<-paste(pcomp.DIV.samp.df.T1$Treatment, pcomp.DIV.samp.df.T1$Clade)
  
set.seed(30)
    adonis2(bc.pcomp.T1 ~ Treatment, data = pcomp.DIV.samp.df.T1, by="terms")
set.seed(30)
        adonis2(bc.pcomp.T1 ~ Clade*Treatment, data = pcomp.DIV.samp.df.T1, by="terms")  

# Homogeneity of dispersion test  
  set.seed(30)
    permutest(betadisper(bc.pcomp.T1, pcomp.DIV.samp.df.T1$Treatment, type = "centroid"))
  set.seed(30)
    permutest(betadisper(bc.pcomp.T1, pcomp.DIV.samp.df.T1$Code, type = "centroid"))

#TF
  bc.pcomp.TF <- phyloseq::distance(pcomp.RelRare.TF, method = "bray") 
      pcomp.DIV.samp.df.TF <- as(sample_data(pcomp.RelRare.TF), "data.frame")
      pcomp.DIV.samp.df.TF <- pcomp.DIV.samp.df.TF %>%
  mutate(
    Clade = as.factor(Clade),
    Treatment = as.factor(Treatment)
  )
    pcomp.DIV.samp.df.TF$Code<-paste(pcomp.DIV.samp.df.TF$Treatment, pcomp.DIV.samp.df.TF$Clade)
  
set.seed(30)
    adonis2(bc.pcomp.TF ~ Treatment, data = pcomp.DIV.samp.df.TF, by="terms")
set.seed(30)
        adonis2(bc.pcomp.TF ~ Clade*Treatment, data = pcomp.DIV.samp.df.TF, by="terms")  

# Homogeneity of dispersion test  
  set.seed(30)
    permutest(betadisper(bc.pcomp.TF, pcomp.DIV.samp.df.TF$Treatment, type = "centroid"))
  set.seed(30)
    permutest(betadisper(bc.pcomp.TF, pcomp.DIV.samp.df.TF$Code, type = "centroid"))

# NMDS ####
## create palette for parent.ID and treatments
Pcomp.Parent.colors <- c("313" = "green4",  "314" = "firebrick3", "315" ="deeppink4", "316" = "orange", "317" = "green1","318" = "gold",  "319"= "hotpink", "320" ="darkslateblue", "321" = "cyan", "322" = "darkorange","323" ="darkorchid1", "324" = "lightblue","Ambient" = "#4575B4", "High" = "#D73027",     "C15cj Ambient"="blue","C15cj High"="red", "C15n Ambient"="darkblue", "C15n High"="darkred") 

#T1 and TF
ord.pcomp <- ordinate(pcomp.RelRare.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.pcomp)
scores.pcomp.samp.df <- as.data.frame(cbind(vegan::scores(ord.pcomp, display="sites"))) #make df
scores.pcomp.samp.df$Time.Point <- pcomp.RelRare.T1F.samp.df$Time.Point
scores.pcomp.samp.df$Parent.ID <- pcomp.RelRare.T1F.samp.df$Parent.ID
scores.pcomp.samp.df$Treatment <- pcomp.RelRare.T1F.samp.df$Treatment
scores.pcomp.samp.df$Clade <- pcomp.RelRare.T1F.samp.df$Clade
scores.pcomp.samp.df$Code <- paste(pcomp.RelRare.T1F.samp.df$Parent.ID,pcomp.RelRare.T1F.samp.df$Time.Point)
scores.pcomp.samp.df$Code2 <- paste(pcomp.RelRare.T1F.samp.df$Treatment,pcomp.RelRare.T1F.samp.df$Time.Point)

#Check if Parent.ID shifts within time point
#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-Section-01-Main-Lab.html#procrustes-and-ggplot2 
PcompITS2_T1TF<-ggplot(scores.pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
    geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle(expression(italic("P. compressa")~"ITS2")) +
  theme_classic()+
  facet_wrap(~Time.Point);PcompITS2_T1TF
#ggsave("PcompT1its2.pdf")

PcompITS2_T1TFplot<-ggplot(scores.pcomp.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Treatment), size = 4) +
  scale_color_manual(values=Pcomp.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle(expression(italic("P. compressa")~"ITS2")) +
  theme_classic()+
  facet_wrap(~Time.Point);PcompITS2_T1TFplot

#ggsave("PcompT1its2.pdf")

```
Pavona varians
```{r}       
#DIV_RelA_stats.T1F ####
#pvar
pvar.RelRare.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pavona")
    pvar.RelRare.T1F = prune_taxa(taxa_sums(pvar.RelRare.T1F) > 0, pvar.RelRare.T1F) 
  pvar.RelRare.T1F.samp.df <- as(sample_data(pvar.RelRare.T1F), "data.frame") #sample df

pvar.RelRare.T1 <- subset_samples(pvar.RelRare.T1F, Time.Point == "T1")
    pvar.RelRare.T1 = prune_taxa(taxa_sums(pvar.RelRare.T1) > 0, pvar.RelRare.T1) 
pvar.RelRare.T1.samp.df <- as(sample_data(pvar.RelRare.T1), "data.frame") #sample df

pvar.RelRare.TF <- subset_samples(pvar.RelRare.T1F, Time.Point == "TF")
    pvar.RelRare.TF = prune_taxa(taxa_sums(pvar.RelRare.TF) > 0, pvar.RelRare.TF) 
  pvar.RelRare.TF.samp.df <- as(sample_data(pvar.RelRare.TF), "data.frame") #sample df

#T1
  bc.pvar.T1 <- phyloseq::distance(pvar.RelRare.T1, method = "bray") 
      pvar.DIV.samp.df.T1 <- as(sample_data(pvar.RelRare.T1), "data.frame")
      pvar.DIV.samp.df.T1 <- pvar.DIV.samp.df.T1 %>%
  mutate(
    Clade = as.factor(Clade),
    Treatment = as.factor(Treatment)
  )
    pvar.DIV.samp.df.T1$Code<-paste(pvar.DIV.samp.df.T1$Treatment, pvar.DIV.samp.df.T1$Clade)
  
set.seed(30)
    adonis2(bc.pvar.T1 ~ Treatment, data = pvar.DIV.samp.df.T1, by="terms")
set.seed(30)
        adonis2(bc.pvar.T1 ~ Clade*Treatment, data = pvar.DIV.samp.df.T1, by="terms")  

# Homogeneity of dispersion test  
  set.seed(30)
    permutest(betadisper(bc.pvar.T1, pvar.DIV.samp.df.T1$Treatment, type = "centroid"))
  set.seed(30)
    permutest(betadisper(bc.pvar.T1, pvar.DIV.samp.df.T1$Code, type = "centroid"))

#TF
  bc.pvar.TF <- phyloseq::distance(pvar.RelRare.TF, method = "bray") 
      pvar.DIV.samp.df.TF <- as(sample_data(pvar.RelRare.TF), "data.frame")
      pvar.DIV.samp.df.TF <- pvar.DIV.samp.df.TF %>%
  mutate(
    Clade = as.factor(Clade),
    Treatment = as.factor(Treatment)
  )
    pvar.DIV.samp.df.TF$Code<-paste(pvar.DIV.samp.df.TF$Treatment, pvar.DIV.samp.df.TF$Clade)
  
set.seed(30)
    adonis2(bc.pvar.TF ~ Treatment, data = pvar.DIV.samp.df.TF, by="terms")
set.seed(30)
        adonis2(bc.pvar.TF ~ Clade*Treatment, data = pvar.DIV.samp.df.TF, by="terms")  

# Homogeneity of dispersion test  
  set.seed(30)
    permutest(betadisper(bc.pvar.TF, pvar.DIV.samp.df.TF$Treatment, type = "centroid"))
  set.seed(30)
    permutest(betadisper(bc.pvar.TF, pvar.DIV.samp.df.TF$Code, type = "centroid"))

# NMDS ####
## create palette for parent.ID and treatments
Pvar.Parent.colors <- c("325" = "green4",  "351" = "firebrick3", "353" ="deeppink4", "354" = "orange", "355" = "green1","356" = "gold",  "357"= "hotpink", "358" ="darkslateblue", "359" = "honeydew4", "360" = "khaki4","361" ="ivory4", "362" = "lightblue", "Ambient" = "#4575B4", "High" = "#D73027","C1 Ambient"="blue","C1 High"="red", "C27 Ambient"="darkblue", "C27 High"="darkred") 

#T1 and TF
ord.pvar <- ordinate(pvar.RelRare.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.pvar)
scores.pvar.samp.df <- as.data.frame(cbind(vegan::scores(ord.pvar, display="sites"))) #make df
scores.pvar.samp.df$Time.Point <- pvar.RelRare.T1F.samp.df$Time.Point
scores.pvar.samp.df$Parent.ID <- pvar.RelRare.T1F.samp.df$Parent.ID
scores.pvar.samp.df$Treatment <- pvar.RelRare.T1F.samp.df$Treatment
scores.pvar.samp.df$Clade <- pvar.RelRare.T1F.samp.df$Clade
scores.pvar.samp.df$Code <- paste(pvar.RelRare.T1F.samp.df$Parent.ID,pvar.RelRare.T1F.samp.df$Time.Point)
scores.pvar.samp.df$Code2 <- paste(pvar.RelRare.T1F.samp.df$Treatment,pvar.RelRare.T1F.samp.df$Time.Point)

#Check if Parent.ID shifts within time point
#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-Section-01-Main-Lab.html#procrustes-and-ggplot2 
PvarITS2_T1TF<-ggplot(scores.pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Clade), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
    geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle(expression(italic("P. varians")~"ITS2")) +
  theme_classic()+
  facet_wrap(~Time.Point);PvarITS2_T1TF

PvarITS2_T1TFplot<-ggplot(scores.pvar.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment, shape = Treatment), size = 4) +
  scale_color_manual(values=Pvar.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle(expression(italic("P. varians")~"ITS2")) +
  theme_classic()+
  facet_wrap(~Time.Point);PvarITS2_T1TFplot



#ggsave("PvarT1its2.pdf")
```
Pocillopora acuta
```{r}       
#DIV_RelA_stats.T1F ####
#pacu
pacu.RelRare.T1F <- subset_samples(DIV_RelA_stats.T1F, host_genus == "pocillopora")
    pacu.RelRare.T1F = prune_taxa(taxa_sums(pacu.RelRare.T1F) > 0, pacu.RelRare.T1F) 
  pacu.RelRare.T1F.samp.df <- as(sample_data(pacu.RelRare.T1F), "data.frame") #sample df

pacu.RelRare.T1 <- subset_samples(pacu.RelRare.T1F, Time.Point == "T1")
    pacu.RelRare.T1 = prune_taxa(taxa_sums(pacu.RelRare.T1) > 0, pacu.RelRare.T1) 
pacu.RelRare.T1.samp.df <- as(sample_data(pacu.RelRare.T1), "data.frame") #sample df

pacu.RelRare.TF <- subset_samples(pacu.RelRare.T1F, Time.Point == "TF")
    pacu.RelRare.TF = prune_taxa(taxa_sums(pacu.RelRare.TF) > 0, pacu.RelRare.TF) 
  pacu.RelRare.TF.samp.df <- as(sample_data(pacu.RelRare.TF), "data.frame") #sample df

#T1
  bc.pacu.T1 <- phyloseq::distance(pacu.RelRare.T1, method = "bray") 
      pacu.DIV.samp.df.T1 <- as(sample_data(pacu.RelRare.T1), "data.frame")
      pacu.DIV.samp.df.T1 <- pacu.DIV.samp.df.T1 %>%
  mutate(
    Clade = as.factor(Clade),
    Treatment = as.factor(Treatment)
  )
    pacu.DIV.samp.df.T1$Code<-paste(pacu.DIV.samp.df.T1$Treatment, pacu.DIV.samp.df.T1$Clade)
  
set.seed(30)
    adonis2(bc.pacu.T1 ~ Treatment, data = pacu.DIV.samp.df.T1, by="terms")
set.seed(30)
        adonis2(bc.pacu.T1 ~ Clade*Treatment, data = pacu.DIV.samp.df.T1, by="terms")  

# Homogeneity of dispersion test  
  set.seed(30)
    permutest(betadisper(bc.pacu.T1, pacu.DIV.samp.df.T1$Treatment, type = "centroid"))


#TF
  bc.pacu.TF <- phyloseq::distance(pacu.RelRare.TF, method = "bray") 
      pacu.DIV.samp.df.TF <- as(sample_data(pacu.RelRare.TF), "data.frame")
      pacu.DIV.samp.df.TF <- pacu.DIV.samp.df.TF %>%
  mutate(
    Clade = as.factor(Clade),
    Treatment = as.factor(Treatment)
  )
    pacu.DIV.samp.df.TF$Code<-paste(pacu.DIV.samp.df.TF$Treatment, pacu.DIV.samp.df.TF$Clade)
  
set.seed(30)
    adonis2(bc.pacu.TF ~ Treatment, data = pacu.DIV.samp.df.TF, by="terms")
set.seed(30)
        adonis2(bc.pacu.TF ~ Clade*Treatment, data = pacu.DIV.samp.df.TF, by="terms")  

# Homogeneity of dispersion test  
  set.seed(30)
    permutest(betadisper(bc.pacu.TF, pacu.DIV.samp.df.TF$Treatment, type = "centroid"))
  set.seed(30)
    permutest(betadisper(bc.pacu.TF, pacu.DIV.samp.df.TF$Code, type = "centroid"))

# NMDS ####
## create palette for parent.ID and treatments
Pacu.Parent.colors <- c("305" = "green4",  "306" = "firebrick3", "307" ="deeppink4", "308" = "orange", "309" = "green1","310" = "gold",  "311"= "hotpink", "312" ="darkslateblue", "302" = "honeydew4", "303" = "khaki4","304" ="ivory4", "305" = "lightblue", "Ambient" = "#4575B4", "High" = "#D73027","CCC Ambient"="blue","CCC High"="red", "C42 Ambient"="darkblue", "C42 High"="darkred") 


#T1 and TF
ord.pacu <- ordinate(pacu.RelRare.T1F, method="NMDS", "bray", trymax = 100, set.seed(30)) 
stressplot(ord.pacu)
scores.pacu.samp.df <- as.data.frame(cbind(vegan::scores(ord.pacu, display="sites"))) #make df
scores.pacu.samp.df$Time.Point <- pacu.RelRare.T1F.samp.df$Time.Point
scores.pacu.samp.df$Parent.ID <- pacu.RelRare.T1F.samp.df$Parent.ID
scores.pacu.samp.df$Treatment <- pacu.RelRare.T1F.samp.df$Treatment
scores.pacu.samp.df$Clade <- pacu.RelRare.T1F.samp.df$Clade
scores.pacu.samp.df$Code <- paste(pacu.RelRare.T1F.samp.df$Parent.ID,pacu.RelRare.T1F.samp.df$Time.Point)
scores.pacu.samp.df$Code2 <- paste(pacu.RelRare.T1F.samp.df$Treatment,pacu.RelRare.T1F.samp.df$Time.Point)

#Check if Parent.ID shifts within time point
#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-Section-01-Main-Lab.html#procrustes-and-ggplot2 
PacuITS2_T1TF<-ggplot(scores.pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
    geom_line(aes(group = Code), color = "black") + 
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle(expression(italic("P. acuta")~"ITS2")) +
  theme_classic()+
  facet_wrap(~Time.Point);PacuITS2_T1TF
#ggsave("PacuT1its2.pdf")

PacuITS2_T1TFplot<-ggplot(scores.pacu.samp.df, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Treatment), size = 4) +
  scale_color_manual(values=Pacu.Parent.colors)+
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle(expression(italic("P. acuta")~"ITS2")) +
  theme_classic()+
  facet_wrap(~Time.Point);PacuITS2_T1TFplot
#ggsave("PacuT1its2.pdf")

```
All plots by lineage: Figure S19
```{r}
p<-grid.arrange(McapITS2_T1TF, PcompITS2_T1TF, PvarITS2_T1TF,PacuITS2_T1TF, nrow = 1)
ggsave( "ITS2_T1_TF_NMDS.pdf", plot= p, width=16, height=7, units="in")

```
All plots: Figure S15
```{r}
p<-grid.arrange(McapITS2_T1TFplot, PcompITS2_T1TFplot, PvarITS2_T1TFplot,PacuITS2_T1TFplot, nrow = 4)
ggsave( "ITS2_T1_TF_NMDS_FigS15.pdf", plot= p, width=7, height=16, units="in")

```



#Alpha diversity: observed richness, Shannon (+evenness) - microbiome packages  
Use rarefied data, not relative abunances   
df:coral_DIV2_rare
```{r}         

Sp.colors <- c("Montipora_capitata"= "#D43F3AFF","Porites_compressa"="#EEA236FF","Pavona_varians"="#5CB85CFF","Pocillopora_acuta"="#46B8DAFF", "High"="darkRed","Ambient"="navyBlue")

its2.al.alpha<-coral_DIV2_rare #make a copy of df, use COUNTS

#its2.al.alpha<-subset_samples(its2.al.alpha, Time.Point=="T1" |Time.Point=="TF")  # ONLY KEEP T1 and TF!
its2.al.alpha2 <- prune_taxa(taxa_sums(its2.al.alpha) > 0, its2.al.alpha) #prune taxa 0s
its2.al.alpha2.sd <- as(sample_data(its2.al.alpha2), "data.frame") #create sample data frame to view


#####try microbiome package for evenness and alpha diversity. use raw counts
library(microbiome)  
library(knitr)

erich.tab <-microbiome::alpha(its2.al.alpha2, index = "all") # pull out: observed, diversity_shannon, evenness_pielou

erich.tab2its<-erich.tab[-c(2:4,6:8,10:22)] #keep the cols you need

 erich.tab2its$Treatment <- its2.al.alpha2.sd$Treatment       #add its2k in meta
  erich.tab2its$Time.Point <- its2.al.alpha2.sd$Time.Point
  erich.tab2its$Parent.ID <- its2.al.alpha2.sd$Parent.ID
  erich.tab2its$ID <- its2.al.alpha2.sd$Frag.ID
  erich.tab2its$Species <- its2.al.alpha2.sd$Species
    erich.tab2its$Clade <- its2.al.alpha2.sd$Clade
    erich.tab2its$Tank.num <- its2.al.alpha2.sd$Tank.num

  erich.tab2its$Time.Point = factor(erich.tab2its$Time.Point, levels = c("T0", "T1", "TF"))
```
#Observed Richness 
Table S9, S17
```{r}     
library(stats)
#T0 only ####
  erich.tab2its.T0<-subset(erich.tab2its, Time.Point=="T0")
  hist(erich.tab2its.T0$observed)
  
  observed_alpha_mod<-lmer(observed ~ Species+(1|Tank.num), data=erich.tab2its.T0) 
set.seed(30)
  anova(observed_alpha_mod) 
  qqPlot(residuals(observed_alpha_mod)) 
  emm<-emmeans(observed_alpha_mod,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #Richness
p1 <- ggviolin(erich.tab2its.T0, x = "Species", y = "observed",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)
#Evenness
p1 <- ggviolin(erich.tab2its.T0, x = "Species", y = "evenness_pielou",
   add = "boxplot", fill = "Species", palette = c(Sp.colors)) 
  print(p1)

#Mcap T0
   erich.tab2its.T0mcap<-subset(erich.tab2its.T0, Species=="Montipora_capitata")
  hist(erich.tab2its.T0mcap$observed)
  
  observed_alpha_mod<-lmer(observed ~ Clade+(1|Tank.num), data=erich.tab2its.T0mcap) 
set.seed(30)
  anova(observed_alpha_mod) 
  emm<-emmeans(observed_alpha_mod,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
#Pcomp T0
   erich.tab2its.T0pcomp<-subset(erich.tab2its.T0, Species=="Porites_compressa")
  hist(erich.tab2its.T0pcomp$observed)
  
  observed_alpha_mod<-lmer(observed ~ Clade+(1|Tank.num), data=erich.tab2its.T0pcomp) 
set.seed(30)
  anova(observed_alpha_mod) 
  qqPlot(residuals(observed_alpha_mod)) 
  emm<-emmeans(observed_alpha_mod,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
#Pvar T0
   erich.tab2its.T0pvar<-subset(erich.tab2its.T0, Species=="Pavona_varians")
  hist(erich.tab2its.T0pvar$observed)
  
  observed_alpha_mod<-lmer(observed ~ Clade+(1|Tank.num), data=erich.tab2its.T0pvar) 
set.seed(30)
  anova(observed_alpha_mod) 
  qqPlot(residuals(observed_alpha_mod)) 
  emm<-emmeans(observed_alpha_mod,  ~ Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
```
```

T1TF
```{r}
##Mcap only ####
  erich.tab2its.mcap<-subset(erich.tab2its, Species=="Montipora_capitata")
  
  hist(erich.tab2its.mcap$observed)
ggqqplot(erich.tab2its.mcap$observed)

  erich.tab2its.mcap.t1tf<-subset(erich.tab2its.mcap, Time.Point=="T1"|Time.Point=="TF")
 #t1TF
  Mcap_alpha_mod<-lmer(observed ~ Time.Point*Treatment+(1|Parent.ID), data=erich.tab2its.mcap.t1tf) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed ~ Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2its.mcap.t1tf) #adding tank is singularity
  anova(Mcap_alpha_mod) 
qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2its.mcap.T1<-subset(erich.tab2its.mcap, Time.Point=="T1")

  Mcap_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID), data=erich.tab2its.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
#T1 by clade richness
 #C
  erich.tab2its.mcap.T1.C<-subset(erich.tab2its.mcap.T1, Clade=="C")

  Mcap_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID), data=erich.tab2its.mcap.T1.C) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  #CD
  erich.tab2its.mcap.T1.CD<-subset(erich.tab2its.mcap.T1, Clade=="CD")

  Mcap_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID), data=erich.tab2its.mcap.T1.CD) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  
  #tf
  erich.tab2its.mcap.TF<-subset(erich.tab2its.mcap, Time.Point=="TF")

   Mcap_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID), data=erich.tab2its.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(observed ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)

#t0  <- WHY IS THIS ONE BEING WEIRD CAN YOU USE OR NO
  erich.tab2its.mcap.T0<-subset(erich.tab2its.mcap, Time.Point=="T0")
hist(erich.tab2its.mcap.T0$observed)
   Mcap_alpha_mod<-lmer(observed ~ Clade+(1|Tank.num), data=erich.tab2its.mcap.T0) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)   
  
####### PCOMP only ####
   
  #t1TF
  erich.tab2its.pcompT1TF<-subset(erich.tab2its.pcomp, Time.Point=="T1"|Time.Point=="TF")
  pcomp_alpha_mod<-lmer(observed ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pcompT1TF) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pcomp) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2its.pcomp.T1<-subset(erich.tab2its.pcomp, Time.Point=="T1")

  pcomp_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID), data=erich.tab2its.pcomp.T1) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pcomp.T1)
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2its.pcomp.TF<-subset(erich.tab2its.pcomp, Time.Point=="TF")

   pcomp_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(observed ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  #t0
  erich.tab2its.pcomp.T0<-subset(erich.tab2its.pcomp, Time.Point=="T0")
hist(erich.tab2its.pcomp.T0$observed)
   pcomp_alpha_mod<-lmer(observed ~ Clade+(1|Tank.num), data=erich.tab2its.pcomp.T0) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)  

####### pvar only####
   
  #t1TF
  erich.tab2its.pvarT1TF<-subset(erich.tab2its.pvar, Time.Point=="T1"|Time.Point=="TF")

  pvar_alpha_mod<-lmer(observed ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pvarT1TF) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pvar) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2its.pvar.T1<-subset(erich.tab2its.pvar, Time.Point=="T1")

  pvar_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2its.pvar.TF<-subset(erich.tab2its.pvar, Time.Point=="TF")

   pvar_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(observed ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
#t0
  erich.tab2its.pvar.T0<-subset(erich.tab2its.pvar, Time.Point=="T0")
hist(erich.tab2its.pvar.T0$observed)
   pvar_alpha_mod<-lmer(observed ~ Clade+(1|Tank.num), data=erich.tab2its.pvar.T0) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)    
  ####### pacu only####

  #t1TF
    erich.tab2its.pacuT1TF<-subset(erich.tab2its.pacu, Time.Point=="T1"|Time.Point=="TF")

  pacu_alpha_mod<-lmer(observed ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pacuT1TF) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pacu) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2its.pacu.T1<-subset(erich.tab2its.pacu, Time.Point=="T1")

  pacu_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID), data=erich.tab2its.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2its.pacu.TF<-subset(erich.tab2its.pacu, Time.Point=="TF")

   pacu_alpha_mod<-lmer(observed ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(observed ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pacu.TF) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t0
  erich.tab2its.pacu.T0<-subset(erich.tab2its.pacu, Time.Point=="T0")
hist(erich.tab2its.pacu.T0$observed)
   pacu_alpha_mod<-lmer(observed ~ Clade+(1|Tank.num), data=erich.tab2its.pacu.T0) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)     
  
  
  
``` 
alpha Observed Plots, no clade
```{r} 
#Means
observed_mean <- ddply(erich.tab2its, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                 N    = length(observed[!is.na(observed)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(observed, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(observed, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(observed, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
observed_mean

observed_mean<-subset(observed_mean, Time.Point=="T1"|Time.Point=="TF")
#Mcap 
Mcap_observed_mean<-subset(observed_mean, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_observed_mean<-subset(observed_mean, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_observed_mean<-subset(observed_mean, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_observed_mean<-subset(observed_mean, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_observed_plot<-ggplot(data=Mcap_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
 ylim(0, 40) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_observed_plot


#plot Pcomp
Pcomp_observed_plot<-ggplot(data=Pcomp_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
 ylim(0, 40) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_observed_plot

#Pvar plots
#plot Pvar
Pvar_observed_plot<-ggplot(data=Pvar_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
 ylim(0, 40) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_observed_plot


#Pacu plots
#plot Pacu
Pacu_observed_plot<-ggplot(data=Pacu_observed_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  ylim(0, 40) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_observed_plot

###### All plots together
Mcap_observed_plot<-Mcap_observed_plot+ theme(legend.position = "none") #remove legend
Pcomp_observed_plot<-Pcomp_observed_plot+ theme(legend.position = "none") #remove legend
Pvar_observed_plot<-Pvar_observed_plot+ theme(legend.position = "none") #remove legend
Pacu_observed_plot<-Pacu_observed_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_observed_plot, Pcomp_observed_plot, Pvar_observed_plot,Pacu_observed_plot, nrow = 1)
 
```  

alpha Observed Plots, clade  
```{r}   
# color pal from eta mcap.g.Parent.colors
alpha.colors <- c( "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#1F78B4","C High"="#6A3D9A", "CD Ambient"="#E31A1C", "CD High"="#FF7F00") 

#Means
observed_mean.C <- ddply(erich.tab2its, c("Treatment", "Species", "Time.Point","Clade"), summarise, #summarize cell counts
                 N    = length(observed[!is.na(observed)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(observed, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(observed, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(observed, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
observed_mean.C
observed_mean.C$Code<-paste(observed_mean.C$Clade, observed_mean.C$Treatment)

#Mcap 
Mcap_observed_mean.C<-subset(observed_mean.C, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_observed_mean.C<-subset(observed_mean.C, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_observed_mean.C<-subset(observed_mean.C, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_observed_mean.C<-subset(observed_mean.C, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap 
Mcap_observed_plot<-ggplot(data=Mcap_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_observed_plot

#plot for defense Mcap T1 and TF
Mcap_observed_mean.C_T1TF<-subset(Mcap_observed_mean.C, Time.Point=="T1"|Time.Point=="TF")
Mcap_observed_plot<-ggplot(data=Mcap_observed_mean.C_T1TF, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
 ylim(0, 45) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_observed_plot


#Pcomp plots
#plot Pcomp
Pcomp_observed_mean.C_T1TF<-subset(Pcomp_observed_mean.C, Time.Point=="T1"|Time.Point=="TF")

Pcomp_observed_plot<-ggplot(data=Pcomp_observed_mean.C_T1TF, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
  ylim(0, 45) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_observed_plot

#Pvar plots
#plot Pvar
Pvar_observed_mean.C_T1TF<-subset(Pvar_observed_mean.C, Time.Point=="T1"|Time.Point=="TF")

Pvar_observed_plot<-ggplot(data=Pvar_observed_mean.C_T1TF, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
 xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
   ylim(0, 45) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_observed_plot


#plot Pacu
Pacu_observed_plot<-ggplot(data=Pacu_observed_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0,45)+
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_observed_plot

#plot Pacu FOR DEFENSE T1TF
Pacu_observed_mean.C_T1TF<-subset(Pacu_observed_mean.C, Time.Point=="T1"|Time.Point=="TF")
Pacu_observed_plot<-ggplot(data=Pacu_observed_mean.C_T1TF, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 45) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_observed_plot

###### All plots together
Mcap_observed_plot<-Mcap_observed_plot+ theme(legend.position = "none") #remove legend
Pcomp_observed_plot<-Pcomp_observed_plot+ theme(legend.position = "none") #remove legend
Pvar_observed_plot<-Pvar_observed_plot+ theme(legend.position = "none") #remove legend
Pacu_observed_plot<-Pacu_observed_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_observed_plot, Pcomp_observed_plot, Pvar_observed_plot,Pacu_observed_plot, nrow = 1)
 
```
 
#evenness_pielou Richness 
```{r}     
#evenness_pielou  
#not transformed

evenness_pielou_alpha_mod<-lmer(evenness_pielou ~ Species*Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its) #adding tank is singularity
  anova(evenness_pielou_alpha_mod) 
  qqPlot(residuals(evenness_pielou_alpha_mod)) 
  emm<-emmeans(evenness_pielou_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  evenness_pielou_alpha_mod<-lmer(evenness_pielou ~ Species*Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2its) #adding tank is singularity
  anova(evenness_pielou_alpha_mod) 
  qqPlot(residuals(evenness_pielou_alpha_mod)) 
  emm<-emmeans(evenness_pielou_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

 
 #T0 only
  erich.tab2its.T0<-subset(erich.tab2its, Time.Point=="T0")
  evenness_pielou_alpha_mod<-lmer(evenness_pielou ~ Species+(1|Tank.num), data=erich.tab2its.T0) #adding tank is singularity
  anova(evenness_pielou_alpha_mod) 
  qqPlot(residuals(evenness_pielou_alpha_mod)) 
  emm<-emmeans(evenness_pielou_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
   
   #T0 only clade
  evenness_pielou_alpha_mod<-lmer(evenness_pielou ~ Species*Clade+(1|Tank.num), data=erich.tab2its.T0) #adding tank is singularity
  anova(evenness_pielou_alpha_mod) 
  qqPlot(residuals(evenness_pielou_alpha_mod)) 
  emm<-emmeans(evenness_pielou_alpha_mod,  ~ Clade*Species, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  
  
  
  

################Mcap only
 #t1TF
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID), data=erich.tab2its.mcap.t1tf) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID), data=erich.tab2its.mcap.t1tf) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Time.Point*Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2its.mcap.T1<-subset(erich.tab2its.mcap, Time.Point=="T1")

  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2its.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.mcap.T1) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Clade*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2its.mcap.TF<-subset(erich.tab2its.mcap, Time.Point=="TF")

   Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2its.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  Mcap_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.mcap.TF) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Treatment*Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)

  #t0
  erich.tab2its.mcap.T0<-subset(erich.tab2its.mcap, Time.Point=="T0")
hist(erich.tab2its.mcap.T0$observed)
   Mcap_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2its.mcap.T0) #adding tank is singularity
  anova(Mcap_alpha_mod) 
  qqPlot(residuals(Mcap_alpha_mod)) 
  emm<-emmeans(Mcap_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
####### PCOMP only
  #t1TF
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pcompT1TF) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pcompT1TF) 
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2its.pcomp.T1<-subset(erich.tab2its.pcomp, Time.Point=="T1")

  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2its.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pcomp.T1) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2its.pcomp.TF<-subset(erich.tab2its.pcomp, Time.Point=="TF")

   pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pcomp_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pcomp.TF) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
#t0
  erich.tab2its.pcomp.T0<-subset(erich.tab2its.pcomp, Time.Point=="T0")
hist(erich.tab2its.pcomp.T0$observed)
   pcomp_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2its.pcomp.T0) #adding tank is singularity
  anova(pcomp_alpha_mod) 
  qqPlot(residuals(pcomp_alpha_mod)) 
  emm<-emmeans(pcomp_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
####### pvar only ####
  #t1TF
  pvar_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pvarT1TF) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pvarT1TF) 
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2its.pvar.T1<-subset(erich.tab2its.pvar, Time.Point=="T1")

  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2its.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pvar.T1) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  #tf
  erich.tab2its.pvar.TF<-subset(erich.tab2its.pvar, Time.Point=="TF")

   pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pvar_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pvar.TF) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)

  #t0
  erich.tab2its.pvar.T0<-subset(erich.tab2its.pvar, Time.Point=="T0")
hist(erich.tab2its.pvar.T0$observed)
   pvar_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2its.pvar.T0) #adding tank is singularity
  anova(pvar_alpha_mod) 
  qqPlot(residuals(pvar_alpha_mod)) 
  emm<-emmeans(pvar_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)
    
  
  ####### pacu only####
  #t1TF
  pacu_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pacuT1TF) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
  pacu_alpha_mod<-lmer(evenness_pielou ~ Time.Point*Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pacuT1TF) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Time.Point*Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t1
  erich.tab2its.pacu.T1<-subset(erich.tab2its.pacu, Time.Point=="T1")

  pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID), data=erich.tab2its.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
   pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID), data=erich.tab2its.pacu.T1) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
    
  
  
  #tf
  erich.tab2its.pacu.TF<-subset(erich.tab2its.pacu, Time.Point=="TF")

   pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pacu.TF) 
  anova(pacu_alpha_mod) 
  pacu_alpha_mod<-lmer(evenness_pielou ~ Treatment*Clade+(1|Parent.ID)+(1|Tank.num), data=erich.tab2its.pacu.TF) 
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Treatment, adjust="Tukey")
  cld(emm)
  pairs(emm)
  
 #t0
  erich.tab2its.pacu.T0<-subset(erich.tab2its.pacu, Time.Point=="T0")
hist(erich.tab2its.pacu.T0$observed)
   pacu_alpha_mod<-lmer(evenness_pielou ~ Clade+(1|Tank.num), data=erich.tab2its.pacu.T0) #adding tank is singularity
  anova(pacu_alpha_mod) 
  qqPlot(residuals(pacu_alpha_mod)) 
  emm<-emmeans(pacu_alpha_mod,  ~ Clade, adjust="Tukey")
  cld(emm)
  pairs(emm)  
 
```
alpha evenness_pielou Plots, no clade
```{r} 
#Means 
evenness_pielou_mean <- ddply(erich.tab2its, c("Treatment", "Species", "Time.Point"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
evenness_pielou_mean
 
#Mcap 
Mcap_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_evenness_pielou_mean<-subset(evenness_pielou_mean, Species=="Pavona_varians") #Subset Pvar
 
#mcap plots
#plot Mcap
Mcap_evenness_pielou_plot<-ggplot(data=Mcap_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_evenness_pielou_plot



#by clade
erich.tab2its.mcapT1<-subset(erich.tab2its.mcap, Time.Point=="T1")
erich.tab2its.mcapT1mean <- ddply(erich.tab2its.mcapT1, c("Treatment", "Clade"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
erich.tab2its.mcapT1mean

Mcap_evenness_pielou_plot<-ggplot(data=erich.tab2its.mcapT1mean, aes(x=Time.Point, y=mean, group = Clade, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
 geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
   xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("Alpha Diversity"))) +
  ggtitle("Mcap Evenness by clade")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_evenness_pielou_plot

#by clade
erich.tab2its.mcapTF<-subset(erich.tab2its.mcap, Time.Point=="TF")
erich.tab2its.mcapTFmean <- ddply(erich.tab2its.mcapTF, c("Treatment", "Clade"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
erich.tab2its.mcapTFmean

Mcap_evenness_pielou_plot<-ggplot(data=erich.tab2its.mcapTFmean, aes(x=Clade, y=mean, group = Clade, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
 # geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  ylab(expression(paste("Alpha Diversity"))) +
  ggtitle("Mcap Evenness by clade TF")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_evenness_pielou_plot


#plot Pcomp
Pcomp_evenness_pielou_plot<-ggplot(data=Pcomp_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
  xlab("") + #Label the X Axis
  ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_evenness_pielou_plot

#Pvar plots
#plot Pvar
Pvar_evenness_pielou_plot<-ggplot(data=Pvar_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
xlab("") + #Label the X Axis
  ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_evenness_pielou_plot


#Pacu plots
#plot Pacu
Pacu_evenness_pielou_plot<-ggplot(data=Pacu_evenness_pielou_mean, aes(x=Time.Point, y=mean, group = Treatment, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment), size=4, show.legend = F)+
  geom_line(aes(color=Treatment), linetype=2, show.legend = F)+
  scale_color_manual(values=c("dodgerblue3", "darkred"))+
xlab("") + #Label the X Axis
  ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits
 # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Richness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_evenness_pielou_plot

###### All plots together
Mcap_evenness_pielou_plot<-Mcap_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pcomp_evenness_pielou_plot<-Pcomp_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pvar_evenness_pielou_plot<-Pvar_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pacu_evenness_pielou_plot<-Pacu_evenness_pielou_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_evenness_pielou_plot, Pcomp_evenness_pielou_plot, Pvar_evenness_pielou_plot,Pacu_evenness_pielou_plot, nrow = 1)
 
```  

alpha evenness_pielou Plots, clade  
```{r}  
# color pal from eta mcap.g.Parent.colors
alpha.colors <- c( "Ambient" = "dodgerblue", "High" = "darkred",
                        "C Ambient"="#1F78B4","C High"="#6A3D9A", "CD Ambient"="#E31A1C", "CD High"="#FF7F00") 

#Means
evenness_pielou_mean.C <- ddply(erich.tab2its, c("Treatment", "Species", "Time.Point","Clade"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NAs
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
)
evenness_pielou_mean.C
evenness_pielou_mean.C$Code<-paste(evenness_pielou_mean.C$Clade, evenness_pielou_mean.C$Treatment)

#Mcap 
Mcap_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Montipora_capitata") #Subset Mcap
# Pcomp 
Pcomp_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Porites_compressa") #Subset Pcomp
# Pacu 
Pacu_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Pocillopora_acuta") #Subset Pacu
# Pvar 
Pvar_evenness_pielou_mean.C<-subset(evenness_pielou_mean.C, Species=="Pavona_varians") #Subset Pvar

#mcap plots
#plot Mcap
Mcap_evenness_pielou_plot<-ggplot(data=Mcap_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Mcap Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Mcap_evenness_pielou_plot

#Pcomp plots
#plot Pcomp
Pcomp_evenness_pielou_plot<-ggplot(data=Pcomp_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits
  # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pcomp Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pcomp_evenness_pielou_plot

#Pvar plots
#plot Pvar
Pvar_evenness_pielou_plot<-ggplot(data=Pvar_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits
  # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pvar Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pvar_evenness_pielou_plot


#plot Pacu
Pacu_evenness_pielou_plot<-ggplot(data=Pacu_evenness_pielou_mean.C, aes(x=Time.Point, y=mean, group = Code, color=Treatment)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=Treatment, shape=Clade), size=4, show.legend = T)+
  geom_line(aes(color=Treatment, linetype=Clade), show.legend = T)+
  scale_color_manual(values=alpha.colors)+
xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  ylim(0, 1) + #set Y limits
  # ylim(0, 11) + #set Y limits
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
  # ylab(expression(paste("Alpha Diversity"))) +
  # ggtitle("Pacu Evenness")+
  theme(plot.title = element_text(size=20, face = "italic"));Pacu_evenness_pielou_plot

###### All plots together
Mcap_evenness_pielou_plot<-Mcap_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pcomp_evenness_pielou_plot<-Pcomp_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pvar_evenness_pielou_plot<-Pvar_evenness_pielou_plot+ theme(legend.position = "none") #remove legend
Pacu_evenness_pielou_plot<-Pacu_evenness_pielou_plot+ theme(legend.position = "none") #remove legend

grid.arrange(Mcap_evenness_pielou_plot, Pcomp_evenness_pielou_plot, Pvar_evenness_pielou_plot,Pacu_evenness_pielou_plot, nrow = 1)
 
```
 
